{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import json\n",
    "import numpy as np\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "from collections import Counter\n",
    "from scipy.stats import pearsonr\n",
    "import math\n",
    "import pydicom\n",
    "from show_keypoints import show_keypoints\n",
    "import matplotlib.cm as cm\n",
    "import scipy.stats as stats\n",
    "from PIL import Image, ImageDraw\n",
    "import os\n",
    "import random\n",
    "from bs4 import BeautifulSoup\n",
    "import requests\n",
    "from sklearn.model_selection import train_test_split"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Get software versions and device serial number"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "path = \"../UKB_xray_image_info/dcm/\"\n",
    "\n",
    "df = pd.DataFrame(columns=['file_name', 'software_version', 'device_serial_number', 'SERIAL_SOFTWARE'])\n",
    "\n",
    "for f in os.listdir(path):\n",
    "    if f.endswith(\".dcm\"):\n",
    "        dcm = pydicom.dcmread(os.path.join(path, f))\n",
    "        file_name = f.split(\".dcm\")[0]\n",
    "        software_version = dcm.get((0x0018, 0x1020), 'Unknown').value\n",
    "        device_serial_number = dcm.get((0x0018, 0x1000), 'Unknown').value\n",
    "        SERIAL_SOFTWARE = \"SS_\" + str(device_serial_number) + \"_\" + str(software_version)\n",
    "\n",
    "        # Append the data to the DataFrame\n",
    "        df = df.append({\n",
    "            'file_name': file_name,\n",
    "            'software_version': software_version,\n",
    "            'device_serial_number': device_serial_number,\n",
    "            'SERIAL_SOFTWARE': SERIAL_SOFTWARE\n",
    "        }, ignore_index=True)\n",
    "\n",
    "# merge the serial_software with the EID\n",
    "hip_pheno = pd.read_csv('key_results/hip_select_pheno_cm.csv')[['file_name', 'eid']]\n",
    "dcm_info = hip_pheno.merge(df, on='file_name', how='inner')\n",
    "dcm_info.to_csv(\"../UKB_xray_image_info/dicom_info.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>file_name</th>\n",
       "      <th>eid</th>\n",
       "      <th>software_version</th>\n",
       "      <th>device_serial_number</th>\n",
       "      <th>SERIAL_SOFTWARE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1.2.840.113619.2.110.212038.20190713145954.1.1...</td>\n",
       "      <td>1000308</td>\n",
       "      <td>16.20.059</td>\n",
       "      <td>212038</td>\n",
       "      <td>SS_212038_16.20.059</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1.2.840.113619.2.110.210419.20180924120844.1.1...</td>\n",
       "      <td>1000320</td>\n",
       "      <td>16.20.059</td>\n",
       "      <td>210419</td>\n",
       "      <td>SS_210419_16.20.059</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1.2.840.113619.2.110.212038.20190307144208.1.1...</td>\n",
       "      <td>1000951</td>\n",
       "      <td>16.20.059</td>\n",
       "      <td>212038</td>\n",
       "      <td>SS_212038_16.20.059</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1.2.840.113619.2.110.212038.20171116132027.1.1...</td>\n",
       "      <td>1001329</td>\n",
       "      <td>16.20.059</td>\n",
       "      <td>212038</td>\n",
       "      <td>SS_212038_16.20.059</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1.2.840.113619.2.110.210419.20180720150454.1.1...</td>\n",
       "      <td>1001512</td>\n",
       "      <td>16.20.059</td>\n",
       "      <td>210419</td>\n",
       "      <td>SS_210419_16.20.059</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39342</th>\n",
       "      <td>1.2.840.113619.2.110.210419.20170408102812.1.1...</td>\n",
       "      <td>6021811</td>\n",
       "      <td>16.00.158</td>\n",
       "      <td>210419</td>\n",
       "      <td>SS_210419_16.00.158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39343</th>\n",
       "      <td>1.2.840.113619.2.110.210419.20170702173904.1.1...</td>\n",
       "      <td>6022591</td>\n",
       "      <td>16.00.158</td>\n",
       "      <td>210419</td>\n",
       "      <td>SS_210419_16.00.158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39344</th>\n",
       "      <td>1.2.840.113619.2.110.210419.20150918144642.2.1...</td>\n",
       "      <td>6022737</td>\n",
       "      <td>15.20.002</td>\n",
       "      <td>210419</td>\n",
       "      <td>SS_210419_15.20.002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39345</th>\n",
       "      <td>1.2.840.113619.2.110.210419.20171202183632.1.1...</td>\n",
       "      <td>6023431</td>\n",
       "      <td>16.00.158</td>\n",
       "      <td>210419</td>\n",
       "      <td>SS_210419_16.00.158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39346</th>\n",
       "      <td>1.2.840.113619.2.110.210419.20170418184226.1.1...</td>\n",
       "      <td>6024029</td>\n",
       "      <td>16.00.158</td>\n",
       "      <td>210419</td>\n",
       "      <td>SS_210419_16.00.158</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>39347 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                               file_name      eid  \\\n",
       "0      1.2.840.113619.2.110.212038.20190713145954.1.1...  1000308   \n",
       "1      1.2.840.113619.2.110.210419.20180924120844.1.1...  1000320   \n",
       "2      1.2.840.113619.2.110.212038.20190307144208.1.1...  1000951   \n",
       "3      1.2.840.113619.2.110.212038.20171116132027.1.1...  1001329   \n",
       "4      1.2.840.113619.2.110.210419.20180720150454.1.1...  1001512   \n",
       "...                                                  ...      ...   \n",
       "39342  1.2.840.113619.2.110.210419.20170408102812.1.1...  6021811   \n",
       "39343  1.2.840.113619.2.110.210419.20170702173904.1.1...  6022591   \n",
       "39344  1.2.840.113619.2.110.210419.20150918144642.2.1...  6022737   \n",
       "39345  1.2.840.113619.2.110.210419.20171202183632.1.1...  6023431   \n",
       "39346  1.2.840.113619.2.110.210419.20170418184226.1.1...  6024029   \n",
       "\n",
       "      software_version device_serial_number      SERIAL_SOFTWARE  \n",
       "0            16.20.059               212038  SS_212038_16.20.059  \n",
       "1            16.20.059               210419  SS_210419_16.20.059  \n",
       "2            16.20.059               212038  SS_212038_16.20.059  \n",
       "3            16.20.059               212038  SS_212038_16.20.059  \n",
       "4            16.20.059               210419  SS_210419_16.20.059  \n",
       "...                ...                  ...                  ...  \n",
       "39342        16.00.158               210419  SS_210419_16.00.158  \n",
       "39343        16.00.158               210419  SS_210419_16.00.158  \n",
       "39344        15.20.002               210419  SS_210419_15.20.002  \n",
       "39345        16.00.158               210419  SS_210419_16.00.158  \n",
       "39346        16.00.158               210419  SS_210419_16.00.158  \n",
       "\n",
       "[39347 rows x 5 columns]"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dcm_info"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Make covar file for imaging individuals white british"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "# get all fid and iid for white british\n",
    "column_names = ['FID', 'IID', 'PID', 'MID', 'SEX', 'PHENOTYPE']\n",
    "fam_df = pd.read_csv('./UKB_Imaging_Genetics/merged_sub_maf0.001_biallel_bbf_400k_white_british.fam', sep='\\s+', header=None, names=column_names)\n",
    "\n",
    "# rename PC columns and remove PC21-40\n",
    "PC_df = pd.read_csv('./UKB_Imaging_Genetics/fid22009.csv', sep = ',')\n",
    "PC_df.columns = [\"eid\"] + [f\"PC{i}\" for i in range(1, 41)]\n",
    "\n",
    "columns_to_remove = [f\"PC{i}\" for i in range(21, 41)]  \n",
    "PC_df = PC_df.drop(columns=columns_to_remove, errors='ignore')\n",
    "\n",
    "covar_df = fam_df.merge(PC_df, left_on='IID', right_on='eid', how='inner')\n",
    "\n",
    "# get age, sex, height information\n",
    "fid_info = pd.read_csv(\"../UKB_xray_image_info/fids/fid_disease/fid_info_400k_20240104.csv\", sep = ',')[['eid', 'age_imaging_visit', 'sex', 'standing_height_imaging_visit']]; fid_info\n",
    "covar_df = covar_df.merge(fid_info, left_on='eid', right_on='eid', how='inner')\n",
    "\n",
    "# remove redundant columns\n",
    "covar_df.drop(['PID', 'MID', 'SEX', 'PHENOTYPE', 'eid'], axis=1, inplace=True)\n",
    "\n",
    "# rename columns\n",
    "covar_df.rename({'age_imaging_visit': 'AGE', 'sex': \"SEX\", 'standing_height_imaging_visit': 'HEIGHT'}, axis=1, inplace=True)\n",
    "covar_df['HEIGHT2'] = covar_df['HEIGHT']**2\n",
    "\n",
    "# add some covariates\n",
    "covar_df['AGE2'] = covar_df['AGE']**2\n",
    "covar_df['SEXAGE'] = covar_df['SEX'] * covar_df['AGE']\n",
    "covar_df['SEXAGE2'] = covar_df['SEX'] * covar_df['AGE']**2\n",
    "\n",
    "# add serial_software\n",
    "dcm_info = pd.read_csv(\"../UKB_xray_image_info/dicom_info.csv\")[['file_name', 'eid', 'SERIAL_SOFTWARE']]\n",
    "dcm_info.sort_values(by = ['eid', 'file_name'], ascending=[True, True], inplace=True)\n",
    "dcm_info.drop_duplicates(subset=['eid'], keep='first', inplace=True)\n",
    "covar_df = covar_df.merge(dcm_info, left_on='IID', right_on='eid', how='inner')\n",
    "\n",
    "# remove redundant columns\n",
    "covar_df.drop(['eid'], axis=1, inplace=True)\n",
    "\n",
    "# remove NA\n",
    "covar_df.dropna(inplace=True)\n",
    "\n",
    "# save\n",
    "covar_df.to_csv('./UKB_Imaging_Genetics/GWAS_INPUT_DATA/covar_40k_imaging_20240104.txt', sep = \" \", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['FID', 'IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'PC6', 'PC7', 'PC8',\n",
       "       'PC9', 'PC10', 'PC11', 'PC12', 'PC13', 'PC14', 'PC15', 'PC16', 'PC17',\n",
       "       'PC18', 'PC19', 'PC20', 'AGE', 'SEX', 'HEIGHT', 'HEIGHT2', 'AGE2',\n",
       "       'SEXAGE', 'SEXAGE2', 'file_name', 'SERIAL_SOFTWARE'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "covar_df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "covar_df.drop(columns=['HEIGHT2', 'AGE2', 'SEXAGE', 'SEXAGE2', 'file_name'], inplace=True)\n",
    "covar_df.to_csv('./UKB_Imaging_Genetics/GWAS_INPUT_DATA/plink_covar_40k_imaging_20240104.txt', sep = \" \", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['FID', 'IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'PC6', 'PC7', 'PC8',\n",
       "       'PC9', 'PC10', 'PC11', 'PC12', 'PC13', 'PC14', 'PC15', 'PC16', 'PC17',\n",
       "       'PC18', 'PC19', 'PC20', 'AGE', 'SEX', 'HEIGHT', 'SERIAL_SOFTWARE'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "covar_df.columns"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Make covar file for 400k individuals white british"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# get all fid and iid for white british\n",
    "column_names = ['FID', 'IID', 'PID', 'MID', 'SEX', 'PHENOTYPE']\n",
    "fam_df = pd.read_csv('./UKB_Imaging_Genetics/merged_sub_maf0.001_biallel_bbf_400k_white_british.fam', sep='\\s+', header=None, names=column_names)\n",
    "\n",
    "# rename PC columns and remove PC21-40\n",
    "PC_df = pd.read_csv('./UKB_Imaging_Genetics/fid22009.csv', sep = ',')\n",
    "PC_df.columns = [\"eid\"] + [f\"PC{i}\" for i in range(1, 41)]\n",
    "\n",
    "columns_to_remove = [f\"PC{i}\" for i in range(21, 41)]  \n",
    "PC_df = PC_df.drop(columns=columns_to_remove, errors='ignore')\n",
    "\n",
    "covar_df = fam_df.merge(PC_df, left_on='IID', right_on='eid', how='inner')\n",
    "\n",
    "# get age, sex, height information\n",
    "fid_info = pd.read_csv(\"../UKB_xray_image_info/fids/fid_disease/fid_info_400k_20240104.csv\", sep = ',')[['eid', 'age', 'sex']]; fid_info\n",
    "covar_df = covar_df.merge(fid_info, left_on='eid', right_on='eid', how='inner')\n",
    "\n",
    "# remove redundant columns\n",
    "covar_df.drop(['PID', 'MID', 'SEX', 'PHENOTYPE', 'eid'], axis=1, inplace=True)\n",
    "\n",
    "# rename columns\n",
    "covar_df.rename({'age': 'AGE', 'sex': \"SEX\"}, axis=1, inplace=True)\n",
    "\n",
    "# add some covariates\n",
    "covar_df['AGE2'] = covar_df['AGE']**2\n",
    "covar_df['SEXAGE'] = covar_df['SEX'] * covar_df['AGE']\n",
    "covar_df['SEXAGE2'] = covar_df['SEX'] * covar_df['AGE']**2\n",
    "\n",
    "# remove NA\n",
    "covar_df.dropna(inplace=True)\n",
    "\n",
    "# save\n",
    "covar_df.to_csv('./UKB_Imaging_Genetics/GWAS_INPUT_DATA/covar_400k_20240320.txt', sep = \" \", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PC1</th>\n",
       "      <th>PC2</th>\n",
       "      <th>PC3</th>\n",
       "      <th>PC4</th>\n",
       "      <th>PC5</th>\n",
       "      <th>PC6</th>\n",
       "      <th>PC7</th>\n",
       "      <th>PC8</th>\n",
       "      <th>...</th>\n",
       "      <th>PC16</th>\n",
       "      <th>PC17</th>\n",
       "      <th>PC18</th>\n",
       "      <th>PC19</th>\n",
       "      <th>PC20</th>\n",
       "      <th>AGE</th>\n",
       "      <th>SEX</th>\n",
       "      <th>AGE2</th>\n",
       "      <th>SEXAGE</th>\n",
       "      <th>SEXAGE2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1000011</td>\n",
       "      <td>1000011</td>\n",
       "      <td>-13.84930</td>\n",
       "      <td>4.66820</td>\n",
       "      <td>0.629980</td>\n",
       "      <td>7.883860</td>\n",
       "      <td>14.791100</td>\n",
       "      <td>-0.527804</td>\n",
       "      <td>2.797600</td>\n",
       "      <td>-0.876043</td>\n",
       "      <td>...</td>\n",
       "      <td>0.442455</td>\n",
       "      <td>-1.466750</td>\n",
       "      <td>-8.013810</td>\n",
       "      <td>4.055920</td>\n",
       "      <td>-5.199500</td>\n",
       "      <td>50.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2500.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1000026</td>\n",
       "      <td>1000026</td>\n",
       "      <td>-11.33050</td>\n",
       "      <td>6.43170</td>\n",
       "      <td>-0.301415</td>\n",
       "      <td>-2.537360</td>\n",
       "      <td>-9.305720</td>\n",
       "      <td>4.765020</td>\n",
       "      <td>-1.391040</td>\n",
       "      <td>0.194816</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.731087</td>\n",
       "      <td>0.659766</td>\n",
       "      <td>0.386768</td>\n",
       "      <td>1.957890</td>\n",
       "      <td>1.015310</td>\n",
       "      <td>59.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3481.0</td>\n",
       "      <td>59.0</td>\n",
       "      <td>3481.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1000032</td>\n",
       "      <td>1000032</td>\n",
       "      <td>-13.52700</td>\n",
       "      <td>4.60185</td>\n",
       "      <td>-2.528160</td>\n",
       "      <td>-1.424620</td>\n",
       "      <td>0.960400</td>\n",
       "      <td>-2.086600</td>\n",
       "      <td>-0.866667</td>\n",
       "      <td>0.392374</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.225871</td>\n",
       "      <td>-3.112810</td>\n",
       "      <td>-2.109150</td>\n",
       "      <td>-4.088490</td>\n",
       "      <td>1.411410</td>\n",
       "      <td>50.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2500.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1000060</td>\n",
       "      <td>1000060</td>\n",
       "      <td>-10.07460</td>\n",
       "      <td>5.66969</td>\n",
       "      <td>-1.121770</td>\n",
       "      <td>3.096290</td>\n",
       "      <td>-5.462910</td>\n",
       "      <td>-1.409950</td>\n",
       "      <td>0.417993</td>\n",
       "      <td>-2.350320</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.198975</td>\n",
       "      <td>0.612011</td>\n",
       "      <td>-1.860850</td>\n",
       "      <td>0.580956</td>\n",
       "      <td>-0.143335</td>\n",
       "      <td>54.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2916.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1000083</td>\n",
       "      <td>1000083</td>\n",
       "      <td>-13.64920</td>\n",
       "      <td>6.30138</td>\n",
       "      <td>-1.464490</td>\n",
       "      <td>0.597857</td>\n",
       "      <td>3.287830</td>\n",
       "      <td>1.383300</td>\n",
       "      <td>1.249120</td>\n",
       "      <td>-0.224952</td>\n",
       "      <td>...</td>\n",
       "      <td>7.953920</td>\n",
       "      <td>1.053860</td>\n",
       "      <td>1.705720</td>\n",
       "      <td>-3.111490</td>\n",
       "      <td>-0.222030</td>\n",
       "      <td>56.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3136.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>402227</th>\n",
       "      <td>6024877</td>\n",
       "      <td>6024877</td>\n",
       "      <td>-11.53160</td>\n",
       "      <td>4.61181</td>\n",
       "      <td>-2.277910</td>\n",
       "      <td>5.479920</td>\n",
       "      <td>12.915600</td>\n",
       "      <td>-0.358599</td>\n",
       "      <td>-1.015170</td>\n",
       "      <td>-1.687950</td>\n",
       "      <td>...</td>\n",
       "      <td>-4.178370</td>\n",
       "      <td>4.812280</td>\n",
       "      <td>-2.353500</td>\n",
       "      <td>-0.966063</td>\n",
       "      <td>-2.422980</td>\n",
       "      <td>62.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3844.0</td>\n",
       "      <td>62.0</td>\n",
       "      <td>3844.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>402228</th>\n",
       "      <td>6024884</td>\n",
       "      <td>6024884</td>\n",
       "      <td>-9.67937</td>\n",
       "      <td>2.28478</td>\n",
       "      <td>-1.682700</td>\n",
       "      <td>3.770500</td>\n",
       "      <td>2.406000</td>\n",
       "      <td>-0.040925</td>\n",
       "      <td>-2.444220</td>\n",
       "      <td>-1.601790</td>\n",
       "      <td>...</td>\n",
       "      <td>1.165680</td>\n",
       "      <td>0.002156</td>\n",
       "      <td>-2.970630</td>\n",
       "      <td>-2.303800</td>\n",
       "      <td>-1.155280</td>\n",
       "      <td>48.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2304.0</td>\n",
       "      <td>48.0</td>\n",
       "      <td>2304.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>402229</th>\n",
       "      <td>6024895</td>\n",
       "      <td>6024895</td>\n",
       "      <td>-11.65080</td>\n",
       "      <td>3.19475</td>\n",
       "      <td>-2.315910</td>\n",
       "      <td>3.687250</td>\n",
       "      <td>-0.551556</td>\n",
       "      <td>-1.632460</td>\n",
       "      <td>-0.020148</td>\n",
       "      <td>-2.682440</td>\n",
       "      <td>...</td>\n",
       "      <td>0.279108</td>\n",
       "      <td>-0.040651</td>\n",
       "      <td>3.622240</td>\n",
       "      <td>2.741980</td>\n",
       "      <td>-4.252590</td>\n",
       "      <td>55.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3025.0</td>\n",
       "      <td>55.0</td>\n",
       "      <td>3025.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>402230</th>\n",
       "      <td>6024913</td>\n",
       "      <td>6024913</td>\n",
       "      <td>-14.66110</td>\n",
       "      <td>2.91677</td>\n",
       "      <td>-1.624580</td>\n",
       "      <td>5.574640</td>\n",
       "      <td>7.682750</td>\n",
       "      <td>-0.079685</td>\n",
       "      <td>1.285760</td>\n",
       "      <td>1.790390</td>\n",
       "      <td>...</td>\n",
       "      <td>-13.309300</td>\n",
       "      <td>-0.918372</td>\n",
       "      <td>4.496170</td>\n",
       "      <td>-0.237348</td>\n",
       "      <td>-2.360360</td>\n",
       "      <td>64.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4096.0</td>\n",
       "      <td>64.0</td>\n",
       "      <td>4096.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>402231</th>\n",
       "      <td>6024922</td>\n",
       "      <td>6024922</td>\n",
       "      <td>-14.11170</td>\n",
       "      <td>4.46003</td>\n",
       "      <td>-1.826900</td>\n",
       "      <td>4.504550</td>\n",
       "      <td>0.311829</td>\n",
       "      <td>-2.406860</td>\n",
       "      <td>-0.065846</td>\n",
       "      <td>-2.944000</td>\n",
       "      <td>...</td>\n",
       "      <td>-8.077880</td>\n",
       "      <td>-1.255310</td>\n",
       "      <td>0.989865</td>\n",
       "      <td>-2.437380</td>\n",
       "      <td>-5.268610</td>\n",
       "      <td>48.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2304.0</td>\n",
       "      <td>48.0</td>\n",
       "      <td>2304.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>402232 rows × 27 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            FID      IID       PC1      PC2       PC3       PC4        PC5  \\\n",
       "0       1000011  1000011 -13.84930  4.66820  0.629980  7.883860  14.791100   \n",
       "1       1000026  1000026 -11.33050  6.43170 -0.301415 -2.537360  -9.305720   \n",
       "2       1000032  1000032 -13.52700  4.60185 -2.528160 -1.424620   0.960400   \n",
       "3       1000060  1000060 -10.07460  5.66969 -1.121770  3.096290  -5.462910   \n",
       "4       1000083  1000083 -13.64920  6.30138 -1.464490  0.597857   3.287830   \n",
       "...         ...      ...       ...      ...       ...       ...        ...   \n",
       "402227  6024877  6024877 -11.53160  4.61181 -2.277910  5.479920  12.915600   \n",
       "402228  6024884  6024884  -9.67937  2.28478 -1.682700  3.770500   2.406000   \n",
       "402229  6024895  6024895 -11.65080  3.19475 -2.315910  3.687250  -0.551556   \n",
       "402230  6024913  6024913 -14.66110  2.91677 -1.624580  5.574640   7.682750   \n",
       "402231  6024922  6024922 -14.11170  4.46003 -1.826900  4.504550   0.311829   \n",
       "\n",
       "             PC6       PC7       PC8  ...       PC16      PC17      PC18  \\\n",
       "0      -0.527804  2.797600 -0.876043  ...   0.442455 -1.466750 -8.013810   \n",
       "1       4.765020 -1.391040  0.194816  ...  -0.731087  0.659766  0.386768   \n",
       "2      -2.086600 -0.866667  0.392374  ...  -0.225871 -3.112810 -2.109150   \n",
       "3      -1.409950  0.417993 -2.350320  ...  -0.198975  0.612011 -1.860850   \n",
       "4       1.383300  1.249120 -0.224952  ...   7.953920  1.053860  1.705720   \n",
       "...          ...       ...       ...  ...        ...       ...       ...   \n",
       "402227 -0.358599 -1.015170 -1.687950  ...  -4.178370  4.812280 -2.353500   \n",
       "402228 -0.040925 -2.444220 -1.601790  ...   1.165680  0.002156 -2.970630   \n",
       "402229 -1.632460 -0.020148 -2.682440  ...   0.279108 -0.040651  3.622240   \n",
       "402230 -0.079685  1.285760  1.790390  ... -13.309300 -0.918372  4.496170   \n",
       "402231 -2.406860 -0.065846 -2.944000  ...  -8.077880 -1.255310  0.989865   \n",
       "\n",
       "            PC19      PC20   AGE  SEX    AGE2  SEXAGE  SEXAGE2  \n",
       "0       4.055920 -5.199500  50.0  0.0  2500.0     0.0      0.0  \n",
       "1       1.957890  1.015310  59.0  1.0  3481.0    59.0   3481.0  \n",
       "2      -4.088490  1.411410  50.0  0.0  2500.0     0.0      0.0  \n",
       "3       0.580956 -0.143335  54.0  0.0  2916.0     0.0      0.0  \n",
       "4      -3.111490 -0.222030  56.0  0.0  3136.0     0.0      0.0  \n",
       "...          ...       ...   ...  ...     ...     ...      ...  \n",
       "402227 -0.966063 -2.422980  62.0  1.0  3844.0    62.0   3844.0  \n",
       "402228 -2.303800 -1.155280  48.0  1.0  2304.0    48.0   2304.0  \n",
       "402229  2.741980 -4.252590  55.0  1.0  3025.0    55.0   3025.0  \n",
       "402230 -0.237348 -2.360360  64.0  1.0  4096.0    64.0   4096.0  \n",
       "402231 -2.437380 -5.268610  48.0  1.0  2304.0    48.0   2304.0  \n",
       "\n",
       "[402232 rows x 27 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "covar_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "covar_df = pd.read_csv(\"./UKB_Imaging_Genetics/GWAS_INPUT_DATA/covar_400k_20240320.txt\", sep = \" \")\n",
    "plink_covar_df = covar_df.drop(columns=['AGE2', 'SEXAGE', 'SEXAGE2'])\n",
    "plink_covar_df.to_csv('./UKB_Imaging_Genetics/GWAS_INPUT_DATA/plink_covar_400k_20240320.txt', sep = \" \", index=False)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Data cleaning"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>image_id</th>\n",
       "      <th>file_name</th>\n",
       "      <th>eid</th>\n",
       "      <th>sex</th>\n",
       "      <th>standing_height_imaging_visit</th>\n",
       "      <th>weight_imaging_visit</th>\n",
       "      <th>bmi_imaging_visit</th>\n",
       "      <th>age_imaging_visit</th>\n",
       "      <th>pelvic_height</th>\n",
       "      <th>pelvic_width</th>\n",
       "      <th>...</th>\n",
       "      <th>acetabular_diameter</th>\n",
       "      <th>acetabular_inclination</th>\n",
       "      <th>humerus</th>\n",
       "      <th>femur</th>\n",
       "      <th>forearm</th>\n",
       "      <th>tibia</th>\n",
       "      <th>arm_length</th>\n",
       "      <th>leg_length</th>\n",
       "      <th>leg_divide_torso</th>\n",
       "      <th>pelvic_inlet_area</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>15998_82</td>\n",
       "      <td>1.2.840.113619.2.110.212038.20190713145954.1.1...</td>\n",
       "      <td>1000308</td>\n",
       "      <td>1.0</td>\n",
       "      <td>182.0</td>\n",
       "      <td>91.5</td>\n",
       "      <td>27.6235</td>\n",
       "      <td>67.0</td>\n",
       "      <td>25.278134</td>\n",
       "      <td>31.968543</td>\n",
       "      <td>...</td>\n",
       "      <td>6.811823</td>\n",
       "      <td>56.275191</td>\n",
       "      <td>30.641372</td>\n",
       "      <td>49.354825</td>\n",
       "      <td>29.255057</td>\n",
       "      <td>44.262392</td>\n",
       "      <td>59.896430</td>\n",
       "      <td>93.617217</td>\n",
       "      <td>1.446595</td>\n",
       "      <td>130.830780</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>02879_82</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20180924120844.1.1...</td>\n",
       "      <td>1000320</td>\n",
       "      <td>0.0</td>\n",
       "      <td>167.0</td>\n",
       "      <td>80.0</td>\n",
       "      <td>28.6851</td>\n",
       "      <td>64.0</td>\n",
       "      <td>23.340543</td>\n",
       "      <td>29.261568</td>\n",
       "      <td>...</td>\n",
       "      <td>5.206508</td>\n",
       "      <td>67.136555</td>\n",
       "      <td>28.182432</td>\n",
       "      <td>45.652312</td>\n",
       "      <td>25.412075</td>\n",
       "      <td>39.512738</td>\n",
       "      <td>53.594507</td>\n",
       "      <td>85.165049</td>\n",
       "      <td>1.486175</td>\n",
       "      <td>127.566679</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14366_82</td>\n",
       "      <td>1.2.840.113619.2.110.212038.20190307144208.1.1...</td>\n",
       "      <td>1000951</td>\n",
       "      <td>1.0</td>\n",
       "      <td>166.0</td>\n",
       "      <td>60.8</td>\n",
       "      <td>22.0642</td>\n",
       "      <td>78.0</td>\n",
       "      <td>22.297293</td>\n",
       "      <td>30.113663</td>\n",
       "      <td>...</td>\n",
       "      <td>6.141783</td>\n",
       "      <td>60.747735</td>\n",
       "      <td>30.816804</td>\n",
       "      <td>48.489427</td>\n",
       "      <td>26.856522</td>\n",
       "      <td>40.743734</td>\n",
       "      <td>57.673326</td>\n",
       "      <td>89.233162</td>\n",
       "      <td>1.651141</td>\n",
       "      <td>108.190429</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>01877_82</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20180720150454.1.1...</td>\n",
       "      <td>1001512</td>\n",
       "      <td>1.0</td>\n",
       "      <td>169.0</td>\n",
       "      <td>77.3</td>\n",
       "      <td>27.0649</td>\n",
       "      <td>66.0</td>\n",
       "      <td>23.742297</td>\n",
       "      <td>30.901775</td>\n",
       "      <td>...</td>\n",
       "      <td>6.035855</td>\n",
       "      <td>59.189839</td>\n",
       "      <td>28.115456</td>\n",
       "      <td>44.984030</td>\n",
       "      <td>26.628187</td>\n",
       "      <td>40.081134</td>\n",
       "      <td>54.743642</td>\n",
       "      <td>85.065164</td>\n",
       "      <td>1.449931</td>\n",
       "      <td>102.660623</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>05431_82</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20190328125146.1.1...</td>\n",
       "      <td>1001867</td>\n",
       "      <td>0.0</td>\n",
       "      <td>153.0</td>\n",
       "      <td>56.1</td>\n",
       "      <td>23.9651</td>\n",
       "      <td>75.0</td>\n",
       "      <td>22.226944</td>\n",
       "      <td>30.934457</td>\n",
       "      <td>...</td>\n",
       "      <td>5.531657</td>\n",
       "      <td>59.017487</td>\n",
       "      <td>26.778209</td>\n",
       "      <td>42.934407</td>\n",
       "      <td>23.988531</td>\n",
       "      <td>35.818183</td>\n",
       "      <td>50.766740</td>\n",
       "      <td>78.752591</td>\n",
       "      <td>1.537721</td>\n",
       "      <td>139.989191</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38196</th>\n",
       "      <td>12857_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20170408102812.1.1...</td>\n",
       "      <td>6021811</td>\n",
       "      <td>0.0</td>\n",
       "      <td>170.0</td>\n",
       "      <td>57.4</td>\n",
       "      <td>19.8616</td>\n",
       "      <td>69.0</td>\n",
       "      <td>22.195469</td>\n",
       "      <td>30.523987</td>\n",
       "      <td>...</td>\n",
       "      <td>5.132306</td>\n",
       "      <td>58.792442</td>\n",
       "      <td>29.838081</td>\n",
       "      <td>47.704132</td>\n",
       "      <td>27.818466</td>\n",
       "      <td>41.904510</td>\n",
       "      <td>57.656547</td>\n",
       "      <td>89.608642</td>\n",
       "      <td>1.530112</td>\n",
       "      <td>112.668303</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38197</th>\n",
       "      <td>13966_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20170702173904.1.1...</td>\n",
       "      <td>6022591</td>\n",
       "      <td>0.0</td>\n",
       "      <td>156.0</td>\n",
       "      <td>87.7</td>\n",
       "      <td>36.0371</td>\n",
       "      <td>69.0</td>\n",
       "      <td>20.946633</td>\n",
       "      <td>26.922261</td>\n",
       "      <td>...</td>\n",
       "      <td>4.959191</td>\n",
       "      <td>67.937764</td>\n",
       "      <td>28.030085</td>\n",
       "      <td>42.718048</td>\n",
       "      <td>24.880466</td>\n",
       "      <td>37.908781</td>\n",
       "      <td>52.910551</td>\n",
       "      <td>80.626829</td>\n",
       "      <td>1.461921</td>\n",
       "      <td>126.477899</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38198</th>\n",
       "      <td>00922_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20150918144642.2.1...</td>\n",
       "      <td>6022737</td>\n",
       "      <td>1.0</td>\n",
       "      <td>186.0</td>\n",
       "      <td>72.8</td>\n",
       "      <td>21.0429</td>\n",
       "      <td>63.0</td>\n",
       "      <td>25.396006</td>\n",
       "      <td>32.074461</td>\n",
       "      <td>...</td>\n",
       "      <td>6.812095</td>\n",
       "      <td>61.621522</td>\n",
       "      <td>34.381037</td>\n",
       "      <td>52.893919</td>\n",
       "      <td>31.659763</td>\n",
       "      <td>45.982489</td>\n",
       "      <td>66.040801</td>\n",
       "      <td>98.876408</td>\n",
       "      <td>1.635700</td>\n",
       "      <td>102.976979</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38199</th>\n",
       "      <td>16309_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20171202183632.1.1...</td>\n",
       "      <td>6023431</td>\n",
       "      <td>0.0</td>\n",
       "      <td>170.0</td>\n",
       "      <td>61.5</td>\n",
       "      <td>21.2803</td>\n",
       "      <td>51.0</td>\n",
       "      <td>22.787944</td>\n",
       "      <td>29.871832</td>\n",
       "      <td>...</td>\n",
       "      <td>5.201943</td>\n",
       "      <td>60.779334</td>\n",
       "      <td>29.276081</td>\n",
       "      <td>45.262378</td>\n",
       "      <td>25.847555</td>\n",
       "      <td>38.992764</td>\n",
       "      <td>55.123637</td>\n",
       "      <td>84.255142</td>\n",
       "      <td>1.384857</td>\n",
       "      <td>150.788401</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38200</th>\n",
       "      <td>12964_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20170418184226.1.1...</td>\n",
       "      <td>6024029</td>\n",
       "      <td>1.0</td>\n",
       "      <td>179.0</td>\n",
       "      <td>80.4</td>\n",
       "      <td>25.0928</td>\n",
       "      <td>50.0</td>\n",
       "      <td>24.323043</td>\n",
       "      <td>29.804727</td>\n",
       "      <td>...</td>\n",
       "      <td>6.171569</td>\n",
       "      <td>59.544963</td>\n",
       "      <td>32.678000</td>\n",
       "      <td>50.968299</td>\n",
       "      <td>29.406987</td>\n",
       "      <td>42.910290</td>\n",
       "      <td>62.084987</td>\n",
       "      <td>93.878589</td>\n",
       "      <td>1.475647</td>\n",
       "      <td>138.796708</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>38201 rows × 32 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       image_id                                          file_name      eid  \\\n",
       "0      15998_82  1.2.840.113619.2.110.212038.20190713145954.1.1...  1000308   \n",
       "1      02879_82  1.2.840.113619.2.110.210419.20180924120844.1.1...  1000320   \n",
       "2      14366_82  1.2.840.113619.2.110.212038.20190307144208.1.1...  1000951   \n",
       "3      01877_82  1.2.840.113619.2.110.210419.20180720150454.1.1...  1001512   \n",
       "4      05431_82  1.2.840.113619.2.110.210419.20190328125146.1.1...  1001867   \n",
       "...         ...                                                ...      ...   \n",
       "38196  12857_93  1.2.840.113619.2.110.210419.20170408102812.1.1...  6021811   \n",
       "38197  13966_93  1.2.840.113619.2.110.210419.20170702173904.1.1...  6022591   \n",
       "38198  00922_93  1.2.840.113619.2.110.210419.20150918144642.2.1...  6022737   \n",
       "38199  16309_93  1.2.840.113619.2.110.210419.20171202183632.1.1...  6023431   \n",
       "38200  12964_93  1.2.840.113619.2.110.210419.20170418184226.1.1...  6024029   \n",
       "\n",
       "       sex  standing_height_imaging_visit  weight_imaging_visit  \\\n",
       "0      1.0                          182.0                  91.5   \n",
       "1      0.0                          167.0                  80.0   \n",
       "2      1.0                          166.0                  60.8   \n",
       "3      1.0                          169.0                  77.3   \n",
       "4      0.0                          153.0                  56.1   \n",
       "...    ...                            ...                   ...   \n",
       "38196  0.0                          170.0                  57.4   \n",
       "38197  0.0                          156.0                  87.7   \n",
       "38198  1.0                          186.0                  72.8   \n",
       "38199  0.0                          170.0                  61.5   \n",
       "38200  1.0                          179.0                  80.4   \n",
       "\n",
       "       bmi_imaging_visit  age_imaging_visit  pelvic_height  pelvic_width  ...  \\\n",
       "0                27.6235               67.0      25.278134     31.968543  ...   \n",
       "1                28.6851               64.0      23.340543     29.261568  ...   \n",
       "2                22.0642               78.0      22.297293     30.113663  ...   \n",
       "3                27.0649               66.0      23.742297     30.901775  ...   \n",
       "4                23.9651               75.0      22.226944     30.934457  ...   \n",
       "...                  ...                ...            ...           ...  ...   \n",
       "38196            19.8616               69.0      22.195469     30.523987  ...   \n",
       "38197            36.0371               69.0      20.946633     26.922261  ...   \n",
       "38198            21.0429               63.0      25.396006     32.074461  ...   \n",
       "38199            21.2803               51.0      22.787944     29.871832  ...   \n",
       "38200            25.0928               50.0      24.323043     29.804727  ...   \n",
       "\n",
       "       acetabular_diameter  acetabular_inclination    humerus      femur  \\\n",
       "0                 6.811823               56.275191  30.641372  49.354825   \n",
       "1                 5.206508               67.136555  28.182432  45.652312   \n",
       "2                 6.141783               60.747735  30.816804  48.489427   \n",
       "3                 6.035855               59.189839  28.115456  44.984030   \n",
       "4                 5.531657               59.017487  26.778209  42.934407   \n",
       "...                    ...                     ...        ...        ...   \n",
       "38196             5.132306               58.792442  29.838081  47.704132   \n",
       "38197             4.959191               67.937764  28.030085  42.718048   \n",
       "38198             6.812095               61.621522  34.381037  52.893919   \n",
       "38199             5.201943               60.779334  29.276081  45.262378   \n",
       "38200             6.171569               59.544963  32.678000  50.968299   \n",
       "\n",
       "         forearm      tibia  arm_length  leg_length  leg_divide_torso  \\\n",
       "0      29.255057  44.262392   59.896430   93.617217          1.446595   \n",
       "1      25.412075  39.512738   53.594507   85.165049          1.486175   \n",
       "2      26.856522  40.743734   57.673326   89.233162          1.651141   \n",
       "3      26.628187  40.081134   54.743642   85.065164          1.449931   \n",
       "4      23.988531  35.818183   50.766740   78.752591          1.537721   \n",
       "...          ...        ...         ...         ...               ...   \n",
       "38196  27.818466  41.904510   57.656547   89.608642          1.530112   \n",
       "38197  24.880466  37.908781   52.910551   80.626829          1.461921   \n",
       "38198  31.659763  45.982489   66.040801   98.876408          1.635700   \n",
       "38199  25.847555  38.992764   55.123637   84.255142          1.384857   \n",
       "38200  29.406987  42.910290   62.084987   93.878589          1.475647   \n",
       "\n",
       "       pelvic_inlet_area  \n",
       "0             130.830780  \n",
       "1             127.566679  \n",
       "2             108.190429  \n",
       "3             102.660623  \n",
       "4             139.989191  \n",
       "...                  ...  \n",
       "38196         112.668303  \n",
       "38197         126.477899  \n",
       "38198         102.976979  \n",
       "38199         150.788401  \n",
       "38200         138.796708  \n",
       "\n",
       "[38201 rows x 32 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hip_pheno = pd.read_csv('key_results/hip_select_pheno_cm.csv'); hip_pheno"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PC1</th>\n",
       "      <th>PC2</th>\n",
       "      <th>PC3</th>\n",
       "      <th>PC4</th>\n",
       "      <th>PC5</th>\n",
       "      <th>PC6</th>\n",
       "      <th>PC7</th>\n",
       "      <th>PC8</th>\n",
       "      <th>...</th>\n",
       "      <th>PC20</th>\n",
       "      <th>AGE</th>\n",
       "      <th>SEX</th>\n",
       "      <th>HEIGHT</th>\n",
       "      <th>HEIGHT2</th>\n",
       "      <th>AGE2</th>\n",
       "      <th>SEXAGE</th>\n",
       "      <th>SEXAGE2</th>\n",
       "      <th>file_name</th>\n",
       "      <th>SERIAL_SOFTWARE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1000276</td>\n",
       "      <td>1000276</td>\n",
       "      <td>-12.4105</td>\n",
       "      <td>4.64514</td>\n",
       "      <td>-3.622420</td>\n",
       "      <td>5.980920</td>\n",
       "      <td>-7.061210</td>\n",
       "      <td>-2.844940</td>\n",
       "      <td>1.447260</td>\n",
       "      <td>-4.110470</td>\n",
       "      <td>...</td>\n",
       "      <td>11.190400</td>\n",
       "      <td>67.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>160.0</td>\n",
       "      <td>25600.00</td>\n",
       "      <td>4489.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20160630134650.1.1...</td>\n",
       "      <td>SS_210419_16.00.158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1000742</td>\n",
       "      <td>1000742</td>\n",
       "      <td>-13.4871</td>\n",
       "      <td>3.94973</td>\n",
       "      <td>-2.277800</td>\n",
       "      <td>5.638660</td>\n",
       "      <td>21.510600</td>\n",
       "      <td>-1.485880</td>\n",
       "      <td>-0.150935</td>\n",
       "      <td>3.719550</td>\n",
       "      <td>...</td>\n",
       "      <td>-2.106240</td>\n",
       "      <td>67.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>160.0</td>\n",
       "      <td>25600.00</td>\n",
       "      <td>4489.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20170305174208.1.1...</td>\n",
       "      <td>SS_210419_16.00.158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1000951</td>\n",
       "      <td>1000951</td>\n",
       "      <td>-16.5284</td>\n",
       "      <td>3.53244</td>\n",
       "      <td>-0.421914</td>\n",
       "      <td>5.432070</td>\n",
       "      <td>7.310350</td>\n",
       "      <td>-1.167890</td>\n",
       "      <td>-0.547475</td>\n",
       "      <td>2.062850</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.357651</td>\n",
       "      <td>78.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>166.0</td>\n",
       "      <td>27556.00</td>\n",
       "      <td>6084.0</td>\n",
       "      <td>78.0</td>\n",
       "      <td>6084.0</td>\n",
       "      <td>1.2.840.113619.2.110.212038.20190307144208.1.1...</td>\n",
       "      <td>SS_212038_16.20.059</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1001025</td>\n",
       "      <td>1001025</td>\n",
       "      <td>-13.8205</td>\n",
       "      <td>3.38118</td>\n",
       "      <td>0.495384</td>\n",
       "      <td>-1.111690</td>\n",
       "      <td>-5.541310</td>\n",
       "      <td>-1.830790</td>\n",
       "      <td>0.781520</td>\n",
       "      <td>-2.161940</td>\n",
       "      <td>...</td>\n",
       "      <td>-2.059780</td>\n",
       "      <td>56.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>167.0</td>\n",
       "      <td>27889.00</td>\n",
       "      <td>3136.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20151211144320.1.1...</td>\n",
       "      <td>SS_210419_16.00.158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1001091</td>\n",
       "      <td>1001091</td>\n",
       "      <td>-13.0812</td>\n",
       "      <td>3.87649</td>\n",
       "      <td>-3.759090</td>\n",
       "      <td>0.539327</td>\n",
       "      <td>-11.600100</td>\n",
       "      <td>-0.399333</td>\n",
       "      <td>-1.265270</td>\n",
       "      <td>-1.461750</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.683705</td>\n",
       "      <td>60.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>161.5</td>\n",
       "      <td>26082.25</td>\n",
       "      <td>3600.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20170902122319.1.1...</td>\n",
       "      <td>SS_210419_16.00.158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31207</th>\n",
       "      <td>6023985</td>\n",
       "      <td>6023985</td>\n",
       "      <td>-13.1687</td>\n",
       "      <td>3.20078</td>\n",
       "      <td>-1.536440</td>\n",
       "      <td>1.981020</td>\n",
       "      <td>-4.450010</td>\n",
       "      <td>-2.889470</td>\n",
       "      <td>0.840123</td>\n",
       "      <td>0.333759</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.024228</td>\n",
       "      <td>66.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>164.6</td>\n",
       "      <td>27093.16</td>\n",
       "      <td>4356.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20190117162724.1.1...</td>\n",
       "      <td>SS_210419_16.20.059</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31208</th>\n",
       "      <td>6024029</td>\n",
       "      <td>6024029</td>\n",
       "      <td>-11.3007</td>\n",
       "      <td>1.94620</td>\n",
       "      <td>-2.961530</td>\n",
       "      <td>3.255970</td>\n",
       "      <td>-0.093339</td>\n",
       "      <td>-0.585923</td>\n",
       "      <td>0.048063</td>\n",
       "      <td>-1.632140</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.785720</td>\n",
       "      <td>50.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>179.0</td>\n",
       "      <td>32041.00</td>\n",
       "      <td>2500.0</td>\n",
       "      <td>50.0</td>\n",
       "      <td>2500.0</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20170418184226.1.1...</td>\n",
       "      <td>SS_210419_16.00.158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31209</th>\n",
       "      <td>6024037</td>\n",
       "      <td>6024037</td>\n",
       "      <td>-12.5736</td>\n",
       "      <td>1.53991</td>\n",
       "      <td>-2.019910</td>\n",
       "      <td>2.278480</td>\n",
       "      <td>-9.960630</td>\n",
       "      <td>1.128140</td>\n",
       "      <td>-0.393647</td>\n",
       "      <td>-3.298780</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.063200</td>\n",
       "      <td>61.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>181.0</td>\n",
       "      <td>32761.00</td>\n",
       "      <td>3721.0</td>\n",
       "      <td>61.0</td>\n",
       "      <td>3721.0</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20180811105258.1.1...</td>\n",
       "      <td>SS_210419_16.20.059</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31210</th>\n",
       "      <td>6024257</td>\n",
       "      <td>6024257</td>\n",
       "      <td>-14.4691</td>\n",
       "      <td>1.81586</td>\n",
       "      <td>0.069078</td>\n",
       "      <td>2.070290</td>\n",
       "      <td>2.293480</td>\n",
       "      <td>0.174149</td>\n",
       "      <td>-0.233781</td>\n",
       "      <td>-0.208208</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.087590</td>\n",
       "      <td>71.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>167.0</td>\n",
       "      <td>27889.00</td>\n",
       "      <td>5041.0</td>\n",
       "      <td>71.0</td>\n",
       "      <td>5041.0</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20190219154149.1.1...</td>\n",
       "      <td>SS_210419_16.20.059</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31211</th>\n",
       "      <td>6024532</td>\n",
       "      <td>6024532</td>\n",
       "      <td>-14.0294</td>\n",
       "      <td>4.36160</td>\n",
       "      <td>-3.006160</td>\n",
       "      <td>0.157262</td>\n",
       "      <td>-10.510700</td>\n",
       "      <td>-2.422590</td>\n",
       "      <td>-0.055734</td>\n",
       "      <td>-3.922910</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.386740</td>\n",
       "      <td>67.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>173.0</td>\n",
       "      <td>29929.00</td>\n",
       "      <td>4489.0</td>\n",
       "      <td>67.0</td>\n",
       "      <td>4489.0</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20180912183226.1.1...</td>\n",
       "      <td>SS_210419_16.20.059</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>31212 rows × 31 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           FID      IID      PC1      PC2       PC3       PC4        PC5  \\\n",
       "0      1000276  1000276 -12.4105  4.64514 -3.622420  5.980920  -7.061210   \n",
       "1      1000742  1000742 -13.4871  3.94973 -2.277800  5.638660  21.510600   \n",
       "2      1000951  1000951 -16.5284  3.53244 -0.421914  5.432070   7.310350   \n",
       "3      1001025  1001025 -13.8205  3.38118  0.495384 -1.111690  -5.541310   \n",
       "4      1001091  1001091 -13.0812  3.87649 -3.759090  0.539327 -11.600100   \n",
       "...        ...      ...      ...      ...       ...       ...        ...   \n",
       "31207  6023985  6023985 -13.1687  3.20078 -1.536440  1.981020  -4.450010   \n",
       "31208  6024029  6024029 -11.3007  1.94620 -2.961530  3.255970  -0.093339   \n",
       "31209  6024037  6024037 -12.5736  1.53991 -2.019910  2.278480  -9.960630   \n",
       "31210  6024257  6024257 -14.4691  1.81586  0.069078  2.070290   2.293480   \n",
       "31211  6024532  6024532 -14.0294  4.36160 -3.006160  0.157262 -10.510700   \n",
       "\n",
       "            PC6       PC7       PC8  ...       PC20   AGE  SEX  HEIGHT  \\\n",
       "0     -2.844940  1.447260 -4.110470  ...  11.190400  67.0  0.0   160.0   \n",
       "1     -1.485880 -0.150935  3.719550  ...  -2.106240  67.0  0.0   160.0   \n",
       "2     -1.167890 -0.547475  2.062850  ...  -0.357651  78.0  1.0   166.0   \n",
       "3     -1.830790  0.781520 -2.161940  ...  -2.059780  56.0  0.0   167.0   \n",
       "4     -0.399333 -1.265270 -1.461750  ...  -0.683705  60.0  0.0   161.5   \n",
       "...         ...       ...       ...  ...        ...   ...  ...     ...   \n",
       "31207 -2.889470  0.840123  0.333759  ...  -0.024228  66.0  0.0   164.6   \n",
       "31208 -0.585923  0.048063 -1.632140  ...  -0.785720  50.0  1.0   179.0   \n",
       "31209  1.128140 -0.393647 -3.298780  ...  -0.063200  61.0  1.0   181.0   \n",
       "31210  0.174149 -0.233781 -0.208208  ...  -1.087590  71.0  1.0   167.0   \n",
       "31211 -2.422590 -0.055734 -3.922910  ...  -1.386740  67.0  1.0   173.0   \n",
       "\n",
       "        HEIGHT2    AGE2  SEXAGE  SEXAGE2  \\\n",
       "0      25600.00  4489.0     0.0      0.0   \n",
       "1      25600.00  4489.0     0.0      0.0   \n",
       "2      27556.00  6084.0    78.0   6084.0   \n",
       "3      27889.00  3136.0     0.0      0.0   \n",
       "4      26082.25  3600.0     0.0      0.0   \n",
       "...         ...     ...     ...      ...   \n",
       "31207  27093.16  4356.0     0.0      0.0   \n",
       "31208  32041.00  2500.0    50.0   2500.0   \n",
       "31209  32761.00  3721.0    61.0   3721.0   \n",
       "31210  27889.00  5041.0    71.0   5041.0   \n",
       "31211  29929.00  4489.0    67.0   4489.0   \n",
       "\n",
       "                                               file_name      SERIAL_SOFTWARE  \n",
       "0      1.2.840.113619.2.110.210419.20160630134650.1.1...  SS_210419_16.00.158  \n",
       "1      1.2.840.113619.2.110.210419.20170305174208.1.1...  SS_210419_16.00.158  \n",
       "2      1.2.840.113619.2.110.212038.20190307144208.1.1...  SS_212038_16.20.059  \n",
       "3      1.2.840.113619.2.110.210419.20151211144320.1.1...  SS_210419_16.00.158  \n",
       "4      1.2.840.113619.2.110.210419.20170902122319.1.1...  SS_210419_16.00.158  \n",
       "...                                                  ...                  ...  \n",
       "31207  1.2.840.113619.2.110.210419.20190117162724.1.1...  SS_210419_16.20.059  \n",
       "31208  1.2.840.113619.2.110.210419.20170418184226.1.1...  SS_210419_16.00.158  \n",
       "31209  1.2.840.113619.2.110.210419.20180811105258.1.1...  SS_210419_16.20.059  \n",
       "31210  1.2.840.113619.2.110.210419.20190219154149.1.1...  SS_210419_16.20.059  \n",
       "31211  1.2.840.113619.2.110.210419.20180912183226.1.1...  SS_210419_16.20.059  \n",
       "\n",
       "[31212 rows x 31 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "covar_df = pd.read_csv('./UKB_Imaging_Genetics/GWAS_INPUT_DATA/covar_40k_imaging_20240104.txt', sep = \" \"); covar_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "# keep white british\n",
    "hip_pheno = hip_pheno[hip_pheno['file_name'].isin(covar_df['file_name'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "# filter ids that patients who don't want to share any more\n",
    "remove_ids = pd.read_csv(\"../UKB_xray_image_info/remove_eids_20230504.csv\", header=None)[0].tolist()\n",
    "hip_pheno = hip_pheno[~hip_pheno['eid'].isin(remove_ids)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>image_id</th>\n",
       "      <th>file_name</th>\n",
       "      <th>eid</th>\n",
       "      <th>sex</th>\n",
       "      <th>standing_height_imaging_visit</th>\n",
       "      <th>weight_imaging_visit</th>\n",
       "      <th>bmi_imaging_visit</th>\n",
       "      <th>age_imaging_visit</th>\n",
       "      <th>pelvic_height</th>\n",
       "      <th>pelvic_width</th>\n",
       "      <th>...</th>\n",
       "      <th>acetabular_diameter</th>\n",
       "      <th>acetabular_inclination</th>\n",
       "      <th>humerus</th>\n",
       "      <th>femur</th>\n",
       "      <th>forearm</th>\n",
       "      <th>tibia</th>\n",
       "      <th>arm_length</th>\n",
       "      <th>leg_length</th>\n",
       "      <th>leg_divide_torso</th>\n",
       "      <th>pelvic_inlet_area</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14366_82</td>\n",
       "      <td>1.2.840.113619.2.110.212038.20190307144208.1.1...</td>\n",
       "      <td>1000951</td>\n",
       "      <td>1.0</td>\n",
       "      <td>166.0</td>\n",
       "      <td>60.8</td>\n",
       "      <td>22.0642</td>\n",
       "      <td>78.0</td>\n",
       "      <td>22.297293</td>\n",
       "      <td>30.113663</td>\n",
       "      <td>...</td>\n",
       "      <td>6.141783</td>\n",
       "      <td>60.747735</td>\n",
       "      <td>30.816804</td>\n",
       "      <td>48.489427</td>\n",
       "      <td>26.856522</td>\n",
       "      <td>40.743734</td>\n",
       "      <td>57.673326</td>\n",
       "      <td>89.233162</td>\n",
       "      <td>1.651141</td>\n",
       "      <td>108.190429</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>01877_82</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20180720150454.1.1...</td>\n",
       "      <td>1001512</td>\n",
       "      <td>1.0</td>\n",
       "      <td>169.0</td>\n",
       "      <td>77.3</td>\n",
       "      <td>27.0649</td>\n",
       "      <td>66.0</td>\n",
       "      <td>23.742297</td>\n",
       "      <td>30.901775</td>\n",
       "      <td>...</td>\n",
       "      <td>6.035855</td>\n",
       "      <td>59.189839</td>\n",
       "      <td>28.115456</td>\n",
       "      <td>44.984030</td>\n",
       "      <td>26.628187</td>\n",
       "      <td>40.081134</td>\n",
       "      <td>54.743642</td>\n",
       "      <td>85.065164</td>\n",
       "      <td>1.449931</td>\n",
       "      <td>102.660623</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>05431_82</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20190328125146.1.1...</td>\n",
       "      <td>1001867</td>\n",
       "      <td>0.0</td>\n",
       "      <td>153.0</td>\n",
       "      <td>56.1</td>\n",
       "      <td>23.9651</td>\n",
       "      <td>75.0</td>\n",
       "      <td>22.226944</td>\n",
       "      <td>30.934457</td>\n",
       "      <td>...</td>\n",
       "      <td>5.531657</td>\n",
       "      <td>59.017487</td>\n",
       "      <td>26.778209</td>\n",
       "      <td>42.934407</td>\n",
       "      <td>23.988531</td>\n",
       "      <td>35.818183</td>\n",
       "      <td>50.766740</td>\n",
       "      <td>78.752591</td>\n",
       "      <td>1.537721</td>\n",
       "      <td>139.989191</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>10557_82</td>\n",
       "      <td>1.2.840.113619.2.110.212038.20180228105159.1.1...</td>\n",
       "      <td>1001921</td>\n",
       "      <td>1.0</td>\n",
       "      <td>166.0</td>\n",
       "      <td>64.2</td>\n",
       "      <td>23.2980</td>\n",
       "      <td>62.0</td>\n",
       "      <td>22.861258</td>\n",
       "      <td>28.863430</td>\n",
       "      <td>...</td>\n",
       "      <td>5.834113</td>\n",
       "      <td>58.027399</td>\n",
       "      <td>29.972729</td>\n",
       "      <td>46.587679</td>\n",
       "      <td>28.561098</td>\n",
       "      <td>39.566371</td>\n",
       "      <td>58.533827</td>\n",
       "      <td>86.154050</td>\n",
       "      <td>1.493503</td>\n",
       "      <td>110.520822</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>01561_82</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20180630141714.1.1...</td>\n",
       "      <td>1001950</td>\n",
       "      <td>1.0</td>\n",
       "      <td>165.0</td>\n",
       "      <td>89.6</td>\n",
       "      <td>32.9109</td>\n",
       "      <td>62.0</td>\n",
       "      <td>23.080688</td>\n",
       "      <td>29.385649</td>\n",
       "      <td>...</td>\n",
       "      <td>5.731393</td>\n",
       "      <td>57.335593</td>\n",
       "      <td>29.012823</td>\n",
       "      <td>45.245573</td>\n",
       "      <td>27.292476</td>\n",
       "      <td>39.041382</td>\n",
       "      <td>56.305299</td>\n",
       "      <td>84.286956</td>\n",
       "      <td>1.465162</td>\n",
       "      <td>109.946859</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38194</th>\n",
       "      <td>10039_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20160923114938.1.1...</td>\n",
       "      <td>6021112</td>\n",
       "      <td>1.0</td>\n",
       "      <td>177.0</td>\n",
       "      <td>89.0</td>\n",
       "      <td>28.4082</td>\n",
       "      <td>71.0</td>\n",
       "      <td>25.247090</td>\n",
       "      <td>32.848104</td>\n",
       "      <td>...</td>\n",
       "      <td>6.149698</td>\n",
       "      <td>60.427962</td>\n",
       "      <td>31.540230</td>\n",
       "      <td>50.329159</td>\n",
       "      <td>30.128478</td>\n",
       "      <td>44.856165</td>\n",
       "      <td>61.668708</td>\n",
       "      <td>95.185324</td>\n",
       "      <td>1.583562</td>\n",
       "      <td>134.603261</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38195</th>\n",
       "      <td>14018_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20170705180448.1.1...</td>\n",
       "      <td>6021539</td>\n",
       "      <td>1.0</td>\n",
       "      <td>179.0</td>\n",
       "      <td>71.0</td>\n",
       "      <td>22.1591</td>\n",
       "      <td>69.0</td>\n",
       "      <td>25.232678</td>\n",
       "      <td>30.970184</td>\n",
       "      <td>...</td>\n",
       "      <td>6.504511</td>\n",
       "      <td>59.444884</td>\n",
       "      <td>30.777846</td>\n",
       "      <td>48.408803</td>\n",
       "      <td>28.966865</td>\n",
       "      <td>42.070750</td>\n",
       "      <td>59.744710</td>\n",
       "      <td>90.479553</td>\n",
       "      <td>1.479675</td>\n",
       "      <td>132.776803</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38197</th>\n",
       "      <td>13966_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20170702173904.1.1...</td>\n",
       "      <td>6022591</td>\n",
       "      <td>0.0</td>\n",
       "      <td>156.0</td>\n",
       "      <td>87.7</td>\n",
       "      <td>36.0371</td>\n",
       "      <td>69.0</td>\n",
       "      <td>20.946633</td>\n",
       "      <td>26.922261</td>\n",
       "      <td>...</td>\n",
       "      <td>4.959191</td>\n",
       "      <td>67.937764</td>\n",
       "      <td>28.030085</td>\n",
       "      <td>42.718048</td>\n",
       "      <td>24.880466</td>\n",
       "      <td>37.908781</td>\n",
       "      <td>52.910551</td>\n",
       "      <td>80.626829</td>\n",
       "      <td>1.461921</td>\n",
       "      <td>126.477899</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38199</th>\n",
       "      <td>16309_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20171202183632.1.1...</td>\n",
       "      <td>6023431</td>\n",
       "      <td>0.0</td>\n",
       "      <td>170.0</td>\n",
       "      <td>61.5</td>\n",
       "      <td>21.2803</td>\n",
       "      <td>51.0</td>\n",
       "      <td>22.787944</td>\n",
       "      <td>29.871832</td>\n",
       "      <td>...</td>\n",
       "      <td>5.201943</td>\n",
       "      <td>60.779334</td>\n",
       "      <td>29.276081</td>\n",
       "      <td>45.262378</td>\n",
       "      <td>25.847555</td>\n",
       "      <td>38.992764</td>\n",
       "      <td>55.123637</td>\n",
       "      <td>84.255142</td>\n",
       "      <td>1.384857</td>\n",
       "      <td>150.788401</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38200</th>\n",
       "      <td>12964_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20170418184226.1.1...</td>\n",
       "      <td>6024029</td>\n",
       "      <td>1.0</td>\n",
       "      <td>179.0</td>\n",
       "      <td>80.4</td>\n",
       "      <td>25.0928</td>\n",
       "      <td>50.0</td>\n",
       "      <td>24.323043</td>\n",
       "      <td>29.804727</td>\n",
       "      <td>...</td>\n",
       "      <td>6.171569</td>\n",
       "      <td>59.544963</td>\n",
       "      <td>32.678000</td>\n",
       "      <td>50.968299</td>\n",
       "      <td>29.406987</td>\n",
       "      <td>42.910290</td>\n",
       "      <td>62.084987</td>\n",
       "      <td>93.878589</td>\n",
       "      <td>1.475647</td>\n",
       "      <td>138.796708</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>31134 rows × 32 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       image_id                                          file_name      eid  \\\n",
       "2      14366_82  1.2.840.113619.2.110.212038.20190307144208.1.1...  1000951   \n",
       "3      01877_82  1.2.840.113619.2.110.210419.20180720150454.1.1...  1001512   \n",
       "4      05431_82  1.2.840.113619.2.110.210419.20190328125146.1.1...  1001867   \n",
       "5      10557_82  1.2.840.113619.2.110.212038.20180228105159.1.1...  1001921   \n",
       "6      01561_82  1.2.840.113619.2.110.210419.20180630141714.1.1...  1001950   \n",
       "...         ...                                                ...      ...   \n",
       "38194  10039_93  1.2.840.113619.2.110.210419.20160923114938.1.1...  6021112   \n",
       "38195  14018_93  1.2.840.113619.2.110.210419.20170705180448.1.1...  6021539   \n",
       "38197  13966_93  1.2.840.113619.2.110.210419.20170702173904.1.1...  6022591   \n",
       "38199  16309_93  1.2.840.113619.2.110.210419.20171202183632.1.1...  6023431   \n",
       "38200  12964_93  1.2.840.113619.2.110.210419.20170418184226.1.1...  6024029   \n",
       "\n",
       "       sex  standing_height_imaging_visit  weight_imaging_visit  \\\n",
       "2      1.0                          166.0                  60.8   \n",
       "3      1.0                          169.0                  77.3   \n",
       "4      0.0                          153.0                  56.1   \n",
       "5      1.0                          166.0                  64.2   \n",
       "6      1.0                          165.0                  89.6   \n",
       "...    ...                            ...                   ...   \n",
       "38194  1.0                          177.0                  89.0   \n",
       "38195  1.0                          179.0                  71.0   \n",
       "38197  0.0                          156.0                  87.7   \n",
       "38199  0.0                          170.0                  61.5   \n",
       "38200  1.0                          179.0                  80.4   \n",
       "\n",
       "       bmi_imaging_visit  age_imaging_visit  pelvic_height  pelvic_width  ...  \\\n",
       "2                22.0642               78.0      22.297293     30.113663  ...   \n",
       "3                27.0649               66.0      23.742297     30.901775  ...   \n",
       "4                23.9651               75.0      22.226944     30.934457  ...   \n",
       "5                23.2980               62.0      22.861258     28.863430  ...   \n",
       "6                32.9109               62.0      23.080688     29.385649  ...   \n",
       "...                  ...                ...            ...           ...  ...   \n",
       "38194            28.4082               71.0      25.247090     32.848104  ...   \n",
       "38195            22.1591               69.0      25.232678     30.970184  ...   \n",
       "38197            36.0371               69.0      20.946633     26.922261  ...   \n",
       "38199            21.2803               51.0      22.787944     29.871832  ...   \n",
       "38200            25.0928               50.0      24.323043     29.804727  ...   \n",
       "\n",
       "       acetabular_diameter  acetabular_inclination    humerus      femur  \\\n",
       "2                 6.141783               60.747735  30.816804  48.489427   \n",
       "3                 6.035855               59.189839  28.115456  44.984030   \n",
       "4                 5.531657               59.017487  26.778209  42.934407   \n",
       "5                 5.834113               58.027399  29.972729  46.587679   \n",
       "6                 5.731393               57.335593  29.012823  45.245573   \n",
       "...                    ...                     ...        ...        ...   \n",
       "38194             6.149698               60.427962  31.540230  50.329159   \n",
       "38195             6.504511               59.444884  30.777846  48.408803   \n",
       "38197             4.959191               67.937764  28.030085  42.718048   \n",
       "38199             5.201943               60.779334  29.276081  45.262378   \n",
       "38200             6.171569               59.544963  32.678000  50.968299   \n",
       "\n",
       "         forearm      tibia  arm_length  leg_length  leg_divide_torso  \\\n",
       "2      26.856522  40.743734   57.673326   89.233162          1.651141   \n",
       "3      26.628187  40.081134   54.743642   85.065164          1.449931   \n",
       "4      23.988531  35.818183   50.766740   78.752591          1.537721   \n",
       "5      28.561098  39.566371   58.533827   86.154050          1.493503   \n",
       "6      27.292476  39.041382   56.305299   84.286956          1.465162   \n",
       "...          ...        ...         ...         ...               ...   \n",
       "38194  30.128478  44.856165   61.668708   95.185324          1.583562   \n",
       "38195  28.966865  42.070750   59.744710   90.479553          1.479675   \n",
       "38197  24.880466  37.908781   52.910551   80.626829          1.461921   \n",
       "38199  25.847555  38.992764   55.123637   84.255142          1.384857   \n",
       "38200  29.406987  42.910290   62.084987   93.878589          1.475647   \n",
       "\n",
       "       pelvic_inlet_area  \n",
       "2             108.190429  \n",
       "3             102.660623  \n",
       "4             139.989191  \n",
       "5             110.520822  \n",
       "6             109.946859  \n",
       "...                  ...  \n",
       "38194         134.603261  \n",
       "38195         132.776803  \n",
       "38197         126.477899  \n",
       "38199         150.788401  \n",
       "38200         138.796708  \n",
       "\n",
       "[31134 rows x 32 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hip_pheno"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "hip_pheno.to_csv('key_results/hip_select_pheno_cm_eid_flt.csv', index=False)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Z-score filter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "hip_pheno = pd.read_csv('key_results/hip_select_pheno_cm_eid_flt.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>image_id</th>\n",
       "      <th>file_name</th>\n",
       "      <th>eid</th>\n",
       "      <th>sex</th>\n",
       "      <th>standing_height_imaging_visit</th>\n",
       "      <th>weight_imaging_visit</th>\n",
       "      <th>bmi_imaging_visit</th>\n",
       "      <th>age_imaging_visit</th>\n",
       "      <th>pelvic_height</th>\n",
       "      <th>pelvic_width</th>\n",
       "      <th>...</th>\n",
       "      <th>arm_devide_torso</th>\n",
       "      <th>bi_acetabular_width</th>\n",
       "      <th>torso_length</th>\n",
       "      <th>iliac_isthmus_breadth</th>\n",
       "      <th>acetabular_diameter</th>\n",
       "      <th>acetabular_inclination</th>\n",
       "      <th>arm_length</th>\n",
       "      <th>leg_length</th>\n",
       "      <th>leg_divide_torso</th>\n",
       "      <th>pelvic_inlet_area</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14366_82</td>\n",
       "      <td>1.2.840.113619.2.110.212038.20190307144208.1.1...</td>\n",
       "      <td>1000951</td>\n",
       "      <td>1.0</td>\n",
       "      <td>166.0</td>\n",
       "      <td>60.8</td>\n",
       "      <td>22.0642</td>\n",
       "      <td>78.0</td>\n",
       "      <td>22.297293</td>\n",
       "      <td>30.113663</td>\n",
       "      <td>...</td>\n",
       "      <td>1.067168</td>\n",
       "      <td>20.324819</td>\n",
       "      <td>54.043332</td>\n",
       "      <td>4.164529</td>\n",
       "      <td>6.141783</td>\n",
       "      <td>60.747735</td>\n",
       "      <td>57.673326</td>\n",
       "      <td>89.233162</td>\n",
       "      <td>1.651141</td>\n",
       "      <td>108.190429</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>01877_82</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20180720150454.1.1...</td>\n",
       "      <td>1001512</td>\n",
       "      <td>1.0</td>\n",
       "      <td>169.0</td>\n",
       "      <td>77.3</td>\n",
       "      <td>27.0649</td>\n",
       "      <td>66.0</td>\n",
       "      <td>23.742297</td>\n",
       "      <td>30.901775</td>\n",
       "      <td>...</td>\n",
       "      <td>0.933102</td>\n",
       "      <td>20.756184</td>\n",
       "      <td>58.668433</td>\n",
       "      <td>4.657824</td>\n",
       "      <td>6.035855</td>\n",
       "      <td>59.189839</td>\n",
       "      <td>54.743642</td>\n",
       "      <td>85.065164</td>\n",
       "      <td>1.449931</td>\n",
       "      <td>102.660623</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>05431_82</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20190328125146.1.1...</td>\n",
       "      <td>1001867</td>\n",
       "      <td>0.0</td>\n",
       "      <td>153.0</td>\n",
       "      <td>56.1</td>\n",
       "      <td>23.9651</td>\n",
       "      <td>75.0</td>\n",
       "      <td>22.226944</td>\n",
       "      <td>30.934457</td>\n",
       "      <td>...</td>\n",
       "      <td>0.991270</td>\n",
       "      <td>21.701987</td>\n",
       "      <td>51.213839</td>\n",
       "      <td>3.718747</td>\n",
       "      <td>5.531657</td>\n",
       "      <td>59.017487</td>\n",
       "      <td>50.766740</td>\n",
       "      <td>78.752591</td>\n",
       "      <td>1.537721</td>\n",
       "      <td>139.989191</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10557_82</td>\n",
       "      <td>1.2.840.113619.2.110.212038.20180228105159.1.1...</td>\n",
       "      <td>1001921</td>\n",
       "      <td>1.0</td>\n",
       "      <td>166.0</td>\n",
       "      <td>64.2</td>\n",
       "      <td>23.2980</td>\n",
       "      <td>62.0</td>\n",
       "      <td>22.861258</td>\n",
       "      <td>28.863430</td>\n",
       "      <td>...</td>\n",
       "      <td>1.014700</td>\n",
       "      <td>19.784228</td>\n",
       "      <td>57.685872</td>\n",
       "      <td>4.238295</td>\n",
       "      <td>5.834113</td>\n",
       "      <td>58.027399</td>\n",
       "      <td>58.533827</td>\n",
       "      <td>86.154050</td>\n",
       "      <td>1.493503</td>\n",
       "      <td>110.520822</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>01561_82</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20180630141714.1.1...</td>\n",
       "      <td>1001950</td>\n",
       "      <td>1.0</td>\n",
       "      <td>165.0</td>\n",
       "      <td>89.6</td>\n",
       "      <td>32.9109</td>\n",
       "      <td>62.0</td>\n",
       "      <td>23.080688</td>\n",
       "      <td>29.385649</td>\n",
       "      <td>...</td>\n",
       "      <td>0.978756</td>\n",
       "      <td>19.821637</td>\n",
       "      <td>57.527410</td>\n",
       "      <td>4.438966</td>\n",
       "      <td>5.731393</td>\n",
       "      <td>57.335593</td>\n",
       "      <td>56.305299</td>\n",
       "      <td>84.286956</td>\n",
       "      <td>1.465162</td>\n",
       "      <td>109.946859</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31129</th>\n",
       "      <td>10039_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20160923114938.1.1...</td>\n",
       "      <td>6021112</td>\n",
       "      <td>1.0</td>\n",
       "      <td>177.0</td>\n",
       "      <td>89.0</td>\n",
       "      <td>28.4082</td>\n",
       "      <td>71.0</td>\n",
       "      <td>25.247090</td>\n",
       "      <td>32.848104</td>\n",
       "      <td>...</td>\n",
       "      <td>1.025959</td>\n",
       "      <td>20.686542</td>\n",
       "      <td>60.108352</td>\n",
       "      <td>4.014342</td>\n",
       "      <td>6.149698</td>\n",
       "      <td>60.427962</td>\n",
       "      <td>61.668708</td>\n",
       "      <td>95.185324</td>\n",
       "      <td>1.583562</td>\n",
       "      <td>134.603261</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31130</th>\n",
       "      <td>14018_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20170705180448.1.1...</td>\n",
       "      <td>6021539</td>\n",
       "      <td>1.0</td>\n",
       "      <td>179.0</td>\n",
       "      <td>71.0</td>\n",
       "      <td>22.1591</td>\n",
       "      <td>69.0</td>\n",
       "      <td>25.232678</td>\n",
       "      <td>30.970184</td>\n",
       "      <td>...</td>\n",
       "      <td>0.977047</td>\n",
       "      <td>20.729378</td>\n",
       "      <td>61.148256</td>\n",
       "      <td>4.381059</td>\n",
       "      <td>6.504511</td>\n",
       "      <td>59.444884</td>\n",
       "      <td>59.744710</td>\n",
       "      <td>90.479553</td>\n",
       "      <td>1.479675</td>\n",
       "      <td>132.776803</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31131</th>\n",
       "      <td>13966_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20170702173904.1.1...</td>\n",
       "      <td>6022591</td>\n",
       "      <td>0.0</td>\n",
       "      <td>156.0</td>\n",
       "      <td>87.7</td>\n",
       "      <td>36.0371</td>\n",
       "      <td>69.0</td>\n",
       "      <td>20.946633</td>\n",
       "      <td>26.922261</td>\n",
       "      <td>...</td>\n",
       "      <td>0.959371</td>\n",
       "      <td>19.152575</td>\n",
       "      <td>55.151275</td>\n",
       "      <td>3.211435</td>\n",
       "      <td>4.959191</td>\n",
       "      <td>67.937764</td>\n",
       "      <td>52.910551</td>\n",
       "      <td>80.626829</td>\n",
       "      <td>1.461921</td>\n",
       "      <td>126.477899</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31132</th>\n",
       "      <td>16309_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20171202183632.1.1...</td>\n",
       "      <td>6023431</td>\n",
       "      <td>0.0</td>\n",
       "      <td>170.0</td>\n",
       "      <td>61.5</td>\n",
       "      <td>21.2803</td>\n",
       "      <td>51.0</td>\n",
       "      <td>22.787944</td>\n",
       "      <td>29.871832</td>\n",
       "      <td>...</td>\n",
       "      <td>0.906038</td>\n",
       "      <td>19.990121</td>\n",
       "      <td>60.840325</td>\n",
       "      <td>3.935749</td>\n",
       "      <td>5.201943</td>\n",
       "      <td>60.779334</td>\n",
       "      <td>55.123637</td>\n",
       "      <td>84.255142</td>\n",
       "      <td>1.384857</td>\n",
       "      <td>150.788401</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31133</th>\n",
       "      <td>12964_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20170418184226.1.1...</td>\n",
       "      <td>6024029</td>\n",
       "      <td>1.0</td>\n",
       "      <td>179.0</td>\n",
       "      <td>80.4</td>\n",
       "      <td>25.0928</td>\n",
       "      <td>50.0</td>\n",
       "      <td>24.323043</td>\n",
       "      <td>29.804727</td>\n",
       "      <td>...</td>\n",
       "      <td>0.975894</td>\n",
       "      <td>20.408280</td>\n",
       "      <td>63.618586</td>\n",
       "      <td>4.298543</td>\n",
       "      <td>6.171569</td>\n",
       "      <td>59.544963</td>\n",
       "      <td>62.084987</td>\n",
       "      <td>93.878589</td>\n",
       "      <td>1.475647</td>\n",
       "      <td>138.796708</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>31134 rows × 28 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       image_id                                          file_name      eid  \\\n",
       "0      14366_82  1.2.840.113619.2.110.212038.20190307144208.1.1...  1000951   \n",
       "1      01877_82  1.2.840.113619.2.110.210419.20180720150454.1.1...  1001512   \n",
       "2      05431_82  1.2.840.113619.2.110.210419.20190328125146.1.1...  1001867   \n",
       "3      10557_82  1.2.840.113619.2.110.212038.20180228105159.1.1...  1001921   \n",
       "4      01561_82  1.2.840.113619.2.110.210419.20180630141714.1.1...  1001950   \n",
       "...         ...                                                ...      ...   \n",
       "31129  10039_93  1.2.840.113619.2.110.210419.20160923114938.1.1...  6021112   \n",
       "31130  14018_93  1.2.840.113619.2.110.210419.20170705180448.1.1...  6021539   \n",
       "31131  13966_93  1.2.840.113619.2.110.210419.20170702173904.1.1...  6022591   \n",
       "31132  16309_93  1.2.840.113619.2.110.210419.20171202183632.1.1...  6023431   \n",
       "31133  12964_93  1.2.840.113619.2.110.210419.20170418184226.1.1...  6024029   \n",
       "\n",
       "       sex  standing_height_imaging_visit  weight_imaging_visit  \\\n",
       "0      1.0                          166.0                  60.8   \n",
       "1      1.0                          169.0                  77.3   \n",
       "2      0.0                          153.0                  56.1   \n",
       "3      1.0                          166.0                  64.2   \n",
       "4      1.0                          165.0                  89.6   \n",
       "...    ...                            ...                   ...   \n",
       "31129  1.0                          177.0                  89.0   \n",
       "31130  1.0                          179.0                  71.0   \n",
       "31131  0.0                          156.0                  87.7   \n",
       "31132  0.0                          170.0                  61.5   \n",
       "31133  1.0                          179.0                  80.4   \n",
       "\n",
       "       bmi_imaging_visit  age_imaging_visit  pelvic_height  pelvic_width  ...  \\\n",
       "0                22.0642               78.0      22.297293     30.113663  ...   \n",
       "1                27.0649               66.0      23.742297     30.901775  ...   \n",
       "2                23.9651               75.0      22.226944     30.934457  ...   \n",
       "3                23.2980               62.0      22.861258     28.863430  ...   \n",
       "4                32.9109               62.0      23.080688     29.385649  ...   \n",
       "...                  ...                ...            ...           ...  ...   \n",
       "31129            28.4082               71.0      25.247090     32.848104  ...   \n",
       "31130            22.1591               69.0      25.232678     30.970184  ...   \n",
       "31131            36.0371               69.0      20.946633     26.922261  ...   \n",
       "31132            21.2803               51.0      22.787944     29.871832  ...   \n",
       "31133            25.0928               50.0      24.323043     29.804727  ...   \n",
       "\n",
       "       arm_devide_torso  bi_acetabular_width  torso_length  \\\n",
       "0              1.067168            20.324819     54.043332   \n",
       "1              0.933102            20.756184     58.668433   \n",
       "2              0.991270            21.701987     51.213839   \n",
       "3              1.014700            19.784228     57.685872   \n",
       "4              0.978756            19.821637     57.527410   \n",
       "...                 ...                  ...           ...   \n",
       "31129          1.025959            20.686542     60.108352   \n",
       "31130          0.977047            20.729378     61.148256   \n",
       "31131          0.959371            19.152575     55.151275   \n",
       "31132          0.906038            19.990121     60.840325   \n",
       "31133          0.975894            20.408280     63.618586   \n",
       "\n",
       "       iliac_isthmus_breadth  acetabular_diameter  acetabular_inclination  \\\n",
       "0                   4.164529             6.141783               60.747735   \n",
       "1                   4.657824             6.035855               59.189839   \n",
       "2                   3.718747             5.531657               59.017487   \n",
       "3                   4.238295             5.834113               58.027399   \n",
       "4                   4.438966             5.731393               57.335593   \n",
       "...                      ...                  ...                     ...   \n",
       "31129               4.014342             6.149698               60.427962   \n",
       "31130               4.381059             6.504511               59.444884   \n",
       "31131               3.211435             4.959191               67.937764   \n",
       "31132               3.935749             5.201943               60.779334   \n",
       "31133               4.298543             6.171569               59.544963   \n",
       "\n",
       "       arm_length  leg_length  leg_divide_torso  pelvic_inlet_area  \n",
       "0       57.673326   89.233162          1.651141         108.190429  \n",
       "1       54.743642   85.065164          1.449931         102.660623  \n",
       "2       50.766740   78.752591          1.537721         139.989191  \n",
       "3       58.533827   86.154050          1.493503         110.520822  \n",
       "4       56.305299   84.286956          1.465162         109.946859  \n",
       "...           ...         ...               ...                ...  \n",
       "31129   61.668708   95.185324          1.583562         134.603261  \n",
       "31130   59.744710   90.479553          1.479675         132.776803  \n",
       "31131   52.910551   80.626829          1.461921         126.477899  \n",
       "31132   55.123637   84.255142          1.384857         150.788401  \n",
       "31133   62.084987   93.878589          1.475647         138.796708  \n",
       "\n",
       "[31134 rows x 28 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hip_pheno"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>image_id</th>\n",
       "      <th>file_name</th>\n",
       "      <th>eid</th>\n",
       "      <th>sex</th>\n",
       "      <th>standing_height_imaging_visit</th>\n",
       "      <th>weight_imaging_visit</th>\n",
       "      <th>bmi_imaging_visit</th>\n",
       "      <th>age_imaging_visit</th>\n",
       "      <th>pelvic_height</th>\n",
       "      <th>pelvic_width</th>\n",
       "      <th>pelvic_inlet_width</th>\n",
       "      <th>oblique_pelvic_inlet_length</th>\n",
       "      <th>subpubic_angle</th>\n",
       "      <th>bi_acetabular_width</th>\n",
       "      <th>iliac_isthmus_breadth</th>\n",
       "      <th>acetabular_diameter</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14366_82</td>\n",
       "      <td>1.2.840.113619.2.110.212038.20190307144208.1.1...</td>\n",
       "      <td>1000951</td>\n",
       "      <td>1.0</td>\n",
       "      <td>166.0</td>\n",
       "      <td>60.8</td>\n",
       "      <td>22.0642</td>\n",
       "      <td>78.0</td>\n",
       "      <td>22.297293</td>\n",
       "      <td>30.113663</td>\n",
       "      <td>13.937174</td>\n",
       "      <td>9.883806</td>\n",
       "      <td>116.365190</td>\n",
       "      <td>20.324819</td>\n",
       "      <td>4.164529</td>\n",
       "      <td>6.141783</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>01877_82</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20180720150454.1.1...</td>\n",
       "      <td>1001512</td>\n",
       "      <td>1.0</td>\n",
       "      <td>169.0</td>\n",
       "      <td>77.3</td>\n",
       "      <td>27.0649</td>\n",
       "      <td>66.0</td>\n",
       "      <td>23.742297</td>\n",
       "      <td>30.901775</td>\n",
       "      <td>13.669524</td>\n",
       "      <td>9.562262</td>\n",
       "      <td>117.491690</td>\n",
       "      <td>20.756184</td>\n",
       "      <td>4.657824</td>\n",
       "      <td>6.035855</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>05431_82</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20190328125146.1.1...</td>\n",
       "      <td>1001867</td>\n",
       "      <td>0.0</td>\n",
       "      <td>153.0</td>\n",
       "      <td>56.1</td>\n",
       "      <td>23.9651</td>\n",
       "      <td>75.0</td>\n",
       "      <td>22.226944</td>\n",
       "      <td>30.934457</td>\n",
       "      <td>15.755275</td>\n",
       "      <td>11.313022</td>\n",
       "      <td>139.787157</td>\n",
       "      <td>21.701987</td>\n",
       "      <td>3.718747</td>\n",
       "      <td>5.531657</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10557_82</td>\n",
       "      <td>1.2.840.113619.2.110.212038.20180228105159.1.1...</td>\n",
       "      <td>1001921</td>\n",
       "      <td>1.0</td>\n",
       "      <td>166.0</td>\n",
       "      <td>64.2</td>\n",
       "      <td>23.2980</td>\n",
       "      <td>62.0</td>\n",
       "      <td>22.861258</td>\n",
       "      <td>28.863430</td>\n",
       "      <td>14.217750</td>\n",
       "      <td>9.897451</td>\n",
       "      <td>116.159185</td>\n",
       "      <td>19.784228</td>\n",
       "      <td>4.238295</td>\n",
       "      <td>5.834113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>01561_82</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20180630141714.1.1...</td>\n",
       "      <td>1001950</td>\n",
       "      <td>1.0</td>\n",
       "      <td>165.0</td>\n",
       "      <td>89.6</td>\n",
       "      <td>32.9109</td>\n",
       "      <td>62.0</td>\n",
       "      <td>23.080688</td>\n",
       "      <td>29.385649</td>\n",
       "      <td>13.583558</td>\n",
       "      <td>10.305746</td>\n",
       "      <td>130.647738</td>\n",
       "      <td>19.821637</td>\n",
       "      <td>4.438966</td>\n",
       "      <td>5.731393</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31129</th>\n",
       "      <td>10039_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20160923114938.1.1...</td>\n",
       "      <td>6021112</td>\n",
       "      <td>1.0</td>\n",
       "      <td>177.0</td>\n",
       "      <td>89.0</td>\n",
       "      <td>28.4082</td>\n",
       "      <td>71.0</td>\n",
       "      <td>25.247090</td>\n",
       "      <td>32.848104</td>\n",
       "      <td>14.645469</td>\n",
       "      <td>11.702063</td>\n",
       "      <td>126.858511</td>\n",
       "      <td>20.686542</td>\n",
       "      <td>4.014342</td>\n",
       "      <td>6.149698</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31130</th>\n",
       "      <td>14018_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20170705180448.1.1...</td>\n",
       "      <td>6021539</td>\n",
       "      <td>1.0</td>\n",
       "      <td>179.0</td>\n",
       "      <td>71.0</td>\n",
       "      <td>22.1591</td>\n",
       "      <td>69.0</td>\n",
       "      <td>25.232678</td>\n",
       "      <td>30.970184</td>\n",
       "      <td>14.210350</td>\n",
       "      <td>11.896728</td>\n",
       "      <td>120.568969</td>\n",
       "      <td>20.729378</td>\n",
       "      <td>4.381059</td>\n",
       "      <td>6.504511</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31131</th>\n",
       "      <td>13966_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20170702173904.1.1...</td>\n",
       "      <td>6022591</td>\n",
       "      <td>0.0</td>\n",
       "      <td>156.0</td>\n",
       "      <td>87.7</td>\n",
       "      <td>36.0371</td>\n",
       "      <td>69.0</td>\n",
       "      <td>20.946633</td>\n",
       "      <td>26.922261</td>\n",
       "      <td>13.540263</td>\n",
       "      <td>11.893171</td>\n",
       "      <td>143.403514</td>\n",
       "      <td>19.152575</td>\n",
       "      <td>3.211435</td>\n",
       "      <td>4.959191</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31132</th>\n",
       "      <td>16309_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20171202183632.1.1...</td>\n",
       "      <td>6023431</td>\n",
       "      <td>0.0</td>\n",
       "      <td>170.0</td>\n",
       "      <td>61.5</td>\n",
       "      <td>21.2803</td>\n",
       "      <td>51.0</td>\n",
       "      <td>22.787944</td>\n",
       "      <td>29.871832</td>\n",
       "      <td>13.586974</td>\n",
       "      <td>14.130428</td>\n",
       "      <td>157.984881</td>\n",
       "      <td>19.990121</td>\n",
       "      <td>3.935749</td>\n",
       "      <td>5.201943</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31133</th>\n",
       "      <td>12964_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20170418184226.1.1...</td>\n",
       "      <td>6024029</td>\n",
       "      <td>1.0</td>\n",
       "      <td>179.0</td>\n",
       "      <td>80.4</td>\n",
       "      <td>25.0928</td>\n",
       "      <td>50.0</td>\n",
       "      <td>24.323043</td>\n",
       "      <td>29.804727</td>\n",
       "      <td>13.883682</td>\n",
       "      <td>12.728717</td>\n",
       "      <td>124.424809</td>\n",
       "      <td>20.408280</td>\n",
       "      <td>4.298543</td>\n",
       "      <td>6.171569</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>31134 rows × 16 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       image_id                                          file_name      eid  \\\n",
       "0      14366_82  1.2.840.113619.2.110.212038.20190307144208.1.1...  1000951   \n",
       "1      01877_82  1.2.840.113619.2.110.210419.20180720150454.1.1...  1001512   \n",
       "2      05431_82  1.2.840.113619.2.110.210419.20190328125146.1.1...  1001867   \n",
       "3      10557_82  1.2.840.113619.2.110.212038.20180228105159.1.1...  1001921   \n",
       "4      01561_82  1.2.840.113619.2.110.210419.20180630141714.1.1...  1001950   \n",
       "...         ...                                                ...      ...   \n",
       "31129  10039_93  1.2.840.113619.2.110.210419.20160923114938.1.1...  6021112   \n",
       "31130  14018_93  1.2.840.113619.2.110.210419.20170705180448.1.1...  6021539   \n",
       "31131  13966_93  1.2.840.113619.2.110.210419.20170702173904.1.1...  6022591   \n",
       "31132  16309_93  1.2.840.113619.2.110.210419.20171202183632.1.1...  6023431   \n",
       "31133  12964_93  1.2.840.113619.2.110.210419.20170418184226.1.1...  6024029   \n",
       "\n",
       "       sex  standing_height_imaging_visit  weight_imaging_visit  \\\n",
       "0      1.0                          166.0                  60.8   \n",
       "1      1.0                          169.0                  77.3   \n",
       "2      0.0                          153.0                  56.1   \n",
       "3      1.0                          166.0                  64.2   \n",
       "4      1.0                          165.0                  89.6   \n",
       "...    ...                            ...                   ...   \n",
       "31129  1.0                          177.0                  89.0   \n",
       "31130  1.0                          179.0                  71.0   \n",
       "31131  0.0                          156.0                  87.7   \n",
       "31132  0.0                          170.0                  61.5   \n",
       "31133  1.0                          179.0                  80.4   \n",
       "\n",
       "       bmi_imaging_visit  age_imaging_visit  pelvic_height  pelvic_width  \\\n",
       "0                22.0642               78.0      22.297293     30.113663   \n",
       "1                27.0649               66.0      23.742297     30.901775   \n",
       "2                23.9651               75.0      22.226944     30.934457   \n",
       "3                23.2980               62.0      22.861258     28.863430   \n",
       "4                32.9109               62.0      23.080688     29.385649   \n",
       "...                  ...                ...            ...           ...   \n",
       "31129            28.4082               71.0      25.247090     32.848104   \n",
       "31130            22.1591               69.0      25.232678     30.970184   \n",
       "31131            36.0371               69.0      20.946633     26.922261   \n",
       "31132            21.2803               51.0      22.787944     29.871832   \n",
       "31133            25.0928               50.0      24.323043     29.804727   \n",
       "\n",
       "       pelvic_inlet_width  oblique_pelvic_inlet_length  subpubic_angle  \\\n",
       "0               13.937174                     9.883806      116.365190   \n",
       "1               13.669524                     9.562262      117.491690   \n",
       "2               15.755275                    11.313022      139.787157   \n",
       "3               14.217750                     9.897451      116.159185   \n",
       "4               13.583558                    10.305746      130.647738   \n",
       "...                   ...                          ...             ...   \n",
       "31129           14.645469                    11.702063      126.858511   \n",
       "31130           14.210350                    11.896728      120.568969   \n",
       "31131           13.540263                    11.893171      143.403514   \n",
       "31132           13.586974                    14.130428      157.984881   \n",
       "31133           13.883682                    12.728717      124.424809   \n",
       "\n",
       "       bi_acetabular_width  iliac_isthmus_breadth  acetabular_diameter  \n",
       "0                20.324819               4.164529             6.141783  \n",
       "1                20.756184               4.657824             6.035855  \n",
       "2                21.701987               3.718747             5.531657  \n",
       "3                19.784228               4.238295             5.834113  \n",
       "4                19.821637               4.438966             5.731393  \n",
       "...                    ...                    ...                  ...  \n",
       "31129            20.686542               4.014342             6.149698  \n",
       "31130            20.729378               4.381059             6.504511  \n",
       "31131            19.152575               3.211435             4.959191  \n",
       "31132            19.990121               3.935749             5.201943  \n",
       "31133            20.408280               4.298543             6.171569  \n",
       "\n",
       "[31134 rows x 16 columns]"
      ]
     },
     "execution_count": 119,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# filter phenotypes\n",
    "\n",
    "pheno = ['image_id', 'file_name', 'eid', 'sex', 'standing_height_imaging_visit', 'weight_imaging_visit', 'bmi_imaging_visit', 'age_imaging_visit',\n",
    "         'pelvic_height', 'pelvic_width', 'pelvic_inlet_width', 'oblique_pelvic_inlet_length', \n",
    "         'subpubic_angle', 'bi_acetabular_width', 'iliac_isthmus_breadth', 'acetabular_diameter']\n",
    "hip_pheno_flt = hip_pheno[pheno]; hip_pheno_flt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "31115"
      ]
     },
     "execution_count": 120,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hip_pheno_flt_eid_list = hip_pheno_flt[(np.abs(stats.zscore(hip_pheno_flt.iloc[:, 8:])) < 4).all(axis=1)]['eid']\n",
    "len(hip_pheno_flt_eid_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>image_id</th>\n",
       "      <th>file_name</th>\n",
       "      <th>eid</th>\n",
       "      <th>sex</th>\n",
       "      <th>standing_height_imaging_visit</th>\n",
       "      <th>weight_imaging_visit</th>\n",
       "      <th>bmi_imaging_visit</th>\n",
       "      <th>age_imaging_visit</th>\n",
       "      <th>pelvic_height</th>\n",
       "      <th>pelvic_width</th>\n",
       "      <th>...</th>\n",
       "      <th>arm_devide_torso</th>\n",
       "      <th>bi_acetabular_width</th>\n",
       "      <th>torso_length</th>\n",
       "      <th>iliac_isthmus_breadth</th>\n",
       "      <th>acetabular_diameter</th>\n",
       "      <th>acetabular_inclination</th>\n",
       "      <th>arm_length</th>\n",
       "      <th>leg_length</th>\n",
       "      <th>leg_divide_torso</th>\n",
       "      <th>pelvic_inlet_area</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14366_82</td>\n",
       "      <td>1.2.840.113619.2.110.212038.20190307144208.1.1...</td>\n",
       "      <td>1000951</td>\n",
       "      <td>1.0</td>\n",
       "      <td>166.0</td>\n",
       "      <td>60.8</td>\n",
       "      <td>22.0642</td>\n",
       "      <td>78.0</td>\n",
       "      <td>22.297293</td>\n",
       "      <td>30.113663</td>\n",
       "      <td>...</td>\n",
       "      <td>1.067168</td>\n",
       "      <td>20.324819</td>\n",
       "      <td>54.043332</td>\n",
       "      <td>4.164529</td>\n",
       "      <td>6.141783</td>\n",
       "      <td>60.747735</td>\n",
       "      <td>57.673326</td>\n",
       "      <td>89.233162</td>\n",
       "      <td>1.651141</td>\n",
       "      <td>108.190429</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>01877_82</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20180720150454.1.1...</td>\n",
       "      <td>1001512</td>\n",
       "      <td>1.0</td>\n",
       "      <td>169.0</td>\n",
       "      <td>77.3</td>\n",
       "      <td>27.0649</td>\n",
       "      <td>66.0</td>\n",
       "      <td>23.742297</td>\n",
       "      <td>30.901775</td>\n",
       "      <td>...</td>\n",
       "      <td>0.933102</td>\n",
       "      <td>20.756184</td>\n",
       "      <td>58.668433</td>\n",
       "      <td>4.657824</td>\n",
       "      <td>6.035855</td>\n",
       "      <td>59.189839</td>\n",
       "      <td>54.743642</td>\n",
       "      <td>85.065164</td>\n",
       "      <td>1.449931</td>\n",
       "      <td>102.660623</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>05431_82</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20190328125146.1.1...</td>\n",
       "      <td>1001867</td>\n",
       "      <td>0.0</td>\n",
       "      <td>153.0</td>\n",
       "      <td>56.1</td>\n",
       "      <td>23.9651</td>\n",
       "      <td>75.0</td>\n",
       "      <td>22.226944</td>\n",
       "      <td>30.934457</td>\n",
       "      <td>...</td>\n",
       "      <td>0.991270</td>\n",
       "      <td>21.701987</td>\n",
       "      <td>51.213839</td>\n",
       "      <td>3.718747</td>\n",
       "      <td>5.531657</td>\n",
       "      <td>59.017487</td>\n",
       "      <td>50.766740</td>\n",
       "      <td>78.752591</td>\n",
       "      <td>1.537721</td>\n",
       "      <td>139.989191</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10557_82</td>\n",
       "      <td>1.2.840.113619.2.110.212038.20180228105159.1.1...</td>\n",
       "      <td>1001921</td>\n",
       "      <td>1.0</td>\n",
       "      <td>166.0</td>\n",
       "      <td>64.2</td>\n",
       "      <td>23.2980</td>\n",
       "      <td>62.0</td>\n",
       "      <td>22.861258</td>\n",
       "      <td>28.863430</td>\n",
       "      <td>...</td>\n",
       "      <td>1.014700</td>\n",
       "      <td>19.784228</td>\n",
       "      <td>57.685872</td>\n",
       "      <td>4.238295</td>\n",
       "      <td>5.834113</td>\n",
       "      <td>58.027399</td>\n",
       "      <td>58.533827</td>\n",
       "      <td>86.154050</td>\n",
       "      <td>1.493503</td>\n",
       "      <td>110.520822</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>01561_82</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20180630141714.1.1...</td>\n",
       "      <td>1001950</td>\n",
       "      <td>1.0</td>\n",
       "      <td>165.0</td>\n",
       "      <td>89.6</td>\n",
       "      <td>32.9109</td>\n",
       "      <td>62.0</td>\n",
       "      <td>23.080688</td>\n",
       "      <td>29.385649</td>\n",
       "      <td>...</td>\n",
       "      <td>0.978756</td>\n",
       "      <td>19.821637</td>\n",
       "      <td>57.527410</td>\n",
       "      <td>4.438966</td>\n",
       "      <td>5.731393</td>\n",
       "      <td>57.335593</td>\n",
       "      <td>56.305299</td>\n",
       "      <td>84.286956</td>\n",
       "      <td>1.465162</td>\n",
       "      <td>109.946859</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31129</th>\n",
       "      <td>10039_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20160923114938.1.1...</td>\n",
       "      <td>6021112</td>\n",
       "      <td>1.0</td>\n",
       "      <td>177.0</td>\n",
       "      <td>89.0</td>\n",
       "      <td>28.4082</td>\n",
       "      <td>71.0</td>\n",
       "      <td>25.247090</td>\n",
       "      <td>32.848104</td>\n",
       "      <td>...</td>\n",
       "      <td>1.025959</td>\n",
       "      <td>20.686542</td>\n",
       "      <td>60.108352</td>\n",
       "      <td>4.014342</td>\n",
       "      <td>6.149698</td>\n",
       "      <td>60.427962</td>\n",
       "      <td>61.668708</td>\n",
       "      <td>95.185324</td>\n",
       "      <td>1.583562</td>\n",
       "      <td>134.603261</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31130</th>\n",
       "      <td>14018_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20170705180448.1.1...</td>\n",
       "      <td>6021539</td>\n",
       "      <td>1.0</td>\n",
       "      <td>179.0</td>\n",
       "      <td>71.0</td>\n",
       "      <td>22.1591</td>\n",
       "      <td>69.0</td>\n",
       "      <td>25.232678</td>\n",
       "      <td>30.970184</td>\n",
       "      <td>...</td>\n",
       "      <td>0.977047</td>\n",
       "      <td>20.729378</td>\n",
       "      <td>61.148256</td>\n",
       "      <td>4.381059</td>\n",
       "      <td>6.504511</td>\n",
       "      <td>59.444884</td>\n",
       "      <td>59.744710</td>\n",
       "      <td>90.479553</td>\n",
       "      <td>1.479675</td>\n",
       "      <td>132.776803</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31131</th>\n",
       "      <td>13966_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20170702173904.1.1...</td>\n",
       "      <td>6022591</td>\n",
       "      <td>0.0</td>\n",
       "      <td>156.0</td>\n",
       "      <td>87.7</td>\n",
       "      <td>36.0371</td>\n",
       "      <td>69.0</td>\n",
       "      <td>20.946633</td>\n",
       "      <td>26.922261</td>\n",
       "      <td>...</td>\n",
       "      <td>0.959371</td>\n",
       "      <td>19.152575</td>\n",
       "      <td>55.151275</td>\n",
       "      <td>3.211435</td>\n",
       "      <td>4.959191</td>\n",
       "      <td>67.937764</td>\n",
       "      <td>52.910551</td>\n",
       "      <td>80.626829</td>\n",
       "      <td>1.461921</td>\n",
       "      <td>126.477899</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31132</th>\n",
       "      <td>16309_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20171202183632.1.1...</td>\n",
       "      <td>6023431</td>\n",
       "      <td>0.0</td>\n",
       "      <td>170.0</td>\n",
       "      <td>61.5</td>\n",
       "      <td>21.2803</td>\n",
       "      <td>51.0</td>\n",
       "      <td>22.787944</td>\n",
       "      <td>29.871832</td>\n",
       "      <td>...</td>\n",
       "      <td>0.906038</td>\n",
       "      <td>19.990121</td>\n",
       "      <td>60.840325</td>\n",
       "      <td>3.935749</td>\n",
       "      <td>5.201943</td>\n",
       "      <td>60.779334</td>\n",
       "      <td>55.123637</td>\n",
       "      <td>84.255142</td>\n",
       "      <td>1.384857</td>\n",
       "      <td>150.788401</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31133</th>\n",
       "      <td>12964_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20170418184226.1.1...</td>\n",
       "      <td>6024029</td>\n",
       "      <td>1.0</td>\n",
       "      <td>179.0</td>\n",
       "      <td>80.4</td>\n",
       "      <td>25.0928</td>\n",
       "      <td>50.0</td>\n",
       "      <td>24.323043</td>\n",
       "      <td>29.804727</td>\n",
       "      <td>...</td>\n",
       "      <td>0.975894</td>\n",
       "      <td>20.408280</td>\n",
       "      <td>63.618586</td>\n",
       "      <td>4.298543</td>\n",
       "      <td>6.171569</td>\n",
       "      <td>59.544963</td>\n",
       "      <td>62.084987</td>\n",
       "      <td>93.878589</td>\n",
       "      <td>1.475647</td>\n",
       "      <td>138.796708</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>31115 rows × 28 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       image_id                                          file_name      eid  \\\n",
       "0      14366_82  1.2.840.113619.2.110.212038.20190307144208.1.1...  1000951   \n",
       "1      01877_82  1.2.840.113619.2.110.210419.20180720150454.1.1...  1001512   \n",
       "2      05431_82  1.2.840.113619.2.110.210419.20190328125146.1.1...  1001867   \n",
       "3      10557_82  1.2.840.113619.2.110.212038.20180228105159.1.1...  1001921   \n",
       "4      01561_82  1.2.840.113619.2.110.210419.20180630141714.1.1...  1001950   \n",
       "...         ...                                                ...      ...   \n",
       "31129  10039_93  1.2.840.113619.2.110.210419.20160923114938.1.1...  6021112   \n",
       "31130  14018_93  1.2.840.113619.2.110.210419.20170705180448.1.1...  6021539   \n",
       "31131  13966_93  1.2.840.113619.2.110.210419.20170702173904.1.1...  6022591   \n",
       "31132  16309_93  1.2.840.113619.2.110.210419.20171202183632.1.1...  6023431   \n",
       "31133  12964_93  1.2.840.113619.2.110.210419.20170418184226.1.1...  6024029   \n",
       "\n",
       "       sex  standing_height_imaging_visit  weight_imaging_visit  \\\n",
       "0      1.0                          166.0                  60.8   \n",
       "1      1.0                          169.0                  77.3   \n",
       "2      0.0                          153.0                  56.1   \n",
       "3      1.0                          166.0                  64.2   \n",
       "4      1.0                          165.0                  89.6   \n",
       "...    ...                            ...                   ...   \n",
       "31129  1.0                          177.0                  89.0   \n",
       "31130  1.0                          179.0                  71.0   \n",
       "31131  0.0                          156.0                  87.7   \n",
       "31132  0.0                          170.0                  61.5   \n",
       "31133  1.0                          179.0                  80.4   \n",
       "\n",
       "       bmi_imaging_visit  age_imaging_visit  pelvic_height  pelvic_width  ...  \\\n",
       "0                22.0642               78.0      22.297293     30.113663  ...   \n",
       "1                27.0649               66.0      23.742297     30.901775  ...   \n",
       "2                23.9651               75.0      22.226944     30.934457  ...   \n",
       "3                23.2980               62.0      22.861258     28.863430  ...   \n",
       "4                32.9109               62.0      23.080688     29.385649  ...   \n",
       "...                  ...                ...            ...           ...  ...   \n",
       "31129            28.4082               71.0      25.247090     32.848104  ...   \n",
       "31130            22.1591               69.0      25.232678     30.970184  ...   \n",
       "31131            36.0371               69.0      20.946633     26.922261  ...   \n",
       "31132            21.2803               51.0      22.787944     29.871832  ...   \n",
       "31133            25.0928               50.0      24.323043     29.804727  ...   \n",
       "\n",
       "       arm_devide_torso  bi_acetabular_width  torso_length  \\\n",
       "0              1.067168            20.324819     54.043332   \n",
       "1              0.933102            20.756184     58.668433   \n",
       "2              0.991270            21.701987     51.213839   \n",
       "3              1.014700            19.784228     57.685872   \n",
       "4              0.978756            19.821637     57.527410   \n",
       "...                 ...                  ...           ...   \n",
       "31129          1.025959            20.686542     60.108352   \n",
       "31130          0.977047            20.729378     61.148256   \n",
       "31131          0.959371            19.152575     55.151275   \n",
       "31132          0.906038            19.990121     60.840325   \n",
       "31133          0.975894            20.408280     63.618586   \n",
       "\n",
       "       iliac_isthmus_breadth  acetabular_diameter  acetabular_inclination  \\\n",
       "0                   4.164529             6.141783               60.747735   \n",
       "1                   4.657824             6.035855               59.189839   \n",
       "2                   3.718747             5.531657               59.017487   \n",
       "3                   4.238295             5.834113               58.027399   \n",
       "4                   4.438966             5.731393               57.335593   \n",
       "...                      ...                  ...                     ...   \n",
       "31129               4.014342             6.149698               60.427962   \n",
       "31130               4.381059             6.504511               59.444884   \n",
       "31131               3.211435             4.959191               67.937764   \n",
       "31132               3.935749             5.201943               60.779334   \n",
       "31133               4.298543             6.171569               59.544963   \n",
       "\n",
       "       arm_length  leg_length  leg_divide_torso  pelvic_inlet_area  \n",
       "0       57.673326   89.233162          1.651141         108.190429  \n",
       "1       54.743642   85.065164          1.449931         102.660623  \n",
       "2       50.766740   78.752591          1.537721         139.989191  \n",
       "3       58.533827   86.154050          1.493503         110.520822  \n",
       "4       56.305299   84.286956          1.465162         109.946859  \n",
       "...           ...         ...               ...                ...  \n",
       "31129   61.668708   95.185324          1.583562         134.603261  \n",
       "31130   59.744710   90.479553          1.479675         132.776803  \n",
       "31131   52.910551   80.626829          1.461921         126.477899  \n",
       "31132   55.123637   84.255142          1.384857         150.788401  \n",
       "31133   62.084987   93.878589          1.475647         138.796708  \n",
       "\n",
       "[31115 rows x 28 columns]"
      ]
     },
     "execution_count": 121,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hip_pheno_flt = hip_pheno[hip_pheno['eid'].isin(hip_pheno_flt_eid_list)]; hip_pheno_flt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {},
   "outputs": [],
   "source": [
    "hip_pheno_flt.to_csv('key_results/hip_select_pheno_cm_eid_flt_z_flt.csv', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Other skeletal traits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "hip_pheno = pd.read_csv('key_results/hip_select_pheno_cm_eid_flt.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['image_id', 'file_name', 'eid', 'sex', 'standing_height_imaging_visit',\n",
       "       'weight_imaging_visit', 'bmi_imaging_visit', 'age_imaging_visit',\n",
       "       'pelvic_height', 'pelvic_width', 'pelvic_inlet_width',\n",
       "       'iliac_flare_ratio', 'oblique_pelvic_inlet_length', 'iliac_flare_angle',\n",
       "       'subpubic_angle', 'head_diameter', 'trochanter_distance',\n",
       "       'shoulder_width', 'arm_devide_torso', 'bi_acetabular_width',\n",
       "       'torso_length', 'iliac_isthmus_breadth', 'acetabular_diameter',\n",
       "       'acetabular_inclination', 'humerus', 'femur', 'forearm', 'tibia',\n",
       "       'arm_length', 'leg_length', 'leg_divide_torso', 'pelvic_inlet_area'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hip_pheno.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>image_id</th>\n",
       "      <th>file_name</th>\n",
       "      <th>eid</th>\n",
       "      <th>sex</th>\n",
       "      <th>standing_height_imaging_visit</th>\n",
       "      <th>weight_imaging_visit</th>\n",
       "      <th>bmi_imaging_visit</th>\n",
       "      <th>age_imaging_visit</th>\n",
       "      <th>head_diameter</th>\n",
       "      <th>trochanter_distance</th>\n",
       "      <th>shoulder_width</th>\n",
       "      <th>torso_length</th>\n",
       "      <th>humerus</th>\n",
       "      <th>femur</th>\n",
       "      <th>forearm</th>\n",
       "      <th>tibia</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14366_82</td>\n",
       "      <td>1.2.840.113619.2.110.212038.20190307144208.1.1...</td>\n",
       "      <td>1000951</td>\n",
       "      <td>1.0</td>\n",
       "      <td>166.0</td>\n",
       "      <td>60.8</td>\n",
       "      <td>22.0642</td>\n",
       "      <td>78.0</td>\n",
       "      <td>14.112233</td>\n",
       "      <td>30.091232</td>\n",
       "      <td>36.696399</td>\n",
       "      <td>54.043332</td>\n",
       "      <td>30.816804</td>\n",
       "      <td>48.489427</td>\n",
       "      <td>26.856522</td>\n",
       "      <td>40.743734</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>01877_82</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20180720150454.1.1...</td>\n",
       "      <td>1001512</td>\n",
       "      <td>1.0</td>\n",
       "      <td>169.0</td>\n",
       "      <td>77.3</td>\n",
       "      <td>27.0649</td>\n",
       "      <td>66.0</td>\n",
       "      <td>13.903518</td>\n",
       "      <td>30.813208</td>\n",
       "      <td>39.203992</td>\n",
       "      <td>58.668433</td>\n",
       "      <td>28.115456</td>\n",
       "      <td>44.984030</td>\n",
       "      <td>26.628187</td>\n",
       "      <td>40.081134</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>05431_82</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20190328125146.1.1...</td>\n",
       "      <td>1001867</td>\n",
       "      <td>0.0</td>\n",
       "      <td>153.0</td>\n",
       "      <td>56.1</td>\n",
       "      <td>23.9651</td>\n",
       "      <td>75.0</td>\n",
       "      <td>13.029582</td>\n",
       "      <td>31.173284</td>\n",
       "      <td>35.909407</td>\n",
       "      <td>51.213839</td>\n",
       "      <td>26.778209</td>\n",
       "      <td>42.934407</td>\n",
       "      <td>23.988531</td>\n",
       "      <td>35.818183</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10557_82</td>\n",
       "      <td>1.2.840.113619.2.110.212038.20180228105159.1.1...</td>\n",
       "      <td>1001921</td>\n",
       "      <td>1.0</td>\n",
       "      <td>166.0</td>\n",
       "      <td>64.2</td>\n",
       "      <td>23.2980</td>\n",
       "      <td>62.0</td>\n",
       "      <td>14.480994</td>\n",
       "      <td>29.777444</td>\n",
       "      <td>33.219853</td>\n",
       "      <td>57.685872</td>\n",
       "      <td>29.972729</td>\n",
       "      <td>46.587679</td>\n",
       "      <td>28.561098</td>\n",
       "      <td>39.566371</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>01561_82</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20180630141714.1.1...</td>\n",
       "      <td>1001950</td>\n",
       "      <td>1.0</td>\n",
       "      <td>165.0</td>\n",
       "      <td>89.6</td>\n",
       "      <td>32.9109</td>\n",
       "      <td>62.0</td>\n",
       "      <td>14.042468</td>\n",
       "      <td>28.838122</td>\n",
       "      <td>39.066233</td>\n",
       "      <td>57.527410</td>\n",
       "      <td>29.012823</td>\n",
       "      <td>45.245573</td>\n",
       "      <td>27.292476</td>\n",
       "      <td>39.041382</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31129</th>\n",
       "      <td>10039_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20160923114938.1.1...</td>\n",
       "      <td>6021112</td>\n",
       "      <td>1.0</td>\n",
       "      <td>177.0</td>\n",
       "      <td>89.0</td>\n",
       "      <td>28.4082</td>\n",
       "      <td>71.0</td>\n",
       "      <td>13.402493</td>\n",
       "      <td>30.995640</td>\n",
       "      <td>41.118337</td>\n",
       "      <td>60.108352</td>\n",
       "      <td>31.540230</td>\n",
       "      <td>50.329159</td>\n",
       "      <td>30.128478</td>\n",
       "      <td>44.856165</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31130</th>\n",
       "      <td>14018_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20170705180448.1.1...</td>\n",
       "      <td>6021539</td>\n",
       "      <td>1.0</td>\n",
       "      <td>179.0</td>\n",
       "      <td>71.0</td>\n",
       "      <td>22.1591</td>\n",
       "      <td>69.0</td>\n",
       "      <td>13.781823</td>\n",
       "      <td>31.116022</td>\n",
       "      <td>38.579135</td>\n",
       "      <td>61.148256</td>\n",
       "      <td>30.777846</td>\n",
       "      <td>48.408803</td>\n",
       "      <td>28.966865</td>\n",
       "      <td>42.070750</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31131</th>\n",
       "      <td>13966_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20170702173904.1.1...</td>\n",
       "      <td>6022591</td>\n",
       "      <td>0.0</td>\n",
       "      <td>156.0</td>\n",
       "      <td>87.7</td>\n",
       "      <td>36.0371</td>\n",
       "      <td>69.0</td>\n",
       "      <td>12.551823</td>\n",
       "      <td>27.342555</td>\n",
       "      <td>35.846661</td>\n",
       "      <td>55.151275</td>\n",
       "      <td>28.030085</td>\n",
       "      <td>42.718048</td>\n",
       "      <td>24.880466</td>\n",
       "      <td>37.908781</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31132</th>\n",
       "      <td>16309_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20171202183632.1.1...</td>\n",
       "      <td>6023431</td>\n",
       "      <td>0.0</td>\n",
       "      <td>170.0</td>\n",
       "      <td>61.5</td>\n",
       "      <td>21.2803</td>\n",
       "      <td>51.0</td>\n",
       "      <td>12.309345</td>\n",
       "      <td>27.698720</td>\n",
       "      <td>36.369916</td>\n",
       "      <td>60.840325</td>\n",
       "      <td>29.276081</td>\n",
       "      <td>45.262378</td>\n",
       "      <td>25.847555</td>\n",
       "      <td>38.992764</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31133</th>\n",
       "      <td>12964_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20170418184226.1.1...</td>\n",
       "      <td>6024029</td>\n",
       "      <td>1.0</td>\n",
       "      <td>179.0</td>\n",
       "      <td>80.4</td>\n",
       "      <td>25.0928</td>\n",
       "      <td>50.0</td>\n",
       "      <td>13.271879</td>\n",
       "      <td>30.125650</td>\n",
       "      <td>39.486231</td>\n",
       "      <td>63.618586</td>\n",
       "      <td>32.678000</td>\n",
       "      <td>50.968299</td>\n",
       "      <td>29.406987</td>\n",
       "      <td>42.910290</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>31134 rows × 16 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       image_id                                          file_name      eid  \\\n",
       "0      14366_82  1.2.840.113619.2.110.212038.20190307144208.1.1...  1000951   \n",
       "1      01877_82  1.2.840.113619.2.110.210419.20180720150454.1.1...  1001512   \n",
       "2      05431_82  1.2.840.113619.2.110.210419.20190328125146.1.1...  1001867   \n",
       "3      10557_82  1.2.840.113619.2.110.212038.20180228105159.1.1...  1001921   \n",
       "4      01561_82  1.2.840.113619.2.110.210419.20180630141714.1.1...  1001950   \n",
       "...         ...                                                ...      ...   \n",
       "31129  10039_93  1.2.840.113619.2.110.210419.20160923114938.1.1...  6021112   \n",
       "31130  14018_93  1.2.840.113619.2.110.210419.20170705180448.1.1...  6021539   \n",
       "31131  13966_93  1.2.840.113619.2.110.210419.20170702173904.1.1...  6022591   \n",
       "31132  16309_93  1.2.840.113619.2.110.210419.20171202183632.1.1...  6023431   \n",
       "31133  12964_93  1.2.840.113619.2.110.210419.20170418184226.1.1...  6024029   \n",
       "\n",
       "       sex  standing_height_imaging_visit  weight_imaging_visit  \\\n",
       "0      1.0                          166.0                  60.8   \n",
       "1      1.0                          169.0                  77.3   \n",
       "2      0.0                          153.0                  56.1   \n",
       "3      1.0                          166.0                  64.2   \n",
       "4      1.0                          165.0                  89.6   \n",
       "...    ...                            ...                   ...   \n",
       "31129  1.0                          177.0                  89.0   \n",
       "31130  1.0                          179.0                  71.0   \n",
       "31131  0.0                          156.0                  87.7   \n",
       "31132  0.0                          170.0                  61.5   \n",
       "31133  1.0                          179.0                  80.4   \n",
       "\n",
       "       bmi_imaging_visit  age_imaging_visit  head_diameter  \\\n",
       "0                22.0642               78.0      14.112233   \n",
       "1                27.0649               66.0      13.903518   \n",
       "2                23.9651               75.0      13.029582   \n",
       "3                23.2980               62.0      14.480994   \n",
       "4                32.9109               62.0      14.042468   \n",
       "...                  ...                ...            ...   \n",
       "31129            28.4082               71.0      13.402493   \n",
       "31130            22.1591               69.0      13.781823   \n",
       "31131            36.0371               69.0      12.551823   \n",
       "31132            21.2803               51.0      12.309345   \n",
       "31133            25.0928               50.0      13.271879   \n",
       "\n",
       "       trochanter_distance  shoulder_width  torso_length    humerus  \\\n",
       "0                30.091232       36.696399     54.043332  30.816804   \n",
       "1                30.813208       39.203992     58.668433  28.115456   \n",
       "2                31.173284       35.909407     51.213839  26.778209   \n",
       "3                29.777444       33.219853     57.685872  29.972729   \n",
       "4                28.838122       39.066233     57.527410  29.012823   \n",
       "...                    ...             ...           ...        ...   \n",
       "31129            30.995640       41.118337     60.108352  31.540230   \n",
       "31130            31.116022       38.579135     61.148256  30.777846   \n",
       "31131            27.342555       35.846661     55.151275  28.030085   \n",
       "31132            27.698720       36.369916     60.840325  29.276081   \n",
       "31133            30.125650       39.486231     63.618586  32.678000   \n",
       "\n",
       "           femur    forearm      tibia  \n",
       "0      48.489427  26.856522  40.743734  \n",
       "1      44.984030  26.628187  40.081134  \n",
       "2      42.934407  23.988531  35.818183  \n",
       "3      46.587679  28.561098  39.566371  \n",
       "4      45.245573  27.292476  39.041382  \n",
       "...          ...        ...        ...  \n",
       "31129  50.329159  30.128478  44.856165  \n",
       "31130  48.408803  28.966865  42.070750  \n",
       "31131  42.718048  24.880466  37.908781  \n",
       "31132  45.262378  25.847555  38.992764  \n",
       "31133  50.968299  29.406987  42.910290  \n",
       "\n",
       "[31134 rows x 16 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# filter phenotypes\n",
    "\n",
    "pheno = ['image_id', 'file_name', 'eid', 'sex', 'standing_height_imaging_visit', 'weight_imaging_visit', 'bmi_imaging_visit', 'age_imaging_visit',\n",
    "         'head_diameter', 'trochanter_distance', 'shoulder_width', 'torso_length', 'humerus', 'femur', 'forearm', 'tibia']\n",
    "hip_pheno_flt = hip_pheno[pheno]; hip_pheno_flt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "z_keep_eids = pd.read_csv(\"key_results/hip_select_pheno_cm_eid_flt_z_flt.csv\")['eid'].tolist()\n",
    "\n",
    "hip_pheno_flt = hip_pheno_flt[hip_pheno_flt['eid'].isin(z_keep_eids)]\n",
    "\n",
    "hip_pheno_flt.to_csv(\"key_results/skeletal_cm_eid_flt_z_flt\", index = False)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Heritability"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Prepare data for h2g"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### use raw phenotype data (cm)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>image_id</th>\n",
       "      <th>file_name</th>\n",
       "      <th>eid</th>\n",
       "      <th>sex</th>\n",
       "      <th>standing_height_imaging_visit</th>\n",
       "      <th>weight_imaging_visit</th>\n",
       "      <th>bmi_imaging_visit</th>\n",
       "      <th>age_imaging_visit</th>\n",
       "      <th>pelvic_height</th>\n",
       "      <th>pelvic_width</th>\n",
       "      <th>...</th>\n",
       "      <th>trochanter_left2trochanter_right</th>\n",
       "      <th>shoulder_width</th>\n",
       "      <th>arm_devide_torso</th>\n",
       "      <th>bi_acetabular_width</th>\n",
       "      <th>iliac_isthmus_breadth</th>\n",
       "      <th>acetabular_diameter</th>\n",
       "      <th>acetabular_inclination</th>\n",
       "      <th>arm_length</th>\n",
       "      <th>leg_length</th>\n",
       "      <th>pelvic_inlet_area</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14366_82</td>\n",
       "      <td>1.2.840.113619.2.110.212038.20190307144208.1.1...</td>\n",
       "      <td>1000951</td>\n",
       "      <td>1.0</td>\n",
       "      <td>166.0</td>\n",
       "      <td>60.8</td>\n",
       "      <td>22.0642</td>\n",
       "      <td>78.0</td>\n",
       "      <td>22.297293</td>\n",
       "      <td>30.113663</td>\n",
       "      <td>...</td>\n",
       "      <td>30.091232</td>\n",
       "      <td>36.696399</td>\n",
       "      <td>1.067168</td>\n",
       "      <td>20.324819</td>\n",
       "      <td>4.164529</td>\n",
       "      <td>6.141783</td>\n",
       "      <td>60.747735</td>\n",
       "      <td>57.673326</td>\n",
       "      <td>89.233162</td>\n",
       "      <td>108.190429</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>01877_82</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20180720150454.1.1...</td>\n",
       "      <td>1001512</td>\n",
       "      <td>1.0</td>\n",
       "      <td>169.0</td>\n",
       "      <td>77.3</td>\n",
       "      <td>27.0649</td>\n",
       "      <td>66.0</td>\n",
       "      <td>23.742297</td>\n",
       "      <td>30.901775</td>\n",
       "      <td>...</td>\n",
       "      <td>30.813208</td>\n",
       "      <td>39.203992</td>\n",
       "      <td>0.933102</td>\n",
       "      <td>20.756184</td>\n",
       "      <td>4.657824</td>\n",
       "      <td>6.035855</td>\n",
       "      <td>59.189839</td>\n",
       "      <td>54.743642</td>\n",
       "      <td>85.065164</td>\n",
       "      <td>102.660623</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>05431_82</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20190328125146.1.1...</td>\n",
       "      <td>1001867</td>\n",
       "      <td>0.0</td>\n",
       "      <td>153.0</td>\n",
       "      <td>56.1</td>\n",
       "      <td>23.9651</td>\n",
       "      <td>75.0</td>\n",
       "      <td>22.226944</td>\n",
       "      <td>30.934457</td>\n",
       "      <td>...</td>\n",
       "      <td>31.173284</td>\n",
       "      <td>35.909407</td>\n",
       "      <td>0.991270</td>\n",
       "      <td>21.701987</td>\n",
       "      <td>3.718747</td>\n",
       "      <td>5.531657</td>\n",
       "      <td>59.017487</td>\n",
       "      <td>50.766740</td>\n",
       "      <td>78.752591</td>\n",
       "      <td>139.989191</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10557_82</td>\n",
       "      <td>1.2.840.113619.2.110.212038.20180228105159.1.1...</td>\n",
       "      <td>1001921</td>\n",
       "      <td>1.0</td>\n",
       "      <td>166.0</td>\n",
       "      <td>64.2</td>\n",
       "      <td>23.2980</td>\n",
       "      <td>62.0</td>\n",
       "      <td>22.861258</td>\n",
       "      <td>28.863430</td>\n",
       "      <td>...</td>\n",
       "      <td>29.777444</td>\n",
       "      <td>33.219853</td>\n",
       "      <td>1.014700</td>\n",
       "      <td>19.784228</td>\n",
       "      <td>4.238295</td>\n",
       "      <td>5.834113</td>\n",
       "      <td>58.027399</td>\n",
       "      <td>58.533827</td>\n",
       "      <td>86.154050</td>\n",
       "      <td>110.520822</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>01561_82</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20180630141714.1.1...</td>\n",
       "      <td>1001950</td>\n",
       "      <td>1.0</td>\n",
       "      <td>165.0</td>\n",
       "      <td>89.6</td>\n",
       "      <td>32.9109</td>\n",
       "      <td>62.0</td>\n",
       "      <td>23.080688</td>\n",
       "      <td>29.385649</td>\n",
       "      <td>...</td>\n",
       "      <td>28.838122</td>\n",
       "      <td>39.066233</td>\n",
       "      <td>0.978756</td>\n",
       "      <td>19.821637</td>\n",
       "      <td>4.438966</td>\n",
       "      <td>5.731393</td>\n",
       "      <td>57.335593</td>\n",
       "      <td>56.305299</td>\n",
       "      <td>84.286956</td>\n",
       "      <td>109.946859</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31110</th>\n",
       "      <td>10039_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20160923114938.1.1...</td>\n",
       "      <td>6021112</td>\n",
       "      <td>1.0</td>\n",
       "      <td>177.0</td>\n",
       "      <td>89.0</td>\n",
       "      <td>28.4082</td>\n",
       "      <td>71.0</td>\n",
       "      <td>25.247090</td>\n",
       "      <td>32.848104</td>\n",
       "      <td>...</td>\n",
       "      <td>30.995640</td>\n",
       "      <td>41.118337</td>\n",
       "      <td>1.025959</td>\n",
       "      <td>20.686542</td>\n",
       "      <td>4.014342</td>\n",
       "      <td>6.149698</td>\n",
       "      <td>60.427962</td>\n",
       "      <td>61.668708</td>\n",
       "      <td>95.185324</td>\n",
       "      <td>134.603261</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31111</th>\n",
       "      <td>14018_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20170705180448.1.1...</td>\n",
       "      <td>6021539</td>\n",
       "      <td>1.0</td>\n",
       "      <td>179.0</td>\n",
       "      <td>71.0</td>\n",
       "      <td>22.1591</td>\n",
       "      <td>69.0</td>\n",
       "      <td>25.232678</td>\n",
       "      <td>30.970184</td>\n",
       "      <td>...</td>\n",
       "      <td>31.116022</td>\n",
       "      <td>38.579135</td>\n",
       "      <td>0.977047</td>\n",
       "      <td>20.729378</td>\n",
       "      <td>4.381059</td>\n",
       "      <td>6.504511</td>\n",
       "      <td>59.444884</td>\n",
       "      <td>59.744710</td>\n",
       "      <td>90.479553</td>\n",
       "      <td>132.776803</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31112</th>\n",
       "      <td>13966_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20170702173904.1.1...</td>\n",
       "      <td>6022591</td>\n",
       "      <td>0.0</td>\n",
       "      <td>156.0</td>\n",
       "      <td>87.7</td>\n",
       "      <td>36.0371</td>\n",
       "      <td>69.0</td>\n",
       "      <td>20.946633</td>\n",
       "      <td>26.922261</td>\n",
       "      <td>...</td>\n",
       "      <td>27.342555</td>\n",
       "      <td>35.846661</td>\n",
       "      <td>0.959371</td>\n",
       "      <td>19.152575</td>\n",
       "      <td>3.211435</td>\n",
       "      <td>4.959191</td>\n",
       "      <td>67.937764</td>\n",
       "      <td>52.910551</td>\n",
       "      <td>80.626829</td>\n",
       "      <td>126.477899</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31113</th>\n",
       "      <td>16309_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20171202183632.1.1...</td>\n",
       "      <td>6023431</td>\n",
       "      <td>0.0</td>\n",
       "      <td>170.0</td>\n",
       "      <td>61.5</td>\n",
       "      <td>21.2803</td>\n",
       "      <td>51.0</td>\n",
       "      <td>22.787944</td>\n",
       "      <td>29.871832</td>\n",
       "      <td>...</td>\n",
       "      <td>27.698720</td>\n",
       "      <td>36.369916</td>\n",
       "      <td>0.906038</td>\n",
       "      <td>19.990121</td>\n",
       "      <td>3.935749</td>\n",
       "      <td>5.201943</td>\n",
       "      <td>60.779334</td>\n",
       "      <td>55.123637</td>\n",
       "      <td>84.255142</td>\n",
       "      <td>150.788401</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31114</th>\n",
       "      <td>12964_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20170418184226.1.1...</td>\n",
       "      <td>6024029</td>\n",
       "      <td>1.0</td>\n",
       "      <td>179.0</td>\n",
       "      <td>80.4</td>\n",
       "      <td>25.0928</td>\n",
       "      <td>50.0</td>\n",
       "      <td>24.323043</td>\n",
       "      <td>29.804727</td>\n",
       "      <td>...</td>\n",
       "      <td>30.125650</td>\n",
       "      <td>39.486231</td>\n",
       "      <td>0.975894</td>\n",
       "      <td>20.408280</td>\n",
       "      <td>4.298543</td>\n",
       "      <td>6.171569</td>\n",
       "      <td>59.544963</td>\n",
       "      <td>62.084987</td>\n",
       "      <td>93.878589</td>\n",
       "      <td>138.796708</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>31115 rows × 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       image_id                                          file_name      eid  \\\n",
       "0      14366_82  1.2.840.113619.2.110.212038.20190307144208.1.1...  1000951   \n",
       "1      01877_82  1.2.840.113619.2.110.210419.20180720150454.1.1...  1001512   \n",
       "2      05431_82  1.2.840.113619.2.110.210419.20190328125146.1.1...  1001867   \n",
       "3      10557_82  1.2.840.113619.2.110.212038.20180228105159.1.1...  1001921   \n",
       "4      01561_82  1.2.840.113619.2.110.210419.20180630141714.1.1...  1001950   \n",
       "...         ...                                                ...      ...   \n",
       "31110  10039_93  1.2.840.113619.2.110.210419.20160923114938.1.1...  6021112   \n",
       "31111  14018_93  1.2.840.113619.2.110.210419.20170705180448.1.1...  6021539   \n",
       "31112  13966_93  1.2.840.113619.2.110.210419.20170702173904.1.1...  6022591   \n",
       "31113  16309_93  1.2.840.113619.2.110.210419.20171202183632.1.1...  6023431   \n",
       "31114  12964_93  1.2.840.113619.2.110.210419.20170418184226.1.1...  6024029   \n",
       "\n",
       "       sex  standing_height_imaging_visit  weight_imaging_visit  \\\n",
       "0      1.0                          166.0                  60.8   \n",
       "1      1.0                          169.0                  77.3   \n",
       "2      0.0                          153.0                  56.1   \n",
       "3      1.0                          166.0                  64.2   \n",
       "4      1.0                          165.0                  89.6   \n",
       "...    ...                            ...                   ...   \n",
       "31110  1.0                          177.0                  89.0   \n",
       "31111  1.0                          179.0                  71.0   \n",
       "31112  0.0                          156.0                  87.7   \n",
       "31113  0.0                          170.0                  61.5   \n",
       "31114  1.0                          179.0                  80.4   \n",
       "\n",
       "       bmi_imaging_visit  age_imaging_visit  pelvic_height  pelvic_width  ...  \\\n",
       "0                22.0642               78.0      22.297293     30.113663  ...   \n",
       "1                27.0649               66.0      23.742297     30.901775  ...   \n",
       "2                23.9651               75.0      22.226944     30.934457  ...   \n",
       "3                23.2980               62.0      22.861258     28.863430  ...   \n",
       "4                32.9109               62.0      23.080688     29.385649  ...   \n",
       "...                  ...                ...            ...           ...  ...   \n",
       "31110            28.4082               71.0      25.247090     32.848104  ...   \n",
       "31111            22.1591               69.0      25.232678     30.970184  ...   \n",
       "31112            36.0371               69.0      20.946633     26.922261  ...   \n",
       "31113            21.2803               51.0      22.787944     29.871832  ...   \n",
       "31114            25.0928               50.0      24.323043     29.804727  ...   \n",
       "\n",
       "       trochanter_left2trochanter_right  shoulder_width  arm_devide_torso  \\\n",
       "0                             30.091232       36.696399          1.067168   \n",
       "1                             30.813208       39.203992          0.933102   \n",
       "2                             31.173284       35.909407          0.991270   \n",
       "3                             29.777444       33.219853          1.014700   \n",
       "4                             28.838122       39.066233          0.978756   \n",
       "...                                 ...             ...               ...   \n",
       "31110                         30.995640       41.118337          1.025959   \n",
       "31111                         31.116022       38.579135          0.977047   \n",
       "31112                         27.342555       35.846661          0.959371   \n",
       "31113                         27.698720       36.369916          0.906038   \n",
       "31114                         30.125650       39.486231          0.975894   \n",
       "\n",
       "       bi_acetabular_width  iliac_isthmus_breadth  acetabular_diameter  \\\n",
       "0                20.324819               4.164529             6.141783   \n",
       "1                20.756184               4.657824             6.035855   \n",
       "2                21.701987               3.718747             5.531657   \n",
       "3                19.784228               4.238295             5.834113   \n",
       "4                19.821637               4.438966             5.731393   \n",
       "...                    ...                    ...                  ...   \n",
       "31110            20.686542               4.014342             6.149698   \n",
       "31111            20.729378               4.381059             6.504511   \n",
       "31112            19.152575               3.211435             4.959191   \n",
       "31113            19.990121               3.935749             5.201943   \n",
       "31114            20.408280               4.298543             6.171569   \n",
       "\n",
       "       acetabular_inclination  arm_length  leg_length  pelvic_inlet_area  \n",
       "0                   60.747735   57.673326   89.233162         108.190429  \n",
       "1                   59.189839   54.743642   85.065164         102.660623  \n",
       "2                   59.017487   50.766740   78.752591         139.989191  \n",
       "3                   58.027399   58.533827   86.154050         110.520822  \n",
       "4                   57.335593   56.305299   84.286956         109.946859  \n",
       "...                       ...         ...         ...                ...  \n",
       "31110               60.427962   61.668708   95.185324         134.603261  \n",
       "31111               59.444884   59.744710   90.479553         132.776803  \n",
       "31112               67.937764   52.910551   80.626829         126.477899  \n",
       "31113               60.779334   55.123637   84.255142         150.788401  \n",
       "31114               59.544963   62.084987   93.878589         138.796708  \n",
       "\n",
       "[31115 rows x 26 columns]"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hip_pheno_flt_eid_flt_cm = pd.read_csv('key_results/hip_select_pheno_cm_eid_flt_z_flt.csv'); hip_pheno_flt_eid_flt_cm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>pelvic_height</th>\n",
       "      <th>pelvic_width</th>\n",
       "      <th>pelvic_inlet_width</th>\n",
       "      <th>iliac_flare_ratio</th>\n",
       "      <th>oblique_pelvic_inlet_length</th>\n",
       "      <th>iliac_flare_angle</th>\n",
       "      <th>subpubic_angle</th>\n",
       "      <th>ear_left2ear_right</th>\n",
       "      <th>trochanter_left2trochanter_right</th>\n",
       "      <th>shoulder_width</th>\n",
       "      <th>arm_devide_torso</th>\n",
       "      <th>bi_acetabular_width</th>\n",
       "      <th>iliac_isthmus_breadth</th>\n",
       "      <th>acetabular_diameter</th>\n",
       "      <th>acetabular_inclination</th>\n",
       "      <th>arm_length</th>\n",
       "      <th>leg_length</th>\n",
       "      <th>pelvic_inlet_area</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1000951</td>\n",
       "      <td>1000951</td>\n",
       "      <td>22.297293</td>\n",
       "      <td>30.113663</td>\n",
       "      <td>13.937174</td>\n",
       "      <td>0.462819</td>\n",
       "      <td>9.883806</td>\n",
       "      <td>97.533181</td>\n",
       "      <td>116.365190</td>\n",
       "      <td>14.112233</td>\n",
       "      <td>30.091232</td>\n",
       "      <td>36.696399</td>\n",
       "      <td>1.067168</td>\n",
       "      <td>20.324819</td>\n",
       "      <td>4.164529</td>\n",
       "      <td>6.141783</td>\n",
       "      <td>60.747735</td>\n",
       "      <td>57.673326</td>\n",
       "      <td>89.233162</td>\n",
       "      <td>108.190429</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1001512</td>\n",
       "      <td>1001512</td>\n",
       "      <td>23.742297</td>\n",
       "      <td>30.901775</td>\n",
       "      <td>13.669524</td>\n",
       "      <td>0.442354</td>\n",
       "      <td>9.562262</td>\n",
       "      <td>95.714405</td>\n",
       "      <td>117.491690</td>\n",
       "      <td>13.903518</td>\n",
       "      <td>30.813208</td>\n",
       "      <td>39.203992</td>\n",
       "      <td>0.933102</td>\n",
       "      <td>20.756184</td>\n",
       "      <td>4.657824</td>\n",
       "      <td>6.035855</td>\n",
       "      <td>59.189839</td>\n",
       "      <td>54.743642</td>\n",
       "      <td>85.065164</td>\n",
       "      <td>102.660623</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1001867</td>\n",
       "      <td>1001867</td>\n",
       "      <td>22.226944</td>\n",
       "      <td>30.934457</td>\n",
       "      <td>15.755275</td>\n",
       "      <td>0.509312</td>\n",
       "      <td>11.313022</td>\n",
       "      <td>101.418560</td>\n",
       "      <td>139.787157</td>\n",
       "      <td>13.029582</td>\n",
       "      <td>31.173284</td>\n",
       "      <td>35.909407</td>\n",
       "      <td>0.991270</td>\n",
       "      <td>21.701987</td>\n",
       "      <td>3.718747</td>\n",
       "      <td>5.531657</td>\n",
       "      <td>59.017487</td>\n",
       "      <td>50.766740</td>\n",
       "      <td>78.752591</td>\n",
       "      <td>139.989191</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1001921</td>\n",
       "      <td>1001921</td>\n",
       "      <td>22.861258</td>\n",
       "      <td>28.863430</td>\n",
       "      <td>14.217750</td>\n",
       "      <td>0.492587</td>\n",
       "      <td>9.897451</td>\n",
       "      <td>95.473293</td>\n",
       "      <td>116.159185</td>\n",
       "      <td>14.480994</td>\n",
       "      <td>29.777444</td>\n",
       "      <td>33.219853</td>\n",
       "      <td>1.014700</td>\n",
       "      <td>19.784228</td>\n",
       "      <td>4.238295</td>\n",
       "      <td>5.834113</td>\n",
       "      <td>58.027399</td>\n",
       "      <td>58.533827</td>\n",
       "      <td>86.154050</td>\n",
       "      <td>110.520822</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1001950</td>\n",
       "      <td>1001950</td>\n",
       "      <td>23.080688</td>\n",
       "      <td>29.385649</td>\n",
       "      <td>13.583558</td>\n",
       "      <td>0.462251</td>\n",
       "      <td>10.305746</td>\n",
       "      <td>95.952879</td>\n",
       "      <td>130.647738</td>\n",
       "      <td>14.042468</td>\n",
       "      <td>28.838122</td>\n",
       "      <td>39.066233</td>\n",
       "      <td>0.978756</td>\n",
       "      <td>19.821637</td>\n",
       "      <td>4.438966</td>\n",
       "      <td>5.731393</td>\n",
       "      <td>57.335593</td>\n",
       "      <td>56.305299</td>\n",
       "      <td>84.286956</td>\n",
       "      <td>109.946859</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31110</th>\n",
       "      <td>6021112</td>\n",
       "      <td>6021112</td>\n",
       "      <td>25.247090</td>\n",
       "      <td>32.848104</td>\n",
       "      <td>14.645469</td>\n",
       "      <td>0.445854</td>\n",
       "      <td>11.702063</td>\n",
       "      <td>95.353305</td>\n",
       "      <td>126.858511</td>\n",
       "      <td>13.402493</td>\n",
       "      <td>30.995640</td>\n",
       "      <td>41.118337</td>\n",
       "      <td>1.025959</td>\n",
       "      <td>20.686542</td>\n",
       "      <td>4.014342</td>\n",
       "      <td>6.149698</td>\n",
       "      <td>60.427962</td>\n",
       "      <td>61.668708</td>\n",
       "      <td>95.185324</td>\n",
       "      <td>134.603261</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31111</th>\n",
       "      <td>6021539</td>\n",
       "      <td>6021539</td>\n",
       "      <td>25.232678</td>\n",
       "      <td>30.970184</td>\n",
       "      <td>14.210350</td>\n",
       "      <td>0.458840</td>\n",
       "      <td>11.896728</td>\n",
       "      <td>87.564668</td>\n",
       "      <td>120.568969</td>\n",
       "      <td>13.781823</td>\n",
       "      <td>31.116022</td>\n",
       "      <td>38.579135</td>\n",
       "      <td>0.977047</td>\n",
       "      <td>20.729378</td>\n",
       "      <td>4.381059</td>\n",
       "      <td>6.504511</td>\n",
       "      <td>59.444884</td>\n",
       "      <td>59.744710</td>\n",
       "      <td>90.479553</td>\n",
       "      <td>132.776803</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31112</th>\n",
       "      <td>6022591</td>\n",
       "      <td>6022591</td>\n",
       "      <td>20.946633</td>\n",
       "      <td>26.922261</td>\n",
       "      <td>13.540263</td>\n",
       "      <td>0.502939</td>\n",
       "      <td>11.893171</td>\n",
       "      <td>88.868037</td>\n",
       "      <td>143.403514</td>\n",
       "      <td>12.551823</td>\n",
       "      <td>27.342555</td>\n",
       "      <td>35.846661</td>\n",
       "      <td>0.959371</td>\n",
       "      <td>19.152575</td>\n",
       "      <td>3.211435</td>\n",
       "      <td>4.959191</td>\n",
       "      <td>67.937764</td>\n",
       "      <td>52.910551</td>\n",
       "      <td>80.626829</td>\n",
       "      <td>126.477899</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31113</th>\n",
       "      <td>6023431</td>\n",
       "      <td>6023431</td>\n",
       "      <td>22.787944</td>\n",
       "      <td>29.871832</td>\n",
       "      <td>13.586974</td>\n",
       "      <td>0.454842</td>\n",
       "      <td>14.130428</td>\n",
       "      <td>90.205445</td>\n",
       "      <td>157.984881</td>\n",
       "      <td>12.309345</td>\n",
       "      <td>27.698720</td>\n",
       "      <td>36.369916</td>\n",
       "      <td>0.906038</td>\n",
       "      <td>19.990121</td>\n",
       "      <td>3.935749</td>\n",
       "      <td>5.201943</td>\n",
       "      <td>60.779334</td>\n",
       "      <td>55.123637</td>\n",
       "      <td>84.255142</td>\n",
       "      <td>150.788401</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31114</th>\n",
       "      <td>6024029</td>\n",
       "      <td>6024029</td>\n",
       "      <td>24.323043</td>\n",
       "      <td>29.804727</td>\n",
       "      <td>13.883682</td>\n",
       "      <td>0.465821</td>\n",
       "      <td>12.728717</td>\n",
       "      <td>85.308035</td>\n",
       "      <td>124.424809</td>\n",
       "      <td>13.271879</td>\n",
       "      <td>30.125650</td>\n",
       "      <td>39.486231</td>\n",
       "      <td>0.975894</td>\n",
       "      <td>20.408280</td>\n",
       "      <td>4.298543</td>\n",
       "      <td>6.171569</td>\n",
       "      <td>59.544963</td>\n",
       "      <td>62.084987</td>\n",
       "      <td>93.878589</td>\n",
       "      <td>138.796708</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>31115 rows × 20 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           FID      IID  pelvic_height  pelvic_width  pelvic_inlet_width  \\\n",
       "0      1000951  1000951      22.297293     30.113663           13.937174   \n",
       "1      1001512  1001512      23.742297     30.901775           13.669524   \n",
       "2      1001867  1001867      22.226944     30.934457           15.755275   \n",
       "3      1001921  1001921      22.861258     28.863430           14.217750   \n",
       "4      1001950  1001950      23.080688     29.385649           13.583558   \n",
       "...        ...      ...            ...           ...                 ...   \n",
       "31110  6021112  6021112      25.247090     32.848104           14.645469   \n",
       "31111  6021539  6021539      25.232678     30.970184           14.210350   \n",
       "31112  6022591  6022591      20.946633     26.922261           13.540263   \n",
       "31113  6023431  6023431      22.787944     29.871832           13.586974   \n",
       "31114  6024029  6024029      24.323043     29.804727           13.883682   \n",
       "\n",
       "       iliac_flare_ratio  oblique_pelvic_inlet_length  iliac_flare_angle  \\\n",
       "0               0.462819                     9.883806          97.533181   \n",
       "1               0.442354                     9.562262          95.714405   \n",
       "2               0.509312                    11.313022         101.418560   \n",
       "3               0.492587                     9.897451          95.473293   \n",
       "4               0.462251                    10.305746          95.952879   \n",
       "...                  ...                          ...                ...   \n",
       "31110           0.445854                    11.702063          95.353305   \n",
       "31111           0.458840                    11.896728          87.564668   \n",
       "31112           0.502939                    11.893171          88.868037   \n",
       "31113           0.454842                    14.130428          90.205445   \n",
       "31114           0.465821                    12.728717          85.308035   \n",
       "\n",
       "       subpubic_angle  ear_left2ear_right  trochanter_left2trochanter_right  \\\n",
       "0          116.365190           14.112233                         30.091232   \n",
       "1          117.491690           13.903518                         30.813208   \n",
       "2          139.787157           13.029582                         31.173284   \n",
       "3          116.159185           14.480994                         29.777444   \n",
       "4          130.647738           14.042468                         28.838122   \n",
       "...               ...                 ...                               ...   \n",
       "31110      126.858511           13.402493                         30.995640   \n",
       "31111      120.568969           13.781823                         31.116022   \n",
       "31112      143.403514           12.551823                         27.342555   \n",
       "31113      157.984881           12.309345                         27.698720   \n",
       "31114      124.424809           13.271879                         30.125650   \n",
       "\n",
       "       shoulder_width  arm_devide_torso  bi_acetabular_width  \\\n",
       "0           36.696399          1.067168            20.324819   \n",
       "1           39.203992          0.933102            20.756184   \n",
       "2           35.909407          0.991270            21.701987   \n",
       "3           33.219853          1.014700            19.784228   \n",
       "4           39.066233          0.978756            19.821637   \n",
       "...               ...               ...                  ...   \n",
       "31110       41.118337          1.025959            20.686542   \n",
       "31111       38.579135          0.977047            20.729378   \n",
       "31112       35.846661          0.959371            19.152575   \n",
       "31113       36.369916          0.906038            19.990121   \n",
       "31114       39.486231          0.975894            20.408280   \n",
       "\n",
       "       iliac_isthmus_breadth  acetabular_diameter  acetabular_inclination  \\\n",
       "0                   4.164529             6.141783               60.747735   \n",
       "1                   4.657824             6.035855               59.189839   \n",
       "2                   3.718747             5.531657               59.017487   \n",
       "3                   4.238295             5.834113               58.027399   \n",
       "4                   4.438966             5.731393               57.335593   \n",
       "...                      ...                  ...                     ...   \n",
       "31110               4.014342             6.149698               60.427962   \n",
       "31111               4.381059             6.504511               59.444884   \n",
       "31112               3.211435             4.959191               67.937764   \n",
       "31113               3.935749             5.201943               60.779334   \n",
       "31114               4.298543             6.171569               59.544963   \n",
       "\n",
       "       arm_length  leg_length  pelvic_inlet_area  \n",
       "0       57.673326   89.233162         108.190429  \n",
       "1       54.743642   85.065164         102.660623  \n",
       "2       50.766740   78.752591         139.989191  \n",
       "3       58.533827   86.154050         110.520822  \n",
       "4       56.305299   84.286956         109.946859  \n",
       "...           ...         ...                ...  \n",
       "31110   61.668708   95.185324         134.603261  \n",
       "31111   59.744710   90.479553         132.776803  \n",
       "31112   52.910551   80.626829         126.477899  \n",
       "31113   55.123637   84.255142         150.788401  \n",
       "31114   62.084987   93.878589         138.796708  \n",
       "\n",
       "[31115 rows x 20 columns]"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# select the phenotypes for h2g\n",
    "hip_pheno_gwas_cm = hip_pheno_flt_eid_flt_cm.iloc[:, 8:]\n",
    "\n",
    "ids = hip_pheno_flt_eid_flt_cm['eid'].tolist()\n",
    "\n",
    "hip_pheno_gwas_cm['FID'] = ids\n",
    "hip_pheno_gwas_cm['IID'] = ids\n",
    "\n",
    "# put FID and IID to the first two columns\n",
    "cols = hip_pheno_gwas_cm.columns\n",
    "cols = cols[-2:].tolist() + cols[:-2].tolist()\n",
    "hip_pheno_gwas_cm = hip_pheno_gwas_cm[cols]\n",
    "\n",
    "hip_pheno_gwas_cm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>pelvic_height</th>\n",
       "      <th>pelvic_width</th>\n",
       "      <th>pelvic_inlet_width</th>\n",
       "      <th>oblique_pelvic_inlet_length</th>\n",
       "      <th>subpubic_angle</th>\n",
       "      <th>bi_acetabular_width</th>\n",
       "      <th>iliac_isthmus_breadth</th>\n",
       "      <th>acetabular_diameter</th>\n",
       "      <th>pelvic_inlet_area</th>\n",
       "      <th>leg_length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1000951</td>\n",
       "      <td>1000951</td>\n",
       "      <td>22.297293</td>\n",
       "      <td>30.113663</td>\n",
       "      <td>13.937174</td>\n",
       "      <td>9.883806</td>\n",
       "      <td>116.365190</td>\n",
       "      <td>20.324819</td>\n",
       "      <td>4.164529</td>\n",
       "      <td>6.141783</td>\n",
       "      <td>108.190429</td>\n",
       "      <td>89.233162</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1001512</td>\n",
       "      <td>1001512</td>\n",
       "      <td>23.742297</td>\n",
       "      <td>30.901775</td>\n",
       "      <td>13.669524</td>\n",
       "      <td>9.562262</td>\n",
       "      <td>117.491690</td>\n",
       "      <td>20.756184</td>\n",
       "      <td>4.657824</td>\n",
       "      <td>6.035855</td>\n",
       "      <td>102.660623</td>\n",
       "      <td>85.065164</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1001867</td>\n",
       "      <td>1001867</td>\n",
       "      <td>22.226944</td>\n",
       "      <td>30.934457</td>\n",
       "      <td>15.755275</td>\n",
       "      <td>11.313022</td>\n",
       "      <td>139.787157</td>\n",
       "      <td>21.701987</td>\n",
       "      <td>3.718747</td>\n",
       "      <td>5.531657</td>\n",
       "      <td>139.989191</td>\n",
       "      <td>78.752591</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1001921</td>\n",
       "      <td>1001921</td>\n",
       "      <td>22.861258</td>\n",
       "      <td>28.863430</td>\n",
       "      <td>14.217750</td>\n",
       "      <td>9.897451</td>\n",
       "      <td>116.159185</td>\n",
       "      <td>19.784228</td>\n",
       "      <td>4.238295</td>\n",
       "      <td>5.834113</td>\n",
       "      <td>110.520822</td>\n",
       "      <td>86.154050</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1001950</td>\n",
       "      <td>1001950</td>\n",
       "      <td>23.080688</td>\n",
       "      <td>29.385649</td>\n",
       "      <td>13.583558</td>\n",
       "      <td>10.305746</td>\n",
       "      <td>130.647738</td>\n",
       "      <td>19.821637</td>\n",
       "      <td>4.438966</td>\n",
       "      <td>5.731393</td>\n",
       "      <td>109.946859</td>\n",
       "      <td>84.286956</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31110</th>\n",
       "      <td>6021112</td>\n",
       "      <td>6021112</td>\n",
       "      <td>25.247090</td>\n",
       "      <td>32.848104</td>\n",
       "      <td>14.645469</td>\n",
       "      <td>11.702063</td>\n",
       "      <td>126.858511</td>\n",
       "      <td>20.686542</td>\n",
       "      <td>4.014342</td>\n",
       "      <td>6.149698</td>\n",
       "      <td>134.603261</td>\n",
       "      <td>95.185324</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31111</th>\n",
       "      <td>6021539</td>\n",
       "      <td>6021539</td>\n",
       "      <td>25.232678</td>\n",
       "      <td>30.970184</td>\n",
       "      <td>14.210350</td>\n",
       "      <td>11.896728</td>\n",
       "      <td>120.568969</td>\n",
       "      <td>20.729378</td>\n",
       "      <td>4.381059</td>\n",
       "      <td>6.504511</td>\n",
       "      <td>132.776803</td>\n",
       "      <td>90.479553</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31112</th>\n",
       "      <td>6022591</td>\n",
       "      <td>6022591</td>\n",
       "      <td>20.946633</td>\n",
       "      <td>26.922261</td>\n",
       "      <td>13.540263</td>\n",
       "      <td>11.893171</td>\n",
       "      <td>143.403514</td>\n",
       "      <td>19.152575</td>\n",
       "      <td>3.211435</td>\n",
       "      <td>4.959191</td>\n",
       "      <td>126.477899</td>\n",
       "      <td>80.626829</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31113</th>\n",
       "      <td>6023431</td>\n",
       "      <td>6023431</td>\n",
       "      <td>22.787944</td>\n",
       "      <td>29.871832</td>\n",
       "      <td>13.586974</td>\n",
       "      <td>14.130428</td>\n",
       "      <td>157.984881</td>\n",
       "      <td>19.990121</td>\n",
       "      <td>3.935749</td>\n",
       "      <td>5.201943</td>\n",
       "      <td>150.788401</td>\n",
       "      <td>84.255142</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31114</th>\n",
       "      <td>6024029</td>\n",
       "      <td>6024029</td>\n",
       "      <td>24.323043</td>\n",
       "      <td>29.804727</td>\n",
       "      <td>13.883682</td>\n",
       "      <td>12.728717</td>\n",
       "      <td>124.424809</td>\n",
       "      <td>20.408280</td>\n",
       "      <td>4.298543</td>\n",
       "      <td>6.171569</td>\n",
       "      <td>138.796708</td>\n",
       "      <td>93.878589</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>31115 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           FID      IID  pelvic_height  pelvic_width  pelvic_inlet_width  \\\n",
       "0      1000951  1000951      22.297293     30.113663           13.937174   \n",
       "1      1001512  1001512      23.742297     30.901775           13.669524   \n",
       "2      1001867  1001867      22.226944     30.934457           15.755275   \n",
       "3      1001921  1001921      22.861258     28.863430           14.217750   \n",
       "4      1001950  1001950      23.080688     29.385649           13.583558   \n",
       "...        ...      ...            ...           ...                 ...   \n",
       "31110  6021112  6021112      25.247090     32.848104           14.645469   \n",
       "31111  6021539  6021539      25.232678     30.970184           14.210350   \n",
       "31112  6022591  6022591      20.946633     26.922261           13.540263   \n",
       "31113  6023431  6023431      22.787944     29.871832           13.586974   \n",
       "31114  6024029  6024029      24.323043     29.804727           13.883682   \n",
       "\n",
       "       oblique_pelvic_inlet_length  subpubic_angle  bi_acetabular_width  \\\n",
       "0                         9.883806      116.365190            20.324819   \n",
       "1                         9.562262      117.491690            20.756184   \n",
       "2                        11.313022      139.787157            21.701987   \n",
       "3                         9.897451      116.159185            19.784228   \n",
       "4                        10.305746      130.647738            19.821637   \n",
       "...                            ...             ...                  ...   \n",
       "31110                    11.702063      126.858511            20.686542   \n",
       "31111                    11.896728      120.568969            20.729378   \n",
       "31112                    11.893171      143.403514            19.152575   \n",
       "31113                    14.130428      157.984881            19.990121   \n",
       "31114                    12.728717      124.424809            20.408280   \n",
       "\n",
       "       iliac_isthmus_breadth  acetabular_diameter  pelvic_inlet_area  \\\n",
       "0                   4.164529             6.141783         108.190429   \n",
       "1                   4.657824             6.035855         102.660623   \n",
       "2                   3.718747             5.531657         139.989191   \n",
       "3                   4.238295             5.834113         110.520822   \n",
       "4                   4.438966             5.731393         109.946859   \n",
       "...                      ...                  ...                ...   \n",
       "31110               4.014342             6.149698         134.603261   \n",
       "31111               4.381059             6.504511         132.776803   \n",
       "31112               3.211435             4.959191         126.477899   \n",
       "31113               3.935749             5.201943         150.788401   \n",
       "31114               4.298543             6.171569         138.796708   \n",
       "\n",
       "       leg_length  \n",
       "0       89.233162  \n",
       "1       85.065164  \n",
       "2       78.752591  \n",
       "3       86.154050  \n",
       "4       84.286956  \n",
       "...           ...  \n",
       "31110   95.185324  \n",
       "31111   90.479553  \n",
       "31112   80.626829  \n",
       "31113   84.255142  \n",
       "31114   93.878589  \n",
       "\n",
       "[31115 rows x 12 columns]"
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hip_pheno_gwas_cm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "hip_pheno_flt_eid_flt_cm[['eid', 'eid']].to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_eids_20240104.txt', index=False, header=False, sep=' ')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "hip_pheno_gwas_cm.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_cm_20240104.txt', index=False, sep = \" \")\n",
    "hip_pheno_gwas_cm.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_cm_no_header_20240104.txt', index=False, header=False, sep = \" \")"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Run gwas for height"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>eid</th>\n",
       "      <th>sex</th>\n",
       "      <th>age</th>\n",
       "      <th>standing_height</th>\n",
       "      <th>bmi</th>\n",
       "      <th>weight</th>\n",
       "      <th>household_income</th>\n",
       "      <th>pelvis_bmd</th>\n",
       "      <th>patient_birth_weight</th>\n",
       "      <th>hip_circumference</th>\n",
       "      <th>...</th>\n",
       "      <th>female_baby</th>\n",
       "      <th>n_pregnancy</th>\n",
       "      <th>n_weeks_gestation</th>\n",
       "      <th>baby_birth_weight</th>\n",
       "      <th>postnatal_stay_duration</th>\n",
       "      <th>ethnic_group</th>\n",
       "      <th>keep_allna_incontinence</th>\n",
       "      <th>remove_allna_incontinence</th>\n",
       "      <th>sitting_height</th>\n",
       "      <th>leg_length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1000011</td>\n",
       "      <td>0.0</td>\n",
       "      <td>50.0</td>\n",
       "      <td>167.0</td>\n",
       "      <td>25.1712</td>\n",
       "      <td>70.2</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.74</td>\n",
       "      <td>103.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>white</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>89.0</td>\n",
       "      <td>78.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1000026</td>\n",
       "      <td>1.0</td>\n",
       "      <td>59.0</td>\n",
       "      <td>174.0</td>\n",
       "      <td>24.6400</td>\n",
       "      <td>74.6</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>99.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>white</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>91.0</td>\n",
       "      <td>83.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1000032</td>\n",
       "      <td>0.0</td>\n",
       "      <td>50.0</td>\n",
       "      <td>170.0</td>\n",
       "      <td>32.2837</td>\n",
       "      <td>93.3</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.65</td>\n",
       "      <td>110.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>white</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>86.0</td>\n",
       "      <td>84.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1000044</td>\n",
       "      <td>0.0</td>\n",
       "      <td>53.0</td>\n",
       "      <td>154.0</td>\n",
       "      <td>32.3832</td>\n",
       "      <td>76.8</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>116.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>white</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>78.0</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1000058</td>\n",
       "      <td>0.0</td>\n",
       "      <td>50.0</td>\n",
       "      <td>155.0</td>\n",
       "      <td>32.3829</td>\n",
       "      <td>77.8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>111.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>81.0</td>\n",
       "      <td>74.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502485</th>\n",
       "      <td>6024895</td>\n",
       "      <td>1.0</td>\n",
       "      <td>55.0</td>\n",
       "      <td>164.4</td>\n",
       "      <td>16.9828</td>\n",
       "      <td>45.9</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.09</td>\n",
       "      <td>84.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>white</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>83.4</td>\n",
       "      <td>81.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502486</th>\n",
       "      <td>6024901</td>\n",
       "      <td>1.0</td>\n",
       "      <td>59.0</td>\n",
       "      <td>177.0</td>\n",
       "      <td>29.0147</td>\n",
       "      <td>90.9</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.54</td>\n",
       "      <td>104.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>white</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>94.0</td>\n",
       "      <td>83.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502487</th>\n",
       "      <td>6024913</td>\n",
       "      <td>1.0</td>\n",
       "      <td>64.0</td>\n",
       "      <td>167.0</td>\n",
       "      <td>26.0676</td>\n",
       "      <td>72.7</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>97.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>white</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>102.0</td>\n",
       "      <td>65.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502488</th>\n",
       "      <td>6024922</td>\n",
       "      <td>1.0</td>\n",
       "      <td>48.0</td>\n",
       "      <td>180.0</td>\n",
       "      <td>30.3395</td>\n",
       "      <td>98.3</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.77</td>\n",
       "      <td>109.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>white</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>95.0</td>\n",
       "      <td>85.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502489</th>\n",
       "      <td>2108075</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.09882</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>502490 rows × 120 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            eid  sex   age  standing_height      bmi  weight  \\\n",
       "0       1000011  0.0  50.0            167.0  25.1712    70.2   \n",
       "1       1000026  1.0  59.0            174.0  24.6400    74.6   \n",
       "2       1000032  0.0  50.0            170.0  32.2837    93.3   \n",
       "3       1000044  0.0  53.0            154.0  32.3832    76.8   \n",
       "4       1000058  0.0  50.0            155.0  32.3829    77.8   \n",
       "...         ...  ...   ...              ...      ...     ...   \n",
       "502485  6024895  1.0  55.0            164.4  16.9828    45.9   \n",
       "502486  6024901  1.0  59.0            177.0  29.0147    90.9   \n",
       "502487  6024913  1.0  64.0            167.0  26.0676    72.7   \n",
       "502488  6024922  1.0  48.0            180.0  30.3395    98.3   \n",
       "502489  2108075  NaN   NaN              NaN      NaN     NaN   \n",
       "\n",
       "        household_income  pelvis_bmd  patient_birth_weight  hip_circumference  \\\n",
       "0                    3.0         NaN                  3.74              103.0   \n",
       "1                    3.0         NaN                   NaN               99.0   \n",
       "2                    4.0         NaN                  4.65              110.0   \n",
       "3                    3.0         NaN                   NaN              116.0   \n",
       "4                    NaN         NaN                   NaN              111.0   \n",
       "...                  ...         ...                   ...                ...   \n",
       "502485               4.0         NaN                  3.09               84.0   \n",
       "502486               3.0         NaN                  4.54              104.0   \n",
       "502487               1.0         NaN                   NaN               97.0   \n",
       "502488               4.0         NaN                  3.77              109.0   \n",
       "502489               NaN     1.09882                   NaN                NaN   \n",
       "\n",
       "        ...  female_baby  n_pregnancy  n_weeks_gestation  baby_birth_weight  \\\n",
       "0       ...          NaN          NaN                NaN                NaN   \n",
       "1       ...          NaN          NaN                NaN                NaN   \n",
       "2       ...          NaN          NaN                NaN                NaN   \n",
       "3       ...          NaN          NaN                NaN                NaN   \n",
       "4       ...          NaN          NaN                NaN                NaN   \n",
       "...     ...          ...          ...                ...                ...   \n",
       "502485  ...          NaN          NaN                NaN                NaN   \n",
       "502486  ...          NaN          NaN                NaN                NaN   \n",
       "502487  ...          NaN          NaN                NaN                NaN   \n",
       "502488  ...          NaN          NaN                NaN                NaN   \n",
       "502489  ...          NaN          NaN                NaN                NaN   \n",
       "\n",
       "        postnatal_stay_duration  ethnic_group  keep_allna_incontinence  \\\n",
       "0                           NaN         white                        0   \n",
       "1                           NaN         white                        1   \n",
       "2                           NaN         white                        0   \n",
       "3                           NaN         white                        1   \n",
       "4                           NaN           NaN                        0   \n",
       "...                         ...           ...                      ...   \n",
       "502485                      NaN         white                        0   \n",
       "502486                      NaN         white                        0   \n",
       "502487                      NaN         white                        0   \n",
       "502488                      NaN         white                        0   \n",
       "502489                      NaN           NaN                        0   \n",
       "\n",
       "        remove_allna_incontinence  sitting_height  leg_length  \n",
       "0                             NaN            89.0        78.0  \n",
       "1                             1.0            91.0        83.0  \n",
       "2                             0.0            86.0        84.0  \n",
       "3                             1.0            78.0        76.0  \n",
       "4                             NaN            81.0        74.0  \n",
       "...                           ...             ...         ...  \n",
       "502485                        0.0            83.4        81.0  \n",
       "502486                        NaN            94.0        83.0  \n",
       "502487                        0.0           102.0        65.0  \n",
       "502488                        NaN            95.0        85.0  \n",
       "502489                        NaN             NaN         NaN  \n",
       "\n",
       "[502490 rows x 120 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fid_info = pd.read_csv('../UKB_xray_image_info/fids/fid_disease/fid_info_400k_20231028.csv'); fid_info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "both_height = fid_info[['eid', 'standing_height']].dropna()\n",
    "male_height = fid_info[fid_info['sex'] == 1.0][['eid', 'standing_height']].dropna()\n",
    "female_height = fid_info[fid_info['sex'] == 0.0][['eid', 'standing_height']].dropna()\n",
    "\n",
    "both_height_eid = both_height['eid'].tolist()\n",
    "male_height_eid = male_height['eid'].tolist()\n",
    "female_height_eid = female_height['eid'].tolist()\n",
    "\n",
    "gwas_standing_height_both = pd.DataFrame({\"FID\": both_height_eid, \"IID\": both_height_eid, \"standing_height\": both_height['standing_height'].tolist()})\n",
    "gwas_standing_height_male = pd.DataFrame({\"FID\": male_height_eid, \"IID\": male_height_eid, \"standing_height\": male_height['standing_height'].tolist()})\n",
    "gwas_standing_height_female = pd.DataFrame({\"FID\": female_height_eid, \"IID\": female_height_eid, \"standing_height\": female_height['standing_height'].tolist()})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>standing_height</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1000011</td>\n",
       "      <td>1000011</td>\n",
       "      <td>167.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1000026</td>\n",
       "      <td>1000026</td>\n",
       "      <td>174.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1000032</td>\n",
       "      <td>1000032</td>\n",
       "      <td>170.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1000044</td>\n",
       "      <td>1000044</td>\n",
       "      <td>154.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1000058</td>\n",
       "      <td>1000058</td>\n",
       "      <td>155.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499923</th>\n",
       "      <td>6024884</td>\n",
       "      <td>6024884</td>\n",
       "      <td>179.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499924</th>\n",
       "      <td>6024895</td>\n",
       "      <td>6024895</td>\n",
       "      <td>164.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499925</th>\n",
       "      <td>6024901</td>\n",
       "      <td>6024901</td>\n",
       "      <td>177.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499926</th>\n",
       "      <td>6024913</td>\n",
       "      <td>6024913</td>\n",
       "      <td>167.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499927</th>\n",
       "      <td>6024922</td>\n",
       "      <td>6024922</td>\n",
       "      <td>180.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>499928 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            FID      IID  standing_height\n",
       "0       1000011  1000011            167.0\n",
       "1       1000026  1000026            174.0\n",
       "2       1000032  1000032            170.0\n",
       "3       1000044  1000044            154.0\n",
       "4       1000058  1000058            155.0\n",
       "...         ...      ...              ...\n",
       "499923  6024884  6024884            179.0\n",
       "499924  6024895  6024895            164.4\n",
       "499925  6024901  6024901            177.0\n",
       "499926  6024913  6024913            167.0\n",
       "499927  6024922  6024922            180.0\n",
       "\n",
       "[499928 rows x 3 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gwas_standing_height_both"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 156,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "both_height_shape: (499928, 3)\n",
      "male_height_shape: (227722, 3)\n",
      "female_height_shape: (272206, 3)\n"
     ]
    }
   ],
   "source": [
    "print(f'both_height_shape: {gwas_standing_height_both.shape}')\n",
    "print(f'male_height_shape: {gwas_standing_height_male.shape}')\n",
    "print(f'female_height_shape: {gwas_standing_height_female.shape}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 157,
   "metadata": {},
   "outputs": [],
   "source": [
    "gwas_standing_height_both.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/gwas_standing_height_both_231107.txt', index=False, sep = \" \")\n",
    "gwas_standing_height_both.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/gwas_standing_height_both_no_header_231107.txt', index=False, header=False, sep = \" \")\n",
    "\n",
    "gwas_standing_height_male.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/gwas_standing_height_male_231107.txt', index=False, sep = \" \")\n",
    "gwas_standing_height_male.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/gwas_standing_height_male_no_header_231107.txt', index=False, header=False, sep = \" \")\n",
    "\n",
    "gwas_standing_height_female.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/gwas_standing_height_female_231107.txt', index=False, sep = \" \")\n",
    "gwas_standing_height_female.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/gwas_standing_height_female_no_header_231107.txt', index=False, header=False, sep = \" \")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# removed imaged individuals\n",
    "\n",
    "# get all fid and iid for imaged individuals\n",
    "iid_img = pd.read_csv('key_results/hip_pheno_23_cm.csv')['eid'].tolist()\n",
    "iid_400k = pd.read_csv(\"UKB_Imaging_Genetics/merged_sub_maf0.01_biallel_bbf_400k_white_british.fam\", sep = \" \", header=None)[0].tolist()\n",
    "\n",
    "# both\n",
    "gwas_standing_height_both = pd.read_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/gwas_standing_height_both_231107.txt', sep = \" \")\n",
    "gwas_standing_height_both = gwas_standing_height_both[gwas_standing_height_both['IID'].isin(iid_400k)]\n",
    "gwas_standing_height_both_train = gwas_standing_height_both[~gwas_standing_height_both['IID'].isin(iid_img)]\n",
    "gwas_standing_height_both_train.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/gwas_standing_height_both_train_231128.txt', index=False, sep = \" \")\n",
    "gwas_standing_height_both_train.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/gwas_standing_height_both_train_no_header_231128.txt', index=False, header=False, sep = \" \")\n",
    "\n",
    "# female\n",
    "gwas_standing_height_female = pd.read_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/gwas_standing_height_female_231107.txt', sep = \" \")\n",
    "gwas_standing_height_female = gwas_standing_height_female[gwas_standing_height_female['IID'].isin(iid_400k)]\n",
    "gwas_standing_height_female_train = gwas_standing_height_female[~gwas_standing_height_female['IID'].isin(iid_img)]\n",
    "gwas_standing_height_female_train.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/gwas_standing_height_female_train_231128.txt', index=False, sep = \" \")\n",
    "gwas_standing_height_female_train.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/gwas_standing_height_female_train_no_header_231128.txt', index=False, header=False, sep = \" \")\n",
    "\n",
    "# male\n",
    "gwas_standing_height_male = pd.read_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/gwas_standing_height_male_231107.txt', sep = \" \")\n",
    "gwas_standing_height_male = gwas_standing_height_male[gwas_standing_height_male['IID'].isin(iid_400k)]\n",
    "gwas_standing_height_male_train = gwas_standing_height_male[~gwas_standing_height_male['IID'].isin(iid_img)]\n",
    "gwas_standing_height_male_train.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/gwas_standing_height_male_train_231128.txt', index=False, sep = \" \")\n",
    "gwas_standing_height_male_train.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/gwas_standing_height_male_train_no_header_231128.txt', index=False, header=False, sep = \" \")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "both_train_shape: (369296, 3)\n",
      "female_train_shape: (200661, 3)\n",
      "male_train_shape: (168635, 3)\n"
     ]
    }
   ],
   "source": [
    "print(f'both_train_shape: {gwas_standing_height_both_train.shape}')\n",
    "\n",
    "print(f'female_train_shape: {gwas_standing_height_female_train.shape}')\n",
    "\n",
    "print(f'male_train_shape: {gwas_standing_height_male_train.shape}')"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### add sex new phenotypes (head or shoulder / pelvic inlet size)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>pelvic_height</th>\n",
       "      <th>pelvic_width</th>\n",
       "      <th>pelvic_inlet_width</th>\n",
       "      <th>iliac_flare_ratio</th>\n",
       "      <th>oblique_pelvic_inlet_length</th>\n",
       "      <th>iliac_flare_angle</th>\n",
       "      <th>subpubic_angle</th>\n",
       "      <th>ear_left2ear_right</th>\n",
       "      <th>trochanter_left2trochanter_right</th>\n",
       "      <th>shoulder_width</th>\n",
       "      <th>arm_devide_torso</th>\n",
       "      <th>bi_acetabular_width</th>\n",
       "      <th>iliac_isthmus_breadth</th>\n",
       "      <th>acetabular_diameter</th>\n",
       "      <th>acetabular_inclination</th>\n",
       "      <th>arm_length</th>\n",
       "      <th>leg_length</th>\n",
       "      <th>pelvic_inlet_area</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1000951</td>\n",
       "      <td>1000951</td>\n",
       "      <td>22.297293</td>\n",
       "      <td>30.113663</td>\n",
       "      <td>13.937174</td>\n",
       "      <td>0.462819</td>\n",
       "      <td>9.883806</td>\n",
       "      <td>97.533181</td>\n",
       "      <td>116.365190</td>\n",
       "      <td>14.112233</td>\n",
       "      <td>30.091232</td>\n",
       "      <td>36.696399</td>\n",
       "      <td>1.067168</td>\n",
       "      <td>20.324819</td>\n",
       "      <td>4.164529</td>\n",
       "      <td>6.141783</td>\n",
       "      <td>60.747735</td>\n",
       "      <td>57.673326</td>\n",
       "      <td>89.233162</td>\n",
       "      <td>108.190429</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1001329</td>\n",
       "      <td>1001329</td>\n",
       "      <td>25.363221</td>\n",
       "      <td>33.085310</td>\n",
       "      <td>13.943964</td>\n",
       "      <td>0.421455</td>\n",
       "      <td>9.796084</td>\n",
       "      <td>99.812799</td>\n",
       "      <td>108.368022</td>\n",
       "      <td>14.487676</td>\n",
       "      <td>31.474656</td>\n",
       "      <td>43.357959</td>\n",
       "      <td>1.023811</td>\n",
       "      <td>21.085789</td>\n",
       "      <td>5.090414</td>\n",
       "      <td>6.729616</td>\n",
       "      <td>56.943430</td>\n",
       "      <td>64.205477</td>\n",
       "      <td>95.795086</td>\n",
       "      <td>107.282431</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1001512</td>\n",
       "      <td>1001512</td>\n",
       "      <td>23.742297</td>\n",
       "      <td>30.901775</td>\n",
       "      <td>13.669524</td>\n",
       "      <td>0.442354</td>\n",
       "      <td>9.562262</td>\n",
       "      <td>95.714405</td>\n",
       "      <td>117.491690</td>\n",
       "      <td>13.903518</td>\n",
       "      <td>30.813208</td>\n",
       "      <td>39.203992</td>\n",
       "      <td>0.933102</td>\n",
       "      <td>20.756184</td>\n",
       "      <td>4.657824</td>\n",
       "      <td>6.035855</td>\n",
       "      <td>59.189839</td>\n",
       "      <td>54.743642</td>\n",
       "      <td>85.065164</td>\n",
       "      <td>102.660623</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1001867</td>\n",
       "      <td>1001867</td>\n",
       "      <td>22.226944</td>\n",
       "      <td>30.934457</td>\n",
       "      <td>15.755275</td>\n",
       "      <td>0.509312</td>\n",
       "      <td>11.313022</td>\n",
       "      <td>101.418560</td>\n",
       "      <td>139.787157</td>\n",
       "      <td>13.029582</td>\n",
       "      <td>31.173284</td>\n",
       "      <td>35.909407</td>\n",
       "      <td>0.991270</td>\n",
       "      <td>21.701987</td>\n",
       "      <td>3.718747</td>\n",
       "      <td>5.531657</td>\n",
       "      <td>59.017487</td>\n",
       "      <td>50.766740</td>\n",
       "      <td>78.752591</td>\n",
       "      <td>139.989191</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1001921</td>\n",
       "      <td>1001921</td>\n",
       "      <td>22.861258</td>\n",
       "      <td>28.863430</td>\n",
       "      <td>14.217750</td>\n",
       "      <td>0.492587</td>\n",
       "      <td>9.897451</td>\n",
       "      <td>95.473293</td>\n",
       "      <td>116.159185</td>\n",
       "      <td>14.480994</td>\n",
       "      <td>29.777444</td>\n",
       "      <td>33.219853</td>\n",
       "      <td>1.014700</td>\n",
       "      <td>19.784228</td>\n",
       "      <td>4.238295</td>\n",
       "      <td>5.834113</td>\n",
       "      <td>58.027399</td>\n",
       "      <td>58.533827</td>\n",
       "      <td>86.154050</td>\n",
       "      <td>110.520822</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32040</th>\n",
       "      <td>6021112</td>\n",
       "      <td>6021112</td>\n",
       "      <td>25.247090</td>\n",
       "      <td>32.848104</td>\n",
       "      <td>14.645469</td>\n",
       "      <td>0.445854</td>\n",
       "      <td>11.702063</td>\n",
       "      <td>95.353305</td>\n",
       "      <td>126.858511</td>\n",
       "      <td>13.402493</td>\n",
       "      <td>30.995640</td>\n",
       "      <td>41.118337</td>\n",
       "      <td>1.025959</td>\n",
       "      <td>20.686542</td>\n",
       "      <td>4.014342</td>\n",
       "      <td>6.149698</td>\n",
       "      <td>60.427962</td>\n",
       "      <td>61.668708</td>\n",
       "      <td>95.185324</td>\n",
       "      <td>134.603261</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32041</th>\n",
       "      <td>6021539</td>\n",
       "      <td>6021539</td>\n",
       "      <td>25.232678</td>\n",
       "      <td>30.970184</td>\n",
       "      <td>14.210350</td>\n",
       "      <td>0.458840</td>\n",
       "      <td>11.896728</td>\n",
       "      <td>87.564668</td>\n",
       "      <td>120.568969</td>\n",
       "      <td>13.781823</td>\n",
       "      <td>31.116022</td>\n",
       "      <td>38.579135</td>\n",
       "      <td>0.977047</td>\n",
       "      <td>20.729378</td>\n",
       "      <td>4.381059</td>\n",
       "      <td>6.504511</td>\n",
       "      <td>59.444884</td>\n",
       "      <td>59.744710</td>\n",
       "      <td>90.479553</td>\n",
       "      <td>132.776803</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32042</th>\n",
       "      <td>6022591</td>\n",
       "      <td>6022591</td>\n",
       "      <td>20.946633</td>\n",
       "      <td>26.922261</td>\n",
       "      <td>13.540263</td>\n",
       "      <td>0.502939</td>\n",
       "      <td>11.893171</td>\n",
       "      <td>88.868037</td>\n",
       "      <td>143.403514</td>\n",
       "      <td>12.551823</td>\n",
       "      <td>27.342555</td>\n",
       "      <td>35.846661</td>\n",
       "      <td>0.959371</td>\n",
       "      <td>19.152575</td>\n",
       "      <td>3.211435</td>\n",
       "      <td>4.959191</td>\n",
       "      <td>67.937764</td>\n",
       "      <td>52.910551</td>\n",
       "      <td>80.626829</td>\n",
       "      <td>126.477899</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32043</th>\n",
       "      <td>6023431</td>\n",
       "      <td>6023431</td>\n",
       "      <td>22.787944</td>\n",
       "      <td>29.871832</td>\n",
       "      <td>13.586974</td>\n",
       "      <td>0.454842</td>\n",
       "      <td>14.130428</td>\n",
       "      <td>90.205445</td>\n",
       "      <td>157.984881</td>\n",
       "      <td>12.309345</td>\n",
       "      <td>27.698720</td>\n",
       "      <td>36.369916</td>\n",
       "      <td>0.906038</td>\n",
       "      <td>19.990121</td>\n",
       "      <td>3.935749</td>\n",
       "      <td>5.201943</td>\n",
       "      <td>60.779334</td>\n",
       "      <td>55.123637</td>\n",
       "      <td>84.255142</td>\n",
       "      <td>150.788401</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32044</th>\n",
       "      <td>6024029</td>\n",
       "      <td>6024029</td>\n",
       "      <td>24.323043</td>\n",
       "      <td>29.804727</td>\n",
       "      <td>13.883682</td>\n",
       "      <td>0.465821</td>\n",
       "      <td>12.728717</td>\n",
       "      <td>85.308035</td>\n",
       "      <td>124.424809</td>\n",
       "      <td>13.271879</td>\n",
       "      <td>30.125650</td>\n",
       "      <td>39.486231</td>\n",
       "      <td>0.975894</td>\n",
       "      <td>20.408280</td>\n",
       "      <td>4.298543</td>\n",
       "      <td>6.171569</td>\n",
       "      <td>59.544963</td>\n",
       "      <td>62.084987</td>\n",
       "      <td>93.878589</td>\n",
       "      <td>138.796708</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>32045 rows × 20 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           FID      IID  pelvic_height  pelvic_width  pelvic_inlet_width  \\\n",
       "0      1000951  1000951      22.297293     30.113663           13.937174   \n",
       "1      1001329  1001329      25.363221     33.085310           13.943964   \n",
       "2      1001512  1001512      23.742297     30.901775           13.669524   \n",
       "3      1001867  1001867      22.226944     30.934457           15.755275   \n",
       "4      1001921  1001921      22.861258     28.863430           14.217750   \n",
       "...        ...      ...            ...           ...                 ...   \n",
       "32040  6021112  6021112      25.247090     32.848104           14.645469   \n",
       "32041  6021539  6021539      25.232678     30.970184           14.210350   \n",
       "32042  6022591  6022591      20.946633     26.922261           13.540263   \n",
       "32043  6023431  6023431      22.787944     29.871832           13.586974   \n",
       "32044  6024029  6024029      24.323043     29.804727           13.883682   \n",
       "\n",
       "       iliac_flare_ratio  oblique_pelvic_inlet_length  iliac_flare_angle  \\\n",
       "0               0.462819                     9.883806          97.533181   \n",
       "1               0.421455                     9.796084          99.812799   \n",
       "2               0.442354                     9.562262          95.714405   \n",
       "3               0.509312                    11.313022         101.418560   \n",
       "4               0.492587                     9.897451          95.473293   \n",
       "...                  ...                          ...                ...   \n",
       "32040           0.445854                    11.702063          95.353305   \n",
       "32041           0.458840                    11.896728          87.564668   \n",
       "32042           0.502939                    11.893171          88.868037   \n",
       "32043           0.454842                    14.130428          90.205445   \n",
       "32044           0.465821                    12.728717          85.308035   \n",
       "\n",
       "       subpubic_angle  ear_left2ear_right  trochanter_left2trochanter_right  \\\n",
       "0          116.365190           14.112233                         30.091232   \n",
       "1          108.368022           14.487676                         31.474656   \n",
       "2          117.491690           13.903518                         30.813208   \n",
       "3          139.787157           13.029582                         31.173284   \n",
       "4          116.159185           14.480994                         29.777444   \n",
       "...               ...                 ...                               ...   \n",
       "32040      126.858511           13.402493                         30.995640   \n",
       "32041      120.568969           13.781823                         31.116022   \n",
       "32042      143.403514           12.551823                         27.342555   \n",
       "32043      157.984881           12.309345                         27.698720   \n",
       "32044      124.424809           13.271879                         30.125650   \n",
       "\n",
       "       shoulder_width  arm_devide_torso  bi_acetabular_width  \\\n",
       "0           36.696399          1.067168            20.324819   \n",
       "1           43.357959          1.023811            21.085789   \n",
       "2           39.203992          0.933102            20.756184   \n",
       "3           35.909407          0.991270            21.701987   \n",
       "4           33.219853          1.014700            19.784228   \n",
       "...               ...               ...                  ...   \n",
       "32040       41.118337          1.025959            20.686542   \n",
       "32041       38.579135          0.977047            20.729378   \n",
       "32042       35.846661          0.959371            19.152575   \n",
       "32043       36.369916          0.906038            19.990121   \n",
       "32044       39.486231          0.975894            20.408280   \n",
       "\n",
       "       iliac_isthmus_breadth  acetabular_diameter  acetabular_inclination  \\\n",
       "0                   4.164529             6.141783               60.747735   \n",
       "1                   5.090414             6.729616               56.943430   \n",
       "2                   4.657824             6.035855               59.189839   \n",
       "3                   3.718747             5.531657               59.017487   \n",
       "4                   4.238295             5.834113               58.027399   \n",
       "...                      ...                  ...                     ...   \n",
       "32040               4.014342             6.149698               60.427962   \n",
       "32041               4.381059             6.504511               59.444884   \n",
       "32042               3.211435             4.959191               67.937764   \n",
       "32043               3.935749             5.201943               60.779334   \n",
       "32044               4.298543             6.171569               59.544963   \n",
       "\n",
       "       arm_length  leg_length  pelvic_inlet_area  \n",
       "0       57.673326   89.233162         108.190429  \n",
       "1       64.205477   95.795086         107.282431  \n",
       "2       54.743642   85.065164         102.660623  \n",
       "3       50.766740   78.752591         139.989191  \n",
       "4       58.533827   86.154050         110.520822  \n",
       "...           ...         ...                ...  \n",
       "32040   61.668708   95.185324         134.603261  \n",
       "32041   59.744710   90.479553         132.776803  \n",
       "32042   52.910551   80.626829         126.477899  \n",
       "32043   55.123637   84.255142         150.788401  \n",
       "32044   62.084987   93.878589         138.796708  \n",
       "\n",
       "[32045 rows x 20 columns]"
      ]
     },
     "execution_count": 119,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hip_pheno_gwas_cm = pd.read_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_cm_231109.txt', sep = \" \"); hip_pheno_gwas_cm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {},
   "outputs": [],
   "source": [
    "# 2 for head or shoulder / oblique pelvic inlet width\n",
    "hip_pheno_gwas_cm['head_divide_inlet_width'] = hip_pheno_gwas_cm['ear_left2ear_right'] / hip_pheno_gwas_cm['pelvic_inlet_width']\n",
    "hip_pheno_gwas_cm['shoulder_divide_inlet_width'] = hip_pheno_gwas_cm['shoulder_width'] / hip_pheno_gwas_cm['pelvic_inlet_width']\n",
    "\n",
    "# 2 for head or shoulder / oblique pelvic inlet length\n",
    "hip_pheno_gwas_cm['head_divide_oblique_inlet_length'] = hip_pheno_gwas_cm['ear_left2ear_right'] / hip_pheno_gwas_cm['oblique_pelvic_inlet_length']\n",
    "hip_pheno_gwas_cm['shoulder_divide_oblique_inlet_length'] = hip_pheno_gwas_cm['shoulder_width'] / hip_pheno_gwas_cm['oblique_pelvic_inlet_length']\n",
    "\n",
    "\n",
    "# 2 for head or shoulder / pelvic inlet area\n",
    "hip_pheno_gwas_cm['head_area_divide_pelvic_inlet_area'] = ((hip_pheno_gwas_cm['ear_left2ear_right'] / 2)**2 * math.pi) / hip_pheno_gwas_cm['pelvic_inlet_area']\n",
    "hip_pheno_gwas_cm['shoulder_area_divide_pelvic_inlet_area'] = ((hip_pheno_gwas_cm['shoulder_width'] / 2)**2 * math.pi) / hip_pheno_gwas_cm['pelvic_inlet_area']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>image_id</th>\n",
       "      <th>file_name</th>\n",
       "      <th>eid</th>\n",
       "      <th>sex</th>\n",
       "      <th>standing_height</th>\n",
       "      <th>weight</th>\n",
       "      <th>bmi</th>\n",
       "      <th>age</th>\n",
       "      <th>pelvic_height</th>\n",
       "      <th>pelvic_width</th>\n",
       "      <th>...</th>\n",
       "      <th>trochanter_left2trochanter_right</th>\n",
       "      <th>shoulder_width</th>\n",
       "      <th>arm_devide_torso</th>\n",
       "      <th>bi_acetabular_width</th>\n",
       "      <th>iliac_isthmus_breadth</th>\n",
       "      <th>acetabular_diameter</th>\n",
       "      <th>acetabular_inclination</th>\n",
       "      <th>arm_length</th>\n",
       "      <th>leg_length</th>\n",
       "      <th>pelvic_inlet_area</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14366_82</td>\n",
       "      <td>1.2.840.113619.2.110.212038.20190307144208.1.1...</td>\n",
       "      <td>1000951</td>\n",
       "      <td>1.0</td>\n",
       "      <td>167.0</td>\n",
       "      <td>62.9</td>\n",
       "      <td>22.5537</td>\n",
       "      <td>66.0</td>\n",
       "      <td>22.297293</td>\n",
       "      <td>30.113663</td>\n",
       "      <td>...</td>\n",
       "      <td>30.091232</td>\n",
       "      <td>36.696399</td>\n",
       "      <td>1.067168</td>\n",
       "      <td>20.324819</td>\n",
       "      <td>4.164529</td>\n",
       "      <td>6.141783</td>\n",
       "      <td>60.747735</td>\n",
       "      <td>57.673326</td>\n",
       "      <td>89.233162</td>\n",
       "      <td>108.190429</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>09648_82</td>\n",
       "      <td>1.2.840.113619.2.110.212038.20171116132027.1.1...</td>\n",
       "      <td>1001329</td>\n",
       "      <td>1.0</td>\n",
       "      <td>185.0</td>\n",
       "      <td>84.9</td>\n",
       "      <td>24.8064</td>\n",
       "      <td>58.0</td>\n",
       "      <td>25.363221</td>\n",
       "      <td>33.085310</td>\n",
       "      <td>...</td>\n",
       "      <td>31.474656</td>\n",
       "      <td>43.357959</td>\n",
       "      <td>1.023811</td>\n",
       "      <td>21.085789</td>\n",
       "      <td>5.090414</td>\n",
       "      <td>6.729616</td>\n",
       "      <td>56.943430</td>\n",
       "      <td>64.205477</td>\n",
       "      <td>95.795086</td>\n",
       "      <td>107.282431</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>01877_82</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20180720150454.1.1...</td>\n",
       "      <td>1001512</td>\n",
       "      <td>1.0</td>\n",
       "      <td>171.0</td>\n",
       "      <td>81.4</td>\n",
       "      <td>27.8376</td>\n",
       "      <td>57.0</td>\n",
       "      <td>23.742297</td>\n",
       "      <td>30.901775</td>\n",
       "      <td>...</td>\n",
       "      <td>30.813208</td>\n",
       "      <td>39.203992</td>\n",
       "      <td>0.933102</td>\n",
       "      <td>20.756184</td>\n",
       "      <td>4.657824</td>\n",
       "      <td>6.035855</td>\n",
       "      <td>59.189839</td>\n",
       "      <td>54.743642</td>\n",
       "      <td>85.065164</td>\n",
       "      <td>102.660623</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>05431_82</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20190328125146.1.1...</td>\n",
       "      <td>1001867</td>\n",
       "      <td>0.0</td>\n",
       "      <td>155.0</td>\n",
       "      <td>66.1</td>\n",
       "      <td>27.5130</td>\n",
       "      <td>64.0</td>\n",
       "      <td>22.226944</td>\n",
       "      <td>30.934457</td>\n",
       "      <td>...</td>\n",
       "      <td>31.173284</td>\n",
       "      <td>35.909407</td>\n",
       "      <td>0.991270</td>\n",
       "      <td>21.701987</td>\n",
       "      <td>3.718747</td>\n",
       "      <td>5.531657</td>\n",
       "      <td>59.017487</td>\n",
       "      <td>50.766740</td>\n",
       "      <td>78.752591</td>\n",
       "      <td>139.989191</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>10557_82</td>\n",
       "      <td>1.2.840.113619.2.110.212038.20180228105159.1.1...</td>\n",
       "      <td>1001921</td>\n",
       "      <td>1.0</td>\n",
       "      <td>167.0</td>\n",
       "      <td>62.6</td>\n",
       "      <td>22.4461</td>\n",
       "      <td>52.0</td>\n",
       "      <td>22.861258</td>\n",
       "      <td>28.863430</td>\n",
       "      <td>...</td>\n",
       "      <td>29.777444</td>\n",
       "      <td>33.219853</td>\n",
       "      <td>1.014700</td>\n",
       "      <td>19.784228</td>\n",
       "      <td>4.238295</td>\n",
       "      <td>5.834113</td>\n",
       "      <td>58.027399</td>\n",
       "      <td>58.533827</td>\n",
       "      <td>86.154050</td>\n",
       "      <td>110.520822</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32060</th>\n",
       "      <td>10039_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20160923114938.1.1...</td>\n",
       "      <td>6021112</td>\n",
       "      <td>1.0</td>\n",
       "      <td>178.0</td>\n",
       "      <td>87.0</td>\n",
       "      <td>27.4587</td>\n",
       "      <td>64.0</td>\n",
       "      <td>25.247090</td>\n",
       "      <td>32.848104</td>\n",
       "      <td>...</td>\n",
       "      <td>30.995640</td>\n",
       "      <td>41.118337</td>\n",
       "      <td>1.025959</td>\n",
       "      <td>20.686542</td>\n",
       "      <td>4.014342</td>\n",
       "      <td>6.149698</td>\n",
       "      <td>60.427962</td>\n",
       "      <td>61.668708</td>\n",
       "      <td>95.185324</td>\n",
       "      <td>134.603261</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32061</th>\n",
       "      <td>14018_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20170705180448.1.1...</td>\n",
       "      <td>6021539</td>\n",
       "      <td>1.0</td>\n",
       "      <td>179.0</td>\n",
       "      <td>74.7</td>\n",
       "      <td>23.3139</td>\n",
       "      <td>59.0</td>\n",
       "      <td>25.232678</td>\n",
       "      <td>30.970184</td>\n",
       "      <td>...</td>\n",
       "      <td>31.116022</td>\n",
       "      <td>38.579135</td>\n",
       "      <td>0.977047</td>\n",
       "      <td>20.729378</td>\n",
       "      <td>4.381059</td>\n",
       "      <td>6.504511</td>\n",
       "      <td>59.444884</td>\n",
       "      <td>59.744710</td>\n",
       "      <td>90.479553</td>\n",
       "      <td>132.776803</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32062</th>\n",
       "      <td>13966_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20170702173904.1.1...</td>\n",
       "      <td>6022591</td>\n",
       "      <td>0.0</td>\n",
       "      <td>156.0</td>\n",
       "      <td>96.2</td>\n",
       "      <td>39.5299</td>\n",
       "      <td>62.0</td>\n",
       "      <td>20.946633</td>\n",
       "      <td>26.922261</td>\n",
       "      <td>...</td>\n",
       "      <td>27.342555</td>\n",
       "      <td>35.846661</td>\n",
       "      <td>0.959371</td>\n",
       "      <td>19.152575</td>\n",
       "      <td>3.211435</td>\n",
       "      <td>4.959191</td>\n",
       "      <td>67.937764</td>\n",
       "      <td>52.910551</td>\n",
       "      <td>80.626829</td>\n",
       "      <td>126.477899</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32063</th>\n",
       "      <td>16309_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20171202183632.1.1...</td>\n",
       "      <td>6023431</td>\n",
       "      <td>0.0</td>\n",
       "      <td>168.0</td>\n",
       "      <td>58.0</td>\n",
       "      <td>20.5499</td>\n",
       "      <td>43.0</td>\n",
       "      <td>22.787944</td>\n",
       "      <td>29.871832</td>\n",
       "      <td>...</td>\n",
       "      <td>27.698720</td>\n",
       "      <td>36.369916</td>\n",
       "      <td>0.906038</td>\n",
       "      <td>19.990121</td>\n",
       "      <td>3.935749</td>\n",
       "      <td>5.201943</td>\n",
       "      <td>60.779334</td>\n",
       "      <td>55.123637</td>\n",
       "      <td>84.255142</td>\n",
       "      <td>150.788401</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32064</th>\n",
       "      <td>12964_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20170418184226.1.1...</td>\n",
       "      <td>6024029</td>\n",
       "      <td>1.0</td>\n",
       "      <td>179.0</td>\n",
       "      <td>77.5</td>\n",
       "      <td>24.1878</td>\n",
       "      <td>41.0</td>\n",
       "      <td>24.323043</td>\n",
       "      <td>29.804727</td>\n",
       "      <td>...</td>\n",
       "      <td>30.125650</td>\n",
       "      <td>39.486231</td>\n",
       "      <td>0.975894</td>\n",
       "      <td>20.408280</td>\n",
       "      <td>4.298543</td>\n",
       "      <td>6.171569</td>\n",
       "      <td>59.544963</td>\n",
       "      <td>62.084987</td>\n",
       "      <td>93.878589</td>\n",
       "      <td>138.796708</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>32065 rows × 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       image_id                                          file_name      eid  \\\n",
       "0      14366_82  1.2.840.113619.2.110.212038.20190307144208.1.1...  1000951   \n",
       "1      09648_82  1.2.840.113619.2.110.212038.20171116132027.1.1...  1001329   \n",
       "2      01877_82  1.2.840.113619.2.110.210419.20180720150454.1.1...  1001512   \n",
       "3      05431_82  1.2.840.113619.2.110.210419.20190328125146.1.1...  1001867   \n",
       "4      10557_82  1.2.840.113619.2.110.212038.20180228105159.1.1...  1001921   \n",
       "...         ...                                                ...      ...   \n",
       "32060  10039_93  1.2.840.113619.2.110.210419.20160923114938.1.1...  6021112   \n",
       "32061  14018_93  1.2.840.113619.2.110.210419.20170705180448.1.1...  6021539   \n",
       "32062  13966_93  1.2.840.113619.2.110.210419.20170702173904.1.1...  6022591   \n",
       "32063  16309_93  1.2.840.113619.2.110.210419.20171202183632.1.1...  6023431   \n",
       "32064  12964_93  1.2.840.113619.2.110.210419.20170418184226.1.1...  6024029   \n",
       "\n",
       "       sex  standing_height  weight      bmi   age  pelvic_height  \\\n",
       "0      1.0            167.0    62.9  22.5537  66.0      22.297293   \n",
       "1      1.0            185.0    84.9  24.8064  58.0      25.363221   \n",
       "2      1.0            171.0    81.4  27.8376  57.0      23.742297   \n",
       "3      0.0            155.0    66.1  27.5130  64.0      22.226944   \n",
       "4      1.0            167.0    62.6  22.4461  52.0      22.861258   \n",
       "...    ...              ...     ...      ...   ...            ...   \n",
       "32060  1.0            178.0    87.0  27.4587  64.0      25.247090   \n",
       "32061  1.0            179.0    74.7  23.3139  59.0      25.232678   \n",
       "32062  0.0            156.0    96.2  39.5299  62.0      20.946633   \n",
       "32063  0.0            168.0    58.0  20.5499  43.0      22.787944   \n",
       "32064  1.0            179.0    77.5  24.1878  41.0      24.323043   \n",
       "\n",
       "       pelvic_width  ...  trochanter_left2trochanter_right  shoulder_width  \\\n",
       "0         30.113663  ...                         30.091232       36.696399   \n",
       "1         33.085310  ...                         31.474656       43.357959   \n",
       "2         30.901775  ...                         30.813208       39.203992   \n",
       "3         30.934457  ...                         31.173284       35.909407   \n",
       "4         28.863430  ...                         29.777444       33.219853   \n",
       "...             ...  ...                               ...             ...   \n",
       "32060     32.848104  ...                         30.995640       41.118337   \n",
       "32061     30.970184  ...                         31.116022       38.579135   \n",
       "32062     26.922261  ...                         27.342555       35.846661   \n",
       "32063     29.871832  ...                         27.698720       36.369916   \n",
       "32064     29.804727  ...                         30.125650       39.486231   \n",
       "\n",
       "       arm_devide_torso  bi_acetabular_width  iliac_isthmus_breadth  \\\n",
       "0              1.067168            20.324819               4.164529   \n",
       "1              1.023811            21.085789               5.090414   \n",
       "2              0.933102            20.756184               4.657824   \n",
       "3              0.991270            21.701987               3.718747   \n",
       "4              1.014700            19.784228               4.238295   \n",
       "...                 ...                  ...                    ...   \n",
       "32060          1.025959            20.686542               4.014342   \n",
       "32061          0.977047            20.729378               4.381059   \n",
       "32062          0.959371            19.152575               3.211435   \n",
       "32063          0.906038            19.990121               3.935749   \n",
       "32064          0.975894            20.408280               4.298543   \n",
       "\n",
       "       acetabular_diameter  acetabular_inclination  arm_length  leg_length  \\\n",
       "0                 6.141783               60.747735   57.673326   89.233162   \n",
       "1                 6.729616               56.943430   64.205477   95.795086   \n",
       "2                 6.035855               59.189839   54.743642   85.065164   \n",
       "3                 5.531657               59.017487   50.766740   78.752591   \n",
       "4                 5.834113               58.027399   58.533827   86.154050   \n",
       "...                    ...                     ...         ...         ...   \n",
       "32060             6.149698               60.427962   61.668708   95.185324   \n",
       "32061             6.504511               59.444884   59.744710   90.479553   \n",
       "32062             4.959191               67.937764   52.910551   80.626829   \n",
       "32063             5.201943               60.779334   55.123637   84.255142   \n",
       "32064             6.171569               59.544963   62.084987   93.878589   \n",
       "\n",
       "       pelvic_inlet_area  \n",
       "0             108.190429  \n",
       "1             107.282431  \n",
       "2             102.660623  \n",
       "3             139.989191  \n",
       "4             110.520822  \n",
       "...                  ...  \n",
       "32060         134.603261  \n",
       "32061         132.776803  \n",
       "32062         126.477899  \n",
       "32063         150.788401  \n",
       "32064         138.796708  \n",
       "\n",
       "[32065 rows x 26 columns]"
      ]
     },
     "execution_count": 121,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hip_pheno"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### add 2 new phenotype as postive control: 1. arm:leg, 2. leg:torso"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {},
   "outputs": [],
   "source": [
    "hip_pheno_all = pd.read_csv('key_results/hip_pheno_23_cm.csv')\n",
    "\n",
    "image_ids = pd.read_csv('key_results/hip_select_pheno_cm_eid_flt_z_flt.csv')['image_id'].tolist()\n",
    "\n",
    "# filter individuals\n",
    "hip_pheno_all = hip_pheno_all[hip_pheno_all['image_id'].isin(image_ids)]\n",
    "\n",
    "# arm to torso ratio\n",
    "hip_pheno_all['arm_divide_leg'] = ((hip_pheno_all['arm_123_left'] + hip_pheno_all['arm_123_right']) / 2) / ((hip_pheno_all['leg_123_left'] + hip_pheno_all['leg_123_right']) / 2)\n",
    "hip_pheno_all['leg_divide_torso'] = ((hip_pheno_all['leg_123_left'] + hip_pheno_all['leg_123_right']) / 2) / hip_pheno_all['torso_length']\n",
    "add_pheno = hip_pheno_all[['eid', 'arm_divide_leg', 'leg_divide_torso']]\n",
    "\n",
    "# merge previous phenotypes\n",
    "hip_pheno_gwas_cm = hip_pheno_gwas_cm.merge(add_pheno, left_on='FID', right_on='eid', how='left')\n",
    "hip_pheno_gwas_cm.drop(columns=['eid'], inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>pelvic_height</th>\n",
       "      <th>pelvic_width</th>\n",
       "      <th>pelvic_inlet_width</th>\n",
       "      <th>iliac_flare_ratio</th>\n",
       "      <th>oblique_pelvic_inlet_length</th>\n",
       "      <th>iliac_flare_angle</th>\n",
       "      <th>subpubic_angle</th>\n",
       "      <th>ear_left2ear_right</th>\n",
       "      <th>...</th>\n",
       "      <th>leg_length</th>\n",
       "      <th>pelvic_inlet_area</th>\n",
       "      <th>head_divide_inlet_width</th>\n",
       "      <th>shoulder_divide_inlet_width</th>\n",
       "      <th>head_divide_oblique_inlet_length</th>\n",
       "      <th>shoulder_divide_oblique_inlet_length</th>\n",
       "      <th>head_area_divide_pelvic_inlet_area</th>\n",
       "      <th>shoulder_area_divide_pelvic_inlet_area</th>\n",
       "      <th>arm_divide_leg</th>\n",
       "      <th>leg_divide_torso</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1000951</td>\n",
       "      <td>1000951</td>\n",
       "      <td>22.297293</td>\n",
       "      <td>30.113663</td>\n",
       "      <td>13.937174</td>\n",
       "      <td>0.462819</td>\n",
       "      <td>9.883806</td>\n",
       "      <td>97.533181</td>\n",
       "      <td>116.365190</td>\n",
       "      <td>14.112233</td>\n",
       "      <td>...</td>\n",
       "      <td>89.233162</td>\n",
       "      <td>108.190429</td>\n",
       "      <td>1.012561</td>\n",
       "      <td>2.632987</td>\n",
       "      <td>1.427814</td>\n",
       "      <td>3.712780</td>\n",
       "      <td>1.445748</td>\n",
       "      <td>9.775701</td>\n",
       "      <td>0.646322</td>\n",
       "      <td>1.651141</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1001329</td>\n",
       "      <td>1001329</td>\n",
       "      <td>25.363221</td>\n",
       "      <td>33.085310</td>\n",
       "      <td>13.943964</td>\n",
       "      <td>0.421455</td>\n",
       "      <td>9.796084</td>\n",
       "      <td>99.812799</td>\n",
       "      <td>108.368022</td>\n",
       "      <td>14.487676</td>\n",
       "      <td>...</td>\n",
       "      <td>95.795086</td>\n",
       "      <td>107.282431</td>\n",
       "      <td>1.038993</td>\n",
       "      <td>3.109443</td>\n",
       "      <td>1.478925</td>\n",
       "      <td>4.426050</td>\n",
       "      <td>1.536593</td>\n",
       "      <td>13.762551</td>\n",
       "      <td>0.670238</td>\n",
       "      <td>1.527534</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1001512</td>\n",
       "      <td>1001512</td>\n",
       "      <td>23.742297</td>\n",
       "      <td>30.901775</td>\n",
       "      <td>13.669524</td>\n",
       "      <td>0.442354</td>\n",
       "      <td>9.562262</td>\n",
       "      <td>95.714405</td>\n",
       "      <td>117.491690</td>\n",
       "      <td>13.903518</td>\n",
       "      <td>...</td>\n",
       "      <td>85.065164</td>\n",
       "      <td>102.660623</td>\n",
       "      <td>1.017118</td>\n",
       "      <td>2.867985</td>\n",
       "      <td>1.453999</td>\n",
       "      <td>4.099866</td>\n",
       "      <td>1.478888</td>\n",
       "      <td>11.758355</td>\n",
       "      <td>0.643549</td>\n",
       "      <td>1.449931</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1001867</td>\n",
       "      <td>1001867</td>\n",
       "      <td>22.226944</td>\n",
       "      <td>30.934457</td>\n",
       "      <td>15.755275</td>\n",
       "      <td>0.509312</td>\n",
       "      <td>11.313022</td>\n",
       "      <td>101.418560</td>\n",
       "      <td>139.787157</td>\n",
       "      <td>13.029582</td>\n",
       "      <td>...</td>\n",
       "      <td>78.752591</td>\n",
       "      <td>139.989191</td>\n",
       "      <td>0.826998</td>\n",
       "      <td>2.279199</td>\n",
       "      <td>1.151733</td>\n",
       "      <td>3.174166</td>\n",
       "      <td>0.952481</td>\n",
       "      <td>7.234555</td>\n",
       "      <td>0.644636</td>\n",
       "      <td>1.537721</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1001921</td>\n",
       "      <td>1001921</td>\n",
       "      <td>22.861258</td>\n",
       "      <td>28.863430</td>\n",
       "      <td>14.217750</td>\n",
       "      <td>0.492587</td>\n",
       "      <td>9.897451</td>\n",
       "      <td>95.473293</td>\n",
       "      <td>116.159185</td>\n",
       "      <td>14.480994</td>\n",
       "      <td>...</td>\n",
       "      <td>86.154050</td>\n",
       "      <td>110.520822</td>\n",
       "      <td>1.018515</td>\n",
       "      <td>2.336506</td>\n",
       "      <td>1.463103</td>\n",
       "      <td>3.356405</td>\n",
       "      <td>1.490193</td>\n",
       "      <td>7.842259</td>\n",
       "      <td>0.679409</td>\n",
       "      <td>1.493503</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32040</th>\n",
       "      <td>6021112</td>\n",
       "      <td>6021112</td>\n",
       "      <td>25.247090</td>\n",
       "      <td>32.848104</td>\n",
       "      <td>14.645469</td>\n",
       "      <td>0.445854</td>\n",
       "      <td>11.702063</td>\n",
       "      <td>95.353305</td>\n",
       "      <td>126.858511</td>\n",
       "      <td>13.402493</td>\n",
       "      <td>...</td>\n",
       "      <td>95.185324</td>\n",
       "      <td>134.603261</td>\n",
       "      <td>0.915129</td>\n",
       "      <td>2.807581</td>\n",
       "      <td>1.145310</td>\n",
       "      <td>3.513768</td>\n",
       "      <td>1.048107</td>\n",
       "      <td>9.865188</td>\n",
       "      <td>0.647880</td>\n",
       "      <td>1.583562</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32041</th>\n",
       "      <td>6021539</td>\n",
       "      <td>6021539</td>\n",
       "      <td>25.232678</td>\n",
       "      <td>30.970184</td>\n",
       "      <td>14.210350</td>\n",
       "      <td>0.458840</td>\n",
       "      <td>11.896728</td>\n",
       "      <td>87.564668</td>\n",
       "      <td>120.568969</td>\n",
       "      <td>13.781823</td>\n",
       "      <td>...</td>\n",
       "      <td>90.479553</td>\n",
       "      <td>132.776803</td>\n",
       "      <td>0.969844</td>\n",
       "      <td>2.714862</td>\n",
       "      <td>1.158455</td>\n",
       "      <td>3.242836</td>\n",
       "      <td>1.123520</td>\n",
       "      <td>8.803850</td>\n",
       "      <td>0.660312</td>\n",
       "      <td>1.479675</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32042</th>\n",
       "      <td>6022591</td>\n",
       "      <td>6022591</td>\n",
       "      <td>20.946633</td>\n",
       "      <td>26.922261</td>\n",
       "      <td>13.540263</td>\n",
       "      <td>0.502939</td>\n",
       "      <td>11.893171</td>\n",
       "      <td>88.868037</td>\n",
       "      <td>143.403514</td>\n",
       "      <td>12.551823</td>\n",
       "      <td>...</td>\n",
       "      <td>80.626829</td>\n",
       "      <td>126.477899</td>\n",
       "      <td>0.927000</td>\n",
       "      <td>2.647412</td>\n",
       "      <td>1.055381</td>\n",
       "      <td>3.014054</td>\n",
       "      <td>0.978338</td>\n",
       "      <td>7.979445</td>\n",
       "      <td>0.656240</td>\n",
       "      <td>1.461921</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32043</th>\n",
       "      <td>6023431</td>\n",
       "      <td>6023431</td>\n",
       "      <td>22.787944</td>\n",
       "      <td>29.871832</td>\n",
       "      <td>13.586974</td>\n",
       "      <td>0.454842</td>\n",
       "      <td>14.130428</td>\n",
       "      <td>90.205445</td>\n",
       "      <td>157.984881</td>\n",
       "      <td>12.309345</td>\n",
       "      <td>...</td>\n",
       "      <td>84.255142</td>\n",
       "      <td>150.788401</td>\n",
       "      <td>0.905967</td>\n",
       "      <td>2.676822</td>\n",
       "      <td>0.871123</td>\n",
       "      <td>2.573872</td>\n",
       "      <td>0.789209</td>\n",
       "      <td>6.889799</td>\n",
       "      <td>0.654247</td>\n",
       "      <td>1.384857</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32044</th>\n",
       "      <td>6024029</td>\n",
       "      <td>6024029</td>\n",
       "      <td>24.323043</td>\n",
       "      <td>29.804727</td>\n",
       "      <td>13.883682</td>\n",
       "      <td>0.465821</td>\n",
       "      <td>12.728717</td>\n",
       "      <td>85.308035</td>\n",
       "      <td>124.424809</td>\n",
       "      <td>13.271879</td>\n",
       "      <td>...</td>\n",
       "      <td>93.878589</td>\n",
       "      <td>138.796708</td>\n",
       "      <td>0.955934</td>\n",
       "      <td>2.844075</td>\n",
       "      <td>1.042672</td>\n",
       "      <td>3.102138</td>\n",
       "      <td>0.996726</td>\n",
       "      <td>8.822711</td>\n",
       "      <td>0.661333</td>\n",
       "      <td>1.475647</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>32045 rows × 28 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           FID      IID  pelvic_height  pelvic_width  pelvic_inlet_width  \\\n",
       "0      1000951  1000951      22.297293     30.113663           13.937174   \n",
       "1      1001329  1001329      25.363221     33.085310           13.943964   \n",
       "2      1001512  1001512      23.742297     30.901775           13.669524   \n",
       "3      1001867  1001867      22.226944     30.934457           15.755275   \n",
       "4      1001921  1001921      22.861258     28.863430           14.217750   \n",
       "...        ...      ...            ...           ...                 ...   \n",
       "32040  6021112  6021112      25.247090     32.848104           14.645469   \n",
       "32041  6021539  6021539      25.232678     30.970184           14.210350   \n",
       "32042  6022591  6022591      20.946633     26.922261           13.540263   \n",
       "32043  6023431  6023431      22.787944     29.871832           13.586974   \n",
       "32044  6024029  6024029      24.323043     29.804727           13.883682   \n",
       "\n",
       "       iliac_flare_ratio  oblique_pelvic_inlet_length  iliac_flare_angle  \\\n",
       "0               0.462819                     9.883806          97.533181   \n",
       "1               0.421455                     9.796084          99.812799   \n",
       "2               0.442354                     9.562262          95.714405   \n",
       "3               0.509312                    11.313022         101.418560   \n",
       "4               0.492587                     9.897451          95.473293   \n",
       "...                  ...                          ...                ...   \n",
       "32040           0.445854                    11.702063          95.353305   \n",
       "32041           0.458840                    11.896728          87.564668   \n",
       "32042           0.502939                    11.893171          88.868037   \n",
       "32043           0.454842                    14.130428          90.205445   \n",
       "32044           0.465821                    12.728717          85.308035   \n",
       "\n",
       "       subpubic_angle  ear_left2ear_right  ...  leg_length  pelvic_inlet_area  \\\n",
       "0          116.365190           14.112233  ...   89.233162         108.190429   \n",
       "1          108.368022           14.487676  ...   95.795086         107.282431   \n",
       "2          117.491690           13.903518  ...   85.065164         102.660623   \n",
       "3          139.787157           13.029582  ...   78.752591         139.989191   \n",
       "4          116.159185           14.480994  ...   86.154050         110.520822   \n",
       "...               ...                 ...  ...         ...                ...   \n",
       "32040      126.858511           13.402493  ...   95.185324         134.603261   \n",
       "32041      120.568969           13.781823  ...   90.479553         132.776803   \n",
       "32042      143.403514           12.551823  ...   80.626829         126.477899   \n",
       "32043      157.984881           12.309345  ...   84.255142         150.788401   \n",
       "32044      124.424809           13.271879  ...   93.878589         138.796708   \n",
       "\n",
       "       head_divide_inlet_width  shoulder_divide_inlet_width  \\\n",
       "0                     1.012561                     2.632987   \n",
       "1                     1.038993                     3.109443   \n",
       "2                     1.017118                     2.867985   \n",
       "3                     0.826998                     2.279199   \n",
       "4                     1.018515                     2.336506   \n",
       "...                        ...                          ...   \n",
       "32040                 0.915129                     2.807581   \n",
       "32041                 0.969844                     2.714862   \n",
       "32042                 0.927000                     2.647412   \n",
       "32043                 0.905967                     2.676822   \n",
       "32044                 0.955934                     2.844075   \n",
       "\n",
       "       head_divide_oblique_inlet_length  shoulder_divide_oblique_inlet_length  \\\n",
       "0                              1.427814                              3.712780   \n",
       "1                              1.478925                              4.426050   \n",
       "2                              1.453999                              4.099866   \n",
       "3                              1.151733                              3.174166   \n",
       "4                              1.463103                              3.356405   \n",
       "...                                 ...                                   ...   \n",
       "32040                          1.145310                              3.513768   \n",
       "32041                          1.158455                              3.242836   \n",
       "32042                          1.055381                              3.014054   \n",
       "32043                          0.871123                              2.573872   \n",
       "32044                          1.042672                              3.102138   \n",
       "\n",
       "       head_area_divide_pelvic_inlet_area  \\\n",
       "0                                1.445748   \n",
       "1                                1.536593   \n",
       "2                                1.478888   \n",
       "3                                0.952481   \n",
       "4                                1.490193   \n",
       "...                                   ...   \n",
       "32040                            1.048107   \n",
       "32041                            1.123520   \n",
       "32042                            0.978338   \n",
       "32043                            0.789209   \n",
       "32044                            0.996726   \n",
       "\n",
       "       shoulder_area_divide_pelvic_inlet_area  arm_divide_leg  \\\n",
       "0                                    9.775701        0.646322   \n",
       "1                                   13.762551        0.670238   \n",
       "2                                   11.758355        0.643549   \n",
       "3                                    7.234555        0.644636   \n",
       "4                                    7.842259        0.679409   \n",
       "...                                       ...             ...   \n",
       "32040                                9.865188        0.647880   \n",
       "32041                                8.803850        0.660312   \n",
       "32042                                7.979445        0.656240   \n",
       "32043                                6.889799        0.654247   \n",
       "32044                                8.822711        0.661333   \n",
       "\n",
       "       leg_divide_torso  \n",
       "0              1.651141  \n",
       "1              1.527534  \n",
       "2              1.449931  \n",
       "3              1.537721  \n",
       "4              1.493503  \n",
       "...                 ...  \n",
       "32040          1.583562  \n",
       "32041          1.479675  \n",
       "32042          1.461921  \n",
       "32043          1.384857  \n",
       "32044          1.475647  \n",
       "\n",
       "[32045 rows x 28 columns]"
      ]
     },
     "execution_count": 123,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hip_pheno_gwas_cm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['FID', 'IID', 'pelvic_height', 'pelvic_width', 'pelvic_inlet_width',\n",
       "       'iliac_flare_ratio', 'oblique_pelvic_inlet_length', 'iliac_flare_angle',\n",
       "       'subpubic_angle', 'ear_left2ear_right',\n",
       "       'trochanter_left2trochanter_right', 'shoulder_width',\n",
       "       'arm_devide_torso', 'bi_acetabular_width', 'iliac_isthmus_breadth',\n",
       "       'acetabular_diameter', 'acetabular_inclination', 'arm_length',\n",
       "       'leg_length', 'pelvic_inlet_area', 'head_divide_inlet_width',\n",
       "       'shoulder_divide_inlet_width', 'head_divide_oblique_inlet_length',\n",
       "       'shoulder_divide_oblique_inlet_length',\n",
       "       'head_area_divide_pelvic_inlet_area',\n",
       "       'shoulder_area_divide_pelvic_inlet_area', 'arm_divide_leg',\n",
       "       'leg_divide_torso'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 124,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hip_pheno_gwas_cm.columns"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### filter phenotypes for gwas"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "pheno = ['FID', 'IID', 'pelvic_height', 'pelvic_width', 'pelvic_inlet_width',\n",
    "         'oblique_pelvic_inlet_length', 'subpubic_angle', 'bi_acetabular_width', \n",
    "         'iliac_isthmus_breadth', 'acetabular_diameter', 'pelvic_inlet_area',\n",
    "         'leg_length']\n",
    "\n",
    "hip_pheno_gwas_cm = hip_pheno_gwas_cm[pheno]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>pelvic_height</th>\n",
       "      <th>pelvic_width</th>\n",
       "      <th>pelvic_inlet_width</th>\n",
       "      <th>oblique_pelvic_inlet_length</th>\n",
       "      <th>subpubic_angle</th>\n",
       "      <th>bi_acetabular_width</th>\n",
       "      <th>iliac_isthmus_breadth</th>\n",
       "      <th>acetabular_diameter</th>\n",
       "      <th>pelvic_inlet_area</th>\n",
       "      <th>leg_length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1000951</td>\n",
       "      <td>1000951</td>\n",
       "      <td>22.297293</td>\n",
       "      <td>30.113663</td>\n",
       "      <td>13.937174</td>\n",
       "      <td>9.883806</td>\n",
       "      <td>116.365190</td>\n",
       "      <td>20.324819</td>\n",
       "      <td>4.164529</td>\n",
       "      <td>6.141783</td>\n",
       "      <td>108.190429</td>\n",
       "      <td>89.233162</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1001512</td>\n",
       "      <td>1001512</td>\n",
       "      <td>23.742297</td>\n",
       "      <td>30.901775</td>\n",
       "      <td>13.669524</td>\n",
       "      <td>9.562262</td>\n",
       "      <td>117.491690</td>\n",
       "      <td>20.756184</td>\n",
       "      <td>4.657824</td>\n",
       "      <td>6.035855</td>\n",
       "      <td>102.660623</td>\n",
       "      <td>85.065164</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1001867</td>\n",
       "      <td>1001867</td>\n",
       "      <td>22.226944</td>\n",
       "      <td>30.934457</td>\n",
       "      <td>15.755275</td>\n",
       "      <td>11.313022</td>\n",
       "      <td>139.787157</td>\n",
       "      <td>21.701987</td>\n",
       "      <td>3.718747</td>\n",
       "      <td>5.531657</td>\n",
       "      <td>139.989191</td>\n",
       "      <td>78.752591</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1001921</td>\n",
       "      <td>1001921</td>\n",
       "      <td>22.861258</td>\n",
       "      <td>28.863430</td>\n",
       "      <td>14.217750</td>\n",
       "      <td>9.897451</td>\n",
       "      <td>116.159185</td>\n",
       "      <td>19.784228</td>\n",
       "      <td>4.238295</td>\n",
       "      <td>5.834113</td>\n",
       "      <td>110.520822</td>\n",
       "      <td>86.154050</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1001950</td>\n",
       "      <td>1001950</td>\n",
       "      <td>23.080688</td>\n",
       "      <td>29.385649</td>\n",
       "      <td>13.583558</td>\n",
       "      <td>10.305746</td>\n",
       "      <td>130.647738</td>\n",
       "      <td>19.821637</td>\n",
       "      <td>4.438966</td>\n",
       "      <td>5.731393</td>\n",
       "      <td>109.946859</td>\n",
       "      <td>84.286956</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31110</th>\n",
       "      <td>6021112</td>\n",
       "      <td>6021112</td>\n",
       "      <td>25.247090</td>\n",
       "      <td>32.848104</td>\n",
       "      <td>14.645469</td>\n",
       "      <td>11.702063</td>\n",
       "      <td>126.858511</td>\n",
       "      <td>20.686542</td>\n",
       "      <td>4.014342</td>\n",
       "      <td>6.149698</td>\n",
       "      <td>134.603261</td>\n",
       "      <td>95.185324</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31111</th>\n",
       "      <td>6021539</td>\n",
       "      <td>6021539</td>\n",
       "      <td>25.232678</td>\n",
       "      <td>30.970184</td>\n",
       "      <td>14.210350</td>\n",
       "      <td>11.896728</td>\n",
       "      <td>120.568969</td>\n",
       "      <td>20.729378</td>\n",
       "      <td>4.381059</td>\n",
       "      <td>6.504511</td>\n",
       "      <td>132.776803</td>\n",
       "      <td>90.479553</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31112</th>\n",
       "      <td>6022591</td>\n",
       "      <td>6022591</td>\n",
       "      <td>20.946633</td>\n",
       "      <td>26.922261</td>\n",
       "      <td>13.540263</td>\n",
       "      <td>11.893171</td>\n",
       "      <td>143.403514</td>\n",
       "      <td>19.152575</td>\n",
       "      <td>3.211435</td>\n",
       "      <td>4.959191</td>\n",
       "      <td>126.477899</td>\n",
       "      <td>80.626829</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31113</th>\n",
       "      <td>6023431</td>\n",
       "      <td>6023431</td>\n",
       "      <td>22.787944</td>\n",
       "      <td>29.871832</td>\n",
       "      <td>13.586974</td>\n",
       "      <td>14.130428</td>\n",
       "      <td>157.984881</td>\n",
       "      <td>19.990121</td>\n",
       "      <td>3.935749</td>\n",
       "      <td>5.201943</td>\n",
       "      <td>150.788401</td>\n",
       "      <td>84.255142</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31114</th>\n",
       "      <td>6024029</td>\n",
       "      <td>6024029</td>\n",
       "      <td>24.323043</td>\n",
       "      <td>29.804727</td>\n",
       "      <td>13.883682</td>\n",
       "      <td>12.728717</td>\n",
       "      <td>124.424809</td>\n",
       "      <td>20.408280</td>\n",
       "      <td>4.298543</td>\n",
       "      <td>6.171569</td>\n",
       "      <td>138.796708</td>\n",
       "      <td>93.878589</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>31115 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           FID      IID  pelvic_height  pelvic_width  pelvic_inlet_width  \\\n",
       "0      1000951  1000951      22.297293     30.113663           13.937174   \n",
       "1      1001512  1001512      23.742297     30.901775           13.669524   \n",
       "2      1001867  1001867      22.226944     30.934457           15.755275   \n",
       "3      1001921  1001921      22.861258     28.863430           14.217750   \n",
       "4      1001950  1001950      23.080688     29.385649           13.583558   \n",
       "...        ...      ...            ...           ...                 ...   \n",
       "31110  6021112  6021112      25.247090     32.848104           14.645469   \n",
       "31111  6021539  6021539      25.232678     30.970184           14.210350   \n",
       "31112  6022591  6022591      20.946633     26.922261           13.540263   \n",
       "31113  6023431  6023431      22.787944     29.871832           13.586974   \n",
       "31114  6024029  6024029      24.323043     29.804727           13.883682   \n",
       "\n",
       "       oblique_pelvic_inlet_length  subpubic_angle  bi_acetabular_width  \\\n",
       "0                         9.883806      116.365190            20.324819   \n",
       "1                         9.562262      117.491690            20.756184   \n",
       "2                        11.313022      139.787157            21.701987   \n",
       "3                         9.897451      116.159185            19.784228   \n",
       "4                        10.305746      130.647738            19.821637   \n",
       "...                            ...             ...                  ...   \n",
       "31110                    11.702063      126.858511            20.686542   \n",
       "31111                    11.896728      120.568969            20.729378   \n",
       "31112                    11.893171      143.403514            19.152575   \n",
       "31113                    14.130428      157.984881            19.990121   \n",
       "31114                    12.728717      124.424809            20.408280   \n",
       "\n",
       "       iliac_isthmus_breadth  acetabular_diameter  pelvic_inlet_area  \\\n",
       "0                   4.164529             6.141783         108.190429   \n",
       "1                   4.657824             6.035855         102.660623   \n",
       "2                   3.718747             5.531657         139.989191   \n",
       "3                   4.238295             5.834113         110.520822   \n",
       "4                   4.438966             5.731393         109.946859   \n",
       "...                      ...                  ...                ...   \n",
       "31110               4.014342             6.149698         134.603261   \n",
       "31111               4.381059             6.504511         132.776803   \n",
       "31112               3.211435             4.959191         126.477899   \n",
       "31113               3.935749             5.201943         150.788401   \n",
       "31114               4.298543             6.171569         138.796708   \n",
       "\n",
       "       leg_length  \n",
       "0       89.233162  \n",
       "1       85.065164  \n",
       "2       78.752591  \n",
       "3       86.154050  \n",
       "4       84.286956  \n",
       "...           ...  \n",
       "31110   95.185324  \n",
       "31111   90.479553  \n",
       "31112   80.626829  \n",
       "31113   84.255142  \n",
       "31114   93.878589  \n",
       "\n",
       "[31115 rows x 12 columns]"
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hip_pheno_gwas_cm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "hip_pheno_gwas_cm.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_both_cm_20240104.txt', index=False, sep = \" \")\n",
    "hip_pheno_gwas_cm.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_both_cm_no_header_20240104.txt', index=False, header=False, sep = \" \")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [],
   "source": [
    "hip_pheno_gwas_cm[['FID', 'IID']].to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_both_eids_20240104.txt', index=False, header=False, sep=' ')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Divide to male and female 231106"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>pelvic_height</th>\n",
       "      <th>pelvic_width</th>\n",
       "      <th>pelvic_inlet_width</th>\n",
       "      <th>oblique_pelvic_inlet_length</th>\n",
       "      <th>subpubic_angle</th>\n",
       "      <th>bi_acetabular_width</th>\n",
       "      <th>iliac_isthmus_breadth</th>\n",
       "      <th>acetabular_diameter</th>\n",
       "      <th>pelvic_inlet_area</th>\n",
       "      <th>leg_length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1000951</td>\n",
       "      <td>1000951</td>\n",
       "      <td>22.297293</td>\n",
       "      <td>30.113663</td>\n",
       "      <td>13.937174</td>\n",
       "      <td>9.883806</td>\n",
       "      <td>116.365190</td>\n",
       "      <td>20.324819</td>\n",
       "      <td>4.164529</td>\n",
       "      <td>6.141783</td>\n",
       "      <td>108.190429</td>\n",
       "      <td>89.233162</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1001329</td>\n",
       "      <td>1001329</td>\n",
       "      <td>25.363221</td>\n",
       "      <td>33.085310</td>\n",
       "      <td>13.943964</td>\n",
       "      <td>9.796084</td>\n",
       "      <td>108.368022</td>\n",
       "      <td>21.085789</td>\n",
       "      <td>5.090414</td>\n",
       "      <td>6.729616</td>\n",
       "      <td>107.282431</td>\n",
       "      <td>95.795086</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1001512</td>\n",
       "      <td>1001512</td>\n",
       "      <td>23.742297</td>\n",
       "      <td>30.901775</td>\n",
       "      <td>13.669524</td>\n",
       "      <td>9.562262</td>\n",
       "      <td>117.491690</td>\n",
       "      <td>20.756184</td>\n",
       "      <td>4.657824</td>\n",
       "      <td>6.035855</td>\n",
       "      <td>102.660623</td>\n",
       "      <td>85.065164</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1001867</td>\n",
       "      <td>1001867</td>\n",
       "      <td>22.226944</td>\n",
       "      <td>30.934457</td>\n",
       "      <td>15.755275</td>\n",
       "      <td>11.313022</td>\n",
       "      <td>139.787157</td>\n",
       "      <td>21.701987</td>\n",
       "      <td>3.718747</td>\n",
       "      <td>5.531657</td>\n",
       "      <td>139.989191</td>\n",
       "      <td>78.752591</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1001921</td>\n",
       "      <td>1001921</td>\n",
       "      <td>22.861258</td>\n",
       "      <td>28.863430</td>\n",
       "      <td>14.217750</td>\n",
       "      <td>9.897451</td>\n",
       "      <td>116.159185</td>\n",
       "      <td>19.784228</td>\n",
       "      <td>4.238295</td>\n",
       "      <td>5.834113</td>\n",
       "      <td>110.520822</td>\n",
       "      <td>86.154050</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32040</th>\n",
       "      <td>6021112</td>\n",
       "      <td>6021112</td>\n",
       "      <td>25.247090</td>\n",
       "      <td>32.848104</td>\n",
       "      <td>14.645469</td>\n",
       "      <td>11.702063</td>\n",
       "      <td>126.858511</td>\n",
       "      <td>20.686542</td>\n",
       "      <td>4.014342</td>\n",
       "      <td>6.149698</td>\n",
       "      <td>134.603261</td>\n",
       "      <td>95.185324</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32041</th>\n",
       "      <td>6021539</td>\n",
       "      <td>6021539</td>\n",
       "      <td>25.232678</td>\n",
       "      <td>30.970184</td>\n",
       "      <td>14.210350</td>\n",
       "      <td>11.896728</td>\n",
       "      <td>120.568969</td>\n",
       "      <td>20.729378</td>\n",
       "      <td>4.381059</td>\n",
       "      <td>6.504511</td>\n",
       "      <td>132.776803</td>\n",
       "      <td>90.479553</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32042</th>\n",
       "      <td>6022591</td>\n",
       "      <td>6022591</td>\n",
       "      <td>20.946633</td>\n",
       "      <td>26.922261</td>\n",
       "      <td>13.540263</td>\n",
       "      <td>11.893171</td>\n",
       "      <td>143.403514</td>\n",
       "      <td>19.152575</td>\n",
       "      <td>3.211435</td>\n",
       "      <td>4.959191</td>\n",
       "      <td>126.477899</td>\n",
       "      <td>80.626829</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32043</th>\n",
       "      <td>6023431</td>\n",
       "      <td>6023431</td>\n",
       "      <td>22.787944</td>\n",
       "      <td>29.871832</td>\n",
       "      <td>13.586974</td>\n",
       "      <td>14.130428</td>\n",
       "      <td>157.984881</td>\n",
       "      <td>19.990121</td>\n",
       "      <td>3.935749</td>\n",
       "      <td>5.201943</td>\n",
       "      <td>150.788401</td>\n",
       "      <td>84.255142</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32044</th>\n",
       "      <td>6024029</td>\n",
       "      <td>6024029</td>\n",
       "      <td>24.323043</td>\n",
       "      <td>29.804727</td>\n",
       "      <td>13.883682</td>\n",
       "      <td>12.728717</td>\n",
       "      <td>124.424809</td>\n",
       "      <td>20.408280</td>\n",
       "      <td>4.298543</td>\n",
       "      <td>6.171569</td>\n",
       "      <td>138.796708</td>\n",
       "      <td>93.878589</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>32045 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           FID      IID  pelvic_height  pelvic_width  pelvic_inlet_width  \\\n",
       "0      1000951  1000951      22.297293     30.113663           13.937174   \n",
       "1      1001329  1001329      25.363221     33.085310           13.943964   \n",
       "2      1001512  1001512      23.742297     30.901775           13.669524   \n",
       "3      1001867  1001867      22.226944     30.934457           15.755275   \n",
       "4      1001921  1001921      22.861258     28.863430           14.217750   \n",
       "...        ...      ...            ...           ...                 ...   \n",
       "32040  6021112  6021112      25.247090     32.848104           14.645469   \n",
       "32041  6021539  6021539      25.232678     30.970184           14.210350   \n",
       "32042  6022591  6022591      20.946633     26.922261           13.540263   \n",
       "32043  6023431  6023431      22.787944     29.871832           13.586974   \n",
       "32044  6024029  6024029      24.323043     29.804727           13.883682   \n",
       "\n",
       "       oblique_pelvic_inlet_length  subpubic_angle  bi_acetabular_width  \\\n",
       "0                         9.883806      116.365190            20.324819   \n",
       "1                         9.796084      108.368022            21.085789   \n",
       "2                         9.562262      117.491690            20.756184   \n",
       "3                        11.313022      139.787157            21.701987   \n",
       "4                         9.897451      116.159185            19.784228   \n",
       "...                            ...             ...                  ...   \n",
       "32040                    11.702063      126.858511            20.686542   \n",
       "32041                    11.896728      120.568969            20.729378   \n",
       "32042                    11.893171      143.403514            19.152575   \n",
       "32043                    14.130428      157.984881            19.990121   \n",
       "32044                    12.728717      124.424809            20.408280   \n",
       "\n",
       "       iliac_isthmus_breadth  acetabular_diameter  pelvic_inlet_area  \\\n",
       "0                   4.164529             6.141783         108.190429   \n",
       "1                   5.090414             6.729616         107.282431   \n",
       "2                   4.657824             6.035855         102.660623   \n",
       "3                   3.718747             5.531657         139.989191   \n",
       "4                   4.238295             5.834113         110.520822   \n",
       "...                      ...                  ...                ...   \n",
       "32040               4.014342             6.149698         134.603261   \n",
       "32041               4.381059             6.504511         132.776803   \n",
       "32042               3.211435             4.959191         126.477899   \n",
       "32043               3.935749             5.201943         150.788401   \n",
       "32044               4.298543             6.171569         138.796708   \n",
       "\n",
       "       leg_length  \n",
       "0       89.233162  \n",
       "1       95.795086  \n",
       "2       85.065164  \n",
       "3       78.752591  \n",
       "4       86.154050  \n",
       "...           ...  \n",
       "32040   95.185324  \n",
       "32041   90.479553  \n",
       "32042   80.626829  \n",
       "32043   84.255142  \n",
       "32044   93.878589  \n",
       "\n",
       "[32045 rows x 12 columns]"
      ]
     },
     "execution_count": 128,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hip_pheno_gwas_cm = pd.read_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_both_cm_231109.txt', sep = \" \"); hip_pheno_gwas_cm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "metadata": {},
   "outputs": [],
   "source": [
    "sex_df = pd.read_csv('key_results/hip_select_pheno_cm_eid_flt_z_flt.csv')[['eid', 'sex']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "male_eids_length = 15716\n",
      "female_eids_length = 16329\n"
     ]
    }
   ],
   "source": [
    "male_eids = sex_df[sex_df['sex'] == 1.0]['eid'].tolist()\n",
    "female_eids = sex_df[sex_df['sex'] == 0.0]['eid'].tolist()\n",
    "print(f\"male_eids_length = {len(male_eids)}\")\n",
    "print(f\"female_eids_length = {len(female_eids)}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "male shape = (15716, 12)\n",
      "female shape = (16329, 12)\n"
     ]
    }
   ],
   "source": [
    "hip_pheno_gwas_male = hip_pheno_gwas_cm[hip_pheno_gwas_cm['IID'].isin(male_eids)]\n",
    "hip_pheno_gwas_female = hip_pheno_gwas_cm[hip_pheno_gwas_cm['IID'].isin(female_eids)]\n",
    "\n",
    "print(f\"male shape = {hip_pheno_gwas_male.shape}\")\n",
    "print(f\"female shape = {hip_pheno_gwas_female.shape}\")\n",
    "\n",
    "hip_pheno_gwas_male.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_male_cm_231109.txt', index=False, sep = \" \")\n",
    "hip_pheno_gwas_male.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_male_cm_no_header_231109.txt', index=False, header=False, sep = \" \")\n",
    "\n",
    "\n",
    "hip_pheno_gwas_female.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_female_cm_231109.txt', index=False, sep = \" \")\n",
    "hip_pheno_gwas_female.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_female_cm_no_header_231109.txt', index=False, header=False, sep = \" \")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### add birth weight/birth canal phenotypes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>BW_devide_oblique_pelvic_inlet_length</th>\n",
       "      <th>BW_devide_pelvic_inlet_width</th>\n",
       "      <th>BW_devide_pelvic_inlet_area</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1001867</td>\n",
       "      <td>1001867</td>\n",
       "      <td>0.441969</td>\n",
       "      <td>0.317354</td>\n",
       "      <td>0.035717</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1002480</td>\n",
       "      <td>1002480</td>\n",
       "      <td>0.707786</td>\n",
       "      <td>0.523539</td>\n",
       "      <td>0.058975</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1002534</td>\n",
       "      <td>1002534</td>\n",
       "      <td>0.567041</td>\n",
       "      <td>0.497296</td>\n",
       "      <td>0.051291</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1003290</td>\n",
       "      <td>1003290</td>\n",
       "      <td>0.525050</td>\n",
       "      <td>0.444291</td>\n",
       "      <td>0.049502</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>1004043</td>\n",
       "      <td>1004043</td>\n",
       "      <td>0.555485</td>\n",
       "      <td>0.458327</td>\n",
       "      <td>0.046308</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16320</th>\n",
       "      <td>6018479</td>\n",
       "      <td>6018479</td>\n",
       "      <td>0.357021</td>\n",
       "      <td>0.286317</td>\n",
       "      <td>0.032538</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16322</th>\n",
       "      <td>6019199</td>\n",
       "      <td>6019199</td>\n",
       "      <td>0.515592</td>\n",
       "      <td>0.456401</td>\n",
       "      <td>0.042802</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16323</th>\n",
       "      <td>6020165</td>\n",
       "      <td>6020165</td>\n",
       "      <td>0.558464</td>\n",
       "      <td>0.478672</td>\n",
       "      <td>0.048623</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16324</th>\n",
       "      <td>6020398</td>\n",
       "      <td>6020398</td>\n",
       "      <td>0.450347</td>\n",
       "      <td>0.406607</td>\n",
       "      <td>0.038858</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16327</th>\n",
       "      <td>6022591</td>\n",
       "      <td>6022591</td>\n",
       "      <td>0.504491</td>\n",
       "      <td>0.443123</td>\n",
       "      <td>0.047439</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>12766 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           FID      IID  BW_devide_oblique_pelvic_inlet_length  \\\n",
       "0      1001867  1001867                               0.441969   \n",
       "1      1002480  1002480                               0.707786   \n",
       "2      1002534  1002534                               0.567041   \n",
       "4      1003290  1003290                               0.525050   \n",
       "5      1004043  1004043                               0.555485   \n",
       "...        ...      ...                                    ...   \n",
       "16320  6018479  6018479                               0.357021   \n",
       "16322  6019199  6019199                               0.515592   \n",
       "16323  6020165  6020165                               0.558464   \n",
       "16324  6020398  6020398                               0.450347   \n",
       "16327  6022591  6022591                               0.504491   \n",
       "\n",
       "       BW_devide_pelvic_inlet_width  BW_devide_pelvic_inlet_area  \n",
       "0                          0.317354                     0.035717  \n",
       "1                          0.523539                     0.058975  \n",
       "2                          0.497296                     0.051291  \n",
       "4                          0.444291                     0.049502  \n",
       "5                          0.458327                     0.046308  \n",
       "...                             ...                          ...  \n",
       "16320                      0.286317                     0.032538  \n",
       "16322                      0.456401                     0.042802  \n",
       "16323                      0.478672                     0.048623  \n",
       "16324                      0.406607                     0.038858  \n",
       "16327                      0.443123                     0.047439  \n",
       "\n",
       "[12766 rows x 5 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hip_pheno_gwas_female = pd.read_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_female_cm_231109.txt', sep = \" \")\n",
    "outcome_pheno = pd.read_csv(\"../UKB_xray_image_info/fids/fid_disease/fid_info_400k_20230916.csv\")[['eid', 'birth_weight_first_child']]\n",
    "\n",
    "hip_pheno_gwas_female = hip_pheno_gwas_female.merge(outcome_pheno, left_on=\"IID\", right_on=\"eid\")\n",
    "hip_pheno_gwas_female.dropna(inplace=True)\n",
    "\n",
    "hip_pheno_gwas_female['BW_devide_oblique_pelvic_inlet_length'] = \\\n",
    "    hip_pheno_gwas_female['birth_weight_first_child'] / hip_pheno_gwas_female['oblique_pelvic_inlet_length']\n",
    "hip_pheno_gwas_female['BW_devide_pelvic_inlet_width'] = \\\n",
    "    hip_pheno_gwas_female['birth_weight_first_child'] / hip_pheno_gwas_female['pelvic_inlet_width']\n",
    "hip_pheno_gwas_female['BW_devide_pelvic_inlet_area'] = \\\n",
    "    hip_pheno_gwas_female['birth_weight_first_child'] / hip_pheno_gwas_female['pelvic_inlet_area']\n",
    "\n",
    "hip_pheno_gwas_female = hip_pheno_gwas_female[['FID', 'IID', 'BW_devide_oblique_pelvic_inlet_length', 'BW_devide_pelvic_inlet_width', 'BW_devide_pelvic_inlet_area']]\n",
    "\n",
    "hip_pheno_gwas_female.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_female_BW_devide_birth_canal_231228.txt', index=False, sep = \" \")\n",
    "hip_pheno_gwas_female.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_female_BW_devide_birth_canal_no_header_231228.txt', index=False, header=False, sep = \" \")\n",
    "\n",
    "hip_pheno_gwas_female"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Divide to male and female 20240104"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>image_id</th>\n",
       "      <th>file_name</th>\n",
       "      <th>eid</th>\n",
       "      <th>sex</th>\n",
       "      <th>standing_height_imaging_visit</th>\n",
       "      <th>weight_imaging_visit</th>\n",
       "      <th>bmi_imaging_visit</th>\n",
       "      <th>age_imaging_visit</th>\n",
       "      <th>pelvic_height</th>\n",
       "      <th>pelvic_width</th>\n",
       "      <th>...</th>\n",
       "      <th>acetabular_diameter</th>\n",
       "      <th>acetabular_inclination</th>\n",
       "      <th>humerus</th>\n",
       "      <th>femur</th>\n",
       "      <th>forearm</th>\n",
       "      <th>tibia</th>\n",
       "      <th>arm_length</th>\n",
       "      <th>leg_length</th>\n",
       "      <th>leg_divide_torso</th>\n",
       "      <th>pelvic_inlet_area</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14366_82</td>\n",
       "      <td>1.2.840.113619.2.110.212038.20190307144208.1.1...</td>\n",
       "      <td>1000951</td>\n",
       "      <td>1.0</td>\n",
       "      <td>166.0</td>\n",
       "      <td>60.8</td>\n",
       "      <td>22.0642</td>\n",
       "      <td>78.0</td>\n",
       "      <td>22.297293</td>\n",
       "      <td>30.113663</td>\n",
       "      <td>...</td>\n",
       "      <td>6.141783</td>\n",
       "      <td>60.747735</td>\n",
       "      <td>30.816804</td>\n",
       "      <td>48.489427</td>\n",
       "      <td>26.856522</td>\n",
       "      <td>40.743734</td>\n",
       "      <td>57.673326</td>\n",
       "      <td>89.233162</td>\n",
       "      <td>1.651141</td>\n",
       "      <td>108.190429</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>01877_82</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20180720150454.1.1...</td>\n",
       "      <td>1001512</td>\n",
       "      <td>1.0</td>\n",
       "      <td>169.0</td>\n",
       "      <td>77.3</td>\n",
       "      <td>27.0649</td>\n",
       "      <td>66.0</td>\n",
       "      <td>23.742297</td>\n",
       "      <td>30.901775</td>\n",
       "      <td>...</td>\n",
       "      <td>6.035855</td>\n",
       "      <td>59.189839</td>\n",
       "      <td>28.115456</td>\n",
       "      <td>44.984030</td>\n",
       "      <td>26.628187</td>\n",
       "      <td>40.081134</td>\n",
       "      <td>54.743642</td>\n",
       "      <td>85.065164</td>\n",
       "      <td>1.449931</td>\n",
       "      <td>102.660623</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>05431_82</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20190328125146.1.1...</td>\n",
       "      <td>1001867</td>\n",
       "      <td>0.0</td>\n",
       "      <td>153.0</td>\n",
       "      <td>56.1</td>\n",
       "      <td>23.9651</td>\n",
       "      <td>75.0</td>\n",
       "      <td>22.226944</td>\n",
       "      <td>30.934457</td>\n",
       "      <td>...</td>\n",
       "      <td>5.531657</td>\n",
       "      <td>59.017487</td>\n",
       "      <td>26.778209</td>\n",
       "      <td>42.934407</td>\n",
       "      <td>23.988531</td>\n",
       "      <td>35.818183</td>\n",
       "      <td>50.766740</td>\n",
       "      <td>78.752591</td>\n",
       "      <td>1.537721</td>\n",
       "      <td>139.989191</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10557_82</td>\n",
       "      <td>1.2.840.113619.2.110.212038.20180228105159.1.1...</td>\n",
       "      <td>1001921</td>\n",
       "      <td>1.0</td>\n",
       "      <td>166.0</td>\n",
       "      <td>64.2</td>\n",
       "      <td>23.2980</td>\n",
       "      <td>62.0</td>\n",
       "      <td>22.861258</td>\n",
       "      <td>28.863430</td>\n",
       "      <td>...</td>\n",
       "      <td>5.834113</td>\n",
       "      <td>58.027399</td>\n",
       "      <td>29.972729</td>\n",
       "      <td>46.587679</td>\n",
       "      <td>28.561098</td>\n",
       "      <td>39.566371</td>\n",
       "      <td>58.533827</td>\n",
       "      <td>86.154050</td>\n",
       "      <td>1.493503</td>\n",
       "      <td>110.520822</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>01561_82</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20180630141714.1.1...</td>\n",
       "      <td>1001950</td>\n",
       "      <td>1.0</td>\n",
       "      <td>165.0</td>\n",
       "      <td>89.6</td>\n",
       "      <td>32.9109</td>\n",
       "      <td>62.0</td>\n",
       "      <td>23.080688</td>\n",
       "      <td>29.385649</td>\n",
       "      <td>...</td>\n",
       "      <td>5.731393</td>\n",
       "      <td>57.335593</td>\n",
       "      <td>29.012823</td>\n",
       "      <td>45.245573</td>\n",
       "      <td>27.292476</td>\n",
       "      <td>39.041382</td>\n",
       "      <td>56.305299</td>\n",
       "      <td>84.286956</td>\n",
       "      <td>1.465162</td>\n",
       "      <td>109.946859</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31129</th>\n",
       "      <td>10039_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20160923114938.1.1...</td>\n",
       "      <td>6021112</td>\n",
       "      <td>1.0</td>\n",
       "      <td>177.0</td>\n",
       "      <td>89.0</td>\n",
       "      <td>28.4082</td>\n",
       "      <td>71.0</td>\n",
       "      <td>25.247090</td>\n",
       "      <td>32.848104</td>\n",
       "      <td>...</td>\n",
       "      <td>6.149698</td>\n",
       "      <td>60.427962</td>\n",
       "      <td>31.540230</td>\n",
       "      <td>50.329159</td>\n",
       "      <td>30.128478</td>\n",
       "      <td>44.856165</td>\n",
       "      <td>61.668708</td>\n",
       "      <td>95.185324</td>\n",
       "      <td>1.583562</td>\n",
       "      <td>134.603261</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31130</th>\n",
       "      <td>14018_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20170705180448.1.1...</td>\n",
       "      <td>6021539</td>\n",
       "      <td>1.0</td>\n",
       "      <td>179.0</td>\n",
       "      <td>71.0</td>\n",
       "      <td>22.1591</td>\n",
       "      <td>69.0</td>\n",
       "      <td>25.232678</td>\n",
       "      <td>30.970184</td>\n",
       "      <td>...</td>\n",
       "      <td>6.504511</td>\n",
       "      <td>59.444884</td>\n",
       "      <td>30.777846</td>\n",
       "      <td>48.408803</td>\n",
       "      <td>28.966865</td>\n",
       "      <td>42.070750</td>\n",
       "      <td>59.744710</td>\n",
       "      <td>90.479553</td>\n",
       "      <td>1.479675</td>\n",
       "      <td>132.776803</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31131</th>\n",
       "      <td>13966_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20170702173904.1.1...</td>\n",
       "      <td>6022591</td>\n",
       "      <td>0.0</td>\n",
       "      <td>156.0</td>\n",
       "      <td>87.7</td>\n",
       "      <td>36.0371</td>\n",
       "      <td>69.0</td>\n",
       "      <td>20.946633</td>\n",
       "      <td>26.922261</td>\n",
       "      <td>...</td>\n",
       "      <td>4.959191</td>\n",
       "      <td>67.937764</td>\n",
       "      <td>28.030085</td>\n",
       "      <td>42.718048</td>\n",
       "      <td>24.880466</td>\n",
       "      <td>37.908781</td>\n",
       "      <td>52.910551</td>\n",
       "      <td>80.626829</td>\n",
       "      <td>1.461921</td>\n",
       "      <td>126.477899</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31132</th>\n",
       "      <td>16309_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20171202183632.1.1...</td>\n",
       "      <td>6023431</td>\n",
       "      <td>0.0</td>\n",
       "      <td>170.0</td>\n",
       "      <td>61.5</td>\n",
       "      <td>21.2803</td>\n",
       "      <td>51.0</td>\n",
       "      <td>22.787944</td>\n",
       "      <td>29.871832</td>\n",
       "      <td>...</td>\n",
       "      <td>5.201943</td>\n",
       "      <td>60.779334</td>\n",
       "      <td>29.276081</td>\n",
       "      <td>45.262378</td>\n",
       "      <td>25.847555</td>\n",
       "      <td>38.992764</td>\n",
       "      <td>55.123637</td>\n",
       "      <td>84.255142</td>\n",
       "      <td>1.384857</td>\n",
       "      <td>150.788401</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31133</th>\n",
       "      <td>12964_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20170418184226.1.1...</td>\n",
       "      <td>6024029</td>\n",
       "      <td>1.0</td>\n",
       "      <td>179.0</td>\n",
       "      <td>80.4</td>\n",
       "      <td>25.0928</td>\n",
       "      <td>50.0</td>\n",
       "      <td>24.323043</td>\n",
       "      <td>29.804727</td>\n",
       "      <td>...</td>\n",
       "      <td>6.171569</td>\n",
       "      <td>59.544963</td>\n",
       "      <td>32.678000</td>\n",
       "      <td>50.968299</td>\n",
       "      <td>29.406987</td>\n",
       "      <td>42.910290</td>\n",
       "      <td>62.084987</td>\n",
       "      <td>93.878589</td>\n",
       "      <td>1.475647</td>\n",
       "      <td>138.796708</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>31134 rows × 32 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       image_id                                          file_name      eid  \\\n",
       "0      14366_82  1.2.840.113619.2.110.212038.20190307144208.1.1...  1000951   \n",
       "1      01877_82  1.2.840.113619.2.110.210419.20180720150454.1.1...  1001512   \n",
       "2      05431_82  1.2.840.113619.2.110.210419.20190328125146.1.1...  1001867   \n",
       "3      10557_82  1.2.840.113619.2.110.212038.20180228105159.1.1...  1001921   \n",
       "4      01561_82  1.2.840.113619.2.110.210419.20180630141714.1.1...  1001950   \n",
       "...         ...                                                ...      ...   \n",
       "31129  10039_93  1.2.840.113619.2.110.210419.20160923114938.1.1...  6021112   \n",
       "31130  14018_93  1.2.840.113619.2.110.210419.20170705180448.1.1...  6021539   \n",
       "31131  13966_93  1.2.840.113619.2.110.210419.20170702173904.1.1...  6022591   \n",
       "31132  16309_93  1.2.840.113619.2.110.210419.20171202183632.1.1...  6023431   \n",
       "31133  12964_93  1.2.840.113619.2.110.210419.20170418184226.1.1...  6024029   \n",
       "\n",
       "       sex  standing_height_imaging_visit  weight_imaging_visit  \\\n",
       "0      1.0                          166.0                  60.8   \n",
       "1      1.0                          169.0                  77.3   \n",
       "2      0.0                          153.0                  56.1   \n",
       "3      1.0                          166.0                  64.2   \n",
       "4      1.0                          165.0                  89.6   \n",
       "...    ...                            ...                   ...   \n",
       "31129  1.0                          177.0                  89.0   \n",
       "31130  1.0                          179.0                  71.0   \n",
       "31131  0.0                          156.0                  87.7   \n",
       "31132  0.0                          170.0                  61.5   \n",
       "31133  1.0                          179.0                  80.4   \n",
       "\n",
       "       bmi_imaging_visit  age_imaging_visit  pelvic_height  pelvic_width  ...  \\\n",
       "0                22.0642               78.0      22.297293     30.113663  ...   \n",
       "1                27.0649               66.0      23.742297     30.901775  ...   \n",
       "2                23.9651               75.0      22.226944     30.934457  ...   \n",
       "3                23.2980               62.0      22.861258     28.863430  ...   \n",
       "4                32.9109               62.0      23.080688     29.385649  ...   \n",
       "...                  ...                ...            ...           ...  ...   \n",
       "31129            28.4082               71.0      25.247090     32.848104  ...   \n",
       "31130            22.1591               69.0      25.232678     30.970184  ...   \n",
       "31131            36.0371               69.0      20.946633     26.922261  ...   \n",
       "31132            21.2803               51.0      22.787944     29.871832  ...   \n",
       "31133            25.0928               50.0      24.323043     29.804727  ...   \n",
       "\n",
       "       acetabular_diameter  acetabular_inclination    humerus      femur  \\\n",
       "0                 6.141783               60.747735  30.816804  48.489427   \n",
       "1                 6.035855               59.189839  28.115456  44.984030   \n",
       "2                 5.531657               59.017487  26.778209  42.934407   \n",
       "3                 5.834113               58.027399  29.972729  46.587679   \n",
       "4                 5.731393               57.335593  29.012823  45.245573   \n",
       "...                    ...                     ...        ...        ...   \n",
       "31129             6.149698               60.427962  31.540230  50.329159   \n",
       "31130             6.504511               59.444884  30.777846  48.408803   \n",
       "31131             4.959191               67.937764  28.030085  42.718048   \n",
       "31132             5.201943               60.779334  29.276081  45.262378   \n",
       "31133             6.171569               59.544963  32.678000  50.968299   \n",
       "\n",
       "         forearm      tibia  arm_length  leg_length  leg_divide_torso  \\\n",
       "0      26.856522  40.743734   57.673326   89.233162          1.651141   \n",
       "1      26.628187  40.081134   54.743642   85.065164          1.449931   \n",
       "2      23.988531  35.818183   50.766740   78.752591          1.537721   \n",
       "3      28.561098  39.566371   58.533827   86.154050          1.493503   \n",
       "4      27.292476  39.041382   56.305299   84.286956          1.465162   \n",
       "...          ...        ...         ...         ...               ...   \n",
       "31129  30.128478  44.856165   61.668708   95.185324          1.583562   \n",
       "31130  28.966865  42.070750   59.744710   90.479553          1.479675   \n",
       "31131  24.880466  37.908781   52.910551   80.626829          1.461921   \n",
       "31132  25.847555  38.992764   55.123637   84.255142          1.384857   \n",
       "31133  29.406987  42.910290   62.084987   93.878589          1.475647   \n",
       "\n",
       "       pelvic_inlet_area  \n",
       "0             108.190429  \n",
       "1             102.660623  \n",
       "2             139.989191  \n",
       "3             110.520822  \n",
       "4             109.946859  \n",
       "...                  ...  \n",
       "31129         134.603261  \n",
       "31130         132.776803  \n",
       "31131         126.477899  \n",
       "31132         150.788401  \n",
       "31133         138.796708  \n",
       "\n",
       "[31134 rows x 32 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hip_pheno = pd.read_csv('key_results/hip_select_pheno_cm_eid_flt.csv'); hip_pheno"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Male dataframe shape: (15277, 32)\n",
      "Female dataframe shape: (15810, 32)\n"
     ]
    }
   ],
   "source": [
    "# filter phenotypes\n",
    "\n",
    "pheno = ['image_id', 'file_name', 'eid', 'sex', 'standing_height_imaging_visit', 'weight_imaging_visit', 'bmi_imaging_visit', 'age_imaging_visit',\n",
    "         'pelvic_height', 'pelvic_width', 'pelvic_inlet_width', 'oblique_pelvic_inlet_length', \n",
    "         'subpubic_angle', 'bi_acetabular_width', 'iliac_isthmus_breadth', 'acetabular_diameter']\n",
    "hip_pheno_flt = hip_pheno[pheno]\n",
    "\n",
    "# male\n",
    "hip_pheno_male_flt = hip_pheno_flt[hip_pheno_flt['sex'] == 1]\n",
    "hip_pheno_male_flt_eid_list = hip_pheno_male_flt[(np.abs(stats.zscore(hip_pheno_male_flt.iloc[:, 8:])) < 4).all(axis=1)]['eid']\n",
    "\n",
    "# z score filter\n",
    "hip_pheno_male = hip_pheno[hip_pheno['eid'].isin(hip_pheno_male_flt_eid_list)]\n",
    "\n",
    "# female\n",
    "hip_pheno_female_flt = hip_pheno_flt[hip_pheno_flt['sex'] == 0]\n",
    "hip_pheno_female_flt_eid_list = hip_pheno_female_flt[(np.abs(stats.zscore(hip_pheno_female_flt.iloc[:, 8:])) < 4).all(axis=1)]['eid']\n",
    "\n",
    "# z score filter\n",
    "hip_pheno_female = hip_pheno[hip_pheno['eid'].isin(hip_pheno_female_flt_eid_list)]\n",
    "\n",
    "print(f\"Male dataframe shape: {hip_pheno_male.shape}\")\n",
    "print(f\"Female dataframe shape: {hip_pheno_female.shape}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "hip_pheno_male.to_csv('key_results/hip_select_pheno_cm_eid_flt_z_flt_male.csv', index=False)\n",
    "hip_pheno_female.to_csv('key_results/hip_select_pheno_cm_eid_flt_z_flt_female.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>image_id</th>\n",
       "      <th>file_name</th>\n",
       "      <th>eid</th>\n",
       "      <th>sex</th>\n",
       "      <th>standing_height_imaging_visit</th>\n",
       "      <th>weight_imaging_visit</th>\n",
       "      <th>bmi_imaging_visit</th>\n",
       "      <th>age_imaging_visit</th>\n",
       "      <th>pelvic_height</th>\n",
       "      <th>pelvic_width</th>\n",
       "      <th>...</th>\n",
       "      <th>trochanter_left2trochanter_right</th>\n",
       "      <th>shoulder_width</th>\n",
       "      <th>arm_devide_torso</th>\n",
       "      <th>bi_acetabular_width</th>\n",
       "      <th>iliac_isthmus_breadth</th>\n",
       "      <th>acetabular_diameter</th>\n",
       "      <th>acetabular_inclination</th>\n",
       "      <th>arm_length</th>\n",
       "      <th>leg_length</th>\n",
       "      <th>pelvic_inlet_area</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14366_82</td>\n",
       "      <td>1.2.840.113619.2.110.212038.20190307144208.1.1...</td>\n",
       "      <td>1000951</td>\n",
       "      <td>1.0</td>\n",
       "      <td>166.0</td>\n",
       "      <td>60.8</td>\n",
       "      <td>22.0642</td>\n",
       "      <td>78.0</td>\n",
       "      <td>22.297293</td>\n",
       "      <td>30.113663</td>\n",
       "      <td>...</td>\n",
       "      <td>30.091232</td>\n",
       "      <td>36.696399</td>\n",
       "      <td>1.067168</td>\n",
       "      <td>20.324819</td>\n",
       "      <td>4.164529</td>\n",
       "      <td>6.141783</td>\n",
       "      <td>60.747735</td>\n",
       "      <td>57.673326</td>\n",
       "      <td>89.233162</td>\n",
       "      <td>108.190429</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>01877_82</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20180720150454.1.1...</td>\n",
       "      <td>1001512</td>\n",
       "      <td>1.0</td>\n",
       "      <td>169.0</td>\n",
       "      <td>77.3</td>\n",
       "      <td>27.0649</td>\n",
       "      <td>66.0</td>\n",
       "      <td>23.742297</td>\n",
       "      <td>30.901775</td>\n",
       "      <td>...</td>\n",
       "      <td>30.813208</td>\n",
       "      <td>39.203992</td>\n",
       "      <td>0.933102</td>\n",
       "      <td>20.756184</td>\n",
       "      <td>4.657824</td>\n",
       "      <td>6.035855</td>\n",
       "      <td>59.189839</td>\n",
       "      <td>54.743642</td>\n",
       "      <td>85.065164</td>\n",
       "      <td>102.660623</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10557_82</td>\n",
       "      <td>1.2.840.113619.2.110.212038.20180228105159.1.1...</td>\n",
       "      <td>1001921</td>\n",
       "      <td>1.0</td>\n",
       "      <td>166.0</td>\n",
       "      <td>64.2</td>\n",
       "      <td>23.2980</td>\n",
       "      <td>62.0</td>\n",
       "      <td>22.861258</td>\n",
       "      <td>28.863430</td>\n",
       "      <td>...</td>\n",
       "      <td>29.777444</td>\n",
       "      <td>33.219853</td>\n",
       "      <td>1.014700</td>\n",
       "      <td>19.784228</td>\n",
       "      <td>4.238295</td>\n",
       "      <td>5.834113</td>\n",
       "      <td>58.027399</td>\n",
       "      <td>58.533827</td>\n",
       "      <td>86.154050</td>\n",
       "      <td>110.520822</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>01561_82</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20180630141714.1.1...</td>\n",
       "      <td>1001950</td>\n",
       "      <td>1.0</td>\n",
       "      <td>165.0</td>\n",
       "      <td>89.6</td>\n",
       "      <td>32.9109</td>\n",
       "      <td>62.0</td>\n",
       "      <td>23.080688</td>\n",
       "      <td>29.385649</td>\n",
       "      <td>...</td>\n",
       "      <td>28.838122</td>\n",
       "      <td>39.066233</td>\n",
       "      <td>0.978756</td>\n",
       "      <td>19.821637</td>\n",
       "      <td>4.438966</td>\n",
       "      <td>5.731393</td>\n",
       "      <td>57.335593</td>\n",
       "      <td>56.305299</td>\n",
       "      <td>84.286956</td>\n",
       "      <td>109.946859</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>07611_82</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20190912120016.1.1...</td>\n",
       "      <td>1002851</td>\n",
       "      <td>1.0</td>\n",
       "      <td>173.0</td>\n",
       "      <td>75.0</td>\n",
       "      <td>25.0593</td>\n",
       "      <td>62.0</td>\n",
       "      <td>25.497024</td>\n",
       "      <td>33.172829</td>\n",
       "      <td>...</td>\n",
       "      <td>30.273697</td>\n",
       "      <td>43.356069</td>\n",
       "      <td>0.972052</td>\n",
       "      <td>20.498618</td>\n",
       "      <td>3.873146</td>\n",
       "      <td>6.167024</td>\n",
       "      <td>62.530271</td>\n",
       "      <td>59.633639</td>\n",
       "      <td>91.311639</td>\n",
       "      <td>149.416046</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31127</th>\n",
       "      <td>03183_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20151006143059.2.1...</td>\n",
       "      <td>6020675</td>\n",
       "      <td>1.0</td>\n",
       "      <td>184.0</td>\n",
       "      <td>73.9</td>\n",
       "      <td>21.8277</td>\n",
       "      <td>57.0</td>\n",
       "      <td>25.369030</td>\n",
       "      <td>32.767891</td>\n",
       "      <td>...</td>\n",
       "      <td>32.669869</td>\n",
       "      <td>40.853551</td>\n",
       "      <td>0.937631</td>\n",
       "      <td>21.810641</td>\n",
       "      <td>4.236680</td>\n",
       "      <td>6.538142</td>\n",
       "      <td>56.217517</td>\n",
       "      <td>61.927156</td>\n",
       "      <td>94.220501</td>\n",
       "      <td>143.697841</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31128</th>\n",
       "      <td>11379_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20161221100256.1.1...</td>\n",
       "      <td>6020722</td>\n",
       "      <td>1.0</td>\n",
       "      <td>173.0</td>\n",
       "      <td>75.8</td>\n",
       "      <td>25.3266</td>\n",
       "      <td>67.0</td>\n",
       "      <td>24.944018</td>\n",
       "      <td>30.547164</td>\n",
       "      <td>...</td>\n",
       "      <td>30.356562</td>\n",
       "      <td>39.199608</td>\n",
       "      <td>0.960849</td>\n",
       "      <td>20.506793</td>\n",
       "      <td>4.054677</td>\n",
       "      <td>5.990853</td>\n",
       "      <td>57.652643</td>\n",
       "      <td>58.651204</td>\n",
       "      <td>88.133506</td>\n",
       "      <td>132.004910</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31129</th>\n",
       "      <td>10039_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20160923114938.1.1...</td>\n",
       "      <td>6021112</td>\n",
       "      <td>1.0</td>\n",
       "      <td>177.0</td>\n",
       "      <td>89.0</td>\n",
       "      <td>28.4082</td>\n",
       "      <td>71.0</td>\n",
       "      <td>25.247090</td>\n",
       "      <td>32.848104</td>\n",
       "      <td>...</td>\n",
       "      <td>30.995640</td>\n",
       "      <td>41.118337</td>\n",
       "      <td>1.025959</td>\n",
       "      <td>20.686542</td>\n",
       "      <td>4.014342</td>\n",
       "      <td>6.149698</td>\n",
       "      <td>60.427962</td>\n",
       "      <td>61.668708</td>\n",
       "      <td>95.185324</td>\n",
       "      <td>134.603261</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31130</th>\n",
       "      <td>14018_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20170705180448.1.1...</td>\n",
       "      <td>6021539</td>\n",
       "      <td>1.0</td>\n",
       "      <td>179.0</td>\n",
       "      <td>71.0</td>\n",
       "      <td>22.1591</td>\n",
       "      <td>69.0</td>\n",
       "      <td>25.232678</td>\n",
       "      <td>30.970184</td>\n",
       "      <td>...</td>\n",
       "      <td>31.116022</td>\n",
       "      <td>38.579135</td>\n",
       "      <td>0.977047</td>\n",
       "      <td>20.729378</td>\n",
       "      <td>4.381059</td>\n",
       "      <td>6.504511</td>\n",
       "      <td>59.444884</td>\n",
       "      <td>59.744710</td>\n",
       "      <td>90.479553</td>\n",
       "      <td>132.776803</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31133</th>\n",
       "      <td>12964_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20170418184226.1.1...</td>\n",
       "      <td>6024029</td>\n",
       "      <td>1.0</td>\n",
       "      <td>179.0</td>\n",
       "      <td>80.4</td>\n",
       "      <td>25.0928</td>\n",
       "      <td>50.0</td>\n",
       "      <td>24.323043</td>\n",
       "      <td>29.804727</td>\n",
       "      <td>...</td>\n",
       "      <td>30.125650</td>\n",
       "      <td>39.486231</td>\n",
       "      <td>0.975894</td>\n",
       "      <td>20.408280</td>\n",
       "      <td>4.298543</td>\n",
       "      <td>6.171569</td>\n",
       "      <td>59.544963</td>\n",
       "      <td>62.084987</td>\n",
       "      <td>93.878589</td>\n",
       "      <td>138.796708</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>15277 rows × 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       image_id                                          file_name      eid  \\\n",
       "0      14366_82  1.2.840.113619.2.110.212038.20190307144208.1.1...  1000951   \n",
       "1      01877_82  1.2.840.113619.2.110.210419.20180720150454.1.1...  1001512   \n",
       "3      10557_82  1.2.840.113619.2.110.212038.20180228105159.1.1...  1001921   \n",
       "4      01561_82  1.2.840.113619.2.110.210419.20180630141714.1.1...  1001950   \n",
       "6      07611_82  1.2.840.113619.2.110.210419.20190912120016.1.1...  1002851   \n",
       "...         ...                                                ...      ...   \n",
       "31127  03183_93  1.2.840.113619.2.110.210419.20151006143059.2.1...  6020675   \n",
       "31128  11379_93  1.2.840.113619.2.110.210419.20161221100256.1.1...  6020722   \n",
       "31129  10039_93  1.2.840.113619.2.110.210419.20160923114938.1.1...  6021112   \n",
       "31130  14018_93  1.2.840.113619.2.110.210419.20170705180448.1.1...  6021539   \n",
       "31133  12964_93  1.2.840.113619.2.110.210419.20170418184226.1.1...  6024029   \n",
       "\n",
       "       sex  standing_height_imaging_visit  weight_imaging_visit  \\\n",
       "0      1.0                          166.0                  60.8   \n",
       "1      1.0                          169.0                  77.3   \n",
       "3      1.0                          166.0                  64.2   \n",
       "4      1.0                          165.0                  89.6   \n",
       "6      1.0                          173.0                  75.0   \n",
       "...    ...                            ...                   ...   \n",
       "31127  1.0                          184.0                  73.9   \n",
       "31128  1.0                          173.0                  75.8   \n",
       "31129  1.0                          177.0                  89.0   \n",
       "31130  1.0                          179.0                  71.0   \n",
       "31133  1.0                          179.0                  80.4   \n",
       "\n",
       "       bmi_imaging_visit  age_imaging_visit  pelvic_height  pelvic_width  ...  \\\n",
       "0                22.0642               78.0      22.297293     30.113663  ...   \n",
       "1                27.0649               66.0      23.742297     30.901775  ...   \n",
       "3                23.2980               62.0      22.861258     28.863430  ...   \n",
       "4                32.9109               62.0      23.080688     29.385649  ...   \n",
       "6                25.0593               62.0      25.497024     33.172829  ...   \n",
       "...                  ...                ...            ...           ...  ...   \n",
       "31127            21.8277               57.0      25.369030     32.767891  ...   \n",
       "31128            25.3266               67.0      24.944018     30.547164  ...   \n",
       "31129            28.4082               71.0      25.247090     32.848104  ...   \n",
       "31130            22.1591               69.0      25.232678     30.970184  ...   \n",
       "31133            25.0928               50.0      24.323043     29.804727  ...   \n",
       "\n",
       "       trochanter_left2trochanter_right  shoulder_width  arm_devide_torso  \\\n",
       "0                             30.091232       36.696399          1.067168   \n",
       "1                             30.813208       39.203992          0.933102   \n",
       "3                             29.777444       33.219853          1.014700   \n",
       "4                             28.838122       39.066233          0.978756   \n",
       "6                             30.273697       43.356069          0.972052   \n",
       "...                                 ...             ...               ...   \n",
       "31127                         32.669869       40.853551          0.937631   \n",
       "31128                         30.356562       39.199608          0.960849   \n",
       "31129                         30.995640       41.118337          1.025959   \n",
       "31130                         31.116022       38.579135          0.977047   \n",
       "31133                         30.125650       39.486231          0.975894   \n",
       "\n",
       "       bi_acetabular_width  iliac_isthmus_breadth  acetabular_diameter  \\\n",
       "0                20.324819               4.164529             6.141783   \n",
       "1                20.756184               4.657824             6.035855   \n",
       "3                19.784228               4.238295             5.834113   \n",
       "4                19.821637               4.438966             5.731393   \n",
       "6                20.498618               3.873146             6.167024   \n",
       "...                    ...                    ...                  ...   \n",
       "31127            21.810641               4.236680             6.538142   \n",
       "31128            20.506793               4.054677             5.990853   \n",
       "31129            20.686542               4.014342             6.149698   \n",
       "31130            20.729378               4.381059             6.504511   \n",
       "31133            20.408280               4.298543             6.171569   \n",
       "\n",
       "       acetabular_inclination  arm_length  leg_length  pelvic_inlet_area  \n",
       "0                   60.747735   57.673326   89.233162         108.190429  \n",
       "1                   59.189839   54.743642   85.065164         102.660623  \n",
       "3                   58.027399   58.533827   86.154050         110.520822  \n",
       "4                   57.335593   56.305299   84.286956         109.946859  \n",
       "6                   62.530271   59.633639   91.311639         149.416046  \n",
       "...                       ...         ...         ...                ...  \n",
       "31127               56.217517   61.927156   94.220501         143.697841  \n",
       "31128               57.652643   58.651204   88.133506         132.004910  \n",
       "31129               60.427962   61.668708   95.185324         134.603261  \n",
       "31130               59.444884   59.744710   90.479553         132.776803  \n",
       "31133               59.544963   62.084987   93.878589         138.796708  \n",
       "\n",
       "[15277 rows x 26 columns]"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hip_pheno_male"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Male dataframe shape: (15277, 11)\n",
      "Female dataframe shape: (15810, 11)\n"
     ]
    }
   ],
   "source": [
    "pheno = ['image_id', 'file_name', 'eid', 'sex', 'standing_height_imaging_visit', 'weight_imaging_visit', 'bmi_imaging_visit', 'age_imaging_visit',\n",
    "         'pelvic_height', 'pelvic_width', 'pelvic_inlet_width', 'oblique_pelvic_inlet_length', \n",
    "         'subpubic_angle', 'bi_acetabular_width', 'iliac_isthmus_breadth', 'acetabular_diameter',\n",
    "         'pelvic_inlet_area']\n",
    "\n",
    "# male dataframe for gwas preparation\n",
    "hip_pheno_gwas_male = hip_pheno_male[pheno]\n",
    "\n",
    "hip_pheno_gwas_male = hip_pheno_male.iloc[:, 8:]\n",
    "\n",
    "ids = hip_pheno_male['eid'].tolist()\n",
    "\n",
    "hip_pheno_gwas_male['FID'] = ids\n",
    "hip_pheno_gwas_male['IID'] = ids\n",
    "\n",
    "# put FID and IID to the first two columns\n",
    "cols = hip_pheno_gwas_male.columns\n",
    "cols = cols[-2:].tolist() + cols[:-2].tolist()\n",
    "hip_pheno_gwas_male = hip_pheno_gwas_male[cols]\n",
    "\n",
    "# female dataframe for gwas preparation\n",
    "hip_pheno_gwas_female = hip_pheno_female[pheno]\n",
    "\n",
    "hip_pheno_gwas_female = hip_pheno_female.iloc[:, 8:]\n",
    "\n",
    "ids = hip_pheno_female['eid'].tolist()\n",
    "\n",
    "hip_pheno_gwas_female['FID'] = ids\n",
    "hip_pheno_gwas_female['IID'] = ids\n",
    "\n",
    "# put FID and IID to the first two columns\n",
    "cols = hip_pheno_gwas_female.columns\n",
    "cols = cols[-2:].tolist() + cols[:-2].tolist()\n",
    "hip_pheno_gwas_female = hip_pheno_gwas_female[cols]\n",
    "\n",
    "print(f\"Male dataframe shape: {hip_pheno_gwas_male.shape}\")\n",
    "print(f\"Female dataframe shape: {hip_pheno_gwas_female.shape}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>pelvic_height</th>\n",
       "      <th>pelvic_width</th>\n",
       "      <th>pelvic_inlet_width</th>\n",
       "      <th>oblique_pelvic_inlet_length</th>\n",
       "      <th>subpubic_angle</th>\n",
       "      <th>bi_acetabular_width</th>\n",
       "      <th>iliac_isthmus_breadth</th>\n",
       "      <th>acetabular_diameter</th>\n",
       "      <th>pelvic_inlet_area</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1001867</td>\n",
       "      <td>1001867</td>\n",
       "      <td>22.226944</td>\n",
       "      <td>30.934457</td>\n",
       "      <td>15.755275</td>\n",
       "      <td>11.313022</td>\n",
       "      <td>139.787157</td>\n",
       "      <td>21.701987</td>\n",
       "      <td>3.718747</td>\n",
       "      <td>5.531657</td>\n",
       "      <td>139.989191</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>1002480</td>\n",
       "      <td>1002480</td>\n",
       "      <td>21.362371</td>\n",
       "      <td>31.549402</td>\n",
       "      <td>15.280627</td>\n",
       "      <td>11.302851</td>\n",
       "      <td>140.236585</td>\n",
       "      <td>21.687472</td>\n",
       "      <td>4.314733</td>\n",
       "      <td>5.847069</td>\n",
       "      <td>135.649767</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>1003186</td>\n",
       "      <td>1003186</td>\n",
       "      <td>21.017710</td>\n",
       "      <td>28.052667</td>\n",
       "      <td>13.993619</td>\n",
       "      <td>12.712840</td>\n",
       "      <td>143.098207</td>\n",
       "      <td>20.418713</td>\n",
       "      <td>3.433443</td>\n",
       "      <td>5.261873</td>\n",
       "      <td>139.721260</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>1003290</td>\n",
       "      <td>1003290</td>\n",
       "      <td>22.347859</td>\n",
       "      <td>27.953417</td>\n",
       "      <td>13.504669</td>\n",
       "      <td>11.427479</td>\n",
       "      <td>139.543307</td>\n",
       "      <td>20.447377</td>\n",
       "      <td>3.826107</td>\n",
       "      <td>5.330001</td>\n",
       "      <td>121.206033</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>1004043</td>\n",
       "      <td>1004043</td>\n",
       "      <td>24.095581</td>\n",
       "      <td>29.541051</td>\n",
       "      <td>15.272943</td>\n",
       "      <td>12.601611</td>\n",
       "      <td>139.605451</td>\n",
       "      <td>21.657533</td>\n",
       "      <td>3.656215</td>\n",
       "      <td>5.571609</td>\n",
       "      <td>151.160622</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31124</th>\n",
       "      <td>6020398</td>\n",
       "      <td>6020398</td>\n",
       "      <td>22.443374</td>\n",
       "      <td>28.848989</td>\n",
       "      <td>14.756264</td>\n",
       "      <td>13.323053</td>\n",
       "      <td>158.945855</td>\n",
       "      <td>21.532009</td>\n",
       "      <td>3.299413</td>\n",
       "      <td>5.076335</td>\n",
       "      <td>154.408090</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31125</th>\n",
       "      <td>6020527</td>\n",
       "      <td>6020527</td>\n",
       "      <td>22.381443</td>\n",
       "      <td>30.477040</td>\n",
       "      <td>14.106894</td>\n",
       "      <td>11.087194</td>\n",
       "      <td>122.996570</td>\n",
       "      <td>19.636336</td>\n",
       "      <td>4.190870</td>\n",
       "      <td>5.575862</td>\n",
       "      <td>122.840882</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31126</th>\n",
       "      <td>6020585</td>\n",
       "      <td>6020585</td>\n",
       "      <td>22.561943</td>\n",
       "      <td>31.630376</td>\n",
       "      <td>15.780600</td>\n",
       "      <td>12.424455</td>\n",
       "      <td>150.116885</td>\n",
       "      <td>21.300465</td>\n",
       "      <td>3.549324</td>\n",
       "      <td>5.044865</td>\n",
       "      <td>153.989374</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31131</th>\n",
       "      <td>6022591</td>\n",
       "      <td>6022591</td>\n",
       "      <td>20.946633</td>\n",
       "      <td>26.922261</td>\n",
       "      <td>13.540263</td>\n",
       "      <td>11.893171</td>\n",
       "      <td>143.403514</td>\n",
       "      <td>19.152575</td>\n",
       "      <td>3.211435</td>\n",
       "      <td>4.959191</td>\n",
       "      <td>126.477899</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31132</th>\n",
       "      <td>6023431</td>\n",
       "      <td>6023431</td>\n",
       "      <td>22.787944</td>\n",
       "      <td>29.871832</td>\n",
       "      <td>13.586974</td>\n",
       "      <td>14.130428</td>\n",
       "      <td>157.984881</td>\n",
       "      <td>19.990121</td>\n",
       "      <td>3.935749</td>\n",
       "      <td>5.201943</td>\n",
       "      <td>150.788401</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>15810 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           FID      IID  pelvic_height  pelvic_width  pelvic_inlet_width  \\\n",
       "2      1001867  1001867      22.226944     30.934457           15.755275   \n",
       "5      1002480  1002480      21.362371     31.549402           15.280627   \n",
       "8      1003186  1003186      21.017710     28.052667           13.993619   \n",
       "10     1003290  1003290      22.347859     27.953417           13.504669   \n",
       "11     1004043  1004043      24.095581     29.541051           15.272943   \n",
       "...        ...      ...            ...           ...                 ...   \n",
       "31124  6020398  6020398      22.443374     28.848989           14.756264   \n",
       "31125  6020527  6020527      22.381443     30.477040           14.106894   \n",
       "31126  6020585  6020585      22.561943     31.630376           15.780600   \n",
       "31131  6022591  6022591      20.946633     26.922261           13.540263   \n",
       "31132  6023431  6023431      22.787944     29.871832           13.586974   \n",
       "\n",
       "       oblique_pelvic_inlet_length  subpubic_angle  bi_acetabular_width  \\\n",
       "2                        11.313022      139.787157            21.701987   \n",
       "5                        11.302851      140.236585            21.687472   \n",
       "8                        12.712840      143.098207            20.418713   \n",
       "10                       11.427479      139.543307            20.447377   \n",
       "11                       12.601611      139.605451            21.657533   \n",
       "...                            ...             ...                  ...   \n",
       "31124                    13.323053      158.945855            21.532009   \n",
       "31125                    11.087194      122.996570            19.636336   \n",
       "31126                    12.424455      150.116885            21.300465   \n",
       "31131                    11.893171      143.403514            19.152575   \n",
       "31132                    14.130428      157.984881            19.990121   \n",
       "\n",
       "       iliac_isthmus_breadth  acetabular_diameter  pelvic_inlet_area  \n",
       "2                   3.718747             5.531657         139.989191  \n",
       "5                   4.314733             5.847069         135.649767  \n",
       "8                   3.433443             5.261873         139.721260  \n",
       "10                  3.826107             5.330001         121.206033  \n",
       "11                  3.656215             5.571609         151.160622  \n",
       "...                      ...                  ...                ...  \n",
       "31124               3.299413             5.076335         154.408090  \n",
       "31125               4.190870             5.575862         122.840882  \n",
       "31126               3.549324             5.044865         153.989374  \n",
       "31131               3.211435             4.959191         126.477899  \n",
       "31132               3.935749             5.201943         150.788401  \n",
       "\n",
       "[15810 rows x 11 columns]"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hip_pheno_gwas_female"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [],
   "source": [
    "hip_pheno_gwas_male.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_male_cm_20240104.txt', index=False, sep = \" \")\n",
    "hip_pheno_gwas_male.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_male_cm_no_header_20240104.txt', index=False, header=False, sep = \" \")\n",
    "\n",
    "hip_pheno_gwas_female.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_female_cm_20240104.txt', index=False, sep = \" \")\n",
    "hip_pheno_gwas_female.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_female_cm_no_header_20240104.txt', index=False, header=False, sep = \" \")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [],
   "source": [
    "hip_pheno_gwas_male[['FID', 'IID']].to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_male_eids_20240104.txt', index=False, header=False, sep=' ')\n",
    "hip_pheno_gwas_female[['FID', 'IID']].to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_female_eids_20240104.txt', index=False, header=False, sep=' ')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### add birth weight/birth canal phenotypes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [],
   "source": [
    "outcome_pheno = pd.read_csv(\"../UKB_xray_image_info/fids/fid_disease/fid_info_400k_20240104.csv\")[['eid', 'birth_weight_first_child']]\n",
    "\n",
    "hip_pheno_gwas_female = hip_pheno_gwas_female.merge(outcome_pheno, left_on=\"IID\", right_on=\"eid\")\n",
    "hip_pheno_gwas_female.dropna(inplace=True)\n",
    "\n",
    "hip_pheno_gwas_female['BW_devide_oblique_pelvic_inlet_length'] = \\\n",
    "    hip_pheno_gwas_female['birth_weight_first_child'] / hip_pheno_gwas_female['oblique_pelvic_inlet_length']\n",
    "hip_pheno_gwas_female['BW_devide_pelvic_inlet_width'] = \\\n",
    "    hip_pheno_gwas_female['birth_weight_first_child'] / hip_pheno_gwas_female['pelvic_inlet_width']\n",
    "hip_pheno_gwas_female['BW_devide_pelvic_inlet_area'] = \\\n",
    "    hip_pheno_gwas_female['birth_weight_first_child'] / hip_pheno_gwas_female['pelvic_inlet_area']\n",
    "\n",
    "hip_pheno_gwas_female = hip_pheno_gwas_female[['FID', 'IID', 'BW_devide_oblique_pelvic_inlet_length', 'BW_devide_pelvic_inlet_width', 'BW_devide_pelvic_inlet_area']]\n",
    "\n",
    "hip_pheno_gwas_female.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_female_BW_birth_canal_20240104.txt', index=False, sep = \" \")\n",
    "hip_pheno_gwas_female.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_female_BW_birth_canal_no_header_20240104.txt', index=False, header=False, sep = \" \")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>BW_devide_oblique_pelvic_inlet_length</th>\n",
       "      <th>BW_devide_pelvic_inlet_width</th>\n",
       "      <th>BW_devide_pelvic_inlet_area</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1001867</td>\n",
       "      <td>1001867</td>\n",
       "      <td>0.441969</td>\n",
       "      <td>0.317354</td>\n",
       "      <td>0.035717</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1002480</td>\n",
       "      <td>1002480</td>\n",
       "      <td>0.707786</td>\n",
       "      <td>0.523539</td>\n",
       "      <td>0.058975</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1003290</td>\n",
       "      <td>1003290</td>\n",
       "      <td>0.525050</td>\n",
       "      <td>0.444291</td>\n",
       "      <td>0.049502</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1004043</td>\n",
       "      <td>1004043</td>\n",
       "      <td>0.555485</td>\n",
       "      <td>0.458327</td>\n",
       "      <td>0.046308</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1004300</td>\n",
       "      <td>1004300</td>\n",
       "      <td>0.667039</td>\n",
       "      <td>0.492191</td>\n",
       "      <td>0.059717</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12357</th>\n",
       "      <td>6018479</td>\n",
       "      <td>6018479</td>\n",
       "      <td>0.357021</td>\n",
       "      <td>0.286317</td>\n",
       "      <td>0.032538</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12358</th>\n",
       "      <td>6019199</td>\n",
       "      <td>6019199</td>\n",
       "      <td>0.515592</td>\n",
       "      <td>0.456401</td>\n",
       "      <td>0.042802</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12359</th>\n",
       "      <td>6020165</td>\n",
       "      <td>6020165</td>\n",
       "      <td>0.558464</td>\n",
       "      <td>0.478672</td>\n",
       "      <td>0.048623</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12360</th>\n",
       "      <td>6020398</td>\n",
       "      <td>6020398</td>\n",
       "      <td>0.450347</td>\n",
       "      <td>0.406607</td>\n",
       "      <td>0.038858</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12361</th>\n",
       "      <td>6022591</td>\n",
       "      <td>6022591</td>\n",
       "      <td>0.504491</td>\n",
       "      <td>0.443123</td>\n",
       "      <td>0.047439</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>12362 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           FID      IID  BW_devide_oblique_pelvic_inlet_length  \\\n",
       "0      1001867  1001867                               0.441969   \n",
       "1      1002480  1002480                               0.707786   \n",
       "2      1003290  1003290                               0.525050   \n",
       "3      1004043  1004043                               0.555485   \n",
       "4      1004300  1004300                               0.667039   \n",
       "...        ...      ...                                    ...   \n",
       "12357  6018479  6018479                               0.357021   \n",
       "12358  6019199  6019199                               0.515592   \n",
       "12359  6020165  6020165                               0.558464   \n",
       "12360  6020398  6020398                               0.450347   \n",
       "12361  6022591  6022591                               0.504491   \n",
       "\n",
       "       BW_devide_pelvic_inlet_width  BW_devide_pelvic_inlet_area  \n",
       "0                          0.317354                     0.035717  \n",
       "1                          0.523539                     0.058975  \n",
       "2                          0.444291                     0.049502  \n",
       "3                          0.458327                     0.046308  \n",
       "4                          0.492191                     0.059717  \n",
       "...                             ...                          ...  \n",
       "12357                      0.286317                     0.032538  \n",
       "12358                      0.456401                     0.042802  \n",
       "12359                      0.478672                     0.048623  \n",
       "12360                      0.406607                     0.038858  \n",
       "12361                      0.443123                     0.047439  \n",
       "\n",
       "[12362 rows x 5 columns]"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_female_BW_birth_canal_20240104.txt', sep = \" \")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [],
   "source": [
    "hip_pheno_male[['eid', 'eid']].to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_eids_male.txt', index=False, header=False, sep=' ')\n",
    "hip_pheno_female[['eid', 'eid']].to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_eids_female.txt', index=False, header=False, sep=' ')"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### add two new phenotypes (head or shoulder / pelvic inlet width)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [],
   "source": [
    "# male\n",
    "hip_pheno_gwas_male = pd.read_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_male_cm_230713.txt', sep = \" \")\n",
    "\n",
    "hip_pheno_gwas_male['head_divide_inlet_width'] = hip_pheno_gwas_male['ear_left2ear_right'] / hip_pheno_gwas_male['pelvic_inlet_width']\n",
    "hip_pheno_gwas_male['shoulder_divide_inlet_width'] = hip_pheno_gwas_male['shoulder_width'] / hip_pheno_gwas_male['pelvic_inlet_width']\n",
    "\n",
    "hip_pheno_gwas_male_new = hip_pheno_gwas_male[['FID', 'IID', 'head_divide_inlet_width', 'shoulder_divide_inlet_width']]\n",
    "\n",
    "hip_pheno_gwas_male.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_male_cm_230829.txt', index=False, sep = \" \")\n",
    "hip_pheno_gwas_male_new.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_two_new_pheno_gwas_male_cm_230829.txt', index=False, sep = \" \")\n",
    "\n",
    "# female\n",
    "hip_pheno_gwas_female = pd.read_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_female_cm_230713.txt', sep = \" \")\n",
    "\n",
    "hip_pheno_gwas_female['head_divide_inlet_width'] = hip_pheno_gwas_female['ear_left2ear_right'] / hip_pheno_gwas_female['pelvic_inlet_width']\n",
    "hip_pheno_gwas_female['shoulder_divide_inlet_width'] = hip_pheno_gwas_female['shoulder_width'] / hip_pheno_gwas_female['pelvic_inlet_width']\n",
    "\n",
    "hip_pheno_gwas_female_new = hip_pheno_gwas_female[['FID', 'IID', 'head_divide_inlet_width', 'shoulder_divide_inlet_width']]\n",
    "\n",
    "hip_pheno_gwas_female.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_female_cm_230829.txt', index=False, sep = \" \")\n",
    "hip_pheno_gwas_female_new.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_two_new_pheno_gwas_female_cm_230829.txt', index=False, sep = \" \")"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### add five new phenotypes (2 for head or shoulder / oblique pelvic inlet length, 1 for pelvic inlet area, 2 for head area or shoulder area / pelvic inlet area)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# male\n",
    "hip_pheno_gwas_male = pd.read_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_male_cm_230829.txt', sep = \" \")\n",
    "# 2 for head or shoulder / oblique pelvic inlet length\n",
    "hip_pheno_gwas_male['head_divide_oblique_inlet_length'] = hip_pheno_gwas_male['ear_left2ear_right'] / hip_pheno_gwas_male['oblique_pelvic_inlet_length']\n",
    "hip_pheno_gwas_male['shoulder_divide_oblique_inlet_length'] = hip_pheno_gwas_male['shoulder_width'] / hip_pheno_gwas_male['oblique_pelvic_inlet_length']\n",
    "\n",
    "# 1 for pelvic inlet area\n",
    "hip_pheno_gwas_male['pelvic_inlet_area'] = (hip_pheno_gwas_male['pelvic_inlet_width'] / 2) * (hip_pheno_gwas_male['oblique_pelvic_inlet_length'] / 2) * math.pi\n",
    "\n",
    "# 2 for head or shoulder / pelvic inlet area\n",
    "hip_pheno_gwas_male['head_area_divide_pelvic_inlet_area'] = ((hip_pheno_gwas_male['ear_left2ear_right'] / 2)**2 * math.pi) / hip_pheno_gwas_male['pelvic_inlet_area']\n",
    "hip_pheno_gwas_male['shoulder_area_divide_pelvic_inlet_area'] = ((hip_pheno_gwas_male['shoulder_width'] / 2)**2 * math.pi) / hip_pheno_gwas_male['pelvic_inlet_area']\n",
    "\n",
    "hip_pheno_gwas_male_new = hip_pheno_gwas_male[['FID', 'IID', \n",
    "                                                'head_divide_oblique_inlet_length', 'shoulder_divide_oblique_inlet_length', \n",
    "                                                'pelvic_inlet_area',\n",
    "                                                'head_area_divide_pelvic_inlet_area', 'shoulder_area_divide_pelvic_inlet_area']]\n",
    "\n",
    "hip_pheno_gwas_male.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_male_cm_230908.txt', index=False, sep = \" \")\n",
    "hip_pheno_gwas_male.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_male_cm_no_header_230908.txt', index=False, header=False, sep = \" \")\n",
    "hip_pheno_gwas_male_new.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_two_new_pheno_gwas_male_cm_230908.txt', index=False, sep = \" \")\n",
    "\n",
    "# female\n",
    "hip_pheno_gwas_female = pd.read_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_female_cm_230829.txt', sep = \" \")\n",
    "# 2 for head or shoulder / oblique pelvic inlet length\n",
    "hip_pheno_gwas_female['head_divide_oblique_inlet_length'] = hip_pheno_gwas_female['ear_left2ear_right'] / hip_pheno_gwas_female['oblique_pelvic_inlet_length']\n",
    "hip_pheno_gwas_female['shoulder_divide_oblique_inlet_length'] = hip_pheno_gwas_female['shoulder_width'] / hip_pheno_gwas_female['oblique_pelvic_inlet_length']\n",
    "\n",
    "# 1 for pelvic inlet area\n",
    "hip_pheno_gwas_female['pelvic_inlet_area'] = (hip_pheno_gwas_female['pelvic_inlet_width'] / 2) * (hip_pheno_gwas_female['oblique_pelvic_inlet_length'] / 2) * math.pi\n",
    "\n",
    "# 2 for head or shoulder / pelvic inlet area\n",
    "hip_pheno_gwas_female['head_area_divide_pelvic_inlet_area'] = ((hip_pheno_gwas_female['ear_left2ear_right'] / 2)**2 * math.pi) / hip_pheno_gwas_female['pelvic_inlet_area']\n",
    "hip_pheno_gwas_female['shoulder_area_divide_pelvic_inlet_area'] = ((hip_pheno_gwas_female['shoulder_width'] / 2)**2 * math.pi) / hip_pheno_gwas_female['pelvic_inlet_area']\n",
    "\n",
    "hip_pheno_gwas_female_new = hip_pheno_gwas_female[['FID', 'IID', \n",
    "                                                'head_divide_oblique_inlet_length', 'shoulder_divide_oblique_inlet_length', \n",
    "                                                'pelvic_inlet_area',\n",
    "                                                'head_area_divide_pelvic_inlet_area', 'shoulder_area_divide_pelvic_inlet_area']]\n",
    "\n",
    "hip_pheno_gwas_female.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_female_cm_230908.txt', index=False, sep = \" \")\n",
    "hip_pheno_gwas_female.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_female_cm_no_header_230908.txt', index=False, header=False, sep = \" \")\n",
    "hip_pheno_gwas_female_new.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_two_new_pheno_gwas_female_cm_230908.txt', index=False, sep = \" \")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### add 3 new phenotype for female: birth weight / birth canal phenos (3 phenotypes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "hip_pheno_gwas_female = pd.read_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_female_cm_230908.txt', sep = \" \")\n",
    "\n",
    "outcome_pheno = pd.read_csv(\"../UKB_xray_image_info/fids/fid_disease/fid_info_400k_20230916.csv\")[['eid', 'birth_weight_first_child']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "hip_pheno_gwas_female = hip_pheno_gwas_female.merge(outcome_pheno, left_on=\"IID\", right_on=\"eid\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "hip_pheno_gwas_female.dropna(inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "hip_pheno_gwas_female['BW_devide_oblique_pelvic_inlet_length'] = \\\n",
    "    hip_pheno_gwas_female['birth_weight_first_child'] / hip_pheno_gwas_female['oblique_pelvic_inlet_length']\n",
    "hip_pheno_gwas_female['BW_devide_pelvic_inlet_width'] = \\\n",
    "    hip_pheno_gwas_female['birth_weight_first_child'] / hip_pheno_gwas_female['pelvic_inlet_width']\n",
    "hip_pheno_gwas_female['BW_devide_pelvic_inlet_area'] = \\\n",
    "    hip_pheno_gwas_female['birth_weight_first_child'] / hip_pheno_gwas_female['pelvic_inlet_area']\n",
    "\n",
    "hip_pheno_gwas_female = hip_pheno_gwas_female[['FID', 'IID', 'BW_devide_oblique_pelvic_inlet_length', 'BW_devide_pelvic_inlet_width', 'BW_devide_pelvic_inlet_area']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>BW_devide_oblique_pelvic_inlet_length</th>\n",
       "      <th>BW_devide_pelvic_inlet_width</th>\n",
       "      <th>BW_devide_pelvic_inlet_area</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1000276</td>\n",
       "      <td>1000276</td>\n",
       "      <td>0.526777</td>\n",
       "      <td>0.372393</td>\n",
       "      <td>0.041628</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1000742</td>\n",
       "      <td>1000742</td>\n",
       "      <td>0.499781</td>\n",
       "      <td>0.399397</td>\n",
       "      <td>0.042359</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1001025</td>\n",
       "      <td>1001025</td>\n",
       "      <td>0.298090</td>\n",
       "      <td>0.245285</td>\n",
       "      <td>0.023274</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1001091</td>\n",
       "      <td>1001091</td>\n",
       "      <td>0.499900</td>\n",
       "      <td>0.345763</td>\n",
       "      <td>0.044015</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1001867</td>\n",
       "      <td>1001867</td>\n",
       "      <td>0.441969</td>\n",
       "      <td>0.317354</td>\n",
       "      <td>0.035717</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15424</th>\n",
       "      <td>6021897</td>\n",
       "      <td>6021897</td>\n",
       "      <td>0.628862</td>\n",
       "      <td>0.485557</td>\n",
       "      <td>0.055540</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15425</th>\n",
       "      <td>6022510</td>\n",
       "      <td>6022510</td>\n",
       "      <td>0.777012</td>\n",
       "      <td>0.657959</td>\n",
       "      <td>0.072326</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15426</th>\n",
       "      <td>6022591</td>\n",
       "      <td>6022591</td>\n",
       "      <td>0.504491</td>\n",
       "      <td>0.443123</td>\n",
       "      <td>0.047439</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15427</th>\n",
       "      <td>6022917</td>\n",
       "      <td>6022917</td>\n",
       "      <td>0.487995</td>\n",
       "      <td>0.386972</td>\n",
       "      <td>0.040073</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15429</th>\n",
       "      <td>6023703</td>\n",
       "      <td>6023703</td>\n",
       "      <td>0.628216</td>\n",
       "      <td>0.502256</td>\n",
       "      <td>0.057391</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>12059 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           FID      IID  BW_devide_oblique_pelvic_inlet_length  \\\n",
       "0      1000276  1000276                               0.526777   \n",
       "1      1000742  1000742                               0.499781   \n",
       "2      1001025  1001025                               0.298090   \n",
       "3      1001091  1001091                               0.499900   \n",
       "4      1001867  1001867                               0.441969   \n",
       "...        ...      ...                                    ...   \n",
       "15424  6021897  6021897                               0.628862   \n",
       "15425  6022510  6022510                               0.777012   \n",
       "15426  6022591  6022591                               0.504491   \n",
       "15427  6022917  6022917                               0.487995   \n",
       "15429  6023703  6023703                               0.628216   \n",
       "\n",
       "       BW_devide_pelvic_inlet_width  BW_devide_pelvic_inlet_area  \n",
       "0                          0.372393                     0.041628  \n",
       "1                          0.399397                     0.042359  \n",
       "2                          0.245285                     0.023274  \n",
       "3                          0.345763                     0.044015  \n",
       "4                          0.317354                     0.035717  \n",
       "...                             ...                          ...  \n",
       "15424                      0.485557                     0.055540  \n",
       "15425                      0.657959                     0.072326  \n",
       "15426                      0.443123                     0.047439  \n",
       "15427                      0.386972                     0.040073  \n",
       "15429                      0.502256                     0.057391  \n",
       "\n",
       "[12059 rows x 5 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hip_pheno_gwas_female"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "hip_pheno_gwas_female.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_female_BW_devide_birth_canal_231005.txt', index=False, sep = \" \")\n",
    "hip_pheno_gwas_female.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_female_BW_devide_birth_canal_no_header_231005.txt', index=False, header=False, sep = \" \")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### add 2 new phenotype as postive control: 1. arm:leg, 2. leg:torso"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "hip_pheno = pd.read_csv('key_results/hip_pheno_23_cm.csv')\n",
    "\n",
    "image_ids = pd.read_csv('key_results/hip_select_pheno_cm_eid_flt.csv')['image_id'].tolist()\n",
    "\n",
    "# filter individuals\n",
    "hip_pheno = hip_pheno[hip_pheno['image_id'].isin(image_ids)]\n",
    "\n",
    "# new phenotypes\n",
    "hip_pheno['arm_divide_leg'] = ((hip_pheno['arm_123_left'] + hip_pheno['arm_123_right']) / 2) / ((hip_pheno['leg_123_left'] + hip_pheno['leg_123_right']) / 2)\n",
    "hip_pheno['leg_divide_torso'] = ((hip_pheno['leg_123_left'] + hip_pheno['leg_123_right']) / 2) / hip_pheno['torso_length']\n",
    "add_pheno = hip_pheno[['eid', 'arm_divide_leg', 'leg_divide_torso']]\n",
    "\n",
    "# process male\n",
    "\n",
    "# merge previous phenotypes\n",
    "hip_pheno_gwas_male = pd.read_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_male_cm_230908.txt', sep = \" \")\n",
    "hip_pheno_gwas_male = hip_pheno_gwas_male.merge(add_pheno, left_on='FID', right_on='eid', how='inner')\n",
    "hip_pheno_gwas_male.drop(columns=['eid'], inplace=True)\n",
    "\n",
    "hip_pheno_gwas_male.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_male_cm_231011.txt', index=False, sep = \" \")\n",
    "hip_pheno_gwas_male.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_male_no_header_231011.txt', index=False, header=False, sep = \" \")\n",
    "\n",
    "hip_pheno_gwas_male[['FID', 'IID', 'arm_divide_leg', 'leg_divide_torso']].to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_two_new_pheno_gwas_male_231011.txt', index=False, sep = \" \")\n",
    "hip_pheno_gwas_male[['FID', 'IID', 'arm_divide_leg', 'leg_divide_torso']].to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_two_new_pheno_gwas_male_no_header_231011.txt', index=False, header=False, sep = \" \")\n",
    "\n",
    "# process female\n",
    "\n",
    "# merge previous phenotypes\n",
    "hip_pheno_gwas_female = pd.read_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_female_cm_230908.txt', sep = \" \")\n",
    "hip_pheno_gwas_female = hip_pheno_gwas_female.merge(add_pheno, left_on='FID', right_on='eid', how='inner')\n",
    "hip_pheno_gwas_female.drop(columns=['eid'], inplace=True)\n",
    "\n",
    "hip_pheno_gwas_female.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_female_cm_231011.txt', index=False, sep = \" \")\n",
    "hip_pheno_gwas_female.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_female_cm_no_header_231011.txt', index=False, header=False, sep = \" \")\n",
    "\n",
    "hip_pheno_gwas_female[['FID', 'IID', 'arm_divide_leg', 'leg_divide_torso']].to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_two_new_pheno_gwas_female_231011.txt', index=False, sep = \" \")\n",
    "hip_pheno_gwas_female[['FID', 'IID', 'arm_divide_leg', 'leg_divide_torso']].to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_two_new_pheno_gwas_female_no_header_231011.txt', index=False, header=False, sep = \" \")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>pelvic_height</th>\n",
       "      <th>pelvic_width</th>\n",
       "      <th>pelvic_inlet_width</th>\n",
       "      <th>iliac_flare_ratio</th>\n",
       "      <th>oblique_pelvic_inlet_length</th>\n",
       "      <th>iliac_flare_angle</th>\n",
       "      <th>subpubic_angle</th>\n",
       "      <th>ear_left2ear_right</th>\n",
       "      <th>...</th>\n",
       "      <th>acetabular_inclination</th>\n",
       "      <th>head_divide_inlet_width</th>\n",
       "      <th>shoulder_divide_inlet_width</th>\n",
       "      <th>head_divide_oblique_inlet_length</th>\n",
       "      <th>shoulder_divide_oblique_inlet_length</th>\n",
       "      <th>pelvic_inlet_area</th>\n",
       "      <th>head_area_divide_pelvic_inlet_area</th>\n",
       "      <th>shoulder_area_divide_pelvic_inlet_area</th>\n",
       "      <th>arm_divide_leg</th>\n",
       "      <th>leg_divide_torso</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1000951</td>\n",
       "      <td>1000951</td>\n",
       "      <td>22.297293</td>\n",
       "      <td>30.113663</td>\n",
       "      <td>13.937174</td>\n",
       "      <td>0.462819</td>\n",
       "      <td>9.883806</td>\n",
       "      <td>97.533181</td>\n",
       "      <td>116.365190</td>\n",
       "      <td>14.112233</td>\n",
       "      <td>...</td>\n",
       "      <td>60.747735</td>\n",
       "      <td>1.012561</td>\n",
       "      <td>2.632987</td>\n",
       "      <td>1.427814</td>\n",
       "      <td>3.712780</td>\n",
       "      <td>108.190429</td>\n",
       "      <td>1.445748</td>\n",
       "      <td>9.775701</td>\n",
       "      <td>0.646322</td>\n",
       "      <td>1.651141</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1001512</td>\n",
       "      <td>1001512</td>\n",
       "      <td>23.742297</td>\n",
       "      <td>30.901775</td>\n",
       "      <td>13.669524</td>\n",
       "      <td>0.442354</td>\n",
       "      <td>9.562262</td>\n",
       "      <td>95.714405</td>\n",
       "      <td>117.491690</td>\n",
       "      <td>13.903518</td>\n",
       "      <td>...</td>\n",
       "      <td>59.189839</td>\n",
       "      <td>1.017118</td>\n",
       "      <td>2.867985</td>\n",
       "      <td>1.453999</td>\n",
       "      <td>4.099866</td>\n",
       "      <td>102.660623</td>\n",
       "      <td>1.478888</td>\n",
       "      <td>11.758355</td>\n",
       "      <td>0.643549</td>\n",
       "      <td>1.449931</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1001719</td>\n",
       "      <td>1001719</td>\n",
       "      <td>27.576413</td>\n",
       "      <td>34.451075</td>\n",
       "      <td>15.479994</td>\n",
       "      <td>0.449333</td>\n",
       "      <td>12.957804</td>\n",
       "      <td>95.958279</td>\n",
       "      <td>116.185643</td>\n",
       "      <td>13.372049</td>\n",
       "      <td>...</td>\n",
       "      <td>55.647944</td>\n",
       "      <td>0.863828</td>\n",
       "      <td>2.791644</td>\n",
       "      <td>1.031969</td>\n",
       "      <td>3.335027</td>\n",
       "      <td>157.540451</td>\n",
       "      <td>0.891443</td>\n",
       "      <td>9.310209</td>\n",
       "      <td>0.630605</td>\n",
       "      <td>1.551221</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1001921</td>\n",
       "      <td>1001921</td>\n",
       "      <td>22.861258</td>\n",
       "      <td>28.863430</td>\n",
       "      <td>14.217750</td>\n",
       "      <td>0.492587</td>\n",
       "      <td>9.897451</td>\n",
       "      <td>95.473293</td>\n",
       "      <td>116.159185</td>\n",
       "      <td>14.480994</td>\n",
       "      <td>...</td>\n",
       "      <td>58.027399</td>\n",
       "      <td>1.018515</td>\n",
       "      <td>2.336506</td>\n",
       "      <td>1.463103</td>\n",
       "      <td>3.356405</td>\n",
       "      <td>110.520822</td>\n",
       "      <td>1.490193</td>\n",
       "      <td>7.842259</td>\n",
       "      <td>0.679409</td>\n",
       "      <td>1.493503</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1001950</td>\n",
       "      <td>1001950</td>\n",
       "      <td>23.080688</td>\n",
       "      <td>29.385649</td>\n",
       "      <td>13.583558</td>\n",
       "      <td>0.462251</td>\n",
       "      <td>10.305746</td>\n",
       "      <td>95.952879</td>\n",
       "      <td>130.647738</td>\n",
       "      <td>14.042468</td>\n",
       "      <td>...</td>\n",
       "      <td>57.335593</td>\n",
       "      <td>1.033784</td>\n",
       "      <td>2.875994</td>\n",
       "      <td>1.362586</td>\n",
       "      <td>3.790724</td>\n",
       "      <td>109.946859</td>\n",
       "      <td>1.408620</td>\n",
       "      <td>10.902099</td>\n",
       "      <td>0.668019</td>\n",
       "      <td>1.465162</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14840</th>\n",
       "      <td>6023890</td>\n",
       "      <td>6023890</td>\n",
       "      <td>24.006938</td>\n",
       "      <td>30.013155</td>\n",
       "      <td>13.591668</td>\n",
       "      <td>0.452857</td>\n",
       "      <td>11.950734</td>\n",
       "      <td>94.685360</td>\n",
       "      <td>123.267297</td>\n",
       "      <td>14.496029</td>\n",
       "      <td>...</td>\n",
       "      <td>57.046345</td>\n",
       "      <td>1.066538</td>\n",
       "      <td>2.979412</td>\n",
       "      <td>1.212982</td>\n",
       "      <td>3.388509</td>\n",
       "      <td>127.572540</td>\n",
       "      <td>1.293692</td>\n",
       "      <td>10.095763</td>\n",
       "      <td>0.626011</td>\n",
       "      <td>1.470443</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14841</th>\n",
       "      <td>6024029</td>\n",
       "      <td>6024029</td>\n",
       "      <td>24.323043</td>\n",
       "      <td>29.804727</td>\n",
       "      <td>13.883682</td>\n",
       "      <td>0.465821</td>\n",
       "      <td>12.728717</td>\n",
       "      <td>85.308035</td>\n",
       "      <td>124.424809</td>\n",
       "      <td>13.271879</td>\n",
       "      <td>...</td>\n",
       "      <td>59.544963</td>\n",
       "      <td>0.955934</td>\n",
       "      <td>2.844075</td>\n",
       "      <td>1.042672</td>\n",
       "      <td>3.102138</td>\n",
       "      <td>138.796708</td>\n",
       "      <td>0.996726</td>\n",
       "      <td>8.822711</td>\n",
       "      <td>0.661333</td>\n",
       "      <td>1.475647</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14842</th>\n",
       "      <td>6024037</td>\n",
       "      <td>6024037</td>\n",
       "      <td>24.695284</td>\n",
       "      <td>31.178125</td>\n",
       "      <td>14.634467</td>\n",
       "      <td>0.469383</td>\n",
       "      <td>12.533393</td>\n",
       "      <td>95.014076</td>\n",
       "      <td>130.924304</td>\n",
       "      <td>14.358949</td>\n",
       "      <td>...</td>\n",
       "      <td>58.682502</td>\n",
       "      <td>0.981173</td>\n",
       "      <td>2.870301</td>\n",
       "      <td>1.145655</td>\n",
       "      <td>3.351472</td>\n",
       "      <td>144.057355</td>\n",
       "      <td>1.124087</td>\n",
       "      <td>9.619733</td>\n",
       "      <td>0.659582</td>\n",
       "      <td>1.570756</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14843</th>\n",
       "      <td>6024257</td>\n",
       "      <td>6024257</td>\n",
       "      <td>22.849056</td>\n",
       "      <td>30.738358</td>\n",
       "      <td>14.076906</td>\n",
       "      <td>0.457959</td>\n",
       "      <td>9.722564</td>\n",
       "      <td>102.209266</td>\n",
       "      <td>101.631995</td>\n",
       "      <td>13.457582</td>\n",
       "      <td>...</td>\n",
       "      <td>54.167545</td>\n",
       "      <td>0.956004</td>\n",
       "      <td>2.741431</td>\n",
       "      <td>1.384160</td>\n",
       "      <td>3.969207</td>\n",
       "      <td>107.492432</td>\n",
       "      <td>1.323263</td>\n",
       "      <td>10.881308</td>\n",
       "      <td>0.661403</td>\n",
       "      <td>1.434394</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14844</th>\n",
       "      <td>6024532</td>\n",
       "      <td>6024532</td>\n",
       "      <td>24.004784</td>\n",
       "      <td>32.248290</td>\n",
       "      <td>15.234202</td>\n",
       "      <td>0.472403</td>\n",
       "      <td>10.533469</td>\n",
       "      <td>105.193136</td>\n",
       "      <td>105.144526</td>\n",
       "      <td>14.392064</td>\n",
       "      <td>...</td>\n",
       "      <td>55.439957</td>\n",
       "      <td>0.944721</td>\n",
       "      <td>2.592005</td>\n",
       "      <td>1.366318</td>\n",
       "      <td>3.748730</td>\n",
       "      <td>126.032051</td>\n",
       "      <td>1.290788</td>\n",
       "      <td>9.716727</td>\n",
       "      <td>0.658262</td>\n",
       "      <td>1.432338</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>14845 rows × 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           FID      IID  pelvic_height  pelvic_width  pelvic_inlet_width  \\\n",
       "0      1000951  1000951      22.297293     30.113663           13.937174   \n",
       "1      1001512  1001512      23.742297     30.901775           13.669524   \n",
       "2      1001719  1001719      27.576413     34.451075           15.479994   \n",
       "3      1001921  1001921      22.861258     28.863430           14.217750   \n",
       "4      1001950  1001950      23.080688     29.385649           13.583558   \n",
       "...        ...      ...            ...           ...                 ...   \n",
       "14840  6023890  6023890      24.006938     30.013155           13.591668   \n",
       "14841  6024029  6024029      24.323043     29.804727           13.883682   \n",
       "14842  6024037  6024037      24.695284     31.178125           14.634467   \n",
       "14843  6024257  6024257      22.849056     30.738358           14.076906   \n",
       "14844  6024532  6024532      24.004784     32.248290           15.234202   \n",
       "\n",
       "       iliac_flare_ratio  oblique_pelvic_inlet_length  iliac_flare_angle  \\\n",
       "0               0.462819                     9.883806          97.533181   \n",
       "1               0.442354                     9.562262          95.714405   \n",
       "2               0.449333                    12.957804          95.958279   \n",
       "3               0.492587                     9.897451          95.473293   \n",
       "4               0.462251                    10.305746          95.952879   \n",
       "...                  ...                          ...                ...   \n",
       "14840           0.452857                    11.950734          94.685360   \n",
       "14841           0.465821                    12.728717          85.308035   \n",
       "14842           0.469383                    12.533393          95.014076   \n",
       "14843           0.457959                     9.722564         102.209266   \n",
       "14844           0.472403                    10.533469         105.193136   \n",
       "\n",
       "       subpubic_angle  ear_left2ear_right  ...  acetabular_inclination  \\\n",
       "0          116.365190           14.112233  ...               60.747735   \n",
       "1          117.491690           13.903518  ...               59.189839   \n",
       "2          116.185643           13.372049  ...               55.647944   \n",
       "3          116.159185           14.480994  ...               58.027399   \n",
       "4          130.647738           14.042468  ...               57.335593   \n",
       "...               ...                 ...  ...                     ...   \n",
       "14840      123.267297           14.496029  ...               57.046345   \n",
       "14841      124.424809           13.271879  ...               59.544963   \n",
       "14842      130.924304           14.358949  ...               58.682502   \n",
       "14843      101.631995           13.457582  ...               54.167545   \n",
       "14844      105.144526           14.392064  ...               55.439957   \n",
       "\n",
       "       head_divide_inlet_width  shoulder_divide_inlet_width  \\\n",
       "0                     1.012561                     2.632987   \n",
       "1                     1.017118                     2.867985   \n",
       "2                     0.863828                     2.791644   \n",
       "3                     1.018515                     2.336506   \n",
       "4                     1.033784                     2.875994   \n",
       "...                        ...                          ...   \n",
       "14840                 1.066538                     2.979412   \n",
       "14841                 0.955934                     2.844075   \n",
       "14842                 0.981173                     2.870301   \n",
       "14843                 0.956004                     2.741431   \n",
       "14844                 0.944721                     2.592005   \n",
       "\n",
       "       head_divide_oblique_inlet_length  shoulder_divide_oblique_inlet_length  \\\n",
       "0                              1.427814                              3.712780   \n",
       "1                              1.453999                              4.099866   \n",
       "2                              1.031969                              3.335027   \n",
       "3                              1.463103                              3.356405   \n",
       "4                              1.362586                              3.790724   \n",
       "...                                 ...                                   ...   \n",
       "14840                          1.212982                              3.388509   \n",
       "14841                          1.042672                              3.102138   \n",
       "14842                          1.145655                              3.351472   \n",
       "14843                          1.384160                              3.969207   \n",
       "14844                          1.366318                              3.748730   \n",
       "\n",
       "       pelvic_inlet_area  head_area_divide_pelvic_inlet_area  \\\n",
       "0             108.190429                            1.445748   \n",
       "1             102.660623                            1.478888   \n",
       "2             157.540451                            0.891443   \n",
       "3             110.520822                            1.490193   \n",
       "4             109.946859                            1.408620   \n",
       "...                  ...                                 ...   \n",
       "14840         127.572540                            1.293692   \n",
       "14841         138.796708                            0.996726   \n",
       "14842         144.057355                            1.124087   \n",
       "14843         107.492432                            1.323263   \n",
       "14844         126.032051                            1.290788   \n",
       "\n",
       "       shoulder_area_divide_pelvic_inlet_area  arm_divide_leg  \\\n",
       "0                                    9.775701        0.646322   \n",
       "1                                   11.758355        0.643549   \n",
       "2                                    9.310209        0.630605   \n",
       "3                                    7.842259        0.679409   \n",
       "4                                   10.902099        0.668019   \n",
       "...                                       ...             ...   \n",
       "14840                               10.095763        0.626011   \n",
       "14841                                8.822711        0.661333   \n",
       "14842                                9.619733        0.659582   \n",
       "14843                               10.881308        0.661403   \n",
       "14844                                9.716727        0.658262   \n",
       "\n",
       "       leg_divide_torso  \n",
       "0              1.651141  \n",
       "1              1.449931  \n",
       "2              1.551221  \n",
       "3              1.493503  \n",
       "4              1.465162  \n",
       "...                 ...  \n",
       "14840          1.470443  \n",
       "14841          1.475647  \n",
       "14842          1.570756  \n",
       "14843          1.434394  \n",
       "14844          1.432338  \n",
       "\n",
       "[14845 rows x 25 columns]"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hip_pheno_gwas_male"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>pelvic_height</th>\n",
       "      <th>pelvic_width</th>\n",
       "      <th>pelvic_inlet_width</th>\n",
       "      <th>iliac_flare_ratio</th>\n",
       "      <th>oblique_pelvic_inlet_length</th>\n",
       "      <th>iliac_flare_angle</th>\n",
       "      <th>subpubic_angle</th>\n",
       "      <th>ear_left2ear_right</th>\n",
       "      <th>...</th>\n",
       "      <th>acetabular_inclination</th>\n",
       "      <th>head_divide_inlet_width</th>\n",
       "      <th>shoulder_divide_inlet_width</th>\n",
       "      <th>head_divide_oblique_inlet_length</th>\n",
       "      <th>shoulder_divide_oblique_inlet_length</th>\n",
       "      <th>pelvic_inlet_area</th>\n",
       "      <th>head_area_divide_pelvic_inlet_area</th>\n",
       "      <th>shoulder_area_divide_pelvic_inlet_area</th>\n",
       "      <th>arm_divide_leg</th>\n",
       "      <th>leg_divide_torso</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1000276</td>\n",
       "      <td>1000276</td>\n",
       "      <td>22.979831</td>\n",
       "      <td>31.953718</td>\n",
       "      <td>16.111993</td>\n",
       "      <td>0.504229</td>\n",
       "      <td>11.390012</td>\n",
       "      <td>101.719769</td>\n",
       "      <td>142.678044</td>\n",
       "      <td>13.641873</td>\n",
       "      <td>...</td>\n",
       "      <td>63.070462</td>\n",
       "      <td>0.846691</td>\n",
       "      <td>2.190174</td>\n",
       "      <td>1.197705</td>\n",
       "      <td>3.098159</td>\n",
       "      <td>144.132961</td>\n",
       "      <td>1.014085</td>\n",
       "      <td>6.785507</td>\n",
       "      <td>0.644230</td>\n",
       "      <td>1.486725</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1000742</td>\n",
       "      <td>1000742</td>\n",
       "      <td>22.388605</td>\n",
       "      <td>29.873284</td>\n",
       "      <td>15.022635</td>\n",
       "      <td>0.502879</td>\n",
       "      <td>12.005250</td>\n",
       "      <td>90.615136</td>\n",
       "      <td>141.231641</td>\n",
       "      <td>13.535399</td>\n",
       "      <td>...</td>\n",
       "      <td>58.032066</td>\n",
       "      <td>0.901000</td>\n",
       "      <td>2.449625</td>\n",
       "      <td>1.127457</td>\n",
       "      <td>3.065311</td>\n",
       "      <td>141.646949</td>\n",
       "      <td>1.015839</td>\n",
       "      <td>7.508865</td>\n",
       "      <td>0.618930</td>\n",
       "      <td>1.442429</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1001025</td>\n",
       "      <td>1001025</td>\n",
       "      <td>22.512247</td>\n",
       "      <td>33.454000</td>\n",
       "      <td>16.307558</td>\n",
       "      <td>0.487462</td>\n",
       "      <td>13.418752</td>\n",
       "      <td>102.935020</td>\n",
       "      <td>159.703035</td>\n",
       "      <td>12.863914</td>\n",
       "      <td>...</td>\n",
       "      <td>61.323156</td>\n",
       "      <td>0.788831</td>\n",
       "      <td>2.154641</td>\n",
       "      <td>0.958652</td>\n",
       "      <td>2.618495</td>\n",
       "      <td>171.866378</td>\n",
       "      <td>0.756215</td>\n",
       "      <td>5.641917</td>\n",
       "      <td>0.662228</td>\n",
       "      <td>1.352832</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1001091</td>\n",
       "      <td>1001091</td>\n",
       "      <td>22.574433</td>\n",
       "      <td>30.432563</td>\n",
       "      <td>14.460793</td>\n",
       "      <td>0.475175</td>\n",
       "      <td>10.001994</td>\n",
       "      <td>93.043305</td>\n",
       "      <td>131.343366</td>\n",
       "      <td>12.557424</td>\n",
       "      <td>...</td>\n",
       "      <td>61.729635</td>\n",
       "      <td>0.868377</td>\n",
       "      <td>2.535743</td>\n",
       "      <td>1.255492</td>\n",
       "      <td>3.666154</td>\n",
       "      <td>113.597447</td>\n",
       "      <td>1.090241</td>\n",
       "      <td>9.296422</td>\n",
       "      <td>0.644521</td>\n",
       "      <td>1.674855</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1001867</td>\n",
       "      <td>1001867</td>\n",
       "      <td>22.226944</td>\n",
       "      <td>30.934457</td>\n",
       "      <td>15.755275</td>\n",
       "      <td>0.509312</td>\n",
       "      <td>11.313022</td>\n",
       "      <td>101.418560</td>\n",
       "      <td>139.787157</td>\n",
       "      <td>13.029582</td>\n",
       "      <td>...</td>\n",
       "      <td>59.017487</td>\n",
       "      <td>0.826998</td>\n",
       "      <td>2.279199</td>\n",
       "      <td>1.151733</td>\n",
       "      <td>3.174166</td>\n",
       "      <td>139.989191</td>\n",
       "      <td>0.952481</td>\n",
       "      <td>7.234555</td>\n",
       "      <td>0.644636</td>\n",
       "      <td>1.537721</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15427</th>\n",
       "      <td>6022917</td>\n",
       "      <td>6022917</td>\n",
       "      <td>22.579634</td>\n",
       "      <td>30.210156</td>\n",
       "      <td>15.505008</td>\n",
       "      <td>0.513238</td>\n",
       "      <td>12.295198</td>\n",
       "      <td>91.622387</td>\n",
       "      <td>148.537788</td>\n",
       "      <td>12.820154</td>\n",
       "      <td>...</td>\n",
       "      <td>64.222794</td>\n",
       "      <td>0.826840</td>\n",
       "      <td>2.066416</td>\n",
       "      <td>1.042696</td>\n",
       "      <td>2.605879</td>\n",
       "      <td>149.726056</td>\n",
       "      <td>0.862142</td>\n",
       "      <td>5.384830</td>\n",
       "      <td>0.619555</td>\n",
       "      <td>1.551718</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15428</th>\n",
       "      <td>6023431</td>\n",
       "      <td>6023431</td>\n",
       "      <td>22.787944</td>\n",
       "      <td>29.871832</td>\n",
       "      <td>13.586974</td>\n",
       "      <td>0.454842</td>\n",
       "      <td>14.130428</td>\n",
       "      <td>90.205445</td>\n",
       "      <td>157.984881</td>\n",
       "      <td>12.309345</td>\n",
       "      <td>...</td>\n",
       "      <td>60.779334</td>\n",
       "      <td>0.905967</td>\n",
       "      <td>2.676822</td>\n",
       "      <td>0.871123</td>\n",
       "      <td>2.573872</td>\n",
       "      <td>150.788401</td>\n",
       "      <td>0.789209</td>\n",
       "      <td>6.889799</td>\n",
       "      <td>0.654247</td>\n",
       "      <td>1.384857</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15429</th>\n",
       "      <td>6023703</td>\n",
       "      <td>6023703</td>\n",
       "      <td>21.410998</td>\n",
       "      <td>24.992278</td>\n",
       "      <td>13.937128</td>\n",
       "      <td>0.557657</td>\n",
       "      <td>11.142661</td>\n",
       "      <td>87.546753</td>\n",
       "      <td>128.013696</td>\n",
       "      <td>13.242774</td>\n",
       "      <td>...</td>\n",
       "      <td>68.414739</td>\n",
       "      <td>0.950180</td>\n",
       "      <td>2.503646</td>\n",
       "      <td>1.188475</td>\n",
       "      <td>3.131536</td>\n",
       "      <td>121.969737</td>\n",
       "      <td>1.129265</td>\n",
       "      <td>7.840259</td>\n",
       "      <td>0.621974</td>\n",
       "      <td>1.436634</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15430</th>\n",
       "      <td>6023791</td>\n",
       "      <td>6023791</td>\n",
       "      <td>24.149063</td>\n",
       "      <td>30.566491</td>\n",
       "      <td>16.398959</td>\n",
       "      <td>0.536501</td>\n",
       "      <td>13.166596</td>\n",
       "      <td>85.839271</td>\n",
       "      <td>141.797305</td>\n",
       "      <td>12.818428</td>\n",
       "      <td>...</td>\n",
       "      <td>65.242741</td>\n",
       "      <td>0.781661</td>\n",
       "      <td>2.315466</td>\n",
       "      <td>0.973557</td>\n",
       "      <td>2.883907</td>\n",
       "      <td>169.581969</td>\n",
       "      <td>0.760991</td>\n",
       "      <td>6.677589</td>\n",
       "      <td>0.631468</td>\n",
       "      <td>1.515224</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15431</th>\n",
       "      <td>6023985</td>\n",
       "      <td>6023985</td>\n",
       "      <td>23.103796</td>\n",
       "      <td>30.177756</td>\n",
       "      <td>14.600016</td>\n",
       "      <td>0.483801</td>\n",
       "      <td>12.205770</td>\n",
       "      <td>91.554095</td>\n",
       "      <td>129.150241</td>\n",
       "      <td>13.763440</td>\n",
       "      <td>...</td>\n",
       "      <td>62.802667</td>\n",
       "      <td>0.942700</td>\n",
       "      <td>2.517889</td>\n",
       "      <td>1.127617</td>\n",
       "      <td>3.011790</td>\n",
       "      <td>139.961444</td>\n",
       "      <td>1.063005</td>\n",
       "      <td>7.583353</td>\n",
       "      <td>0.643413</td>\n",
       "      <td>1.508679</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>15432 rows × 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           FID      IID  pelvic_height  pelvic_width  pelvic_inlet_width  \\\n",
       "0      1000276  1000276      22.979831     31.953718           16.111993   \n",
       "1      1000742  1000742      22.388605     29.873284           15.022635   \n",
       "2      1001025  1001025      22.512247     33.454000           16.307558   \n",
       "3      1001091  1001091      22.574433     30.432563           14.460793   \n",
       "4      1001867  1001867      22.226944     30.934457           15.755275   \n",
       "...        ...      ...            ...           ...                 ...   \n",
       "15427  6022917  6022917      22.579634     30.210156           15.505008   \n",
       "15428  6023431  6023431      22.787944     29.871832           13.586974   \n",
       "15429  6023703  6023703      21.410998     24.992278           13.937128   \n",
       "15430  6023791  6023791      24.149063     30.566491           16.398959   \n",
       "15431  6023985  6023985      23.103796     30.177756           14.600016   \n",
       "\n",
       "       iliac_flare_ratio  oblique_pelvic_inlet_length  iliac_flare_angle  \\\n",
       "0               0.504229                    11.390012         101.719769   \n",
       "1               0.502879                    12.005250          90.615136   \n",
       "2               0.487462                    13.418752         102.935020   \n",
       "3               0.475175                    10.001994          93.043305   \n",
       "4               0.509312                    11.313022         101.418560   \n",
       "...                  ...                          ...                ...   \n",
       "15427           0.513238                    12.295198          91.622387   \n",
       "15428           0.454842                    14.130428          90.205445   \n",
       "15429           0.557657                    11.142661          87.546753   \n",
       "15430           0.536501                    13.166596          85.839271   \n",
       "15431           0.483801                    12.205770          91.554095   \n",
       "\n",
       "       subpubic_angle  ear_left2ear_right  ...  acetabular_inclination  \\\n",
       "0          142.678044           13.641873  ...               63.070462   \n",
       "1          141.231641           13.535399  ...               58.032066   \n",
       "2          159.703035           12.863914  ...               61.323156   \n",
       "3          131.343366           12.557424  ...               61.729635   \n",
       "4          139.787157           13.029582  ...               59.017487   \n",
       "...               ...                 ...  ...                     ...   \n",
       "15427      148.537788           12.820154  ...               64.222794   \n",
       "15428      157.984881           12.309345  ...               60.779334   \n",
       "15429      128.013696           13.242774  ...               68.414739   \n",
       "15430      141.797305           12.818428  ...               65.242741   \n",
       "15431      129.150241           13.763440  ...               62.802667   \n",
       "\n",
       "       head_divide_inlet_width  shoulder_divide_inlet_width  \\\n",
       "0                     0.846691                     2.190174   \n",
       "1                     0.901000                     2.449625   \n",
       "2                     0.788831                     2.154641   \n",
       "3                     0.868377                     2.535743   \n",
       "4                     0.826998                     2.279199   \n",
       "...                        ...                          ...   \n",
       "15427                 0.826840                     2.066416   \n",
       "15428                 0.905967                     2.676822   \n",
       "15429                 0.950180                     2.503646   \n",
       "15430                 0.781661                     2.315466   \n",
       "15431                 0.942700                     2.517889   \n",
       "\n",
       "       head_divide_oblique_inlet_length  shoulder_divide_oblique_inlet_length  \\\n",
       "0                              1.197705                              3.098159   \n",
       "1                              1.127457                              3.065311   \n",
       "2                              0.958652                              2.618495   \n",
       "3                              1.255492                              3.666154   \n",
       "4                              1.151733                              3.174166   \n",
       "...                                 ...                                   ...   \n",
       "15427                          1.042696                              2.605879   \n",
       "15428                          0.871123                              2.573872   \n",
       "15429                          1.188475                              3.131536   \n",
       "15430                          0.973557                              2.883907   \n",
       "15431                          1.127617                              3.011790   \n",
       "\n",
       "       pelvic_inlet_area  head_area_divide_pelvic_inlet_area  \\\n",
       "0             144.132961                            1.014085   \n",
       "1             141.646949                            1.015839   \n",
       "2             171.866378                            0.756215   \n",
       "3             113.597447                            1.090241   \n",
       "4             139.989191                            0.952481   \n",
       "...                  ...                                 ...   \n",
       "15427         149.726056                            0.862142   \n",
       "15428         150.788401                            0.789209   \n",
       "15429         121.969737                            1.129265   \n",
       "15430         169.581969                            0.760991   \n",
       "15431         139.961444                            1.063005   \n",
       "\n",
       "       shoulder_area_divide_pelvic_inlet_area  arm_divide_leg  \\\n",
       "0                                    6.785507        0.644230   \n",
       "1                                    7.508865        0.618930   \n",
       "2                                    5.641917        0.662228   \n",
       "3                                    9.296422        0.644521   \n",
       "4                                    7.234555        0.644636   \n",
       "...                                       ...             ...   \n",
       "15427                                5.384830        0.619555   \n",
       "15428                                6.889799        0.654247   \n",
       "15429                                7.840259        0.621974   \n",
       "15430                                6.677589        0.631468   \n",
       "15431                                7.583353        0.643413   \n",
       "\n",
       "       leg_divide_torso  \n",
       "0              1.486725  \n",
       "1              1.442429  \n",
       "2              1.352832  \n",
       "3              1.674855  \n",
       "4              1.537721  \n",
       "...                 ...  \n",
       "15427          1.551718  \n",
       "15428          1.384857  \n",
       "15429          1.436634  \n",
       "15430          1.515224  \n",
       "15431          1.508679  \n",
       "\n",
       "[15432 rows x 25 columns]"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hip_pheno_gwas_female"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### add bi-acetabular width phentoype"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>eid</th>\n",
       "      <th>bi_acetabular_width</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1000276</td>\n",
       "      <td>21.344512</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1000742</td>\n",
       "      <td>20.452948</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1000951</td>\n",
       "      <td>20.324819</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1001025</td>\n",
       "      <td>22.256144</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1001091</td>\n",
       "      <td>20.924496</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30365</th>\n",
       "      <td>6023985</td>\n",
       "      <td>20.140485</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30366</th>\n",
       "      <td>6024029</td>\n",
       "      <td>20.408280</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30367</th>\n",
       "      <td>6024037</td>\n",
       "      <td>20.853574</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30368</th>\n",
       "      <td>6024257</td>\n",
       "      <td>20.581813</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30369</th>\n",
       "      <td>6024532</td>\n",
       "      <td>21.589125</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>30370 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           eid  bi_acetabular_width\n",
       "0      1000276            21.344512\n",
       "1      1000742            20.452948\n",
       "2      1000951            20.324819\n",
       "3      1001025            22.256144\n",
       "4      1001091            20.924496\n",
       "...        ...                  ...\n",
       "30365  6023985            20.140485\n",
       "30366  6024029            20.408280\n",
       "30367  6024037            20.853574\n",
       "30368  6024257            20.581813\n",
       "30369  6024532            21.589125\n",
       "\n",
       "[30370 rows x 2 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bi_acetabular_width = pd.read_csv('key_results/hip_select_pheno_cm_eid_flt.csv')[['eid', 'bi_acetabular_width']]; bi_acetabular_width"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "# process male\n",
    "\n",
    "# merge previous phenotypes\n",
    "hip_pheno_gwas_male = pd.read_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_male_cm_231011.txt', sep = \" \")\n",
    "hip_pheno_gwas_male = hip_pheno_gwas_male.merge(bi_acetabular_width, left_on='FID', right_on='eid', how='inner')\n",
    "hip_pheno_gwas_male.drop(columns=['eid'], inplace=True)\n",
    "\n",
    "hip_pheno_gwas_male.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_male_cm_231025.txt', index=False, sep = \" \")\n",
    "hip_pheno_gwas_male.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_male_no_header_231025.txt', index=False, header=False, sep = \" \")\n",
    "\n",
    "hip_pheno_gwas_male[['FID', 'IID', 'arm_divide_leg', 'leg_divide_torso', 'bi_acetabular_width']].to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_three_new_pheno_gwas_male_231025.txt', index=False, sep = \" \")\n",
    "hip_pheno_gwas_male[['FID', 'IID', 'arm_divide_leg', 'leg_divide_torso', 'bi_acetabular_width']].to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_three_new_pheno_gwas_male_no_header_231025.txt', index=False, header=False, sep = \" \")\n",
    "\n",
    "# process female\n",
    "\n",
    "# merge previous phenotypes\n",
    "hip_pheno_gwas_female = pd.read_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_female_cm_231011.txt', sep = \" \")\n",
    "hip_pheno_gwas_female = hip_pheno_gwas_female.merge(bi_acetabular_width, left_on='FID', right_on='eid', how='inner')\n",
    "hip_pheno_gwas_female.drop(columns=['eid'], inplace=True)\n",
    "\n",
    "hip_pheno_gwas_female.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_female_cm_231025.txt', index=False, sep = \" \")\n",
    "hip_pheno_gwas_female.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_female_cm_no_header_231025.txt', index=False, header=False, sep = \" \")\n",
    "\n",
    "hip_pheno_gwas_female[['FID', 'IID', 'arm_divide_leg', 'leg_divide_torso', 'bi_acetabular_width']].to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_three_new_pheno_gwas_female_231025.txt', index=False, sep = \" \")\n",
    "hip_pheno_gwas_female[['FID', 'IID', 'arm_divide_leg', 'leg_divide_torso', 'bi_acetabular_width']].to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_three_new_pheno_gwas_female_no_header_231025.txt', index=False, header=False, sep = \" \")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(14845, 26)"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hip_pheno_gwas_male.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(15432, 26)"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hip_pheno_gwas_female.shape"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### add leg length phentoype"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "hip_pheno = pd.read_csv('key_results/hip_pheno_23_cm.csv')\n",
    "\n",
    "image_ids = pd.read_csv('key_results/hip_select_pheno_cm_eid_flt.csv')['image_id'].tolist()\n",
    "\n",
    "# filter individuals\n",
    "hip_pheno = hip_pheno[hip_pheno['image_id'].isin(image_ids)]\n",
    "\n",
    "# new phenotypes\n",
    "hip_pheno['leg_length'] = (hip_pheno['leg_123_left'] + hip_pheno['leg_123_right']) / 2\n",
    "add_pheno = hip_pheno[['eid', 'leg_length']]\n",
    "\n",
    "# process male\n",
    "\n",
    "# merge previous phenotypes\n",
    "hip_pheno_gwas_male = pd.read_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_male_cm_231025.txt', sep = \" \")\n",
    "hip_pheno_gwas_male = hip_pheno_gwas_male.merge(add_pheno, left_on='FID', right_on='eid', how='inner')\n",
    "hip_pheno_gwas_male.drop(columns=['eid'], inplace=True)\n",
    "\n",
    "hip_pheno_gwas_male.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_male_cm_240104.txt', index=False, sep = \" \")\n",
    "hip_pheno_gwas_male.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_male_no_header_240104.txt', index=False, header=False, sep = \" \")\n",
    "\n",
    "# process female\n",
    "\n",
    "# merge previous phenotypes\n",
    "hip_pheno_gwas_female = pd.read_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_female_cm_231025.txt', sep = \" \")\n",
    "hip_pheno_gwas_female = hip_pheno_gwas_female.merge(add_pheno, left_on='FID', right_on='eid', how='inner')\n",
    "hip_pheno_gwas_female.drop(columns=['eid'], inplace=True)\n",
    "\n",
    "hip_pheno_gwas_female.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_female_cm_240104.txt', index=False, sep = \" \")\n",
    "hip_pheno_gwas_female.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_female_cm_no_header_240104.txt', index=False, header=False, sep = \" \")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(14831, 27)"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hip_pheno_gwas_male.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(15412, 27)"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hip_pheno_gwas_female.shape"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### gwas for female other skeletal traits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>image_id</th>\n",
       "      <th>file_name</th>\n",
       "      <th>eid</th>\n",
       "      <th>sex</th>\n",
       "      <th>standing_height_imaging_visit</th>\n",
       "      <th>weight_imaging_visit</th>\n",
       "      <th>bmi_imaging_visit</th>\n",
       "      <th>age_imaging_visit</th>\n",
       "      <th>pelvic_height</th>\n",
       "      <th>pelvic_width</th>\n",
       "      <th>...</th>\n",
       "      <th>acetabular_diameter</th>\n",
       "      <th>acetabular_inclination</th>\n",
       "      <th>humerus</th>\n",
       "      <th>femur</th>\n",
       "      <th>forearm</th>\n",
       "      <th>tibia</th>\n",
       "      <th>arm_length</th>\n",
       "      <th>leg_length</th>\n",
       "      <th>leg_divide_torso</th>\n",
       "      <th>pelvic_inlet_area</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>05431_82</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20190328125146.1.1...</td>\n",
       "      <td>1001867</td>\n",
       "      <td>0.0</td>\n",
       "      <td>153.0</td>\n",
       "      <td>56.1</td>\n",
       "      <td>23.9651</td>\n",
       "      <td>75.0</td>\n",
       "      <td>22.226944</td>\n",
       "      <td>30.934457</td>\n",
       "      <td>...</td>\n",
       "      <td>5.531657</td>\n",
       "      <td>59.017487</td>\n",
       "      <td>26.778209</td>\n",
       "      <td>42.934407</td>\n",
       "      <td>23.988531</td>\n",
       "      <td>35.818183</td>\n",
       "      <td>50.766740</td>\n",
       "      <td>78.752591</td>\n",
       "      <td>1.537721</td>\n",
       "      <td>139.989191</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>18261_82</td>\n",
       "      <td>1.2.840.113619.2.110.212174.20181127130214.1.1...</td>\n",
       "      <td>1002480</td>\n",
       "      <td>0.0</td>\n",
       "      <td>163.6</td>\n",
       "      <td>77.6</td>\n",
       "      <td>28.9931</td>\n",
       "      <td>73.0</td>\n",
       "      <td>21.362371</td>\n",
       "      <td>31.549402</td>\n",
       "      <td>...</td>\n",
       "      <td>5.847069</td>\n",
       "      <td>58.261479</td>\n",
       "      <td>27.178109</td>\n",
       "      <td>43.806834</td>\n",
       "      <td>24.890288</td>\n",
       "      <td>36.858990</td>\n",
       "      <td>52.068396</td>\n",
       "      <td>80.665824</td>\n",
       "      <td>1.370988</td>\n",
       "      <td>135.649767</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>08315_82</td>\n",
       "      <td>1.2.840.113619.2.110.212038.20170623172431.1.1...</td>\n",
       "      <td>1003186</td>\n",
       "      <td>0.0</td>\n",
       "      <td>158.0</td>\n",
       "      <td>60.8</td>\n",
       "      <td>24.3551</td>\n",
       "      <td>60.0</td>\n",
       "      <td>21.017710</td>\n",
       "      <td>28.052667</td>\n",
       "      <td>...</td>\n",
       "      <td>5.261873</td>\n",
       "      <td>65.963468</td>\n",
       "      <td>26.305751</td>\n",
       "      <td>39.912014</td>\n",
       "      <td>24.355342</td>\n",
       "      <td>35.738969</td>\n",
       "      <td>50.661093</td>\n",
       "      <td>75.650983</td>\n",
       "      <td>1.325449</td>\n",
       "      <td>139.721260</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10901_82</td>\n",
       "      <td>1.2.840.113619.2.110.212038.20180410153412.1.1...</td>\n",
       "      <td>1003290</td>\n",
       "      <td>0.0</td>\n",
       "      <td>158.0</td>\n",
       "      <td>67.3</td>\n",
       "      <td>26.9588</td>\n",
       "      <td>67.0</td>\n",
       "      <td>22.347859</td>\n",
       "      <td>27.953417</td>\n",
       "      <td>...</td>\n",
       "      <td>5.330001</td>\n",
       "      <td>70.077085</td>\n",
       "      <td>27.443498</td>\n",
       "      <td>43.574953</td>\n",
       "      <td>25.184950</td>\n",
       "      <td>37.695381</td>\n",
       "      <td>52.628449</td>\n",
       "      <td>81.270334</td>\n",
       "      <td>1.437291</td>\n",
       "      <td>121.206033</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>00181_82</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20180325133658.1.1...</td>\n",
       "      <td>1004043</td>\n",
       "      <td>0.0</td>\n",
       "      <td>164.0</td>\n",
       "      <td>65.7</td>\n",
       "      <td>24.4274</td>\n",
       "      <td>76.0</td>\n",
       "      <td>24.095581</td>\n",
       "      <td>29.541051</td>\n",
       "      <td>...</td>\n",
       "      <td>5.571609</td>\n",
       "      <td>66.368688</td>\n",
       "      <td>29.277986</td>\n",
       "      <td>44.588294</td>\n",
       "      <td>26.471927</td>\n",
       "      <td>38.666034</td>\n",
       "      <td>55.749913</td>\n",
       "      <td>83.254328</td>\n",
       "      <td>1.377830</td>\n",
       "      <td>151.160622</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15805</th>\n",
       "      <td>03502_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20151009092926.1.1...</td>\n",
       "      <td>6020398</td>\n",
       "      <td>0.0</td>\n",
       "      <td>157.0</td>\n",
       "      <td>52.5</td>\n",
       "      <td>21.2990</td>\n",
       "      <td>64.0</td>\n",
       "      <td>22.443374</td>\n",
       "      <td>28.848989</td>\n",
       "      <td>...</td>\n",
       "      <td>5.076335</td>\n",
       "      <td>73.754462</td>\n",
       "      <td>26.072229</td>\n",
       "      <td>43.012878</td>\n",
       "      <td>25.616248</td>\n",
       "      <td>38.496799</td>\n",
       "      <td>51.688477</td>\n",
       "      <td>81.509677</td>\n",
       "      <td>1.433212</td>\n",
       "      <td>154.408090</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15806</th>\n",
       "      <td>12000_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20170210165522.1.1...</td>\n",
       "      <td>6020527</td>\n",
       "      <td>0.0</td>\n",
       "      <td>163.0</td>\n",
       "      <td>70.5</td>\n",
       "      <td>26.5347</td>\n",
       "      <td>75.0</td>\n",
       "      <td>22.381443</td>\n",
       "      <td>30.477040</td>\n",
       "      <td>...</td>\n",
       "      <td>5.575862</td>\n",
       "      <td>54.604663</td>\n",
       "      <td>29.256903</td>\n",
       "      <td>45.089598</td>\n",
       "      <td>26.486205</td>\n",
       "      <td>39.392239</td>\n",
       "      <td>55.743108</td>\n",
       "      <td>84.481837</td>\n",
       "      <td>1.466060</td>\n",
       "      <td>122.840882</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15807</th>\n",
       "      <td>14260_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20170720174337.1.1...</td>\n",
       "      <td>6020585</td>\n",
       "      <td>0.0</td>\n",
       "      <td>158.0</td>\n",
       "      <td>69.4</td>\n",
       "      <td>27.8000</td>\n",
       "      <td>62.0</td>\n",
       "      <td>22.561943</td>\n",
       "      <td>31.630376</td>\n",
       "      <td>...</td>\n",
       "      <td>5.044865</td>\n",
       "      <td>62.885530</td>\n",
       "      <td>27.775247</td>\n",
       "      <td>43.821114</td>\n",
       "      <td>25.846634</td>\n",
       "      <td>38.041064</td>\n",
       "      <td>53.621881</td>\n",
       "      <td>81.862178</td>\n",
       "      <td>1.432360</td>\n",
       "      <td>153.989374</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15808</th>\n",
       "      <td>13966_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20170702173904.1.1...</td>\n",
       "      <td>6022591</td>\n",
       "      <td>0.0</td>\n",
       "      <td>156.0</td>\n",
       "      <td>87.7</td>\n",
       "      <td>36.0371</td>\n",
       "      <td>69.0</td>\n",
       "      <td>20.946633</td>\n",
       "      <td>26.922261</td>\n",
       "      <td>...</td>\n",
       "      <td>4.959191</td>\n",
       "      <td>67.937764</td>\n",
       "      <td>28.030085</td>\n",
       "      <td>42.718048</td>\n",
       "      <td>24.880466</td>\n",
       "      <td>37.908781</td>\n",
       "      <td>52.910551</td>\n",
       "      <td>80.626829</td>\n",
       "      <td>1.461921</td>\n",
       "      <td>126.477899</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15809</th>\n",
       "      <td>16309_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20171202183632.1.1...</td>\n",
       "      <td>6023431</td>\n",
       "      <td>0.0</td>\n",
       "      <td>170.0</td>\n",
       "      <td>61.5</td>\n",
       "      <td>21.2803</td>\n",
       "      <td>51.0</td>\n",
       "      <td>22.787944</td>\n",
       "      <td>29.871832</td>\n",
       "      <td>...</td>\n",
       "      <td>5.201943</td>\n",
       "      <td>60.779334</td>\n",
       "      <td>29.276081</td>\n",
       "      <td>45.262378</td>\n",
       "      <td>25.847555</td>\n",
       "      <td>38.992764</td>\n",
       "      <td>55.123637</td>\n",
       "      <td>84.255142</td>\n",
       "      <td>1.384857</td>\n",
       "      <td>150.788401</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>15810 rows × 32 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       image_id                                          file_name      eid  \\\n",
       "0      05431_82  1.2.840.113619.2.110.210419.20190328125146.1.1...  1001867   \n",
       "1      18261_82  1.2.840.113619.2.110.212174.20181127130214.1.1...  1002480   \n",
       "2      08315_82  1.2.840.113619.2.110.212038.20170623172431.1.1...  1003186   \n",
       "3      10901_82  1.2.840.113619.2.110.212038.20180410153412.1.1...  1003290   \n",
       "4      00181_82  1.2.840.113619.2.110.210419.20180325133658.1.1...  1004043   \n",
       "...         ...                                                ...      ...   \n",
       "15805  03502_93  1.2.840.113619.2.110.210419.20151009092926.1.1...  6020398   \n",
       "15806  12000_93  1.2.840.113619.2.110.210419.20170210165522.1.1...  6020527   \n",
       "15807  14260_93  1.2.840.113619.2.110.210419.20170720174337.1.1...  6020585   \n",
       "15808  13966_93  1.2.840.113619.2.110.210419.20170702173904.1.1...  6022591   \n",
       "15809  16309_93  1.2.840.113619.2.110.210419.20171202183632.1.1...  6023431   \n",
       "\n",
       "       sex  standing_height_imaging_visit  weight_imaging_visit  \\\n",
       "0      0.0                          153.0                  56.1   \n",
       "1      0.0                          163.6                  77.6   \n",
       "2      0.0                          158.0                  60.8   \n",
       "3      0.0                          158.0                  67.3   \n",
       "4      0.0                          164.0                  65.7   \n",
       "...    ...                            ...                   ...   \n",
       "15805  0.0                          157.0                  52.5   \n",
       "15806  0.0                          163.0                  70.5   \n",
       "15807  0.0                          158.0                  69.4   \n",
       "15808  0.0                          156.0                  87.7   \n",
       "15809  0.0                          170.0                  61.5   \n",
       "\n",
       "       bmi_imaging_visit  age_imaging_visit  pelvic_height  pelvic_width  ...  \\\n",
       "0                23.9651               75.0      22.226944     30.934457  ...   \n",
       "1                28.9931               73.0      21.362371     31.549402  ...   \n",
       "2                24.3551               60.0      21.017710     28.052667  ...   \n",
       "3                26.9588               67.0      22.347859     27.953417  ...   \n",
       "4                24.4274               76.0      24.095581     29.541051  ...   \n",
       "...                  ...                ...            ...           ...  ...   \n",
       "15805            21.2990               64.0      22.443374     28.848989  ...   \n",
       "15806            26.5347               75.0      22.381443     30.477040  ...   \n",
       "15807            27.8000               62.0      22.561943     31.630376  ...   \n",
       "15808            36.0371               69.0      20.946633     26.922261  ...   \n",
       "15809            21.2803               51.0      22.787944     29.871832  ...   \n",
       "\n",
       "       acetabular_diameter  acetabular_inclination    humerus      femur  \\\n",
       "0                 5.531657               59.017487  26.778209  42.934407   \n",
       "1                 5.847069               58.261479  27.178109  43.806834   \n",
       "2                 5.261873               65.963468  26.305751  39.912014   \n",
       "3                 5.330001               70.077085  27.443498  43.574953   \n",
       "4                 5.571609               66.368688  29.277986  44.588294   \n",
       "...                    ...                     ...        ...        ...   \n",
       "15805             5.076335               73.754462  26.072229  43.012878   \n",
       "15806             5.575862               54.604663  29.256903  45.089598   \n",
       "15807             5.044865               62.885530  27.775247  43.821114   \n",
       "15808             4.959191               67.937764  28.030085  42.718048   \n",
       "15809             5.201943               60.779334  29.276081  45.262378   \n",
       "\n",
       "         forearm      tibia  arm_length  leg_length  leg_divide_torso  \\\n",
       "0      23.988531  35.818183   50.766740   78.752591          1.537721   \n",
       "1      24.890288  36.858990   52.068396   80.665824          1.370988   \n",
       "2      24.355342  35.738969   50.661093   75.650983          1.325449   \n",
       "3      25.184950  37.695381   52.628449   81.270334          1.437291   \n",
       "4      26.471927  38.666034   55.749913   83.254328          1.377830   \n",
       "...          ...        ...         ...         ...               ...   \n",
       "15805  25.616248  38.496799   51.688477   81.509677          1.433212   \n",
       "15806  26.486205  39.392239   55.743108   84.481837          1.466060   \n",
       "15807  25.846634  38.041064   53.621881   81.862178          1.432360   \n",
       "15808  24.880466  37.908781   52.910551   80.626829          1.461921   \n",
       "15809  25.847555  38.992764   55.123637   84.255142          1.384857   \n",
       "\n",
       "       pelvic_inlet_area  \n",
       "0             139.989191  \n",
       "1             135.649767  \n",
       "2             139.721260  \n",
       "3             121.206033  \n",
       "4             151.160622  \n",
       "...                  ...  \n",
       "15805         154.408090  \n",
       "15806         122.840882  \n",
       "15807         153.989374  \n",
       "15808         126.477899  \n",
       "15809         150.788401  \n",
       "\n",
       "[15810 rows x 32 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "female_skeletal = pd.read_csv(\"key_results/hip_select_pheno_cm_eid_flt_z_flt_female.csv\"); female_skeletal"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['image_id', 'file_name', 'eid', 'sex', 'standing_height_imaging_visit',\n",
       "       'weight_imaging_visit', 'bmi_imaging_visit', 'age_imaging_visit',\n",
       "       'pelvic_height', 'pelvic_width', 'pelvic_inlet_width',\n",
       "       'iliac_flare_ratio', 'oblique_pelvic_inlet_length', 'iliac_flare_angle',\n",
       "       'subpubic_angle', 'head_diameter', 'trochanter_distance',\n",
       "       'shoulder_width', 'arm_devide_torso', 'bi_acetabular_width',\n",
       "       'torso_length', 'iliac_isthmus_breadth', 'acetabular_diameter',\n",
       "       'acetabular_inclination', 'humerus', 'femur', 'forearm', 'tibia',\n",
       "       'arm_length', 'leg_length', 'leg_divide_torso', 'pelvic_inlet_area'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "female_skeletal.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "pheno = ['eid', 'standing_height_imaging_visit', 'head_diameter', 'trochanter_distance', 'shoulder_width',\n",
    "         'torso_length', 'humerus', 'femur', 'forearm', 'tibia']\n",
    "\n",
    "# male dataframe for gwas preparation\n",
    "female_skeletal = female_skeletal[pheno]\n",
    "\n",
    "female_skeletal_gwas = female_skeletal.iloc[:, 1:]\n",
    "\n",
    "ids = female_skeletal['eid'].tolist()\n",
    "\n",
    "female_skeletal_gwas['FID'] = ids\n",
    "female_skeletal_gwas['IID'] = ids\n",
    "\n",
    "# put FID and IID to the first two columns\n",
    "cols = female_skeletal_gwas.columns\n",
    "cols = cols[-2:].tolist() + cols[:-2].tolist()\n",
    "female_skeletal_gwas = female_skeletal_gwas[cols]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>standing_height_imaging_visit</th>\n",
       "      <th>head_diameter</th>\n",
       "      <th>trochanter_distance</th>\n",
       "      <th>shoulder_width</th>\n",
       "      <th>torso_length</th>\n",
       "      <th>humerus</th>\n",
       "      <th>femur</th>\n",
       "      <th>forearm</th>\n",
       "      <th>tibia</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1001867</td>\n",
       "      <td>1001867</td>\n",
       "      <td>153.0</td>\n",
       "      <td>13.029582</td>\n",
       "      <td>31.173284</td>\n",
       "      <td>35.909407</td>\n",
       "      <td>51.213839</td>\n",
       "      <td>26.778209</td>\n",
       "      <td>42.934407</td>\n",
       "      <td>23.988531</td>\n",
       "      <td>35.818183</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1002480</td>\n",
       "      <td>1002480</td>\n",
       "      <td>163.6</td>\n",
       "      <td>14.027009</td>\n",
       "      <td>32.626364</td>\n",
       "      <td>36.389501</td>\n",
       "      <td>58.837716</td>\n",
       "      <td>27.178109</td>\n",
       "      <td>43.806834</td>\n",
       "      <td>24.890288</td>\n",
       "      <td>36.858990</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1003186</td>\n",
       "      <td>1003186</td>\n",
       "      <td>158.0</td>\n",
       "      <td>12.450457</td>\n",
       "      <td>29.220709</td>\n",
       "      <td>35.147572</td>\n",
       "      <td>57.075759</td>\n",
       "      <td>26.305751</td>\n",
       "      <td>39.912014</td>\n",
       "      <td>24.355342</td>\n",
       "      <td>35.738969</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1003290</td>\n",
       "      <td>1003290</td>\n",
       "      <td>158.0</td>\n",
       "      <td>13.490861</td>\n",
       "      <td>28.088993</td>\n",
       "      <td>35.241159</td>\n",
       "      <td>56.544099</td>\n",
       "      <td>27.443498</td>\n",
       "      <td>43.574953</td>\n",
       "      <td>25.184950</td>\n",
       "      <td>37.695381</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1004043</td>\n",
       "      <td>1004043</td>\n",
       "      <td>164.0</td>\n",
       "      <td>13.634670</td>\n",
       "      <td>29.834703</td>\n",
       "      <td>34.289392</td>\n",
       "      <td>60.424245</td>\n",
       "      <td>29.277986</td>\n",
       "      <td>44.588294</td>\n",
       "      <td>26.471927</td>\n",
       "      <td>38.666034</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15805</th>\n",
       "      <td>6020398</td>\n",
       "      <td>6020398</td>\n",
       "      <td>157.0</td>\n",
       "      <td>12.705593</td>\n",
       "      <td>29.996145</td>\n",
       "      <td>34.601432</td>\n",
       "      <td>56.872025</td>\n",
       "      <td>26.072229</td>\n",
       "      <td>43.012878</td>\n",
       "      <td>25.616248</td>\n",
       "      <td>38.496799</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15806</th>\n",
       "      <td>6020527</td>\n",
       "      <td>6020527</td>\n",
       "      <td>163.0</td>\n",
       "      <td>13.167859</td>\n",
       "      <td>28.098706</td>\n",
       "      <td>37.714589</td>\n",
       "      <td>57.625105</td>\n",
       "      <td>29.256903</td>\n",
       "      <td>45.089598</td>\n",
       "      <td>26.486205</td>\n",
       "      <td>39.392239</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15807</th>\n",
       "      <td>6020585</td>\n",
       "      <td>6020585</td>\n",
       "      <td>158.0</td>\n",
       "      <td>12.365094</td>\n",
       "      <td>30.759181</td>\n",
       "      <td>37.654639</td>\n",
       "      <td>57.151972</td>\n",
       "      <td>27.775247</td>\n",
       "      <td>43.821114</td>\n",
       "      <td>25.846634</td>\n",
       "      <td>38.041064</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15808</th>\n",
       "      <td>6022591</td>\n",
       "      <td>6022591</td>\n",
       "      <td>156.0</td>\n",
       "      <td>12.551823</td>\n",
       "      <td>27.342555</td>\n",
       "      <td>35.846661</td>\n",
       "      <td>55.151275</td>\n",
       "      <td>28.030085</td>\n",
       "      <td>42.718048</td>\n",
       "      <td>24.880466</td>\n",
       "      <td>37.908781</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15809</th>\n",
       "      <td>6023431</td>\n",
       "      <td>6023431</td>\n",
       "      <td>170.0</td>\n",
       "      <td>12.309345</td>\n",
       "      <td>27.698720</td>\n",
       "      <td>36.369916</td>\n",
       "      <td>60.840325</td>\n",
       "      <td>29.276081</td>\n",
       "      <td>45.262378</td>\n",
       "      <td>25.847555</td>\n",
       "      <td>38.992764</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>15810 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           FID      IID  standing_height_imaging_visit  head_diameter  \\\n",
       "0      1001867  1001867                          153.0      13.029582   \n",
       "1      1002480  1002480                          163.6      14.027009   \n",
       "2      1003186  1003186                          158.0      12.450457   \n",
       "3      1003290  1003290                          158.0      13.490861   \n",
       "4      1004043  1004043                          164.0      13.634670   \n",
       "...        ...      ...                            ...            ...   \n",
       "15805  6020398  6020398                          157.0      12.705593   \n",
       "15806  6020527  6020527                          163.0      13.167859   \n",
       "15807  6020585  6020585                          158.0      12.365094   \n",
       "15808  6022591  6022591                          156.0      12.551823   \n",
       "15809  6023431  6023431                          170.0      12.309345   \n",
       "\n",
       "       trochanter_distance  shoulder_width  torso_length    humerus  \\\n",
       "0                31.173284       35.909407     51.213839  26.778209   \n",
       "1                32.626364       36.389501     58.837716  27.178109   \n",
       "2                29.220709       35.147572     57.075759  26.305751   \n",
       "3                28.088993       35.241159     56.544099  27.443498   \n",
       "4                29.834703       34.289392     60.424245  29.277986   \n",
       "...                    ...             ...           ...        ...   \n",
       "15805            29.996145       34.601432     56.872025  26.072229   \n",
       "15806            28.098706       37.714589     57.625105  29.256903   \n",
       "15807            30.759181       37.654639     57.151972  27.775247   \n",
       "15808            27.342555       35.846661     55.151275  28.030085   \n",
       "15809            27.698720       36.369916     60.840325  29.276081   \n",
       "\n",
       "           femur    forearm      tibia  \n",
       "0      42.934407  23.988531  35.818183  \n",
       "1      43.806834  24.890288  36.858990  \n",
       "2      39.912014  24.355342  35.738969  \n",
       "3      43.574953  25.184950  37.695381  \n",
       "4      44.588294  26.471927  38.666034  \n",
       "...          ...        ...        ...  \n",
       "15805  43.012878  25.616248  38.496799  \n",
       "15806  45.089598  26.486205  39.392239  \n",
       "15807  43.821114  25.846634  38.041064  \n",
       "15808  42.718048  24.880466  37.908781  \n",
       "15809  45.262378  25.847555  38.992764  \n",
       "\n",
       "[15810 rows x 11 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "female_skeletal_gwas"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "female_skeletal_gwas[['FID', 'IID', 'standing_height_imaging_visit']].to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/female_height_gwas_cm_240310.txt', index=False, sep = \" \")\n",
    "female_skeletal_gwas.drop(columns = ['standing_height_imaging_visit']).to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/female_skeletal_gwas_cm_240310.txt', index=False, sep = \" \")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### gwas for male other skeletal traits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>image_id</th>\n",
       "      <th>file_name</th>\n",
       "      <th>eid</th>\n",
       "      <th>sex</th>\n",
       "      <th>standing_height_imaging_visit</th>\n",
       "      <th>weight_imaging_visit</th>\n",
       "      <th>bmi_imaging_visit</th>\n",
       "      <th>age_imaging_visit</th>\n",
       "      <th>pelvic_height</th>\n",
       "      <th>pelvic_width</th>\n",
       "      <th>...</th>\n",
       "      <th>acetabular_diameter</th>\n",
       "      <th>acetabular_inclination</th>\n",
       "      <th>humerus</th>\n",
       "      <th>femur</th>\n",
       "      <th>forearm</th>\n",
       "      <th>tibia</th>\n",
       "      <th>arm_length</th>\n",
       "      <th>leg_length</th>\n",
       "      <th>leg_divide_torso</th>\n",
       "      <th>pelvic_inlet_area</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14366_82</td>\n",
       "      <td>1.2.840.113619.2.110.212038.20190307144208.1.1...</td>\n",
       "      <td>1000951</td>\n",
       "      <td>1.0</td>\n",
       "      <td>166.0</td>\n",
       "      <td>60.8</td>\n",
       "      <td>22.0642</td>\n",
       "      <td>78.0</td>\n",
       "      <td>22.297293</td>\n",
       "      <td>30.113663</td>\n",
       "      <td>...</td>\n",
       "      <td>6.141783</td>\n",
       "      <td>60.747735</td>\n",
       "      <td>30.816804</td>\n",
       "      <td>48.489427</td>\n",
       "      <td>26.856522</td>\n",
       "      <td>40.743734</td>\n",
       "      <td>57.673326</td>\n",
       "      <td>89.233162</td>\n",
       "      <td>1.651141</td>\n",
       "      <td>108.190429</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>01877_82</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20180720150454.1.1...</td>\n",
       "      <td>1001512</td>\n",
       "      <td>1.0</td>\n",
       "      <td>169.0</td>\n",
       "      <td>77.3</td>\n",
       "      <td>27.0649</td>\n",
       "      <td>66.0</td>\n",
       "      <td>23.742297</td>\n",
       "      <td>30.901775</td>\n",
       "      <td>...</td>\n",
       "      <td>6.035855</td>\n",
       "      <td>59.189839</td>\n",
       "      <td>28.115456</td>\n",
       "      <td>44.984030</td>\n",
       "      <td>26.628187</td>\n",
       "      <td>40.081134</td>\n",
       "      <td>54.743642</td>\n",
       "      <td>85.065164</td>\n",
       "      <td>1.449931</td>\n",
       "      <td>102.660623</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10557_82</td>\n",
       "      <td>1.2.840.113619.2.110.212038.20180228105159.1.1...</td>\n",
       "      <td>1001921</td>\n",
       "      <td>1.0</td>\n",
       "      <td>166.0</td>\n",
       "      <td>64.2</td>\n",
       "      <td>23.2980</td>\n",
       "      <td>62.0</td>\n",
       "      <td>22.861258</td>\n",
       "      <td>28.863430</td>\n",
       "      <td>...</td>\n",
       "      <td>5.834113</td>\n",
       "      <td>58.027399</td>\n",
       "      <td>29.972729</td>\n",
       "      <td>46.587679</td>\n",
       "      <td>28.561098</td>\n",
       "      <td>39.566371</td>\n",
       "      <td>58.533827</td>\n",
       "      <td>86.154050</td>\n",
       "      <td>1.493503</td>\n",
       "      <td>110.520822</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>01561_82</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20180630141714.1.1...</td>\n",
       "      <td>1001950</td>\n",
       "      <td>1.0</td>\n",
       "      <td>165.0</td>\n",
       "      <td>89.6</td>\n",
       "      <td>32.9109</td>\n",
       "      <td>62.0</td>\n",
       "      <td>23.080688</td>\n",
       "      <td>29.385649</td>\n",
       "      <td>...</td>\n",
       "      <td>5.731393</td>\n",
       "      <td>57.335593</td>\n",
       "      <td>29.012823</td>\n",
       "      <td>45.245573</td>\n",
       "      <td>27.292476</td>\n",
       "      <td>39.041382</td>\n",
       "      <td>56.305299</td>\n",
       "      <td>84.286956</td>\n",
       "      <td>1.465162</td>\n",
       "      <td>109.946859</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>07611_82</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20190912120016.1.1...</td>\n",
       "      <td>1002851</td>\n",
       "      <td>1.0</td>\n",
       "      <td>173.0</td>\n",
       "      <td>75.0</td>\n",
       "      <td>25.0593</td>\n",
       "      <td>62.0</td>\n",
       "      <td>25.497024</td>\n",
       "      <td>33.172829</td>\n",
       "      <td>...</td>\n",
       "      <td>6.167024</td>\n",
       "      <td>62.530271</td>\n",
       "      <td>30.128320</td>\n",
       "      <td>48.528284</td>\n",
       "      <td>29.505318</td>\n",
       "      <td>42.783355</td>\n",
       "      <td>59.633639</td>\n",
       "      <td>91.311639</td>\n",
       "      <td>1.488416</td>\n",
       "      <td>149.416046</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15272</th>\n",
       "      <td>03183_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20151006143059.2.1...</td>\n",
       "      <td>6020675</td>\n",
       "      <td>1.0</td>\n",
       "      <td>184.0</td>\n",
       "      <td>73.9</td>\n",
       "      <td>21.8277</td>\n",
       "      <td>57.0</td>\n",
       "      <td>25.369030</td>\n",
       "      <td>32.767891</td>\n",
       "      <td>...</td>\n",
       "      <td>6.538142</td>\n",
       "      <td>56.217517</td>\n",
       "      <td>32.354476</td>\n",
       "      <td>50.976459</td>\n",
       "      <td>29.572681</td>\n",
       "      <td>43.244042</td>\n",
       "      <td>61.927156</td>\n",
       "      <td>94.220501</td>\n",
       "      <td>1.426580</td>\n",
       "      <td>143.697841</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15273</th>\n",
       "      <td>11379_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20161221100256.1.1...</td>\n",
       "      <td>6020722</td>\n",
       "      <td>1.0</td>\n",
       "      <td>173.0</td>\n",
       "      <td>75.8</td>\n",
       "      <td>25.3266</td>\n",
       "      <td>67.0</td>\n",
       "      <td>24.944018</td>\n",
       "      <td>30.547164</td>\n",
       "      <td>...</td>\n",
       "      <td>5.990853</td>\n",
       "      <td>57.652643</td>\n",
       "      <td>28.799111</td>\n",
       "      <td>46.356024</td>\n",
       "      <td>29.852094</td>\n",
       "      <td>41.777482</td>\n",
       "      <td>58.651204</td>\n",
       "      <td>88.133506</td>\n",
       "      <td>1.443841</td>\n",
       "      <td>132.004910</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15274</th>\n",
       "      <td>10039_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20160923114938.1.1...</td>\n",
       "      <td>6021112</td>\n",
       "      <td>1.0</td>\n",
       "      <td>177.0</td>\n",
       "      <td>89.0</td>\n",
       "      <td>28.4082</td>\n",
       "      <td>71.0</td>\n",
       "      <td>25.247090</td>\n",
       "      <td>32.848104</td>\n",
       "      <td>...</td>\n",
       "      <td>6.149698</td>\n",
       "      <td>60.427962</td>\n",
       "      <td>31.540230</td>\n",
       "      <td>50.329159</td>\n",
       "      <td>30.128478</td>\n",
       "      <td>44.856165</td>\n",
       "      <td>61.668708</td>\n",
       "      <td>95.185324</td>\n",
       "      <td>1.583562</td>\n",
       "      <td>134.603261</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15275</th>\n",
       "      <td>14018_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20170705180448.1.1...</td>\n",
       "      <td>6021539</td>\n",
       "      <td>1.0</td>\n",
       "      <td>179.0</td>\n",
       "      <td>71.0</td>\n",
       "      <td>22.1591</td>\n",
       "      <td>69.0</td>\n",
       "      <td>25.232678</td>\n",
       "      <td>30.970184</td>\n",
       "      <td>...</td>\n",
       "      <td>6.504511</td>\n",
       "      <td>59.444884</td>\n",
       "      <td>30.777846</td>\n",
       "      <td>48.408803</td>\n",
       "      <td>28.966865</td>\n",
       "      <td>42.070750</td>\n",
       "      <td>59.744710</td>\n",
       "      <td>90.479553</td>\n",
       "      <td>1.479675</td>\n",
       "      <td>132.776803</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15276</th>\n",
       "      <td>12964_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20170418184226.1.1...</td>\n",
       "      <td>6024029</td>\n",
       "      <td>1.0</td>\n",
       "      <td>179.0</td>\n",
       "      <td>80.4</td>\n",
       "      <td>25.0928</td>\n",
       "      <td>50.0</td>\n",
       "      <td>24.323043</td>\n",
       "      <td>29.804727</td>\n",
       "      <td>...</td>\n",
       "      <td>6.171569</td>\n",
       "      <td>59.544963</td>\n",
       "      <td>32.678000</td>\n",
       "      <td>50.968299</td>\n",
       "      <td>29.406987</td>\n",
       "      <td>42.910290</td>\n",
       "      <td>62.084987</td>\n",
       "      <td>93.878589</td>\n",
       "      <td>1.475647</td>\n",
       "      <td>138.796708</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>15277 rows × 32 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       image_id                                          file_name      eid  \\\n",
       "0      14366_82  1.2.840.113619.2.110.212038.20190307144208.1.1...  1000951   \n",
       "1      01877_82  1.2.840.113619.2.110.210419.20180720150454.1.1...  1001512   \n",
       "2      10557_82  1.2.840.113619.2.110.212038.20180228105159.1.1...  1001921   \n",
       "3      01561_82  1.2.840.113619.2.110.210419.20180630141714.1.1...  1001950   \n",
       "4      07611_82  1.2.840.113619.2.110.210419.20190912120016.1.1...  1002851   \n",
       "...         ...                                                ...      ...   \n",
       "15272  03183_93  1.2.840.113619.2.110.210419.20151006143059.2.1...  6020675   \n",
       "15273  11379_93  1.2.840.113619.2.110.210419.20161221100256.1.1...  6020722   \n",
       "15274  10039_93  1.2.840.113619.2.110.210419.20160923114938.1.1...  6021112   \n",
       "15275  14018_93  1.2.840.113619.2.110.210419.20170705180448.1.1...  6021539   \n",
       "15276  12964_93  1.2.840.113619.2.110.210419.20170418184226.1.1...  6024029   \n",
       "\n",
       "       sex  standing_height_imaging_visit  weight_imaging_visit  \\\n",
       "0      1.0                          166.0                  60.8   \n",
       "1      1.0                          169.0                  77.3   \n",
       "2      1.0                          166.0                  64.2   \n",
       "3      1.0                          165.0                  89.6   \n",
       "4      1.0                          173.0                  75.0   \n",
       "...    ...                            ...                   ...   \n",
       "15272  1.0                          184.0                  73.9   \n",
       "15273  1.0                          173.0                  75.8   \n",
       "15274  1.0                          177.0                  89.0   \n",
       "15275  1.0                          179.0                  71.0   \n",
       "15276  1.0                          179.0                  80.4   \n",
       "\n",
       "       bmi_imaging_visit  age_imaging_visit  pelvic_height  pelvic_width  ...  \\\n",
       "0                22.0642               78.0      22.297293     30.113663  ...   \n",
       "1                27.0649               66.0      23.742297     30.901775  ...   \n",
       "2                23.2980               62.0      22.861258     28.863430  ...   \n",
       "3                32.9109               62.0      23.080688     29.385649  ...   \n",
       "4                25.0593               62.0      25.497024     33.172829  ...   \n",
       "...                  ...                ...            ...           ...  ...   \n",
       "15272            21.8277               57.0      25.369030     32.767891  ...   \n",
       "15273            25.3266               67.0      24.944018     30.547164  ...   \n",
       "15274            28.4082               71.0      25.247090     32.848104  ...   \n",
       "15275            22.1591               69.0      25.232678     30.970184  ...   \n",
       "15276            25.0928               50.0      24.323043     29.804727  ...   \n",
       "\n",
       "       acetabular_diameter  acetabular_inclination    humerus      femur  \\\n",
       "0                 6.141783               60.747735  30.816804  48.489427   \n",
       "1                 6.035855               59.189839  28.115456  44.984030   \n",
       "2                 5.834113               58.027399  29.972729  46.587679   \n",
       "3                 5.731393               57.335593  29.012823  45.245573   \n",
       "4                 6.167024               62.530271  30.128320  48.528284   \n",
       "...                    ...                     ...        ...        ...   \n",
       "15272             6.538142               56.217517  32.354476  50.976459   \n",
       "15273             5.990853               57.652643  28.799111  46.356024   \n",
       "15274             6.149698               60.427962  31.540230  50.329159   \n",
       "15275             6.504511               59.444884  30.777846  48.408803   \n",
       "15276             6.171569               59.544963  32.678000  50.968299   \n",
       "\n",
       "         forearm      tibia  arm_length  leg_length  leg_divide_torso  \\\n",
       "0      26.856522  40.743734   57.673326   89.233162          1.651141   \n",
       "1      26.628187  40.081134   54.743642   85.065164          1.449931   \n",
       "2      28.561098  39.566371   58.533827   86.154050          1.493503   \n",
       "3      27.292476  39.041382   56.305299   84.286956          1.465162   \n",
       "4      29.505318  42.783355   59.633639   91.311639          1.488416   \n",
       "...          ...        ...         ...         ...               ...   \n",
       "15272  29.572681  43.244042   61.927156   94.220501          1.426580   \n",
       "15273  29.852094  41.777482   58.651204   88.133506          1.443841   \n",
       "15274  30.128478  44.856165   61.668708   95.185324          1.583562   \n",
       "15275  28.966865  42.070750   59.744710   90.479553          1.479675   \n",
       "15276  29.406987  42.910290   62.084987   93.878589          1.475647   \n",
       "\n",
       "       pelvic_inlet_area  \n",
       "0             108.190429  \n",
       "1             102.660623  \n",
       "2             110.520822  \n",
       "3             109.946859  \n",
       "4             149.416046  \n",
       "...                  ...  \n",
       "15272         143.697841  \n",
       "15273         132.004910  \n",
       "15274         134.603261  \n",
       "15275         132.776803  \n",
       "15276         138.796708  \n",
       "\n",
       "[15277 rows x 32 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "male_skeletal = pd.read_csv(\"key_results/hip_select_pheno_cm_eid_flt_z_flt_male.csv\"); male_skeletal"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>standing_height_imaging_visit</th>\n",
       "      <th>head_diameter</th>\n",
       "      <th>trochanter_distance</th>\n",
       "      <th>shoulder_width</th>\n",
       "      <th>torso_length</th>\n",
       "      <th>humerus</th>\n",
       "      <th>femur</th>\n",
       "      <th>forearm</th>\n",
       "      <th>tibia</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1000951</td>\n",
       "      <td>1000951</td>\n",
       "      <td>166.0</td>\n",
       "      <td>14.112233</td>\n",
       "      <td>30.091232</td>\n",
       "      <td>36.696399</td>\n",
       "      <td>54.043332</td>\n",
       "      <td>30.816804</td>\n",
       "      <td>48.489427</td>\n",
       "      <td>26.856522</td>\n",
       "      <td>40.743734</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1001512</td>\n",
       "      <td>1001512</td>\n",
       "      <td>169.0</td>\n",
       "      <td>13.903518</td>\n",
       "      <td>30.813208</td>\n",
       "      <td>39.203992</td>\n",
       "      <td>58.668433</td>\n",
       "      <td>28.115456</td>\n",
       "      <td>44.984030</td>\n",
       "      <td>26.628187</td>\n",
       "      <td>40.081134</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1001921</td>\n",
       "      <td>1001921</td>\n",
       "      <td>166.0</td>\n",
       "      <td>14.480994</td>\n",
       "      <td>29.777444</td>\n",
       "      <td>33.219853</td>\n",
       "      <td>57.685872</td>\n",
       "      <td>29.972729</td>\n",
       "      <td>46.587679</td>\n",
       "      <td>28.561098</td>\n",
       "      <td>39.566371</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1001950</td>\n",
       "      <td>1001950</td>\n",
       "      <td>165.0</td>\n",
       "      <td>14.042468</td>\n",
       "      <td>28.838122</td>\n",
       "      <td>39.066233</td>\n",
       "      <td>57.527410</td>\n",
       "      <td>29.012823</td>\n",
       "      <td>45.245573</td>\n",
       "      <td>27.292476</td>\n",
       "      <td>39.041382</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1002851</td>\n",
       "      <td>1002851</td>\n",
       "      <td>173.0</td>\n",
       "      <td>13.723949</td>\n",
       "      <td>30.273697</td>\n",
       "      <td>43.356069</td>\n",
       "      <td>61.348184</td>\n",
       "      <td>30.128320</td>\n",
       "      <td>48.528284</td>\n",
       "      <td>29.505318</td>\n",
       "      <td>42.783355</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15272</th>\n",
       "      <td>6020675</td>\n",
       "      <td>6020675</td>\n",
       "      <td>184.0</td>\n",
       "      <td>14.045964</td>\n",
       "      <td>32.669869</td>\n",
       "      <td>40.853551</td>\n",
       "      <td>66.046411</td>\n",
       "      <td>32.354476</td>\n",
       "      <td>50.976459</td>\n",
       "      <td>29.572681</td>\n",
       "      <td>43.244042</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15273</th>\n",
       "      <td>6020722</td>\n",
       "      <td>6020722</td>\n",
       "      <td>173.0</td>\n",
       "      <td>14.056792</td>\n",
       "      <td>30.356562</td>\n",
       "      <td>39.199608</td>\n",
       "      <td>61.041006</td>\n",
       "      <td>28.799111</td>\n",
       "      <td>46.356024</td>\n",
       "      <td>29.852094</td>\n",
       "      <td>41.777482</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15274</th>\n",
       "      <td>6021112</td>\n",
       "      <td>6021112</td>\n",
       "      <td>177.0</td>\n",
       "      <td>13.402493</td>\n",
       "      <td>30.995640</td>\n",
       "      <td>41.118337</td>\n",
       "      <td>60.108352</td>\n",
       "      <td>31.540230</td>\n",
       "      <td>50.329159</td>\n",
       "      <td>30.128478</td>\n",
       "      <td>44.856165</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15275</th>\n",
       "      <td>6021539</td>\n",
       "      <td>6021539</td>\n",
       "      <td>179.0</td>\n",
       "      <td>13.781823</td>\n",
       "      <td>31.116022</td>\n",
       "      <td>38.579135</td>\n",
       "      <td>61.148256</td>\n",
       "      <td>30.777846</td>\n",
       "      <td>48.408803</td>\n",
       "      <td>28.966865</td>\n",
       "      <td>42.070750</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15276</th>\n",
       "      <td>6024029</td>\n",
       "      <td>6024029</td>\n",
       "      <td>179.0</td>\n",
       "      <td>13.271879</td>\n",
       "      <td>30.125650</td>\n",
       "      <td>39.486231</td>\n",
       "      <td>63.618586</td>\n",
       "      <td>32.678000</td>\n",
       "      <td>50.968299</td>\n",
       "      <td>29.406987</td>\n",
       "      <td>42.910290</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>15277 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           FID      IID  standing_height_imaging_visit  head_diameter  \\\n",
       "0      1000951  1000951                          166.0      14.112233   \n",
       "1      1001512  1001512                          169.0      13.903518   \n",
       "2      1001921  1001921                          166.0      14.480994   \n",
       "3      1001950  1001950                          165.0      14.042468   \n",
       "4      1002851  1002851                          173.0      13.723949   \n",
       "...        ...      ...                            ...            ...   \n",
       "15272  6020675  6020675                          184.0      14.045964   \n",
       "15273  6020722  6020722                          173.0      14.056792   \n",
       "15274  6021112  6021112                          177.0      13.402493   \n",
       "15275  6021539  6021539                          179.0      13.781823   \n",
       "15276  6024029  6024029                          179.0      13.271879   \n",
       "\n",
       "       trochanter_distance  shoulder_width  torso_length    humerus  \\\n",
       "0                30.091232       36.696399     54.043332  30.816804   \n",
       "1                30.813208       39.203992     58.668433  28.115456   \n",
       "2                29.777444       33.219853     57.685872  29.972729   \n",
       "3                28.838122       39.066233     57.527410  29.012823   \n",
       "4                30.273697       43.356069     61.348184  30.128320   \n",
       "...                    ...             ...           ...        ...   \n",
       "15272            32.669869       40.853551     66.046411  32.354476   \n",
       "15273            30.356562       39.199608     61.041006  28.799111   \n",
       "15274            30.995640       41.118337     60.108352  31.540230   \n",
       "15275            31.116022       38.579135     61.148256  30.777846   \n",
       "15276            30.125650       39.486231     63.618586  32.678000   \n",
       "\n",
       "           femur    forearm      tibia  \n",
       "0      48.489427  26.856522  40.743734  \n",
       "1      44.984030  26.628187  40.081134  \n",
       "2      46.587679  28.561098  39.566371  \n",
       "3      45.245573  27.292476  39.041382  \n",
       "4      48.528284  29.505318  42.783355  \n",
       "...          ...        ...        ...  \n",
       "15272  50.976459  29.572681  43.244042  \n",
       "15273  46.356024  29.852094  41.777482  \n",
       "15274  50.329159  30.128478  44.856165  \n",
       "15275  48.408803  28.966865  42.070750  \n",
       "15276  50.968299  29.406987  42.910290  \n",
       "\n",
       "[15277 rows x 11 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pheno = ['eid', 'standing_height_imaging_visit', 'head_diameter', 'trochanter_distance', 'shoulder_width',\n",
    "         'torso_length', 'humerus', 'femur', 'forearm', 'tibia']\n",
    "\n",
    "# male dataframe for gwas preparation\n",
    "male_skeletal = male_skeletal[pheno]\n",
    "\n",
    "male_skeletal_gwas = male_skeletal.iloc[:, 1:]\n",
    "\n",
    "ids = male_skeletal['eid'].tolist()\n",
    "\n",
    "male_skeletal_gwas['FID'] = ids\n",
    "male_skeletal_gwas['IID'] = ids\n",
    "\n",
    "# put FID and IID to the first two columns\n",
    "cols = male_skeletal_gwas.columns\n",
    "cols = cols[-2:].tolist() + cols[:-2].tolist()\n",
    "male_skeletal_gwas = male_skeletal_gwas[cols]\n",
    "\n",
    "male_skeletal_gwas"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "male_skeletal_gwas[['FID', 'IID', 'standing_height_imaging_visit']].to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/male_height_gwas_cm_240318.txt', index=False, sep = \" \")\n",
    "male_skeletal_gwas.drop(columns = ['standing_height_imaging_visit']).to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/male_skeletal_gwas_cm_240318.txt', index=False, sep = \" \")"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### get male and female no imaging eids in 400k population"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 184,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "402233"
      ]
     },
     "execution_count": 184,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_eids = pd.read_csv(\"UKB_Imaging_Genetics/GWAS_INPUT_DATA/geno_qc_eids_2022-07-28.txt\", header=None, sep = ' ')[0].tolist()\n",
    "\n",
    "len(all_eids)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 185,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>eid</th>\n",
       "      <th>sex</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1000011</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1000026</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1000032</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1000044</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1000058</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502485</th>\n",
       "      <td>6024895</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502486</th>\n",
       "      <td>6024901</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502487</th>\n",
       "      <td>6024913</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502488</th>\n",
       "      <td>6024922</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502489</th>\n",
       "      <td>2108075</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>502490 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            eid  sex\n",
       "0       1000011  0.0\n",
       "1       1000026  1.0\n",
       "2       1000032  0.0\n",
       "3       1000044  0.0\n",
       "4       1000058  0.0\n",
       "...         ...  ...\n",
       "502485  6024895  1.0\n",
       "502486  6024901  1.0\n",
       "502487  6024913  1.0\n",
       "502488  6024922  1.0\n",
       "502489  2108075  NaN\n",
       "\n",
       "[502490 rows x 2 columns]"
      ]
     },
     "execution_count": 185,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fid_info = pd.read_csv(\"../UKB_xray_image_info/fids/fid_disease/fid_info.csv\")[['eid', 'sex']]; fid_info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 186,
   "metadata": {},
   "outputs": [],
   "source": [
    "fid_info = fid_info[fid_info['eid'].isin(all_eids)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 187,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/alexxu/opt/miniconda3/envs/ml/lib/python3.8/site-packages/pandas/util/_decorators.py:311: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  return func(*args, **kwargs)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>eid</th>\n",
       "      <th>sex</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1000011</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1000026</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1000032</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>1000060</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>1000083</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502483</th>\n",
       "      <td>6024877</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502484</th>\n",
       "      <td>6024884</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502485</th>\n",
       "      <td>6024895</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502487</th>\n",
       "      <td>6024913</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502488</th>\n",
       "      <td>6024922</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>402233 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            eid  sex\n",
       "0       1000011  0.0\n",
       "1       1000026  1.0\n",
       "2       1000032  0.0\n",
       "5       1000060  0.0\n",
       "7       1000083  0.0\n",
       "...         ...  ...\n",
       "502483  6024877  1.0\n",
       "502484  6024884  1.0\n",
       "502485  6024895  1.0\n",
       "502487  6024913  1.0\n",
       "502488  6024922  1.0\n",
       "\n",
       "[402233 rows x 2 columns]"
      ]
     },
     "execution_count": 187,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fid_info.dropna(inplace=True); fid_info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 188,
   "metadata": {},
   "outputs": [],
   "source": [
    "fid_male = fid_info[fid_info['sex'] == 1]\n",
    "fid_female = fid_info[fid_info['sex'] == 0]\n",
    "\n",
    "male_imaging_eids = hip_pheno_male['eid'].tolist()\n",
    "female_imaging_eids = hip_pheno_female['eid'].tolist()\n",
    "\n",
    "fid_male_no_img = fid_male[~fid_male['eid'].isin(male_imaging_eids)]\n",
    "fid_female_no_img = fid_female[~fid_female['eid'].isin(female_imaging_eids)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 189,
   "metadata": {},
   "outputs": [],
   "source": [
    "fid_male_no_img[['eid', 'eid']].to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/eids_male_400k_no_img.txt', index=False, header=False, sep=' ')\n",
    "fid_female_no_img[['eid', 'eid']].to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/eids_female_400k_no_img.txt', index=False, header=False, sep=' ')"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### divide to train and test dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "hip_pheno_gwas_male = pd.read_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_male_cm_230713.txt', sep = \" \")\n",
    "hip_pheno_gwas_female = pd.read_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_female_cm_230713.txt', sep = \" \")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>pelvic_height</th>\n",
       "      <th>pelvic_width</th>\n",
       "      <th>pelvic_inlet_width</th>\n",
       "      <th>iliac_flare_ratio</th>\n",
       "      <th>oblique_pelvic_inlet_length</th>\n",
       "      <th>iliac_flare_angle</th>\n",
       "      <th>subpubic_angle</th>\n",
       "      <th>ear_left2ear_right</th>\n",
       "      <th>trochanter_left2trochanter_right</th>\n",
       "      <th>shoulder_width</th>\n",
       "      <th>arm_devide_torso</th>\n",
       "      <th>iliac_isthmus_breadth</th>\n",
       "      <th>acetabular_diameter</th>\n",
       "      <th>acetabular_inclination</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1000951</td>\n",
       "      <td>1000951</td>\n",
       "      <td>22.297293</td>\n",
       "      <td>30.113663</td>\n",
       "      <td>13.937174</td>\n",
       "      <td>0.462819</td>\n",
       "      <td>9.883806</td>\n",
       "      <td>97.533181</td>\n",
       "      <td>116.365190</td>\n",
       "      <td>14.112233</td>\n",
       "      <td>30.091232</td>\n",
       "      <td>36.696399</td>\n",
       "      <td>1.067168</td>\n",
       "      <td>4.164529</td>\n",
       "      <td>6.141783</td>\n",
       "      <td>60.747735</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1001512</td>\n",
       "      <td>1001512</td>\n",
       "      <td>23.742297</td>\n",
       "      <td>30.901775</td>\n",
       "      <td>13.669524</td>\n",
       "      <td>0.442354</td>\n",
       "      <td>9.562262</td>\n",
       "      <td>95.714405</td>\n",
       "      <td>117.491690</td>\n",
       "      <td>13.903518</td>\n",
       "      <td>30.813208</td>\n",
       "      <td>39.203992</td>\n",
       "      <td>0.933102</td>\n",
       "      <td>4.657824</td>\n",
       "      <td>6.035855</td>\n",
       "      <td>59.189839</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1001719</td>\n",
       "      <td>1001719</td>\n",
       "      <td>27.576413</td>\n",
       "      <td>34.451075</td>\n",
       "      <td>15.479994</td>\n",
       "      <td>0.449333</td>\n",
       "      <td>12.957804</td>\n",
       "      <td>95.958279</td>\n",
       "      <td>116.185643</td>\n",
       "      <td>13.372049</td>\n",
       "      <td>34.830301</td>\n",
       "      <td>43.214632</td>\n",
       "      <td>0.978208</td>\n",
       "      <td>4.809205</td>\n",
       "      <td>6.716948</td>\n",
       "      <td>55.647944</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1001921</td>\n",
       "      <td>1001921</td>\n",
       "      <td>22.861258</td>\n",
       "      <td>28.863430</td>\n",
       "      <td>14.217750</td>\n",
       "      <td>0.492587</td>\n",
       "      <td>9.897451</td>\n",
       "      <td>95.473293</td>\n",
       "      <td>116.159185</td>\n",
       "      <td>14.480994</td>\n",
       "      <td>29.777444</td>\n",
       "      <td>33.219853</td>\n",
       "      <td>1.014700</td>\n",
       "      <td>4.238295</td>\n",
       "      <td>5.834113</td>\n",
       "      <td>58.027399</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1001950</td>\n",
       "      <td>1001950</td>\n",
       "      <td>23.080688</td>\n",
       "      <td>29.385649</td>\n",
       "      <td>13.583558</td>\n",
       "      <td>0.462251</td>\n",
       "      <td>10.305746</td>\n",
       "      <td>95.952879</td>\n",
       "      <td>130.647738</td>\n",
       "      <td>14.042468</td>\n",
       "      <td>28.838122</td>\n",
       "      <td>39.066233</td>\n",
       "      <td>0.978756</td>\n",
       "      <td>4.438966</td>\n",
       "      <td>5.731393</td>\n",
       "      <td>57.335593</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14840</th>\n",
       "      <td>6023890</td>\n",
       "      <td>6023890</td>\n",
       "      <td>24.006938</td>\n",
       "      <td>30.013155</td>\n",
       "      <td>13.591668</td>\n",
       "      <td>0.452857</td>\n",
       "      <td>11.950734</td>\n",
       "      <td>94.685360</td>\n",
       "      <td>123.267297</td>\n",
       "      <td>14.496029</td>\n",
       "      <td>28.958036</td>\n",
       "      <td>40.495171</td>\n",
       "      <td>0.920514</td>\n",
       "      <td>4.505524</td>\n",
       "      <td>5.849752</td>\n",
       "      <td>57.046345</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14841</th>\n",
       "      <td>6024029</td>\n",
       "      <td>6024029</td>\n",
       "      <td>24.323043</td>\n",
       "      <td>29.804727</td>\n",
       "      <td>13.883682</td>\n",
       "      <td>0.465821</td>\n",
       "      <td>12.728717</td>\n",
       "      <td>85.308035</td>\n",
       "      <td>124.424809</td>\n",
       "      <td>13.271879</td>\n",
       "      <td>30.125650</td>\n",
       "      <td>39.486231</td>\n",
       "      <td>0.975894</td>\n",
       "      <td>4.298543</td>\n",
       "      <td>6.171569</td>\n",
       "      <td>59.544963</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14842</th>\n",
       "      <td>6024037</td>\n",
       "      <td>6024037</td>\n",
       "      <td>24.695284</td>\n",
       "      <td>31.178125</td>\n",
       "      <td>14.634467</td>\n",
       "      <td>0.469383</td>\n",
       "      <td>12.533393</td>\n",
       "      <td>95.014076</td>\n",
       "      <td>130.924304</td>\n",
       "      <td>14.358949</td>\n",
       "      <td>31.715789</td>\n",
       "      <td>42.005320</td>\n",
       "      <td>1.036042</td>\n",
       "      <td>4.439059</td>\n",
       "      <td>6.281782</td>\n",
       "      <td>58.682502</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14843</th>\n",
       "      <td>6024257</td>\n",
       "      <td>6024257</td>\n",
       "      <td>22.849056</td>\n",
       "      <td>30.738358</td>\n",
       "      <td>14.076906</td>\n",
       "      <td>0.457959</td>\n",
       "      <td>9.722564</td>\n",
       "      <td>102.209266</td>\n",
       "      <td>101.631995</td>\n",
       "      <td>13.457582</td>\n",
       "      <td>31.007227</td>\n",
       "      <td>38.590868</td>\n",
       "      <td>0.948712</td>\n",
       "      <td>4.628839</td>\n",
       "      <td>6.287657</td>\n",
       "      <td>54.167545</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14844</th>\n",
       "      <td>6024532</td>\n",
       "      <td>6024532</td>\n",
       "      <td>24.004784</td>\n",
       "      <td>32.248290</td>\n",
       "      <td>15.234202</td>\n",
       "      <td>0.472403</td>\n",
       "      <td>10.533469</td>\n",
       "      <td>105.193136</td>\n",
       "      <td>105.144526</td>\n",
       "      <td>14.392064</td>\n",
       "      <td>30.197318</td>\n",
       "      <td>39.487130</td>\n",
       "      <td>0.942854</td>\n",
       "      <td>4.722146</td>\n",
       "      <td>5.989986</td>\n",
       "      <td>55.439957</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>14845 rows × 16 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           FID      IID  pelvic_height  pelvic_width  pelvic_inlet_width  \\\n",
       "0      1000951  1000951      22.297293     30.113663           13.937174   \n",
       "1      1001512  1001512      23.742297     30.901775           13.669524   \n",
       "2      1001719  1001719      27.576413     34.451075           15.479994   \n",
       "3      1001921  1001921      22.861258     28.863430           14.217750   \n",
       "4      1001950  1001950      23.080688     29.385649           13.583558   \n",
       "...        ...      ...            ...           ...                 ...   \n",
       "14840  6023890  6023890      24.006938     30.013155           13.591668   \n",
       "14841  6024029  6024029      24.323043     29.804727           13.883682   \n",
       "14842  6024037  6024037      24.695284     31.178125           14.634467   \n",
       "14843  6024257  6024257      22.849056     30.738358           14.076906   \n",
       "14844  6024532  6024532      24.004784     32.248290           15.234202   \n",
       "\n",
       "       iliac_flare_ratio  oblique_pelvic_inlet_length  iliac_flare_angle  \\\n",
       "0               0.462819                     9.883806          97.533181   \n",
       "1               0.442354                     9.562262          95.714405   \n",
       "2               0.449333                    12.957804          95.958279   \n",
       "3               0.492587                     9.897451          95.473293   \n",
       "4               0.462251                    10.305746          95.952879   \n",
       "...                  ...                          ...                ...   \n",
       "14840           0.452857                    11.950734          94.685360   \n",
       "14841           0.465821                    12.728717          85.308035   \n",
       "14842           0.469383                    12.533393          95.014076   \n",
       "14843           0.457959                     9.722564         102.209266   \n",
       "14844           0.472403                    10.533469         105.193136   \n",
       "\n",
       "       subpubic_angle  ear_left2ear_right  trochanter_left2trochanter_right  \\\n",
       "0          116.365190           14.112233                         30.091232   \n",
       "1          117.491690           13.903518                         30.813208   \n",
       "2          116.185643           13.372049                         34.830301   \n",
       "3          116.159185           14.480994                         29.777444   \n",
       "4          130.647738           14.042468                         28.838122   \n",
       "...               ...                 ...                               ...   \n",
       "14840      123.267297           14.496029                         28.958036   \n",
       "14841      124.424809           13.271879                         30.125650   \n",
       "14842      130.924304           14.358949                         31.715789   \n",
       "14843      101.631995           13.457582                         31.007227   \n",
       "14844      105.144526           14.392064                         30.197318   \n",
       "\n",
       "       shoulder_width  arm_devide_torso  iliac_isthmus_breadth  \\\n",
       "0           36.696399          1.067168               4.164529   \n",
       "1           39.203992          0.933102               4.657824   \n",
       "2           43.214632          0.978208               4.809205   \n",
       "3           33.219853          1.014700               4.238295   \n",
       "4           39.066233          0.978756               4.438966   \n",
       "...               ...               ...                    ...   \n",
       "14840       40.495171          0.920514               4.505524   \n",
       "14841       39.486231          0.975894               4.298543   \n",
       "14842       42.005320          1.036042               4.439059   \n",
       "14843       38.590868          0.948712               4.628839   \n",
       "14844       39.487130          0.942854               4.722146   \n",
       "\n",
       "       acetabular_diameter  acetabular_inclination  \n",
       "0                 6.141783               60.747735  \n",
       "1                 6.035855               59.189839  \n",
       "2                 6.716948               55.647944  \n",
       "3                 5.834113               58.027399  \n",
       "4                 5.731393               57.335593  \n",
       "...                    ...                     ...  \n",
       "14840             5.849752               57.046345  \n",
       "14841             6.171569               59.544963  \n",
       "14842             6.281782               58.682502  \n",
       "14843             6.287657               54.167545  \n",
       "14844             5.989986               55.439957  \n",
       "\n",
       "[14845 rows x 16 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hip_pheno_gwas_male"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Male train dataframe shape: (10000, 16)\n",
      "Male test dataframe shape: (4845, 16)\n",
      "Female train dataframe shape: (10000, 16)\n",
      "Female test dataframe shape: (5432, 16)\n"
     ]
    }
   ],
   "source": [
    "# male\n",
    "\n",
    "random.seed(42)\n",
    "\n",
    "male_eids = hip_pheno_gwas_male['IID'].tolist()\n",
    "random.shuffle(male_eids)\n",
    "\n",
    "# Split the list into two parts\n",
    "male_eids_train = male_eids[:10000]  # First 12000 elements\n",
    "male_eids_test = male_eids[10000:]  # Remaining elements\n",
    "\n",
    "hip_pheno_gwas_male_train = hip_pheno_gwas_male[hip_pheno_gwas_male['FID'].isin(male_eids_train)]\n",
    "hip_pheno_gwas_male_test = hip_pheno_gwas_male[hip_pheno_gwas_male['FID'].isin(male_eids_test)]\n",
    "\n",
    "hip_pheno_gwas_male_train.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_male_train_cm_230801.txt', index=False, sep = \" \")\n",
    "hip_pheno_gwas_male_train.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_male_train_cm_no_header_230801.txt', index=False, header=False, sep = \" \")\n",
    "\n",
    "hip_pheno_gwas_male_test.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_male_test_cm_230801.txt', index=False, sep = \" \")\n",
    "hip_pheno_gwas_male_test.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_male_test_cm_no_header_230801.txt', index=False, header=False, sep = \" \")\n",
    "\n",
    "pd.DataFrame({\"FID\": male_eids_train, \"IID\": male_eids_train}).to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_eids_male_train.txt', index=False, header=False, sep=' ')\n",
    "pd.DataFrame({\"FID\": male_eids_test, \"IID\": male_eids_test}).to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_eids_male_test.txt', index=False, header=False, sep=' ')\n",
    "\n",
    "print(f\"Male train dataframe shape: {hip_pheno_gwas_male_train.shape}\")\n",
    "print(f\"Male test dataframe shape: {hip_pheno_gwas_male_test.shape}\")\n",
    "\n",
    "\n",
    "# female\n",
    "female_eids = hip_pheno_gwas_female['IID'].tolist()\n",
    "random.shuffle(female_eids)\n",
    "\n",
    "# Split the list into two parts\n",
    "female_eids_train = female_eids[:10000]  # First 12000 elements\n",
    "female_eids_test = female_eids[10000:]  # Remaining elements\n",
    "\n",
    "hip_pheno_gwas_female_train = hip_pheno_gwas_female[hip_pheno_gwas_female['FID'].isin(female_eids_train)]\n",
    "hip_pheno_gwas_female_test = hip_pheno_gwas_female[hip_pheno_gwas_female['FID'].isin(female_eids_test)]\n",
    "\n",
    "hip_pheno_gwas_female_train.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_female_train_cm_230801.txt', index=False, sep = \" \")\n",
    "hip_pheno_gwas_female_train.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_female_train_cm_no_header_230801.txt', index=False, header=False, sep = \" \")\n",
    "\n",
    "hip_pheno_gwas_female_test.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_female_test_cm_230801.txt', index=False, sep = \" \")\n",
    "hip_pheno_gwas_female_test.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_gwas_female_test_cm_no_header_230801.txt', index=False, header=False, sep = \" \")\n",
    "\n",
    "pd.DataFrame({\"FID\": female_eids_train, \"IID\": female_eids_train}).to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_eids_female_train.txt', index=False, header=False, sep=' ')\n",
    "pd.DataFrame({\"FID\": female_eids_test, \"IID\": female_eids_test}).to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_select_pheno_eids_female_test.txt', index=False, header=False, sep=' ')\n",
    "\n",
    "print(f\"Female train dataframe shape: {hip_pheno_gwas_female_train.shape}\")\n",
    "print(f\"Female test dataframe shape: {hip_pheno_gwas_female_test.shape}\")"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Generate categorical covar for male and female"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 428,
   "metadata": {},
   "outputs": [],
   "source": [
    "covar = pd.read_csv(\"UKB_Imaging_Genetics/GCTA_INPUT_DATA/cat_covar_all_eid_2021-10-25_no_header.txt\", sep = \" \", header = None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 431,
   "metadata": {},
   "outputs": [],
   "source": [
    "covar[[0, 1, 3]].to_csv('UKB_Imaging_Genetics/GCTA_INPUT_DATA/cat_covar_all_eid_2021-10-25_no_header_for_single_sex.txt', index=False, header=False, sep = \" \")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 432,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1000276</td>\n",
       "      <td>1000276</td>\n",
       "      <td>SS_210419_16.00.158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1000742</td>\n",
       "      <td>1000742</td>\n",
       "      <td>SS_210419_16.00.158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1000951</td>\n",
       "      <td>1000951</td>\n",
       "      <td>SS_212038_16.20.059</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1001025</td>\n",
       "      <td>1001025</td>\n",
       "      <td>SS_210419_16.00.158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1001091</td>\n",
       "      <td>1001091</td>\n",
       "      <td>SS_210419_16.00.158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31271</th>\n",
       "      <td>6024029</td>\n",
       "      <td>6024029</td>\n",
       "      <td>SS_210419_16.00.158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31272</th>\n",
       "      <td>6024037</td>\n",
       "      <td>6024037</td>\n",
       "      <td>SS_210419_16.20.059</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31273</th>\n",
       "      <td>6024119</td>\n",
       "      <td>6024119</td>\n",
       "      <td>SS_212038_16.20.059</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31274</th>\n",
       "      <td>6024257</td>\n",
       "      <td>6024257</td>\n",
       "      <td>SS_210419_16.20.059</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31275</th>\n",
       "      <td>6024532</td>\n",
       "      <td>6024532</td>\n",
       "      <td>SS_210419_16.20.059</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>31276 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             0        1                    3\n",
       "0      1000276  1000276  SS_210419_16.00.158\n",
       "1      1000742  1000742  SS_210419_16.00.158\n",
       "2      1000951  1000951  SS_212038_16.20.059\n",
       "3      1001025  1001025  SS_210419_16.00.158\n",
       "4      1001091  1001091  SS_210419_16.00.158\n",
       "...        ...      ...                  ...\n",
       "31271  6024029  6024029  SS_210419_16.00.158\n",
       "31272  6024037  6024037  SS_210419_16.20.059\n",
       "31273  6024119  6024119  SS_212038_16.20.059\n",
       "31274  6024257  6024257  SS_210419_16.20.059\n",
       "31275  6024532  6024532  SS_210419_16.20.059\n",
       "\n",
       "[31276 rows x 3 columns]"
      ]
     },
     "execution_count": 432,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "covar[[0, 1, 3]]"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Divide into 2 parts of tables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "cols = hip_pheno_gwas.columns.tolist()[2:]\n",
    "\n",
    "cols_part1 = cols[:len(cols)//2]\n",
    "cols_part2 = cols[len(cols)//2:]\n",
    "\n",
    "cols_part1 = ['FID', 'IID'] + cols_part1\n",
    "cols_part2 = ['FID', 'IID'] + cols_part2\n",
    "\n",
    "hip_pheno_gwas_part1 = hip_pheno_gwas[cols_part1]\n",
    "hip_pheno_gwas_part2 = hip_pheno_gwas[cols_part2]\n",
    "\n",
    "hip_pheno_gwas_part1.to_csv('UKB_Imaging_Genetics/GCTA_INPUT_DATA/hip_pheno_gwas_part1.txt', index=False, sep = \" \")\n",
    "hip_pheno_gwas_part1.to_csv('UKB_Imaging_Genetics/GCTA_INPUT_DATA/hip_pheno_gwas_part1_no_header.txt', index=False, header=False, sep = \" \")\n",
    "\n",
    "hip_pheno_gwas_part2.to_csv('UKB_Imaging_Genetics/GCTA_INPUT_DATA/hip_pheno_gwas_part2.txt', index=False, sep = \" \")\n",
    "hip_pheno_gwas_part2.to_csv('UKB_Imaging_Genetics/GCTA_INPUT_DATA/hip_pheno_gwas_part2_no_header.txt', index=False, header=False, sep = \" \")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>p_age</th>\n",
       "      <th>p_weight</th>\n",
       "      <th>p_height</th>\n",
       "      <th>iliac_crest_left2iliac_crest_right</th>\n",
       "      <th>iliac_crest_left2iliac_spine_left</th>\n",
       "      <th>iliac_crest_left2iliac_spine_right</th>\n",
       "      <th>iliac_crest_left2iliopubic_eminence_left</th>\n",
       "      <th>iliac_crest_left2iliopubic_eminence_right</th>\n",
       "      <th>...</th>\n",
       "      <th>inferior_pubic_ramus_left2sciatic_notch_right</th>\n",
       "      <th>inferior_pubic_ramus_left2sacrum</th>\n",
       "      <th>inferior_pubic_ramus_left2pubic_tubercle</th>\n",
       "      <th>inferior_pubic_ramus_left2trochanter_left</th>\n",
       "      <th>inferior_pubic_ramus_left2trochanter_right</th>\n",
       "      <th>inferior_pubic_ramus_left2obturator_left</th>\n",
       "      <th>inferior_pubic_ramus_left2obturator_right</th>\n",
       "      <th>inferior_pubic_ramus_left2sacrum_left</th>\n",
       "      <th>inferior_pubic_ramus_left2sacrum_right</th>\n",
       "      <th>inferior_pubic_ramus_left2inferior_iliac_spine_left</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>21811</th>\n",
       "      <td>1000276</td>\n",
       "      <td>1000276</td>\n",
       "      <td>67.0</td>\n",
       "      <td>60.0</td>\n",
       "      <td>159.0</td>\n",
       "      <td>0.119177</td>\n",
       "      <td>0.057544</td>\n",
       "      <td>0.161311</td>\n",
       "      <td>0.092231</td>\n",
       "      <td>0.161052</td>\n",
       "      <td>...</td>\n",
       "      <td>0.110155</td>\n",
       "      <td>0.104636</td>\n",
       "      <td>0.044298</td>\n",
       "      <td>0.069056</td>\n",
       "      <td>0.137080</td>\n",
       "      <td>0.013766</td>\n",
       "      <td>0.073653</td>\n",
       "      <td>0.086145</td>\n",
       "      <td>0.110220</td>\n",
       "      <td>0.074858</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21812</th>\n",
       "      <td>1000742</td>\n",
       "      <td>1000742</td>\n",
       "      <td>67.0</td>\n",
       "      <td>77.0</td>\n",
       "      <td>161.0</td>\n",
       "      <td>0.123848</td>\n",
       "      <td>0.038570</td>\n",
       "      <td>0.155576</td>\n",
       "      <td>0.084050</td>\n",
       "      <td>0.159198</td>\n",
       "      <td>...</td>\n",
       "      <td>0.104073</td>\n",
       "      <td>0.104152</td>\n",
       "      <td>0.041364</td>\n",
       "      <td>0.073949</td>\n",
       "      <td>0.130034</td>\n",
       "      <td>0.014362</td>\n",
       "      <td>0.065348</td>\n",
       "      <td>0.088393</td>\n",
       "      <td>0.109329</td>\n",
       "      <td>0.080233</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1000951</td>\n",
       "      <td>1000951</td>\n",
       "      <td>78.0</td>\n",
       "      <td>60.0</td>\n",
       "      <td>170.0</td>\n",
       "      <td>0.113652</td>\n",
       "      <td>0.036685</td>\n",
       "      <td>0.147139</td>\n",
       "      <td>0.075637</td>\n",
       "      <td>0.147038</td>\n",
       "      <td>...</td>\n",
       "      <td>0.095160</td>\n",
       "      <td>0.095916</td>\n",
       "      <td>0.043712</td>\n",
       "      <td>0.077659</td>\n",
       "      <td>0.118688</td>\n",
       "      <td>0.015033</td>\n",
       "      <td>0.053254</td>\n",
       "      <td>0.081753</td>\n",
       "      <td>0.096841</td>\n",
       "      <td>0.081542</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21813</th>\n",
       "      <td>1001025</td>\n",
       "      <td>1001025</td>\n",
       "      <td>55.0</td>\n",
       "      <td>69.0</td>\n",
       "      <td>167.0</td>\n",
       "      <td>0.115559</td>\n",
       "      <td>0.055194</td>\n",
       "      <td>0.162427</td>\n",
       "      <td>0.083520</td>\n",
       "      <td>0.157159</td>\n",
       "      <td>...</td>\n",
       "      <td>0.109018</td>\n",
       "      <td>0.107804</td>\n",
       "      <td>0.045590</td>\n",
       "      <td>0.068298</td>\n",
       "      <td>0.138567</td>\n",
       "      <td>0.013027</td>\n",
       "      <td>0.076324</td>\n",
       "      <td>0.090239</td>\n",
       "      <td>0.113508</td>\n",
       "      <td>0.073854</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21814</th>\n",
       "      <td>1001091</td>\n",
       "      <td>1001091</td>\n",
       "      <td>60.0</td>\n",
       "      <td>65.0</td>\n",
       "      <td>162.0</td>\n",
       "      <td>0.119257</td>\n",
       "      <td>0.037780</td>\n",
       "      <td>0.155524</td>\n",
       "      <td>0.079466</td>\n",
       "      <td>0.155074</td>\n",
       "      <td>...</td>\n",
       "      <td>0.106365</td>\n",
       "      <td>0.101284</td>\n",
       "      <td>0.049616</td>\n",
       "      <td>0.074924</td>\n",
       "      <td>0.139770</td>\n",
       "      <td>0.017390</td>\n",
       "      <td>0.072765</td>\n",
       "      <td>0.082077</td>\n",
       "      <td>0.104506</td>\n",
       "      <td>0.079882</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39042</th>\n",
       "      <td>6024029</td>\n",
       "      <td>6024029</td>\n",
       "      <td>50.0</td>\n",
       "      <td>79.0</td>\n",
       "      <td>180.0</td>\n",
       "      <td>0.115447</td>\n",
       "      <td>0.030139</td>\n",
       "      <td>0.141083</td>\n",
       "      <td>0.076052</td>\n",
       "      <td>0.142842</td>\n",
       "      <td>...</td>\n",
       "      <td>0.099910</td>\n",
       "      <td>0.102659</td>\n",
       "      <td>0.040403</td>\n",
       "      <td>0.072571</td>\n",
       "      <td>0.122383</td>\n",
       "      <td>0.013948</td>\n",
       "      <td>0.055061</td>\n",
       "      <td>0.084047</td>\n",
       "      <td>0.102126</td>\n",
       "      <td>0.080325</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21805</th>\n",
       "      <td>6024037</td>\n",
       "      <td>6024037</td>\n",
       "      <td>61.0</td>\n",
       "      <td>84.0</td>\n",
       "      <td>181.0</td>\n",
       "      <td>0.109468</td>\n",
       "      <td>0.046472</td>\n",
       "      <td>0.139892</td>\n",
       "      <td>0.080300</td>\n",
       "      <td>0.143638</td>\n",
       "      <td>...</td>\n",
       "      <td>0.094230</td>\n",
       "      <td>0.102061</td>\n",
       "      <td>0.037650</td>\n",
       "      <td>0.076924</td>\n",
       "      <td>0.121409</td>\n",
       "      <td>0.013391</td>\n",
       "      <td>0.051631</td>\n",
       "      <td>0.083635</td>\n",
       "      <td>0.098101</td>\n",
       "      <td>0.080632</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21806</th>\n",
       "      <td>6024119</td>\n",
       "      <td>6024119</td>\n",
       "      <td>68.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>165.0</td>\n",
       "      <td>0.117222</td>\n",
       "      <td>0.031103</td>\n",
       "      <td>0.146952</td>\n",
       "      <td>0.078930</td>\n",
       "      <td>0.149576</td>\n",
       "      <td>...</td>\n",
       "      <td>0.097808</td>\n",
       "      <td>0.095440</td>\n",
       "      <td>0.042180</td>\n",
       "      <td>0.070944</td>\n",
       "      <td>0.119109</td>\n",
       "      <td>0.013254</td>\n",
       "      <td>0.052838</td>\n",
       "      <td>0.077908</td>\n",
       "      <td>0.097668</td>\n",
       "      <td>0.081110</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21807</th>\n",
       "      <td>6024257</td>\n",
       "      <td>6024257</td>\n",
       "      <td>71.0</td>\n",
       "      <td>74.0</td>\n",
       "      <td>169.0</td>\n",
       "      <td>0.121064</td>\n",
       "      <td>0.039582</td>\n",
       "      <td>0.151179</td>\n",
       "      <td>0.079783</td>\n",
       "      <td>0.153874</td>\n",
       "      <td>...</td>\n",
       "      <td>0.099110</td>\n",
       "      <td>0.101200</td>\n",
       "      <td>0.048538</td>\n",
       "      <td>0.077974</td>\n",
       "      <td>0.129572</td>\n",
       "      <td>0.017587</td>\n",
       "      <td>0.053602</td>\n",
       "      <td>0.081821</td>\n",
       "      <td>0.100447</td>\n",
       "      <td>0.080782</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21809</th>\n",
       "      <td>6024532</td>\n",
       "      <td>6024532</td>\n",
       "      <td>67.0</td>\n",
       "      <td>75.0</td>\n",
       "      <td>175.0</td>\n",
       "      <td>0.112109</td>\n",
       "      <td>0.043879</td>\n",
       "      <td>0.150154</td>\n",
       "      <td>0.082683</td>\n",
       "      <td>0.151725</td>\n",
       "      <td>...</td>\n",
       "      <td>0.100897</td>\n",
       "      <td>0.105690</td>\n",
       "      <td>0.050147</td>\n",
       "      <td>0.074927</td>\n",
       "      <td>0.124752</td>\n",
       "      <td>0.016608</td>\n",
       "      <td>0.059372</td>\n",
       "      <td>0.084845</td>\n",
       "      <td>0.103813</td>\n",
       "      <td>0.080637</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>30997 rows × 135 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           FID      IID  p_age  p_weight  p_height  \\\n",
       "21811  1000276  1000276   67.0      60.0     159.0   \n",
       "21812  1000742  1000742   67.0      77.0     161.0   \n",
       "2      1000951  1000951   78.0      60.0     170.0   \n",
       "21813  1001025  1001025   55.0      69.0     167.0   \n",
       "21814  1001091  1001091   60.0      65.0     162.0   \n",
       "...        ...      ...    ...       ...       ...   \n",
       "39042  6024029  6024029   50.0      79.0     180.0   \n",
       "21805  6024037  6024037   61.0      84.0     181.0   \n",
       "21806  6024119  6024119   68.0      70.0     165.0   \n",
       "21807  6024257  6024257   71.0      74.0     169.0   \n",
       "21809  6024532  6024532   67.0      75.0     175.0   \n",
       "\n",
       "       iliac_crest_left2iliac_crest_right  iliac_crest_left2iliac_spine_left  \\\n",
       "21811                            0.119177                           0.057544   \n",
       "21812                            0.123848                           0.038570   \n",
       "2                                0.113652                           0.036685   \n",
       "21813                            0.115559                           0.055194   \n",
       "21814                            0.119257                           0.037780   \n",
       "...                                   ...                                ...   \n",
       "39042                            0.115447                           0.030139   \n",
       "21805                            0.109468                           0.046472   \n",
       "21806                            0.117222                           0.031103   \n",
       "21807                            0.121064                           0.039582   \n",
       "21809                            0.112109                           0.043879   \n",
       "\n",
       "       iliac_crest_left2iliac_spine_right  \\\n",
       "21811                            0.161311   \n",
       "21812                            0.155576   \n",
       "2                                0.147139   \n",
       "21813                            0.162427   \n",
       "21814                            0.155524   \n",
       "...                                   ...   \n",
       "39042                            0.141083   \n",
       "21805                            0.139892   \n",
       "21806                            0.146952   \n",
       "21807                            0.151179   \n",
       "21809                            0.150154   \n",
       "\n",
       "       iliac_crest_left2iliopubic_eminence_left  \\\n",
       "21811                                  0.092231   \n",
       "21812                                  0.084050   \n",
       "2                                      0.075637   \n",
       "21813                                  0.083520   \n",
       "21814                                  0.079466   \n",
       "...                                         ...   \n",
       "39042                                  0.076052   \n",
       "21805                                  0.080300   \n",
       "21806                                  0.078930   \n",
       "21807                                  0.079783   \n",
       "21809                                  0.082683   \n",
       "\n",
       "       iliac_crest_left2iliopubic_eminence_right  ...  \\\n",
       "21811                                   0.161052  ...   \n",
       "21812                                   0.159198  ...   \n",
       "2                                       0.147038  ...   \n",
       "21813                                   0.157159  ...   \n",
       "21814                                   0.155074  ...   \n",
       "...                                          ...  ...   \n",
       "39042                                   0.142842  ...   \n",
       "21805                                   0.143638  ...   \n",
       "21806                                   0.149576  ...   \n",
       "21807                                   0.153874  ...   \n",
       "21809                                   0.151725  ...   \n",
       "\n",
       "       inferior_pubic_ramus_left2sciatic_notch_right  \\\n",
       "21811                                       0.110155   \n",
       "21812                                       0.104073   \n",
       "2                                           0.095160   \n",
       "21813                                       0.109018   \n",
       "21814                                       0.106365   \n",
       "...                                              ...   \n",
       "39042                                       0.099910   \n",
       "21805                                       0.094230   \n",
       "21806                                       0.097808   \n",
       "21807                                       0.099110   \n",
       "21809                                       0.100897   \n",
       "\n",
       "       inferior_pubic_ramus_left2sacrum  \\\n",
       "21811                          0.104636   \n",
       "21812                          0.104152   \n",
       "2                              0.095916   \n",
       "21813                          0.107804   \n",
       "21814                          0.101284   \n",
       "...                                 ...   \n",
       "39042                          0.102659   \n",
       "21805                          0.102061   \n",
       "21806                          0.095440   \n",
       "21807                          0.101200   \n",
       "21809                          0.105690   \n",
       "\n",
       "       inferior_pubic_ramus_left2pubic_tubercle  \\\n",
       "21811                                  0.044298   \n",
       "21812                                  0.041364   \n",
       "2                                      0.043712   \n",
       "21813                                  0.045590   \n",
       "21814                                  0.049616   \n",
       "...                                         ...   \n",
       "39042                                  0.040403   \n",
       "21805                                  0.037650   \n",
       "21806                                  0.042180   \n",
       "21807                                  0.048538   \n",
       "21809                                  0.050147   \n",
       "\n",
       "       inferior_pubic_ramus_left2trochanter_left  \\\n",
       "21811                                   0.069056   \n",
       "21812                                   0.073949   \n",
       "2                                       0.077659   \n",
       "21813                                   0.068298   \n",
       "21814                                   0.074924   \n",
       "...                                          ...   \n",
       "39042                                   0.072571   \n",
       "21805                                   0.076924   \n",
       "21806                                   0.070944   \n",
       "21807                                   0.077974   \n",
       "21809                                   0.074927   \n",
       "\n",
       "       inferior_pubic_ramus_left2trochanter_right  \\\n",
       "21811                                    0.137080   \n",
       "21812                                    0.130034   \n",
       "2                                        0.118688   \n",
       "21813                                    0.138567   \n",
       "21814                                    0.139770   \n",
       "...                                           ...   \n",
       "39042                                    0.122383   \n",
       "21805                                    0.121409   \n",
       "21806                                    0.119109   \n",
       "21807                                    0.129572   \n",
       "21809                                    0.124752   \n",
       "\n",
       "       inferior_pubic_ramus_left2obturator_left  \\\n",
       "21811                                  0.013766   \n",
       "21812                                  0.014362   \n",
       "2                                      0.015033   \n",
       "21813                                  0.013027   \n",
       "21814                                  0.017390   \n",
       "...                                         ...   \n",
       "39042                                  0.013948   \n",
       "21805                                  0.013391   \n",
       "21806                                  0.013254   \n",
       "21807                                  0.017587   \n",
       "21809                                  0.016608   \n",
       "\n",
       "       inferior_pubic_ramus_left2obturator_right  \\\n",
       "21811                                   0.073653   \n",
       "21812                                   0.065348   \n",
       "2                                       0.053254   \n",
       "21813                                   0.076324   \n",
       "21814                                   0.072765   \n",
       "...                                          ...   \n",
       "39042                                   0.055061   \n",
       "21805                                   0.051631   \n",
       "21806                                   0.052838   \n",
       "21807                                   0.053602   \n",
       "21809                                   0.059372   \n",
       "\n",
       "       inferior_pubic_ramus_left2sacrum_left  \\\n",
       "21811                               0.086145   \n",
       "21812                               0.088393   \n",
       "2                                   0.081753   \n",
       "21813                               0.090239   \n",
       "21814                               0.082077   \n",
       "...                                      ...   \n",
       "39042                               0.084047   \n",
       "21805                               0.083635   \n",
       "21806                               0.077908   \n",
       "21807                               0.081821   \n",
       "21809                               0.084845   \n",
       "\n",
       "       inferior_pubic_ramus_left2sacrum_right  \\\n",
       "21811                                0.110220   \n",
       "21812                                0.109329   \n",
       "2                                    0.096841   \n",
       "21813                                0.113508   \n",
       "21814                                0.104506   \n",
       "...                                       ...   \n",
       "39042                                0.102126   \n",
       "21805                                0.098101   \n",
       "21806                                0.097668   \n",
       "21807                                0.100447   \n",
       "21809                                0.103813   \n",
       "\n",
       "       inferior_pubic_ramus_left2inferior_iliac_spine_left  \n",
       "21811                                           0.074858    \n",
       "21812                                           0.080233    \n",
       "2                                               0.081542    \n",
       "21813                                           0.073854    \n",
       "21814                                           0.079882    \n",
       "...                                                  ...    \n",
       "39042                                           0.080325    \n",
       "21805                                           0.080632    \n",
       "21806                                           0.081110    \n",
       "21807                                           0.080782    \n",
       "21809                                           0.080637    \n",
       "\n",
       "[30997 rows x 135 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hip_pheno_gwas_part1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>inferior_pubic_ramus_left2inferior_iliac_spine_right</th>\n",
       "      <th>inferior_pubic_ramus_left2acetabular_inferior_left</th>\n",
       "      <th>inferior_pubic_ramus_left2acetabular_inferior_right</th>\n",
       "      <th>inferior_pubic_ramus_right2pubic_arch</th>\n",
       "      <th>inferior_pubic_ramus_right2sciatic_notch_left</th>\n",
       "      <th>inferior_pubic_ramus_right2sciatic_notch_right</th>\n",
       "      <th>inferior_pubic_ramus_right2sacrum</th>\n",
       "      <th>inferior_pubic_ramus_right2pubic_tubercle</th>\n",
       "      <th>...</th>\n",
       "      <th>hip_height</th>\n",
       "      <th>sacrum_ratio_w2h</th>\n",
       "      <th>hip_ratio_w2h</th>\n",
       "      <th>sacrum2pubic_tubercle_devide_ear_left2ear_right</th>\n",
       "      <th>sacrum_left2sacrum_right_devide_ear_left2ear_right</th>\n",
       "      <th>sciatic_notch_left2sciatic_notch_right_divide_ear_left2ear_right</th>\n",
       "      <th>pubic_arch_angle</th>\n",
       "      <th>acetabular_inclination_left</th>\n",
       "      <th>acetabular_inclination_right</th>\n",
       "      <th>acetabular_inclination_diff</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>21811</th>\n",
       "      <td>1000276</td>\n",
       "      <td>1000276</td>\n",
       "      <td>0.129266</td>\n",
       "      <td>0.033702</td>\n",
       "      <td>0.097956</td>\n",
       "      <td>0.039689</td>\n",
       "      <td>0.113500</td>\n",
       "      <td>0.070933</td>\n",
       "      <td>0.107401</td>\n",
       "      <td>0.048149</td>\n",
       "      <td>...</td>\n",
       "      <td>0.144527</td>\n",
       "      <td>1.414572</td>\n",
       "      <td>1.390511</td>\n",
       "      <td>0.834930</td>\n",
       "      <td>0.828164</td>\n",
       "      <td>1.181069</td>\n",
       "      <td>142.678044</td>\n",
       "      <td>53.033866</td>\n",
       "      <td>57.859459</td>\n",
       "      <td>-4.825592</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21812</th>\n",
       "      <td>1000742</td>\n",
       "      <td>1000742</td>\n",
       "      <td>0.122910</td>\n",
       "      <td>0.034450</td>\n",
       "      <td>0.090896</td>\n",
       "      <td>0.035200</td>\n",
       "      <td>0.108085</td>\n",
       "      <td>0.070176</td>\n",
       "      <td>0.106577</td>\n",
       "      <td>0.041911</td>\n",
       "      <td>...</td>\n",
       "      <td>0.139060</td>\n",
       "      <td>1.251339</td>\n",
       "      <td>1.334308</td>\n",
       "      <td>0.886952</td>\n",
       "      <td>0.765835</td>\n",
       "      <td>1.109878</td>\n",
       "      <td>141.231641</td>\n",
       "      <td>54.070014</td>\n",
       "      <td>54.815992</td>\n",
       "      <td>-0.745978</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1000951</td>\n",
       "      <td>1000951</td>\n",
       "      <td>0.115608</td>\n",
       "      <td>0.035556</td>\n",
       "      <td>0.080277</td>\n",
       "      <td>0.030913</td>\n",
       "      <td>0.098890</td>\n",
       "      <td>0.070933</td>\n",
       "      <td>0.097910</td>\n",
       "      <td>0.044078</td>\n",
       "      <td>...</td>\n",
       "      <td>0.131161</td>\n",
       "      <td>1.410102</td>\n",
       "      <td>1.350552</td>\n",
       "      <td>0.700372</td>\n",
       "      <td>0.739401</td>\n",
       "      <td>0.987595</td>\n",
       "      <td>116.365190</td>\n",
       "      <td>51.546225</td>\n",
       "      <td>55.132446</td>\n",
       "      <td>-3.586221</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21813</th>\n",
       "      <td>1001025</td>\n",
       "      <td>1001025</td>\n",
       "      <td>0.130874</td>\n",
       "      <td>0.032311</td>\n",
       "      <td>0.101586</td>\n",
       "      <td>0.038092</td>\n",
       "      <td>0.111774</td>\n",
       "      <td>0.066583</td>\n",
       "      <td>0.108825</td>\n",
       "      <td>0.042733</td>\n",
       "      <td>...</td>\n",
       "      <td>0.134804</td>\n",
       "      <td>1.215281</td>\n",
       "      <td>1.486036</td>\n",
       "      <td>1.043131</td>\n",
       "      <td>0.845409</td>\n",
       "      <td>1.267698</td>\n",
       "      <td>159.703035</td>\n",
       "      <td>56.689420</td>\n",
       "      <td>56.406852</td>\n",
       "      <td>0.282569</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21814</th>\n",
       "      <td>1001091</td>\n",
       "      <td>1001091</td>\n",
       "      <td>0.128806</td>\n",
       "      <td>0.037393</td>\n",
       "      <td>0.099376</td>\n",
       "      <td>0.041285</td>\n",
       "      <td>0.111245</td>\n",
       "      <td>0.070655</td>\n",
       "      <td>0.102082</td>\n",
       "      <td>0.049782</td>\n",
       "      <td>...</td>\n",
       "      <td>0.139348</td>\n",
       "      <td>1.445791</td>\n",
       "      <td>1.348099</td>\n",
       "      <td>0.796500</td>\n",
       "      <td>0.836422</td>\n",
       "      <td>1.151573</td>\n",
       "      <td>131.343366</td>\n",
       "      <td>59.651556</td>\n",
       "      <td>61.152404</td>\n",
       "      <td>-1.500849</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39042</th>\n",
       "      <td>6024029</td>\n",
       "      <td>6024029</td>\n",
       "      <td>0.117885</td>\n",
       "      <td>0.034738</td>\n",
       "      <td>0.081298</td>\n",
       "      <td>0.030278</td>\n",
       "      <td>0.098501</td>\n",
       "      <td>0.075739</td>\n",
       "      <td>0.102122</td>\n",
       "      <td>0.038018</td>\n",
       "      <td>...</td>\n",
       "      <td>0.135128</td>\n",
       "      <td>1.090737</td>\n",
       "      <td>1.225370</td>\n",
       "      <td>0.959074</td>\n",
       "      <td>0.752686</td>\n",
       "      <td>1.046098</td>\n",
       "      <td>124.424809</td>\n",
       "      <td>57.535146</td>\n",
       "      <td>57.062575</td>\n",
       "      <td>0.472571</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21805</th>\n",
       "      <td>6024037</td>\n",
       "      <td>6024037</td>\n",
       "      <td>0.112367</td>\n",
       "      <td>0.036132</td>\n",
       "      <td>0.077881</td>\n",
       "      <td>0.032549</td>\n",
       "      <td>0.099637</td>\n",
       "      <td>0.069446</td>\n",
       "      <td>0.102245</td>\n",
       "      <td>0.042428</td>\n",
       "      <td>...</td>\n",
       "      <td>0.136438</td>\n",
       "      <td>1.167638</td>\n",
       "      <td>1.262513</td>\n",
       "      <td>0.872863</td>\n",
       "      <td>0.770157</td>\n",
       "      <td>1.019188</td>\n",
       "      <td>130.924304</td>\n",
       "      <td>53.310400</td>\n",
       "      <td>57.006480</td>\n",
       "      <td>-3.696080</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21806</th>\n",
       "      <td>6024119</td>\n",
       "      <td>6024119</td>\n",
       "      <td>0.121178</td>\n",
       "      <td>0.032904</td>\n",
       "      <td>0.077474</td>\n",
       "      <td>0.029162</td>\n",
       "      <td>0.095000</td>\n",
       "      <td>0.072398</td>\n",
       "      <td>0.095824</td>\n",
       "      <td>0.039736</td>\n",
       "      <td>...</td>\n",
       "      <td>0.132217</td>\n",
       "      <td>1.317724</td>\n",
       "      <td>1.301696</td>\n",
       "      <td>0.762527</td>\n",
       "      <td>0.765581</td>\n",
       "      <td>1.004801</td>\n",
       "      <td>115.933030</td>\n",
       "      <td>52.860920</td>\n",
       "      <td>50.817846</td>\n",
       "      <td>2.043073</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21807</th>\n",
       "      <td>6024257</td>\n",
       "      <td>6024257</td>\n",
       "      <td>0.121377</td>\n",
       "      <td>0.034333</td>\n",
       "      <td>0.081964</td>\n",
       "      <td>0.033230</td>\n",
       "      <td>0.097496</td>\n",
       "      <td>0.072068</td>\n",
       "      <td>0.100179</td>\n",
       "      <td>0.046828</td>\n",
       "      <td>...</td>\n",
       "      <td>0.135202</td>\n",
       "      <td>1.447859</td>\n",
       "      <td>1.345279</td>\n",
       "      <td>0.722460</td>\n",
       "      <td>0.804753</td>\n",
       "      <td>1.046020</td>\n",
       "      <td>101.631995</td>\n",
       "      <td>53.945919</td>\n",
       "      <td>48.201155</td>\n",
       "      <td>5.744764</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21809</th>\n",
       "      <td>6024532</td>\n",
       "      <td>6024532</td>\n",
       "      <td>0.121740</td>\n",
       "      <td>0.036339</td>\n",
       "      <td>0.085064</td>\n",
       "      <td>0.037902</td>\n",
       "      <td>0.103527</td>\n",
       "      <td>0.072056</td>\n",
       "      <td>0.106921</td>\n",
       "      <td>0.052569</td>\n",
       "      <td>...</td>\n",
       "      <td>0.137170</td>\n",
       "      <td>1.446266</td>\n",
       "      <td>1.343411</td>\n",
       "      <td>0.731894</td>\n",
       "      <td>0.771329</td>\n",
       "      <td>1.058514</td>\n",
       "      <td>105.144526</td>\n",
       "      <td>53.038293</td>\n",
       "      <td>53.927739</td>\n",
       "      <td>-0.889446</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>30997 rows × 136 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           FID      IID  inferior_pubic_ramus_left2inferior_iliac_spine_right  \\\n",
       "21811  1000276  1000276                                           0.129266      \n",
       "21812  1000742  1000742                                           0.122910      \n",
       "2      1000951  1000951                                           0.115608      \n",
       "21813  1001025  1001025                                           0.130874      \n",
       "21814  1001091  1001091                                           0.128806      \n",
       "...        ...      ...                                                ...      \n",
       "39042  6024029  6024029                                           0.117885      \n",
       "21805  6024037  6024037                                           0.112367      \n",
       "21806  6024119  6024119                                           0.121178      \n",
       "21807  6024257  6024257                                           0.121377      \n",
       "21809  6024532  6024532                                           0.121740      \n",
       "\n",
       "       inferior_pubic_ramus_left2acetabular_inferior_left  \\\n",
       "21811                                           0.033702    \n",
       "21812                                           0.034450    \n",
       "2                                               0.035556    \n",
       "21813                                           0.032311    \n",
       "21814                                           0.037393    \n",
       "...                                                  ...    \n",
       "39042                                           0.034738    \n",
       "21805                                           0.036132    \n",
       "21806                                           0.032904    \n",
       "21807                                           0.034333    \n",
       "21809                                           0.036339    \n",
       "\n",
       "       inferior_pubic_ramus_left2acetabular_inferior_right  \\\n",
       "21811                                           0.097956     \n",
       "21812                                           0.090896     \n",
       "2                                               0.080277     \n",
       "21813                                           0.101586     \n",
       "21814                                           0.099376     \n",
       "...                                                  ...     \n",
       "39042                                           0.081298     \n",
       "21805                                           0.077881     \n",
       "21806                                           0.077474     \n",
       "21807                                           0.081964     \n",
       "21809                                           0.085064     \n",
       "\n",
       "       inferior_pubic_ramus_right2pubic_arch  \\\n",
       "21811                               0.039689   \n",
       "21812                               0.035200   \n",
       "2                                   0.030913   \n",
       "21813                               0.038092   \n",
       "21814                               0.041285   \n",
       "...                                      ...   \n",
       "39042                               0.030278   \n",
       "21805                               0.032549   \n",
       "21806                               0.029162   \n",
       "21807                               0.033230   \n",
       "21809                               0.037902   \n",
       "\n",
       "       inferior_pubic_ramus_right2sciatic_notch_left  \\\n",
       "21811                                       0.113500   \n",
       "21812                                       0.108085   \n",
       "2                                           0.098890   \n",
       "21813                                       0.111774   \n",
       "21814                                       0.111245   \n",
       "...                                              ...   \n",
       "39042                                       0.098501   \n",
       "21805                                       0.099637   \n",
       "21806                                       0.095000   \n",
       "21807                                       0.097496   \n",
       "21809                                       0.103527   \n",
       "\n",
       "       inferior_pubic_ramus_right2sciatic_notch_right  \\\n",
       "21811                                        0.070933   \n",
       "21812                                        0.070176   \n",
       "2                                            0.070933   \n",
       "21813                                        0.066583   \n",
       "21814                                        0.070655   \n",
       "...                                               ...   \n",
       "39042                                        0.075739   \n",
       "21805                                        0.069446   \n",
       "21806                                        0.072398   \n",
       "21807                                        0.072068   \n",
       "21809                                        0.072056   \n",
       "\n",
       "       inferior_pubic_ramus_right2sacrum  \\\n",
       "21811                           0.107401   \n",
       "21812                           0.106577   \n",
       "2                               0.097910   \n",
       "21813                           0.108825   \n",
       "21814                           0.102082   \n",
       "...                                  ...   \n",
       "39042                           0.102122   \n",
       "21805                           0.102245   \n",
       "21806                           0.095824   \n",
       "21807                           0.100179   \n",
       "21809                           0.106921   \n",
       "\n",
       "       inferior_pubic_ramus_right2pubic_tubercle  ...  hip_height  \\\n",
       "21811                                   0.048149  ...    0.144527   \n",
       "21812                                   0.041911  ...    0.139060   \n",
       "2                                       0.044078  ...    0.131161   \n",
       "21813                                   0.042733  ...    0.134804   \n",
       "21814                                   0.049782  ...    0.139348   \n",
       "...                                          ...  ...         ...   \n",
       "39042                                   0.038018  ...    0.135128   \n",
       "21805                                   0.042428  ...    0.136438   \n",
       "21806                                   0.039736  ...    0.132217   \n",
       "21807                                   0.046828  ...    0.135202   \n",
       "21809                                   0.052569  ...    0.137170   \n",
       "\n",
       "       sacrum_ratio_w2h  hip_ratio_w2h  \\\n",
       "21811          1.414572       1.390511   \n",
       "21812          1.251339       1.334308   \n",
       "2              1.410102       1.350552   \n",
       "21813          1.215281       1.486036   \n",
       "21814          1.445791       1.348099   \n",
       "...                 ...            ...   \n",
       "39042          1.090737       1.225370   \n",
       "21805          1.167638       1.262513   \n",
       "21806          1.317724       1.301696   \n",
       "21807          1.447859       1.345279   \n",
       "21809          1.446266       1.343411   \n",
       "\n",
       "       sacrum2pubic_tubercle_devide_ear_left2ear_right  \\\n",
       "21811                                         0.834930   \n",
       "21812                                         0.886952   \n",
       "2                                             0.700372   \n",
       "21813                                         1.043131   \n",
       "21814                                         0.796500   \n",
       "...                                                ...   \n",
       "39042                                         0.959074   \n",
       "21805                                         0.872863   \n",
       "21806                                         0.762527   \n",
       "21807                                         0.722460   \n",
       "21809                                         0.731894   \n",
       "\n",
       "       sacrum_left2sacrum_right_devide_ear_left2ear_right  \\\n",
       "21811                                           0.828164    \n",
       "21812                                           0.765835    \n",
       "2                                               0.739401    \n",
       "21813                                           0.845409    \n",
       "21814                                           0.836422    \n",
       "...                                                  ...    \n",
       "39042                                           0.752686    \n",
       "21805                                           0.770157    \n",
       "21806                                           0.765581    \n",
       "21807                                           0.804753    \n",
       "21809                                           0.771329    \n",
       "\n",
       "       sciatic_notch_left2sciatic_notch_right_divide_ear_left2ear_right  \\\n",
       "21811                                           1.181069                  \n",
       "21812                                           1.109878                  \n",
       "2                                               0.987595                  \n",
       "21813                                           1.267698                  \n",
       "21814                                           1.151573                  \n",
       "...                                                  ...                  \n",
       "39042                                           1.046098                  \n",
       "21805                                           1.019188                  \n",
       "21806                                           1.004801                  \n",
       "21807                                           1.046020                  \n",
       "21809                                           1.058514                  \n",
       "\n",
       "       pubic_arch_angle  acetabular_inclination_left  \\\n",
       "21811        142.678044                    53.033866   \n",
       "21812        141.231641                    54.070014   \n",
       "2            116.365190                    51.546225   \n",
       "21813        159.703035                    56.689420   \n",
       "21814        131.343366                    59.651556   \n",
       "...                 ...                          ...   \n",
       "39042        124.424809                    57.535146   \n",
       "21805        130.924304                    53.310400   \n",
       "21806        115.933030                    52.860920   \n",
       "21807        101.631995                    53.945919   \n",
       "21809        105.144526                    53.038293   \n",
       "\n",
       "       acetabular_inclination_right  acetabular_inclination_diff  \n",
       "21811                     57.859459                    -4.825592  \n",
       "21812                     54.815992                    -0.745978  \n",
       "2                         55.132446                    -3.586221  \n",
       "21813                     56.406852                     0.282569  \n",
       "21814                     61.152404                    -1.500849  \n",
       "...                             ...                          ...  \n",
       "39042                     57.062575                     0.472571  \n",
       "21805                     57.006480                    -3.696080  \n",
       "21806                     50.817846                     2.043073  \n",
       "21807                     48.201155                     5.744764  \n",
       "21809                     53.927739                    -0.889446  \n",
       "\n",
       "[30997 rows x 136 columns]"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hip_pheno_gwas_part2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Double check with Eucharist result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>image_id</th>\n",
       "      <th>file_name</th>\n",
       "      <th>eid</th>\n",
       "      <th>sex</th>\n",
       "      <th>standing_height</th>\n",
       "      <th>weight</th>\n",
       "      <th>bmi</th>\n",
       "      <th>age</th>\n",
       "      <th>pelvic_height</th>\n",
       "      <th>pelvic_width</th>\n",
       "      <th>...</th>\n",
       "      <th>oblique_pelvic_inlet_length</th>\n",
       "      <th>iliac_flare_angle</th>\n",
       "      <th>subpubic_angle</th>\n",
       "      <th>ear_left2ear_right</th>\n",
       "      <th>trochanter_left2trochanter_right</th>\n",
       "      <th>shoulder_width</th>\n",
       "      <th>arm_devide_torso</th>\n",
       "      <th>iliac_isthmus_breadth</th>\n",
       "      <th>acetabular_diameter</th>\n",
       "      <th>acetabular_inclination</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>08760_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20160630134650.1.1...</td>\n",
       "      <td>1000276</td>\n",
       "      <td>0.0</td>\n",
       "      <td>160.0</td>\n",
       "      <td>61.5</td>\n",
       "      <td>24.0234</td>\n",
       "      <td>60.0</td>\n",
       "      <td>22.979831</td>\n",
       "      <td>31.953718</td>\n",
       "      <td>...</td>\n",
       "      <td>11.390012</td>\n",
       "      <td>101.719769</td>\n",
       "      <td>142.678044</td>\n",
       "      <td>13.641873</td>\n",
       "      <td>29.618177</td>\n",
       "      <td>35.288068</td>\n",
       "      <td>0.957793</td>\n",
       "      <td>3.648129</td>\n",
       "      <td>5.369424</td>\n",
       "      <td>63.070462</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>12347_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20170305174208.1.1...</td>\n",
       "      <td>1000742</td>\n",
       "      <td>0.0</td>\n",
       "      <td>160.0</td>\n",
       "      <td>77.4</td>\n",
       "      <td>30.2344</td>\n",
       "      <td>60.0</td>\n",
       "      <td>22.388605</td>\n",
       "      <td>29.873284</td>\n",
       "      <td>...</td>\n",
       "      <td>12.005250</td>\n",
       "      <td>90.615136</td>\n",
       "      <td>141.231641</td>\n",
       "      <td>13.535399</td>\n",
       "      <td>29.360838</td>\n",
       "      <td>36.799830</td>\n",
       "      <td>0.892762</td>\n",
       "      <td>3.725955</td>\n",
       "      <td>5.469268</td>\n",
       "      <td>58.032066</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14366_82</td>\n",
       "      <td>1.2.840.113619.2.110.212038.20190307144208.1.1...</td>\n",
       "      <td>1000951</td>\n",
       "      <td>1.0</td>\n",
       "      <td>166.0</td>\n",
       "      <td>60.8</td>\n",
       "      <td>22.0642</td>\n",
       "      <td>66.0</td>\n",
       "      <td>22.297293</td>\n",
       "      <td>30.113663</td>\n",
       "      <td>...</td>\n",
       "      <td>9.883806</td>\n",
       "      <td>97.533181</td>\n",
       "      <td>116.365190</td>\n",
       "      <td>14.112233</td>\n",
       "      <td>30.091232</td>\n",
       "      <td>36.696399</td>\n",
       "      <td>1.067168</td>\n",
       "      <td>4.164529</td>\n",
       "      <td>6.141783</td>\n",
       "      <td>60.747735</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>06366_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20151211144320.1.1...</td>\n",
       "      <td>1001025</td>\n",
       "      <td>0.0</td>\n",
       "      <td>167.0</td>\n",
       "      <td>68.9</td>\n",
       "      <td>24.7051</td>\n",
       "      <td>49.0</td>\n",
       "      <td>22.512247</td>\n",
       "      <td>33.454000</td>\n",
       "      <td>...</td>\n",
       "      <td>13.418752</td>\n",
       "      <td>102.935020</td>\n",
       "      <td>159.703035</td>\n",
       "      <td>12.863914</td>\n",
       "      <td>30.934761</td>\n",
       "      <td>35.136934</td>\n",
       "      <td>0.895883</td>\n",
       "      <td>4.080365</td>\n",
       "      <td>5.154912</td>\n",
       "      <td>61.323156</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>14857_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20170902122319.1.1...</td>\n",
       "      <td>1001091</td>\n",
       "      <td>0.0</td>\n",
       "      <td>161.5</td>\n",
       "      <td>65.3</td>\n",
       "      <td>25.0362</td>\n",
       "      <td>51.0</td>\n",
       "      <td>22.574433</td>\n",
       "      <td>30.432563</td>\n",
       "      <td>...</td>\n",
       "      <td>10.001994</td>\n",
       "      <td>93.043305</td>\n",
       "      <td>131.343366</td>\n",
       "      <td>12.557424</td>\n",
       "      <td>30.240635</td>\n",
       "      <td>36.668847</td>\n",
       "      <td>1.079479</td>\n",
       "      <td>3.817204</td>\n",
       "      <td>5.370990</td>\n",
       "      <td>61.729635</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30319</th>\n",
       "      <td>04448_82</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20190117162724.1.1...</td>\n",
       "      <td>6023985</td>\n",
       "      <td>0.0</td>\n",
       "      <td>164.6</td>\n",
       "      <td>71.4</td>\n",
       "      <td>26.3535</td>\n",
       "      <td>55.0</td>\n",
       "      <td>23.103796</td>\n",
       "      <td>30.177756</td>\n",
       "      <td>...</td>\n",
       "      <td>12.205770</td>\n",
       "      <td>91.554095</td>\n",
       "      <td>129.150241</td>\n",
       "      <td>13.763440</td>\n",
       "      <td>29.604453</td>\n",
       "      <td>36.761220</td>\n",
       "      <td>0.970704</td>\n",
       "      <td>3.598762</td>\n",
       "      <td>5.443171</td>\n",
       "      <td>62.802667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30320</th>\n",
       "      <td>12964_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20170418184226.1.1...</td>\n",
       "      <td>6024029</td>\n",
       "      <td>1.0</td>\n",
       "      <td>179.0</td>\n",
       "      <td>80.4</td>\n",
       "      <td>25.0928</td>\n",
       "      <td>41.0</td>\n",
       "      <td>24.323043</td>\n",
       "      <td>29.804727</td>\n",
       "      <td>...</td>\n",
       "      <td>12.728717</td>\n",
       "      <td>85.308035</td>\n",
       "      <td>124.424809</td>\n",
       "      <td>13.271879</td>\n",
       "      <td>30.125650</td>\n",
       "      <td>39.486231</td>\n",
       "      <td>0.975894</td>\n",
       "      <td>4.298543</td>\n",
       "      <td>6.171569</td>\n",
       "      <td>59.544963</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30321</th>\n",
       "      <td>02193_82</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20180811105258.1.1...</td>\n",
       "      <td>6024037</td>\n",
       "      <td>1.0</td>\n",
       "      <td>181.0</td>\n",
       "      <td>85.0</td>\n",
       "      <td>25.9455</td>\n",
       "      <td>53.0</td>\n",
       "      <td>24.695284</td>\n",
       "      <td>31.178125</td>\n",
       "      <td>...</td>\n",
       "      <td>12.533393</td>\n",
       "      <td>95.014076</td>\n",
       "      <td>130.924304</td>\n",
       "      <td>14.358949</td>\n",
       "      <td>31.715789</td>\n",
       "      <td>42.005320</td>\n",
       "      <td>1.036042</td>\n",
       "      <td>4.439059</td>\n",
       "      <td>6.281782</td>\n",
       "      <td>58.682502</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30322</th>\n",
       "      <td>04905_82</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20190219154149.1.1...</td>\n",
       "      <td>6024257</td>\n",
       "      <td>1.0</td>\n",
       "      <td>167.0</td>\n",
       "      <td>73.0</td>\n",
       "      <td>26.1752</td>\n",
       "      <td>60.0</td>\n",
       "      <td>22.849056</td>\n",
       "      <td>30.738358</td>\n",
       "      <td>...</td>\n",
       "      <td>9.722564</td>\n",
       "      <td>102.209266</td>\n",
       "      <td>101.631995</td>\n",
       "      <td>13.457582</td>\n",
       "      <td>31.007227</td>\n",
       "      <td>38.590868</td>\n",
       "      <td>0.948712</td>\n",
       "      <td>4.628839</td>\n",
       "      <td>6.287657</td>\n",
       "      <td>54.167545</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30323</th>\n",
       "      <td>02700_82</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20180912183226.1.1...</td>\n",
       "      <td>6024532</td>\n",
       "      <td>1.0</td>\n",
       "      <td>173.0</td>\n",
       "      <td>74.2</td>\n",
       "      <td>24.7920</td>\n",
       "      <td>57.0</td>\n",
       "      <td>24.004784</td>\n",
       "      <td>32.248290</td>\n",
       "      <td>...</td>\n",
       "      <td>10.533469</td>\n",
       "      <td>105.193136</td>\n",
       "      <td>105.144526</td>\n",
       "      <td>14.392064</td>\n",
       "      <td>30.197318</td>\n",
       "      <td>39.487130</td>\n",
       "      <td>0.942854</td>\n",
       "      <td>4.722146</td>\n",
       "      <td>5.989986</td>\n",
       "      <td>55.439957</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>30324 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       image_id                                          file_name      eid  \\\n",
       "0      08760_93  1.2.840.113619.2.110.210419.20160630134650.1.1...  1000276   \n",
       "1      12347_93  1.2.840.113619.2.110.210419.20170305174208.1.1...  1000742   \n",
       "2      14366_82  1.2.840.113619.2.110.212038.20190307144208.1.1...  1000951   \n",
       "3      06366_93  1.2.840.113619.2.110.210419.20151211144320.1.1...  1001025   \n",
       "4      14857_93  1.2.840.113619.2.110.210419.20170902122319.1.1...  1001091   \n",
       "...         ...                                                ...      ...   \n",
       "30319  04448_82  1.2.840.113619.2.110.210419.20190117162724.1.1...  6023985   \n",
       "30320  12964_93  1.2.840.113619.2.110.210419.20170418184226.1.1...  6024029   \n",
       "30321  02193_82  1.2.840.113619.2.110.210419.20180811105258.1.1...  6024037   \n",
       "30322  04905_82  1.2.840.113619.2.110.210419.20190219154149.1.1...  6024257   \n",
       "30323  02700_82  1.2.840.113619.2.110.210419.20180912183226.1.1...  6024532   \n",
       "\n",
       "       sex  standing_height  weight      bmi   age  pelvic_height  \\\n",
       "0      0.0            160.0    61.5  24.0234  60.0      22.979831   \n",
       "1      0.0            160.0    77.4  30.2344  60.0      22.388605   \n",
       "2      1.0            166.0    60.8  22.0642  66.0      22.297293   \n",
       "3      0.0            167.0    68.9  24.7051  49.0      22.512247   \n",
       "4      0.0            161.5    65.3  25.0362  51.0      22.574433   \n",
       "...    ...              ...     ...      ...   ...            ...   \n",
       "30319  0.0            164.6    71.4  26.3535  55.0      23.103796   \n",
       "30320  1.0            179.0    80.4  25.0928  41.0      24.323043   \n",
       "30321  1.0            181.0    85.0  25.9455  53.0      24.695284   \n",
       "30322  1.0            167.0    73.0  26.1752  60.0      22.849056   \n",
       "30323  1.0            173.0    74.2  24.7920  57.0      24.004784   \n",
       "\n",
       "       pelvic_width  ...  oblique_pelvic_inlet_length  iliac_flare_angle  \\\n",
       "0         31.953718  ...                    11.390012         101.719769   \n",
       "1         29.873284  ...                    12.005250          90.615136   \n",
       "2         30.113663  ...                     9.883806          97.533181   \n",
       "3         33.454000  ...                    13.418752         102.935020   \n",
       "4         30.432563  ...                    10.001994          93.043305   \n",
       "...             ...  ...                          ...                ...   \n",
       "30319     30.177756  ...                    12.205770          91.554095   \n",
       "30320     29.804727  ...                    12.728717          85.308035   \n",
       "30321     31.178125  ...                    12.533393          95.014076   \n",
       "30322     30.738358  ...                     9.722564         102.209266   \n",
       "30323     32.248290  ...                    10.533469         105.193136   \n",
       "\n",
       "       subpubic_angle  ear_left2ear_right  trochanter_left2trochanter_right  \\\n",
       "0          142.678044           13.641873                         29.618177   \n",
       "1          141.231641           13.535399                         29.360838   \n",
       "2          116.365190           14.112233                         30.091232   \n",
       "3          159.703035           12.863914                         30.934761   \n",
       "4          131.343366           12.557424                         30.240635   \n",
       "...               ...                 ...                               ...   \n",
       "30319      129.150241           13.763440                         29.604453   \n",
       "30320      124.424809           13.271879                         30.125650   \n",
       "30321      130.924304           14.358949                         31.715789   \n",
       "30322      101.631995           13.457582                         31.007227   \n",
       "30323      105.144526           14.392064                         30.197318   \n",
       "\n",
       "       shoulder_width  arm_devide_torso  iliac_isthmus_breadth  \\\n",
       "0           35.288068          0.957793               3.648129   \n",
       "1           36.799830          0.892762               3.725955   \n",
       "2           36.696399          1.067168               4.164529   \n",
       "3           35.136934          0.895883               4.080365   \n",
       "4           36.668847          1.079479               3.817204   \n",
       "...               ...               ...                    ...   \n",
       "30319       36.761220          0.970704               3.598762   \n",
       "30320       39.486231          0.975894               4.298543   \n",
       "30321       42.005320          1.036042               4.439059   \n",
       "30322       38.590868          0.948712               4.628839   \n",
       "30323       39.487130          0.942854               4.722146   \n",
       "\n",
       "       acetabular_diameter  acetabular_inclination  \n",
       "0                 5.369424               63.070462  \n",
       "1                 5.469268               58.032066  \n",
       "2                 6.141783               60.747735  \n",
       "3                 5.154912               61.323156  \n",
       "4                 5.370990               61.729635  \n",
       "...                    ...                     ...  \n",
       "30319             5.443171               62.802667  \n",
       "30320             6.171569               59.544963  \n",
       "30321             6.281782               58.682502  \n",
       "30322             6.287657               54.167545  \n",
       "30323             5.989986               55.439957  \n",
       "\n",
       "[30324 rows x 22 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hip_pheno = pd.read_csv('key_results/hip_select_pheno_cm_eid_flt_z_flt.csv'); hip_pheno"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/alexxu/opt/miniconda3/envs/ml/lib/python3.8/site-packages/pandas/core/frame.py:3607: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  self._set_item(key, value)\n"
     ]
    }
   ],
   "source": [
    "# use ratio\n",
    "hip_pheno['shoulder_width_ratio'] = hip_pheno['shoulder_width'] / hip_pheno['standing_height']\n",
    "hip_pheno['trochanter_left2trochanter_right_ratio'] = hip_pheno['trochanter_left2trochanter_right'] / hip_pheno['standing_height']\n",
    "hip_pheno = hip_pheno[['eid', 'shoulder_width_ratio', 'trochanter_left2trochanter_right_ratio']]\n",
    "\n",
    "ids = hip_pheno['eid'].tolist()\n",
    "\n",
    "hip_pheno['FID'] = ids\n",
    "hip_pheno['IID'] = ids\n",
    "\n",
    "hip_pheno = hip_pheno[['FID', \"IID\", \"shoulder_width_ratio\", \"trochanter_left2trochanter_right_ratio\"]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>shoulder_width_ratio</th>\n",
       "      <th>trochanter_left2trochanter_right_ratio</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1000276</td>\n",
       "      <td>1000276</td>\n",
       "      <td>0.220550</td>\n",
       "      <td>0.185114</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1000742</td>\n",
       "      <td>1000742</td>\n",
       "      <td>0.229999</td>\n",
       "      <td>0.183505</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1000951</td>\n",
       "      <td>1000951</td>\n",
       "      <td>0.221063</td>\n",
       "      <td>0.181272</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1001025</td>\n",
       "      <td>1001025</td>\n",
       "      <td>0.210401</td>\n",
       "      <td>0.185238</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1001091</td>\n",
       "      <td>1001091</td>\n",
       "      <td>0.227052</td>\n",
       "      <td>0.187249</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30319</th>\n",
       "      <td>6023985</td>\n",
       "      <td>6023985</td>\n",
       "      <td>0.223337</td>\n",
       "      <td>0.179857</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30320</th>\n",
       "      <td>6024029</td>\n",
       "      <td>6024029</td>\n",
       "      <td>0.220593</td>\n",
       "      <td>0.168300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30321</th>\n",
       "      <td>6024037</td>\n",
       "      <td>6024037</td>\n",
       "      <td>0.232074</td>\n",
       "      <td>0.175225</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30322</th>\n",
       "      <td>6024257</td>\n",
       "      <td>6024257</td>\n",
       "      <td>0.231083</td>\n",
       "      <td>0.185672</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30323</th>\n",
       "      <td>6024532</td>\n",
       "      <td>6024532</td>\n",
       "      <td>0.228249</td>\n",
       "      <td>0.174551</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>30324 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           FID      IID  shoulder_width_ratio  \\\n",
       "0      1000276  1000276              0.220550   \n",
       "1      1000742  1000742              0.229999   \n",
       "2      1000951  1000951              0.221063   \n",
       "3      1001025  1001025              0.210401   \n",
       "4      1001091  1001091              0.227052   \n",
       "...        ...      ...                   ...   \n",
       "30319  6023985  6023985              0.223337   \n",
       "30320  6024029  6024029              0.220593   \n",
       "30321  6024037  6024037              0.232074   \n",
       "30322  6024257  6024257              0.231083   \n",
       "30323  6024532  6024532              0.228249   \n",
       "\n",
       "       trochanter_left2trochanter_right_ratio  \n",
       "0                                    0.185114  \n",
       "1                                    0.183505  \n",
       "2                                    0.181272  \n",
       "3                                    0.185238  \n",
       "4                                    0.187249  \n",
       "...                                       ...  \n",
       "30319                                0.179857  \n",
       "30320                                0.168300  \n",
       "30321                                0.175225  \n",
       "30322                                0.185672  \n",
       "30323                                0.174551  \n",
       "\n",
       "[30324 rows x 4 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hip_pheno"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "hip_pheno.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_pheno_ratio_check.txt', index=False, sep = \" \")\n",
    "hip_pheno.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_pheno_ratio_check_no_header.txt', index=False, header=False, sep = \" \")"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Get h2g"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Get h2g from GCTA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "# both\n",
    "vg2vp = pd.DataFrame(columns=['pheno', 'var', 'se'])\n",
    "path = \"UKB_Imaging_Genetics/GCTA_GRM_OUTPUT/nonimputed_hip_both_hapmap3_20230918\"\n",
    "\n",
    "for file in os.listdir(path):\n",
    "    if file.endswith(\".hsq\"):\n",
    "        # get phenotype name\n",
    "        pheno = file.split('.')[0]\n",
    "        if pheno == '':\n",
    "            continue\n",
    "        with open(os.path.join(path, file), \"r\") as f:\n",
    "            for line in f:\n",
    "                # get V(G)/Vp\n",
    "                if line.startswith(\"V(G)/Vp\"):\n",
    "                    var = line.split()[1]\n",
    "                    se = line.split()[2]\n",
    "                    vg2vp = vg2vp.append({'pheno': pheno, 'var': var, 'se': se}, ignore_index=True)\n",
    "\n",
    "vg2vp = vg2vp.sort_values(by=['var'], ascending=False)\n",
    "vg2vp.reset_index(drop=True, inplace=True)\n",
    "vg2vp[['var', 'se']] = vg2vp[['var', 'se']].apply(pd.to_numeric)\n",
    "vg2vp.to_csv(\"key_results/h2g/h2g_both_20230918.csv\", index=False)\n",
    "\n",
    "# male\n",
    "vg2vp_male = pd.DataFrame(columns=['pheno', 'var', 'se'])\n",
    "path = \"UKB_Imaging_Genetics/GCTA_GRM_OUTPUT/nonimputed_hip_male_hapmap3_20230918\"\n",
    "\n",
    "for file in os.listdir(path):\n",
    "    if file.endswith(\".hsq\"):\n",
    "        # get phenotype name\n",
    "        pheno = file.split('.')[0]\n",
    "        if pheno == '':\n",
    "            continue\n",
    "        with open(os.path.join(path, file), \"r\") as f:\n",
    "            for line in f:\n",
    "                # get V(G)/Vp\n",
    "                if line.startswith(\"V(G)/Vp\"):\n",
    "                    var = line.split()[1]\n",
    "                    se = line.split()[2]\n",
    "                    vg2vp_male = vg2vp_male.append({'pheno': pheno, 'var': var, 'se': se}, ignore_index=True)\n",
    "\n",
    "vg2vp_male = vg2vp_male.sort_values(by=['var'], ascending=False)\n",
    "vg2vp_male.reset_index(drop=True, inplace=True)\n",
    "vg2vp_male[['var', 'se']] = vg2vp_male[['var', 'se']].apply(pd.to_numeric)\n",
    "vg2vp_male.to_csv(\"key_results/h2g/h2g_male_20230918.csv\", index=False)\n",
    "\n",
    "# female\n",
    "vg2vp_female = pd.DataFrame(columns=['pheno', 'var', 'se'])\n",
    "path = \"UKB_Imaging_Genetics/GCTA_GRM_OUTPUT/nonimputed_hip_female_hapmap3_20230918\"\n",
    "\n",
    "for file in os.listdir(path):\n",
    "    if file.endswith(\".hsq\"):\n",
    "        # get phenotype name\n",
    "        pheno = file.split('.')[0]\n",
    "        if pheno == '':\n",
    "            continue\n",
    "        with open(os.path.join(path, file), \"r\") as f:\n",
    "            for line in f:\n",
    "                # get V(G)/Vp\n",
    "                if line.startswith(\"V(G)/Vp\"):\n",
    "                    var = line.split()[1]\n",
    "                    se = line.split()[2]\n",
    "                    vg2vp_female = vg2vp_female.append({'pheno': pheno, 'var': var, 'se': se}, ignore_index=True)\n",
    "\n",
    "vg2vp_female = vg2vp_female.sort_values(by=['var'], ascending=False)\n",
    "vg2vp_female.reset_index(drop=True, inplace=True)\n",
    "vg2vp_female[['var', 'se']] = vg2vp_female[['var', 'se']].apply(pd.to_numeric)\n",
    "vg2vp_female.to_csv(\"key_results/h2g/h2g_female_20230918.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "# concatenate\n",
    "vg2vp['sex'] = \"both\"\n",
    "vg2vp_male['sex'] = \"male\"\n",
    "vg2vp_female['sex'] = \"female\"\n",
    "\n",
    "df = pd.concat([vg2vp, vg2vp_male, vg2vp_female])\n",
    "df.to_csv(\"key_results/h2g/h2g_all_20230918.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pheno</th>\n",
       "      <th>var</th>\n",
       "      <th>se</th>\n",
       "      <th>sex</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>pelvic_height</td>\n",
       "      <td>0.400520</td>\n",
       "      <td>0.012617</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>trochanter_left2trochanter_right</td>\n",
       "      <td>0.389605</td>\n",
       "      <td>0.012701</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>pelvic_inlet_width</td>\n",
       "      <td>0.387759</td>\n",
       "      <td>0.012670</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>pelvic_width</td>\n",
       "      <td>0.381625</td>\n",
       "      <td>0.012746</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>iliac_flare_angle</td>\n",
       "      <td>0.361388</td>\n",
       "      <td>0.012720</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>subpubic_angle</td>\n",
       "      <td>0.304433</td>\n",
       "      <td>0.023329</td>\n",
       "      <td>female</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>head_divide_oblique_inlet_length</td>\n",
       "      <td>0.300734</td>\n",
       "      <td>0.023590</td>\n",
       "      <td>female</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>head_area_divide_pelvic_inlet_area</td>\n",
       "      <td>0.300339</td>\n",
       "      <td>0.023653</td>\n",
       "      <td>female</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>oblique_pelvic_inlet_length</td>\n",
       "      <td>0.296099</td>\n",
       "      <td>0.023594</td>\n",
       "      <td>female</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>acetabular_inclination</td>\n",
       "      <td>0.283247</td>\n",
       "      <td>0.023464</td>\n",
       "      <td>female</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>63 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 pheno       var        se     sex\n",
       "0                        pelvic_height  0.400520  0.012617    both\n",
       "1     trochanter_left2trochanter_right  0.389605  0.012701    both\n",
       "2                   pelvic_inlet_width  0.387759  0.012670    both\n",
       "3                         pelvic_width  0.381625  0.012746    both\n",
       "4                    iliac_flare_angle  0.361388  0.012720    both\n",
       "..                                 ...       ...       ...     ...\n",
       "16                      subpubic_angle  0.304433  0.023329  female\n",
       "17    head_divide_oblique_inlet_length  0.300734  0.023590  female\n",
       "18  head_area_divide_pelvic_inlet_area  0.300339  0.023653  female\n",
       "19         oblique_pelvic_inlet_length  0.296099  0.023594  female\n",
       "20              acetabular_inclination  0.283247  0.023464  female\n",
       "\n",
       "[63 rows x 4 columns]"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Get h2g from LDSC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [],
   "source": [
    "path = \"/Users/alexxu/Library/CloudStorage/Box-Box/Narasimhan_lab/hip_shape/UKB_Imaging_Genetics/LDSC_OUTPUT\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [],
   "source": [
    "group = ['both', 'female', 'male']\n",
    "\n",
    "h2g_df = pd.DataFrame()\n",
    "\n",
    "for sex in group:\n",
    "    folder_name = f\"h2g_hip_{group}_20230909\"\n",
    "    for file_name in os.listdir(os.path.join(path, folder_name)):\n",
    "        if file_name.endswith(\".log\"):\n",
    "            pheno = file_name.split('.')[0]\n",
    "            with open(os.path.join(path, folder_name, file_name), \"r\") as f:\n",
    "                for line in f:\n",
    "                    if line.startswith(\"Total Observed scale h2\"):\n",
    "                        h2g = float(line.split(\":\")[-1].strip().split()[0])\n",
    "                        se = float(line.split(\":\")[-1].strip().split()[1].replace(\"(\" , \"\").replace(\")\", \"\"))\n",
    "                        h2g_df = h2g_df.append({'pheno': pheno, 'h2g': h2g, 'se': se, 'sex': sex}, ignore_index=True)\n",
    "                        break\n",
    "h2g_df.to_csv(\"key_results/h2g/h2g_ldsc_all_20230909.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>phenotype</th>\n",
       "      <th>h2g</th>\n",
       "      <th>se</th>\n",
       "      <th>sex</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>pelvic_width</td>\n",
       "      <td>0.4123</td>\n",
       "      <td>0.0444</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>shoulder_divide_inlet_width</td>\n",
       "      <td>0.3711</td>\n",
       "      <td>0.0400</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>shoulder_area_divide_pelvic_inlet_area</td>\n",
       "      <td>0.3092</td>\n",
       "      <td>0.0419</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>acetabular_inclination</td>\n",
       "      <td>0.3043</td>\n",
       "      <td>0.0392</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>arm_devide_torso</td>\n",
       "      <td>0.3850</td>\n",
       "      <td>0.0410</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>head_divide_inlet_width</td>\n",
       "      <td>0.3596</td>\n",
       "      <td>0.0404</td>\n",
       "      <td>male</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>subpubic_angle</td>\n",
       "      <td>0.3474</td>\n",
       "      <td>0.0394</td>\n",
       "      <td>male</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>trochanter_left2trochanter_right</td>\n",
       "      <td>0.3424</td>\n",
       "      <td>0.0458</td>\n",
       "      <td>male</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>shoulder_width</td>\n",
       "      <td>0.3013</td>\n",
       "      <td>0.0389</td>\n",
       "      <td>male</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>pelvic_height</td>\n",
       "      <td>0.5152</td>\n",
       "      <td>0.0520</td>\n",
       "      <td>male</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>63 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 phenotype     h2g      se   sex\n",
       "0                             pelvic_width  0.4123  0.0444  both\n",
       "1              shoulder_divide_inlet_width  0.3711  0.0400  both\n",
       "2   shoulder_area_divide_pelvic_inlet_area  0.3092  0.0419  both\n",
       "3                   acetabular_inclination  0.3043  0.0392  both\n",
       "4                         arm_devide_torso  0.3850  0.0410  both\n",
       "..                                     ...     ...     ...   ...\n",
       "58                 head_divide_inlet_width  0.3596  0.0404  male\n",
       "59                          subpubic_angle  0.3474  0.0394  male\n",
       "60        trochanter_left2trochanter_right  0.3424  0.0458  male\n",
       "61                          shoulder_width  0.3013  0.0389  male\n",
       "62                           pelvic_height  0.5152  0.0520  male\n",
       "\n",
       "[63 rows x 4 columns]"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "h2g_df"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## GWAS"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Try with PLINK"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>p_age</th>\n",
       "      <th>p_weight</th>\n",
       "      <th>p_height</th>\n",
       "      <th>iliac_crest_left2iliac_crest_right</th>\n",
       "      <th>iliac_crest_left2iliac_spine_left</th>\n",
       "      <th>iliac_crest_left2iliac_spine_right</th>\n",
       "      <th>iliac_crest_left2iliopubic_eminence_left</th>\n",
       "      <th>iliac_crest_left2iliopubic_eminence_right</th>\n",
       "      <th>...</th>\n",
       "      <th>hip_height</th>\n",
       "      <th>sacrum_ratio_w2h</th>\n",
       "      <th>hip_ratio_w2h</th>\n",
       "      <th>sacrum2pubic_tubercle_devide_ear_left2ear_right</th>\n",
       "      <th>sacrum_left2sacrum_right_devide_ear_left2ear_right</th>\n",
       "      <th>sciatic_notch_left2sciatic_notch_right_divide_ear_left2ear_right</th>\n",
       "      <th>pubic_arch_angle</th>\n",
       "      <th>acetabular_inclination_left</th>\n",
       "      <th>acetabular_inclination_right</th>\n",
       "      <th>acetabular_inclination_diff</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1000276</td>\n",
       "      <td>1000276</td>\n",
       "      <td>67.0</td>\n",
       "      <td>60.0</td>\n",
       "      <td>159.0</td>\n",
       "      <td>0.119177</td>\n",
       "      <td>0.057544</td>\n",
       "      <td>0.161311</td>\n",
       "      <td>0.092231</td>\n",
       "      <td>0.161052</td>\n",
       "      <td>...</td>\n",
       "      <td>0.144527</td>\n",
       "      <td>1.414572</td>\n",
       "      <td>1.390511</td>\n",
       "      <td>0.834930</td>\n",
       "      <td>0.828164</td>\n",
       "      <td>1.181069</td>\n",
       "      <td>142.678044</td>\n",
       "      <td>53.033866</td>\n",
       "      <td>57.859459</td>\n",
       "      <td>-4.825592</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1000742</td>\n",
       "      <td>1000742</td>\n",
       "      <td>67.0</td>\n",
       "      <td>77.0</td>\n",
       "      <td>161.0</td>\n",
       "      <td>0.123848</td>\n",
       "      <td>0.038570</td>\n",
       "      <td>0.155576</td>\n",
       "      <td>0.084050</td>\n",
       "      <td>0.159198</td>\n",
       "      <td>...</td>\n",
       "      <td>0.139060</td>\n",
       "      <td>1.251339</td>\n",
       "      <td>1.334308</td>\n",
       "      <td>0.886952</td>\n",
       "      <td>0.765835</td>\n",
       "      <td>1.109878</td>\n",
       "      <td>141.231641</td>\n",
       "      <td>54.070014</td>\n",
       "      <td>54.815992</td>\n",
       "      <td>-0.745978</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1000951</td>\n",
       "      <td>1000951</td>\n",
       "      <td>78.0</td>\n",
       "      <td>60.0</td>\n",
       "      <td>170.0</td>\n",
       "      <td>0.113652</td>\n",
       "      <td>0.036685</td>\n",
       "      <td>0.147139</td>\n",
       "      <td>0.075637</td>\n",
       "      <td>0.147038</td>\n",
       "      <td>...</td>\n",
       "      <td>0.131161</td>\n",
       "      <td>1.410102</td>\n",
       "      <td>1.350552</td>\n",
       "      <td>0.700372</td>\n",
       "      <td>0.739401</td>\n",
       "      <td>0.987595</td>\n",
       "      <td>116.365190</td>\n",
       "      <td>51.546225</td>\n",
       "      <td>55.132446</td>\n",
       "      <td>-3.586221</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1001025</td>\n",
       "      <td>1001025</td>\n",
       "      <td>55.0</td>\n",
       "      <td>69.0</td>\n",
       "      <td>167.0</td>\n",
       "      <td>0.115559</td>\n",
       "      <td>0.055194</td>\n",
       "      <td>0.162427</td>\n",
       "      <td>0.083520</td>\n",
       "      <td>0.157159</td>\n",
       "      <td>...</td>\n",
       "      <td>0.134804</td>\n",
       "      <td>1.215281</td>\n",
       "      <td>1.486036</td>\n",
       "      <td>1.043131</td>\n",
       "      <td>0.845409</td>\n",
       "      <td>1.267698</td>\n",
       "      <td>159.703035</td>\n",
       "      <td>56.689420</td>\n",
       "      <td>56.406852</td>\n",
       "      <td>0.282569</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1001091</td>\n",
       "      <td>1001091</td>\n",
       "      <td>60.0</td>\n",
       "      <td>65.0</td>\n",
       "      <td>162.0</td>\n",
       "      <td>0.119257</td>\n",
       "      <td>0.037780</td>\n",
       "      <td>0.155524</td>\n",
       "      <td>0.079466</td>\n",
       "      <td>0.155074</td>\n",
       "      <td>...</td>\n",
       "      <td>0.139348</td>\n",
       "      <td>1.445791</td>\n",
       "      <td>1.348099</td>\n",
       "      <td>0.796500</td>\n",
       "      <td>0.836422</td>\n",
       "      <td>1.151573</td>\n",
       "      <td>131.343366</td>\n",
       "      <td>59.651556</td>\n",
       "      <td>61.152404</td>\n",
       "      <td>-1.500849</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30992</th>\n",
       "      <td>6024029</td>\n",
       "      <td>6024029</td>\n",
       "      <td>50.0</td>\n",
       "      <td>79.0</td>\n",
       "      <td>180.0</td>\n",
       "      <td>0.115447</td>\n",
       "      <td>0.030139</td>\n",
       "      <td>0.141083</td>\n",
       "      <td>0.076052</td>\n",
       "      <td>0.142842</td>\n",
       "      <td>...</td>\n",
       "      <td>0.135128</td>\n",
       "      <td>1.090737</td>\n",
       "      <td>1.225370</td>\n",
       "      <td>0.959074</td>\n",
       "      <td>0.752686</td>\n",
       "      <td>1.046098</td>\n",
       "      <td>124.424809</td>\n",
       "      <td>57.535146</td>\n",
       "      <td>57.062575</td>\n",
       "      <td>0.472571</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30993</th>\n",
       "      <td>6024037</td>\n",
       "      <td>6024037</td>\n",
       "      <td>61.0</td>\n",
       "      <td>84.0</td>\n",
       "      <td>181.0</td>\n",
       "      <td>0.109468</td>\n",
       "      <td>0.046472</td>\n",
       "      <td>0.139892</td>\n",
       "      <td>0.080300</td>\n",
       "      <td>0.143638</td>\n",
       "      <td>...</td>\n",
       "      <td>0.136438</td>\n",
       "      <td>1.167638</td>\n",
       "      <td>1.262513</td>\n",
       "      <td>0.872863</td>\n",
       "      <td>0.770157</td>\n",
       "      <td>1.019188</td>\n",
       "      <td>130.924304</td>\n",
       "      <td>53.310400</td>\n",
       "      <td>57.006480</td>\n",
       "      <td>-3.696080</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30994</th>\n",
       "      <td>6024119</td>\n",
       "      <td>6024119</td>\n",
       "      <td>68.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>165.0</td>\n",
       "      <td>0.117222</td>\n",
       "      <td>0.031103</td>\n",
       "      <td>0.146952</td>\n",
       "      <td>0.078930</td>\n",
       "      <td>0.149576</td>\n",
       "      <td>...</td>\n",
       "      <td>0.132217</td>\n",
       "      <td>1.317724</td>\n",
       "      <td>1.301696</td>\n",
       "      <td>0.762527</td>\n",
       "      <td>0.765581</td>\n",
       "      <td>1.004801</td>\n",
       "      <td>115.933030</td>\n",
       "      <td>52.860920</td>\n",
       "      <td>50.817846</td>\n",
       "      <td>2.043073</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30995</th>\n",
       "      <td>6024257</td>\n",
       "      <td>6024257</td>\n",
       "      <td>71.0</td>\n",
       "      <td>74.0</td>\n",
       "      <td>169.0</td>\n",
       "      <td>0.121064</td>\n",
       "      <td>0.039582</td>\n",
       "      <td>0.151179</td>\n",
       "      <td>0.079783</td>\n",
       "      <td>0.153874</td>\n",
       "      <td>...</td>\n",
       "      <td>0.135202</td>\n",
       "      <td>1.447859</td>\n",
       "      <td>1.345279</td>\n",
       "      <td>0.722460</td>\n",
       "      <td>0.804753</td>\n",
       "      <td>1.046020</td>\n",
       "      <td>101.631995</td>\n",
       "      <td>53.945919</td>\n",
       "      <td>48.201155</td>\n",
       "      <td>5.744764</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30996</th>\n",
       "      <td>6024532</td>\n",
       "      <td>6024532</td>\n",
       "      <td>67.0</td>\n",
       "      <td>75.0</td>\n",
       "      <td>175.0</td>\n",
       "      <td>0.112109</td>\n",
       "      <td>0.043879</td>\n",
       "      <td>0.150154</td>\n",
       "      <td>0.082683</td>\n",
       "      <td>0.151725</td>\n",
       "      <td>...</td>\n",
       "      <td>0.137170</td>\n",
       "      <td>1.446266</td>\n",
       "      <td>1.343411</td>\n",
       "      <td>0.731894</td>\n",
       "      <td>0.771329</td>\n",
       "      <td>1.058514</td>\n",
       "      <td>105.144526</td>\n",
       "      <td>53.038293</td>\n",
       "      <td>53.927739</td>\n",
       "      <td>-0.889446</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>30997 rows × 269 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           FID      IID  p_age  p_weight  p_height  \\\n",
       "0      1000276  1000276   67.0      60.0     159.0   \n",
       "1      1000742  1000742   67.0      77.0     161.0   \n",
       "2      1000951  1000951   78.0      60.0     170.0   \n",
       "3      1001025  1001025   55.0      69.0     167.0   \n",
       "4      1001091  1001091   60.0      65.0     162.0   \n",
       "...        ...      ...    ...       ...       ...   \n",
       "30992  6024029  6024029   50.0      79.0     180.0   \n",
       "30993  6024037  6024037   61.0      84.0     181.0   \n",
       "30994  6024119  6024119   68.0      70.0     165.0   \n",
       "30995  6024257  6024257   71.0      74.0     169.0   \n",
       "30996  6024532  6024532   67.0      75.0     175.0   \n",
       "\n",
       "       iliac_crest_left2iliac_crest_right  iliac_crest_left2iliac_spine_left  \\\n",
       "0                                0.119177                           0.057544   \n",
       "1                                0.123848                           0.038570   \n",
       "2                                0.113652                           0.036685   \n",
       "3                                0.115559                           0.055194   \n",
       "4                                0.119257                           0.037780   \n",
       "...                                   ...                                ...   \n",
       "30992                            0.115447                           0.030139   \n",
       "30993                            0.109468                           0.046472   \n",
       "30994                            0.117222                           0.031103   \n",
       "30995                            0.121064                           0.039582   \n",
       "30996                            0.112109                           0.043879   \n",
       "\n",
       "       iliac_crest_left2iliac_spine_right  \\\n",
       "0                                0.161311   \n",
       "1                                0.155576   \n",
       "2                                0.147139   \n",
       "3                                0.162427   \n",
       "4                                0.155524   \n",
       "...                                   ...   \n",
       "30992                            0.141083   \n",
       "30993                            0.139892   \n",
       "30994                            0.146952   \n",
       "30995                            0.151179   \n",
       "30996                            0.150154   \n",
       "\n",
       "       iliac_crest_left2iliopubic_eminence_left  \\\n",
       "0                                      0.092231   \n",
       "1                                      0.084050   \n",
       "2                                      0.075637   \n",
       "3                                      0.083520   \n",
       "4                                      0.079466   \n",
       "...                                         ...   \n",
       "30992                                  0.076052   \n",
       "30993                                  0.080300   \n",
       "30994                                  0.078930   \n",
       "30995                                  0.079783   \n",
       "30996                                  0.082683   \n",
       "\n",
       "       iliac_crest_left2iliopubic_eminence_right  ...  hip_height  \\\n",
       "0                                       0.161052  ...    0.144527   \n",
       "1                                       0.159198  ...    0.139060   \n",
       "2                                       0.147038  ...    0.131161   \n",
       "3                                       0.157159  ...    0.134804   \n",
       "4                                       0.155074  ...    0.139348   \n",
       "...                                          ...  ...         ...   \n",
       "30992                                   0.142842  ...    0.135128   \n",
       "30993                                   0.143638  ...    0.136438   \n",
       "30994                                   0.149576  ...    0.132217   \n",
       "30995                                   0.153874  ...    0.135202   \n",
       "30996                                   0.151725  ...    0.137170   \n",
       "\n",
       "       sacrum_ratio_w2h  hip_ratio_w2h  \\\n",
       "0              1.414572       1.390511   \n",
       "1              1.251339       1.334308   \n",
       "2              1.410102       1.350552   \n",
       "3              1.215281       1.486036   \n",
       "4              1.445791       1.348099   \n",
       "...                 ...            ...   \n",
       "30992          1.090737       1.225370   \n",
       "30993          1.167638       1.262513   \n",
       "30994          1.317724       1.301696   \n",
       "30995          1.447859       1.345279   \n",
       "30996          1.446266       1.343411   \n",
       "\n",
       "       sacrum2pubic_tubercle_devide_ear_left2ear_right  \\\n",
       "0                                             0.834930   \n",
       "1                                             0.886952   \n",
       "2                                             0.700372   \n",
       "3                                             1.043131   \n",
       "4                                             0.796500   \n",
       "...                                                ...   \n",
       "30992                                         0.959074   \n",
       "30993                                         0.872863   \n",
       "30994                                         0.762527   \n",
       "30995                                         0.722460   \n",
       "30996                                         0.731894   \n",
       "\n",
       "       sacrum_left2sacrum_right_devide_ear_left2ear_right  \\\n",
       "0                                               0.828164    \n",
       "1                                               0.765835    \n",
       "2                                               0.739401    \n",
       "3                                               0.845409    \n",
       "4                                               0.836422    \n",
       "...                                                  ...    \n",
       "30992                                           0.752686    \n",
       "30993                                           0.770157    \n",
       "30994                                           0.765581    \n",
       "30995                                           0.804753    \n",
       "30996                                           0.771329    \n",
       "\n",
       "       sciatic_notch_left2sciatic_notch_right_divide_ear_left2ear_right  \\\n",
       "0                                               1.181069                  \n",
       "1                                               1.109878                  \n",
       "2                                               0.987595                  \n",
       "3                                               1.267698                  \n",
       "4                                               1.151573                  \n",
       "...                                                  ...                  \n",
       "30992                                           1.046098                  \n",
       "30993                                           1.019188                  \n",
       "30994                                           1.004801                  \n",
       "30995                                           1.046020                  \n",
       "30996                                           1.058514                  \n",
       "\n",
       "       pubic_arch_angle  acetabular_inclination_left  \\\n",
       "0            142.678044                    53.033866   \n",
       "1            141.231641                    54.070014   \n",
       "2            116.365190                    51.546225   \n",
       "3            159.703035                    56.689420   \n",
       "4            131.343366                    59.651556   \n",
       "...                 ...                          ...   \n",
       "30992        124.424809                    57.535146   \n",
       "30993        130.924304                    53.310400   \n",
       "30994        115.933030                    52.860920   \n",
       "30995        101.631995                    53.945919   \n",
       "30996        105.144526                    53.038293   \n",
       "\n",
       "       acetabular_inclination_right  acetabular_inclination_diff  \n",
       "0                         57.859459                    -4.825592  \n",
       "1                         54.815992                    -0.745978  \n",
       "2                         55.132446                    -3.586221  \n",
       "3                         56.406852                     0.282569  \n",
       "4                         61.152404                    -1.500849  \n",
       "...                             ...                          ...  \n",
       "30992                     57.062575                     0.472571  \n",
       "30993                     57.006480                    -3.696080  \n",
       "30994                     50.817846                     2.043073  \n",
       "30995                     48.201155                     5.744764  \n",
       "30996                     53.927739                    -0.889446  \n",
       "\n",
       "[30997 rows x 269 columns]"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hip_pheno_gwas = pd.read_csv('UKB_Imaging_Genetics/GCTA_INPUT_DATA/hip_pheno_gwas.txt', sep=\" \"); hip_pheno_gwas"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>hip_height</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1000276</td>\n",
       "      <td>1000276</td>\n",
       "      <td>0.144527</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1000742</td>\n",
       "      <td>1000742</td>\n",
       "      <td>0.139060</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1000951</td>\n",
       "      <td>1000951</td>\n",
       "      <td>0.131161</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1001025</td>\n",
       "      <td>1001025</td>\n",
       "      <td>0.134804</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1001091</td>\n",
       "      <td>1001091</td>\n",
       "      <td>0.139348</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31005</th>\n",
       "      <td>6024029</td>\n",
       "      <td>6024029</td>\n",
       "      <td>0.135128</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31006</th>\n",
       "      <td>6024037</td>\n",
       "      <td>6024037</td>\n",
       "      <td>0.136438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31007</th>\n",
       "      <td>6024119</td>\n",
       "      <td>6024119</td>\n",
       "      <td>0.132217</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31008</th>\n",
       "      <td>6024257</td>\n",
       "      <td>6024257</td>\n",
       "      <td>0.135202</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31009</th>\n",
       "      <td>6024532</td>\n",
       "      <td>6024532</td>\n",
       "      <td>0.137170</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>31010 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           FID      IID  hip_height\n",
       "0      1000276  1000276    0.144527\n",
       "1      1000742  1000742    0.139060\n",
       "2      1000951  1000951    0.131161\n",
       "3      1001025  1001025    0.134804\n",
       "4      1001091  1001091    0.139348\n",
       "...        ...      ...         ...\n",
       "31005  6024029  6024029    0.135128\n",
       "31006  6024037  6024037    0.136438\n",
       "31007  6024119  6024119    0.132217\n",
       "31008  6024257  6024257    0.135202\n",
       "31009  6024532  6024532    0.137170\n",
       "\n",
       "[31010 rows x 3 columns]"
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hip_height_gwas = hip_pheno_gwas[['FID', 'IID', 'hip_height']]; hip_height_gwas"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [],
   "source": [
    "hip_height_gwas.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_height_gwas.txt', index=False, sep = \" \")"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Using BOLT-LMM ~2:30min per trait"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>p_height</th>\n",
       "      <th>iliac_crest_left2iliac_crest_right</th>\n",
       "      <th>iliac_crest_left2iliac_spine_left</th>\n",
       "      <th>iliac_crest_left2iliac_spine_right</th>\n",
       "      <th>iliac_crest_left2iliopubic_eminence_left</th>\n",
       "      <th>iliac_crest_left2iliopubic_eminence_right</th>\n",
       "      <th>iliac_crest_left2inferior_pubic_ramus_left</th>\n",
       "      <th>iliac_crest_left2inferior_pubic_ramus_right</th>\n",
       "      <th>...</th>\n",
       "      <th>inferior_iliac_spine_left2acetabular_inferior_right</th>\n",
       "      <th>inferior_iliac_spine_right2acetabular_inferior_left</th>\n",
       "      <th>inferior_iliac_spine_right2acetabular_inferior_right</th>\n",
       "      <th>acetabular_inferior_left2acetabular_inferior_right</th>\n",
       "      <th>hip_height</th>\n",
       "      <th>sacrum_ratio_w/h</th>\n",
       "      <th>hip_ratio_w/h</th>\n",
       "      <th>pubic_arch_angle</th>\n",
       "      <th>acetabular_inclination_left</th>\n",
       "      <th>acetabular_inclination_right</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1000276</td>\n",
       "      <td>1000276</td>\n",
       "      <td>159.0</td>\n",
       "      <td>0.119177</td>\n",
       "      <td>0.057544</td>\n",
       "      <td>0.161311</td>\n",
       "      <td>0.092231</td>\n",
       "      <td>0.161052</td>\n",
       "      <td>0.145886</td>\n",
       "      <td>0.174953</td>\n",
       "      <td>...</td>\n",
       "      <td>0.138035</td>\n",
       "      <td>0.141101</td>\n",
       "      <td>0.046274</td>\n",
       "      <td>0.118927</td>\n",
       "      <td>0.144527</td>\n",
       "      <td>1.414572</td>\n",
       "      <td>1.390511</td>\n",
       "      <td>142.678044</td>\n",
       "      <td>53.033866</td>\n",
       "      <td>57.859459</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1000742</td>\n",
       "      <td>1000742</td>\n",
       "      <td>161.0</td>\n",
       "      <td>0.123848</td>\n",
       "      <td>0.038570</td>\n",
       "      <td>0.155576</td>\n",
       "      <td>0.084050</td>\n",
       "      <td>0.159198</td>\n",
       "      <td>0.142365</td>\n",
       "      <td>0.169182</td>\n",
       "      <td>...</td>\n",
       "      <td>0.132737</td>\n",
       "      <td>0.130718</td>\n",
       "      <td>0.045501</td>\n",
       "      <td>0.109055</td>\n",
       "      <td>0.139060</td>\n",
       "      <td>1.251339</td>\n",
       "      <td>1.334308</td>\n",
       "      <td>141.231641</td>\n",
       "      <td>54.070014</td>\n",
       "      <td>54.815992</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1000951</td>\n",
       "      <td>1000951</td>\n",
       "      <td>170.0</td>\n",
       "      <td>0.113652</td>\n",
       "      <td>0.036685</td>\n",
       "      <td>0.147139</td>\n",
       "      <td>0.075637</td>\n",
       "      <td>0.147038</td>\n",
       "      <td>0.135329</td>\n",
       "      <td>0.157257</td>\n",
       "      <td>...</td>\n",
       "      <td>0.124839</td>\n",
       "      <td>0.125665</td>\n",
       "      <td>0.048471</td>\n",
       "      <td>0.101909</td>\n",
       "      <td>0.131161</td>\n",
       "      <td>1.410102</td>\n",
       "      <td>1.350552</td>\n",
       "      <td>116.365190</td>\n",
       "      <td>51.546225</td>\n",
       "      <td>55.132446</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1001025</td>\n",
       "      <td>1001025</td>\n",
       "      <td>167.0</td>\n",
       "      <td>0.115559</td>\n",
       "      <td>0.055194</td>\n",
       "      <td>0.162427</td>\n",
       "      <td>0.083520</td>\n",
       "      <td>0.157159</td>\n",
       "      <td>0.136728</td>\n",
       "      <td>0.167379</td>\n",
       "      <td>...</td>\n",
       "      <td>0.137322</td>\n",
       "      <td>0.139501</td>\n",
       "      <td>0.042405</td>\n",
       "      <td>0.118486</td>\n",
       "      <td>0.134804</td>\n",
       "      <td>1.215281</td>\n",
       "      <td>1.486036</td>\n",
       "      <td>159.703035</td>\n",
       "      <td>56.689420</td>\n",
       "      <td>56.406852</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1001091</td>\n",
       "      <td>1001091</td>\n",
       "      <td>162.0</td>\n",
       "      <td>0.119257</td>\n",
       "      <td>0.037780</td>\n",
       "      <td>0.155524</td>\n",
       "      <td>0.079466</td>\n",
       "      <td>0.155074</td>\n",
       "      <td>0.142456</td>\n",
       "      <td>0.172080</td>\n",
       "      <td>...</td>\n",
       "      <td>0.131174</td>\n",
       "      <td>0.131803</td>\n",
       "      <td>0.043280</td>\n",
       "      <td>0.113473</td>\n",
       "      <td>0.139348</td>\n",
       "      <td>1.445791</td>\n",
       "      <td>1.348099</td>\n",
       "      <td>131.343366</td>\n",
       "      <td>59.651556</td>\n",
       "      <td>61.152404</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31005</th>\n",
       "      <td>6024029</td>\n",
       "      <td>6024029</td>\n",
       "      <td>180.0</td>\n",
       "      <td>0.115447</td>\n",
       "      <td>0.030139</td>\n",
       "      <td>0.141083</td>\n",
       "      <td>0.076052</td>\n",
       "      <td>0.142842</td>\n",
       "      <td>0.136692</td>\n",
       "      <td>0.157629</td>\n",
       "      <td>...</td>\n",
       "      <td>0.116595</td>\n",
       "      <td>0.121475</td>\n",
       "      <td>0.046520</td>\n",
       "      <td>0.096047</td>\n",
       "      <td>0.135128</td>\n",
       "      <td>1.090737</td>\n",
       "      <td>1.225370</td>\n",
       "      <td>124.424809</td>\n",
       "      <td>57.535146</td>\n",
       "      <td>57.062575</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31006</th>\n",
       "      <td>6024037</td>\n",
       "      <td>6024037</td>\n",
       "      <td>181.0</td>\n",
       "      <td>0.109468</td>\n",
       "      <td>0.046472</td>\n",
       "      <td>0.139892</td>\n",
       "      <td>0.080300</td>\n",
       "      <td>0.143638</td>\n",
       "      <td>0.139164</td>\n",
       "      <td>0.159407</td>\n",
       "      <td>...</td>\n",
       "      <td>0.122303</td>\n",
       "      <td>0.120067</td>\n",
       "      <td>0.045412</td>\n",
       "      <td>0.097170</td>\n",
       "      <td>0.136438</td>\n",
       "      <td>1.167638</td>\n",
       "      <td>1.262513</td>\n",
       "      <td>130.924304</td>\n",
       "      <td>53.310400</td>\n",
       "      <td>57.006480</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31007</th>\n",
       "      <td>6024119</td>\n",
       "      <td>6024119</td>\n",
       "      <td>165.0</td>\n",
       "      <td>0.117222</td>\n",
       "      <td>0.031103</td>\n",
       "      <td>0.146952</td>\n",
       "      <td>0.078930</td>\n",
       "      <td>0.149576</td>\n",
       "      <td>0.136405</td>\n",
       "      <td>0.156596</td>\n",
       "      <td>...</td>\n",
       "      <td>0.119512</td>\n",
       "      <td>0.126894</td>\n",
       "      <td>0.052668</td>\n",
       "      <td>0.093897</td>\n",
       "      <td>0.132217</td>\n",
       "      <td>1.317724</td>\n",
       "      <td>1.301696</td>\n",
       "      <td>115.933030</td>\n",
       "      <td>52.860920</td>\n",
       "      <td>50.817846</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31008</th>\n",
       "      <td>6024257</td>\n",
       "      <td>6024257</td>\n",
       "      <td>169.0</td>\n",
       "      <td>0.121064</td>\n",
       "      <td>0.039582</td>\n",
       "      <td>0.151179</td>\n",
       "      <td>0.079783</td>\n",
       "      <td>0.153874</td>\n",
       "      <td>0.139239</td>\n",
       "      <td>0.158895</td>\n",
       "      <td>...</td>\n",
       "      <td>0.125164</td>\n",
       "      <td>0.128997</td>\n",
       "      <td>0.049194</td>\n",
       "      <td>0.100022</td>\n",
       "      <td>0.135202</td>\n",
       "      <td>1.447859</td>\n",
       "      <td>1.345279</td>\n",
       "      <td>101.631995</td>\n",
       "      <td>53.945919</td>\n",
       "      <td>48.201155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31009</th>\n",
       "      <td>6024532</td>\n",
       "      <td>6024532</td>\n",
       "      <td>175.0</td>\n",
       "      <td>0.112109</td>\n",
       "      <td>0.043879</td>\n",
       "      <td>0.150154</td>\n",
       "      <td>0.082683</td>\n",
       "      <td>0.151725</td>\n",
       "      <td>0.140263</td>\n",
       "      <td>0.162785</td>\n",
       "      <td>...</td>\n",
       "      <td>0.127369</td>\n",
       "      <td>0.131003</td>\n",
       "      <td>0.045023</td>\n",
       "      <td>0.103967</td>\n",
       "      <td>0.137170</td>\n",
       "      <td>1.446266</td>\n",
       "      <td>1.343411</td>\n",
       "      <td>105.144526</td>\n",
       "      <td>53.038293</td>\n",
       "      <td>53.927739</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>31010 rows × 262 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           FID      IID  p_height  iliac_crest_left2iliac_crest_right  \\\n",
       "0      1000276  1000276     159.0                            0.119177   \n",
       "1      1000742  1000742     161.0                            0.123848   \n",
       "2      1000951  1000951     170.0                            0.113652   \n",
       "3      1001025  1001025     167.0                            0.115559   \n",
       "4      1001091  1001091     162.0                            0.119257   \n",
       "...        ...      ...       ...                                 ...   \n",
       "31005  6024029  6024029     180.0                            0.115447   \n",
       "31006  6024037  6024037     181.0                            0.109468   \n",
       "31007  6024119  6024119     165.0                            0.117222   \n",
       "31008  6024257  6024257     169.0                            0.121064   \n",
       "31009  6024532  6024532     175.0                            0.112109   \n",
       "\n",
       "       iliac_crest_left2iliac_spine_left  iliac_crest_left2iliac_spine_right  \\\n",
       "0                               0.057544                            0.161311   \n",
       "1                               0.038570                            0.155576   \n",
       "2                               0.036685                            0.147139   \n",
       "3                               0.055194                            0.162427   \n",
       "4                               0.037780                            0.155524   \n",
       "...                                  ...                                 ...   \n",
       "31005                           0.030139                            0.141083   \n",
       "31006                           0.046472                            0.139892   \n",
       "31007                           0.031103                            0.146952   \n",
       "31008                           0.039582                            0.151179   \n",
       "31009                           0.043879                            0.150154   \n",
       "\n",
       "       iliac_crest_left2iliopubic_eminence_left  \\\n",
       "0                                      0.092231   \n",
       "1                                      0.084050   \n",
       "2                                      0.075637   \n",
       "3                                      0.083520   \n",
       "4                                      0.079466   \n",
       "...                                         ...   \n",
       "31005                                  0.076052   \n",
       "31006                                  0.080300   \n",
       "31007                                  0.078930   \n",
       "31008                                  0.079783   \n",
       "31009                                  0.082683   \n",
       "\n",
       "       iliac_crest_left2iliopubic_eminence_right  \\\n",
       "0                                       0.161052   \n",
       "1                                       0.159198   \n",
       "2                                       0.147038   \n",
       "3                                       0.157159   \n",
       "4                                       0.155074   \n",
       "...                                          ...   \n",
       "31005                                   0.142842   \n",
       "31006                                   0.143638   \n",
       "31007                                   0.149576   \n",
       "31008                                   0.153874   \n",
       "31009                                   0.151725   \n",
       "\n",
       "       iliac_crest_left2inferior_pubic_ramus_left  \\\n",
       "0                                        0.145886   \n",
       "1                                        0.142365   \n",
       "2                                        0.135329   \n",
       "3                                        0.136728   \n",
       "4                                        0.142456   \n",
       "...                                           ...   \n",
       "31005                                    0.136692   \n",
       "31006                                    0.139164   \n",
       "31007                                    0.136405   \n",
       "31008                                    0.139239   \n",
       "31009                                    0.140263   \n",
       "\n",
       "       iliac_crest_left2inferior_pubic_ramus_right  ...  \\\n",
       "0                                         0.174953  ...   \n",
       "1                                         0.169182  ...   \n",
       "2                                         0.157257  ...   \n",
       "3                                         0.167379  ...   \n",
       "4                                         0.172080  ...   \n",
       "...                                            ...  ...   \n",
       "31005                                     0.157629  ...   \n",
       "31006                                     0.159407  ...   \n",
       "31007                                     0.156596  ...   \n",
       "31008                                     0.158895  ...   \n",
       "31009                                     0.162785  ...   \n",
       "\n",
       "       inferior_iliac_spine_left2acetabular_inferior_right  \\\n",
       "0                                               0.138035     \n",
       "1                                               0.132737     \n",
       "2                                               0.124839     \n",
       "3                                               0.137322     \n",
       "4                                               0.131174     \n",
       "...                                                  ...     \n",
       "31005                                           0.116595     \n",
       "31006                                           0.122303     \n",
       "31007                                           0.119512     \n",
       "31008                                           0.125164     \n",
       "31009                                           0.127369     \n",
       "\n",
       "       inferior_iliac_spine_right2acetabular_inferior_left  \\\n",
       "0                                               0.141101     \n",
       "1                                               0.130718     \n",
       "2                                               0.125665     \n",
       "3                                               0.139501     \n",
       "4                                               0.131803     \n",
       "...                                                  ...     \n",
       "31005                                           0.121475     \n",
       "31006                                           0.120067     \n",
       "31007                                           0.126894     \n",
       "31008                                           0.128997     \n",
       "31009                                           0.131003     \n",
       "\n",
       "       inferior_iliac_spine_right2acetabular_inferior_right  \\\n",
       "0                                               0.046274      \n",
       "1                                               0.045501      \n",
       "2                                               0.048471      \n",
       "3                                               0.042405      \n",
       "4                                               0.043280      \n",
       "...                                                  ...      \n",
       "31005                                           0.046520      \n",
       "31006                                           0.045412      \n",
       "31007                                           0.052668      \n",
       "31008                                           0.049194      \n",
       "31009                                           0.045023      \n",
       "\n",
       "       acetabular_inferior_left2acetabular_inferior_right  hip_height  \\\n",
       "0                                               0.118927     0.144527   \n",
       "1                                               0.109055     0.139060   \n",
       "2                                               0.101909     0.131161   \n",
       "3                                               0.118486     0.134804   \n",
       "4                                               0.113473     0.139348   \n",
       "...                                                  ...          ...   \n",
       "31005                                           0.096047     0.135128   \n",
       "31006                                           0.097170     0.136438   \n",
       "31007                                           0.093897     0.132217   \n",
       "31008                                           0.100022     0.135202   \n",
       "31009                                           0.103967     0.137170   \n",
       "\n",
       "       sacrum_ratio_w/h  hip_ratio_w/h  pubic_arch_angle  \\\n",
       "0              1.414572       1.390511        142.678044   \n",
       "1              1.251339       1.334308        141.231641   \n",
       "2              1.410102       1.350552        116.365190   \n",
       "3              1.215281       1.486036        159.703035   \n",
       "4              1.445791       1.348099        131.343366   \n",
       "...                 ...            ...               ...   \n",
       "31005          1.090737       1.225370        124.424809   \n",
       "31006          1.167638       1.262513        130.924304   \n",
       "31007          1.317724       1.301696        115.933030   \n",
       "31008          1.447859       1.345279        101.631995   \n",
       "31009          1.446266       1.343411        105.144526   \n",
       "\n",
       "       acetabular_inclination_left  acetabular_inclination_right  \n",
       "0                        53.033866                     57.859459  \n",
       "1                        54.070014                     54.815992  \n",
       "2                        51.546225                     55.132446  \n",
       "3                        56.689420                     56.406852  \n",
       "4                        59.651556                     61.152404  \n",
       "...                            ...                           ...  \n",
       "31005                    57.535146                     57.062575  \n",
       "31006                    53.310400                     57.006480  \n",
       "31007                    52.860920                     50.817846  \n",
       "31008                    53.945919                     48.201155  \n",
       "31009                    53.038293                     53.927739  \n",
       "\n",
       "[31010 rows x 262 columns]"
      ]
     },
     "execution_count": 89,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hip_pheno_gwas = pd.read_csv('UKB_Imaging_Genetics/GCTA_INPUT_DATA/hip_pheno_gwas.txt', sep=\" \"); hip_pheno_gwas"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [],
   "source": [
    "# devide phenotypes into 4 parts each part has 70 phenotypes, all parts should contain first 2 columns (FID, IID)\n",
    "\n",
    "id_col = hip_pheno_gwas.columns[:2].tolist()\n",
    "col_part1 = hip_pheno_gwas.columns[2:72].tolist()\n",
    "col_part2 = hip_pheno_gwas.columns[72:142].tolist()\n",
    "col_part3 = hip_pheno_gwas.columns[142:212].tolist()\n",
    "col_part4 = hip_pheno_gwas.columns[212:].tolist()\n",
    "\n",
    "hip_pheno_gwas_part1 = pd.concat([hip_pheno_gwas[id_col], hip_pheno_gwas[col_part1]], axis=1)\n",
    "hip_pheno_gwas_part2 = pd.concat([hip_pheno_gwas[id_col], hip_pheno_gwas[col_part2]], axis=1)\n",
    "hip_pheno_gwas_part3 = pd.concat([hip_pheno_gwas[id_col], hip_pheno_gwas[col_part3]], axis=1)\n",
    "hip_pheno_gwas_part4 = pd.concat([hip_pheno_gwas[id_col], hip_pheno_gwas[col_part4]], axis=1)\n",
    "\n",
    "hip_pheno_gwas_part1.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_pheno_gwas_part1.txt', index=False, sep = \" \")\n",
    "hip_pheno_gwas_part2.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_pheno_gwas_part2.txt', index=False, sep = \" \")\n",
    "hip_pheno_gwas_part3.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_pheno_gwas_part3.txt', index=False, sep = \" \")\n",
    "hip_pheno_gwas_part4.to_csv('UKB_Imaging_Genetics/GWAS_INPUT_DATA/hip_pheno_gwas_part4.txt', index=False, sep = \" \")"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Use drinking as an nagative control"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>image_id</th>\n",
       "      <th>file_name</th>\n",
       "      <th>Patient EID</th>\n",
       "      <th>p_sex</th>\n",
       "      <th>p_age</th>\n",
       "      <th>p_weight</th>\n",
       "      <th>p_height</th>\n",
       "      <th>iliac_crest_left2iliac_crest_right</th>\n",
       "      <th>iliac_crest_left2iliac_spine_left</th>\n",
       "      <th>iliac_crest_left2iliac_spine_right</th>\n",
       "      <th>...</th>\n",
       "      <th>inferior_iliac_spine_left2acetabular_inferior_right</th>\n",
       "      <th>inferior_iliac_spine_right2acetabular_inferior_left</th>\n",
       "      <th>inferior_iliac_spine_right2acetabular_inferior_right</th>\n",
       "      <th>acetabular_inferior_left2acetabular_inferior_right</th>\n",
       "      <th>hip_height</th>\n",
       "      <th>sacrum_ratio_w/h</th>\n",
       "      <th>hip_ratio_w/h</th>\n",
       "      <th>pubic_arch_angle</th>\n",
       "      <th>acetabular_inclination_left</th>\n",
       "      <th>acetabular_inclination_right</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>08760_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20160630134650.1.1...</td>\n",
       "      <td>1000276</td>\n",
       "      <td>F</td>\n",
       "      <td>67.0</td>\n",
       "      <td>60.0</td>\n",
       "      <td>159.0</td>\n",
       "      <td>0.119177</td>\n",
       "      <td>0.057544</td>\n",
       "      <td>0.161311</td>\n",
       "      <td>...</td>\n",
       "      <td>0.138035</td>\n",
       "      <td>0.141101</td>\n",
       "      <td>0.046274</td>\n",
       "      <td>0.118927</td>\n",
       "      <td>0.144527</td>\n",
       "      <td>1.414572</td>\n",
       "      <td>1.390511</td>\n",
       "      <td>142.678044</td>\n",
       "      <td>53.033866</td>\n",
       "      <td>57.859459</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>12347_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20170305174208.1.1...</td>\n",
       "      <td>1000742</td>\n",
       "      <td>F</td>\n",
       "      <td>67.0</td>\n",
       "      <td>77.0</td>\n",
       "      <td>161.0</td>\n",
       "      <td>0.123848</td>\n",
       "      <td>0.038570</td>\n",
       "      <td>0.155576</td>\n",
       "      <td>...</td>\n",
       "      <td>0.132737</td>\n",
       "      <td>0.130718</td>\n",
       "      <td>0.045501</td>\n",
       "      <td>0.109055</td>\n",
       "      <td>0.139060</td>\n",
       "      <td>1.251339</td>\n",
       "      <td>1.334308</td>\n",
       "      <td>141.231641</td>\n",
       "      <td>54.070014</td>\n",
       "      <td>54.815992</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14366_82</td>\n",
       "      <td>1.2.840.113619.2.110.212038.20190307144208.1.1...</td>\n",
       "      <td>1000951</td>\n",
       "      <td>M</td>\n",
       "      <td>78.0</td>\n",
       "      <td>60.0</td>\n",
       "      <td>170.0</td>\n",
       "      <td>0.113652</td>\n",
       "      <td>0.036685</td>\n",
       "      <td>0.147139</td>\n",
       "      <td>...</td>\n",
       "      <td>0.124839</td>\n",
       "      <td>0.125665</td>\n",
       "      <td>0.048471</td>\n",
       "      <td>0.101909</td>\n",
       "      <td>0.131161</td>\n",
       "      <td>1.410102</td>\n",
       "      <td>1.350552</td>\n",
       "      <td>116.365190</td>\n",
       "      <td>51.546225</td>\n",
       "      <td>55.132446</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>06366_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20151211144320.1.1...</td>\n",
       "      <td>1001025</td>\n",
       "      <td>F</td>\n",
       "      <td>55.0</td>\n",
       "      <td>69.0</td>\n",
       "      <td>167.0</td>\n",
       "      <td>0.115559</td>\n",
       "      <td>0.055194</td>\n",
       "      <td>0.162427</td>\n",
       "      <td>...</td>\n",
       "      <td>0.137322</td>\n",
       "      <td>0.139501</td>\n",
       "      <td>0.042405</td>\n",
       "      <td>0.118486</td>\n",
       "      <td>0.134804</td>\n",
       "      <td>1.215281</td>\n",
       "      <td>1.486036</td>\n",
       "      <td>159.703035</td>\n",
       "      <td>56.689420</td>\n",
       "      <td>56.406852</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>14857_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20170902122319.1.1...</td>\n",
       "      <td>1001091</td>\n",
       "      <td>F</td>\n",
       "      <td>60.0</td>\n",
       "      <td>65.0</td>\n",
       "      <td>162.0</td>\n",
       "      <td>0.119257</td>\n",
       "      <td>0.037780</td>\n",
       "      <td>0.155524</td>\n",
       "      <td>...</td>\n",
       "      <td>0.131174</td>\n",
       "      <td>0.131803</td>\n",
       "      <td>0.043280</td>\n",
       "      <td>0.113473</td>\n",
       "      <td>0.139348</td>\n",
       "      <td>1.445791</td>\n",
       "      <td>1.348099</td>\n",
       "      <td>131.343366</td>\n",
       "      <td>59.651556</td>\n",
       "      <td>61.152404</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31005</th>\n",
       "      <td>12964_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20170418184226.1.1...</td>\n",
       "      <td>6024029</td>\n",
       "      <td>M</td>\n",
       "      <td>50.0</td>\n",
       "      <td>79.0</td>\n",
       "      <td>180.0</td>\n",
       "      <td>0.115447</td>\n",
       "      <td>0.030139</td>\n",
       "      <td>0.141083</td>\n",
       "      <td>...</td>\n",
       "      <td>0.116595</td>\n",
       "      <td>0.121475</td>\n",
       "      <td>0.046520</td>\n",
       "      <td>0.096047</td>\n",
       "      <td>0.135128</td>\n",
       "      <td>1.090737</td>\n",
       "      <td>1.225370</td>\n",
       "      <td>124.424809</td>\n",
       "      <td>57.535146</td>\n",
       "      <td>57.062575</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31006</th>\n",
       "      <td>02193_82</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20180811105258.1.1...</td>\n",
       "      <td>6024037</td>\n",
       "      <td>M</td>\n",
       "      <td>61.0</td>\n",
       "      <td>84.0</td>\n",
       "      <td>181.0</td>\n",
       "      <td>0.109468</td>\n",
       "      <td>0.046472</td>\n",
       "      <td>0.139892</td>\n",
       "      <td>...</td>\n",
       "      <td>0.122303</td>\n",
       "      <td>0.120067</td>\n",
       "      <td>0.045412</td>\n",
       "      <td>0.097170</td>\n",
       "      <td>0.136438</td>\n",
       "      <td>1.167638</td>\n",
       "      <td>1.262513</td>\n",
       "      <td>130.924304</td>\n",
       "      <td>53.310400</td>\n",
       "      <td>57.006480</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31007</th>\n",
       "      <td>11798_82</td>\n",
       "      <td>1.2.840.113619.2.110.212038.20180618143159.1.1...</td>\n",
       "      <td>6024119</td>\n",
       "      <td>M</td>\n",
       "      <td>68.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>165.0</td>\n",
       "      <td>0.117222</td>\n",
       "      <td>0.031103</td>\n",
       "      <td>0.146952</td>\n",
       "      <td>...</td>\n",
       "      <td>0.119512</td>\n",
       "      <td>0.126894</td>\n",
       "      <td>0.052668</td>\n",
       "      <td>0.093897</td>\n",
       "      <td>0.132217</td>\n",
       "      <td>1.317724</td>\n",
       "      <td>1.301696</td>\n",
       "      <td>115.933030</td>\n",
       "      <td>52.860920</td>\n",
       "      <td>50.817846</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31008</th>\n",
       "      <td>04905_82</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20190219154149.1.1...</td>\n",
       "      <td>6024257</td>\n",
       "      <td>M</td>\n",
       "      <td>71.0</td>\n",
       "      <td>74.0</td>\n",
       "      <td>169.0</td>\n",
       "      <td>0.121064</td>\n",
       "      <td>0.039582</td>\n",
       "      <td>0.151179</td>\n",
       "      <td>...</td>\n",
       "      <td>0.125164</td>\n",
       "      <td>0.128997</td>\n",
       "      <td>0.049194</td>\n",
       "      <td>0.100022</td>\n",
       "      <td>0.135202</td>\n",
       "      <td>1.447859</td>\n",
       "      <td>1.345279</td>\n",
       "      <td>101.631995</td>\n",
       "      <td>53.945919</td>\n",
       "      <td>48.201155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31009</th>\n",
       "      <td>02700_82</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20180912183226.1.1...</td>\n",
       "      <td>6024532</td>\n",
       "      <td>M</td>\n",
       "      <td>67.0</td>\n",
       "      <td>75.0</td>\n",
       "      <td>175.0</td>\n",
       "      <td>0.112109</td>\n",
       "      <td>0.043879</td>\n",
       "      <td>0.150154</td>\n",
       "      <td>...</td>\n",
       "      <td>0.127369</td>\n",
       "      <td>0.131003</td>\n",
       "      <td>0.045023</td>\n",
       "      <td>0.103967</td>\n",
       "      <td>0.137170</td>\n",
       "      <td>1.446266</td>\n",
       "      <td>1.343411</td>\n",
       "      <td>105.144526</td>\n",
       "      <td>53.038293</td>\n",
       "      <td>53.927739</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>31010 rows × 266 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       image_id                                          file_name  \\\n",
       "0      08760_93  1.2.840.113619.2.110.210419.20160630134650.1.1...   \n",
       "1      12347_93  1.2.840.113619.2.110.210419.20170305174208.1.1...   \n",
       "2      14366_82  1.2.840.113619.2.110.212038.20190307144208.1.1...   \n",
       "3      06366_93  1.2.840.113619.2.110.210419.20151211144320.1.1...   \n",
       "4      14857_93  1.2.840.113619.2.110.210419.20170902122319.1.1...   \n",
       "...         ...                                                ...   \n",
       "31005  12964_93  1.2.840.113619.2.110.210419.20170418184226.1.1...   \n",
       "31006  02193_82  1.2.840.113619.2.110.210419.20180811105258.1.1...   \n",
       "31007  11798_82  1.2.840.113619.2.110.212038.20180618143159.1.1...   \n",
       "31008  04905_82  1.2.840.113619.2.110.210419.20190219154149.1.1...   \n",
       "31009  02700_82  1.2.840.113619.2.110.210419.20180912183226.1.1...   \n",
       "\n",
       "       Patient EID p_sex  p_age  p_weight  p_height  \\\n",
       "0          1000276     F   67.0      60.0     159.0   \n",
       "1          1000742     F   67.0      77.0     161.0   \n",
       "2          1000951     M   78.0      60.0     170.0   \n",
       "3          1001025     F   55.0      69.0     167.0   \n",
       "4          1001091     F   60.0      65.0     162.0   \n",
       "...            ...   ...    ...       ...       ...   \n",
       "31005      6024029     M   50.0      79.0     180.0   \n",
       "31006      6024037     M   61.0      84.0     181.0   \n",
       "31007      6024119     M   68.0      70.0     165.0   \n",
       "31008      6024257     M   71.0      74.0     169.0   \n",
       "31009      6024532     M   67.0      75.0     175.0   \n",
       "\n",
       "       iliac_crest_left2iliac_crest_right  iliac_crest_left2iliac_spine_left  \\\n",
       "0                                0.119177                           0.057544   \n",
       "1                                0.123848                           0.038570   \n",
       "2                                0.113652                           0.036685   \n",
       "3                                0.115559                           0.055194   \n",
       "4                                0.119257                           0.037780   \n",
       "...                                   ...                                ...   \n",
       "31005                            0.115447                           0.030139   \n",
       "31006                            0.109468                           0.046472   \n",
       "31007                            0.117222                           0.031103   \n",
       "31008                            0.121064                           0.039582   \n",
       "31009                            0.112109                           0.043879   \n",
       "\n",
       "       iliac_crest_left2iliac_spine_right  ...  \\\n",
       "0                                0.161311  ...   \n",
       "1                                0.155576  ...   \n",
       "2                                0.147139  ...   \n",
       "3                                0.162427  ...   \n",
       "4                                0.155524  ...   \n",
       "...                                   ...  ...   \n",
       "31005                            0.141083  ...   \n",
       "31006                            0.139892  ...   \n",
       "31007                            0.146952  ...   \n",
       "31008                            0.151179  ...   \n",
       "31009                            0.150154  ...   \n",
       "\n",
       "       inferior_iliac_spine_left2acetabular_inferior_right  \\\n",
       "0                                               0.138035     \n",
       "1                                               0.132737     \n",
       "2                                               0.124839     \n",
       "3                                               0.137322     \n",
       "4                                               0.131174     \n",
       "...                                                  ...     \n",
       "31005                                           0.116595     \n",
       "31006                                           0.122303     \n",
       "31007                                           0.119512     \n",
       "31008                                           0.125164     \n",
       "31009                                           0.127369     \n",
       "\n",
       "       inferior_iliac_spine_right2acetabular_inferior_left  \\\n",
       "0                                               0.141101     \n",
       "1                                               0.130718     \n",
       "2                                               0.125665     \n",
       "3                                               0.139501     \n",
       "4                                               0.131803     \n",
       "...                                                  ...     \n",
       "31005                                           0.121475     \n",
       "31006                                           0.120067     \n",
       "31007                                           0.126894     \n",
       "31008                                           0.128997     \n",
       "31009                                           0.131003     \n",
       "\n",
       "       inferior_iliac_spine_right2acetabular_inferior_right  \\\n",
       "0                                               0.046274      \n",
       "1                                               0.045501      \n",
       "2                                               0.048471      \n",
       "3                                               0.042405      \n",
       "4                                               0.043280      \n",
       "...                                                  ...      \n",
       "31005                                           0.046520      \n",
       "31006                                           0.045412      \n",
       "31007                                           0.052668      \n",
       "31008                                           0.049194      \n",
       "31009                                           0.045023      \n",
       "\n",
       "       acetabular_inferior_left2acetabular_inferior_right  hip_height  \\\n",
       "0                                               0.118927     0.144527   \n",
       "1                                               0.109055     0.139060   \n",
       "2                                               0.101909     0.131161   \n",
       "3                                               0.118486     0.134804   \n",
       "4                                               0.113473     0.139348   \n",
       "...                                                  ...          ...   \n",
       "31005                                           0.096047     0.135128   \n",
       "31006                                           0.097170     0.136438   \n",
       "31007                                           0.093897     0.132217   \n",
       "31008                                           0.100022     0.135202   \n",
       "31009                                           0.103967     0.137170   \n",
       "\n",
       "       sacrum_ratio_w/h  hip_ratio_w/h  pubic_arch_angle  \\\n",
       "0              1.414572       1.390511        142.678044   \n",
       "1              1.251339       1.334308        141.231641   \n",
       "2              1.410102       1.350552        116.365190   \n",
       "3              1.215281       1.486036        159.703035   \n",
       "4              1.445791       1.348099        131.343366   \n",
       "...                 ...            ...               ...   \n",
       "31005          1.090737       1.225370        124.424809   \n",
       "31006          1.167638       1.262513        130.924304   \n",
       "31007          1.317724       1.301696        115.933030   \n",
       "31008          1.447859       1.345279        101.631995   \n",
       "31009          1.446266       1.343411        105.144526   \n",
       "\n",
       "       acetabular_inclination_left  acetabular_inclination_right  \n",
       "0                        53.033866                     57.859459  \n",
       "1                        54.070014                     54.815992  \n",
       "2                        51.546225                     55.132446  \n",
       "3                        56.689420                     56.406852  \n",
       "4                        59.651556                     61.152404  \n",
       "...                            ...                           ...  \n",
       "31005                    57.535146                     57.062575  \n",
       "31006                    53.310400                     57.006480  \n",
       "31007                    52.860920                     50.817846  \n",
       "31008                    53.945919                     48.201155  \n",
       "31009                    53.038293                     53.927739  \n",
       "\n",
       "[31010 rows x 266 columns]"
      ]
     },
     "execution_count": 125,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hip_pheno_gwas = pd.read_csv(\"key_results/hip_pheno_23_eid_flt.csv\"); hip_pheno_gwas"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>eid</th>\n",
       "      <th>1568-0.0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1000011</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1000026</td>\n",
       "      <td>8.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1000032</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1000044</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1000058</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502485</th>\n",
       "      <td>6024884</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502486</th>\n",
       "      <td>6024895</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502487</th>\n",
       "      <td>6024901</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502488</th>\n",
       "      <td>6024913</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502489</th>\n",
       "      <td>6024922</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>502490 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            eid  1568-0.0\n",
       "0       1000011       NaN\n",
       "1       1000026       8.0\n",
       "2       1000032       1.0\n",
       "3       1000044       NaN\n",
       "4       1000058       NaN\n",
       "...         ...       ...\n",
       "502485  6024884       6.0\n",
       "502486  6024895       0.0\n",
       "502487  6024901       2.0\n",
       "502488  6024913       0.0\n",
       "502489  6024922       1.0\n",
       "\n",
       "[502490 rows x 2 columns]"
      ]
     },
     "execution_count": 118,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# fid for Average weekly red wine intake\n",
    "drinking_fid = pd.read_csv(\"gwas/fid1568.csv\")[['eid', '1568-0.0']]; drinking_fid"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>image_id</th>\n",
       "      <th>file_name</th>\n",
       "      <th>Patient EID</th>\n",
       "      <th>p_sex</th>\n",
       "      <th>p_age</th>\n",
       "      <th>p_weight</th>\n",
       "      <th>p_height</th>\n",
       "      <th>iliac_crest_left2iliac_crest_right</th>\n",
       "      <th>iliac_crest_left2iliac_spine_left</th>\n",
       "      <th>iliac_crest_left2iliac_spine_right</th>\n",
       "      <th>...</th>\n",
       "      <th>inferior_iliac_spine_left2acetabular_inferior_right</th>\n",
       "      <th>inferior_iliac_spine_right2acetabular_inferior_left</th>\n",
       "      <th>inferior_iliac_spine_right2acetabular_inferior_right</th>\n",
       "      <th>acetabular_inferior_left2acetabular_inferior_right</th>\n",
       "      <th>hip_height</th>\n",
       "      <th>sacrum_ratio_w/h</th>\n",
       "      <th>hip_ratio_w/h</th>\n",
       "      <th>pubic_arch_angle</th>\n",
       "      <th>acetabular_inclination_left</th>\n",
       "      <th>acetabular_inclination_right</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>08760_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20160630134650.1.1...</td>\n",
       "      <td>1000276</td>\n",
       "      <td>F</td>\n",
       "      <td>67.0</td>\n",
       "      <td>60.0</td>\n",
       "      <td>159.0</td>\n",
       "      <td>0.119177</td>\n",
       "      <td>0.057544</td>\n",
       "      <td>0.161311</td>\n",
       "      <td>...</td>\n",
       "      <td>0.138035</td>\n",
       "      <td>0.141101</td>\n",
       "      <td>0.046274</td>\n",
       "      <td>0.118927</td>\n",
       "      <td>0.144527</td>\n",
       "      <td>1.414572</td>\n",
       "      <td>1.390511</td>\n",
       "      <td>142.678044</td>\n",
       "      <td>53.033866</td>\n",
       "      <td>57.859459</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>12347_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20170305174208.1.1...</td>\n",
       "      <td>1000742</td>\n",
       "      <td>F</td>\n",
       "      <td>67.0</td>\n",
       "      <td>77.0</td>\n",
       "      <td>161.0</td>\n",
       "      <td>0.123848</td>\n",
       "      <td>0.038570</td>\n",
       "      <td>0.155576</td>\n",
       "      <td>...</td>\n",
       "      <td>0.132737</td>\n",
       "      <td>0.130718</td>\n",
       "      <td>0.045501</td>\n",
       "      <td>0.109055</td>\n",
       "      <td>0.139060</td>\n",
       "      <td>1.251339</td>\n",
       "      <td>1.334308</td>\n",
       "      <td>141.231641</td>\n",
       "      <td>54.070014</td>\n",
       "      <td>54.815992</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14366_82</td>\n",
       "      <td>1.2.840.113619.2.110.212038.20190307144208.1.1...</td>\n",
       "      <td>1000951</td>\n",
       "      <td>M</td>\n",
       "      <td>78.0</td>\n",
       "      <td>60.0</td>\n",
       "      <td>170.0</td>\n",
       "      <td>0.113652</td>\n",
       "      <td>0.036685</td>\n",
       "      <td>0.147139</td>\n",
       "      <td>...</td>\n",
       "      <td>0.124839</td>\n",
       "      <td>0.125665</td>\n",
       "      <td>0.048471</td>\n",
       "      <td>0.101909</td>\n",
       "      <td>0.131161</td>\n",
       "      <td>1.410102</td>\n",
       "      <td>1.350552</td>\n",
       "      <td>116.365190</td>\n",
       "      <td>51.546225</td>\n",
       "      <td>55.132446</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>06366_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20151211144320.1.1...</td>\n",
       "      <td>1001025</td>\n",
       "      <td>F</td>\n",
       "      <td>55.0</td>\n",
       "      <td>69.0</td>\n",
       "      <td>167.0</td>\n",
       "      <td>0.115559</td>\n",
       "      <td>0.055194</td>\n",
       "      <td>0.162427</td>\n",
       "      <td>...</td>\n",
       "      <td>0.137322</td>\n",
       "      <td>0.139501</td>\n",
       "      <td>0.042405</td>\n",
       "      <td>0.118486</td>\n",
       "      <td>0.134804</td>\n",
       "      <td>1.215281</td>\n",
       "      <td>1.486036</td>\n",
       "      <td>159.703035</td>\n",
       "      <td>56.689420</td>\n",
       "      <td>56.406852</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>14857_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20170902122319.1.1...</td>\n",
       "      <td>1001091</td>\n",
       "      <td>F</td>\n",
       "      <td>60.0</td>\n",
       "      <td>65.0</td>\n",
       "      <td>162.0</td>\n",
       "      <td>0.119257</td>\n",
       "      <td>0.037780</td>\n",
       "      <td>0.155524</td>\n",
       "      <td>...</td>\n",
       "      <td>0.131174</td>\n",
       "      <td>0.131803</td>\n",
       "      <td>0.043280</td>\n",
       "      <td>0.113473</td>\n",
       "      <td>0.139348</td>\n",
       "      <td>1.445791</td>\n",
       "      <td>1.348099</td>\n",
       "      <td>131.343366</td>\n",
       "      <td>59.651556</td>\n",
       "      <td>61.152404</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31005</th>\n",
       "      <td>12964_93</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20170418184226.1.1...</td>\n",
       "      <td>6024029</td>\n",
       "      <td>M</td>\n",
       "      <td>50.0</td>\n",
       "      <td>79.0</td>\n",
       "      <td>180.0</td>\n",
       "      <td>0.115447</td>\n",
       "      <td>0.030139</td>\n",
       "      <td>0.141083</td>\n",
       "      <td>...</td>\n",
       "      <td>0.116595</td>\n",
       "      <td>0.121475</td>\n",
       "      <td>0.046520</td>\n",
       "      <td>0.096047</td>\n",
       "      <td>0.135128</td>\n",
       "      <td>1.090737</td>\n",
       "      <td>1.225370</td>\n",
       "      <td>124.424809</td>\n",
       "      <td>57.535146</td>\n",
       "      <td>57.062575</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31006</th>\n",
       "      <td>02193_82</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20180811105258.1.1...</td>\n",
       "      <td>6024037</td>\n",
       "      <td>M</td>\n",
       "      <td>61.0</td>\n",
       "      <td>84.0</td>\n",
       "      <td>181.0</td>\n",
       "      <td>0.109468</td>\n",
       "      <td>0.046472</td>\n",
       "      <td>0.139892</td>\n",
       "      <td>...</td>\n",
       "      <td>0.122303</td>\n",
       "      <td>0.120067</td>\n",
       "      <td>0.045412</td>\n",
       "      <td>0.097170</td>\n",
       "      <td>0.136438</td>\n",
       "      <td>1.167638</td>\n",
       "      <td>1.262513</td>\n",
       "      <td>130.924304</td>\n",
       "      <td>53.310400</td>\n",
       "      <td>57.006480</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31007</th>\n",
       "      <td>11798_82</td>\n",
       "      <td>1.2.840.113619.2.110.212038.20180618143159.1.1...</td>\n",
       "      <td>6024119</td>\n",
       "      <td>M</td>\n",
       "      <td>68.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>165.0</td>\n",
       "      <td>0.117222</td>\n",
       "      <td>0.031103</td>\n",
       "      <td>0.146952</td>\n",
       "      <td>...</td>\n",
       "      <td>0.119512</td>\n",
       "      <td>0.126894</td>\n",
       "      <td>0.052668</td>\n",
       "      <td>0.093897</td>\n",
       "      <td>0.132217</td>\n",
       "      <td>1.317724</td>\n",
       "      <td>1.301696</td>\n",
       "      <td>115.933030</td>\n",
       "      <td>52.860920</td>\n",
       "      <td>50.817846</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31008</th>\n",
       "      <td>04905_82</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20190219154149.1.1...</td>\n",
       "      <td>6024257</td>\n",
       "      <td>M</td>\n",
       "      <td>71.0</td>\n",
       "      <td>74.0</td>\n",
       "      <td>169.0</td>\n",
       "      <td>0.121064</td>\n",
       "      <td>0.039582</td>\n",
       "      <td>0.151179</td>\n",
       "      <td>...</td>\n",
       "      <td>0.125164</td>\n",
       "      <td>0.128997</td>\n",
       "      <td>0.049194</td>\n",
       "      <td>0.100022</td>\n",
       "      <td>0.135202</td>\n",
       "      <td>1.447859</td>\n",
       "      <td>1.345279</td>\n",
       "      <td>101.631995</td>\n",
       "      <td>53.945919</td>\n",
       "      <td>48.201155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31009</th>\n",
       "      <td>02700_82</td>\n",
       "      <td>1.2.840.113619.2.110.210419.20180912183226.1.1...</td>\n",
       "      <td>6024532</td>\n",
       "      <td>M</td>\n",
       "      <td>67.0</td>\n",
       "      <td>75.0</td>\n",
       "      <td>175.0</td>\n",
       "      <td>0.112109</td>\n",
       "      <td>0.043879</td>\n",
       "      <td>0.150154</td>\n",
       "      <td>...</td>\n",
       "      <td>0.127369</td>\n",
       "      <td>0.131003</td>\n",
       "      <td>0.045023</td>\n",
       "      <td>0.103967</td>\n",
       "      <td>0.137170</td>\n",
       "      <td>1.446266</td>\n",
       "      <td>1.343411</td>\n",
       "      <td>105.144526</td>\n",
       "      <td>53.038293</td>\n",
       "      <td>53.927739</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>31010 rows × 266 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       image_id                                          file_name  \\\n",
       "0      08760_93  1.2.840.113619.2.110.210419.20160630134650.1.1...   \n",
       "1      12347_93  1.2.840.113619.2.110.210419.20170305174208.1.1...   \n",
       "2      14366_82  1.2.840.113619.2.110.212038.20190307144208.1.1...   \n",
       "3      06366_93  1.2.840.113619.2.110.210419.20151211144320.1.1...   \n",
       "4      14857_93  1.2.840.113619.2.110.210419.20170902122319.1.1...   \n",
       "...         ...                                                ...   \n",
       "31005  12964_93  1.2.840.113619.2.110.210419.20170418184226.1.1...   \n",
       "31006  02193_82  1.2.840.113619.2.110.210419.20180811105258.1.1...   \n",
       "31007  11798_82  1.2.840.113619.2.110.212038.20180618143159.1.1...   \n",
       "31008  04905_82  1.2.840.113619.2.110.210419.20190219154149.1.1...   \n",
       "31009  02700_82  1.2.840.113619.2.110.210419.20180912183226.1.1...   \n",
       "\n",
       "       Patient EID p_sex  p_age  p_weight  p_height  \\\n",
       "0          1000276     F   67.0      60.0     159.0   \n",
       "1          1000742     F   67.0      77.0     161.0   \n",
       "2          1000951     M   78.0      60.0     170.0   \n",
       "3          1001025     F   55.0      69.0     167.0   \n",
       "4          1001091     F   60.0      65.0     162.0   \n",
       "...            ...   ...    ...       ...       ...   \n",
       "31005      6024029     M   50.0      79.0     180.0   \n",
       "31006      6024037     M   61.0      84.0     181.0   \n",
       "31007      6024119     M   68.0      70.0     165.0   \n",
       "31008      6024257     M   71.0      74.0     169.0   \n",
       "31009      6024532     M   67.0      75.0     175.0   \n",
       "\n",
       "       iliac_crest_left2iliac_crest_right  iliac_crest_left2iliac_spine_left  \\\n",
       "0                                0.119177                           0.057544   \n",
       "1                                0.123848                           0.038570   \n",
       "2                                0.113652                           0.036685   \n",
       "3                                0.115559                           0.055194   \n",
       "4                                0.119257                           0.037780   \n",
       "...                                   ...                                ...   \n",
       "31005                            0.115447                           0.030139   \n",
       "31006                            0.109468                           0.046472   \n",
       "31007                            0.117222                           0.031103   \n",
       "31008                            0.121064                           0.039582   \n",
       "31009                            0.112109                           0.043879   \n",
       "\n",
       "       iliac_crest_left2iliac_spine_right  ...  \\\n",
       "0                                0.161311  ...   \n",
       "1                                0.155576  ...   \n",
       "2                                0.147139  ...   \n",
       "3                                0.162427  ...   \n",
       "4                                0.155524  ...   \n",
       "...                                   ...  ...   \n",
       "31005                            0.141083  ...   \n",
       "31006                            0.139892  ...   \n",
       "31007                            0.146952  ...   \n",
       "31008                            0.151179  ...   \n",
       "31009                            0.150154  ...   \n",
       "\n",
       "       inferior_iliac_spine_left2acetabular_inferior_right  \\\n",
       "0                                               0.138035     \n",
       "1                                               0.132737     \n",
       "2                                               0.124839     \n",
       "3                                               0.137322     \n",
       "4                                               0.131174     \n",
       "...                                                  ...     \n",
       "31005                                           0.116595     \n",
       "31006                                           0.122303     \n",
       "31007                                           0.119512     \n",
       "31008                                           0.125164     \n",
       "31009                                           0.127369     \n",
       "\n",
       "       inferior_iliac_spine_right2acetabular_inferior_left  \\\n",
       "0                                               0.141101     \n",
       "1                                               0.130718     \n",
       "2                                               0.125665     \n",
       "3                                               0.139501     \n",
       "4                                               0.131803     \n",
       "...                                                  ...     \n",
       "31005                                           0.121475     \n",
       "31006                                           0.120067     \n",
       "31007                                           0.126894     \n",
       "31008                                           0.128997     \n",
       "31009                                           0.131003     \n",
       "\n",
       "       inferior_iliac_spine_right2acetabular_inferior_right  \\\n",
       "0                                               0.046274      \n",
       "1                                               0.045501      \n",
       "2                                               0.048471      \n",
       "3                                               0.042405      \n",
       "4                                               0.043280      \n",
       "...                                                  ...      \n",
       "31005                                           0.046520      \n",
       "31006                                           0.045412      \n",
       "31007                                           0.052668      \n",
       "31008                                           0.049194      \n",
       "31009                                           0.045023      \n",
       "\n",
       "       acetabular_inferior_left2acetabular_inferior_right  hip_height  \\\n",
       "0                                               0.118927     0.144527   \n",
       "1                                               0.109055     0.139060   \n",
       "2                                               0.101909     0.131161   \n",
       "3                                               0.118486     0.134804   \n",
       "4                                               0.113473     0.139348   \n",
       "...                                                  ...          ...   \n",
       "31005                                           0.096047     0.135128   \n",
       "31006                                           0.097170     0.136438   \n",
       "31007                                           0.093897     0.132217   \n",
       "31008                                           0.100022     0.135202   \n",
       "31009                                           0.103967     0.137170   \n",
       "\n",
       "       sacrum_ratio_w/h  hip_ratio_w/h  pubic_arch_angle  \\\n",
       "0              1.414572       1.390511        142.678044   \n",
       "1              1.251339       1.334308        141.231641   \n",
       "2              1.410102       1.350552        116.365190   \n",
       "3              1.215281       1.486036        159.703035   \n",
       "4              1.445791       1.348099        131.343366   \n",
       "...                 ...            ...               ...   \n",
       "31005          1.090737       1.225370        124.424809   \n",
       "31006          1.167638       1.262513        130.924304   \n",
       "31007          1.317724       1.301696        115.933030   \n",
       "31008          1.447859       1.345279        101.631995   \n",
       "31009          1.446266       1.343411        105.144526   \n",
       "\n",
       "       acetabular_inclination_left  acetabular_inclination_right  \n",
       "0                        53.033866                     57.859459  \n",
       "1                        54.070014                     54.815992  \n",
       "2                        51.546225                     55.132446  \n",
       "3                        56.689420                     56.406852  \n",
       "4                        59.651556                     61.152404  \n",
       "...                            ...                           ...  \n",
       "31005                    57.535146                     57.062575  \n",
       "31006                    53.310400                     57.006480  \n",
       "31007                    52.860920                     50.817846  \n",
       "31008                    53.945919                     48.201155  \n",
       "31009                    53.038293                     53.927739  \n",
       "\n",
       "[31010 rows x 266 columns]"
      ]
     },
     "execution_count": 128,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hip_pheno_gwas"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>drinking</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1000276</td>\n",
       "      <td>1000276</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1000742</td>\n",
       "      <td>1000742</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1000951</td>\n",
       "      <td>1000951</td>\n",
       "      <td>25.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1001025</td>\n",
       "      <td>1001025</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1001091</td>\n",
       "      <td>1001091</td>\n",
       "      <td>10.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31005</th>\n",
       "      <td>6024029</td>\n",
       "      <td>6024029</td>\n",
       "      <td>12.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31006</th>\n",
       "      <td>6024037</td>\n",
       "      <td>6024037</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31007</th>\n",
       "      <td>6024119</td>\n",
       "      <td>6024119</td>\n",
       "      <td>9.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31008</th>\n",
       "      <td>6024257</td>\n",
       "      <td>6024257</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31009</th>\n",
       "      <td>6024532</td>\n",
       "      <td>6024532</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>31010 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           FID      IID  drinking\n",
       "0      1000276  1000276       4.0\n",
       "1      1000742  1000742       NaN\n",
       "2      1000951  1000951      25.0\n",
       "3      1001025  1001025       4.0\n",
       "4      1001091  1001091      10.0\n",
       "...        ...      ...       ...\n",
       "31005  6024029  6024029      12.0\n",
       "31006  6024037  6024037       0.0\n",
       "31007  6024119  6024119       9.0\n",
       "31008  6024257  6024257       6.0\n",
       "31009  6024532  6024532       0.0\n",
       "\n",
       "[31010 rows x 3 columns]"
      ]
     },
     "execution_count": 131,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hip_pheno_gwas_eid =  hip_pheno_gwas[['Patient EID']]\n",
    "neg_pheno_gwas = hip_pheno_gwas_eid.merge(drinking_fid, left_on='Patient EID', right_on='eid', how='inner').rename(columns={'Patient EID': \"FID\",\n",
    "                                                                                                                            'eid': \"IID\", \n",
    "                                                                                                                            '1568-0.0': 'drinking'}); neg_pheno_gwas"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {},
   "outputs": [],
   "source": [
    "neg_pheno_gwas.dropna(inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "metadata": {},
   "outputs": [],
   "source": [
    "neg_pheno_gwas.to_csv(\"key_results/neg_pheno_gwas.txt\", index=False, sep=\" \")"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Clumped SNPs"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### for both sexes, male, and female separately"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 165,
   "metadata": {},
   "outputs": [],
   "source": [
    "path = 'UKB_Imaging_Genetics/PLINK_CLUMP_OUTPUT/'\n",
    "\n",
    "clump_both = pd.DataFrame()\n",
    "\n",
    "for file in os.listdir(os.path.join(path, 'hip_both_1000_noverbose')):\n",
    "    if file.endswith(\".ranges\"):\n",
    "        tmp_df = pd.read_csv(os.path.join(path, 'hip_both_1000_noverbose', file), sep=\"\\s+\")\n",
    "        pheno_name = file.split(\".\")[1]\n",
    "        tmp_df['pheno'] = pheno_name\n",
    "        if clump_both.empty:\n",
    "            clump_both = tmp_df\n",
    "        else:\n",
    "            clump_both = pd.concat([clump_both, tmp_df], axis=0)\n",
    "\n",
    "clump_male = pd.DataFrame()\n",
    "\n",
    "for file in os.listdir(os.path.join(path, 'hip_male_1000_noverbose')):\n",
    "    if file.endswith(\".ranges\"):\n",
    "        tmp_df = pd.read_csv(os.path.join(path, 'hip_male_1000_noverbose', file), sep=\"\\s+\")\n",
    "        pheno_name = file.split(\".\")[1]\n",
    "        tmp_df['pheno'] = pheno_name\n",
    "        if clump_male.empty:\n",
    "            clump_male = tmp_df\n",
    "        else:\n",
    "            clump_male = pd.concat([clump_male, tmp_df], axis=0)\n",
    "\n",
    "clump_female = pd.DataFrame()\n",
    "\n",
    "for file in os.listdir(os.path.join(path, 'hip_female_1000_noverbose')):\n",
    "    if file.endswith(\".ranges\"):\n",
    "        tmp_df = pd.read_csv(os.path.join(path, 'hip_female_1000_noverbose', file), sep=\"\\s+\")\n",
    "        pheno_name = file.split(\".\")[1]\n",
    "        tmp_df['pheno'] = pheno_name\n",
    "        if clump_female.empty:\n",
    "            clump_female = tmp_df\n",
    "        else:\n",
    "            clump_female = pd.concat([clump_female, tmp_df], axis=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 175,
   "metadata": {},
   "outputs": [],
   "source": [
    "clump_both.to_csv(\"UKB_Imaging_Genetics/PLINK_CLUMP_OUTPUT/hip_1000_noverbose_summary/clump_both.txt\", index=False, sep=\" \")\n",
    "clump_female.to_csv(\"UKB_Imaging_Genetics/PLINK_CLUMP_OUTPUT/hip_1000_noverbose_summary/clump_female.txt\", index=False, sep=\" \")\n",
    "clump_male.to_csv(\"UKB_Imaging_Genetics/PLINK_CLUMP_OUTPUT/hip_1000_noverbose_summary/clump_male.txt\", index=False, sep=\" \")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 177,
   "metadata": {},
   "outputs": [],
   "source": [
    "clump_both[['SNP']].to_csv(\"UKB_Imaging_Genetics/PLINK_CLUMP_OUTPUT/hip_1000_noverbose_summary/clump_both_snp.txt\", index=False, sep=\" \", header=False)\n",
    "clump_female[['SNP']].to_csv(\"UKB_Imaging_Genetics/PLINK_CLUMP_OUTPUT/hip_1000_noverbose_summary/clump_female_snp.txt\", index=False, sep=\" \", header=False)\n",
    "clump_male[['SNP']].to_csv(\"UKB_Imaging_Genetics/PLINK_CLUMP_OUTPUT/hip_1000_noverbose_summary/clump_male_snp.txt\", index=False, sep=\" \", header=False)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### after doing pruning, merge the pruned snps with chr position"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 191,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "clump_both: 314, clump_both_pruned: 233\n",
      "clump_female: 74, clump_female_pruned: 59\n",
      "clump_male: 94, clump_male_pruned: 79\n"
     ]
    }
   ],
   "source": [
    "# get pruned snps\n",
    "\n",
    "pruned_both = pd.read_csv(\"UKB_Imaging_Genetics/PLINK_CLUMP_OUTPUT/hip_1000_noverbose_summary/clump_both_snp.prune.in\", sep=\"\\s+\", header=None)\n",
    "pruned_female = pd.read_csv(\"UKB_Imaging_Genetics/PLINK_CLUMP_OUTPUT/hip_1000_noverbose_summary/clump_female_snp.prune.in\", sep=\"\\s+\", header=None)\n",
    "pruned_male = pd.read_csv(\"UKB_Imaging_Genetics/PLINK_CLUMP_OUTPUT/hip_1000_noverbose_summary/clump_male_snp.prune.in\", sep=\"\\s+\", header=None)\n",
    "\n",
    "pruned_both_snps = pruned_both[0].tolist()\n",
    "pruned_female_snps = pruned_female[0].tolist()\n",
    "pruned_male_snps = pruned_male[0].tolist()\n",
    "\n",
    "clump_both_pruned = clump_both[clump_both['SNP'].isin(pruned_both_snps)]\n",
    "clump_female_pruned = clump_female[clump_female['SNP'].isin(pruned_female_snps)]\n",
    "clump_male_pruned = clump_male[clump_male['SNP'].isin(pruned_male_snps)]\n",
    "\n",
    "print(f\"clump_both: {clump_both['SNP'].unique().shape[0]}, clump_both_pruned: {clump_both_pruned['SNP'].unique().shape[0]}\")\n",
    "print(f\"clump_female: {clump_female['SNP'].unique().shape[0]}, clump_female_pruned: {clump_female_pruned['SNP'].unique().shape[0]}\")\n",
    "print(f\"clump_male: {clump_male['SNP'].unique().shape[0]}, clump_male_pruned: {clump_male_pruned['SNP'].unique().shape[0]}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 193,
   "metadata": {},
   "outputs": [],
   "source": [
    "clump_both_pruned.to_csv(\"UKB_Imaging_Genetics/PLINK_CLUMP_OUTPUT/hip_1000_noverbose_summary/clump_both_pruned.csv\", index=False,)\n",
    "clump_female_pruned.to_csv(\"UKB_Imaging_Genetics/PLINK_CLUMP_OUTPUT/hip_1000_noverbose_summary/clump_female_pruned.csv\", index=False)\n",
    "clump_male_pruned.to_csv(\"UKB_Imaging_Genetics/PLINK_CLUMP_OUTPUT/hip_1000_noverbose_summary/clump_male_pruned.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 194,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "clump_both: 314, clump_both_pruned: 201\n",
      "clump_female: 74, clump_female_pruned: 50\n",
      "clump_male: 94, clump_male_pruned: 68\n"
     ]
    }
   ],
   "source": [
    "# get pruned snps\n",
    "\n",
    "pruned_both = pd.read_csv(\"UKB_Imaging_Genetics/PLINK_CLUMP_OUTPUT/hip_1000_noverbose_summary/clump_both_snp_vif1.1.prune.in\", sep=\"\\s+\", header=None)\n",
    "pruned_female = pd.read_csv(\"UKB_Imaging_Genetics/PLINK_CLUMP_OUTPUT/hip_1000_noverbose_summary/clump_female_snp_vif1.1.prune.in\", sep=\"\\s+\", header=None)\n",
    "pruned_male = pd.read_csv(\"UKB_Imaging_Genetics/PLINK_CLUMP_OUTPUT/hip_1000_noverbose_summary/clump_male_snp_vif1.1.prune.in\", sep=\"\\s+\", header=None)\n",
    "\n",
    "pruned_both_snps = pruned_both[0].tolist()\n",
    "pruned_female_snps = pruned_female[0].tolist()\n",
    "pruned_male_snps = pruned_male[0].tolist()\n",
    "\n",
    "clump_both_pruned = clump_both[clump_both['SNP'].isin(pruned_both_snps)]\n",
    "clump_female_pruned = clump_female[clump_female['SNP'].isin(pruned_female_snps)]\n",
    "clump_male_pruned = clump_male[clump_male['SNP'].isin(pruned_male_snps)]\n",
    "\n",
    "print(f\"clump_both: {clump_both['SNP'].unique().shape[0]}, clump_both_pruned: {clump_both_pruned['SNP'].unique().shape[0]}\")\n",
    "print(f\"clump_female: {clump_female['SNP'].unique().shape[0]}, clump_female_pruned: {clump_female_pruned['SNP'].unique().shape[0]}\")\n",
    "print(f\"clump_male: {clump_male['SNP'].unique().shape[0]}, clump_male_pruned: {clump_male_pruned['SNP'].unique().shape[0]}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 229,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>SNP</th>\n",
       "      <th>P</th>\n",
       "      <th>N</th>\n",
       "      <th>POS</th>\n",
       "      <th>KB</th>\n",
       "      <th>RANGES</th>\n",
       "      <th>pheno</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>9</td>\n",
       "      <td>rs7855648</td>\n",
       "      <td>2.431000e-18</td>\n",
       "      <td>209</td>\n",
       "      <td>chr9:126134542..126708175</td>\n",
       "      <td>573.634</td>\n",
       "      <td>[CRB2,DENND1A,MIR601,MIR7150]</td>\n",
       "      <td>pelvic_width</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>11</td>\n",
       "      <td>rs947791</td>\n",
       "      <td>1.615000e-16</td>\n",
       "      <td>159</td>\n",
       "      <td>chr11:64824179..65368434</td>\n",
       "      <td>544.256</td>\n",
       "      <td>[CAPN1,CDC42EP2,CDCA5,DPF2,EHBP1L1,FAM89B,FAU,...</td>\n",
       "      <td>pelvic_width</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>15</td>\n",
       "      <td>rs11070646</td>\n",
       "      <td>4.098000e-14</td>\n",
       "      <td>203</td>\n",
       "      <td>chr15:48692609..48933821</td>\n",
       "      <td>241.213</td>\n",
       "      <td>[FBN1]</td>\n",
       "      <td>pelvic_width</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>8</td>\n",
       "      <td>rs7839249</td>\n",
       "      <td>8.703000e-12</td>\n",
       "      <td>104</td>\n",
       "      <td>chr8:72551864..72675903</td>\n",
       "      <td>124.040</td>\n",
       "      <td>[]</td>\n",
       "      <td>pelvic_width</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>12</td>\n",
       "      <td>rs17811547</td>\n",
       "      <td>2.247000e-11</td>\n",
       "      <td>791</td>\n",
       "      <td>chr12:83857937..84678383</td>\n",
       "      <td>820.447</td>\n",
       "      <td>[]</td>\n",
       "      <td>pelvic_width</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>5</td>\n",
       "      <td>rs493578</td>\n",
       "      <td>2.188000e-08</td>\n",
       "      <td>44</td>\n",
       "      <td>chr5:176325177..176531772</td>\n",
       "      <td>206.596</td>\n",
       "      <td>[FGFR4,HK3,UIMC1,ZNF346]</td>\n",
       "      <td>ear_left2ear_right</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>17</td>\n",
       "      <td>rs2663345</td>\n",
       "      <td>2.769000e-08</td>\n",
       "      <td>13</td>\n",
       "      <td>chr17:895549..929741</td>\n",
       "      <td>34.193</td>\n",
       "      <td>[ABR,MIR3183,TIMM22]</td>\n",
       "      <td>ear_left2ear_right</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>17</td>\n",
       "      <td>rs2006122</td>\n",
       "      <td>3.168000e-08</td>\n",
       "      <td>182</td>\n",
       "      <td>chr17:61621330..62021568</td>\n",
       "      <td>400.239</td>\n",
       "      <td>[CCDC47,CD79B,CSH1,CSH2,CSHL1,DCAF7,DDX42,FTSJ...</td>\n",
       "      <td>ear_left2ear_right</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>17</td>\n",
       "      <td>rs17690703</td>\n",
       "      <td>3.585000e-08</td>\n",
       "      <td>2641</td>\n",
       "      <td>chr17:43463493..44863133</td>\n",
       "      <td>1399.640</td>\n",
       "      <td>[ARHGAP27,ARL17A,ARL17B,CRHR1,CRHR1-IT1,KANSL1...</td>\n",
       "      <td>ear_left2ear_right</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>6</td>\n",
       "      <td>rs2858332</td>\n",
       "      <td>4.962000e-08</td>\n",
       "      <td>361</td>\n",
       "      <td>chr6:31840794..32681631</td>\n",
       "      <td>840.838</td>\n",
       "      <td>[AGER,AGPAT1,ATF6B,BTNL2,C2,C4A,C4B,C4B_2,C6or...</td>\n",
       "      <td>ear_left2ear_right</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>221 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    CHR         SNP             P     N                        POS        KB  \\\n",
       "0     9   rs7855648  2.431000e-18   209  chr9:126134542..126708175   573.634   \n",
       "1    11    rs947791  1.615000e-16   159   chr11:64824179..65368434   544.256   \n",
       "4    15  rs11070646  4.098000e-14   203   chr15:48692609..48933821   241.213   \n",
       "6     8   rs7839249  8.703000e-12   104    chr8:72551864..72675903   124.040   \n",
       "9    12  rs17811547  2.247000e-11   791   chr12:83857937..84678383   820.447   \n",
       "..  ...         ...           ...   ...                        ...       ...   \n",
       "21    5    rs493578  2.188000e-08    44  chr5:176325177..176531772   206.596   \n",
       "22   17   rs2663345  2.769000e-08    13       chr17:895549..929741    34.193   \n",
       "23   17   rs2006122  3.168000e-08   182   chr17:61621330..62021568   400.239   \n",
       "24   17  rs17690703  3.585000e-08  2641   chr17:43463493..44863133  1399.640   \n",
       "25    6   rs2858332  4.962000e-08   361    chr6:31840794..32681631   840.838   \n",
       "\n",
       "                                               RANGES               pheno  \n",
       "0                       [CRB2,DENND1A,MIR601,MIR7150]        pelvic_width  \n",
       "1   [CAPN1,CDC42EP2,CDCA5,DPF2,EHBP1L1,FAM89B,FAU,...        pelvic_width  \n",
       "4                                              [FBN1]        pelvic_width  \n",
       "6                                                  []        pelvic_width  \n",
       "9                                                  []        pelvic_width  \n",
       "..                                                ...                 ...  \n",
       "21                           [FGFR4,HK3,UIMC1,ZNF346]  ear_left2ear_right  \n",
       "22                               [ABR,MIR3183,TIMM22]  ear_left2ear_right  \n",
       "23  [CCDC47,CD79B,CSH1,CSH2,CSHL1,DCAF7,DDX42,FTSJ...  ear_left2ear_right  \n",
       "24  [ARHGAP27,ARL17A,ARL17B,CRHR1,CRHR1-IT1,KANSL1...  ear_left2ear_right  \n",
       "25  [AGER,AGPAT1,ATF6B,BTNL2,C2,C4A,C4B,C4B_2,C6or...  ear_left2ear_right  \n",
       "\n",
       "[221 rows x 8 columns]"
      ]
     },
     "execution_count": 229,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clump_both_pruned"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 195,
   "metadata": {},
   "outputs": [],
   "source": [
    "clump_both_pruned.to_csv(\"UKB_Imaging_Genetics/PLINK_CLUMP_OUTPUT/hip_1000_noverbose_summary/clump_both_pruned_vif1.1.csv\", index=False,)\n",
    "clump_female_pruned.to_csv(\"UKB_Imaging_Genetics/PLINK_CLUMP_OUTPUT/hip_1000_noverbose_summary/clump_female_pruned_vif1.1.csv\", index=False)\n",
    "clump_male_pruned.to_csv(\"UKB_Imaging_Genetics/PLINK_CLUMP_OUTPUT/hip_1000_noverbose_summary/clump_male_pruned_vif1.1.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 244,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv(\"UKB_Imaging_Genetics/PLINK_CLUMP_OUTPUT/hip_1000_noverbose_summary/clump_both_pruned_vif1.1.csv\")\n",
    "\n",
    "# Sort the dataframe by chromosome and start position\n",
    "df['POS'] = df['POS'].apply(lambda x: x.split(':')[1])\n",
    "df['start'], df['end'] = df['POS'].apply(lambda x: x.split('..')[0]), df['POS'].apply(lambda x: x.split('..')[1])\n",
    "df[['start', 'end']] = df[['start', 'end']].apply(pd.to_numeric)\n",
    "df.sort_values(['CHR', 'start'], inplace=True)\n",
    "\n",
    "# Calculate overlaps\n",
    "overlaps = []\n",
    "for chr_id in df['CHR'].unique():\n",
    "    chr_data = df[df['CHR'] == chr_id]\n",
    "    for i in range(len(chr_data) - 1):\n",
    "        for j in range(i + 1, len(chr_data)):\n",
    "            if chr_data.iloc[i]['SNP'] != chr_data.iloc[j]['SNP']:\n",
    "                start1, end1 = chr_data.iloc[i][['start', 'end']]\n",
    "                start2, end2 = chr_data.iloc[j][['start', 'end']]\n",
    "                overlap = max(0, min(end1, end2) - max(start1, start2))\n",
    "                overlaps.append({'CHR': chr_id, 'SNP1': chr_data.iloc[i]['SNP'], 'SNP2': chr_data.iloc[j]['SNP'], 'Overlap': overlap})\n",
    "\n",
    "# Convert to a dataframe\n",
    "overlaps_df = pd.DataFrame(overlaps)\n",
    "overlaps_df.sort_values('Overlap', ascending=False, inplace=True)\n",
    "overlaps_df.drop_duplicates(subset=['SNP1', 'SNP2'], inplace=True)\n",
    "overlaps_df.to_csv(\"UKB_Imaging_Genetics/PLINK_CLUMP_OUTPUT/hip_1000_noverbose_summary/clump_both_pruned_vif1.1_overlaps.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 245,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>SNP1</th>\n",
       "      <th>SNP2</th>\n",
       "      <th>Overlap</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>541</th>\n",
       "      <td>6</td>\n",
       "      <td>rs3134951</td>\n",
       "      <td>rs2858332</td>\n",
       "      <td>840837</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1149</th>\n",
       "      <td>15</td>\n",
       "      <td>rs6602997</td>\n",
       "      <td>rs4842854</td>\n",
       "      <td>809879</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>616</th>\n",
       "      <td>6</td>\n",
       "      <td>rs113308638</td>\n",
       "      <td>rs13206206</td>\n",
       "      <td>700675</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>477</th>\n",
       "      <td>4</td>\n",
       "      <td>rs13141641</td>\n",
       "      <td>rs11727676</td>\n",
       "      <td>545882</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>476</th>\n",
       "      <td>4</td>\n",
       "      <td>rs13141641</td>\n",
       "      <td>rs62346126</td>\n",
       "      <td>522697</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>497</th>\n",
       "      <td>5</td>\n",
       "      <td>rs7734399</td>\n",
       "      <td>rs61749613</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>495</th>\n",
       "      <td>4</td>\n",
       "      <td>rs17276260</td>\n",
       "      <td>rs12511271</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>494</th>\n",
       "      <td>4</td>\n",
       "      <td>rs10031912</td>\n",
       "      <td>rs12511271</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>493</th>\n",
       "      <td>4</td>\n",
       "      <td>rs10031912</td>\n",
       "      <td>rs17276260</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1448</th>\n",
       "      <td>22</td>\n",
       "      <td>rs131274</td>\n",
       "      <td>rs1569414</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1202 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHR         SNP1        SNP2  Overlap\n",
       "541     6    rs3134951   rs2858332   840837\n",
       "1149   15    rs6602997   rs4842854   809879\n",
       "616     6  rs113308638  rs13206206   700675\n",
       "477     4   rs13141641  rs11727676   545882\n",
       "476     4   rs13141641  rs62346126   522697\n",
       "...   ...          ...         ...      ...\n",
       "497     5    rs7734399  rs61749613        0\n",
       "495     4   rs17276260  rs12511271        0\n",
       "494     4   rs10031912  rs12511271        0\n",
       "493     4   rs10031912  rs17276260        0\n",
       "1448   22     rs131274   rs1569414        0\n",
       "\n",
       "[1202 rows x 4 columns]"
      ]
     },
     "execution_count": 245,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "overlaps_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 247,
   "metadata": {},
   "outputs": [],
   "source": [
    "overlaps_df[['SNP1', 'SNP2']].to_csv(\"UKB_Imaging_Genetics/PLINK_CLUMP_OUTPUT/hip_1000_noverbose_summary/clump_both_pruned_vif1.1_overlaps_snps.txt\", index=False, sep = ' ', header=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 166,
   "metadata": {},
   "outputs": [],
   "source": [
    "unique_snp_male = clump_male['SNP'].unique()\n",
    "unique_snp_female = clump_female['SNP'].unique()\n",
    "unique_snp_both = clump_both['SNP'].unique()\n",
    "\n",
    "fm_overlap = np.intersect1d(unique_snp_male, unique_snp_female)\n",
    "both_fm_overlap = [i for i in unique_snp_both if i in unique_snp_male or i in unique_snp_female]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 169,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "314"
      ]
     },
     "execution_count": 169,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(unique_snp_both)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 167,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "94"
      ]
     },
     "execution_count": 167,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(unique_snp_male)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 168,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "74"
      ]
     },
     "execution_count": 168,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(unique_snp_female)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_one_gene_snp(range):\n",
    "    list_element = range.replace('[', '').replace(']', '').split(',')\n",
    "    if len(list_element) == 1 and list_element[0] != '':\n",
    "        return True\n",
    "    return False"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "metadata": {},
   "outputs": [],
   "source": [
    "clump_male_one_gene = clump_male[clump_male['RANGES'].apply(lambda x: get_one_gene_snp(x))]\n",
    "clump_female_one_gene = clump_female[clump_female['RANGES'].apply(lambda x: get_one_gene_snp(x))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>SNP</th>\n",
       "      <th>P</th>\n",
       "      <th>N</th>\n",
       "      <th>POS</th>\n",
       "      <th>KB</th>\n",
       "      <th>RANGES</th>\n",
       "      <th>pheno</th>\n",
       "      <th>gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>20</td>\n",
       "      <td>rs1884902</td>\n",
       "      <td>1.833000e-09</td>\n",
       "      <td>50</td>\n",
       "      <td>chr20:6487687..6635509</td>\n",
       "      <td>147.823</td>\n",
       "      <td>[CASC20]</td>\n",
       "      <td>pelvic_width</td>\n",
       "      <td>NC_000020.11:g.6627855C&gt;T,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>15</td>\n",
       "      <td>rs28370880</td>\n",
       "      <td>7.197000e-09</td>\n",
       "      <td>126</td>\n",
       "      <td>chr15:48692609..48933821</td>\n",
       "      <td>241.213</td>\n",
       "      <td>[FBN1]</td>\n",
       "      <td>pelvic_width</td>\n",
       "      <td>FBN1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>2</td>\n",
       "      <td>rs13003786</td>\n",
       "      <td>8.609000e-09</td>\n",
       "      <td>74</td>\n",
       "      <td>chr2:45588067..45842844</td>\n",
       "      <td>254.778</td>\n",
       "      <td>[SRBD1]</td>\n",
       "      <td>pelvic_width</td>\n",
       "      <td>SRBD1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>20</td>\n",
       "      <td>rs232659</td>\n",
       "      <td>4.070000e-08</td>\n",
       "      <td>6</td>\n",
       "      <td>chr20:7085280..7226483</td>\n",
       "      <td>141.204</td>\n",
       "      <td>[LOC101929265]</td>\n",
       "      <td>pelvic_width</td>\n",
       "      <td>NC_000020.11:g.7113491A&gt;C,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10</td>\n",
       "      <td>rs1932542</td>\n",
       "      <td>4.005000e-11</td>\n",
       "      <td>86</td>\n",
       "      <td>chr10:120116771..120215373</td>\n",
       "      <td>98.603</td>\n",
       "      <td>[LINC00867]</td>\n",
       "      <td>acetabular_inclination</td>\n",
       "      <td>NC_000010.11:g.118453277C&gt;T,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>9</td>\n",
       "      <td>rs4961732</td>\n",
       "      <td>3.856000e-12</td>\n",
       "      <td>248</td>\n",
       "      <td>chr9:16665850..16833127</td>\n",
       "      <td>167.278</td>\n",
       "      <td>[BNC2]</td>\n",
       "      <td>oblique_pelvic_inlet_length</td>\n",
       "      <td>BNC2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>12</td>\n",
       "      <td>rs6487691</td>\n",
       "      <td>1.236000e-08</td>\n",
       "      <td>124</td>\n",
       "      <td>chr12:28277338..28735596</td>\n",
       "      <td>458.259</td>\n",
       "      <td>[CCDC91]</td>\n",
       "      <td>oblique_pelvic_inlet_length</td>\n",
       "      <td>NC_000012.12:g.28571283T&gt;C,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>20</td>\n",
       "      <td>rs2145270</td>\n",
       "      <td>2.011000e-24</td>\n",
       "      <td>133</td>\n",
       "      <td>chr20:6474221..6635509</td>\n",
       "      <td>161.289</td>\n",
       "      <td>[CASC20]</td>\n",
       "      <td>pelvic_height</td>\n",
       "      <td>NC_000020.11:g.6641038C&gt;A,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>20</td>\n",
       "      <td>rs6066101</td>\n",
       "      <td>1.194000e-10</td>\n",
       "      <td>23</td>\n",
       "      <td>chr20:45486991..45571692</td>\n",
       "      <td>84.702</td>\n",
       "      <td>[EYA2]</td>\n",
       "      <td>pelvic_height</td>\n",
       "      <td>NC_000020.11:g.46891513A&gt;G,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>2</td>\n",
       "      <td>rs3103295</td>\n",
       "      <td>1.267000e-09</td>\n",
       "      <td>138</td>\n",
       "      <td>chr2:232797966..233226220</td>\n",
       "      <td>428.255</td>\n",
       "      <td>[DIS3L2]</td>\n",
       "      <td>pelvic_height</td>\n",
       "      <td>DIS3L2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>10</td>\n",
       "      <td>rs10776630</td>\n",
       "      <td>4.043000e-09</td>\n",
       "      <td>24</td>\n",
       "      <td>chr10:49818441..49856467</td>\n",
       "      <td>38.027</td>\n",
       "      <td>[ARHGAP22]</td>\n",
       "      <td>pelvic_height</td>\n",
       "      <td>ARHGAP22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>15</td>\n",
       "      <td>rs1062033</td>\n",
       "      <td>6.547000e-09</td>\n",
       "      <td>36</td>\n",
       "      <td>chr15:51544871..51564341</td>\n",
       "      <td>19.471</td>\n",
       "      <td>[CYP19A1]</td>\n",
       "      <td>pelvic_height</td>\n",
       "      <td>CYP19A1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>9</td>\n",
       "      <td>rs4246133</td>\n",
       "      <td>6.556000e-09</td>\n",
       "      <td>53</td>\n",
       "      <td>chr9:16862358..16915716</td>\n",
       "      <td>53.359</td>\n",
       "      <td>[BNC2]</td>\n",
       "      <td>pelvic_height</td>\n",
       "      <td>NC_000009.12:g.16912921A&gt;C,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>12</td>\n",
       "      <td>rs7960396</td>\n",
       "      <td>1.107000e-08</td>\n",
       "      <td>56</td>\n",
       "      <td>chr12:114708140..114830700</td>\n",
       "      <td>122.561</td>\n",
       "      <td>[TBX5]</td>\n",
       "      <td>pelvic_height</td>\n",
       "      <td>TBX5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>8</td>\n",
       "      <td>rs11786277</td>\n",
       "      <td>1.939000e-08</td>\n",
       "      <td>101</td>\n",
       "      <td>chr8:75855004..75995961</td>\n",
       "      <td>140.958</td>\n",
       "      <td>[CRISPLD1]</td>\n",
       "      <td>acetabular_diameter</td>\n",
       "      <td>NC_000008.11:g.74969957G&gt;A,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>20</td>\n",
       "      <td>rs2145272</td>\n",
       "      <td>8.750000e-15</td>\n",
       "      <td>75</td>\n",
       "      <td>chr20:6479737..6635509</td>\n",
       "      <td>155.773</td>\n",
       "      <td>[CASC20]</td>\n",
       "      <td>pelvic_inlet_width</td>\n",
       "      <td>NC_000020.11:g.6645571G&gt;A,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>11</td>\n",
       "      <td>rs7945719</td>\n",
       "      <td>3.846000e-13</td>\n",
       "      <td>174</td>\n",
       "      <td>chr11:12649610..12901978</td>\n",
       "      <td>252.369</td>\n",
       "      <td>[TEAD1]</td>\n",
       "      <td>pelvic_inlet_width</td>\n",
       "      <td>TEAD1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>12</td>\n",
       "      <td>rs7305790</td>\n",
       "      <td>4.447000e-08</td>\n",
       "      <td>125</td>\n",
       "      <td>chr12:28224353..28716044</td>\n",
       "      <td>491.692</td>\n",
       "      <td>[CCDC91]</td>\n",
       "      <td>pelvic_inlet_width</td>\n",
       "      <td>NC_000012.12:g.28136791A&gt;C,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>rs34580448</td>\n",
       "      <td>2.006000e-09</td>\n",
       "      <td>32</td>\n",
       "      <td>chr5:82724176..82888471</td>\n",
       "      <td>164.296</td>\n",
       "      <td>[VCAN]</td>\n",
       "      <td>iliac_flare_angle</td>\n",
       "      <td>VCAN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>1</td>\n",
       "      <td>rs11118411</td>\n",
       "      <td>1.456000e-08</td>\n",
       "      <td>223</td>\n",
       "      <td>chr1:219835486..220066014</td>\n",
       "      <td>230.529</td>\n",
       "      <td>[RNU5F-1]</td>\n",
       "      <td>iliac_flare_angle</td>\n",
       "      <td>LOC105372926</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>12</td>\n",
       "      <td>rs4359252</td>\n",
       "      <td>3.497000e-16</td>\n",
       "      <td>548</td>\n",
       "      <td>chr12:28241649..28737797</td>\n",
       "      <td>496.149</td>\n",
       "      <td>[CCDC91]</td>\n",
       "      <td>iliac_isthmus_breadth</td>\n",
       "      <td>CCDC91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>1</td>\n",
       "      <td>rs12756209</td>\n",
       "      <td>4.892000e-10</td>\n",
       "      <td>208</td>\n",
       "      <td>chr1:103113416..103754185</td>\n",
       "      <td>640.770</td>\n",
       "      <td>[COL11A1]</td>\n",
       "      <td>iliac_isthmus_breadth</td>\n",
       "      <td>COL11A1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>12</td>\n",
       "      <td>rs11049715</td>\n",
       "      <td>6.172000e-10</td>\n",
       "      <td>161</td>\n",
       "      <td>chr12:28194223..28739627</td>\n",
       "      <td>545.405</td>\n",
       "      <td>[CCDC91]</td>\n",
       "      <td>iliac_isthmus_breadth</td>\n",
       "      <td>NC_000012.12:g.28561520A&gt;G,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>20</td>\n",
       "      <td>rs232658</td>\n",
       "      <td>5.292000e-09</td>\n",
       "      <td>77</td>\n",
       "      <td>chr20:7085280..7175682</td>\n",
       "      <td>90.403</td>\n",
       "      <td>[LOC101929265]</td>\n",
       "      <td>subpubic_angle</td>\n",
       "      <td>NC_000020.11:g.7113723G&gt;A,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>13</td>\n",
       "      <td>rs3118919</td>\n",
       "      <td>1.876000e-10</td>\n",
       "      <td>125</td>\n",
       "      <td>chr13:50887164..51252064</td>\n",
       "      <td>364.901</td>\n",
       "      <td>[DLEU1]</td>\n",
       "      <td>trochanter_left2trochanter_right</td>\n",
       "      <td>LOC107984567</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>12</td>\n",
       "      <td>rs6487691</td>\n",
       "      <td>1.748000e-10</td>\n",
       "      <td>172</td>\n",
       "      <td>chr12:28241649..28737797</td>\n",
       "      <td>496.149</td>\n",
       "      <td>[CCDC91]</td>\n",
       "      <td>iliac_flare_ratio</td>\n",
       "      <td>NC_000012.12:g.28571283T&gt;C,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>7</td>\n",
       "      <td>rs6965195</td>\n",
       "      <td>2.675000e-09</td>\n",
       "      <td>235</td>\n",
       "      <td>chr7:120724335..120904727</td>\n",
       "      <td>180.393</td>\n",
       "      <td>[CPED1]</td>\n",
       "      <td>ear_left2ear_right</td>\n",
       "      <td>CPED1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    CHR         SNP             P    N                         POS       KB  \\\n",
       "2    20   rs1884902  1.833000e-09   50      chr20:6487687..6635509  147.823   \n",
       "4    15  rs28370880  7.197000e-09  126    chr15:48692609..48933821  241.213   \n",
       "5     2  rs13003786  8.609000e-09   74     chr2:45588067..45842844  254.778   \n",
       "6    20    rs232659  4.070000e-08    6      chr20:7085280..7226483  141.204   \n",
       "1    10   rs1932542  4.005000e-11   86  chr10:120116771..120215373   98.603   \n",
       "0     9   rs4961732  3.856000e-12  248     chr9:16665850..16833127  167.278   \n",
       "3    12   rs6487691  1.236000e-08  124    chr12:28277338..28735596  458.259   \n",
       "0    20   rs2145270  2.011000e-24  133      chr20:6474221..6635509  161.289   \n",
       "7    20   rs6066101  1.194000e-10   23    chr20:45486991..45571692   84.702   \n",
       "9     2   rs3103295  1.267000e-09  138   chr2:232797966..233226220  428.255   \n",
       "13   10  rs10776630  4.043000e-09   24    chr10:49818441..49856467   38.027   \n",
       "16   15   rs1062033  6.547000e-09   36    chr15:51544871..51564341   19.471   \n",
       "17    9   rs4246133  6.556000e-09   53     chr9:16862358..16915716   53.359   \n",
       "19   12   rs7960396  1.107000e-08   56  chr12:114708140..114830700  122.561   \n",
       "5     8  rs11786277  1.939000e-08  101     chr8:75855004..75995961  140.958   \n",
       "1    20   rs2145272  8.750000e-15   75      chr20:6479737..6635509  155.773   \n",
       "2    11   rs7945719  3.846000e-13  174    chr11:12649610..12901978  252.369   \n",
       "9    12   rs7305790  4.447000e-08  125    chr12:28224353..28716044  491.692   \n",
       "4     5  rs34580448  2.006000e-09   32     chr5:82724176..82888471  164.296   \n",
       "6     1  rs11118411  1.456000e-08  223   chr1:219835486..220066014  230.529   \n",
       "0    12   rs4359252  3.497000e-16  548    chr12:28241649..28737797  496.149   \n",
       "5     1  rs12756209  4.892000e-10  208   chr1:103113416..103754185  640.770   \n",
       "6    12  rs11049715  6.172000e-10  161    chr12:28194223..28739627  545.405   \n",
       "0    20    rs232658  5.292000e-09   77      chr20:7085280..7175682   90.403   \n",
       "3    13   rs3118919  1.876000e-10  125    chr13:50887164..51252064  364.901   \n",
       "3    12   rs6487691  1.748000e-10  172    chr12:28241649..28737797  496.149   \n",
       "2     7   rs6965195  2.675000e-09  235   chr7:120724335..120904727  180.393   \n",
       "\n",
       "            RANGES                             pheno  \\\n",
       "2         [CASC20]                      pelvic_width   \n",
       "4           [FBN1]                      pelvic_width   \n",
       "5          [SRBD1]                      pelvic_width   \n",
       "6   [LOC101929265]                      pelvic_width   \n",
       "1      [LINC00867]            acetabular_inclination   \n",
       "0           [BNC2]       oblique_pelvic_inlet_length   \n",
       "3         [CCDC91]       oblique_pelvic_inlet_length   \n",
       "0         [CASC20]                     pelvic_height   \n",
       "7           [EYA2]                     pelvic_height   \n",
       "9         [DIS3L2]                     pelvic_height   \n",
       "13      [ARHGAP22]                     pelvic_height   \n",
       "16       [CYP19A1]                     pelvic_height   \n",
       "17          [BNC2]                     pelvic_height   \n",
       "19          [TBX5]                     pelvic_height   \n",
       "5       [CRISPLD1]               acetabular_diameter   \n",
       "1         [CASC20]                pelvic_inlet_width   \n",
       "2          [TEAD1]                pelvic_inlet_width   \n",
       "9         [CCDC91]                pelvic_inlet_width   \n",
       "4           [VCAN]                 iliac_flare_angle   \n",
       "6        [RNU5F-1]                 iliac_flare_angle   \n",
       "0         [CCDC91]             iliac_isthmus_breadth   \n",
       "5        [COL11A1]             iliac_isthmus_breadth   \n",
       "6         [CCDC91]             iliac_isthmus_breadth   \n",
       "0   [LOC101929265]                    subpubic_angle   \n",
       "3          [DLEU1]  trochanter_left2trochanter_right   \n",
       "3         [CCDC91]                 iliac_flare_ratio   \n",
       "2          [CPED1]                ear_left2ear_right   \n",
       "\n",
       "                            gene  \n",
       "2     NC_000020.11:g.6627855C>T,  \n",
       "4                           FBN1  \n",
       "5                          SRBD1  \n",
       "6     NC_000020.11:g.7113491A>C,  \n",
       "1   NC_000010.11:g.118453277C>T,  \n",
       "0                           BNC2  \n",
       "3    NC_000012.12:g.28571283T>C,  \n",
       "0     NC_000020.11:g.6641038C>A,  \n",
       "7    NC_000020.11:g.46891513A>G,  \n",
       "9                         DIS3L2  \n",
       "13                      ARHGAP22  \n",
       "16                       CYP19A1  \n",
       "17   NC_000009.12:g.16912921A>C,  \n",
       "19                          TBX5  \n",
       "5    NC_000008.11:g.74969957G>A,  \n",
       "1     NC_000020.11:g.6645571G>A,  \n",
       "2                          TEAD1  \n",
       "9    NC_000012.12:g.28136791A>C,  \n",
       "4                           VCAN  \n",
       "6                   LOC105372926  \n",
       "0                         CCDC91  \n",
       "5                        COL11A1  \n",
       "6    NC_000012.12:g.28561520A>G,  \n",
       "0     NC_000020.11:g.7113723G>A,  \n",
       "3                   LOC107984567  \n",
       "3    NC_000012.12:g.28571283T>C,  \n",
       "2                          CPED1  "
      ]
     },
     "execution_count": 147,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clump_male_one_gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{}"
      ]
     },
     "execution_count": 151,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "{\n",
    "    \"CASC20\": \"heterotopic ossification\",\n",
    "    \"FBN1\": \"Marfan syndrome\",\n",
    "    \"SRBD1\": \"Glaucoma\",\n",
    "    \"BNC2\": \"adolescent idiopathic scoliosis\",\n",
    "    \"CCDC91\": \"ossification of the posterior longitudinal ligament of the spine\",\n",
    "    \"EYA2\": \"hypaxial somitic myogenesis in the mouse embryo\",\n",
    "    \"DIS3L2\": \"perlman syndrome\",\n",
    "    \"ARHGAP22\": \"type 2 diabetic retinopathy\",\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>SNP</th>\n",
       "      <th>P</th>\n",
       "      <th>N</th>\n",
       "      <th>POS</th>\n",
       "      <th>KB</th>\n",
       "      <th>RANGES</th>\n",
       "      <th>pheno</th>\n",
       "      <th>gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>6</td>\n",
       "      <td>rs6459105</td>\n",
       "      <td>3.021000e-12</td>\n",
       "      <td>36</td>\n",
       "      <td>chr6:55476272..55642001</td>\n",
       "      <td>165.730</td>\n",
       "      <td>[BMP5]</td>\n",
       "      <td>acetabular_inclination</td>\n",
       "      <td>BMP5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>6</td>\n",
       "      <td>rs4714858</td>\n",
       "      <td>1.981000e-08</td>\n",
       "      <td>65</td>\n",
       "      <td>chr6:45407654..45510107</td>\n",
       "      <td>102.454</td>\n",
       "      <td>[RUNX2]</td>\n",
       "      <td>acetabular_inclination</td>\n",
       "      <td>RUNX2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>9</td>\n",
       "      <td>rs1337399</td>\n",
       "      <td>4.996000e-15</td>\n",
       "      <td>268</td>\n",
       "      <td>chr9:16661820..16833127</td>\n",
       "      <td>171.308</td>\n",
       "      <td>[BNC2]</td>\n",
       "      <td>oblique_pelvic_inlet_length</td>\n",
       "      <td>BNC2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>rs3116187</td>\n",
       "      <td>1.523000e-12</td>\n",
       "      <td>228</td>\n",
       "      <td>chr2:232805666..233226220</td>\n",
       "      <td>420.555</td>\n",
       "      <td>[DIS3L2]</td>\n",
       "      <td>oblique_pelvic_inlet_length</td>\n",
       "      <td>DIS3L2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>9</td>\n",
       "      <td>rs5015536</td>\n",
       "      <td>3.000000e-08</td>\n",
       "      <td>45</td>\n",
       "      <td>chr9:16755509..16800557</td>\n",
       "      <td>45.049</td>\n",
       "      <td>[BNC2]</td>\n",
       "      <td>oblique_pelvic_inlet_length</td>\n",
       "      <td>BNC2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>rs13386977</td>\n",
       "      <td>7.212000e-18</td>\n",
       "      <td>360</td>\n",
       "      <td>chr2:232797966..233210465</td>\n",
       "      <td>412.500</td>\n",
       "      <td>[DIS3L2]</td>\n",
       "      <td>pelvic_height</td>\n",
       "      <td>DIS3L2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>15</td>\n",
       "      <td>rs17361098</td>\n",
       "      <td>1.534000e-08</td>\n",
       "      <td>124</td>\n",
       "      <td>chr15:48692609..48933821</td>\n",
       "      <td>241.213</td>\n",
       "      <td>[FBN1]</td>\n",
       "      <td>pelvic_height</td>\n",
       "      <td>FBN1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>12</td>\n",
       "      <td>rs7964793</td>\n",
       "      <td>2.878000e-08</td>\n",
       "      <td>521</td>\n",
       "      <td>chr12:28264703..28739627</td>\n",
       "      <td>474.925</td>\n",
       "      <td>[CCDC91]</td>\n",
       "      <td>pelvic_height</td>\n",
       "      <td>CCDC91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>8</td>\n",
       "      <td>rs960856</td>\n",
       "      <td>3.900000e-09</td>\n",
       "      <td>124</td>\n",
       "      <td>chr8:75847989..75995961</td>\n",
       "      <td>147.973</td>\n",
       "      <td>[CRISPLD1]</td>\n",
       "      <td>acetabular_diameter</td>\n",
       "      <td>CRISPLD1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>8</td>\n",
       "      <td>rs7839059</td>\n",
       "      <td>2.164000e-12</td>\n",
       "      <td>199</td>\n",
       "      <td>chr8:119887127..120063542</td>\n",
       "      <td>176.416</td>\n",
       "      <td>[TNFRSF11B]</td>\n",
       "      <td>pelvic_inlet_width</td>\n",
       "      <td>COLEC10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14</td>\n",
       "      <td>rs2074562</td>\n",
       "      <td>5.279000e-09</td>\n",
       "      <td>20</td>\n",
       "      <td>chr14:68732422..68959711</td>\n",
       "      <td>227.290</td>\n",
       "      <td>[RAD51B]</td>\n",
       "      <td>pelvic_inlet_width</td>\n",
       "      <td>RAD51B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>11</td>\n",
       "      <td>rs9888258</td>\n",
       "      <td>1.748000e-08</td>\n",
       "      <td>140</td>\n",
       "      <td>chr11:12658440..12903077</td>\n",
       "      <td>244.638</td>\n",
       "      <td>[TEAD1]</td>\n",
       "      <td>pelvic_inlet_width</td>\n",
       "      <td>TEAD1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>8</td>\n",
       "      <td>rs6470771</td>\n",
       "      <td>2.361000e-09</td>\n",
       "      <td>91</td>\n",
       "      <td>chr8:130717716..130763236</td>\n",
       "      <td>45.521</td>\n",
       "      <td>[GSDMC]</td>\n",
       "      <td>iliac_isthmus_breadth</td>\n",
       "      <td>GSDMC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>12</td>\n",
       "      <td>rs11049418</td>\n",
       "      <td>2.521000e-08</td>\n",
       "      <td>565</td>\n",
       "      <td>chr12:28247221..28739627</td>\n",
       "      <td>492.407</td>\n",
       "      <td>[CCDC91]</td>\n",
       "      <td>iliac_isthmus_breadth</td>\n",
       "      <td>CCDC91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>rs12475479</td>\n",
       "      <td>2.235000e-09</td>\n",
       "      <td>191</td>\n",
       "      <td>chr2:232809066..233161818</td>\n",
       "      <td>352.753</td>\n",
       "      <td>[DIS3L2]</td>\n",
       "      <td>subpubic_angle</td>\n",
       "      <td>DIS3L2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>20</td>\n",
       "      <td>rs6047788</td>\n",
       "      <td>1.903000e-08</td>\n",
       "      <td>373</td>\n",
       "      <td>chr20:21815192..22192446</td>\n",
       "      <td>377.255</td>\n",
       "      <td>[LOC100270679]</td>\n",
       "      <td>subpubic_angle</td>\n",
       "      <td>NC_000020.11:g.21996668A&gt;C,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>6</td>\n",
       "      <td>rs12204049</td>\n",
       "      <td>2.615000e-08</td>\n",
       "      <td>73</td>\n",
       "      <td>chr6:45407654..45510107</td>\n",
       "      <td>102.454</td>\n",
       "      <td>[RUNX2]</td>\n",
       "      <td>trochanter_left2trochanter_right</td>\n",
       "      <td>RUNX2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    CHR         SNP             P    N                        POS       KB  \\\n",
       "3     6   rs6459105  3.021000e-12   36    chr6:55476272..55642001  165.730   \n",
       "5     6   rs4714858  1.981000e-08   65    chr6:45407654..45510107  102.454   \n",
       "0     9   rs1337399  4.996000e-15  268    chr9:16661820..16833127  171.308   \n",
       "1     2   rs3116187  1.523000e-12  228  chr2:232805666..233226220  420.555   \n",
       "3     9   rs5015536  3.000000e-08   45    chr9:16755509..16800557   45.049   \n",
       "1     2  rs13386977  7.212000e-18  360  chr2:232797966..233210465  412.500   \n",
       "13   15  rs17361098  1.534000e-08  124   chr15:48692609..48933821  241.213   \n",
       "14   12   rs7964793  2.878000e-08  521   chr12:28264703..28739627  474.925   \n",
       "4     8    rs960856  3.900000e-09  124    chr8:75847989..75995961  147.973   \n",
       "0     8   rs7839059  2.164000e-12  199  chr8:119887127..120063542  176.416   \n",
       "2    14   rs2074562  5.279000e-09   20   chr14:68732422..68959711  227.290   \n",
       "5    11   rs9888258  1.748000e-08  140   chr11:12658440..12903077  244.638   \n",
       "1     8   rs6470771  2.361000e-09   91  chr8:130717716..130763236   45.521   \n",
       "3    12  rs11049418  2.521000e-08  565   chr12:28247221..28739627  492.407   \n",
       "2     2  rs12475479  2.235000e-09  191  chr2:232809066..233161818  352.753   \n",
       "6    20   rs6047788  1.903000e-08  373   chr20:21815192..22192446  377.255   \n",
       "3     6  rs12204049  2.615000e-08   73    chr6:45407654..45510107  102.454   \n",
       "\n",
       "            RANGES                             pheno  \\\n",
       "3           [BMP5]            acetabular_inclination   \n",
       "5          [RUNX2]            acetabular_inclination   \n",
       "0           [BNC2]       oblique_pelvic_inlet_length   \n",
       "1         [DIS3L2]       oblique_pelvic_inlet_length   \n",
       "3           [BNC2]       oblique_pelvic_inlet_length   \n",
       "1         [DIS3L2]                     pelvic_height   \n",
       "13          [FBN1]                     pelvic_height   \n",
       "14        [CCDC91]                     pelvic_height   \n",
       "4       [CRISPLD1]               acetabular_diameter   \n",
       "0      [TNFRSF11B]                pelvic_inlet_width   \n",
       "2         [RAD51B]                pelvic_inlet_width   \n",
       "5          [TEAD1]                pelvic_inlet_width   \n",
       "1          [GSDMC]             iliac_isthmus_breadth   \n",
       "3         [CCDC91]             iliac_isthmus_breadth   \n",
       "2         [DIS3L2]                    subpubic_angle   \n",
       "6   [LOC100270679]                    subpubic_angle   \n",
       "3          [RUNX2]  trochanter_left2trochanter_right   \n",
       "\n",
       "                           gene  \n",
       "3                          BMP5  \n",
       "5                         RUNX2  \n",
       "0                          BNC2  \n",
       "1                        DIS3L2  \n",
       "3                          BNC2  \n",
       "1                        DIS3L2  \n",
       "13                         FBN1  \n",
       "14                       CCDC91  \n",
       "4                      CRISPLD1  \n",
       "0                       COLEC10  \n",
       "2                        RAD51B  \n",
       "5                         TEAD1  \n",
       "1                         GSDMC  \n",
       "3                        CCDC91  \n",
       "2                        DIS3L2  \n",
       "6   NC_000020.11:g.21996668A>C,  \n",
       "3                         RUNX2  "
      ]
     },
     "execution_count": 148,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clump_female_one_gene"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Genetic correlation"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Cross phenotypes rg"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "!mkdir -p key_results/genetic_correlation/rg_20230909"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/7f/jzt97g754236ldjj13l5pgd00000gn/T/ipykernel_20099/2294302090.py:55: RuntimeWarning: divide by zero encountered in log10\n",
      "  neglog10p = -np.log10(p_value)\n",
      "/var/folders/7f/jzt97g754236ldjj13l5pgd00000gn/T/ipykernel_20099/2294302090.py:120: RuntimeWarning: divide by zero encountered in log10\n",
      "  neglog10p = -np.log10(p_value)\n",
      "/var/folders/7f/jzt97g754236ldjj13l5pgd00000gn/T/ipykernel_20099/2294302090.py:185: RuntimeWarning: divide by zero encountered in log10\n",
      "  neglog10p = -np.log10(p_value)\n"
     ]
    }
   ],
   "source": [
    "# both rg\n",
    "rg_df_both = pd.DataFrame(columns=['pheno1', 'pheno1_h2g', 'pheno1_h2g_se', 'pheno1_lambda',\n",
    "                                   'pheno2', 'pheno2_h2g', 'pheno2_h2g_se', 'pheno2_lambda',\n",
    "                                   'cor', 'se', 'z', 'p_value'])\n",
    "path = \"UKB_Imaging_Genetics/LDSC_OUTPUT/rg_hip_both_20230909\"\n",
    "for file in os.listdir(path):\n",
    "    if file.endswith(\".log\"):\n",
    "        # get phenotype name\n",
    "        pheno1, pheno2 = file.split('-')\n",
    "        pheno1 = pheno1.split('.')[0]\n",
    "        pheno2 = pheno2.split('.')[0]\n",
    "\n",
    "        with open(os.path.join(path, file), \"r\") as f:\n",
    "            lines = f.readlines()\n",
    "\n",
    "        for i, line in enumerate(lines):\n",
    "            # get h2g\n",
    "            if line.startswith(\"Heritability of phenotype 1\") and i + 2 < len(lines):\n",
    "                h2g_1 = lines[i + 2].split(\" \")[4].strip()\n",
    "                h2g_1 = float(h2g_1)\n",
    "\n",
    "                h2g_1_se = lines[i + 2].split(\" \")[5].strip()\n",
    "                h2g_1_se = h2g_1_se[1:-1]\n",
    "                h2g_1_se = float(h2g_1_se)\n",
    "\n",
    "                pheno1_lambda = lines[i + 3].split(\" \")[2].strip()\n",
    "                pheno1_lambda = float(pheno1_lambda)\n",
    "            \n",
    "            if line.startswith(\"Heritability of phenotype 2/2\") and i + 2 < len(lines):\n",
    "                h2g_2 = lines[i + 2].split(\" \")[4].strip()\n",
    "                h2g_2 = float(h2g_2)\n",
    "                \n",
    "                h2g_2_se = lines[i + 2].split(\" \")[5].strip()\n",
    "                h2g_2_se = h2g_2_se[1:-1]\n",
    "                h2g_2_se = float(h2g_2_se)\n",
    "\n",
    "                pheno2_lambda = lines[i + 3].split(\" \")[2].strip()\n",
    "                pheno2_lambda = float(pheno1_lambda)\n",
    "\n",
    "            # get rg\n",
    "            if line.startswith(\"Genetic Correlation:\"):\n",
    "                # print(line)\n",
    "                rg = line.split(\" \")[2].strip()\n",
    "                rg = float(rg)\n",
    "                # print(rg)\n",
    "                se = line.split(\" \")[3].strip()\n",
    "                se = se[1:-1]\n",
    "                se = float(se)\n",
    "            if line.startswith(\"Z-score:\"):\n",
    "                z = line.split(\" \")[1].strip()\n",
    "                z = float(z)\n",
    "            if line.startswith(\"P:\"):\n",
    "                p_value = line.split(\" \")[1].strip()\n",
    "                p_value = float(p_value)\n",
    "                neglog10p = -np.log10(p_value)\n",
    "\n",
    "        rg_df_both = rg_df_both.append({'pheno1': pheno1, 'pheno1_h2g': h2g_1, 'pheno1_h2g_se': h2g_1_se, 'pheno1_lambda': pheno1_lambda,\n",
    "                                        'pheno2': pheno2, 'pheno2_h2g': h2g_2, 'pheno2_h2g_se': h2g_2_se, 'pheno2_lambda': pheno2_lambda,\n",
    "                                        'cor': rg, 'se': se, 'z': z, 'p_value': p_value, 'neglog10p': neglog10p}, \n",
    "                                        ignore_index=True)\n",
    "\n",
    "rg_df_both.sort_values(by = ['pheno1', 'pheno2'], inplace=True)\n",
    "rg_df_both['gender'] = 'both'\n",
    "rg_df_both.to_csv(\"key_results/genetic_correlation/rg_20230909/rg_hip_both.csv\", index=False)\n",
    "\n",
    "# female rg\n",
    "rg_df_female = pd.DataFrame(columns=['pheno1', 'pheno1_h2g', 'pheno1_h2g_se', 'pheno1_lambda',\n",
    "                                   'pheno2', 'pheno2_h2g', 'pheno2_h2g_se', 'pheno2_lambda',\n",
    "                                   'cor', 'se', 'z', 'p_value'])\n",
    "path = \"UKB_Imaging_Genetics/LDSC_OUTPUT/rg_hip_female_20230909\"\n",
    "for file in os.listdir(path):\n",
    "    if file.endswith(\".log\"):\n",
    "        # get phenotype name\n",
    "        pheno1, pheno2 = file.split('-')\n",
    "        pheno1 = pheno1.split('.')[0]\n",
    "        pheno2 = pheno2.split('.')[0]\n",
    "\n",
    "        with open(os.path.join(path, file), \"r\") as f:\n",
    "            lines = f.readlines()\n",
    "\n",
    "        for i, line in enumerate(lines):\n",
    "            # get h2g\n",
    "            if line.startswith(\"Heritability of phenotype 1\") and i + 2 < len(lines):\n",
    "                h2g_1 = lines[i + 2].split(\" \")[4].strip()\n",
    "                h2g_1 = float(h2g_1)\n",
    "\n",
    "                h2g_1_se = lines[i + 2].split(\" \")[5].strip()\n",
    "                h2g_1_se = h2g_1_se[1:-1]\n",
    "                h2g_1_se = float(h2g_1_se)\n",
    "\n",
    "                pheno1_lambda = lines[i + 3].split(\" \")[2].strip()\n",
    "                pheno1_lambda = float(pheno1_lambda)\n",
    "            \n",
    "            if line.startswith(\"Heritability of phenotype 2/2\") and i + 2 < len(lines):\n",
    "                h2g_2 = lines[i + 2].split(\" \")[4].strip()\n",
    "                h2g_2 = float(h2g_2)\n",
    "                \n",
    "                h2g_2_se = lines[i + 2].split(\" \")[5].strip()\n",
    "                h2g_2_se = h2g_2_se[1:-1]\n",
    "                h2g_2_se = float(h2g_2_se)\n",
    "\n",
    "                pheno2_lambda = lines[i + 3].split(\" \")[2].strip()\n",
    "                pheno2_lambda = float(pheno1_lambda)\n",
    "\n",
    "            # get rg\n",
    "            if line.startswith(\"Genetic Correlation:\"):\n",
    "                # print(line)\n",
    "                rg = line.split(\" \")[2].strip()\n",
    "                rg = float(rg)\n",
    "                # print(rg)\n",
    "                se = line.split(\" \")[3].strip()\n",
    "                se = se[1:-1]\n",
    "                se = float(se)\n",
    "            if line.startswith(\"Z-score:\"):\n",
    "                z = line.split(\" \")[1].strip()\n",
    "                z = float(z)\n",
    "            if line.startswith(\"P:\"):\n",
    "                p_value = line.split(\" \")[1].strip()\n",
    "                p_value = float(p_value)\n",
    "                neglog10p = -np.log10(p_value)\n",
    "\n",
    "        rg_df_female = rg_df_female.append({'pheno1': pheno1, 'pheno1_h2g': h2g_1, 'pheno1_h2g_se': h2g_1_se, 'pheno1_lambda': pheno1_lambda,\n",
    "                                          'pheno2': pheno2, 'pheno2_h2g': h2g_2, 'pheno2_h2g_se': h2g_2_se, 'pheno2_lambda': pheno2_lambda,\n",
    "                                          'cor': rg, 'se': se, 'z': z, 'p_value': p_value, 'neglog10p': neglog10p}, \n",
    "                                          ignore_index=True)\n",
    "\n",
    "rg_df_female.sort_values(by = ['pheno1', 'pheno2'], inplace=True)\n",
    "rg_df_female['gender'] = 'female'\n",
    "rg_df_female.to_csv(\"key_results/genetic_correlation/rg_20230909/rg_hip_female.csv\", index=False)\n",
    "\n",
    "# male rg\n",
    "rg_df_male = pd.DataFrame(columns=['pheno1', 'pheno1_h2g', 'pheno1_h2g_se', 'pheno1_lambda',\n",
    "                                   'pheno2', 'pheno2_h2g', 'pheno2_h2g_se', 'pheno2_lambda',\n",
    "                                   'cor', 'se', 'z', 'p_value'])\n",
    "path = \"UKB_Imaging_Genetics/LDSC_OUTPUT/rg_hip_male_20230909\"\n",
    "for file in os.listdir(path):\n",
    "    if file.endswith(\".log\"):\n",
    "        # get phenotype name\n",
    "        pheno1, pheno2 = file.split('-')\n",
    "        pheno1 = pheno1.split('.')[0]\n",
    "        pheno2 = pheno2.split('.')[0]\n",
    "\n",
    "        with open(os.path.join(path, file), \"r\") as f:\n",
    "            lines = f.readlines()\n",
    "\n",
    "        for i, line in enumerate(lines):\n",
    "            # get h2g\n",
    "            if line.startswith(\"Heritability of phenotype 1\") and i + 2 < len(lines):\n",
    "                h2g_1 = lines[i + 2].split(\" \")[4].strip()\n",
    "                h2g_1 = float(h2g_1)\n",
    "\n",
    "                h2g_1_se = lines[i + 2].split(\" \")[5].strip()\n",
    "                h2g_1_se = h2g_1_se[1:-1]\n",
    "                h2g_1_se = float(h2g_1_se)\n",
    "\n",
    "                pheno1_lambda = lines[i + 3].split(\" \")[2].strip()\n",
    "                pheno1_lambda = float(pheno1_lambda)\n",
    "            \n",
    "            if line.startswith(\"Heritability of phenotype 2/2\") and i + 2 < len(lines):\n",
    "                h2g_2 = lines[i + 2].split(\" \")[4].strip()\n",
    "                h2g_2 = float(h2g_2)\n",
    "                \n",
    "                h2g_2_se = lines[i + 2].split(\" \")[5].strip()\n",
    "                h2g_2_se = h2g_2_se[1:-1]\n",
    "                h2g_2_se = float(h2g_2_se)\n",
    "\n",
    "                pheno2_lambda = lines[i + 3].split(\" \")[2].strip()\n",
    "                pheno2_lambda = float(pheno1_lambda)\n",
    "\n",
    "            # get rg\n",
    "            if line.startswith(\"Genetic Correlation:\"):\n",
    "                # print(line)\n",
    "                rg = line.split(\" \")[2].strip()\n",
    "                rg = float(rg)\n",
    "                # print(rg)\n",
    "                se = line.split(\" \")[3].strip()\n",
    "                se = se[1:-1]\n",
    "                se = float(se)\n",
    "            if line.startswith(\"Z-score:\"):\n",
    "                z = line.split(\" \")[1].strip()\n",
    "                z = float(z)\n",
    "            if line.startswith(\"P:\"):\n",
    "                p_value = line.split(\" \")[1].strip()\n",
    "                p_value = float(p_value)\n",
    "                neglog10p = -np.log10(p_value)\n",
    "\n",
    "        rg_df_male = rg_df_male.append({'pheno1': pheno1, 'pheno1_h2g': h2g_1, 'pheno1_h2g_se': h2g_1_se, 'pheno1_lambda': pheno1_lambda,\n",
    "                                          'pheno2': pheno2, 'pheno2_h2g': h2g_2, 'pheno2_h2g_se': h2g_2_se, 'pheno2_lambda': pheno2_lambda,\n",
    "                                          'cor': rg, 'se': se, 'z': z, 'p_value': p_value, 'neglog10p': neglog10p}, \n",
    "                                          ignore_index=True)\n",
    "\n",
    "rg_df_male.sort_values(by = ['pheno1', 'pheno2'], inplace=True)\n",
    "rg_df_male['gender'] = 'male'\n",
    "rg_df_male.to_csv(\"key_results/genetic_correlation/rg_20230909/rg_hip_male.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pheno1</th>\n",
       "      <th>pheno1_h2g</th>\n",
       "      <th>pheno1_h2g_se</th>\n",
       "      <th>pheno1_lambda</th>\n",
       "      <th>pheno2</th>\n",
       "      <th>pheno2_h2g</th>\n",
       "      <th>pheno2_h2g_se</th>\n",
       "      <th>pheno2_lambda</th>\n",
       "      <th>cor</th>\n",
       "      <th>se</th>\n",
       "      <th>z</th>\n",
       "      <th>p_value</th>\n",
       "      <th>neglog10p</th>\n",
       "      <th>gender</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>acetabular_diameter</td>\n",
       "      <td>0.3180</td>\n",
       "      <td>0.0420</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>acetabular_inclination</td>\n",
       "      <td>0.3033</td>\n",
       "      <td>0.0390</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>-0.4099</td>\n",
       "      <td>0.0774</td>\n",
       "      <td>-5.2948</td>\n",
       "      <td>1.191500e-07</td>\n",
       "      <td>6.923906</td>\n",
       "      <td>male</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>93</th>\n",
       "      <td>acetabular_diameter</td>\n",
       "      <td>0.3180</td>\n",
       "      <td>0.0420</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>arm_devide_torso</td>\n",
       "      <td>0.3869</td>\n",
       "      <td>0.0412</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>-0.0195</td>\n",
       "      <td>0.0742</td>\n",
       "      <td>-0.2631</td>\n",
       "      <td>7.925000e-01</td>\n",
       "      <td>0.101001</td>\n",
       "      <td>male</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>202</th>\n",
       "      <td>acetabular_diameter</td>\n",
       "      <td>0.3180</td>\n",
       "      <td>0.0420</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>ear_left2ear_right</td>\n",
       "      <td>0.2993</td>\n",
       "      <td>0.0390</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>0.0019</td>\n",
       "      <td>0.0781</td>\n",
       "      <td>0.0244</td>\n",
       "      <td>9.805000e-01</td>\n",
       "      <td>0.008552</td>\n",
       "      <td>male</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162</th>\n",
       "      <td>acetabular_diameter</td>\n",
       "      <td>0.3180</td>\n",
       "      <td>0.0420</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>head_area_divide_pelvic_inlet_area</td>\n",
       "      <td>0.2877</td>\n",
       "      <td>0.0423</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>0.0427</td>\n",
       "      <td>0.0930</td>\n",
       "      <td>0.4590</td>\n",
       "      <td>6.462000e-01</td>\n",
       "      <td>0.189633</td>\n",
       "      <td>male</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>acetabular_diameter</td>\n",
       "      <td>0.3180</td>\n",
       "      <td>0.0420</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>head_divide_inlet_width</td>\n",
       "      <td>0.3588</td>\n",
       "      <td>0.0403</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>-0.0136</td>\n",
       "      <td>0.0744</td>\n",
       "      <td>-0.1823</td>\n",
       "      <td>8.554000e-01</td>\n",
       "      <td>0.067831</td>\n",
       "      <td>male</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>175</th>\n",
       "      <td>shoulder_divide_oblique_inlet_length</td>\n",
       "      <td>0.2687</td>\n",
       "      <td>0.0431</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>subpubic_angle</td>\n",
       "      <td>0.3472</td>\n",
       "      <td>0.0394</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>-0.5050</td>\n",
       "      <td>0.0670</td>\n",
       "      <td>-7.5421</td>\n",
       "      <td>4.624200e-14</td>\n",
       "      <td>13.334963</td>\n",
       "      <td>male</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>shoulder_divide_oblique_inlet_length</td>\n",
       "      <td>0.2687</td>\n",
       "      <td>0.0431</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>trochanter_left2trochanter_right</td>\n",
       "      <td>0.3494</td>\n",
       "      <td>0.0474</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>-0.0041</td>\n",
       "      <td>0.1113</td>\n",
       "      <td>-0.0368</td>\n",
       "      <td>9.707000e-01</td>\n",
       "      <td>0.012915</td>\n",
       "      <td>male</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>166</th>\n",
       "      <td>shoulder_width</td>\n",
       "      <td>0.3064</td>\n",
       "      <td>0.0394</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>subpubic_angle</td>\n",
       "      <td>0.3472</td>\n",
       "      <td>0.0394</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>-0.0829</td>\n",
       "      <td>0.0762</td>\n",
       "      <td>-1.0875</td>\n",
       "      <td>2.768000e-01</td>\n",
       "      <td>0.557834</td>\n",
       "      <td>male</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140</th>\n",
       "      <td>shoulder_width</td>\n",
       "      <td>0.3064</td>\n",
       "      <td>0.0394</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>trochanter_left2trochanter_right</td>\n",
       "      <td>0.3494</td>\n",
       "      <td>0.0474</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>0.1003</td>\n",
       "      <td>0.0770</td>\n",
       "      <td>1.3023</td>\n",
       "      <td>1.928000e-01</td>\n",
       "      <td>0.714893</td>\n",
       "      <td>male</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117</th>\n",
       "      <td>subpubic_angle</td>\n",
       "      <td>0.3472</td>\n",
       "      <td>0.0394</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>trochanter_left2trochanter_right</td>\n",
       "      <td>0.3494</td>\n",
       "      <td>0.0474</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>-0.0214</td>\n",
       "      <td>0.0826</td>\n",
       "      <td>-0.2586</td>\n",
       "      <td>7.959000e-01</td>\n",
       "      <td>0.099141</td>\n",
       "      <td>male</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>210 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   pheno1  pheno1_h2g  pheno1_h2g_se  \\\n",
       "36                    acetabular_diameter      0.3180         0.0420   \n",
       "93                    acetabular_diameter      0.3180         0.0420   \n",
       "202                   acetabular_diameter      0.3180         0.0420   \n",
       "162                   acetabular_diameter      0.3180         0.0420   \n",
       "53                    acetabular_diameter      0.3180         0.0420   \n",
       "..                                    ...         ...            ...   \n",
       "175  shoulder_divide_oblique_inlet_length      0.2687         0.0431   \n",
       "44   shoulder_divide_oblique_inlet_length      0.2687         0.0431   \n",
       "166                        shoulder_width      0.3064         0.0394   \n",
       "140                        shoulder_width      0.3064         0.0394   \n",
       "117                        subpubic_angle      0.3472         0.0394   \n",
       "\n",
       "     pheno1_lambda                              pheno2  pheno2_h2g  \\\n",
       "36          1.0957              acetabular_inclination      0.3033   \n",
       "93          1.0957                    arm_devide_torso      0.3869   \n",
       "202         1.0957                  ear_left2ear_right      0.2993   \n",
       "162         1.0957  head_area_divide_pelvic_inlet_area      0.2877   \n",
       "53          1.0957             head_divide_inlet_width      0.3588   \n",
       "..             ...                                 ...         ...   \n",
       "175         1.0957                      subpubic_angle      0.3472   \n",
       "44          1.0957    trochanter_left2trochanter_right      0.3494   \n",
       "166         1.0957                      subpubic_angle      0.3472   \n",
       "140         1.0957    trochanter_left2trochanter_right      0.3494   \n",
       "117         1.0957    trochanter_left2trochanter_right      0.3494   \n",
       "\n",
       "     pheno2_h2g_se  pheno2_lambda     cor      se       z       p_value  \\\n",
       "36          0.0390         1.0957 -0.4099  0.0774 -5.2948  1.191500e-07   \n",
       "93          0.0412         1.0957 -0.0195  0.0742 -0.2631  7.925000e-01   \n",
       "202         0.0390         1.0957  0.0019  0.0781  0.0244  9.805000e-01   \n",
       "162         0.0423         1.0957  0.0427  0.0930  0.4590  6.462000e-01   \n",
       "53          0.0403         1.0957 -0.0136  0.0744 -0.1823  8.554000e-01   \n",
       "..             ...            ...     ...     ...     ...           ...   \n",
       "175         0.0394         1.0957 -0.5050  0.0670 -7.5421  4.624200e-14   \n",
       "44          0.0474         1.0957 -0.0041  0.1113 -0.0368  9.707000e-01   \n",
       "166         0.0394         1.0957 -0.0829  0.0762 -1.0875  2.768000e-01   \n",
       "140         0.0474         1.0957  0.1003  0.0770  1.3023  1.928000e-01   \n",
       "117         0.0474         1.0957 -0.0214  0.0826 -0.2586  7.959000e-01   \n",
       "\n",
       "     neglog10p gender  \n",
       "36    6.923906   male  \n",
       "93    0.101001   male  \n",
       "202   0.008552   male  \n",
       "162   0.189633   male  \n",
       "53    0.067831   male  \n",
       "..         ...    ...  \n",
       "175  13.334963   male  \n",
       "44    0.012915   male  \n",
       "166   0.557834   male  \n",
       "140   0.714893   male  \n",
       "117   0.099141   male  \n",
       "\n",
       "[210 rows x 14 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rg_df_male"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Prepare for both rg plot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "acetabular_diameter                       20\n",
       "acetabular_inclination                    19\n",
       "arm_devide_torso                          18\n",
       "ear_left2ear_right                        17\n",
       "head_area_divide_pelvic_inlet_area        16\n",
       "head_divide_inlet_width                   15\n",
       "head_divide_oblique_inlet_length          14\n",
       "iliac_flare_angle                         13\n",
       "iliac_flare_ratio                         12\n",
       "iliac_isthmus_breadth                     11\n",
       "oblique_pelvic_inlet_length               10\n",
       "pelvic_height                              9\n",
       "pelvic_inlet_area                          8\n",
       "pelvic_inlet_width                         7\n",
       "pelvic_width                               6\n",
       "shoulder_area_divide_pelvic_inlet_area     5\n",
       "shoulder_divide_inlet_width                4\n",
       "shoulder_divide_oblique_inlet_length       3\n",
       "shoulder_width                             2\n",
       "subpubic_angle                             1\n",
       "Name: pheno1, dtype: int64"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rg_df_both['pheno1'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "trochanter_left2trochanter_right          20\n",
       "subpubic_angle                            19\n",
       "shoulder_width                            18\n",
       "shoulder_divide_oblique_inlet_length      17\n",
       "shoulder_divide_inlet_width               16\n",
       "shoulder_area_divide_pelvic_inlet_area    15\n",
       "pelvic_width                              14\n",
       "pelvic_inlet_width                        13\n",
       "pelvic_inlet_area                         12\n",
       "pelvic_height                             11\n",
       "oblique_pelvic_inlet_length               10\n",
       "iliac_isthmus_breadth                      9\n",
       "iliac_flare_ratio                          8\n",
       "iliac_flare_angle                          7\n",
       "head_divide_oblique_inlet_length           6\n",
       "head_divide_inlet_width                    5\n",
       "head_area_divide_pelvic_inlet_area         4\n",
       "ear_left2ear_right                         3\n",
       "arm_devide_torso                           2\n",
       "acetabular_inclination                     1\n",
       "Name: pheno2, dtype: int64"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rg_df_both['pheno2'].value_counts()"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Male vs Female rg"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [],
   "source": [
    "# male vs female rg\n",
    "rg_df_fvsm = pd.DataFrame(columns=['pheno_f', 'pheno_f_h2g', 'pheno_f_h2g_se', 'pheno_f_lambda',\n",
    "                                   'pheno_m', 'pheno_m_h2g', 'pheno_m_h2g_se', 'pheno_m_lambda',\n",
    "                                   'cor', 'se', 'z', 'p_value'])\n",
    "# path = \"UKB_Imaging_Genetics/LDSC_OUTPUT/rg_hip_female_vs_male_20230909\"\n",
    "path = \"UKB_Imaging_Genetics/LDSC_OUTPUT/rg_hip_female_vs_male_20240111\"\n",
    "for file in os.listdir(path):\n",
    "    if file.endswith(\".log\"):\n",
    "        # get phenotype name\n",
    "        pheno1, pheno2 = file.split('-')\n",
    "        pheno1 = pheno1.split('.')[0]\n",
    "        pheno2 = pheno2.split('.')[0]\n",
    "\n",
    "        with open(os.path.join(path, file), \"r\") as f:\n",
    "            lines = f.readlines()\n",
    "\n",
    "        for i, line in enumerate(lines):\n",
    "            # get h2g\n",
    "            if line.startswith(\"Heritability of phenotype 1\") and i + 2 < len(lines):\n",
    "                h2g_1 = lines[i + 2].split(\" \")[4].strip()\n",
    "                h2g_1 = float(h2g_1)\n",
    "\n",
    "                h2g_1_se = lines[i + 2].split(\" \")[5].strip()\n",
    "                h2g_1_se = h2g_1_se[1:-1]\n",
    "                h2g_1_se = float(h2g_1_se)\n",
    "\n",
    "                pheno1_lambda = lines[i + 3].split(\" \")[2].strip()\n",
    "                pheno1_lambda = float(pheno1_lambda)\n",
    "            \n",
    "            if line.startswith(\"Heritability of phenotype 2/2\") and i + 2 < len(lines):\n",
    "                h2g_2 = lines[i + 2].split(\" \")[4].strip()\n",
    "                h2g_2 = float(h2g_2)\n",
    "                \n",
    "                h2g_2_se = lines[i + 2].split(\" \")[5].strip()\n",
    "                h2g_2_se = h2g_2_se[1:-1]\n",
    "                h2g_2_se = float(h2g_2_se)\n",
    "\n",
    "                pheno2_lambda = lines[i + 3].split(\" \")[2].strip()\n",
    "                pheno2_lambda = float(pheno1_lambda)\n",
    "\n",
    "            # get rg\n",
    "            if line.startswith(\"Genetic Correlation:\"):\n",
    "                # print(line)\n",
    "                rg = line.split(\" \")[2].strip()\n",
    "                rg = float(rg)\n",
    "                # print(rg)\n",
    "                se = line.split(\" \")[3].strip()\n",
    "                se = se[1:-1]\n",
    "                se = float(se)\n",
    "            if line.startswith(\"Z-score:\"):\n",
    "                z = line.split(\" \")[1].strip()\n",
    "                z = float(z)\n",
    "            if line.startswith(\"P:\"):\n",
    "                p_value = line.split(\" \")[1].strip()\n",
    "                p_value = float(p_value)\n",
    "                neglog10p = -np.log10(p_value)\n",
    "\n",
    "        rg_df_fvsm = rg_df_fvsm.append({'pheno_f': pheno1, 'pheno_f_h2g': h2g_1, 'pheno_f_h2g_se': h2g_1_se, 'pheno_f_lambda': pheno1_lambda,\n",
    "                                        'pheno_m': pheno2, 'pheno_m_h2g': h2g_2, 'pheno_m_h2g_se': h2g_2_se, 'pheno_m_lambda': pheno2_lambda,\n",
    "                                        'cor': rg, 'se': se, 'z': z, 'p_value': p_value, 'neglog10p': neglog10p}, \n",
    "                                        ignore_index=True)\n",
    "\n",
    "rg_df_fvsm.sort_values(by = ['pheno_f', 'pheno_m'], inplace=True)\n",
    "rg_df_fvsm = rg_df_fvsm[rg_df_fvsm['pheno_f'] == rg_df_fvsm['pheno_m']]\n",
    "\n",
    "# add long bone male vs female correlation\n",
    "rg_df_fvsm['pheno'] = rg_df_fvsm['pheno_f']\n",
    "rg_df_fvsm = rg_df_fvsm[['pheno', 'cor', 'se']]\n",
    "\n",
    "long_bone_cor = pd.DataFrame({\n",
    "    'pheno': ['Humerus', 'Forearm', 'Torso length', 'Femur', 'Shoulder width', 'Tibia'],\n",
    "    'cor': [0.99, 0.97, 0.96, 0.95, 0.94, 0.92],\n",
    "    'se': [0.04, 0.04, 0.05, 0.04, 0.05, 0.04]\n",
    "})\n",
    "\n",
    "rg_df_fvsm = pd.concat([rg_df_fvsm, long_bone_cor]).reset_index(drop=True)\n",
    "rg_df_fvsm.to_csv(\"key_results/genetic_correlation/rg_hip_female_vs_male_240112.csv\", index=False)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Combine male and female genetic correlation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "male_rg = pd.read_csv(\"key_results/genetic_correlation/rg_20230909/rg_hip_male.csv\")\n",
    "female_rg = pd.read_csv(\"key_results/genetic_correlation/rg_20230909/rg_hip_female.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "acetabular_diameter                       20\n",
       "acetabular_inclination                    19\n",
       "arm_devide_torso                          18\n",
       "ear_left2ear_right                        17\n",
       "head_area_divide_pelvic_inlet_area        16\n",
       "head_divide_inlet_width                   15\n",
       "head_divide_oblique_inlet_length          14\n",
       "iliac_flare_angle                         13\n",
       "iliac_flare_ratio                         12\n",
       "iliac_isthmus_breadth                     11\n",
       "oblique_pelvic_inlet_length               10\n",
       "pelvic_height                              9\n",
       "pelvic_inlet_area                          8\n",
       "pelvic_inlet_width                         7\n",
       "pelvic_width                               6\n",
       "shoulder_area_divide_pelvic_inlet_area     5\n",
       "shoulder_divide_inlet_width                4\n",
       "shoulder_divide_oblique_inlet_length       3\n",
       "shoulder_width                             2\n",
       "subpubic_angle                             1\n",
       "Name: pheno1, dtype: int64"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "male_rg['pheno1'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "trochanter_left2trochanter_right          20\n",
       "subpubic_angle                            19\n",
       "shoulder_width                            18\n",
       "shoulder_divide_oblique_inlet_length      17\n",
       "shoulder_divide_inlet_width               16\n",
       "shoulder_area_divide_pelvic_inlet_area    15\n",
       "pelvic_width                              14\n",
       "pelvic_inlet_width                        13\n",
       "pelvic_inlet_area                         12\n",
       "pelvic_height                             11\n",
       "oblique_pelvic_inlet_length               10\n",
       "iliac_isthmus_breadth                      9\n",
       "iliac_flare_ratio                          8\n",
       "iliac_flare_angle                          7\n",
       "head_divide_oblique_inlet_length           6\n",
       "head_divide_inlet_width                    5\n",
       "head_area_divide_pelvic_inlet_area         4\n",
       "ear_left2ear_right                         3\n",
       "arm_devide_torso                           2\n",
       "acetabular_inclination                     1\n",
       "Name: pheno2, dtype: int64"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "male_rg['pheno2'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pheno1</th>\n",
       "      <th>pheno1_h2g</th>\n",
       "      <th>pheno1_h2g_se</th>\n",
       "      <th>pheno1_lambda</th>\n",
       "      <th>pheno2</th>\n",
       "      <th>pheno2_h2g</th>\n",
       "      <th>pheno2_h2g_se</th>\n",
       "      <th>pheno2_lambda</th>\n",
       "      <th>cor</th>\n",
       "      <th>se</th>\n",
       "      <th>z</th>\n",
       "      <th>p_value</th>\n",
       "      <th>neglog10p</th>\n",
       "      <th>gender</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>acetabular_diameter</td>\n",
       "      <td>0.3180</td>\n",
       "      <td>0.0420</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>acetabular_inclination</td>\n",
       "      <td>0.3033</td>\n",
       "      <td>0.0390</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>-0.4099</td>\n",
       "      <td>0.0774</td>\n",
       "      <td>-5.2948</td>\n",
       "      <td>1.191500e-07</td>\n",
       "      <td>6.923906</td>\n",
       "      <td>male</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>acetabular_diameter</td>\n",
       "      <td>0.3180</td>\n",
       "      <td>0.0420</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>arm_devide_torso</td>\n",
       "      <td>0.3869</td>\n",
       "      <td>0.0412</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>-0.0195</td>\n",
       "      <td>0.0742</td>\n",
       "      <td>-0.2631</td>\n",
       "      <td>7.925000e-01</td>\n",
       "      <td>0.101001</td>\n",
       "      <td>male</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>acetabular_diameter</td>\n",
       "      <td>0.3180</td>\n",
       "      <td>0.0420</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>ear_left2ear_right</td>\n",
       "      <td>0.2993</td>\n",
       "      <td>0.0390</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>0.0019</td>\n",
       "      <td>0.0781</td>\n",
       "      <td>0.0244</td>\n",
       "      <td>9.805000e-01</td>\n",
       "      <td>0.008552</td>\n",
       "      <td>male</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>acetabular_diameter</td>\n",
       "      <td>0.3180</td>\n",
       "      <td>0.0420</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>head_area_divide_pelvic_inlet_area</td>\n",
       "      <td>0.2877</td>\n",
       "      <td>0.0423</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>0.0427</td>\n",
       "      <td>0.0930</td>\n",
       "      <td>0.4590</td>\n",
       "      <td>6.462000e-01</td>\n",
       "      <td>0.189633</td>\n",
       "      <td>male</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>acetabular_diameter</td>\n",
       "      <td>0.3180</td>\n",
       "      <td>0.0420</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>head_divide_inlet_width</td>\n",
       "      <td>0.3588</td>\n",
       "      <td>0.0403</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>-0.0136</td>\n",
       "      <td>0.0744</td>\n",
       "      <td>-0.1823</td>\n",
       "      <td>8.554000e-01</td>\n",
       "      <td>0.067831</td>\n",
       "      <td>male</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>205</th>\n",
       "      <td>shoulder_divide_oblique_inlet_length</td>\n",
       "      <td>0.2687</td>\n",
       "      <td>0.0431</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>subpubic_angle</td>\n",
       "      <td>0.3472</td>\n",
       "      <td>0.0394</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>-0.5050</td>\n",
       "      <td>0.0670</td>\n",
       "      <td>-7.5421</td>\n",
       "      <td>4.624200e-14</td>\n",
       "      <td>13.334963</td>\n",
       "      <td>male</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>206</th>\n",
       "      <td>shoulder_divide_oblique_inlet_length</td>\n",
       "      <td>0.2687</td>\n",
       "      <td>0.0431</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>trochanter_left2trochanter_right</td>\n",
       "      <td>0.3494</td>\n",
       "      <td>0.0474</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>-0.0041</td>\n",
       "      <td>0.1113</td>\n",
       "      <td>-0.0368</td>\n",
       "      <td>9.707000e-01</td>\n",
       "      <td>0.012915</td>\n",
       "      <td>male</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>207</th>\n",
       "      <td>shoulder_width</td>\n",
       "      <td>0.3064</td>\n",
       "      <td>0.0394</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>subpubic_angle</td>\n",
       "      <td>0.3472</td>\n",
       "      <td>0.0394</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>-0.0829</td>\n",
       "      <td>0.0762</td>\n",
       "      <td>-1.0875</td>\n",
       "      <td>2.768000e-01</td>\n",
       "      <td>0.557834</td>\n",
       "      <td>male</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208</th>\n",
       "      <td>shoulder_width</td>\n",
       "      <td>0.3064</td>\n",
       "      <td>0.0394</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>trochanter_left2trochanter_right</td>\n",
       "      <td>0.3494</td>\n",
       "      <td>0.0474</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>0.1003</td>\n",
       "      <td>0.0770</td>\n",
       "      <td>1.3023</td>\n",
       "      <td>1.928000e-01</td>\n",
       "      <td>0.714893</td>\n",
       "      <td>male</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>209</th>\n",
       "      <td>subpubic_angle</td>\n",
       "      <td>0.3472</td>\n",
       "      <td>0.0394</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>trochanter_left2trochanter_right</td>\n",
       "      <td>0.3494</td>\n",
       "      <td>0.0474</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>-0.0214</td>\n",
       "      <td>0.0826</td>\n",
       "      <td>-0.2586</td>\n",
       "      <td>7.959000e-01</td>\n",
       "      <td>0.099141</td>\n",
       "      <td>male</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>210 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   pheno1  pheno1_h2g  pheno1_h2g_se  \\\n",
       "0                     acetabular_diameter      0.3180         0.0420   \n",
       "1                     acetabular_diameter      0.3180         0.0420   \n",
       "2                     acetabular_diameter      0.3180         0.0420   \n",
       "3                     acetabular_diameter      0.3180         0.0420   \n",
       "4                     acetabular_diameter      0.3180         0.0420   \n",
       "..                                    ...         ...            ...   \n",
       "205  shoulder_divide_oblique_inlet_length      0.2687         0.0431   \n",
       "206  shoulder_divide_oblique_inlet_length      0.2687         0.0431   \n",
       "207                        shoulder_width      0.3064         0.0394   \n",
       "208                        shoulder_width      0.3064         0.0394   \n",
       "209                        subpubic_angle      0.3472         0.0394   \n",
       "\n",
       "     pheno1_lambda                              pheno2  pheno2_h2g  \\\n",
       "0           1.0957              acetabular_inclination      0.3033   \n",
       "1           1.0957                    arm_devide_torso      0.3869   \n",
       "2           1.0957                  ear_left2ear_right      0.2993   \n",
       "3           1.0957  head_area_divide_pelvic_inlet_area      0.2877   \n",
       "4           1.0957             head_divide_inlet_width      0.3588   \n",
       "..             ...                                 ...         ...   \n",
       "205         1.0957                      subpubic_angle      0.3472   \n",
       "206         1.0957    trochanter_left2trochanter_right      0.3494   \n",
       "207         1.0957                      subpubic_angle      0.3472   \n",
       "208         1.0957    trochanter_left2trochanter_right      0.3494   \n",
       "209         1.0957    trochanter_left2trochanter_right      0.3494   \n",
       "\n",
       "     pheno2_h2g_se  pheno2_lambda     cor      se       z       p_value  \\\n",
       "0           0.0390         1.0957 -0.4099  0.0774 -5.2948  1.191500e-07   \n",
       "1           0.0412         1.0957 -0.0195  0.0742 -0.2631  7.925000e-01   \n",
       "2           0.0390         1.0957  0.0019  0.0781  0.0244  9.805000e-01   \n",
       "3           0.0423         1.0957  0.0427  0.0930  0.4590  6.462000e-01   \n",
       "4           0.0403         1.0957 -0.0136  0.0744 -0.1823  8.554000e-01   \n",
       "..             ...            ...     ...     ...     ...           ...   \n",
       "205         0.0394         1.0957 -0.5050  0.0670 -7.5421  4.624200e-14   \n",
       "206         0.0474         1.0957 -0.0041  0.1113 -0.0368  9.707000e-01   \n",
       "207         0.0394         1.0957 -0.0829  0.0762 -1.0875  2.768000e-01   \n",
       "208         0.0474         1.0957  0.1003  0.0770  1.3023  1.928000e-01   \n",
       "209         0.0474         1.0957 -0.0214  0.0826 -0.2586  7.959000e-01   \n",
       "\n",
       "     neglog10p gender  \n",
       "0     6.923906   male  \n",
       "1     0.101001   male  \n",
       "2     0.008552   male  \n",
       "3     0.189633   male  \n",
       "4     0.067831   male  \n",
       "..         ...    ...  \n",
       "205  13.334963   male  \n",
       "206   0.012915   male  \n",
       "207   0.557834   male  \n",
       "208   0.714893   male  \n",
       "209   0.099141   male  \n",
       "\n",
       "[210 rows x 14 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "male_rg"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pheno1</th>\n",
       "      <th>pheno1_h2g</th>\n",
       "      <th>pheno1_h2g_se</th>\n",
       "      <th>pheno1_lambda</th>\n",
       "      <th>pheno2</th>\n",
       "      <th>pheno2_h2g</th>\n",
       "      <th>pheno2_h2g_se</th>\n",
       "      <th>pheno2_lambda</th>\n",
       "      <th>cor</th>\n",
       "      <th>se</th>\n",
       "      <th>z</th>\n",
       "      <th>p_value</th>\n",
       "      <th>neglog10p</th>\n",
       "      <th>gender</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>acetabular_diameter</td>\n",
       "      <td>0.3366</td>\n",
       "      <td>0.0432</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>acetabular_inclination</td>\n",
       "      <td>0.2522</td>\n",
       "      <td>0.0427</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>-0.3536</td>\n",
       "      <td>0.0790</td>\n",
       "      <td>-4.4746</td>\n",
       "      <td>7.653800e-06</td>\n",
       "      <td>5.116123</td>\n",
       "      <td>female</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>acetabular_diameter</td>\n",
       "      <td>0.3366</td>\n",
       "      <td>0.0432</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>arm_devide_torso</td>\n",
       "      <td>0.4150</td>\n",
       "      <td>0.0447</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>-0.0629</td>\n",
       "      <td>0.0794</td>\n",
       "      <td>-0.7929</td>\n",
       "      <td>4.278000e-01</td>\n",
       "      <td>0.368759</td>\n",
       "      <td>female</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>acetabular_diameter</td>\n",
       "      <td>0.3366</td>\n",
       "      <td>0.0432</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>ear_left2ear_right</td>\n",
       "      <td>0.3437</td>\n",
       "      <td>0.0417</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>0.1411</td>\n",
       "      <td>0.0772</td>\n",
       "      <td>1.8278</td>\n",
       "      <td>6.760000e-02</td>\n",
       "      <td>1.170053</td>\n",
       "      <td>female</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>acetabular_diameter</td>\n",
       "      <td>0.3366</td>\n",
       "      <td>0.0432</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>head_area_divide_pelvic_inlet_area</td>\n",
       "      <td>0.3180</td>\n",
       "      <td>0.0367</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>0.2315</td>\n",
       "      <td>0.0784</td>\n",
       "      <td>2.9513</td>\n",
       "      <td>3.200000e-03</td>\n",
       "      <td>2.494850</td>\n",
       "      <td>female</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>acetabular_diameter</td>\n",
       "      <td>0.3366</td>\n",
       "      <td>0.0432</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>head_divide_inlet_width</td>\n",
       "      <td>0.3673</td>\n",
       "      <td>0.0402</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>0.0798</td>\n",
       "      <td>0.0012</td>\n",
       "      <td>9.990000e-01</td>\n",
       "      <td>0.000435</td>\n",
       "      <td>female</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>205</th>\n",
       "      <td>shoulder_divide_oblique_inlet_length</td>\n",
       "      <td>0.3093</td>\n",
       "      <td>0.0368</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>subpubic_angle</td>\n",
       "      <td>0.3718</td>\n",
       "      <td>0.0465</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>-0.6151</td>\n",
       "      <td>0.0572</td>\n",
       "      <td>-10.7614</td>\n",
       "      <td>5.235500e-27</td>\n",
       "      <td>26.281042</td>\n",
       "      <td>female</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>206</th>\n",
       "      <td>shoulder_divide_oblique_inlet_length</td>\n",
       "      <td>0.3093</td>\n",
       "      <td>0.0368</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>trochanter_left2trochanter_right</td>\n",
       "      <td>0.3651</td>\n",
       "      <td>0.0526</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>0.2040</td>\n",
       "      <td>0.0866</td>\n",
       "      <td>2.3561</td>\n",
       "      <td>1.850000e-02</td>\n",
       "      <td>1.732828</td>\n",
       "      <td>female</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>207</th>\n",
       "      <td>shoulder_width</td>\n",
       "      <td>0.3555</td>\n",
       "      <td>0.0380</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>subpubic_angle</td>\n",
       "      <td>0.3718</td>\n",
       "      <td>0.0465</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>-0.0066</td>\n",
       "      <td>0.0772</td>\n",
       "      <td>-0.0856</td>\n",
       "      <td>9.318000e-01</td>\n",
       "      <td>0.030677</td>\n",
       "      <td>female</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208</th>\n",
       "      <td>shoulder_width</td>\n",
       "      <td>0.3555</td>\n",
       "      <td>0.0380</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>trochanter_left2trochanter_right</td>\n",
       "      <td>0.3651</td>\n",
       "      <td>0.0526</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>0.0454</td>\n",
       "      <td>0.0764</td>\n",
       "      <td>0.5947</td>\n",
       "      <td>5.521000e-01</td>\n",
       "      <td>0.257982</td>\n",
       "      <td>female</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>209</th>\n",
       "      <td>subpubic_angle</td>\n",
       "      <td>0.3718</td>\n",
       "      <td>0.0465</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>trochanter_left2trochanter_right</td>\n",
       "      <td>0.3651</td>\n",
       "      <td>0.0526</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>-0.1326</td>\n",
       "      <td>0.0827</td>\n",
       "      <td>-1.6026</td>\n",
       "      <td>1.090000e-01</td>\n",
       "      <td>0.962574</td>\n",
       "      <td>female</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>210 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   pheno1  pheno1_h2g  pheno1_h2g_se  \\\n",
       "0                     acetabular_diameter      0.3366         0.0432   \n",
       "1                     acetabular_diameter      0.3366         0.0432   \n",
       "2                     acetabular_diameter      0.3366         0.0432   \n",
       "3                     acetabular_diameter      0.3366         0.0432   \n",
       "4                     acetabular_diameter      0.3366         0.0432   \n",
       "..                                    ...         ...            ...   \n",
       "205  shoulder_divide_oblique_inlet_length      0.3093         0.0368   \n",
       "206  shoulder_divide_oblique_inlet_length      0.3093         0.0368   \n",
       "207                        shoulder_width      0.3555         0.0380   \n",
       "208                        shoulder_width      0.3555         0.0380   \n",
       "209                        subpubic_angle      0.3718         0.0465   \n",
       "\n",
       "     pheno1_lambda                              pheno2  pheno2_h2g  \\\n",
       "0           1.0957              acetabular_inclination      0.2522   \n",
       "1           1.0957                    arm_devide_torso      0.4150   \n",
       "2           1.0957                  ear_left2ear_right      0.3437   \n",
       "3           1.0957  head_area_divide_pelvic_inlet_area      0.3180   \n",
       "4           1.0957             head_divide_inlet_width      0.3673   \n",
       "..             ...                                 ...         ...   \n",
       "205         1.0957                      subpubic_angle      0.3718   \n",
       "206         1.0957    trochanter_left2trochanter_right      0.3651   \n",
       "207         1.0957                      subpubic_angle      0.3718   \n",
       "208         1.0957    trochanter_left2trochanter_right      0.3651   \n",
       "209         1.0957    trochanter_left2trochanter_right      0.3651   \n",
       "\n",
       "     pheno2_h2g_se  pheno2_lambda     cor      se        z       p_value  \\\n",
       "0           0.0427         1.0957 -0.3536  0.0790  -4.4746  7.653800e-06   \n",
       "1           0.0447         1.0957 -0.0629  0.0794  -0.7929  4.278000e-01   \n",
       "2           0.0417         1.0957  0.1411  0.0772   1.8278  6.760000e-02   \n",
       "3           0.0367         1.0957  0.2315  0.0784   2.9513  3.200000e-03   \n",
       "4           0.0402         1.0957  0.0001  0.0798   0.0012  9.990000e-01   \n",
       "..             ...            ...     ...     ...      ...           ...   \n",
       "205         0.0465         1.0957 -0.6151  0.0572 -10.7614  5.235500e-27   \n",
       "206         0.0526         1.0957  0.2040  0.0866   2.3561  1.850000e-02   \n",
       "207         0.0465         1.0957 -0.0066  0.0772  -0.0856  9.318000e-01   \n",
       "208         0.0526         1.0957  0.0454  0.0764   0.5947  5.521000e-01   \n",
       "209         0.0526         1.0957 -0.1326  0.0827  -1.6026  1.090000e-01   \n",
       "\n",
       "     neglog10p  gender  \n",
       "0     5.116123  female  \n",
       "1     0.368759  female  \n",
       "2     1.170053  female  \n",
       "3     2.494850  female  \n",
       "4     0.000435  female  \n",
       "..         ...     ...  \n",
       "205  26.281042  female  \n",
       "206   1.732828  female  \n",
       "207   0.030677  female  \n",
       "208   0.257982  female  \n",
       "209   0.962574  female  \n",
       "\n",
       "[210 rows x 14 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "female_rg"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "female_rg.rename(columns={'pheno1': 'pheno2', 'pheno2': 'pheno1'}, inplace=True)\n",
    "\n",
    "all_cor = pd.concat([male_rg, \n",
    "                    female_rg], axis=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pheno1</th>\n",
       "      <th>pheno1_h2g</th>\n",
       "      <th>pheno1_h2g_se</th>\n",
       "      <th>pheno1_lambda</th>\n",
       "      <th>pheno2</th>\n",
       "      <th>pheno2_h2g</th>\n",
       "      <th>pheno2_h2g_se</th>\n",
       "      <th>pheno2_lambda</th>\n",
       "      <th>cor</th>\n",
       "      <th>se</th>\n",
       "      <th>z</th>\n",
       "      <th>p_value</th>\n",
       "      <th>neglog10p</th>\n",
       "      <th>gender</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>acetabular_diameter</td>\n",
       "      <td>0.3180</td>\n",
       "      <td>0.0420</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>acetabular_inclination</td>\n",
       "      <td>0.3033</td>\n",
       "      <td>0.0390</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>-0.4099</td>\n",
       "      <td>0.0774</td>\n",
       "      <td>-5.2948</td>\n",
       "      <td>1.191500e-07</td>\n",
       "      <td>6.923906</td>\n",
       "      <td>male</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>acetabular_diameter</td>\n",
       "      <td>0.3180</td>\n",
       "      <td>0.0420</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>arm_devide_torso</td>\n",
       "      <td>0.3869</td>\n",
       "      <td>0.0412</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>-0.0195</td>\n",
       "      <td>0.0742</td>\n",
       "      <td>-0.2631</td>\n",
       "      <td>7.925000e-01</td>\n",
       "      <td>0.101001</td>\n",
       "      <td>male</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>acetabular_diameter</td>\n",
       "      <td>0.3180</td>\n",
       "      <td>0.0420</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>ear_left2ear_right</td>\n",
       "      <td>0.2993</td>\n",
       "      <td>0.0390</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>0.0019</td>\n",
       "      <td>0.0781</td>\n",
       "      <td>0.0244</td>\n",
       "      <td>9.805000e-01</td>\n",
       "      <td>0.008552</td>\n",
       "      <td>male</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>acetabular_diameter</td>\n",
       "      <td>0.3180</td>\n",
       "      <td>0.0420</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>head_area_divide_pelvic_inlet_area</td>\n",
       "      <td>0.2877</td>\n",
       "      <td>0.0423</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>0.0427</td>\n",
       "      <td>0.0930</td>\n",
       "      <td>0.4590</td>\n",
       "      <td>6.462000e-01</td>\n",
       "      <td>0.189633</td>\n",
       "      <td>male</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>acetabular_diameter</td>\n",
       "      <td>0.3180</td>\n",
       "      <td>0.0420</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>head_divide_inlet_width</td>\n",
       "      <td>0.3588</td>\n",
       "      <td>0.0403</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>-0.0136</td>\n",
       "      <td>0.0744</td>\n",
       "      <td>-0.1823</td>\n",
       "      <td>8.554000e-01</td>\n",
       "      <td>0.067831</td>\n",
       "      <td>male</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>205</th>\n",
       "      <td>subpubic_angle</td>\n",
       "      <td>0.3093</td>\n",
       "      <td>0.0368</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>shoulder_divide_oblique_inlet_length</td>\n",
       "      <td>0.3718</td>\n",
       "      <td>0.0465</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>-0.6151</td>\n",
       "      <td>0.0572</td>\n",
       "      <td>-10.7614</td>\n",
       "      <td>5.235500e-27</td>\n",
       "      <td>26.281042</td>\n",
       "      <td>female</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>206</th>\n",
       "      <td>trochanter_left2trochanter_right</td>\n",
       "      <td>0.3093</td>\n",
       "      <td>0.0368</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>shoulder_divide_oblique_inlet_length</td>\n",
       "      <td>0.3651</td>\n",
       "      <td>0.0526</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>0.2040</td>\n",
       "      <td>0.0866</td>\n",
       "      <td>2.3561</td>\n",
       "      <td>1.850000e-02</td>\n",
       "      <td>1.732828</td>\n",
       "      <td>female</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>207</th>\n",
       "      <td>subpubic_angle</td>\n",
       "      <td>0.3555</td>\n",
       "      <td>0.0380</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>shoulder_width</td>\n",
       "      <td>0.3718</td>\n",
       "      <td>0.0465</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>-0.0066</td>\n",
       "      <td>0.0772</td>\n",
       "      <td>-0.0856</td>\n",
       "      <td>9.318000e-01</td>\n",
       "      <td>0.030677</td>\n",
       "      <td>female</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208</th>\n",
       "      <td>trochanter_left2trochanter_right</td>\n",
       "      <td>0.3555</td>\n",
       "      <td>0.0380</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>shoulder_width</td>\n",
       "      <td>0.3651</td>\n",
       "      <td>0.0526</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>0.0454</td>\n",
       "      <td>0.0764</td>\n",
       "      <td>0.5947</td>\n",
       "      <td>5.521000e-01</td>\n",
       "      <td>0.257982</td>\n",
       "      <td>female</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>209</th>\n",
       "      <td>trochanter_left2trochanter_right</td>\n",
       "      <td>0.3718</td>\n",
       "      <td>0.0465</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>subpubic_angle</td>\n",
       "      <td>0.3651</td>\n",
       "      <td>0.0526</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>-0.1326</td>\n",
       "      <td>0.0827</td>\n",
       "      <td>-1.6026</td>\n",
       "      <td>1.090000e-01</td>\n",
       "      <td>0.962574</td>\n",
       "      <td>female</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>420 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                               pheno1  pheno1_h2g  pheno1_h2g_se  \\\n",
       "0                 acetabular_diameter      0.3180         0.0420   \n",
       "1                 acetabular_diameter      0.3180         0.0420   \n",
       "2                 acetabular_diameter      0.3180         0.0420   \n",
       "3                 acetabular_diameter      0.3180         0.0420   \n",
       "4                 acetabular_diameter      0.3180         0.0420   \n",
       "..                                ...         ...            ...   \n",
       "205                    subpubic_angle      0.3093         0.0368   \n",
       "206  trochanter_left2trochanter_right      0.3093         0.0368   \n",
       "207                    subpubic_angle      0.3555         0.0380   \n",
       "208  trochanter_left2trochanter_right      0.3555         0.0380   \n",
       "209  trochanter_left2trochanter_right      0.3718         0.0465   \n",
       "\n",
       "     pheno1_lambda                                pheno2  pheno2_h2g  \\\n",
       "0           1.0957                acetabular_inclination      0.3033   \n",
       "1           1.0957                      arm_devide_torso      0.3869   \n",
       "2           1.0957                    ear_left2ear_right      0.2993   \n",
       "3           1.0957    head_area_divide_pelvic_inlet_area      0.2877   \n",
       "4           1.0957               head_divide_inlet_width      0.3588   \n",
       "..             ...                                   ...         ...   \n",
       "205         1.0957  shoulder_divide_oblique_inlet_length      0.3718   \n",
       "206         1.0957  shoulder_divide_oblique_inlet_length      0.3651   \n",
       "207         1.0957                        shoulder_width      0.3718   \n",
       "208         1.0957                        shoulder_width      0.3651   \n",
       "209         1.0957                        subpubic_angle      0.3651   \n",
       "\n",
       "     pheno2_h2g_se  pheno2_lambda     cor      se        z       p_value  \\\n",
       "0           0.0390         1.0957 -0.4099  0.0774  -5.2948  1.191500e-07   \n",
       "1           0.0412         1.0957 -0.0195  0.0742  -0.2631  7.925000e-01   \n",
       "2           0.0390         1.0957  0.0019  0.0781   0.0244  9.805000e-01   \n",
       "3           0.0423         1.0957  0.0427  0.0930   0.4590  6.462000e-01   \n",
       "4           0.0403         1.0957 -0.0136  0.0744  -0.1823  8.554000e-01   \n",
       "..             ...            ...     ...     ...      ...           ...   \n",
       "205         0.0465         1.0957 -0.6151  0.0572 -10.7614  5.235500e-27   \n",
       "206         0.0526         1.0957  0.2040  0.0866   2.3561  1.850000e-02   \n",
       "207         0.0465         1.0957 -0.0066  0.0772  -0.0856  9.318000e-01   \n",
       "208         0.0526         1.0957  0.0454  0.0764   0.5947  5.521000e-01   \n",
       "209         0.0526         1.0957 -0.1326  0.0827  -1.6026  1.090000e-01   \n",
       "\n",
       "     neglog10p  gender  \n",
       "0     6.923906    male  \n",
       "1     0.101001    male  \n",
       "2     0.008552    male  \n",
       "3     0.189633    male  \n",
       "4     0.067831    male  \n",
       "..         ...     ...  \n",
       "205  26.281042  female  \n",
       "206   1.732828  female  \n",
       "207   0.030677  female  \n",
       "208   0.257982  female  \n",
       "209   0.962574  female  \n",
       "\n",
       "[420 rows x 14 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_cor"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pheno1</th>\n",
       "      <th>pheno1_h2g</th>\n",
       "      <th>pheno1_h2g_se</th>\n",
       "      <th>pheno1_lambda</th>\n",
       "      <th>pheno2</th>\n",
       "      <th>pheno2_h2g</th>\n",
       "      <th>pheno2_h2g_se</th>\n",
       "      <th>pheno2_lambda</th>\n",
       "      <th>cor</th>\n",
       "      <th>se</th>\n",
       "      <th>z</th>\n",
       "      <th>p_value</th>\n",
       "      <th>neglog10p</th>\n",
       "      <th>gender</th>\n",
       "      <th>type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>acetabular_diameter</td>\n",
       "      <td>0.318</td>\n",
       "      <td>0.042</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>acetabular_inclination</td>\n",
       "      <td>0.3033</td>\n",
       "      <td>0.0390</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>-0.4099</td>\n",
       "      <td>0.0774</td>\n",
       "      <td>-5.2948</td>\n",
       "      <td>1.191500e-07</td>\n",
       "      <td>6.923906</td>\n",
       "      <td>male</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>acetabular_diameter</td>\n",
       "      <td>0.318</td>\n",
       "      <td>0.042</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>arm_devide_torso</td>\n",
       "      <td>0.3869</td>\n",
       "      <td>0.0412</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>-0.0195</td>\n",
       "      <td>0.0742</td>\n",
       "      <td>-0.2631</td>\n",
       "      <td>7.925000e-01</td>\n",
       "      <td>0.101001</td>\n",
       "      <td>male</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>acetabular_diameter</td>\n",
       "      <td>0.318</td>\n",
       "      <td>0.042</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>ear_left2ear_right</td>\n",
       "      <td>0.2993</td>\n",
       "      <td>0.0390</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>0.0019</td>\n",
       "      <td>0.0781</td>\n",
       "      <td>0.0244</td>\n",
       "      <td>9.805000e-01</td>\n",
       "      <td>0.008552</td>\n",
       "      <td>male</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>acetabular_diameter</td>\n",
       "      <td>0.318</td>\n",
       "      <td>0.042</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>head_area_divide_pelvic_inlet_area</td>\n",
       "      <td>0.2877</td>\n",
       "      <td>0.0423</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>0.0427</td>\n",
       "      <td>0.0930</td>\n",
       "      <td>0.4590</td>\n",
       "      <td>6.462000e-01</td>\n",
       "      <td>0.189633</td>\n",
       "      <td>male</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>acetabular_diameter</td>\n",
       "      <td>0.318</td>\n",
       "      <td>0.042</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>head_divide_inlet_width</td>\n",
       "      <td>0.3588</td>\n",
       "      <td>0.0403</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>-0.0136</td>\n",
       "      <td>0.0744</td>\n",
       "      <td>-0.1823</td>\n",
       "      <td>8.554000e-01</td>\n",
       "      <td>0.067831</td>\n",
       "      <td>male</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>436</th>\n",
       "      <td>shoulder_divide_inlet_width</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>shoulder_divide_inlet_width</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>437</th>\n",
       "      <td>shoulder_divide_oblique_inlet_length</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>shoulder_divide_oblique_inlet_length</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>438</th>\n",
       "      <td>shoulder_width</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>shoulder_width</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>439</th>\n",
       "      <td>subpubic_angle</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>subpubic_angle</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>440</th>\n",
       "      <td>trochanter_left2trochanter_right</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>trochanter_left2trochanter_right</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>441 rows × 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   pheno1  pheno1_h2g  pheno1_h2g_se  \\\n",
       "0                     acetabular_diameter       0.318          0.042   \n",
       "1                     acetabular_diameter       0.318          0.042   \n",
       "2                     acetabular_diameter       0.318          0.042   \n",
       "3                     acetabular_diameter       0.318          0.042   \n",
       "4                     acetabular_diameter       0.318          0.042   \n",
       "..                                    ...         ...            ...   \n",
       "436           shoulder_divide_inlet_width         NaN            NaN   \n",
       "437  shoulder_divide_oblique_inlet_length         NaN            NaN   \n",
       "438                        shoulder_width         NaN            NaN   \n",
       "439                        subpubic_angle         NaN            NaN   \n",
       "440      trochanter_left2trochanter_right         NaN            NaN   \n",
       "\n",
       "     pheno1_lambda                                pheno2  pheno2_h2g  \\\n",
       "0           1.0957                acetabular_inclination      0.3033   \n",
       "1           1.0957                      arm_devide_torso      0.3869   \n",
       "2           1.0957                    ear_left2ear_right      0.2993   \n",
       "3           1.0957    head_area_divide_pelvic_inlet_area      0.2877   \n",
       "4           1.0957               head_divide_inlet_width      0.3588   \n",
       "..             ...                                   ...         ...   \n",
       "436            NaN           shoulder_divide_inlet_width         NaN   \n",
       "437            NaN  shoulder_divide_oblique_inlet_length         NaN   \n",
       "438            NaN                        shoulder_width         NaN   \n",
       "439            NaN                        subpubic_angle         NaN   \n",
       "440            NaN      trochanter_left2trochanter_right         NaN   \n",
       "\n",
       "     pheno2_h2g_se  pheno2_lambda     cor      se       z       p_value  \\\n",
       "0           0.0390         1.0957 -0.4099  0.0774 -5.2948  1.191500e-07   \n",
       "1           0.0412         1.0957 -0.0195  0.0742 -0.2631  7.925000e-01   \n",
       "2           0.0390         1.0957  0.0019  0.0781  0.0244  9.805000e-01   \n",
       "3           0.0423         1.0957  0.0427  0.0930  0.4590  6.462000e-01   \n",
       "4           0.0403         1.0957 -0.0136  0.0744 -0.1823  8.554000e-01   \n",
       "..             ...            ...     ...     ...     ...           ...   \n",
       "436            NaN            NaN  1.0000     NaN     NaN  0.000000e+00   \n",
       "437            NaN            NaN  1.0000     NaN     NaN  0.000000e+00   \n",
       "438            NaN            NaN  1.0000     NaN     NaN  0.000000e+00   \n",
       "439            NaN            NaN  1.0000     NaN     NaN  0.000000e+00   \n",
       "440            NaN            NaN  1.0000     NaN     NaN  0.000000e+00   \n",
       "\n",
       "     neglog10p gender  type  \n",
       "0     6.923906   male   NaN  \n",
       "1     0.101001   male   NaN  \n",
       "2     0.008552   male   NaN  \n",
       "3     0.189633   male   NaN  \n",
       "4     0.067831   male   NaN  \n",
       "..         ...    ...   ...  \n",
       "436        NaN    NaN   NaN  \n",
       "437        NaN    NaN   NaN  \n",
       "438        NaN    NaN   NaN  \n",
       "439        NaN    NaN   NaN  \n",
       "440        NaN    NaN   NaN  \n",
       "\n",
       "[441 rows x 15 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "for pheno in all_cor.pheno1.unique():\n",
    "    cor = {'pheno1': pheno, 'pheno2': pheno, 'cor': 1, 'p_value': 0, 'type': np.nan}\n",
    "    all_cor = all_cor.append(cor, ignore_index=True)\n",
    "\n",
    "all_cor"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_cor.to_csv(\"key_results/genetic_correlation/male_vs_female_genetic_cor.csv\", index=False)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Combine genetic correlation and phenotypic correlation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "# get genetic correlation\n",
    "rg_df_both = pd.read_csv(\"key_results/genetic_correlation/rg_20230909/rg_hip_both.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/7f/jzt97g754236ldjj13l5pgd00000gn/T/ipykernel_20099/4285930895.py:14: RuntimeWarning: divide by zero encountered in log10\n",
      "  'neglog10p': -np.log10(p_value)},  ignore_index=True)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pheno1</th>\n",
       "      <th>pheno2</th>\n",
       "      <th>cor</th>\n",
       "      <th>p_value</th>\n",
       "      <th>neglog10p</th>\n",
       "      <th>type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>pelvic_height</td>\n",
       "      <td>pelvic_width</td>\n",
       "      <td>0.220268</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>inf</td>\n",
       "      <td>phenotype_cor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>pelvic_height</td>\n",
       "      <td>pelvic_inlet_width</td>\n",
       "      <td>0.182181</td>\n",
       "      <td>1.438844e-224</td>\n",
       "      <td>223.841986</td>\n",
       "      <td>phenotype_cor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>pelvic_height</td>\n",
       "      <td>subpubic_angle</td>\n",
       "      <td>-0.104209</td>\n",
       "      <td>5.605186e-74</td>\n",
       "      <td>73.251410</td>\n",
       "      <td>phenotype_cor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>pelvic_height</td>\n",
       "      <td>trochanter_left2trochanter_right</td>\n",
       "      <td>0.274015</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>inf</td>\n",
       "      <td>phenotype_cor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>pelvic_height</td>\n",
       "      <td>shoulder_width</td>\n",
       "      <td>0.207732</td>\n",
       "      <td>7.950429e-293</td>\n",
       "      <td>292.099609</td>\n",
       "      <td>phenotype_cor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>205</th>\n",
       "      <td>shoulder_area_divide_pelvic_inlet_area</td>\n",
       "      <td>subpubic_angle</td>\n",
       "      <td>-0.805037</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>inf</td>\n",
       "      <td>phenotype_cor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>206</th>\n",
       "      <td>shoulder_area_divide_pelvic_inlet_area</td>\n",
       "      <td>trochanter_left2trochanter_right</td>\n",
       "      <td>-0.149230</td>\n",
       "      <td>1.579664e-150</td>\n",
       "      <td>149.801435</td>\n",
       "      <td>phenotype_cor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>207</th>\n",
       "      <td>shoulder_area_divide_pelvic_inlet_area</td>\n",
       "      <td>shoulder_width</td>\n",
       "      <td>0.590272</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>inf</td>\n",
       "      <td>phenotype_cor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208</th>\n",
       "      <td>shoulder_area_divide_pelvic_inlet_area</td>\n",
       "      <td>shoulder_divide_inlet_width</td>\n",
       "      <td>0.902217</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>inf</td>\n",
       "      <td>phenotype_cor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>209</th>\n",
       "      <td>shoulder_area_divide_pelvic_inlet_area</td>\n",
       "      <td>shoulder_divide_oblique_inlet_length</td>\n",
       "      <td>0.945973</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>inf</td>\n",
       "      <td>phenotype_cor</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>210 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                     pheno1  \\\n",
       "0                             pelvic_height   \n",
       "1                             pelvic_height   \n",
       "2                             pelvic_height   \n",
       "3                             pelvic_height   \n",
       "4                             pelvic_height   \n",
       "..                                      ...   \n",
       "205  shoulder_area_divide_pelvic_inlet_area   \n",
       "206  shoulder_area_divide_pelvic_inlet_area   \n",
       "207  shoulder_area_divide_pelvic_inlet_area   \n",
       "208  shoulder_area_divide_pelvic_inlet_area   \n",
       "209  shoulder_area_divide_pelvic_inlet_area   \n",
       "\n",
       "                                   pheno2       cor        p_value  \\\n",
       "0                            pelvic_width  0.220268   0.000000e+00   \n",
       "1                      pelvic_inlet_width  0.182181  1.438844e-224   \n",
       "2                          subpubic_angle -0.104209   5.605186e-74   \n",
       "3        trochanter_left2trochanter_right  0.274015   0.000000e+00   \n",
       "4                          shoulder_width  0.207732  7.950429e-293   \n",
       "..                                    ...       ...            ...   \n",
       "205                        subpubic_angle -0.805037   0.000000e+00   \n",
       "206      trochanter_left2trochanter_right -0.149230  1.579664e-150   \n",
       "207                        shoulder_width  0.590272   0.000000e+00   \n",
       "208           shoulder_divide_inlet_width  0.902217   0.000000e+00   \n",
       "209  shoulder_divide_oblique_inlet_length  0.945973   0.000000e+00   \n",
       "\n",
       "      neglog10p           type  \n",
       "0           inf  phenotype_cor  \n",
       "1    223.841986  phenotype_cor  \n",
       "2     73.251410  phenotype_cor  \n",
       "3           inf  phenotype_cor  \n",
       "4    292.099609  phenotype_cor  \n",
       "..          ...            ...  \n",
       "205         inf  phenotype_cor  \n",
       "206  149.801435  phenotype_cor  \n",
       "207         inf  phenotype_cor  \n",
       "208         inf  phenotype_cor  \n",
       "209         inf  phenotype_cor  \n",
       "\n",
       "[210 rows x 6 columns]"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# phenotype correlation\n",
    "hip_pheno_flt = pd.read_csv('key_results/hip_select_pheno_cm_residual.csv').drop(columns=['standing_height'])\n",
    "hip_pheno_flt = hip_pheno_flt.iloc[:, 1:]\n",
    "\n",
    "pheno_cor = pd.DataFrame(columns=['pheno1', 'pheno2', 'cor', 'p_value', 'neglog10p'])\n",
    "\n",
    "for col1 in hip_pheno_flt.columns:\n",
    "    for col2 in hip_pheno_flt.columns:\n",
    "        corr, p_value = pearsonr(hip_pheno_flt[col1], hip_pheno_flt[col2])\n",
    "        pheno_cor = pheno_cor.append({'pheno1': col1, \n",
    "                                      'pheno2': col2, \n",
    "                                      'cor': corr, \n",
    "                                      'p_value': p_value,\n",
    "                                      'neglog10p': -np.log10(p_value)},  ignore_index=True)\n",
    "pheno_cor['type'] = 'phenotype_cor'\n",
    "pheno_cor = pheno_cor.merge(rg_df_both[['pheno1', 'pheno2']], on=['pheno1', 'pheno2'], how='inner')\n",
    "\n",
    "pheno_cor"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "# merge genetic correlation and phenotype correlation\n",
    "rg_df_both.rename(columns={'pheno1': 'pheno2', 'pheno2': 'pheno1'}, inplace=True)\n",
    "rg_df_both['type'] = 'genetic_cor'\n",
    "all_cor = pd.concat([rg_df_both[['pheno1', 'pheno2', 'cor', 'p_value', 'neglog10p', 'type']], pheno_cor], axis=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pheno1</th>\n",
       "      <th>pheno2</th>\n",
       "      <th>cor</th>\n",
       "      <th>p_value</th>\n",
       "      <th>neglog10p</th>\n",
       "      <th>type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>acetabular_inclination</td>\n",
       "      <td>acetabular_diameter</td>\n",
       "      <td>-0.357900</td>\n",
       "      <td>3.272200e-14</td>\n",
       "      <td>13.485160</td>\n",
       "      <td>genetic_cor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>arm_devide_torso</td>\n",
       "      <td>acetabular_diameter</td>\n",
       "      <td>-0.001400</td>\n",
       "      <td>9.765000e-01</td>\n",
       "      <td>0.010328</td>\n",
       "      <td>genetic_cor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ear_left2ear_right</td>\n",
       "      <td>acetabular_diameter</td>\n",
       "      <td>0.048100</td>\n",
       "      <td>3.634000e-01</td>\n",
       "      <td>0.439615</td>\n",
       "      <td>genetic_cor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>head_area_divide_pelvic_inlet_area</td>\n",
       "      <td>acetabular_diameter</td>\n",
       "      <td>0.089100</td>\n",
       "      <td>1.598000e-01</td>\n",
       "      <td>0.796423</td>\n",
       "      <td>genetic_cor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>head_divide_inlet_width</td>\n",
       "      <td>acetabular_diameter</td>\n",
       "      <td>-0.040300</td>\n",
       "      <td>4.426000e-01</td>\n",
       "      <td>0.353989</td>\n",
       "      <td>genetic_cor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>205</th>\n",
       "      <td>shoulder_area_divide_pelvic_inlet_area</td>\n",
       "      <td>subpubic_angle</td>\n",
       "      <td>-0.805037</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>inf</td>\n",
       "      <td>phenotype_cor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>206</th>\n",
       "      <td>shoulder_area_divide_pelvic_inlet_area</td>\n",
       "      <td>trochanter_left2trochanter_right</td>\n",
       "      <td>-0.149230</td>\n",
       "      <td>1.579664e-150</td>\n",
       "      <td>149.801435</td>\n",
       "      <td>phenotype_cor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>207</th>\n",
       "      <td>shoulder_area_divide_pelvic_inlet_area</td>\n",
       "      <td>shoulder_width</td>\n",
       "      <td>0.590272</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>inf</td>\n",
       "      <td>phenotype_cor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208</th>\n",
       "      <td>shoulder_area_divide_pelvic_inlet_area</td>\n",
       "      <td>shoulder_divide_inlet_width</td>\n",
       "      <td>0.902217</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>inf</td>\n",
       "      <td>phenotype_cor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>209</th>\n",
       "      <td>shoulder_area_divide_pelvic_inlet_area</td>\n",
       "      <td>shoulder_divide_oblique_inlet_length</td>\n",
       "      <td>0.945973</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>inf</td>\n",
       "      <td>phenotype_cor</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>420 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                     pheno1  \\\n",
       "0                    acetabular_inclination   \n",
       "1                          arm_devide_torso   \n",
       "2                        ear_left2ear_right   \n",
       "3        head_area_divide_pelvic_inlet_area   \n",
       "4                   head_divide_inlet_width   \n",
       "..                                      ...   \n",
       "205  shoulder_area_divide_pelvic_inlet_area   \n",
       "206  shoulder_area_divide_pelvic_inlet_area   \n",
       "207  shoulder_area_divide_pelvic_inlet_area   \n",
       "208  shoulder_area_divide_pelvic_inlet_area   \n",
       "209  shoulder_area_divide_pelvic_inlet_area   \n",
       "\n",
       "                                   pheno2       cor        p_value  \\\n",
       "0                     acetabular_diameter -0.357900   3.272200e-14   \n",
       "1                     acetabular_diameter -0.001400   9.765000e-01   \n",
       "2                     acetabular_diameter  0.048100   3.634000e-01   \n",
       "3                     acetabular_diameter  0.089100   1.598000e-01   \n",
       "4                     acetabular_diameter -0.040300   4.426000e-01   \n",
       "..                                    ...       ...            ...   \n",
       "205                        subpubic_angle -0.805037   0.000000e+00   \n",
       "206      trochanter_left2trochanter_right -0.149230  1.579664e-150   \n",
       "207                        shoulder_width  0.590272   0.000000e+00   \n",
       "208           shoulder_divide_inlet_width  0.902217   0.000000e+00   \n",
       "209  shoulder_divide_oblique_inlet_length  0.945973   0.000000e+00   \n",
       "\n",
       "      neglog10p           type  \n",
       "0     13.485160    genetic_cor  \n",
       "1      0.010328    genetic_cor  \n",
       "2      0.439615    genetic_cor  \n",
       "3      0.796423    genetic_cor  \n",
       "4      0.353989    genetic_cor  \n",
       "..          ...            ...  \n",
       "205         inf  phenotype_cor  \n",
       "206  149.801435  phenotype_cor  \n",
       "207         inf  phenotype_cor  \n",
       "208         inf  phenotype_cor  \n",
       "209         inf  phenotype_cor  \n",
       "\n",
       "[420 rows x 6 columns]"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_cor"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pheno1</th>\n",
       "      <th>pheno2</th>\n",
       "      <th>cor</th>\n",
       "      <th>p_value</th>\n",
       "      <th>neglog10p</th>\n",
       "      <th>type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>acetabular_inclination</td>\n",
       "      <td>acetabular_diameter</td>\n",
       "      <td>-0.3579</td>\n",
       "      <td>3.272200e-14</td>\n",
       "      <td>13.485160</td>\n",
       "      <td>genetic_cor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>arm_devide_torso</td>\n",
       "      <td>acetabular_diameter</td>\n",
       "      <td>-0.0014</td>\n",
       "      <td>9.765000e-01</td>\n",
       "      <td>0.010328</td>\n",
       "      <td>genetic_cor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ear_left2ear_right</td>\n",
       "      <td>acetabular_diameter</td>\n",
       "      <td>0.0481</td>\n",
       "      <td>3.634000e-01</td>\n",
       "      <td>0.439615</td>\n",
       "      <td>genetic_cor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>head_area_divide_pelvic_inlet_area</td>\n",
       "      <td>acetabular_diameter</td>\n",
       "      <td>0.0891</td>\n",
       "      <td>1.598000e-01</td>\n",
       "      <td>0.796423</td>\n",
       "      <td>genetic_cor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>head_divide_inlet_width</td>\n",
       "      <td>acetabular_diameter</td>\n",
       "      <td>-0.0403</td>\n",
       "      <td>4.426000e-01</td>\n",
       "      <td>0.353989</td>\n",
       "      <td>genetic_cor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>436</th>\n",
       "      <td>shoulder_divide_oblique_inlet_length</td>\n",
       "      <td>shoulder_divide_oblique_inlet_length</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>437</th>\n",
       "      <td>shoulder_width</td>\n",
       "      <td>shoulder_width</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>438</th>\n",
       "      <td>subpubic_angle</td>\n",
       "      <td>subpubic_angle</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>439</th>\n",
       "      <td>trochanter_left2trochanter_right</td>\n",
       "      <td>trochanter_left2trochanter_right</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>440</th>\n",
       "      <td>acetabular_diameter</td>\n",
       "      <td>acetabular_diameter</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>441 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   pheno1  \\\n",
       "0                  acetabular_inclination   \n",
       "1                        arm_devide_torso   \n",
       "2                      ear_left2ear_right   \n",
       "3      head_area_divide_pelvic_inlet_area   \n",
       "4                 head_divide_inlet_width   \n",
       "..                                    ...   \n",
       "436  shoulder_divide_oblique_inlet_length   \n",
       "437                        shoulder_width   \n",
       "438                        subpubic_angle   \n",
       "439      trochanter_left2trochanter_right   \n",
       "440                   acetabular_diameter   \n",
       "\n",
       "                                   pheno2     cor       p_value  neglog10p  \\\n",
       "0                     acetabular_diameter -0.3579  3.272200e-14  13.485160   \n",
       "1                     acetabular_diameter -0.0014  9.765000e-01   0.010328   \n",
       "2                     acetabular_diameter  0.0481  3.634000e-01   0.439615   \n",
       "3                     acetabular_diameter  0.0891  1.598000e-01   0.796423   \n",
       "4                     acetabular_diameter -0.0403  4.426000e-01   0.353989   \n",
       "..                                    ...     ...           ...        ...   \n",
       "436  shoulder_divide_oblique_inlet_length  1.0000  0.000000e+00        NaN   \n",
       "437                        shoulder_width  1.0000  0.000000e+00        NaN   \n",
       "438                        subpubic_angle  1.0000  0.000000e+00        NaN   \n",
       "439      trochanter_left2trochanter_right  1.0000  0.000000e+00        NaN   \n",
       "440                   acetabular_diameter  1.0000  0.000000e+00        NaN   \n",
       "\n",
       "            type  \n",
       "0    genetic_cor  \n",
       "1    genetic_cor  \n",
       "2    genetic_cor  \n",
       "3    genetic_cor  \n",
       "4    genetic_cor  \n",
       "..           ...  \n",
       "436          NaN  \n",
       "437          NaN  \n",
       "438          NaN  \n",
       "439          NaN  \n",
       "440          NaN  \n",
       "\n",
       "[441 rows x 6 columns]"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "for pheno in all_cor.pheno1.unique():\n",
    "    cor = {'pheno1': pheno, 'pheno2': pheno, 'cor': 1, 'p_value': 0, 'type': np.nan}\n",
    "    all_cor = all_cor.append(cor, ignore_index=True)\n",
    "\n",
    "all_cor"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_cor.to_csv(\"key_results/genetic_correlation/genetic_and_phenotype_cor.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "acetabular_diameter                       21\n",
       "pelvic_height                             21\n",
       "subpubic_angle                            21\n",
       "shoulder_width                            21\n",
       "shoulder_divide_oblique_inlet_length      21\n",
       "shoulder_divide_inlet_width               21\n",
       "shoulder_area_divide_pelvic_inlet_area    21\n",
       "pelvic_width                              21\n",
       "pelvic_inlet_width                        21\n",
       "pelvic_inlet_area                         21\n",
       "oblique_pelvic_inlet_length               21\n",
       "acetabular_inclination                    21\n",
       "iliac_isthmus_breadth                     21\n",
       "iliac_flare_ratio                         21\n",
       "iliac_flare_angle                         21\n",
       "head_divide_oblique_inlet_length          21\n",
       "head_divide_inlet_width                   21\n",
       "head_area_divide_pelvic_inlet_area        21\n",
       "ear_left2ear_right                        21\n",
       "arm_devide_torso                          21\n",
       "trochanter_left2trochanter_right          21\n",
       "Name: pheno2, dtype: int64"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_cor['pheno2'].value_counts()"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### get rg for regression comparation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/7f/jzt97g754236ldjj13l5pgd00000gn/T/ipykernel_86435/1037662206.py:28: RuntimeWarning: divide by zero encountered in log10\n",
      "  neglog10p = -np.log10(p_value)\n"
     ]
    }
   ],
   "source": [
    "# different kinds of regression rg\n",
    "rg_df = pd.DataFrame(columns=['pheno1', 'pheno2', 'cor', 'se', 'z', 'p_value'])\n",
    "path = \"key_results/genetic_correlation/rg_20230626\"\n",
    "for file in os.listdir(path):\n",
    "    if file.endswith(\".log\"):\n",
    "        # get phenotype name\n",
    "        pheno1, pheno2 = file.split('-')\n",
    "        pheno1 = pheno1.split('.')[0]\n",
    "        pheno2 = pheno2.split('.')[0]\n",
    "\n",
    "        with open(os.path.join(path, file), \"r\") as f:\n",
    "            for line in f:\n",
    "                # get rg info\n",
    "                if line.startswith(\"Genetic Correlation:\"):\n",
    "                    # print(line)\n",
    "                    rg = line.split(\" \")[2].strip()\n",
    "                    rg = float(rg)\n",
    "                    # print(rg)\n",
    "                    se = line.split(\" \")[3].strip()\n",
    "                    se = se[1:-1]\n",
    "                    se = float(se)\n",
    "                if line.startswith(\"Z-score:\"):\n",
    "                    z = line.split(\" \")[1].strip()\n",
    "                    z = float(z)\n",
    "                if line.startswith(\"P:\"):\n",
    "                    p_value = line.split(\" \")[1].strip()\n",
    "                    p_value = float(p_value)\n",
    "                    neglog10p = -np.log10(p_value)\n",
    "\n",
    "            rg_df = rg_df.append({'pheno1': pheno1, 'pheno2': pheno2, 'cor': rg, 'se': se, 'z': z, 'p_value': p_value, 'neglog10p': neglog10p}, ignore_index=True)\n",
    "\n",
    "rg_df.sort_values(by = ['pheno1', 'pheno2'], inplace=True)\n",
    "rg_df.to_csv(\"key_results/genetic_correlation/rg_20230626/rg_reg_compare.csv\", index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pheno1</th>\n",
       "      <th>pheno2</th>\n",
       "      <th>cor</th>\n",
       "      <th>se</th>\n",
       "      <th>z</th>\n",
       "      <th>p_value</th>\n",
       "      <th>neglog10p</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>hip_width:height~height</td>\n",
       "      <td>hip_width:height~nan</td>\n",
       "      <td>0.9750</td>\n",
       "      <td>0.002400</td>\n",
       "      <td>398.4114</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>inf</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>hip_width:height~height</td>\n",
       "      <td>hip_width~height</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.000084</td>\n",
       "      <td>11901.8496</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>inf</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>hip_width:height~height</td>\n",
       "      <td>standing_height</td>\n",
       "      <td>0.0388</td>\n",
       "      <td>0.040800</td>\n",
       "      <td>0.9496</td>\n",
       "      <td>0.342300</td>\n",
       "      <td>0.465593</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>hip_width:height~nan</td>\n",
       "      <td>hip_width~height</td>\n",
       "      <td>0.9742</td>\n",
       "      <td>0.002500</td>\n",
       "      <td>384.4155</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>inf</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>hip_width:height~nan</td>\n",
       "      <td>standing_height</td>\n",
       "      <td>-0.1847</td>\n",
       "      <td>0.037900</td>\n",
       "      <td>-4.8700</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>5.952336</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>hip_width~height</td>\n",
       "      <td>standing_height</td>\n",
       "      <td>0.0419</td>\n",
       "      <td>0.040800</td>\n",
       "      <td>1.0273</td>\n",
       "      <td>0.304300</td>\n",
       "      <td>0.516698</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    pheno1                pheno2     cor        se  \\\n",
       "1  hip_width:height~height  hip_width:height~nan  0.9750  0.002400   \n",
       "3  hip_width:height~height      hip_width~height  1.0000  0.000084   \n",
       "0  hip_width:height~height       standing_height  0.0388  0.040800   \n",
       "2     hip_width:height~nan      hip_width~height  0.9742  0.002500   \n",
       "4     hip_width:height~nan       standing_height -0.1847  0.037900   \n",
       "5         hip_width~height       standing_height  0.0419  0.040800   \n",
       "\n",
       "            z   p_value  neglog10p  \n",
       "1    398.4114  0.000000        inf  \n",
       "3  11901.8496  0.000000        inf  \n",
       "0      0.9496  0.342300   0.465593  \n",
       "2    384.4155  0.000000        inf  \n",
       "4     -4.8700  0.000001   5.952336  \n",
       "5      1.0273  0.304300   0.516698  "
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rg_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAv0AAAKqCAYAAACpTeFbAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjUuMywgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/NK7nSAAAACXBIWXMAAA9hAAAPYQGoP6dpAACZ6ElEQVR4nOzdd1gU1xoG8Hd3aTZ6EREpImJBUVHEErvYRRN7AY0ae8HYu0axxxJr7L3FGCsW1KjR2BF7RVEUkCY2irtz//C6cbOAu8PiAnl/zzPP3T1zZs43I5v77dlzzkgEQRBARERERET5llTfARARERERUc5i0k9ERERElM8x6SciIiIiyueY9BMRERER5XNM+omIiIiI8jkm/URERERE+RyTfiIiIiKifI5JPxERERFRPsekn4iIiIgon2PST0Q6IZFIMHnyZH2HoTc5cf1169ZF3bp1dXpOIiL6b2LST5QLREREYODAgXB3d0fBggVRsGBBlC1bFgMGDEB4eLi+w1M6ePDgV0vsU1JS8PPPP8PHxwdmZmYwMTGBu7s7Bg4ciHv37n2VGL6GW7duYfLkyXj8+LG+Q1E6efIkJBIJdu3aleH+wMBAFC5cOEdjOHv2LCZPnoykpKQcbYeI6L/CQN8BEP3X7d+/Hx06dICBgQG6dOmCihUrQiqV4s6dO9i9ezeWLVuGiIgIODk56TtUHDx4EEuWLMkw8X///j0MDHTzn5S4uDg0adIEly9fRosWLdC5c2cULlwYd+/exbZt27By5UqkpaXppC19u3XrFqZMmYK6devC2dlZZd+RI0f0E1QucPbsWUyZMgWBgYEwNzfXdzhERHkek34iPXr48CE6duwIJycnhIaGwt7eXmX/rFmzsHTpUkiluf9HORMTE52dKzAwEFevXsWuXbvw7bffquybNm0axo0bp5N23r59i0KFCqmVC4KAlJQUFChQQCftiGVkZKTX9omIKP/I/ZkEUT42e/ZsvH37FmvXrlVL+AHAwMAAgwcPhqOjo0r5nTt38N1338HS0hImJibw9vbG3r17VeqsW7cOEokEf/31F4KCgmBjY4NChQqhTZs2ePnypVpbhw4dQu3atVGoUCEUKVIEzZs3x82bN5X7AwMDsWTJEgAfx69/2j7JaEx7VFQUvv/+exQrVgzGxsZwcXFBv379suylP3/+PA4cOIDvv/9eLeEHAGNjY8ydO1el7Pjx48rYzc3N0bp1a9y+fVulzuTJkyGRSHDr1i107twZFhYWqFWrFgDA2dkZLVq0wOHDh+Ht7Y0CBQpgxYoVAICkpCQMHToUjo6OMDY2hpubG2bNmgWFQpHpNQDAkydP0L9/f5QuXRoFChSAlZUV2rVrpzKMZ926dWjXrh0AoF69esp7evLkSQAZj+mPjY3F999/Dzs7O5iYmKBixYpYv369Sp3Hjx9DIpFg7ty5WLlyJUqWLAljY2NUrVoVFy9ezDLu7PjS3xAAhIeHIzAwEK6urjAxMUHRokXRs2dPxMfHK+tMnjwZI0aMAAC4uLgo78uneyeRSDBw4EDs3LkTZcuWRYECBeDr64vr168DAFasWAE3NzeYmJigbt26akOnTp8+jXbt2qFEiRIwNjaGo6Mjhg0bhvfv36vU+zSM6dGjR/Dz80OhQoVQrFgxTJ06FYIg6PjuERHlLPb0E+nR/v374ebmBh8fH42PuXnzJmrWrAkHBweMHj0ahQoVwo4dO+Dv74/ffvsNbdq0Uak/aNAgWFhYYNKkSXj8+DEWLFiAgQMHYvv27co6GzduREBAAPz8/DBr1iy8e/cOy5YtQ61atXD16lU4Ozvjhx9+wPPnz3H06FFs3Ljxi3E+f/4c1apVQ1JSEvr06QMPDw9ERUVh165dePfuXaa92J++vHTr1k2j+3Hs2DE0bdoUrq6umDx5Mt6/f4/FixejZs2auHLlitqQmXbt2qFUqVKYMWOGSuJ29+5ddOrUCT/88AN69+6N0qVL4927d6hTpw6ioqLwww8/oESJEjh79izGjBmDFy9eYMGCBZnGdfHiRZw9exYdO3ZE8eLF8fjxYyxbtgx169bFrVu3ULBgQXzzzTcYPHgwFi1ahLFjx6JMmTIAoPzff3v//j3q1q2LBw8eYODAgXBxccHOnTsRGBiIpKQkDBkyRKX+li1b8Pr1a/zwww+QSCSYPXs22rZti0ePHsHQ0PCL9/b169eIi4tTK09NTVUr0+RvCACOHj2KR48eoUePHihatChu3ryJlStX4ubNm/j7778hkUjQtm1b3Lt3D1u3bsXPP/8Ma2trAICNjY2yvdOnT2Pv3r0YMGAAACA4OBgtWrTAyJEjsXTpUvTv3x+JiYmYPXs2evbsiePHjyuP3blzJ969e4d+/frBysoKFy5cwOLFi/Hs2TPs3LlT5brkcjmaNGmC6tWrY/bs2QgJCcGkSZPw4cMHTJ069Yv3kIgo1xCISC9evXolABD8/f3V9iUmJgovX75Ubu/evVPua9CggeDp6SmkpKQoyxQKhVCjRg2hVKlSyrK1a9cKAISGDRsKCoVCWT5s2DBBJpMJSUlJgiAIwuvXrwVzc3Ohd+/eKjFER0cLZmZmKuUDBgwQMvvPBgBh0qRJyvfdu3cXpFKpcPHiRbW6n8fzb23atBEACImJiZnW+ZyXl5dga2srxMfHK8uuXbsmSKVSoXv37sqySZMmCQCETp06qZ3DyclJACCEhISolE+bNk0oVKiQcO/ePZXy0aNHCzKZTIiMjFSW/fv6P/83++TcuXMCAGHDhg3Ksp07dwoAhBMnTqjVr1OnjlCnTh3l+wULFggAhE2bNinL0tLSBF9fX6Fw4cJCcnKyIAiCEBERIQAQrKyshISEBGXdP/74QwAg7Nu3T62tz504cUIAkOVWqFAhZX1t/oYyui9bt24VAAinTp1Sls2ZM0cAIERERKjVByAYGxur7FuxYoUAQChatKjyPgiCIIwZM0btPBnFEBwcLEgkEuHJkyfKsoCAAAGAMGjQIGWZQqEQmjdvLhgZGQkvX75UOw8RUW7F4T1EepKcnAwAGa6CUrduXdjY2Ci3T8NqEhIScPz4cbRv317ZCxsXF4f4+Hj4+fnh/v37iIqKUjlXnz59VIbh1K5dG3K5HE+ePAHwsec1KSkJnTp1Up4vLi4OMpkMPj4+OHHihNbXplAosGfPHrRs2RLe3t5q+z+PJ7P7UqRIkS+28+LFC4SFhSEwMBCWlpbK8goVKqBRo0Y4ePCg2jF9+/bN8FwuLi7w8/NTKdu5cydq164NCwsLlXvTsGFDyOVynDp1KtPYPp8PkJ6ejvj4eLi5ucHc3BxXrlz54rVl5ODBgyhatCg6deqkLDM0NMTgwYPx5s0b/Pnnnyr1O3ToAAsLC+X72rVrAwAePXqkUXsTJ07E0aNH1bbGjRur1NPmb+jz+5KSkoK4uDhUr14dALS6Lw0aNFD5FefTr2Xffvutyt/Op/LPr/nzGN6+fYu4uDjUqFEDgiDg6tWram0NHDhQ+frT0KK0tDQcO3ZM43iJiPSNw3uI9ORTYvLmzRu1fStWrMDr168RExODrl27KssfPHgAQRAwYcIETJgwIcPzxsbGwsHBQfm+RIkSKvs/JYGJiYkAgPv37wMA6tevn+H5TE1NNb0kpZcvXyI5ORnly5fX+thP7b1+/fqLq7Z8+uJSunRptX1lypTB4cOH1Sbruri4ZHiujMrv37+P8PBwlWEln4uNjc00tvfv3yM4OBhr165FVFSUylCiV69eZXpcVp48eYJSpUqpTez+NBzo0/345Ev/9l/i6emJhg0bqpVv2rRJ5b02f0MJCQmYMmUKtm3bpnb/tLkv/742MzMzAFCb//Kp/PNrjoyMxMSJE7F37161e/HvGKRSKVxdXVXK3N3dASBXLbNKRPQlTPqJ9MTMzAz29va4ceOG2r5PvZP/Tio+TR798ccf1XqlP3Fzc1N5L5PJMqz3KQn9dM6NGzeiaNGiavV0tQynpjw8PAAA169fV/ZM61JmK/JkVK5QKNCoUSOMHDkyw2M+JX8ZGTRoENauXYuhQ4fC19cXZmZmkEgk6Nix4xcnAevKl/7tdUWbv6H27dvj7NmzGDFiBLy8vFC4cGEoFAo0adJEq/uS2bV96ZrlcjkaNWqEhIQEjBo1Ch4eHihUqBCioqIQGBj41f5tiIi+Nib9RHrUvHlzrFq1ChcuXEC1atW+WP9Tj6OhoWGGPbBilCxZEgBga2v7xXNmNSznczY2NjA1Nc3wC82XtGzZEsHBwdi0adMXk/5Pzy64e/eu2r47d+7A2to6wyU5NVWyZEm8efNG1L3etWsXAgICMG/ePGVZSkqK2sOmNL2nwMfrDQ8Ph0KhUOntv3PnjnK/Pmj6N5SYmIjQ0FBMmTIFEydOVJZ/+qXgc9rcF21cv34d9+7dw/r169G9e3dl+dGjRzOsr1Ao8OjRI5UveJ8eDvfvSeJERLkZx/QT6dHIkSNRsGBB9OzZEzExMWr7/90ja2tri7p162LFihV48eKFWv2MluL8Ej8/P5iammLGjBlIT0/P8pyfEugvPSVVKpXC398f+/btw6VLl9T2Z9XT7OvriyZNmmDVqlXYs2eP2v60tDT8+OOPAAB7e3t4eXlh/fr1KjHduHEDR44cQbNmzbKM80vat2+Pc+fO4fDhw2r7kpKS8OHDh0yPlclkate5ePFiyOVylTJN7ykANGvWDNHR0SorL3348AGLFy9G4cKFUadOnS+eIydo+jf0qRf+3/clo1WQtLkv2sgoBkEQsHDhwkyP+eWXX1Tq/vLLLzA0NESDBg10GhsRUU5iTz+RHpUqVQpbtmxBp06dULp0aeUTeQVBQEREBLZs2QKpVIrixYsrj1myZAlq1aoFT09P9O7dG66uroiJicG5c+fw7NkzXLt2TasYTE1NsWzZMnTr1g2VK1dGx44dYWNjg8jISBw4cAA1a9ZUJj1VqlQBAAwePBh+fn6QyWTo2LFjhuedMWMGjhw5gjp16qBPnz4oU6YMXrx4gZ07d+LMmTNZjtffsGEDGjdujLZt26Jly5Zo0KABChUqhPv372Pbtm148eKFcq3+OXPmoGnTpvD19cX333+vXLLTzMwswycHa2PEiBHYu3cvWrRogcDAQFSpUgVv377F9evXsWvXLjx+/Fi5nOS/tWjRAhs3boSZmRnKli2Lc+fO4dixY7CyslKp5+XlBZlMhlmzZuHVq1cwNjZG/fr1YWtrq3bOPn36YMWKFQgMDMTly5fh7OyMXbt24a+//sKCBQs0mvycEzT9GzI1NcU333yD2bNnIz09HQ4ODjhy5AgiIiLUzvnpb23cuHHo2LEjDA0N0bJly2z9cgN8HD5WsmRJ/Pjjj4iKioKpqSl+++23TOc5mJiYICQkBAEBAfDx8cGhQ4dw4MABjB07NtO5HkREuZI+lgwiIlUPHjwQ+vXrJ7i5uQkmJiZCgQIFBA8PD6Fv375CWFiYWv2HDx8K3bt3F4oWLSoYGhoKDg4OQosWLYRdu3Yp63xasvPfS2Z+Wo7x30tEnjhxQvDz8xPMzMwEExMToWTJkkJgYKBw6dIlZZ0PHz4IgwYNEmxsbASJRKKyfCf+tWSlIAjCkydPhO7duws2NjaCsbGx4OrqKgwYMEBITU394j159+6dMHfuXKFq1apC4cKFBSMjI6FUqVLCoEGDhAcPHqjUPXbsmFCzZk2hQIECgqmpqdCyZUvh1q1bKnU+LdmZ0TKLTk5OQvPmzTOM4/Xr18KYMWMENzc3wcjISLC2thZq1KghzJ07V0hLS8v0+hMTE4UePXoI1tbWQuHChQU/Pz/hzp07gpOTkxAQEKDSxq+//iq4uroKMplM5d/m30t2CoIgxMTEKM9rZGQkeHp6CmvXrlWp82nJzjlz5qhdT0b/Tv/26W9k586dGe4PCAhQWbLz8+O+9Df07NkzoU2bNoK5ublgZmYmtGvXTnj+/HmGcU2bNk1wcHAQpFKpyrKbAIQBAwZodM0ZXcutW7eEhg0bCoULFxasra2F3r17C9euXRMAqNzLT9f58OFDoXHjxkLBggUFOzs7YdKkSYJcLs/yHhIR5TYSQeBjBYmIiP4tMDAQu3btynCFLSKivIZj+omIiIiI8jkm/URERERE+RyTfiIiIiKifI5JPxERUQbWrVvH8fxEpHOnTp1Cy5YtUaxYMUgkkgyXp/63kydPonLlyjA2NoabmxvWrVundbtM+omIiIiIvpK3b9+iYsWKWLJkiUb1IyIi0Lx5c9SrVw9hYWEYOnQoevXqleEzZLLC1XuIiIiIiPRAIpHg999/h7+/f6Z1Ro0ahQMHDqg85b5jx45ISkpCSEiIxm2xp5+IiIiISKTU1FQkJyerbKmpqTo7/7lz59CwYUOVMj8/P5w7d06r8+SaJ/IOlrnoOwSiPClNwR/riLS1MDlM3yEQ5UnGRcz1HUKm+kqc9dJu0UmBmDJlikrZpEmTsv1U+E+io6NhZ2enUmZnZ4fk5GS8f/8eBQoU0Og8uSbpJyIiIiLKa8aMGYOgoCCVMmNjYz1Fkzkm/UREREREIhkbG+dokl+0aFHExMSolMXExMDU1FTjXn6AY/qJiIiIiHItX19fhIaGqpQdPXoUvr6+Wp2HST8RERER0Vfy5s0bhIWFISwsDMDHJTnDwsIQGRkJ4ONwoe7duyvr9+3bF48ePcLIkSNx584dLF26FDt27MCwYcO0apfDe4iIiIgoz5NJ9B2BZi5duoR69eop33+aDxAQEIB169bhxYsXyi8AAODi4oIDBw5g2LBhWLhwIYoXL45Vq1bBz89Pq3ZzzTr9XL2HSByu3kOkPa7eQyRObl69Z4DUWS/tLlE81ku72uLwHiIiIiKifI5JPxERERFRPsekn4iIiIgon+NEXiIiIiLK82SSPDKTV0/Y009ERERElM8x6SciIiIiyueY9BMRERER5XNM+omIiIiI8jlO5CUiIiKiPC+vPJFXX9jTT0RERESUz7Gnn4iIiIjyPC7ZmTX29BMRERER5XNM+omIiIiI8jlRSf+pU6fw4cMHtfIPHz7g1KlT2Q6KiIiIiIh0R1TSX69ePSQkJKiVv3r1CvXq1ct2UEREREREpDuiJvIKggBJBpMl4uPjUahQoWwHRURERESkDS7ZmTWtkv62bdsCACQSCQIDA2FsbKzcJ5fLER4ejho1aug2QiIiIiIiyhatkn4zMzMAH3v6ixQpggIFCij3GRkZoXr16ujdu7duIyQiIiIiomzRKulfu3YtAMDZ2Rk//vgjh/IQEREREeUBosb0T5o0SddxEBERERFRDhG1ek9MTAy6deuGYsWKwcDAADKZTGUjIiIiIvqaZBKJXra8QlRPf2BgICIjIzFhwgTY29tnuJIPERERERHlDqKS/jNnzuD06dPw8vLScThERERERNrjkp1ZEzW8x9HREYIg6DoWIiIiIiLKAaKS/gULFmD06NF4/PixjsMhIiIiIiJd03h4j4WFhcrY/bdv36JkyZIoWLAgDA0NVeomJCToLkIiIiIiIsoWjZP+BQsW5GAYRERERESUUzRO+gMCAnIyDiIiIiIi0fLS8pn6IGr1nuTk5AzLJRIJjI2NYWRklK2giIiIiIhId0Ql/ebm5lmuzV+8eHEEBgZi0qRJkEpFzRUmIiIiIiIdEZX0r1u3DuPGjUNgYCCqVasGALhw4QLWr1+P8ePH4+XLl5g7dy6MjY0xduxYnQZMRERERETaEZX0r1+/HvPmzUP79u2VZS1btoSnpydWrFiB0NBQlChRAtOnT2fST0RERESkZ6LG3pw9exaVKlVSK69UqRLOnTsHAKhVqxYiIyOzFx0RERERkQaketryCtFP5F29erVa+erVq+Ho6AgAiI+Ph4WFRfaiIyIiIiKibBM1vGfu3Llo164dDh06hKpVqwIALl26hDt37mDXrl0AgIsXL6JDhw66i5SIiIiIKBNcsjNropL+Vq1a4c6dO1i5ciXu3r0LAGjatCn27NkDZ2dnAEC/fv10FiQREREREYknKukHABcXFwQHB+syFiIiIiIiygEaJ/3h4eEoX748pFIpwsPDs6xboUKFbAdGRERERES6oXHS7+XlhejoaNja2sLLywsSiQSCIKjVk0gkkMvlOg2SiIiIiIjE0zjpj4iIgI2NjfI1EREREVFuIeM83ixpnPQ7OTll+JqIiIiIiHI30c8U2LhxI2rWrIlixYrhyZMnAIAFCxbgjz/+0FlwRERERESUfaKS/mXLliEoKAjNmjVDUlKScgy/ubk5FixYoMv4iIiIiIgom0Ql/YsXL8avv/6KcePGQSaTKcu9vb1x/fp1nQVHRERERETZJ2qd/oiICFSqVEmt3NjYGG/fvs12UERERERE2uATebMmqqffxcUFYWFhauUhISEoU6ZMdmMiIiIiIiIdEtXTHxQUhAEDBiAlJQWCIODChQvYunUrgoODsWrVKl3HSERERESUJS7ZmTVRSX+vXr1QoEABjB8/Hu/evUPnzp1RrFgxLFy4EB07dtR1jERERERElA2ikn4A6NKlC7p06YJ3797hzZs3sLW11WVcRERERESkI6KT/k8KFiyIggUL6iIWIiIiIiLKAaIm8sbExKBbt24oVqwYDAwMIJPJVDYiIiIiIso9RPX0BwYGIjIyEhMmTIC9vT0kXCKJiIiIiPSIS3ZmTVTSf+bMGZw+fRpeXl46DoeIiIiIiHRN1PAeR0dHCIKg61iIiIiIiCgHiEr6FyxYgNGjR+Px48c6DoeIiIiIiHRN4+E9FhYWKmP33759i5IlS6JgwYIwNDRUqZuQkKC7CImIiIiIKFs0TvoXLFiQg2EQEREREYnHJ/JmTeOkPyAgICfjICIiIiKiHJLth3MREREREekbl+zMmqiJvAAgCAJatWqF1atX6zIeIiIiIiLSMdFJv0QiwfTp0zFx4kRdxkNERERERDomOukHAE9PT7x8+RKxsbG6ioeIiIiIiHQsW0n/rVu3YGhoCAsLC13FQ0REREREOpatibwbN25E69at1dbpJyIiIiL6mrhkZ9ay1dP/+++/o1OnTrqKhYiIiIiIckC2kv7ExES4urrqKhYiIiIiIsoB2Ur6GzZsiIMHD+oqFiIiIiIiygHZSvqnTZuGM2fOIDk5WVfxEBERERGRjmVrIq+rqyv++OMPXcVCRERERCQKJ/JmTVRP/4YNG5CamqpWnpaWhg0bNmQ7KCIiIiIi0h1RSX+PHj3w6tUrtfLXr1+jR48e2Q6KiIiIiEgbMolEL1teISrpFwQBkgwu8tmzZzAzM8t2UEREREREpDtajemvVKkSJBIJJBIJGjRoAAODfw6Xy+WIiIhAkyZNdB4kERERERGJp1XS7+/vDwAICwuDn58fChcurNxnZGQEZ2dnfPvttzoNkIiIiIiIskerpH/SpEkAAGdnZ3To0AEmJiY5EhQREREREemOqCU7AwICAHxcrSc2NhYKhUJlf4kSJbIfGRERERGRhrhkZ9ZEJf33799Hz549cfbsWZXyTxN85XK5ToIjIiIiIqLsE5X0BwYGwsDAAPv374e9vX2GK/kQEREREVHuICrpDwsLw+XLl+Hh4aHreIiIiIiISMdErdNftmxZxMXF6ToWIiIiIiLKARr39CcnJytfz5o1CyNHjsSMGTPg6ekJQ0NDlbqmpqa6i5CIiIiI6Avy0tNx9UHjpN/c3Fxl7L4gCGjQoIFKHU7kJSIiIiLKfTRO+k+cOJGTcVAOqd2vG+r/2AemRW0Qde02dg2ZjMiL1zKsKzUwQOPR/VCt+7cwcyiK2LuPsHfMTNw+fEpZZ9LD07ByLq527OmlG7Fz0EQAwKDQrShVt7rK/jMrNmNH//E6vDIi/XGrXQ2NR/RBiSqeMC9mh2X+fXDtjyNZHuNepzq+mz8e9uVKIfHpCxz66RecW7/rK0VMlDO27diJdRs3Iy4+Hu6lSmHMiOHwLF8u0/pHjoXil2Ur8PzFC5RwdMSwQQNQu1ZN5f6lK35FyJGjiI6JgaGhIcqW8cCg/n1RoXx5ZZ3HTyIxf+EihF0LR/qHdLi7uWFAvx9QzdtbWefGzVtY8MsS3L59B5BI4FmuLIYNHojS7u45cyMoV+CSnVnTOOmvU6dOTsZBOaBS++ZoM28ctvcfjyfnw1BnSE/0P7QeP5VpgDcv49Xqt5g2HN5d/LHthzGIufMQZfy+wfe/rcCCWt/iWdgtAMA8n9aQyP6ZCmJfvjQGHtmEq7sOqJzrr1+34uCk+cr36e9Scugqib4+40IF8ezabZxdsxN9f1/xxfpWzsUx4MAanFq+GWu6DIFHg5roumomXr2Ixa0jp754PFFuFHLkKOb8vBATxoyCZ/ly2LR1G/oOGoK9v+2AlaWlWv2wa+EYNW4CBg/ohzq1a+FgyGEM+XEktm/agFJuJQEATk4lMHbkjyju4ICU1FRs3LIVfQcMxv49v8HSwgIAMGhYEEo4OmLV8iUwNjbGpq3bMHDocBzcsxvW1lZ49+4d+g0egrrf1Ma4USMhl8uxdMVK9B00BEcO7IOhgag1TIjyPFETecPDwzPcrl+/jvv37yM1NVXXcZII9Yb2wtlV23F+3S5E336AHf3GIe3de1Tv0S7D+lW7tsHR4KW4degk4iOe4szyzbh16ATqBfVW1nkTl4DXMXHKrXzz+nj54DEe/Hle5Vzp796r1Et5/SZHr5Xoa7oZchJ7J8xD2J7DGtX/pm9XxEU8xW8/Tkf0nYc4uWQDruw6hAbDvs/hSIlyzobNW/Gtf2v4t2qJkq6umDBmNAqYmGDP3n0Z1t+8bTtq+lZHj+7d4OrigoH9+qKMR2ls27FTWad5Ez9U96mG4sUd4FbSFSOGDcGbt29x7/4DAEBiUhKeRD5Fz8DucC9VCk4lSmDowAFISUnBg4cPAQARj5/g1atkDPjhB7g4O8GtpCv69umF+PgEvHjxIudvDFEuJSrp9/LyQqVKldQ2Ly8veHh4wMzMDAEBAUhJYe+uvsgMDeFYpTzuhp5RlgmCgLuhf8HFt3KGxxgYGyH9X1/Y0t+nwrWmd4b1ZYaG8O7ij7/X7lTb5925NWbEXMboayFoOX0EDAuYZONqiPI2V99KuHPsL5WyW4dPwdW3kp4iIsqe9PR03L5zB9V9qinLpFIpfKpVxbXw6xkecy38OnyqVVUpq+FbHdeuZ1w/PT0du37fgyKFC6O0eykAgLmZGZydnLDvwCG8e/8eHz58wM7dv8PS0gJly3xcRtzZqQTMzcyw+4+9SE9PR0pKCn7/Yy9cXZxRzN5eF5dPlCeJ+o3r999/x6hRozBixAhUq/bxA3/hwgXMmzcPkyZNwocPHzB69GiMHz8ec+fO1WnApJlC1haQGRjgdYzq0qqvY+JgV7pkhsfcPnIK9YZ+j4enLiDu4RO4N6iJim38IJVl/N2wgn9jFDA3xfl/jUu+vG0vEp5E4dXzGDhU8ECr4FGwLe2K1d/1083FEeUxpkVtkKz2WXyJAmamMDQxRnoKfx2lvCUxKQlyuVxtGI+VpSUiHj/J8Ji4+PgM68fFqw43/fP0GYwcOx4pKSmwsbbGiiWLYWFuDgCQSCRYuXQxhv44Er7f1INUKoWlhQWWLVqoXDmwUKFCWL1iGYb+OBIrV68BAJRwdMTyXxbCgEN76D9M1F//9OnTsXDhQvj5+SnLPD09Ubx4cUyYMAEXLlxAoUKFMHz4cCb9ecjuoVPRcWUwxt06BkEQEPcwEufX7YJPJsOBqvdsj9shfyL5RaxK+dlftypfv7hxF69exGLQsS2wdi2BuEeROXoNRESUt1X1roKdWzYiMSkJu3//Az+OGYvN69bAytISgiBgxqw5sLSwwLpfV8DYxBi79+zFoKDh2LphHWysrZGSkoJJ06bDq2IFzJo+DXKFHOs3bsaAIUHYumEtTEz4y3N+xSU7syZqeM/169fh5OSkVu7k5ITr//+ZzsvLi2Pn9OhtXCLkHz6giJ21SnkRO2u8jnmZ4TFv4hKwqu0P+LFIWUx2qYXpZRsg9c1bxGeQqFuUcEDpBjVxbvX2L8by5HwYAMDazVnr6yDKD5KjX8JU7bNog/evktnLT3mShbk5ZDIZ4hMSVMrjExJgbaU+iRcArK2sMqlvpVJWsEABlHB0REVPT0yZOB4GMhl+/2MvAOD8xUs4deYvzJ7xEyp5VURZDw+MHz0SJsbG2Lv/44ISB0OO4PmL55g2aQLKlyuLip6emDV9GqKeP8eJPzlxnv67RCX9Hh4emDlzJtLS0pRl6enpmDlzJjw8Po6pi4qKgp2dnW6iJK3J09Px9PINuNf/Zyk0iUSC0vVrIOLclSyP/ZCahlfPYyA1MEDFtk1wfe9RtTrVA7/D69h43Dxw/IuxOHiVBQC1XwSI/isenbsKjwY1VMrKNKqFR+eu6ikiouwxNDREGQ8PnL9wUVmmUChw/uJFVKzgmeExFSt44vzFSyplf5+/gIqeGdf/57wC0tLSAUA5V1AqVU1fJBIpFApBWUcqkao8W0gikUAikUAhCBpeIVH+I2p4z5IlS9CqVSsUL14cFSpUAPCx918ul2P//v0AgEePHqF///66i5S0dmLBKnRdOw9PL4fjyYVrqDukJ4wKFcT5dR/H4HddNw+voqKxb9wcAIBTNS+YOdghKuwWzByKounEIZBIpQido7okoUQigU9gO1zY8BsU/3oQm7VrCVTp1Bq3Dp3A2/hEFKtQBm3njceDP8/j+fU7X+fCiXKYcaGCsPnslytrF0cUr1gWbxOSkPj0OfxnjIS5gx3WBQwHAJxavgl1B3ZH21mj8deanfCo74sq7ZtjSfOeeroCouzr3qUTxk+eirJly8CzXFls2rIN79+nwL9lCwDA2ImTYWdrgyEDBwAAunTsgJ59+mL9ps34plZNHDp8FDdv3cbEsWMAAO/ev8eva9ai7je1YWNtjaSkJGzbsQuxL1+iccOPDwOtWMETpkWKYNykKejb+3sYG5vgtz17EPX8Ob6p9fGLtW/1api/aDGmz5qDzh3aQaEQsGbdehjIZKjmXUUPd4oodxCV9NeoUQMRERHYvHkz7t27BwBo164dOnfujCJFigAAunXrprsoSZSrOw6gsLUVmk0OgmlRazwLu41lzQLxOvbjhEILx2IQFAplfUMTY7SYOhxWriWQ+uYtbh06iY0BQXj/6rXKeUs3rAVLJ4cMV+35kJaO0g1qou6QHjAqVBCJT58jbHcIjkz/JWcvlugrcvKugKCT25Tv2/08AQBwbt0urO/xI8zsbWFZwkG5P/7xMyxp3hPf/TwB9Yb0QNKzaGzqNZpr9FOe1qRxIyQmJmHp8pWIi49HaXd3LFu8AFb/H64THR2j0iPvVbECZk6fhsVLl2PRkmUo4eiIhXNnK9fol0mlePz4CYbvP4jEpCSYm5mhXNkyWPfrCriVdAXwcVjRssULsXjpMvTqNwAfPnxASVdXLJw3R/ngLRdnZyyePxfLf12Fbj16QSKVwqO0O5YuXgAba2sQ/VdJBCF3/NY1WOai7xCI8qQ0Ra74CBPlKQuTw/QdAlGeZFzEXN8hZOr3opk/DTontYm+qZd2taVxT//evXvRtGlTGBoaYu/evVnWbdWqVbYDIyIiIiLKj5YsWYI5c+YgOjoaFStWxOLFi5XL4GdkwYIFWLZsGSIjI2FtbY3vvvsOwcHBWq1GpXHS7+/vj+joaNja2sLf3z/TehKJBPJ/jfMmIiIiIiJg+/btCAoKwvLly+Hj44MFCxbAz88Pd+/eha2trVr9LVu2YPTo0VizZg1q1KiBe/fuITAwEBKJBPPnz9e4XY2TfsVnY78/fy1GamoqUv/15Fe5IHB9VSIiIiLK1+bPn4/evXujR48eAIDly5fjwIEDWLNmDUaPHq1W/+zZs6hZsyY6d+4MAHB2dkanTp1w/vx5rdoVtWTn5z4tn6WN4OBgmJmZqWyXhKTshkJERERE9FWlpqYiOTlZZft35/YnaWlpuHz5Mho2bKgsk0qlaNiwIc6dO5fhMTVq1MDly5dx4cIFAB9XyDx48CCaNWumVZyikn65XI5p06bBwcEBhQsXxqNHjwAAEyZMwOrVq794/JgxY/Dq1SuVzVtiLiYUAlDQ0hzTX1yEpZPDlyvnIkF/7UbFtk30HQb9RxWyNMfsmEuwciqu71C0MvLc76jEzw3lkKSkV6jTqAminj/Xdyhf3Zmz59Cuc9dsj2Yg/ZFJJHrZMurMDg4OzjDGuLg4yOVytWdZ2dnZITo6OsNjOnfujKlTp6JWrVowNDREyZIlUbduXYwdO1ar+yMq6Z8+fTrWrVuH2bNnw8jISFlevnx5rFq16ovHGxsbw9TUVGXj0B7x/MYOwPW9x5DwJCrH2qjdrxsmPTyNeW/vIOjs7yhRteIXj/H6rhnG3TyGeW/vYHTYIZRtWldl/+EZv6DljJEqD1Ah+lqajhuI8D+OIv7Jsxxro07/bpgecQaL39/FqL/3wFmDz03l75ph8u1QLH5/FxPCQ1D+X5+bQz8tRpuZo/i5oRzx65q1qFfnGzgUK5ZjbWzbsRNNWvrDu0ZtdA7oies3sl755NjxE+jYLQA16zZAtVp10K5zV+w7cDDT+tNmzEQFbx9s3LJVWRb1/DkmTf0JTVr5o2rNb9CsdVssWbES6enpyjq1avjCwMAABw6FZP8i6T8lo87sMWPG6Oz8J0+exIwZM7B06VJcuXIFu3fvxoEDBzBt2jStziMq6d+wYQNWrlyJLl26QCaTKcsrVqyIO3f4AKavybCACar3bI+/12zPsTYqtW+ONvPGIWTaQszxboGo8Nvof2g9CttYZXqMi29lBGxeiHNrdmB2leYI/+Moeu1eAfty7so6tw6dhEmRwijzr6SGKKcZFjBBze/b46/VO3KsjSrtW+C7+eOxf8pCzKjcHM+u3cKgwxtQJIvPjatvZXy/dRH+Wr0d0ys1Q9ieI+i7ZyWKffa5uXHoJIyLFEY5fm5Ix96npOD3P/aiTeuWOdZGyJGjmPPzQvTt/T22b1qP0u5u6DtoCOITEjI9xszUFL179sDGtavw27bNaN2yBSZO/Ql/nftbrW7oiZMIv3EDtjY2KuURj59AISgwcexo/L59K0YEDcXO33Zj4ZKlKvVat2iOLdtz7r8LlLOkEoletow6s42NjTOM0draGjKZDDExMSrlMTExKFq0aIbHTJgwAd26dUOvXr3g6emJNm3aYMaMGQgODtbqlylRSX9UVBTc3NzUyhUKhcq3Zsp55ZrVw4fUNDw+HwYAcKvjg0XyCJRtVg+jrh762DP/126VZFtb9Yb2wtlV23F+3S5E336AHf3GIe3de1Tv0S7TY+oM7oHbh//E8XkrEXPnIQ5Omo9nV26i9oDuyjqCQoFbh06gSoec+z8Yoox4NquH9NQ0RJy/CgBwr1Mdy4XHKN+sHsZfO4TF7+9i5LnfVZJtbTUM6oW/ft2Gc+t24sXtB9jSdxzS371HjZ7tMz2m/pCeuBnyJ47OXYnoOw+xb+J8RF65iboDA5R1BIUCNw6egHdHfm5It06f+QuGRkao6OkJALh46TIqePvg1Jkz+LZjF3jXqI0ugT1x/8FD0W1s2LwV3/q3hn+rlijp6ooJY0ajgIkJ9uzdl+kxVb2roEG9unB1cYFj8eLo2qkjSrm54WpYmEq9mNhYBM+Zi+BpU2FgoLpOSa0avpg2aSJqVK+O4sUdUK/ONwjo2gWhJ06q1KvzTW3cvHUbT5/l3C+A9N9mZGSEKlWqIDQ0VFmmUCgQGhoKX1/fDI959+6dyoPuACg73bV53JaoJ/KWLVsWp0+fhpOTk0r5rl27UKlSJTGnJJFca1XF0ys31Mr9Z43Bb8OmIjn6JVpOH4E+f6zCNI/6UHz4AAvHYhh740iW5z0SvBRHZy6FzNAQjlXK4+isf3pDBEHA3dC/4OJbOdPjnatXwsmfVed33D5yChVaN1Ype3LxGhqO7KfJpRLpjFvtaoi8rP65+XbOWOwYMgWvol/Cf8YI9N+3GhPd6yk/N5NuHc3yvCEzliAk+OPnpkSV8ggJVv3c3D72F1yz+Ny4+lbCsfmqn5tbh0/By1/1c/P4wjU0Gc3PDenWlbAwlC3joVY+f+FijBoeBCtrKyxasgyDgoZj3+5dMDQwwIvoaPi365jleXv1CETvnoFIT0/H7Tt30KvHP19ipVIpfKpVxbXw6xrFKAgCzl+8hMdPnmDYoAHKcoVCgbETJyOwW1fl03u/5M2btzAzNVUpsy9aFFZWlrhyNQyOxfPWfB/KO4KCghAQEABvb29Uq1YNCxYswNu3b5Wr+XTv3h0ODg7KeQEtW7bE/PnzUalSJfj4+ODBgweYMGECWrZsqTLi5ktEJf0TJ05EQEAAoqKioFAosHv3bty9excbNmzA/v37xZySRLJ0csCr5zFq5YemLcTdY2cAAJsCh2Nq5DlUbOOHqzsP4NXzGMyq3DzL875LSAIAFLK2gMzAAK9j4lT2v46Jg13pkpkeb1rUBsmx6scUKar6k+ur57GwcLSHRCLR6tsqUXZk9rnZP2Uhbv//c7MuYDhmPvsbldr44fL/PzfTvbJeKeHt/z83hf//uUlW+9y8RFGPL3xuMjjGtKi1Stmr5zH83JDOvXgRDVtra7Xyvr17wbe6DwBg+uSJaNSsJY6fOAm/Rg1hY22NnVs2ZnneT4l1YlIS5HI5rCwtVfZbWVoi4vGTLM/x+s0bNGzaAulpaZDKZBg3aoQyJgBYs34DDGQydOnYQaNrjXz6FFu370DQ0MFq+2ysbfD8RcYTKol0oUOHDnj58iUmTpyI6OhoeHl5ISQkRDm5NzIyUqVnf/z48ZBIJBg/fjyioqJgY2ODli1bYvr06Vq1Kyrpb926Nfbt24epU6eiUKFCmDhxIipXrox9+/ahUaNGYk5JIhkWMMGHlDS18sfnripfv0t8hdi7j2D3/2RDIZcj7mHW/4H9WtLfp0Aqk8HA2AjpKRkvb0Wka4YFTDL8e3t07ory9bvEV4i++whFy3wcyqiQy/GSnxvKx1JTU2FkbKRWXrGCp/K1mZkZnJ2c8CjiMQDAwMAAJRwdczy2QgULYueWjXj37j3OX7yIuT8vRHEHB1T1roJbt29j87bt2L5pg0YT3GNiY9Fv0FA0atgA37XxV9tvYmwsajlyIm0MHDgQAwcOzHDfyZMnVd4bGBhg0qRJmDRpUrbaFJX0A0Dt2rVx9GjWP3VTznsbl4ACFqZfrvgZbYb3vI1LhPzDBxSxU+39KWJnjdcxLzM9Pjn6JUxtMzgmWvWYgpZmSH3zlokLfVVv4xJQ0MJMq2O0Gd7z5v+fG1O1z40NkqO/8LnJ8BjV3v+CluZI4eeGdMzc3BzJya+1Okab4T0W5uaQyWRqk3bjExJgbWWZydEfSaVS5ZcLj9LueBTxGKvXrUdV7yq4fDUMCQmJ8GvRWllfLpdj3oJF2Lx1O0L27VGWx758iV59+6NiBU9MGpfx6iqvkpNhYWGeZTyUO0lkXNUsK6KTfuDjAwZiY2PVZg6XKFEiW0GR5p5dvQXvLv5q5c7VKyHx6cd1lguYm8LG3QUxdz5OvtJmeI88PR1PL9+Ae/2auP7Hx4RHIpGgdP0aOLVkQ6bHP/77Ktwb1MTJRWuVZR4NayHi7ysq9ezLlcazq7e+eJ1EuvT06k1U69pGrdy1eiVc/v/npqC5KezcXRB9+wEAaDW8R56ejsjLN+DRoAau/fHxC7ZEIoFHgxo4+Uvmn5tH567Co0ENHF+4RllWplEtlV8gAKBYeXc8vZr1ModE2vIo7Z7hcpXh12/A/v+riiQnJ+NxZCRcXZwBQKvhPYaGhijj4YHzFy6ift06AD6OxT9/8SI6tc98YYiMCAoF0tI+LhzSslkzVK9WTWV/v0FD0KJZU7Ru2UJZFhMbi159+6OMhwemTZqgNjES+Phrx9Nnz+BRurRW8RDlBaKS/vv376Nnz544e/asSrkgCJBIJJDL5ToJjr7s9pFTaDljBAqYm+J9UrKy3G/8ILyNT8TrmDi0+OlHvI1LRPiej8mHtsN7TixYha5r5+Hp5XA8uXANdYf0hFGhgji/bpeyTtd18/AqKhr7xs0BAPy5aC0Gn9iGesN64ebB46jSoSUcvT2xra/qgyRK1qqKO0dPZ+cWEGnt5uFT8A8eiYLmpnj32eem2cTBePP/z03r6SPwJi4BYZ99brQZ3nNs/ioErp+HJ5eu4/GFMNQf+j2MChXE2bU7lXUC189DUlQM9oydDQA4vnANhv+5HQ2DeuH6gROo2rElnLw9sbmPao9kqdrVcPsIPzekWzV8q2PRL0uRnJwM088muC7/dTXMzMxgZWmJxUuXwcLcXJm0azu8p3uXThg/eSrKli0Dz3JlsWnLNrx/nwL/z5LzsRMnw87WBkMGfpyou2rtOpQrUwaOxYsjLT0Np/86i/0HD2HcmFEAAHNzM5ibq/5yZ2BgACsrS7g4f1xwJCY2Ft//0A/29vYYPnQwEhOTlHWtrf9ZRjf8+g0YGRmpDGkiyi9EJf2BgYEwMDDA/v37YW9vz4fE6NGLG3fx9MpNVGrfHGdX/vMgkn1jZ6PtzxNhW8oZz8JuY2XrXpCLXE716o4DKGxthWaTg2Ba1BrPwm5jWbNAvP5soq6FYzEIn/3iE3HuCtZ3HYrmU4ej5fQfEXv/MVa1/QEvbt5T1jErZgeXGpWxsfswUXERifX8xl1EXrmJKu1b4PTKLcryPaNnof3CScrPzdKW4j83l3fsRxEbS7ScOgymRW3wLOw2FjcJUPncWJZwgKD4ZyLuo3NXsLrzELT6aThazxiB2PuPsdy/D55/9rkxL2YH1xqVsabrUFFxEWXG3c0NHh4eOHz0GNp921ZZPnTQAMyaOx+RT5+itLs7Fv88F4aGhqLaaNK4ERITk7B0+UrExcejtLs7li1eACurfxLv6OgYlV749+9TMH3WbMTEvoSxsTFcnJ0wY9oUNGms+RzCv89fQOTTZ4h8+gyNmqkudxt+6bzy9aHDR9C8iR8KmJiIuj6i3EwiiFj6oVChQrh8+TI8PNSX9hJrsMxFZ+f6rynbrB78Z41BcAU/lPymGgYf34ZRlhXw/pV2YzO/tlbBo1DAwgzb+2r3GGlSlabg6i1ilG9WD9/OGYup5Ruj1Dc+CDq5DcPMK+D9q+QvH6xHbWaORkELU2z+gZ+b7FiYHKbvEHKlU2fOYP7Cxdi9fSsuX7mK7/v2x5kTx2BapIi+Q8txiUlJaPVtO2zdsB7FHXLuicR5nXERc32HkKkDjhX00m7zp+F6aVdbotfpj4uL+3JF+ipuHTwBWzdnmDlk/CS33Op1bDxO/Gstf6Kv5cbBE7At5QLzPPe5icOx+av0HQblU9/UqoUnkU8RG5v5hPP86vnzFxg3aiQT/jxMyom8WdI46U9O/qf3a9asWRg5ciRmzJgBT09PtZ/5TE21W02Gsu/ThFnrknlnEvWJn5m4kH59mjBrU9LpCzVzDyb8lNO6de4EAP+5p9KWK1sG5cqW0XcYRDlG4+E9UqlUZez+p0m7n8vORF4O7yESh8N7iLTH4T1E4uTm4T0hJSvppd0mD69+uVIuoHFP/4kTJ3IyDiIiIiIiyiEaJ/116tTJyTiIiIiIiCiHqD+ZgoiIiIiI8hXRSb+LiwsaNVJdI7dhw4ZwdXXNdlBERERERKQ7opbsBICAgADY2NiolLVp04ZLeRIRERHRVyfhkp1ZEvVwrpzA1XuIxOHqPUTa4+o9ROLk5tV7DrtX1ku7fveu6KVdbXFMPxERERFRPidqeI9cLse6desQGhqK2NhYKBQKlf3Hjx/XSXBERERERJR9opL+IUOGYN26dWjevDnKly+v9pAuIiIiIiLKPUQl/du2bcOOHTvQrFkzXcdDRERERKQ1KSfyZknUmH4jIyO4ubnpOhYiIiIiIsoBopL+4cOHY+HChcglC/8QERER0X+cRCrVy5ZXaDy8p23btirvjx8/jkOHDqFcuXIwNDRU2bd7927dREdERERERNmmcdJvZmam8r5NmzY6D4aIiIiIiHRP46R/7dq1ORkHERERERHlEFEDkerXr4+kpCS18uTkZNSvXz+7MRERERERkQ6JWrLz5MmTSEtLUytPSUnB6dOnsx0UEREREZE2uGRn1rRK+sPDw5Wvb926hejoaOV7uVyOkJAQODg46C46IiIiIiLKNq2Sfi8vL0gkEkgkkgyH8RQoUACLFy/WWXBERERERJR9WiX9EREREAQBrq6uuHDhAmxsbJT7jIyMYGtrC5lMpvMgiYiIiIhIPK2SficnJwCAQqHIkWCIiIiIiEj3NE769+7dq/FJW7VqJSoYIiIiIiIxJJzImyWNk35/f3+V9xKJBIIgqLz/RC6XZz8yIiIiIiLSCY3X6VcoFMrtyJEj8PLywqFDh5CUlISkpCQcPHgQlStXRkhISE7GS0RERESkRiKT6mXLK0St0z906FAsX74ctWrVUpb5+fmhYMGC6NOnD27fvq2zAImIiIiIKHtEfT15+PAhzM3N1crNzMzw+PHjbIZERERERES6JCrpr1q1KoKCghATE6Msi4mJwYgRI1CtWjWdBUdERERERNknKulfs2YNXrx4gRIlSsDNzQ1ubm4oUaIEoqKisHr1al3HSERERERE2SBqTL+bmxvCw8Nx9OhR3LlzBwBQpkwZNGzYUGUVHyIiIiKir0HKJTuzJCrpBz4u0dm4cWM0btxYl/EQEREREZGOaZz0L1q0CH369IGJiQkWLVqUZd3BgwdnOzAiIiIiItINifD5E7ay4OLigkuXLsHKygouLi6Zn1AiwaNHj7QOZLAs83MSUebSFBp9hInoMwuTw/QdAlGeZFzEXN8hZOqUTw29tPvN+bN6aVdbGvf0R0REZPiaiIiIiIhyN1Fj+h89egRXV1ddx0JEREREJIpEyom8WRG9ek/x4sVRp04d1K1bF3Xq1IGbm5uuYyMiIiIiIh0QtU7/06dPERwcjAIFCmD27Nlwd3dH8eLF0aVLF6xatUrXMRIRERERZUkqk+plyys0nsiblfv372P69OnYvHkzFAoF5HK51ufgRF4icTiRl0h7nMhLJE5unsj7V63aemm35pnTemlXW6KG97x79w5nzpzByZMncfLkSVy9ehUeHh4YOHAg6tatq+MQiYiIiIgoO0Ql/ebm5rCwsECXLl0wevRo1K5dGxYWFrqOjYiIiIiIdEBU0t+sWTOcOXMG27ZtQ3R0NKKjo1G3bl24u7vrOj4iIiIiIsomUbMP9uzZg7i4OISEhMDX1xdHjhxB7dq14eDggC5duug6RiIiIiKiLElkEr1seYWonv5PPD098eHDB6SlpSElJQWHDx/G9u3bsXnzZl3FR0RERERE2SSqp3/+/Plo1aoVrKys4OPjg61bt8Ld3R2//fYbXr58qesYiYiIiIgoG0T19G/duhV16tRBnz59ULt2bZiZmek6LiIiIiIi0hFRSf/Fixc1qte/f39MnToV1tbWYpohIiIiIiIdyNHHiG3atAnJyck52QQRERERESfyfkGOJv06eNgvERERERFlU7ZW7yEiIiIiyg2kshzty87zeHeIiIiIiPI5Jv1ERERERPkck34iIiIionwuR5P+rl27wtTUNCebICIiIiKiLxA9kTcxMRGrV6/G7du3AQBlypRBz549YWlpqayzbNmy7EdIRERERPQFeWn5TH0Q1dN/6tQpuLi4YNGiRUhMTERiYiIWL14MFxcXnDp1StcxEhERERFRNojq6R8wYADat2+PZcuWQSaTAQDkcjn69++PAQMG4Pr16zoNkoiIiIiIxBPV0//gwQMMHz5cmfADgEwmQ1BQEB48eKCz4IiIiIiIKPtEJf2VK1dWjuX/3O3bt1GxYsVsB0VERERERLojanjP4MGDMWTIEDx48ADVq1cHAPz9999YsmQJZs6cifDwcGXdChUq6CZSIiIiIqJMSKWcyJsViSAIgrYHSaVZ/0AgkUggCAIkEgnkcrlG5xwsc9E2DCICkKbQ+iNM9J+3MDlM3yEQ5UnGRcz1HUKmLjVroJd2vQ+G6qVdbYnq6Y+IiNB1HEREREREoklkfOZsVkQl/U5OTrqOg4iIiIiIcojGSf/evXvRtGlTGBoaYu/evVnWbdWqVbYDIyIiIiIi3dA46ff390d0dDRsbW3h7++faT1txvETEREREVHO0zjpVygUGb4mIiIiIqLcTdSYfgAIDQ1FaGgoYmNjVb4ESCQSrF69WifBERERERFpQirjkp1ZEZX0T5kyBVOnToW3tzfs7e0hkfAmExERERHlVqKS/uXLl2PdunXo1q2bruMhIiIiIiIdE7WgaVpaGmrUqKHrWIiIiIiIKAeISvp79eqFLVu26DoWIiIiIiLKARoP7wkKClK+VigUWLlyJY4dO4YKFSrA0NBQpe78+fN1FyERERER0RdIOJE3Sxon/VevXlV57+XlBQC4ceOGSjkn9RIRERER5S4aJ/0nTpzIyTiIiIiIiCiHiBrTT0REREREeQeTfiIiIiKifE70E3mJiIiIiHILiYx92Vnh3SEiIiIiyufY009EREREeZ6US3ZmiT39RERERET5HJN+IiIiIqJ8jkk/EREREVE+x6SfiIiIiCif40ReIiIiIsrzJFJO5M0Ke/qJiIiIiPI5Jv1ERERERPkck34iIiIionyOST8RERER0Ve0ZMkSODs7w8TEBD4+Prhw4UKW9ZOSkjBgwADY29vD2NgY7u7uOHjwoFZt5pqJvIvkEfoOgShPGixz0XcIRHnOgzfs8yISo1wRfUeQOaksb3yut2/fjqCgICxfvhw+Pj5YsGAB/Pz8cPfuXdja2qrVT0tLQ6NGjWBra4tdu3bBwcEBT548gbm5uVbt5pqkn4iIiIgov5s/fz569+6NHj16AACWL1+OAwcOYM2aNRg9erRa/TVr1iAhIQFnz56FoaEhAMDZ2VnrdvPGVyIiIiIioixIZBK9bKmpqUhOTlbZUlNTM4wxLS0Nly9fRsOGDZVlUqkUDRs2xLlz5zI8Zu/evfD19cWAAQNgZ2eH8uXLY8aMGZDL5VrdHyb9REREREQiBQcHw8zMTGULDg7OsG5cXBzkcjns7OxUyu3s7BAdHZ3hMY8ePcKuXbsgl8tx8OBBTJgwAfPmzcNPP/2kVZwc3kNEREREJNKYMWMQFBSkUmZsbKyz8ysUCtja2mLlypWQyWSoUqUKoqKiMGfOHEyaNEnj8zDpJyIiIiISydjYWOMk39raGjKZDDExMSrlMTExKFq0aIbH2Nvbw9DQEDKZTFlWpkwZREdHIy0tDUZGRhq1zeE9RERERERfgZGREapUqYLQ0FBlmUKhQGhoKHx9fTM8pmbNmnjw4AEUCoWy7N69e7C3t9c44QeY9BMRERFRPiCRSfWyaSsoKAi//vor1q9fj9u3b6Nfv354+/atcjWf7t27Y8yYMcr6/fr1Q0JCAoYMGYJ79+7hwIEDmDFjBgYMGKBVuxzeQ0RERET0lXTo0AEvX77ExIkTER0dDS8vL4SEhCgn90ZGRkIq/efLhKOjIw4fPoxhw4ahQoUKcHBwwJAhQzBq1Cit2pUIgiDo9EqI6Kviw7mItPfDs2v6DoEoTypnb6rvEDJ1p5e/Xtr1WLVHL+1qi8N7iIiIiIjyOSb9RERERET5HMf0ExEREVGeJ5GyLzsrvDtERERERPkce/qJiIiIKM+Tilg+87+Ed4eIiIiIKJ9j0k9ERERElM8x6SciIiIiyueY9BMRERER5XOcyEtEREREeZ6EE3mzxLtDRERERJTPMeknIiIiIsrnmPQTEREREeVzTPqJiIiIiPI5TuQlIiIiojyPE3mzxrtDRERERJTPsaefiIiIiPI8iZR92Vnh3SEiIiIiyueY9BMRERER5XNM+omIiIiI8jnRY/qTkpJw4cIFxMbGQqFQqOzr3r17tgMjIiIiIiLdEJX079u3D126dMGbN29gamoKiUSi3CeRSJj0ExEREdFXJZHJ9B1CriZqeM/w4cPRs2dPvHnzBklJSUhMTFRuCQkJuo6RiIiIiIiyQVTSHxUVhcGDB6NgwYK6joeIiIiIiHRMVNLv5+eHS5cu6ToWIiIiIiLKAaLG9Ddv3hwjRozArVu34OnpCUNDQ5X9rVq10klwRERERESUfaKS/t69ewMApk6dqrZPIpFALpdnLyoiIiIiIi1IZFyJPiuikv5/L9FJRERERES5l+h1+omIiIiIcguplD39WRGd9L99+xZ//vknIiMjkZaWprJv8ODB2Q6MiIiIiIh0Q1TSf/XqVTRr1gzv3r3D27dvYWlpibi4OBQsWBC2trZM+omIiIiIchFRv4MMGzYMLVu2RGJiIgoUKIC///4bT548QZUqVTB37lxdx0hERERERNkgKukPCwvD8OHDIZVKIZPJkJqaCkdHR8yePRtjx47VdYxERERERJQNoob3GBoaKidL2NraIjIyEmXKlIGZmRmePn2q0wCJiIiIiL6ES3ZmTVTSX6lSJVy8eBGlSpVCnTp1MHHiRMTFxWHjxo0oX768rmMkIiIiIqJsEPWVaMaMGbC3twcATJ8+HRYWFujXrx9evnyJlStX6jRAIiIiIiLKHlE9/d7e3srXtra2CAkJ0VlARERERESkWxz8RERERESUz4lK+mNiYtCtWzcUK1YMBgYGkMlkKhsRERER0dckkUn1suUVoob3BAYGIjIyEhMmTIC9vT0kEomu4yIiIiIiIh0RlfSfOXMGp0+fhpeXl47DISIiIiLSnkSad3rd9UHU3XF0dIQgCLqOhYiIiIiIcoCopH/BggUYPXo0Hj9+rONwiIiIiIhI10QN7+nQoQPevXuHkiVLomDBgjA0NFTZn5CQoJPgiIiIiIgo+0Ql/QsWLNBxGERERERElFNEJf0BAQG6joOIiIiISLS8tHymPvDuEBERERHlc6KTfkEQ0KpVK6xevVqX8RARERERkY6JTvolEgmmT5+OiRMn6jIeIiIiIiLSsWwN7/H09MTLly8RGxurq3iIiIiIiEjHRE3k/eTWrVswNDSEhYWFruIhIiIiItIaJ/JmLVt3Z+PGjWjdurXaOv1ERERERJR7ZKun//fff8ecOXN0FQsRERERkShS9vRnKVt3JzExEa6urrqKhYiIiIiIckC2kv6GDRvi4MGDuoqFiIiIiIhyQLaS/mnTpuHMmTNITk7WVTxERERERKRj2RrT7+rqij/++ENXsRARERERUQ4Q1dO/YcMGpKamqpWnpaVhw4YN2Q6KiIiIiEgbEqlUL1teISrSHj164NWrV2rlr1+/Ro8ePbIdFBERERER6Y6opF8QBEgkErXyZ8+ewczMLNtBERERERGR7mg1pr9SpUqQSCSQSCRo0KABDAz+OVwulyMiIgJNmjTReZBERERERCSeVkm/v78/ACAsLAx+fn4oXLiwcp+RkRGcnZ3x7bff6jRAIiIiIiLKHq2S/kmTJgEAnJ2d0aFDB5iYmORIUERERERE2pDwibxZErVkZ0BAAICPq/XExsZCoVCo7C9RokT2IyMiIiIiIp0QlfTfv38fPXv2xNmzZ1XKP03wlcvlOgmOiIiIiIiyT1TSHxgYCAMDA+zfvx/29vYZruRDRERERES5g6ikPywsDJcvX4aHh4eu4yEiIiIiIh0TlfSXLVsWcXFxuo6FiIiIiEgUTuTNmsZ3Jzk5WbnNmjULI0eOxMmTJxEfH6+yLzk5OSfjJSIiIiIiLWnc029ubq4ydl8QBDRo0EClDifyEhEREZE+SKTs6c+Kxkn/iRMncjIOIiIiIiLKIRon/XXq1MnJOIiIiIiIKIeImsgbHh6eYblEIoGJiQlKlCgBY2PjbAVGRERERES6ISrp9/LyynJtfkNDQ3To0AErVqyAiYmJ6OCIiIiIiCj7RM14+P3331GqVCmsXLkSYWFhCAsLw8qVK1G6dGls2bIFq1evxvHjxzF+/Hhdx0tEREREpEYqk+llyytE9fRPnz4dCxcuhJ+fn7LM09MTxYsXx4QJE3DhwgUUKlQIw4cPx9y5c3UWLBERERERaU9UT//169fh5OSkVu7k5ITr168D+DgE6MWLF9mLjoiIiIiIsk1U0u/h4YGZM2ciLS1NWZaeno6ZM2fCw8MDABAVFQU7OzvdRElERERERKKJGt6zZMkStGrVCsWLF0eFChUAfOz9l8vl2L9/PwDg0aNH6N+/v+4iJSIiIiIiUUQl/TVq1EBERAQ2b96Me/fuAQDatWuHzp07o0iRIgCAbt266S5KIqIcVrtfN9T/sQ9Mi9og6tpt7BoyGZEXr2VYV2pggMaj+6Fa929h5lAUsXcfYe+Ymbh9+JSyzqSHp2HlXFzt2NNLN2LnoIlq5X0PrEXZJnXxa9s+uP7HUQBAsQpl0GhUX7jW9EYha0skPH6Gv1Zsxp+L1+nmool0QBAEbFu7Akf378G7N2/gUb4C+gSNRrHiJTI95ua1K/hj20Y8vHcHifFxGDVtDnxq11Wp8/7dO2xa+QvOn/kTb5Jfwda+GJq37QC/1t9mGMNPo4bg6oVzKuc6fmgffpk1NcMY1vx+GOYWlqKvm3IfiYxP5M2KqKQfAIoUKYK+ffvqMhYiIr2o1L452swbh+39x+PJ+TDUGdIT/Q+tx09lGuDNy3i1+i2mDYd3F39s+2EMYu48RBm/b/D9byuwoNa3eBZ2CwAwz6e1yv8B2ZcvjYFHNuHqrgNq56s7pCcEQVArL1GlPF7HxmNj9yAkPn0OlxpV0HH5DCjkCpxeukGHd4BIvN+3bsCB37Zj8JjJsLUvhq1rlmPaiEFYuG4HjDJ5Zk9qyns4l3RH/WatMHvCyAzrrFv6M65fuYSh46bCtqg9wi79jZU/z4aFtTWq1VR9YOj+XVszXEq8Zv1GqFTNV6Vs8cwpSE9LY8JP/zkaJ/179+5F06ZNYWhoiL1792ZZt1WrVtkOjIjoa6k3tBfOrtqO8+t2AQB29BuHcs3qoXqPdjg2e7la/apd2+DIjCW4degkAODM8s1wb1AT9YJ6Y2P3YQCAN3EJKsc0GtUPLx88xoM/z6uUO1Qsg/pBvTCnWitMf35RZd/fa3eqvI+PeAqX6pVRsY0fk37KFQRBwP5dW/Fdt56oVutjIj54zBT0bOOHC2f+RK0GjTM8rrJPTVT2qZnlue/cCEfdJs1RvlIVAEDjlm1xZN/veHD7lkrSH3H/Lv7YvhlzVqzH9982VTmHsbEJjI3/eV7Qq6RE3Lh6Cf1HThB1vZS7sac/axon/f7+/oiOjoatrS38/f0zrSeRSCCXy3URGxFRjpMZGsKxSnkcnbVUWSYIAu6G/gUX38oZHmNgbIT01FSVsvT3qXCt6Z1pG95d/HHi59Uq5YYFTBCwaSF2DpqE1zFxGsVrYlYE7xJfaVSXKKfFvIhCUkI8KlappiwrVLgwSpUth7u3wjNN+jXhUb4CLv51Cg2atoKltQ1uhF3G86eR6DFgmLJOakoKfv5pAvoMHQkLK+svnvPk4QMwMjaBb536ouMiyqs0TvoVCkWGr4mI8rJC1haQGRioJd2vY+JgV7pkhsfcPnIK9YZ+j4enLiDu4RO4N6iJim38IM2kl6mCf2MUMDfF+fW7VMrbzp+AiHNXcH3vUY1idfGtjMrtm2NFy+81qk+U05ISPg5/M7O0Uik3t7BCYoL60Dht9Bo8AsvmzUDvds0hk8kgkUrR78dxKFfxny/ja5bMR+lyFZS/MnxJ6MG9qN3QT6X3n+i/QvSY/k9SUlJgYsIPDxH9d+weOhUdVwZj3K1jEAQBcQ8jcX7dLvj0aJdh/eo92+N2yJ9IfhGrLCvfsiFK1fPF7CotNGrTvpw7ev++EoemLsKdo6d1ch1E2vrz6CGsmBesfD9u5s851taB3dtx79Z1jJkxDzZ29rh17Sp+XTAbllbWqOjtgwt//YkbVy5h7q+bNDrf3ZvhePYkAkPGTsmxmIlyM1FJv1wux4wZM7B8+XLExMTg3r17cHV1xYQJE+Ds7Izvv2cvFBHlDW/jEiH/8AFF7FSHBhSxs8brmJcZHvMmLgGr2v4AA2MjFLKywKvnMWgVPArxjyLV6lqUcEDpBjWx+rt+KuXu9XxhXdIJsxJUVwj6fucyPDx9EYsbdFKWFS3jhgFHN+OvX7fhyIxfxF4qUbZVq/kN3MuUV75PT//4vJ5XCfGw/Gx4TVJiPFzc3EW3k5qagi2rlmLktDnw9q0FAHAuWQoRD+7hj+2bUNHbB9evXEL082fo1kJ1qM6cSaNQxtML0xauUCk/duAPuLi5o2TpMqLjIsrLRCX906dPx/r16zF79mz07t1bWV6+fHksWLCAST8R5Rny9HQ8vXwD7vVrKpfKlEgkKF2/Bk4tyXqy7IfUNLx6HgOpgQEqtm2CqzvVV+apHvgdXsfG4+aB4yrlR2ctw7nV21XKxoQfxu6gn3Bj/zFlWdGypTDw2BZc2PAbDkyYK/YyiXSiQMFCKFCwkPK9IAgwt7RC+JWLcClVGgDw7u0b3L91E01afSe6HfmHD/jw4QOkUtUVeaQyqXKlq7adA9CweWuV/cN6dkKPAcPgXaO2Svn7d+/w14lj6Np7gOiYKPeTSDmRNyuikv4NGzZg5cqVaNCggcqynRUrVsSdO3d0FhwR0ddwYsEqdF07D08vh+PJhWuoO6QnjAoVVK7m03XdPLyKisa+cXMAAE7VvGDmYIeosFswcyiKphOHQCKVInSOas+iRCKBT2A7XNjwGxT/WuDgdUxchpN3E59GIeHxMwAfh/QMPLYZd46cxomfVyl/jRDkCrXVgYj0QSKRoMV3nbBr4xrYF3eEnb0Dtq5eDktra5Vx9pOC+sGnVj00a9sewMckPDrqqXJ/bPRzRNy/i8KmZrCxK4qChQqjXMXKWL9sEYyMTGBTtChuhl3Bn4cPInDAUACAhZV1hpN3rW2Lws7eQaXsrxNHoZDLUadRU7X6RP8VopL+qKgouLm5qZUrFAqkp6dnOygioq/p6o4DKGxthWaTg2Ba1BrPwm5jWbNAvI79mJRbOBaD8NkCBoYmxmgxdTisXEsg9c1b3Dp0EhsDgvD+1WuV85ZuWAuWTg5qS29qyuvbpihia42qXdugatc2yvL4x88wpWTtLI4k+nradOqO1JT3WD53Bt6+eYMynhUxYfYilTX6o6OikPwqSfn+4d3bmDjsn07DtUs+zg2o59ccg8ZMBgAETZyOTb8uwYLpE/AmORk2dkXRuVc/+LVSfzjXl4Qe/AM+39RFof8/QJTov0giZPREmC+oUqUKhg0bhq5du6JIkSK4du0aXF1dMXXqVBw9ehSnT3OSGdHXMljmou8QiPKcH55l/LRlIspaOXtTfYeQqeS16k87/xpMe2T81OfcRlRP/8SJExEQEICoqCgoFArs3r0bd+/exYYNG7B//35dx0hERERERNkgasZD69atsW/fPhw7dgyFChXCxIkTcfv2bezbtw+NGjXSdYxERERERFmSyKR62fIK0ev0165dG0ePavZAmX9LTU1F6r+eZmlsbAzjz8b/ERERERGRbmTr60laWhqePXuGyMhIle1LgoODYWZmprIFBwd/8TgiIiIioozkpZ7+JUuWwNnZGSYmJvDx8cGFCxc0Om7btm2QSCTw9/fXuk1Rkd6/fx+1a9dGgQIF4OTkBBcXF7i4uMDZ2RkuLl+eVDhmzBi8evVKZRszZoyYUIiIRCloaY7pLy7C0snhy5VzkaC/dqNi2yb6DoP+w16/SkKgf2PEvniu71C0MqpfD5z78/iXKxLlsO3btyMoKAiTJk3ClStXULFiRfj5+SE2NjbL4x4/fowff/wRtWuLW71NVNIfGBgIqVSK/fv34/Lly7hy5QquXLmCq1ev4sqVK1883tjYGKampiobh/YQ0dfkN3YAru89hoQnUTnWRu1+3TDp4WnMe3sHQWd/R4mqFb94jNd3zTDu5jHMe3sHo8MOoWzTuir7D8/4BS1njIREIsn4BEQ5bNemtahW8xvY2hfLsTYO/b4DP3RohQ6NamJUv0Dcv33zi8ecPXkMg7p9hw6NamJoj464/PdfKvvbdeuJTSt/geKz5XeJ9GH+/Pno3bs3evTogbJly2L58uUoWLAg1qxZk+kxcrkcXbp0wZQpU+Dq6iqqXVFJf1hYGFasWIGmTZvCy8sLFStWVNmIiHIzwwImqN6zPf5es/3LlUWq1L452swbh5BpCzHHuwWiwm+j/6H1KGxjlekxLr6VEbB5Ic6t2YHZVZoj/I+j6LV7BezLuSvr3Dp0EiZFCqPMv74MEH0NqSkpCD34Bxo0a/3lyiKdOX4Ea5cuQPvAXpj760Y4lyyFqSMGISkx8wfS3blxDfOnjkeD5q0xb9UmVKtVB7PG/4gnjx4o61TyqYH3797iyvmzORY7/TelpqYiOTlZZfv33NVP0tLScPnyZTRs2FBZJpVK0bBhQ5w7dy7TNqZOnQpbW1t8//33ouMUlfSXLVsWcXHqT5IkIsoLyjWrhw+paXh8PgwA4FbHB4vkESjbrB5GXT30sWf+r90qyba26g3thbOrtuP8ul2Ivv0AO/qNQ9q796jeo12mx9QZ3AO3D/+J4/NWIubOQxycNB/PrtxE7QHdlXUEhQK3Dp1AlQ4tRcdGJNblv/+CgaERSpfzBADcuHoZbetWxaVzZzCsZ6f/98z3UEm2tbVv5xY0au6PBk1bwdHZFT8EjYGxiQmOH9yb6TH7f9uGStV84d+xG4o7uaDz9/3gUsoDh37/58F4MpkMlavXxF/Hj4iOjSgj2sxVjYuLg1wuh52dnUq5nZ0doqOjMzzmzJkzWL16NX799ddsxanx6j3JycnK17NmzcLIkSMxY8YMeHp6wtDQUKWuqWnufXADEZFrrap4euWGWrn/rDH4bdhUJEe/RMvpI9Dnj1WY5lEfig8fYOFYDGNvZJ0sHAleiqMzl0JmaAjHKuVxdNZS5T5BEHA39C+4+FbO9Hjn6pVw8ufVKmW3j5xChdaNVcqeXLyGhiP7aXKpRDp1+/pVlHT3UCvfsHwReg4aDgtLK2z+dQmCxw7HL5t+g4GBAV7GRGNIQPssz9u2aw9817UH0tPT8fDuHbTtHKjcJ5VKUaFKNdy9dT3T4+/dvI6W7TqrlFWqVh3nz/ypUlbKoxx2b1mvwZVSXiSR6mf5zDFjRiEoKEilTFfD1l+/fo1u3brh119/hbW1dbbOpXHSb25urjKGVBAENGjQQKWOIAiQSCSQy+XZCoqIKCdZOjng1fMYtfJD0xbi7rEzAIBNgcMxNfIcKrbxw9WdB/DqeQxmVW6e5XnfJSQBAApZW0BmYIDXMaq/iL6OiYNd6ZKZHm9a1AbJserHFClqo1L26nksLBztIZFIIOKh6kSivYyOhqW1jVp5+4Be8PL2AQAMGjMZvds1x/nTJ1CzXiNYWllj3qrNWZ638P87C1+/SoJCIYe5paXKfnMLS0RFPs70+KSEeJhbqg6dM7OwRFJCvEqZhbU14l/GQKFQQKqnBJHyH22Wnbe2toZMJkNMjOr/B8XExKBo0aJq9R8+fIjHjx+jZct/ft39NC/FwMAAd+/eRcmSmf//yuc0TvpPnDihaVUiolzNsIAJPqSkqZU/PndV+fpd4ivE3n0EO4+P/zFVyOWIe/jkq8WYlfT3KZDKZDAwNkJ6SsbjRolyQlpaKgyNjNTKS5eroHxdxNQMDo5OePbkMQBAZmAA++KOXyvELBkZm0ChUCA9PQ3Gxib6Dof+g4yMjFClShWEhoYql91UKBQIDQ3FwIED1ep7eHjg+nXVX7nGjx+P169fY+HChXB01PyzpXHSX6dOHY1PSkSUm72NS0ABC+2GIWozvOdtXCLkHz6giJ3qT7FF7KzxOuZlpscnR7+EqW0Gx0SrHlPQ0gypb94y4aevroiZGd6+fq3VMdoM7yliZg6pVIakBNVJu0mJCWo9+Z8zt7RS69V/lcExb5JfwcSkABN+0qugoCAEBATA29sb1apVw4IFC/D27Vv06NEDANC9e3c4ODggODgYJiYmKF++vMrx5ubmAKBW/iWin8hLRJRXPbt6C95d/NXKnatXQuLTj2uPFzA3hY27C2LuPAQArYb3yNPT8fTyDbjXr4nrf3x8crlEIkHp+jVwasmGTI9//PdVuDeoiZOL1irLPBrWQsTfqksh25crjWdXb33xOol0zcWtNE4dPaRWfu/WddjYfRya8OZ1Mp4/i0RxJ2cA0Gp4j6GhIUqW9kD4lYvwqV0XwMde0PDLF9GsTeaT4N3LeeL6lYsq4/qvXTqP0mU9VepFRjyES6nSX7xOopzUoUMHvHz5EhMnTkR0dDS8vLwQEhKinNwbGRmZI8PPRCf9Li4ucHNzw9GjR5VlDRs2xKNHj/Do0SOdBEdElBNuHzmFljNGoIC5Kd4n/bNIgd/4QXgbn4jXMXFo8dOPeBuXiPA9H3v3tR3ec2LBKnRdOw9PL4fjyYVrqDukJ4wKFcT5dbuUdbqum4dXUdHYN24OAODPRWsx+MQ21BvWCzcPHkeVDi3h6O2JbX3Hqpy7ZK2quHP0dHZuAZEolar5YvOvS/DmdTIKF/nn17Id61ehiKkZzCwssWX1MhQxM0e1WnUBaD+8p2W7zlgcPAVupcugVJly2LdrK1JT3qN+03/GNC+cMQlW1jbo2ufjcIgW33bEhCE/4I/tm1Clei2cOX4ED+/eRt/hqp+d2+FhqFjVJxt3gHIziVSm7xA0NnDgwAyH8wDAyZMnszx23bp1otoUnfQHBATAxkZ1Mk+bNm24lCcR5XovbtzF0ys3Ual9c5xduVVZvm/sbLT9eSJsSznjWdhtrGzdC/L0dFFtXN1xAIWtrdBschBMi1rjWdhtLGsWiNefTdS1cCwG4bMHBUWcu4L1XYei+dThaDn9R8Tef4xVbX/Ai5v3lHXMitnBpUZlbOw+TFRcRNnh5OoGV3cP/HXiGPxatVWWd+szEKsXz8OLqKdwcXPH2Bnz1Vb201St+o2RnJSErWtXICkhHi5u7pgwe5HKUJ24mGhIP1tcxKN8RQyb8BO2rF6GzauWwt7BEaN+mgsnVzdlnfiXsbh7MxxDxk0VFRdRXicRuPQDUZ42WOai7xDypLLN6sF/1hgEV/BDyW+qYfDxbRhlWQHvX2k3XvlraxU8CgUszLD9X73/pJ0fnl3Tdwh51qVzZ7Bh+SIsWLsNt65dxcRhfbFx33EUKlJE36FlacOKxXj7Ohn9fhyn71DytHL2uXdZ9nc7Z+ul3YLtRuqlXW1xTD8R/SfdOngCtm7OMHNQXyItN3sdG48T/1rLn+hr8vathRfPIpEQF6vvULRiZm6BVv9ay5/ov0RU0i+Xy7Fu3TqEhoYiNjZWuV7oJ8ePH9dJcEREOenThFnrkiX0HInmTvy8St8hECknzEZHRek5Es217tBV3yEQ6ZWopH/IkCFYt24dmjdvjvLly6s8tIuIKK958Od5DpMiEqF8pSrYffKivsMgIg2ISvq3bduGHTt2oFmzZrqOh4iIiIiIdExU0m9kZAQ3N7cvVyQiIiIi+hpyYG37/ETU3Rk+fDgWLlwILvxDRERERJT7adzT37ZtW5X3x48fx6FDh1CuXDm1tXh3796tm+iIiIiIiCjbNE76zczMVN63adNG58EQEREREZHuaZz0r127NifjICIiIiKiHCJqTH/9+vWRlJSkVp6cnIz69etnNyYiIiIiIq1IZDK9bHmFqKT/5MmTSEtLUytPSUnB6dOnsx0UERERERHpjlZLdoaHhytf37p1C9HR0cr3crkcISEhcHBw0F10RERERESakOadXnd90Crp9/LygkQigUQiyXAYT4ECBbB48WKdBUdERERERNmnVdIfEREBQRDg6uqKCxcuwMbGRrnPyMgItra2kOWhsU1ERERERP8FWiX9Tk5OAACFQpEjwRARERERke5pnPTv3btX45O2atVKVDBERERERKR7Gif9/v7+Ku8lEgkEQVB5/4lcLs9+ZEREREREmuJE3ixpvGSnQqFQbkeOHIGXlxcOHTqEpKQkJCUl4eDBg6hcuTJCQkJyMl4iIiIiItKSVmP6Pxk6dCiWL1+OWrVqKcv8/PxQsGBB9OnTB7dv39ZZgERERERElD2iHs718OFDmJubq5WbmZnh8ePH2QyJiIiIiIh0SVTSX7VqVQQFBSEmJkZZFhMTgxEjRqBatWo6C46IiIiIiLJP1PCeNWvWoE2bNihRogQcHR0BAE+fPkWpUqWwZ88eXcZHRERERPRFEqmovuz/DFFJv5ubG8LDw3H06FHcuXMHAFCmTBk0bNhQZRUfIiIiIiLSP1FJP/Bxic7GjRujcePGuoyHiIiIiEh7XLIzSxon/YsWLUKfPn1gYmKCRYsWZVl38ODB2Q6MiIiIiIh0QyJ8/oStLLi4uODSpUuwsrKCi4tL5ieUSPDo0SOdBUhEWRssy/zzSEQZ++HZNX2HQJQnlbM31XcImUoJWamXdk2a9NFLu9rSuKc/IiIiw9dERERERJS7iZrmzJ58IiIiIqK8Q/TqPcWLF0edOnVQt25d1KlTB25ubrqOjYiIiIhIM5zImyVRPf1Pnz5FcHAwChQogNmzZ8Pd3R3FixdHly5dsGrVKl3HSERERERE2aDxRN6s3L9/H9OnT8fmzZuhUCggl8t1ERsRaYATeYm0x4m8ROLk6om8R1brpV2Txt/rpV1tiRre8+7dO5w5cwYnT57EyZMncfXqVXh4eGDgwIGoW7eujkMkIiIiIqLsEJX0m5ubw8LCAl26dMHo0aNRu3ZtWFhY6Do2IiIiIiLSAVFJf7NmzXDmzBls27YN0dHRiI6ORt26deHu7q7r+IiIiIiIvkgiFTVV9T9D1N3Zs2cP4uLiEBISAl9fXxw5cgS1a9eGg4MDunTpousYiYiIiIgoG0T19H/i6emJDx8+IC0tDSkpKTh8+DC2b9+OzZs36yo+IiIiIqIv45KdWRLV0z9//ny0atUKVlZW8PHxwdatW+Hu7o7ffvsNL1++1HWMRERERESUDaJ6+rdu3Yo6deqgT58+qF27NszMzHQdFxERERER6YiopP/ixYsa1evfvz+mTp0Ka2trMc0QEREREZEO5Og0502bNiE5OTknmyAiIiIioi/I1kTeL9HBw36JiIiIiL6ME3mzxAVNiYiIiIjyOSb9RERERET5HJN+IiIiIqJ8jkk/EREREVE+l6MTebt27QpTU9OcbIKIiIiICBIZJ/JmRXTSn5iYiNWrV+P27dsAgDJlyqBnz56wtLRU1lm2bFn2IyQiIiIiomwRNbzn1KlTcHFxwaJFi5CYmIjExEQsXrwYLi4uOHXqlK5jJCIiIiKibBDV0z9gwAC0b98ey5Ytg+z/P6XI5XL0798fAwYMwPXr13UaJBERERERiSeqp//BgwcYPny4MuEHAJlMhqCgIDx48EBnwRERERERUfaJSvorV66sHMv/udu3b6NixYrZDoqIiIiISCtSqX62PELU8J7BgwdjyJAhePDgAapXrw4A+Pvvv7FkyRLMnDkT4eHhyroVKlTQTaRERERERCSKRBAEQduDpF/4ViORSCAIAiQSCeRyuejgiOjLBstc9B0CUZ7zw7Nr+g6BKE8qZ597l2JPO/ebXto18v1WL+1qS1RPf0REhK7jICIiIiKiHCIq6XdyctJ1HERERERElEM0Tvr37t2Lpk2bwtDQEHv37s2ybqtWrbIdGBERERER6YbGSb+/vz+io6Nha2sLf3//TOtxHD8RERERUe6icdKvUCgyfE1EREREpG8SqezLlf7DRI3pB4DQ0FCEhoYiNjZW5UuARCLB6tWrdRIcERERERFln6ikf8qUKZg6dSq8vb1hb28PiUSi67iIiIiIiEhHRCX9y5cvx7p169CtWzddx0NERERERDom6tnBaWlpqFGjhq5jISIiIiKiHCAq6e/Vqxe2bNmi61iIiIiIiMSRSvWz5REaD+8JCgpSvlYoFFi5ciWOHTuGChUqwNDQUKXu/PnzdRchERERERFli8ZJ/9WrV1Xee3l5AQBu3LihUs5JvURERET0tXHJzqxpnPSfOHEiJ+MgIiIiIqIckncGIhERERERkShM+omIiIiI8jkm/URERERE+Zyoh3MREREREeUqnMibJfb0ExERERHlc0z6iYiIiIjyOSb9RERERET5HJN+IiIiIqJ8jhN5iYiIiCjvk7IvOyu8O0RERERE+Rx7+omIiIgoz5PIuGRnVtjTT0RERESUzzHpJyIiIiLK55j0ExERERHlc0z6iYiIiIjyOU7kJSIiIqK8T8qJvFlhTz8RERERUT7HpJ+IiIiIKJ9j0k9ERERElM8x6SciIiIiyuc4kZeIiIiI8j5O5M0Se/qJiIiIiPI59vQTERERUZ4nkbIvOyu8O0RERERE+Rx7+onyuEXyCH2HQJTnpL5O0ncIRERfFXv6iYiIiIi+oiVLlsDZ2RkmJibw8fHBhQsXMq3766+/onbt2rCwsICFhQUaNmyYZf3MMOknIiIiIvpKtm/fjqCgIEyaNAlXrlxBxYoV4efnh9jY2Azrnzx5Ep06dcKJEydw7tw5ODo6onHjxoiKitKqXYkgCIIuLoCIiCiv4PAeInGMi5jrO4RMKR78rZd2pW7Vtarv4+ODqlWr4pdffgEAKBQKODo6YtCgQRg9evQXj5fL5bCwsMAvv/yC7t27ax6nVlESEREREZFSamoqkpOTVbbU1NQM66alpeHy5cto2LChskwqlaJhw4Y4d+6cRu29e/cO6enpsLS01CpOJv1ERERERCIFBwfDzMxMZQsODs6wblxcHORyOezs7FTK7ezsEB0drVF7o0aNQrFixVS+OGiCq/cQEREREYk0ZswYBAUFqZQZGxvnSFszZ87Etm3bcPLkSZiYmGh1LJN+IiIiIiKRjI2NNU7yra2tIZPJEBMTo1IeExODokWLZnns3LlzMXPmTBw7dgwVKlTQOk4O7yEiIiKivE8i1c+mBSMjI1SpUgWhoaHKMoVCgdDQUPj6+mZ63OzZszFt2jSEhITA29tb1O1hTz8RERER0VcSFBSEgIAAeHt7o1q1aliwYAHevn2LHj16AAC6d+8OBwcH5byAWbNmYeLEidiyZQucnZ2VY/8LFy6MwoULa9wuk34iIiIiyvu07HXXlw4dOuDly5eYOHEioqOj4eXlhZCQEOXk3sjISEil/1zLsmXLkJaWhu+++07lPJMmTcLkyZM1bpfr9BMR0X8O1+knEidXr9P/6JJe2pW6ihtu87Xlja9EREREREQkGpN+IiIiIqJ8jkk/EREREVE+x4m8RERERJTnCXlkIq++8O4QEREREeVzTPqJiIiIiPI5Jv1ERERERPkck34iIiIionyOE3mJiIiIKO/jRN4s8e4QEREREeVz7OknIiIiorxPItF3BLkae/qJiIiIiPI5Jv1ERERERPkck34iIiIionyOST8RERERUT7HibxERERElPdJ2ZedFd4dIiIiIqJ8jkk/EREREVE+x6SfiIiIiCifY9JPRERERJTPcSIvEREREeV5goR92Vnh3SEiIiIiyufY009EREREeR97+rPEu0NERERElM8x6SciIiIiyueY9BMRERER5XNM+omIiIiI8jlO5CUiIiKivI8TebPEu0NERERElM8x6SciIiIiyueY9BMRERER5XNM+omIiIiI8jlRSf+GDRuQmpqqVp6WloYNGzZkOygiIiIiIq1IpPrZ8giJIAiCtgfJZDK8ePECtra2KuXx8fGwtbWFXC7XWYBERES6lvo6Sd8hEOVJxkXM9R1Cpj68uK+Xdg3sS+mlXW2J+noiCAIkEola+bNnz2BmZpbtoIiIiIiISHe0Wqe/UqVKkEgkkEgkaNCgAQwM/jlcLpcjIiICTZo00XmQREREREQknlZJv7+/PwAgLCwMfn5+KFy4sHKfkZERnJ2d8e233+o0QCIiIiIiyh6tkv5JkyYBAJydndGhQweYmJjkSFBERERERNoQ8tCkWn0QNZH3k7S0NMTGxkKhUKiUlyhRItuBERER5RRO5CUSJzdP5E2PfqiXdg2LltRLu9rSqqf/k/v376Nnz544e/asSvmnCb5cvYeIiIiIvir29GdJVNIfGBgIAwMD7N+/H/b29hmu5ENERERERLmDqKQ/LCwMly9fhoeHh67jISIiIiIiHRP1O0jZsmURFxen61iIiIiIiCgHaJz0JycnK7dZs2Zh5MiROHnyJOLj41X2JScn52S8RERERESkJY1X75FKpSpj9zN6Ki8n8hIRUV7A1XuIxMnVq/fEPtZLu4a2znppV1saj+k/ceJETsZBREREREQ5ROOkv06dOjkZBxERERER5RBRq/eEh4dnWC6RSGBiYoISJUrA2Ng4W4EREREREZFuiEr6vby8slyb39DQEB06dMCKFStgYmIiOjgiIiIiIso+UUt2/v777yhVqhRWrlyJsLAwhIWFYeXKlShdujS2bNmC1atX4/jx4xg/fryu4yUiIiIiUieR6mfLIzRevedz1apVw7Rp0+Dn56dSfvjwYUyYMAEXLlzAnj17MHz4cDx8+FBnwRIREekCV+8hEidXr97zMlIv7RralNBLu9oSNbzn+vXrcHJyUit3cnLC9evXAXwcAvTixYvsRUdEREREpAEhD/W664Oou+Ph4YGZM2ciLS1NWZaeno6ZM2fCw8MDABAVFQU7OzvdRElERERERKKJ6ulfsmQJWrVqheLFi6NChQoAPvb+y+Vy7N+/HwDw6NEj9O/fX3eREhERERGRKKLG9APA69evsXnzZty7dw8AULp0aXTu3BlFihTRaYBERES6xjH9ROLk5jH9aXHP9NKukXVxvbSrLdFJPxERUV7FpJ9IHCb96vJK0q/x8J69e/eiadOmMDQ0xN69e7Os26pVq2wHRkRERESkMSkn8mZF455+qVSK6Oho2NraQprFTZVIJJDL5ToLkIiISNfY008kTq7u6U94rpd2jSyL6aVdbWnc069QKDJ8TUREREREuVu2fwdJSUnRRRxERERERJRDRCX9crkc06ZNg4ODAwoXLoxHjx4BACZMmIDVq1frNEAiIiIiIsoeUUn/9OnTsW7dOsyePRtGRkbK8vLly2PVqlU6C46IiIiISCMSqX62PEJUpBs2bMDKlSvRpUsXyGQyZXnFihVx584dnQVHRERERETZJ+qJvFFRUXBzc1MrVygUSE9Pz3ZQRERERERayUO97vog6u6ULVsWp0+fVivftWsXKlWqlO2giIiIiIhId0T19E+cOBEBAQGIioqCQqHA7t27cffuXWzYsAH79+/XdYxERERERJQNGj+c699Onz6NqVOn4tq1a3jz5g0qV66MiRMnonHjxrqOkYiISKf4cC4icXL1w7mSYvXSrpG5rV7a1ZbopJ+IiCivYtJPJA6TfnV5JekXNbznk7S0NMTGxqo9obdEiRLZCoqIiIiISCucyJslUUn//fv30bNnT5w9e1alXBAESCQSyOVynQRHRERERETZJyrpDwwMhIGBAfbv3w97e3tIJBJdx0VERERERDoiKukPCwvD5cuX4eHhoet4iIiIiIhIx0Sv0x8XF6frWIiIiIiIKAdovHpPcnKy8vWlS5cwfvx4zJgxA56enjA0NFSpa2pqqtsoiYiIdIir9xCJk5tX70lNTtBLu8amlnppV1saJ/1SqVRl7P6nSbuf40ReIiLKC5j0E4nDpF9dXkn6NR7Tf+LEiZyMg4iIiIhIPC7ZmSU+nIuIiP5z2NNPJE6u7unX0+c6N9+Tz2XrK9Hn4/yJiIjov2Xbjp1o0tIf3jVqo3NAT1y/cTPL+keOhaLVt+3hXaM22nbojNNn/sq07rQZM1HB2wcbt2xVKV+5ei269eyFajW/Qc26DTI89u8LF9GtZy9U/6Ye6vk1xc+LfsGHDx+0v0CifCRbSX9gYCBmzJihq1iIiIgojwg5chRzfl6Ivr2/x/ZN61Ha3Q19Bw1BfELG46rDroVj1LgJaNO6JXZs3oD6db/BkB9H4v6Dh2p1Q0+cRPiNG7C1sVHbl/4hHY0bNED7777NsJ279+5hwJBhqOlbHTs2b8CcGdNx8tRpLPhlSfYumCiPy1bSP2/ePMybN09XsRAREVEesWHzVnzr3xr+rVqipKsrJowZjQImJtizd1+G9Tdv246avtXRo3s3uLq4YGC/vijjURrbduxUqRcTG4vgOXMRPG0qDAzUpx4O+KEPunXphFJuJTNsJ+ToMbiXckPf3r1QwtER3lUqY9jggdi+8ze8ffs2+xdOlEdlK+l3dnbG69evERMTo6t4iIiIKJdLT0/H7Tt3UN2nmrJMKpXCp1pVXAu/nuEx18Kvw6daVZWyGr7Vce36P/UVCgXGTpyMwG5d4VbSVVxsaekwMjJSKTMxNkZqaipu3b4j6pyUR0gk+tnyiGwl/ZcvX0ahQoVgk8HPb0RERJQ/JSYlQS6Xw8pSdalCK0tLxMVnPLwnLj4+k/rxyvdr1m+AgUyGLh07iI6thq8ProVfx8GQw5DL5YiJjcXyVasBAC/5YFH6D8tW0r9u3Tq0a9cOUimXSCIiIiLxbt2+jc3btmPa5IlqzwHSRo3q1RE0eBB+Cp4F7xq10bJtO9SuWQMAmK/Qf5rG6/Rn5ODBg1i5cqWuYiEiIqI8wMLcHDKZTG3SbnxCAqytMn5QkbWVVSb1rQAAl6+GISEhEX4tWiv3y+VyzFuwCJu3bkfIvj0ax9e9a2d069IJL+PiYFqkCJ6/eIGFvyxFcQcHjc9BlN9kK+lPTU2Fra2trmIhIiKiPMDQ0BBlPDxw/sJF1K9bB8DH8fjnL15Ep/btMjymYgVPnL94Cd06d1KW/X3+Aip6egIAWjZrhurVqqkc02/QELRo1hStW7bQOkaJRKJc/efQ4SMoameHMh6ltT4PUX6RraS/ZcuW2L59OypUqKCreIiIiCgP6N6lE8ZPnoqyZcvAs1xZbNqyDe/fp8D//wn62ImTYWdrgyEDBwAAunTsgJ59+mL9ps34plZNHDp8FDdv3cbEsWMAAObmZjA3N1Npw8DAAFZWlnBxdlKWvYiOxqtXyXgRHQ25QoE7d+8BAEo4FkfBggUBAGs3bETNGr6QSqQIPXECq9dtwNyZMyCTyXL8vpAe8Ym8WcpW0j9t2jRMmDABycnJMDU11VVMRERElMs1adwIiYlJWLp8JeLi41Ha3R3LFi+A1f+H60RHx6iMofeqWAEzp0/D4qXLsWjJMpRwdMTCubMzXXozM0uWr8Te/QeU79t36QYAWL18Kap6VwEAnDl7DqvWrENaejrcS7lh4bw5ynH9RP9VEkEQBH0HQURE9DWlvk7SdwhEeZJxEXN9h5CplHf6eQ6DScFCemlXW/wdhIiIiIgonxM1vEcul2PdunUIDQ1FbGwsFAqFyv7jx49neXxqaipSU1NVyoyNjWFsbCwmHCIiIiIiyoKonv4hQ4ZgyJAhkMvlKF++PCpWrKiyfUlwcDDMzMxUtuDgYDGhEBERERHRF4ga029tbY0NGzagWbNmohplTz8REekTx/QTicMx/ery9Zh+IyMjuLm5iW7U2NgYpqamKhsTfiIiIv1KSnqFOo2aIOr5c32H8tWdOXsO7Tp3VRuyTHmIRKqfLY8QFenw4cOxcOFCcOEfIiKi/OPXNWtRr843cChWLMfa2LZjJ5q09Id3jdroHNAT12/czLL+seMn0LFbAGrWbYBqteqgXeeu2HfgoEodQRCwZPkK1Pdrhqo1v0Hv/gPxJDIyw/OlpaWhXeeuqODto1zjHwBq1fCFgYEBDhwKyf5FEn3BkiVL4OzsDBMTE/j4+ODChQtZ1t+5cyc8PDxgYmICT09PHDx4MMv6GdF4Im/btm1V3h8/fhyHDh1CuXLlYGhoqLJv9+7dWgdCRERE+vM+JQW//7EXy35ZmGNthBw5ijk/L8SEMaPgWb4cNm3dhr6DhmDvbztgZWmZ4TFmpqbo3bMHXJydYGhoiD9Pn8HEqT/B0tISNX2rAwDWrt+ILdt24KfJE+HgUAy/LFuBvoOGYM+ObWojCeYvWgwba2vcvXdfra3WLZpjy/YdaNlc3PBlIk1s374dQUFBWL58OXx8fLBgwQL4+fnh7t27sLW1Vat/9uxZdOrUCcHBwWjRogW2bNkCf39/XLlyBeXLl9e4XY17+v898bZNmzaoU6cOrK2t1fYRERFR3nL6zF8wNDJCRU9PAMDFS5dRwdsHp86cwbcdu8C7Rm10CeyJ+w8eim5jw+at+Na/NfxbtURJV1dMGDMaBUxMsGfvvkyPqepdBQ3q1YWriwscixdH104dUcrNDVfDwgB87OXftHUben/fA/Xq1oF7qVKYPnUyXr6Mw/GTf6pe419nce7vCxg+dHCGbdX5pjZu3rqNp8+eib5Goi+ZP38+evfujR49eqBs2bJYvnw5ChYsiDVr1mRYf+HChWjSpAlGjBiBMmXKYNq0aahcuTJ++eUXrdrVuKd/7dq1Wp2YiIiI8o4rYWEoW8ZDrXz+wsUYNTwIVtZWWLRkGQYFDce+3btgaGCAF9HR8G/XMcvz9uoRiN49A5Geno7bd+6gV48A5T6pVAqfalVxLfy6RjEKgoDzFy/h8ZMnGDZoAAAgKuo54uLjUb1aNWW9IoULw7N8OVy7fh1N/RoDAOLj4zFl+gwsnDsHJiYmGZ7fvmhRWFlZ4srVMDgWL65RTETaLFCTlpaGy5cvY8yYMcoyqVSKhg0b4ty5cxme/9y5cwgKClIp8/Pzw549e7SKU9Q6/fXr18fu3bthbm6uUp6cnAx/f/8vrtNPREREucuLF9GwtbZWK+/buxd8q/sAAKZPnohGzVri+ImT8GvUEDbW1ti5ZWOW5zUzNQUAJCYlQS6Xqw3jsbK0RMTjJ1me4/WbN2jYtAXS09IglckwbtQIZUxx8fEfz2Olft74+AQAH78sjJ8yDe3btkW5smWynKhsY22D5y+is4yHcidBItFLuzODgzFlyhSVskmTJmHy5MlqdePi4iCXy2FnZ6dSbmdnhzt37mR4/ujo6AzrR0dr93cqKuk/efIk0tLS1MpTUlJw+vRpMackIiIiPUpNTYWRsZFaecUKnsrXZmZmcHZywqOIxwAAAwMDlHB0zPHYChUsiJ1bNuLdu/c4f/Ei5v68EMUdHFDVu4pGx2/ZvgPv3r7F95/9ypAZE2NjpKSkZDdk+g8ZM2aMWk98blyVUqukPzw8XPn61q1bKt8w5HI5QkJC4ODgoLvoiIiI6KswNzdHcvJrrY7RZniPhbk5ZDIZ4hMSVPbHJyTA2irjSbyfSKVS5ZcLj9LueBTxGKvXrUdV7yqwtrL6eJ74BNh89ktFfEICSruXAgBcuHgJ167fgHeN2irn7dQ9EM2a+GH6lEnKslfJybCwMM/6wilX0teiksYmmj9rytraGjKZDDExMSrlMTExKFq0aIbHFC1aVKv6mdEq6ffy8oJEIoFEIkH9+vXV9hcoUACLFy/WKgAiIiLSP4/S7hkuVxl+/Qbs/59cJCcn43FkJFxdnAFAq+E9hoaGKOPhgfMXLqJ+3ToAAIVCgfMXL6JT+3ZaxSooFEhLSwcAODgUg7WVFc5fvAiP0u4AgDdv3uD6jZto/+3HlQdHjxiOgf36Ko9/GfcSfQcOwewZP8GzfDlleWpqKp4+ewaP0qW1iodIU0ZGRqhSpQpCQ0Ph7+8P4OPnIDQ0FAMHDszwGF9fX4SGhmLo0KHKsqNHj8LX11ertrVK+iMiIiAIAlxdXXHhwgXY2NioXIStrS1kMplWARAREZH+1fCtjkW/LEVycjJM/5+oA8DyX1fDzMwMVpaWWLx0GSzMzZVJu7bDe7p36YTxk6eibNky8CxXFpu2bMP79ynwb9lCWWfsxMmws7XBkIEfJ+quWrsO5cqUgWPx4khLT8Ppv85i/8FDGDdmFABAIpGga6eOWLl6LUo4OsLBoRiWLFsBGxtrZZz2/+oRLViwAADAsXhxFP1srHT49RswMjJSGdJEpGtBQUEICAiAt7c3qlWrhgULFuDt27fo0aMHAKB79+5wcHBAcHAwAGDIkCGoU6cO5s2bh+bNm2Pbtm24dOkSVq5cqVW7WiX9Tk5OAMCn1REREeUz7m5u8PDwwOGjx9Du23+ezTN00ADMmjsfkU+forS7Oxb/PFft+TyaatK4ERITk7B0+UrExcejtLs7li1eAKv/D9EBgOjoGEil/6wo/v59CqbPmo2Y2JcwNjaGi7MTZkybgiaNGynr9Ajohvcp7zF1RjBev36DSl4VsWzRQq3HVR86fATNm/ihQCar+xDpQocOHfDy5UtMnDgR0dHR8PLyQkhIiHKybmRkpMpnoEaNGtiyZQvGjx+PsWPHolSpUtizZ49Wa/QDgETQ8LG6e/fu1fikrVq10ioIIiKiryn1dZK+Q8iVTp05g/kLF2P39q24fOUqvu/bH2dOHINpkSL6Di3HJSYlodW37bB1w3oUd8i5JxLndcZFzPUdQqbevdfPBOyCBfLGl0SNe/o/jTv6RCKR4PPvC5LPlkmSy+XZj4yIiIi+qm9q1cKTyKeIjX2p71C+uufPX2DcqJFM+PMwhb5m8uYRGj+RV6FQKLcjR47Ay8sLhw4dQlJSEpKSknDw4EFUrlwZISHqk4CIiIgob+jW+X/t3XlU1Nfdx/HPKMgShAguoCaACkpcIAQ0GgmkaqK2UZMekzaaauNSazxaTS2nVSNoXRIT02KNxqXibnM0GmyTulXAJTFqwS2IilptiutDXKLHBe7zRx/nccI+DMKM79c5cw6/mTu/+53l/vjOne/vzk8VGNik/IYupu0TETYlQ4CrqXB5z/3atWun+fPnq2vXrjbX79ixQ8OHD1dOTo7DAgQAwNEo7wHsU5vLe67fuFkj/fr834nhtV2FZ/rvl5eXV+zXeKX//mjH6dOnqxgSAAAAAEeyK+mPjY3VuHHjbH4o4Pz58xo/frw6duzosOAAAAAAVF2lluy8589//rNeeuklPf7443rs/9bnPXv2rHUJIQAAAOBB4jTestlV0y9Jxhht2bJFR48elSRFRESoe/fuNqv4AABQG1HTD9inNtf0X6uhmv76TlLTb3fSDwCAsyLpB+xTm5P+K9/VTNLv94hzJP0VLu9JSUnR8OHD5enpqZSUlDLbjh49usqBAQAAAHCMCs/0h4aGat++fQoICFBoaGjpO7RYdPLkSYcFCACAozHTD9iHmf7inGWmn/IeAMBDh6QfsA9Jf3HOkvTbtWQnM/kAAACA87Bryc5WrVqpefPmio+PV0JCguLj49WqVStHxwYAAABUCMUrZbNrpv/s2bOaMWOGvLy89O677yo8PFzNmzfXgAEDtGjRIkfHCAAAAKAKHFLTf/z4cU2bNk0rV65UUVGRCgsLHREbAADVgpp+wD61uab/2+s3aqTfR328a6TfyrKrvOfGjRvauXOn0tPTlZ6erqysLLVp00ajRo1SQkKCg0MEAAAAUBV2zfTXq1dPDRo00IABA5SQkKC4uDg1aNCgOuIDAMDhmOkH7MNMf3EuPdPfu3dv7dy5U2vWrNG5c+d07tw5JSQkKDw83NHxAQAAAOUq4jzeMlWppv/gwYPKyMhQRkaGduzYITc3NyUkJGjlypWOjBEAAIdiph+wT22e6f+fazUz0+9f3zlm+quU9BtjlJWVpe3bt2v79u3atGmTjDG6e/euI2MEAMChSPoB+5D0F+csSb9dS3bOnj1bffr0UUBAgDp16qTVq1crPDxc69at08WLFx0dIwAAAIAqsGumPzY21vrDXHFxcfLz86uO2AAAqBbM9AP2Yaa/OGeZ6XfIOv2lGTlypKZMmaKGDRtWVxcAAFQaST9gn9qc9F+uoaQ/wEmSfrvKeypqxYoVunr1anV2AQAAAKAcdi3ZWVHV+CUCAAAAYMWSnWWr1pl+AAAAADWPpB8AAABwcST9AAAAgIsj6QcAAABcXLWeyDtw4ED5+vpWZxcAAAAAC8iUw+51+gsKCrR48WLl5ORIkiIiIvTGG2/I39/foQECAOBorNMP2Kc2r9N/4cp3NdJvY79HaqTfyrKrvCczM1OhoaFKSUlRQUGBCgoKNGfOHIWGhiozM9PRMQIAAACoArtm+tu3b6/OnTtr3rx5qlu3riSpsLBQI0eO1O7du3Xo0CGHBwoAgKMw0w/Yh5n+4pxlpt+upN/Ly0vZ2dlq3bq1zfW5ubmKiorSzZs3HRYgAACORtIP2IekvzhnSfrtKu+Jjo621vLfLycnR5GRkVUOCgAAAKiMohq6OAu7Vu8ZPXq0xowZoxMnTujpp5+WJH355ZeaO3euZs6cqYMHD1rbdujQwTGRAgAAALCLXeU9deqU/QWBxWKRMUYWi0WFhYV2BwcAQHWgvAewT20u78n/tmbKe4IedY7yHrtm+k+dOuXoOAAAAABUE7uS/uDgYEfHAQAAAKCaVDjpT0tLU69eveTu7q60tLQy2/bp06fKgQEAAABwjArX9NepU0fnzp1T48aNy6zpp44fAFDbUdMP2Iea/uJcrqa/qKioxL8BAACAmlZU6aVpHi521fRL0rZt27Rt2zZduHDB5kOAxWLR4sWLHRIcAAAAgKqzK+lPTk7WlClTFBMTo6CgIFksFkfHBQAAAMBB7Er658+fr9TUVL3++uuOjgcAAACAg5X9K1uluH37trp06eLoWAAAAABUA7uS/qFDh2rVqlWOjgUAAACwizGmRi7OosLlPePGjbP+XVRUpAULFmjr1q3q0KGD3N3dbdrOnj3bcRECAAAAqJIKJ/1ZWVk221FRUZKkw4cP21zPSb0AAAB40FhQvmwVTvq3b99enXEAAAAAqCZ21fQDAAAAcB4k/QAAAICLI+kHAAAAXJxdP84FAAAA1CZOtHpmjWCmHwAAAHBxJP0AAACAiyPpBwAAAFwcST8AAADg4jiRFwAAAE6viDN5y8RMPwAAAODimOkHAACA02Oev2zM9AMAAAAujqQfAAAAcHEk/QAAAICLI+kHAAAAXBwn8gIAAMDpFXEmb5mY6QcAAABcHEk/AAAA4OJI+gEAAAAXR9IPAAAAuDhO5AUAAIDTM5zIWyZm+gEAAAAXx0w/AAAAnF6RmOovCzP9AAAAgIsj6QcAAABcHEk/AAAA4OJI+gEAAAAXx4m8AAAAcHos2Vk2ZvoBAAAAF0fSDwAAALg4kn4AAADAxZH0AwAAAC6OE3kBAADg9Io4kbdMzPQDAAAALo6ZfgAAADg9luwsGzP9AAAAgIsj6QcAAABcHEk/AAAA4OJI+gEAAAAXZzGG0x5Qulu3bmnGjBn67W9/Kw8Pj5oOB3AajB3APowdoHqQ9KNMV69elZ+fn65cuSJfX9+aDgdwGowdwD6MHaB6UN4DAAAAuDiSfgAAAMDFkfQDAAAALo6kH2Xy8PDQ5MmTOZkKqCTGDmAfxg5QPTiRFwAAAHBxzPQDAAAALo6kHwAAAHBxJP0AAACAi3tok/6EhAT96le/KvV2i8WiDRs2PLB47hcSEqI//OEPZbapSHyDBw9Wv379KtV3enq6LBaLvv3220rd7/sq+/w5ql/Uboy7ymNMorb7/riuyFhypNTUVD366KNV2sfp06dlsViUnZ39QPsFHiS3mg6gtsrPz1eDBg1qpO+9e/fqkUceqXD706dPKzQ0VFlZWYqKiqq+wCqhOp6/pKQkbdiwoVIHZTgXxl31YUw+PAYPHqxvv/22xj5AV3Ys1QaPPfaY8vPz1bBhQ4fut6ZfC+B+JP2lCAwMrLG+GzVqVGN9O0pNPn9wXoy76sOYxIPijGOpbt26jBG4vIe2vEeSioqK9Jvf/Eb+/v4KDAxUUlKS9bb7vwq/97XfmjVr1KVLF3l6eqpdu3bKyMioUD8xMTF67733rNv9+vWTu7u7rl+/Lkn697//LYvFohMnTkgq/tXo8ePH9eyzz8rT01NPPPGEtmzZYrP/0NBQSdKTTz4pi8WihIQEm9vfe+89BQUFKSAgQG+++abu3LlTbsz79+9XTEyMvL291aVLF+Xm5trc/umnnyo6Olqenp5q0aKFkpOTdffuXevt3y8l2L17t6KiouTp6amYmBht2LChxK9SS+s3NTVVycnJOnDggCwWiywWi1JTU8t9HN937+vYTZs2KSIiQj4+PurZs6fy8/Otbfbu3asePXqoYcOG8vPzU3x8vP75z3/a7MdisWjRokV66aWX5O3trbCwMKWlpVU6nocR484+rjomYb+1a9eqffv28vLyUkBAgLp3767x48dr6dKl+vTTT62vS3p6uiQpMTFR4eHh8vb2VosWLTRp0iSb92VSUpKioqK0fPlyhYSEyM/PTz/5yU907do1a5vvvvtOP/vZz+Tj46OgoCC9//77xeL6/liqyPEyLS1NYWFh8vT01HPPPaelS5dWurysrOO6JC1atEgRERHy9PRUmzZt9OGHH1pvK6m8p6IxldZvUlJSqa8FUCPMQyo+Pt74+vqapKQkc+zYMbN06VJjsVjM5s2bjTHGSDLr1683xhhz6tQpI8k0b97crF271nz99ddm6NChpn79+ubSpUvl9jVu3Djzwx/+0BhjTFFRkfH39zcNGzY0n3/+uTHGmBUrVphmzZpZ2wcHB5sPPvjAGGNMYWGhadeunenWrZvJzs42GRkZ5sknn7SJ76uvvjKSzNatW01+fr65fPmyMcaYQYMGGV9fXzNixAiTk5NjNm7caLy9vc2CBQusfU2ePNkEBwdbt7dv324kmU6dOpn09HRz5MgRExcXZ7p06WJtk5mZaXx9fU1qaqrJy8szmzdvNiEhISYpKcna5v74rly5Yvz9/c3AgQPNkSNHzGeffWbCw8ONJJOVlVWhfm/cuGHeeust07ZtW5Ofn2/y8/PNjRs3yn3u7z2H9yxZssS4u7ub7t27m71795r9+/ebiIgI89prr1nbbNu2zSxfvtzk5OSYr7/+2gwZMsQ0adLEXL161ebxNW/e3KxatcocP37cjB492vj4+Fife5SMcVdxRUVFxhjXHJOouv/85z/Gzc3NzJ4925w6dcocPHjQzJ0711y7ds288sorpmfPntbX5datW8YYY6ZOnWp27dplTp06ZdLS0kyTJk3MO++8Y93n5MmTjY+Pj3n55ZfNoUOHTGZmpgkMDDS/+93vrG1++ctfmscff9xs3brVHDx40PzoRz8y9evXN2PGjLG2uX8sGVP+8fLkyZPG3d3d/PrXvzZHjx41q1evNs2aNTOSTEFBQbnPRUWO6ytWrDBBQUFm3bp15uTJk2bdunXG39/fpKamGmP+/3hz7/1fkZjK67es1wKoCQ910t+1a1eb62JjY01iYqIxpuTkY+bMmda2d+7cMc2bN7c5YJYmLS3N+Pn5mbt375rs7GwTGBhoxowZY+1r6NChNgen+w+YmzZtMm5ubuabb76x3v7555+XGN+9g9U9gwYNMsHBwebu3bvW6/r3729effVV6/acOXPMD37wA+v2vX/0W7dutV73t7/9zUgyN2/eNMYY061bNzN9+nSbvpYvX26CgoKs2/fHN2/ePBMQEGC9vzHGLFy4sMQEo6x+J0+ebCIjI4s9vydOnDA//elPTaNGjUx4eLhJTEw0hw8fNsYY6z+le5YsWWIkmRMnTlivmzt3rmnSpEmx/d5TWFho6tevbzZu3Gjz+CZOnGjdvn79upFkTShRMsbdf50/f94MHz7cBAUFmZCQEDNy5Ejz1VdfmcLCQnPy5EnTr18/a2LhjGMS1W///v1Gkjl9+nSx2wYNGmT69u1b7j5mzZplnnrqKev25MmTjbe3t80Ex/jx402nTp2MMf9NYuvVq2c+/vhj6+2XL182Xl5e5Sb9ZR0vExMTTbt27WximzBhQqWS/vKO6y1btjSrVq2yud/UqVNN586djTHFx3NFYqpIvxV9LYAH4aEu7+nQoYPNdlBQkC5cuFBq+86dO1v/dnNzU0xMjHJycsrtJy4uTteuXVNWVpYyMjIUHx+vhIQE69d8GRkZxUoD7snJydFjjz2mpk2blhhHedq2bau6detat7//GEeNGqVt27YVu9/9z01QUJAkWe934MABTZkyRT4+PtbLsGHDlJ+frxs3bhTbV25urjp06CBPT0/rdR07diwx3rL6Lc3YsWMVGRmpbdu26d1339Xp06cVGxsrd3d3Pf/883rxxRdt2nt7e6tly5Y2/dzfx/nz5zVs2DCFhYXJz89Pvr6+un79us6cOVNqrI888oh8fX3LjRWMO0maNGmS/P399dlnn+mjjz7SrVu31KNHD3l4eCg6OlrR0dHy9fW12aczjUlUv8jISHXr1k3t27dX//79tXDhQhUUFJR5n7/85S965plnFBgYKB8fH02cOLHYcS0kJET169e3bt//3s3Ly9Pt27fVqVMn6+3+/v5q3bp1ufGWdbzMzc1VbGysTfvS3o+lKeu4/t133ykvL09DhgyxGSO///3vlZeXV+L+KhpTef9PgNrkoT6R193d3WbbYrGoqKjI4f08+uijioyMVHp6ur744gv16NFDzz77rF599VUdO3ZMx48fV3x8vMP7lex/jPffz2KxSJL1ftevX1dycrJefvnlYve7P4mwR1n9lmbZsmXWZdPat2+vvn376tatWyooKCjxxKySnhNjjHV70KBBunz5sv74xz8qODhYHh4e6ty5s27fvl3ufqrj/eNqGHfSO++8Y7PU3/PPP6/58+fr4sWLCgwMtL73S9tnbR+TqH5169bVli1btHv3bm3evFlz5szRhAkTtGfPnhLbf/HFFxowYICSk5P1wgsvyM/PT2vWrClWk19d47O6x31Zx/V75/EsXLjQ5gOLJJsP547uF6htHuqZ/sr68ssvrX/fvXtX+/fvV0RERIXuGx8fr+3btyszM1MJCQny9/dXRESEpk2bpqCgIIWHh5d4v4iICJ09e9bmhKT745CkevXqSZIKCwsr+5DsEh0drdzcXLVq1arYpU6d4m+p1q1b69ChQ7p165b1ur1791a633r16pX4GEtaJ9nDw8PulRh27dql0aNHq3fv3mrbtq08PDx06dIlu/aFqnPFcVfSe9bNzU1BQUElJvzlqW1jEg+GxWLRM888o+TkZGVlZalevXpav359ia/L7t27FRwcrAkTJigmJkZhYWH617/+Van+WrZsKXd3d5sPFgUFBTp27FiVHkfr1q21b98+m+vseT+WpkmTJmratKlOnjxZbHzcOyG/umJijKA2IemvhLlz52r9+vU6evSo3nzzTRUUFOiNN96o0H0TEhK0adMmubm5qU2bNtbrVq5cWeZsY/fu3RUeHq5BgwbpwIED2rFjhyZMmGDTpnHjxvLy8tLf//53nT9/XleuXKnwY/rTn/6kbt26Vbi9JL399ttatmyZkpOTdeTIEeXk5GjNmjWaOHFiie1fe+01FRUVafjw4crJydGmTZusq6pUJsEJCQnRqVOnlJ2drUuXLtkkLI4UFham5cuXKycnR3v27NGAAQPk5eVVLX2hfK447hzN1cckituzZ4+mT5+uffv26cyZM/rkk0908eJFRUREKCQkRAcPHlRubq4uXbqkO3fuKCwsTGfOnNGaNWuUl5enlJQUrV+/vlJ9+vj4aMiQIRo/frz+8Y9/6PDhwxo8eHCJHywr4xe/+IWOHj2qxMREHTt2TB9//LF1JSh7PgSXJDk5WTNmzFBKSoqOHTumQ4cOacmSJZo9e3a1xlTSawHUFJL+Spg5c6ZmzpypyMhI7dy5U2lpaRX+IY+4uDgVFRXZJBoJCQkqLCwsta5YkurUqaP169fr5s2b6tixo4YOHapp06bZtHFzc1NKSoo++ugjNW3aVH379q3wY7p06VKpNY2leeGFF/TXv/5VmzdvVmxsrJ5++ml98MEHCg4OLrG9r6+vNm7cqOzsbEVFRWnChAl6++23JVWu9ODHP/6xevbsqeeee06NGjXS6tWrKxV3RS1evFgFBQWKjo7W66+/rtGjR6tx48bV0hfK54rjztFcfUyiOF9fX2VmZqp3794KDw/XxIkT9f7776tXr14aNmyYWrdurZiYGDVq1Ei7du1Snz59NHbsWI0aNUpRUVHavXu3Jk2aVOl+Z82apbi4OL344ovq3r27unbtqqeeeqpKjyU0NFRr167VJ598og4dOmjevHnWD9keHh5V2vc9Q4cO1aJFi7RkyRK1b99e8fHxSk1NLXWm31ExlfRaADXFYig+K5ez/fKmM1i5cqV+/vOf68qVK8yio0SMuweLMYnaZNq0aZo/f77Onj1b06FY1caYgMp4qE/kxYOzbNkytWjRQs2aNdOBAweUmJioV155heQCqCGMSdQmH374oWJjYxUQEKBdu3Zp1qxZGjVqFDEBDkR5jwOMGDHCZhmw+y8jRoyo6fBqhXPnzmngwIGKiIjQ2LFj1b9/fy1YsKCmw4ITY9xVDWMStcnx48fVt29fPfHEE5o6dareeust66919+rVq9SxPn369BqJCXBGlPc4wIULF3T16tUSb/P19aUeHKgGjDvg4fDNN9/o5s2bJd7m7+8vf3//BxwR4JxI+gEAAAAXR3kPAAAA4OJI+gEAAAAXR9IPAAAAuDiSfgAAAMDFkfQDAAAALo6kHwAAAHBxJP0AAACAiyPpBwAAAFzc/wJxBjutYAfLkAAAAABJRU5ErkJggg==",
      "text/plain": [
       "<Figure size 1000x800 with 2 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Create a pivot table for heatmap\n",
    "correlation_table = rg_df.pivot('pheno1', 'pheno2', 'cor')\n",
    "\n",
    "# Create a pivot table for p-values\n",
    "p_value_table = rg_df.pivot('pheno1', 'pheno2', 'p_value')\n",
    "\n",
    "# Create a new DataFrame to hold the display values\n",
    "display_table = correlation_table.astype(str) + \"\\n(p=\" + p_value_table.round(3).astype(str) + \")\"\n",
    "\n",
    "# Plotting\n",
    "plt.figure(figsize=(10, 8))\n",
    "sns.heatmap(correlation_table, annot=display_table, fmt='', cmap=\"RdBu_r\", center=0)\n",
    "plt.title(\"Genetic Correlation Heatmap\")\n",
    "plt.xlabel(\"\")\n",
    "plt.ylabel(\"\")\n",
    "plt.show()"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Genetic correlation for obstetrical phenotypes and pelvic shape"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Female"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "!mkdir -p key_results/genetic_correlation/rg_20230909"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.DataFrame(columns=['pheno_pelvic', 'pheno_pelvic_h2g', 'pheno_pelvic_h2g_se', 'pheno_pelvic_lambda',\n",
    "                           'pheno_outcome', 'pheno_outcome_h2g', 'pheno_outcome_h2g_se', 'pheno_outcome_lambda',\n",
    "                           'cor', 'se', 'z', 'p_value'])\n",
    "\n",
    "path = \"UKB_Imaging_Genetics/LDSC_OUTPUT/rg_pheno_outcome_20230909/female/\"\n",
    "\n",
    "for file in os.listdir(path):\n",
    "    if file.endswith(\".log\"):\n",
    "        file_modified = file.replace(\"_-_\", \"_\")\n",
    "        # get phenotype name\n",
    "        pheno1 = file_modified.split('-')[0]\n",
    "        pheno2 = file_modified.split('-')[1]\n",
    "        pheno1 = pheno1.split('.')[0]\n",
    "        pheno2 = pheno2.split('_rg.log')[0]\n",
    "\n",
    "        with open(os.path.join(path, file), \"r\") as f:\n",
    "            lines = f.readlines()\n",
    "\n",
    "        for i, line in enumerate(lines):\n",
    "            # get h2g\n",
    "            if line.startswith(\"Heritability of phenotype 1\") and i + 2 < len(lines):\n",
    "                h2g_1 = lines[i + 2].split(\" \")[4].strip()\n",
    "                h2g_1 = float(h2g_1)\n",
    "\n",
    "                h2g_1_se = lines[i + 2].split(\" \")[5].strip()\n",
    "                h2g_1_se = h2g_1_se[1:-1]\n",
    "                h2g_1_se = float(h2g_1_se)\n",
    "\n",
    "                pheno1_lambda = lines[i + 3].split(\" \")[2].strip()\n",
    "                pheno1_lambda = float(pheno1_lambda)\n",
    "            \n",
    "            if line.startswith(\"Heritability of phenotype 2/2\") and i + 2 < len(lines):\n",
    "                h2g_2 = lines[i + 2].split(\" \")[4].strip()\n",
    "                h2g_2 = float(h2g_2)\n",
    "                \n",
    "                h2g_2_se = lines[i + 2].split(\" \")[5].strip()\n",
    "                h2g_2_se = h2g_2_se[1:-1]\n",
    "                h2g_2_se = float(h2g_2_se)\n",
    "\n",
    "                pheno2_lambda = lines[i + 3].split(\" \")[2].strip()\n",
    "                pheno2_lambda = float(pheno1_lambda)\n",
    "\n",
    "            # get rg\n",
    "            if line.startswith(\"Genetic Correlation:\"):\n",
    "                # print(line)\n",
    "                rg = line.split(\" \")[2].strip()\n",
    "                rg = float(rg)\n",
    "                # print(rg)\n",
    "                se = line.split(\" \")[3].strip()\n",
    "                se = se[1:-1]\n",
    "                se = float(se)\n",
    "            if line.startswith(\"Z-score:\"):\n",
    "                z = line.split(\" \")[1].strip()\n",
    "                z = float(z)\n",
    "            if line.startswith(\"P:\"):\n",
    "                p_value = line.split(\" \")[1].strip()\n",
    "                p_value = float(p_value)\n",
    "                neglog10p = -np.log10(p_value)\n",
    "            if line.startswith(\"ERROR\"):\n",
    "                rg = \"NA\"\n",
    "                se = \"NA\"\n",
    "                z = \"NA\"\n",
    "                p_value = \"NA\"\n",
    "                neglog10p = \"NA\"\n",
    "                break\n",
    "\n",
    "        df = df.append({'pheno_pelvic': pheno1, 'pheno_pelvic_h2g': h2g_1, 'pheno_pelvic_h2g_se': h2g_1_se, 'pheno_pelvic_lambda': pheno1_lambda,\n",
    "                        'pheno_outcome': pheno2, 'pheno_outcome_h2g': h2g_2, 'pheno_outcome_h2g_se': h2g_2_se, 'pheno_outcome_lambda': pheno2_lambda,\n",
    "                        'cor': rg, 'se': se, 'z': z, 'p_value': p_value, 'neglog10p': neglog10p}, \n",
    "                        ignore_index=True)\n",
    "\n",
    "df.sort_values(by = ['pheno_pelvic', 'pheno_outcome'], inplace=True)\n",
    "df.to_csv(\"key_results/genetic_correlation/rg_20230909/rg_hip_pelvic_vs_outcome_female.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pheno_pelvic</th>\n",
       "      <th>pheno_pelvic_h2g</th>\n",
       "      <th>pheno_pelvic_h2g_se</th>\n",
       "      <th>pheno_pelvic_lambda</th>\n",
       "      <th>pheno_outcome</th>\n",
       "      <th>pheno_outcome_h2g</th>\n",
       "      <th>pheno_outcome_h2g_se</th>\n",
       "      <th>pheno_outcome_lambda</th>\n",
       "      <th>cor</th>\n",
       "      <th>se</th>\n",
       "      <th>z</th>\n",
       "      <th>p_value</th>\n",
       "      <th>neglog10p</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>465</th>\n",
       "      <td>BW_devide_oblique_pelvic_inlet_length</td>\n",
       "      <td>0.1310</td>\n",
       "      <td>0.0376</td>\n",
       "      <td>1.0466</td>\n",
       "      <td>finngen_R9_O15_LABOUR_PELVIC_ABNORM</td>\n",
       "      <td>0.0217</td>\n",
       "      <td>0.0034</td>\n",
       "      <td>1.0466</td>\n",
       "      <td>0.559</td>\n",
       "      <td>0.1561</td>\n",
       "      <td>3.581</td>\n",
       "      <td>0.0003</td>\n",
       "      <td>3.522879</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1627</th>\n",
       "      <td>BW_devide_pelvic_inlet_area</td>\n",
       "      <td>0.1462</td>\n",
       "      <td>0.0375</td>\n",
       "      <td>1.0466</td>\n",
       "      <td>finngen_R9_O15_LABOUR_PELVIC_ABNORM</td>\n",
       "      <td>0.0217</td>\n",
       "      <td>0.0034</td>\n",
       "      <td>1.0466</td>\n",
       "      <td>0.7444</td>\n",
       "      <td>0.1598</td>\n",
       "      <td>4.657</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>5.493752</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>847</th>\n",
       "      <td>BW_devide_pelvic_inlet_width</td>\n",
       "      <td>0.1176</td>\n",
       "      <td>0.0375</td>\n",
       "      <td>1.0466</td>\n",
       "      <td>finngen_R9_O15_LABOUR_PELVIC_ABNORM</td>\n",
       "      <td>0.0217</td>\n",
       "      <td>0.0034</td>\n",
       "      <td>1.0466</td>\n",
       "      <td>0.4526</td>\n",
       "      <td>0.175</td>\n",
       "      <td>2.5859</td>\n",
       "      <td>0.0097</td>\n",
       "      <td>2.013228</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>851</th>\n",
       "      <td>acetabular_diameter</td>\n",
       "      <td>0.3373</td>\n",
       "      <td>0.0436</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>finngen_R9_O15_LABOUR_PELVIC_ABNORM</td>\n",
       "      <td>0.0217</td>\n",
       "      <td>0.0034</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>0.0743</td>\n",
       "      <td>0.0934</td>\n",
       "      <td>0.7958</td>\n",
       "      <td>0.4262</td>\n",
       "      <td>0.370387</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>709</th>\n",
       "      <td>acetabular_inclination</td>\n",
       "      <td>0.2538</td>\n",
       "      <td>0.0438</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>finngen_R9_O15_LABOUR_PELVIC_ABNORM</td>\n",
       "      <td>0.0217</td>\n",
       "      <td>0.0034</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>-0.3268</td>\n",
       "      <td>0.1065</td>\n",
       "      <td>-3.069</td>\n",
       "      <td>0.0021</td>\n",
       "      <td>2.677781</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>825</th>\n",
       "      <td>arm_devide_torso</td>\n",
       "      <td>0.4144</td>\n",
       "      <td>0.0456</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>finngen_R9_O15_LABOUR_PELVIC_ABNORM</td>\n",
       "      <td>0.0217</td>\n",
       "      <td>0.0034</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>0.008</td>\n",
       "      <td>0.0812</td>\n",
       "      <td>0.098</td>\n",
       "      <td>0.922</td>\n",
       "      <td>0.035269</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1582</th>\n",
       "      <td>arm_divide_leg</td>\n",
       "      <td>0.4272</td>\n",
       "      <td>0.0445</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>finngen_R9_O15_LABOUR_PELVIC_ABNORM</td>\n",
       "      <td>0.0217</td>\n",
       "      <td>0.0034</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>0.0939</td>\n",
       "      <td>0.088</td>\n",
       "      <td>1.0672</td>\n",
       "      <td>0.2859</td>\n",
       "      <td>0.543786</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1428</th>\n",
       "      <td>bi_acetabular_width</td>\n",
       "      <td>0.4971</td>\n",
       "      <td>0.0517</td>\n",
       "      <td>1.1459</td>\n",
       "      <td>finngen_R9_O15_LABOUR_PELVIC_ABNORM</td>\n",
       "      <td>0.0217</td>\n",
       "      <td>0.0034</td>\n",
       "      <td>1.1459</td>\n",
       "      <td>-0.5607</td>\n",
       "      <td>0.0829</td>\n",
       "      <td>-6.7649</td>\n",
       "      <td>0.0</td>\n",
       "      <td>10.874942</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1421</th>\n",
       "      <td>ear_left2ear_right</td>\n",
       "      <td>0.3461</td>\n",
       "      <td>0.0406</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>finngen_R9_O15_LABOUR_PELVIC_ABNORM</td>\n",
       "      <td>0.0217</td>\n",
       "      <td>0.0034</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>-0.0679</td>\n",
       "      <td>0.0861</td>\n",
       "      <td>-0.7891</td>\n",
       "      <td>0.4301</td>\n",
       "      <td>0.366431</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>925</th>\n",
       "      <td>head_area_divide_pelvic_inlet_area</td>\n",
       "      <td>0.3165</td>\n",
       "      <td>0.0378</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>finngen_R9_O15_LABOUR_PELVIC_ABNORM</td>\n",
       "      <td>0.0217</td>\n",
       "      <td>0.0034</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>0.356</td>\n",
       "      <td>0.0864</td>\n",
       "      <td>4.1179</td>\n",
       "      <td>0.000038</td>\n",
       "      <td>4.41755</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>714</th>\n",
       "      <td>head_divide_inlet_width</td>\n",
       "      <td>0.3650</td>\n",
       "      <td>0.0384</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>finngen_R9_O15_LABOUR_PELVIC_ABNORM</td>\n",
       "      <td>0.0217</td>\n",
       "      <td>0.0034</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>0.2165</td>\n",
       "      <td>0.0894</td>\n",
       "      <td>2.4224</td>\n",
       "      <td>0.0154</td>\n",
       "      <td>1.812479</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1619</th>\n",
       "      <td>head_divide_oblique_inlet_length</td>\n",
       "      <td>0.3152</td>\n",
       "      <td>0.0384</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>finngen_R9_O15_LABOUR_PELVIC_ABNORM</td>\n",
       "      <td>0.0217</td>\n",
       "      <td>0.0034</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>0.3264</td>\n",
       "      <td>0.0862</td>\n",
       "      <td>3.7867</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>3.69897</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>623</th>\n",
       "      <td>iliac_flare_angle</td>\n",
       "      <td>0.3366</td>\n",
       "      <td>0.0421</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>finngen_R9_O15_LABOUR_PELVIC_ABNORM</td>\n",
       "      <td>0.0217</td>\n",
       "      <td>0.0034</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>0.1325</td>\n",
       "      <td>0.0849</td>\n",
       "      <td>1.5603</td>\n",
       "      <td>0.1187</td>\n",
       "      <td>0.925549</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>iliac_flare_ratio</td>\n",
       "      <td>0.3732</td>\n",
       "      <td>0.0465</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>finngen_R9_O15_LABOUR_PELVIC_ABNORM</td>\n",
       "      <td>0.0217</td>\n",
       "      <td>0.0034</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>-0.2191</td>\n",
       "      <td>0.0946</td>\n",
       "      <td>-2.3166</td>\n",
       "      <td>0.0205</td>\n",
       "      <td>1.688246</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>221</th>\n",
       "      <td>iliac_isthmus_breadth</td>\n",
       "      <td>0.3036</td>\n",
       "      <td>0.0397</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>finngen_R9_O15_LABOUR_PELVIC_ABNORM</td>\n",
       "      <td>0.0217</td>\n",
       "      <td>0.0034</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>0.2173</td>\n",
       "      <td>0.0942</td>\n",
       "      <td>2.3064</td>\n",
       "      <td>0.0211</td>\n",
       "      <td>1.675718</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>443</th>\n",
       "      <td>leg_divide_torso</td>\n",
       "      <td>0.4256</td>\n",
       "      <td>0.0412</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>finngen_R9_O15_LABOUR_PELVIC_ABNORM</td>\n",
       "      <td>0.0217</td>\n",
       "      <td>0.0034</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>-0.1212</td>\n",
       "      <td>0.0819</td>\n",
       "      <td>-1.4807</td>\n",
       "      <td>0.1387</td>\n",
       "      <td>0.857924</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162</th>\n",
       "      <td>oblique_pelvic_inlet_length</td>\n",
       "      <td>0.3077</td>\n",
       "      <td>0.0393</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>finngen_R9_O15_LABOUR_PELVIC_ABNORM</td>\n",
       "      <td>0.0217</td>\n",
       "      <td>0.0034</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>-0.3937</td>\n",
       "      <td>0.0896</td>\n",
       "      <td>-4.3937</td>\n",
       "      <td>0.000011</td>\n",
       "      <td>4.953037</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>476</th>\n",
       "      <td>pelvic_height</td>\n",
       "      <td>0.4257</td>\n",
       "      <td>0.0531</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>finngen_R9_O15_LABOUR_PELVIC_ABNORM</td>\n",
       "      <td>0.0217</td>\n",
       "      <td>0.0034</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>-0.4011</td>\n",
       "      <td>0.0869</td>\n",
       "      <td>-4.6154</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>5.406348</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1156</th>\n",
       "      <td>pelvic_inlet_area</td>\n",
       "      <td>0.3306</td>\n",
       "      <td>0.0391</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>finngen_R9_O15_LABOUR_PELVIC_ABNORM</td>\n",
       "      <td>0.0217</td>\n",
       "      <td>0.0034</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>-0.4852</td>\n",
       "      <td>0.0895</td>\n",
       "      <td>-5.4182</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7.220389</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>951</th>\n",
       "      <td>pelvic_inlet_width</td>\n",
       "      <td>0.4342</td>\n",
       "      <td>0.0440</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>finngen_R9_O15_LABOUR_PELVIC_ABNORM</td>\n",
       "      <td>0.0217</td>\n",
       "      <td>0.0034</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>-0.2877</td>\n",
       "      <td>0.0852</td>\n",
       "      <td>-3.3769</td>\n",
       "      <td>0.0007</td>\n",
       "      <td>3.154902</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>496</th>\n",
       "      <td>pelvic_width</td>\n",
       "      <td>0.4251</td>\n",
       "      <td>0.0451</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>finngen_R9_O15_LABOUR_PELVIC_ABNORM</td>\n",
       "      <td>0.0217</td>\n",
       "      <td>0.0034</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>-0.0895</td>\n",
       "      <td>0.0761</td>\n",
       "      <td>-1.1765</td>\n",
       "      <td>0.2394</td>\n",
       "      <td>0.620876</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>833</th>\n",
       "      <td>shoulder_area_divide_pelvic_inlet_area</td>\n",
       "      <td>0.3439</td>\n",
       "      <td>0.0376</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>finngen_R9_O15_LABOUR_PELVIC_ABNORM</td>\n",
       "      <td>0.0217</td>\n",
       "      <td>0.0034</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>0.3151</td>\n",
       "      <td>0.0862</td>\n",
       "      <td>3.6542</td>\n",
       "      <td>0.0003</td>\n",
       "      <td>3.522879</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340</th>\n",
       "      <td>shoulder_divide_inlet_width</td>\n",
       "      <td>0.4305</td>\n",
       "      <td>0.0415</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>finngen_R9_O15_LABOUR_PELVIC_ABNORM</td>\n",
       "      <td>0.0217</td>\n",
       "      <td>0.0034</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>0.1686</td>\n",
       "      <td>0.0864</td>\n",
       "      <td>1.9514</td>\n",
       "      <td>0.051</td>\n",
       "      <td>1.29243</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>198</th>\n",
       "      <td>shoulder_divide_oblique_inlet_length</td>\n",
       "      <td>0.3072</td>\n",
       "      <td>0.0379</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>finngen_R9_O15_LABOUR_PELVIC_ABNORM</td>\n",
       "      <td>0.0217</td>\n",
       "      <td>0.0034</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>0.3184</td>\n",
       "      <td>0.0876</td>\n",
       "      <td>3.6334</td>\n",
       "      <td>0.0003</td>\n",
       "      <td>3.522879</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>185</th>\n",
       "      <td>shoulder_width</td>\n",
       "      <td>0.3539</td>\n",
       "      <td>0.0394</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>finngen_R9_O15_LABOUR_PELVIC_ABNORM</td>\n",
       "      <td>0.0217</td>\n",
       "      <td>0.0034</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>-0.0822</td>\n",
       "      <td>0.0894</td>\n",
       "      <td>-0.9193</td>\n",
       "      <td>0.3579</td>\n",
       "      <td>0.446238</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>519</th>\n",
       "      <td>subpubic_angle</td>\n",
       "      <td>0.3723</td>\n",
       "      <td>0.0470</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>finngen_R9_O15_LABOUR_PELVIC_ABNORM</td>\n",
       "      <td>0.0217</td>\n",
       "      <td>0.0034</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>-0.2271</td>\n",
       "      <td>0.0867</td>\n",
       "      <td>-2.6194</td>\n",
       "      <td>0.0088</td>\n",
       "      <td>2.055517</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1028</th>\n",
       "      <td>trochanter_left2trochanter_right</td>\n",
       "      <td>0.3651</td>\n",
       "      <td>0.0533</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>finngen_R9_O15_LABOUR_PELVIC_ABNORM</td>\n",
       "      <td>0.0217</td>\n",
       "      <td>0.0034</td>\n",
       "      <td>1.0957</td>\n",
       "      <td>-0.2445</td>\n",
       "      <td>0.0897</td>\n",
       "      <td>-2.7263</td>\n",
       "      <td>0.0064</td>\n",
       "      <td>2.19382</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                pheno_pelvic  pheno_pelvic_h2g  \\\n",
       "465    BW_devide_oblique_pelvic_inlet_length            0.1310   \n",
       "1627             BW_devide_pelvic_inlet_area            0.1462   \n",
       "847             BW_devide_pelvic_inlet_width            0.1176   \n",
       "851                      acetabular_diameter            0.3373   \n",
       "709                   acetabular_inclination            0.2538   \n",
       "825                         arm_devide_torso            0.4144   \n",
       "1582                          arm_divide_leg            0.4272   \n",
       "1428                     bi_acetabular_width            0.4971   \n",
       "1421                      ear_left2ear_right            0.3461   \n",
       "925       head_area_divide_pelvic_inlet_area            0.3165   \n",
       "714                  head_divide_inlet_width            0.3650   \n",
       "1619        head_divide_oblique_inlet_length            0.3152   \n",
       "623                        iliac_flare_angle            0.3366   \n",
       "56                         iliac_flare_ratio            0.3732   \n",
       "221                    iliac_isthmus_breadth            0.3036   \n",
       "443                         leg_divide_torso            0.4256   \n",
       "162              oblique_pelvic_inlet_length            0.3077   \n",
       "476                            pelvic_height            0.4257   \n",
       "1156                       pelvic_inlet_area            0.3306   \n",
       "951                       pelvic_inlet_width            0.4342   \n",
       "496                             pelvic_width            0.4251   \n",
       "833   shoulder_area_divide_pelvic_inlet_area            0.3439   \n",
       "340              shoulder_divide_inlet_width            0.4305   \n",
       "198     shoulder_divide_oblique_inlet_length            0.3072   \n",
       "185                           shoulder_width            0.3539   \n",
       "519                           subpubic_angle            0.3723   \n",
       "1028        trochanter_left2trochanter_right            0.3651   \n",
       "\n",
       "      pheno_pelvic_h2g_se  pheno_pelvic_lambda  \\\n",
       "465                0.0376               1.0466   \n",
       "1627               0.0375               1.0466   \n",
       "847                0.0375               1.0466   \n",
       "851                0.0436               1.0957   \n",
       "709                0.0438               1.0957   \n",
       "825                0.0456               1.0957   \n",
       "1582               0.0445               1.0957   \n",
       "1428               0.0517               1.1459   \n",
       "1421               0.0406               1.0957   \n",
       "925                0.0378               1.0957   \n",
       "714                0.0384               1.0957   \n",
       "1619               0.0384               1.0957   \n",
       "623                0.0421               1.0957   \n",
       "56                 0.0465               1.0957   \n",
       "221                0.0397               1.0957   \n",
       "443                0.0412               1.0957   \n",
       "162                0.0393               1.0957   \n",
       "476                0.0531               1.0957   \n",
       "1156               0.0391               1.0957   \n",
       "951                0.0440               1.0957   \n",
       "496                0.0451               1.0957   \n",
       "833                0.0376               1.0957   \n",
       "340                0.0415               1.0957   \n",
       "198                0.0379               1.0957   \n",
       "185                0.0394               1.0957   \n",
       "519                0.0470               1.0957   \n",
       "1028               0.0533               1.0957   \n",
       "\n",
       "                            pheno_outcome  pheno_outcome_h2g  \\\n",
       "465   finngen_R9_O15_LABOUR_PELVIC_ABNORM             0.0217   \n",
       "1627  finngen_R9_O15_LABOUR_PELVIC_ABNORM             0.0217   \n",
       "847   finngen_R9_O15_LABOUR_PELVIC_ABNORM             0.0217   \n",
       "851   finngen_R9_O15_LABOUR_PELVIC_ABNORM             0.0217   \n",
       "709   finngen_R9_O15_LABOUR_PELVIC_ABNORM             0.0217   \n",
       "825   finngen_R9_O15_LABOUR_PELVIC_ABNORM             0.0217   \n",
       "1582  finngen_R9_O15_LABOUR_PELVIC_ABNORM             0.0217   \n",
       "1428  finngen_R9_O15_LABOUR_PELVIC_ABNORM             0.0217   \n",
       "1421  finngen_R9_O15_LABOUR_PELVIC_ABNORM             0.0217   \n",
       "925   finngen_R9_O15_LABOUR_PELVIC_ABNORM             0.0217   \n",
       "714   finngen_R9_O15_LABOUR_PELVIC_ABNORM             0.0217   \n",
       "1619  finngen_R9_O15_LABOUR_PELVIC_ABNORM             0.0217   \n",
       "623   finngen_R9_O15_LABOUR_PELVIC_ABNORM             0.0217   \n",
       "56    finngen_R9_O15_LABOUR_PELVIC_ABNORM             0.0217   \n",
       "221   finngen_R9_O15_LABOUR_PELVIC_ABNORM             0.0217   \n",
       "443   finngen_R9_O15_LABOUR_PELVIC_ABNORM             0.0217   \n",
       "162   finngen_R9_O15_LABOUR_PELVIC_ABNORM             0.0217   \n",
       "476   finngen_R9_O15_LABOUR_PELVIC_ABNORM             0.0217   \n",
       "1156  finngen_R9_O15_LABOUR_PELVIC_ABNORM             0.0217   \n",
       "951   finngen_R9_O15_LABOUR_PELVIC_ABNORM             0.0217   \n",
       "496   finngen_R9_O15_LABOUR_PELVIC_ABNORM             0.0217   \n",
       "833   finngen_R9_O15_LABOUR_PELVIC_ABNORM             0.0217   \n",
       "340   finngen_R9_O15_LABOUR_PELVIC_ABNORM             0.0217   \n",
       "198   finngen_R9_O15_LABOUR_PELVIC_ABNORM             0.0217   \n",
       "185   finngen_R9_O15_LABOUR_PELVIC_ABNORM             0.0217   \n",
       "519   finngen_R9_O15_LABOUR_PELVIC_ABNORM             0.0217   \n",
       "1028  finngen_R9_O15_LABOUR_PELVIC_ABNORM             0.0217   \n",
       "\n",
       "      pheno_outcome_h2g_se  pheno_outcome_lambda     cor      se       z  \\\n",
       "465                 0.0034                1.0466   0.559  0.1561   3.581   \n",
       "1627                0.0034                1.0466  0.7444  0.1598   4.657   \n",
       "847                 0.0034                1.0466  0.4526   0.175  2.5859   \n",
       "851                 0.0034                1.0957  0.0743  0.0934  0.7958   \n",
       "709                 0.0034                1.0957 -0.3268  0.1065  -3.069   \n",
       "825                 0.0034                1.0957   0.008  0.0812   0.098   \n",
       "1582                0.0034                1.0957  0.0939   0.088  1.0672   \n",
       "1428                0.0034                1.1459 -0.5607  0.0829 -6.7649   \n",
       "1421                0.0034                1.0957 -0.0679  0.0861 -0.7891   \n",
       "925                 0.0034                1.0957   0.356  0.0864  4.1179   \n",
       "714                 0.0034                1.0957  0.2165  0.0894  2.4224   \n",
       "1619                0.0034                1.0957  0.3264  0.0862  3.7867   \n",
       "623                 0.0034                1.0957  0.1325  0.0849  1.5603   \n",
       "56                  0.0034                1.0957 -0.2191  0.0946 -2.3166   \n",
       "221                 0.0034                1.0957  0.2173  0.0942  2.3064   \n",
       "443                 0.0034                1.0957 -0.1212  0.0819 -1.4807   \n",
       "162                 0.0034                1.0957 -0.3937  0.0896 -4.3937   \n",
       "476                 0.0034                1.0957 -0.4011  0.0869 -4.6154   \n",
       "1156                0.0034                1.0957 -0.4852  0.0895 -5.4182   \n",
       "951                 0.0034                1.0957 -0.2877  0.0852 -3.3769   \n",
       "496                 0.0034                1.0957 -0.0895  0.0761 -1.1765   \n",
       "833                 0.0034                1.0957  0.3151  0.0862  3.6542   \n",
       "340                 0.0034                1.0957  0.1686  0.0864  1.9514   \n",
       "198                 0.0034                1.0957  0.3184  0.0876  3.6334   \n",
       "185                 0.0034                1.0957 -0.0822  0.0894 -0.9193   \n",
       "519                 0.0034                1.0957 -0.2271  0.0867 -2.6194   \n",
       "1028                0.0034                1.0957 -0.2445  0.0897 -2.7263   \n",
       "\n",
       "       p_value  neglog10p  \n",
       "465     0.0003   3.522879  \n",
       "1627  0.000003   5.493752  \n",
       "847     0.0097   2.013228  \n",
       "851     0.4262   0.370387  \n",
       "709     0.0021   2.677781  \n",
       "825      0.922   0.035269  \n",
       "1582    0.2859   0.543786  \n",
       "1428       0.0  10.874942  \n",
       "1421    0.4301   0.366431  \n",
       "925   0.000038    4.41755  \n",
       "714     0.0154   1.812479  \n",
       "1619    0.0002    3.69897  \n",
       "623     0.1187   0.925549  \n",
       "56      0.0205   1.688246  \n",
       "221     0.0211   1.675718  \n",
       "443     0.1387   0.857924  \n",
       "162   0.000011   4.953037  \n",
       "476   0.000004   5.406348  \n",
       "1156       0.0   7.220389  \n",
       "951     0.0007   3.154902  \n",
       "496     0.2394   0.620876  \n",
       "833     0.0003   3.522879  \n",
       "340      0.051    1.29243  \n",
       "198     0.0003   3.522879  \n",
       "185     0.3579   0.446238  \n",
       "519     0.0088   2.055517  \n",
       "1028    0.0064    2.19382  "
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[df['pheno_outcome'] == \"finngen_R9_O15_LABOUR_PELVIC_ABNORM\"]"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Male"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.DataFrame(columns=['pheno_pelvic', 'pheno_pelvic_h2g', 'pheno_pelvic_h2g_se', 'pheno_pelvic_lambda',\n",
    "                           'pheno_outcome', 'pheno_outcome_h2g', 'pheno_outcome_h2g_se', 'pheno_outcome_lambda',\n",
    "                           'cor', 'se', 'z', 'p_value'])\n",
    "\n",
    "path = \"UKB_Imaging_Genetics/LDSC_OUTPUT/rg_pheno_outcome_20230909/male/\"\n",
    "\n",
    "for file in os.listdir(path):\n",
    "    if file.endswith(\".log\"):\n",
    "        file_modified = file.replace(\"_-_\", \"_\")\n",
    "        # get phenotype name\n",
    "        pheno1 = file_modified.split('-')[0]\n",
    "        pheno2 = file_modified.split('-')[1]\n",
    "        pheno1 = pheno1.split('.')[0]\n",
    "        pheno2 = pheno2.split('_rg.log')[0]\n",
    "\n",
    "        with open(os.path.join(path, file), \"r\") as f:\n",
    "            lines = f.readlines()\n",
    "\n",
    "        for i, line in enumerate(lines):\n",
    "            # get h2g\n",
    "            if line.startswith(\"Heritability of phenotype 1\") and i + 2 < len(lines):\n",
    "                h2g_1 = lines[i + 2].split(\" \")[4].strip()\n",
    "                h2g_1 = float(h2g_1)\n",
    "\n",
    "                h2g_1_se = lines[i + 2].split(\" \")[5].strip()\n",
    "                h2g_1_se = h2g_1_se[1:-1]\n",
    "                h2g_1_se = float(h2g_1_se)\n",
    "\n",
    "                pheno1_lambda = lines[i + 3].split(\" \")[2].strip()\n",
    "                pheno1_lambda = float(pheno1_lambda)\n",
    "            \n",
    "            if line.startswith(\"Heritability of phenotype 2/2\") and i + 2 < len(lines):\n",
    "                h2g_2 = lines[i + 2].split(\" \")[4].strip()\n",
    "                h2g_2 = float(h2g_2)\n",
    "                \n",
    "                h2g_2_se = lines[i + 2].split(\" \")[5].strip()\n",
    "                h2g_2_se = h2g_2_se[1:-1]\n",
    "                h2g_2_se = float(h2g_2_se)\n",
    "\n",
    "                pheno2_lambda = lines[i + 3].split(\" \")[2].strip()\n",
    "                pheno2_lambda = float(pheno1_lambda)\n",
    "\n",
    "            # get rg\n",
    "            if line.startswith(\"Genetic Correlation:\"):\n",
    "                # print(line)\n",
    "                rg = line.split(\" \")[2].strip()\n",
    "                rg = float(rg)\n",
    "                # print(rg)\n",
    "                se = line.split(\" \")[3].strip()\n",
    "                se = se[1:-1]\n",
    "                se = float(se)\n",
    "            if line.startswith(\"Z-score:\"):\n",
    "                z = line.split(\" \")[1].strip()\n",
    "                z = float(z)\n",
    "            if line.startswith(\"P:\"):\n",
    "                p_value = line.split(\" \")[1].strip()\n",
    "                p_value = float(p_value)\n",
    "                neglog10p = -np.log10(p_value)\n",
    "            if line.startswith(\"ERROR\"):\n",
    "                rg = \"NA\"\n",
    "                se = \"NA\"\n",
    "                z = \"NA\"\n",
    "                p_value = \"NA\"\n",
    "                neglog10p = \"NA\"\n",
    "                break\n",
    "\n",
    "        df = df.append({'pheno_pelvic': pheno1, 'pheno_pelvic_h2g': h2g_1, 'pheno_pelvic_h2g_se': h2g_1_se, 'pheno_pelvic_lambda': pheno1_lambda,\n",
    "                        'pheno_outcome': pheno2, 'pheno_outcome_h2g': h2g_2, 'pheno_outcome_h2g_se': h2g_2_se, 'pheno_outcome_lambda': pheno2_lambda,\n",
    "                        'cor': rg, 'se': se, 'z': z, 'p_value': p_value, 'neglog10p': neglog10p}, \n",
    "                        ignore_index=True)\n",
    "\n",
    "df.sort_values(by = ['pheno_pelvic', 'pheno_outcome'], inplace=True)\n",
    "df.to_csv(\"key_results/genetic_correlation/rg_20230909/rg_hip_pelvic_vs_outcome_male.csv\", index=False)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Both"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.DataFrame(columns=['pheno_pelvic', 'pheno_pelvic_h2g', 'pheno_pelvic_h2g_se', 'pheno_pelvic_lambda',\n",
    "                           'pheno_outcome', 'pheno_outcome_h2g', 'pheno_outcome_h2g_se', 'pheno_outcome_lambda',\n",
    "                           'cor', 'se', 'z', 'p_value'])\n",
    "\n",
    "path = \"UKB_Imaging_Genetics/LDSC_OUTPUT/rg_pheno_outcome_20230909/both/\"\n",
    "\n",
    "for file in os.listdir(path):\n",
    "    if file.endswith(\".log\"):\n",
    "        file_modified = file.replace(\"_-_\", \"_\")\n",
    "        # get phenotype name\n",
    "        pheno1 = file_modified.split('-')[0]\n",
    "        pheno2 = file_modified.split('-')[1]\n",
    "        pheno1 = pheno1.split('.')[0]\n",
    "        pheno2 = pheno2.split('_rg.log')[0]\n",
    "\n",
    "        with open(os.path.join(path, file), \"r\") as f:\n",
    "            lines = f.readlines()\n",
    "\n",
    "        for i, line in enumerate(lines):\n",
    "            # get h2g\n",
    "            if line.startswith(\"Heritability of phenotype 1\") and i + 2 < len(lines):\n",
    "                h2g_1 = lines[i + 2].split(\" \")[4].strip()\n",
    "                h2g_1 = float(h2g_1)\n",
    "\n",
    "                h2g_1_se = lines[i + 2].split(\" \")[5].strip()\n",
    "                h2g_1_se = h2g_1_se[1:-1]\n",
    "                h2g_1_se = float(h2g_1_se)\n",
    "\n",
    "                pheno1_lambda = lines[i + 3].split(\" \")[2].strip()\n",
    "                pheno1_lambda = float(pheno1_lambda)\n",
    "            \n",
    "            if line.startswith(\"Heritability of phenotype 2/2\") and i + 2 < len(lines):\n",
    "                h2g_2 = lines[i + 2].split(\" \")[4].strip()\n",
    "                h2g_2 = float(h2g_2)\n",
    "                \n",
    "                h2g_2_se = lines[i + 2].split(\" \")[5].strip()\n",
    "                h2g_2_se = h2g_2_se[1:-1]\n",
    "                h2g_2_se = float(h2g_2_se)\n",
    "\n",
    "                pheno2_lambda = lines[i + 3].split(\" \")[2].strip()\n",
    "                pheno2_lambda = float(pheno1_lambda)\n",
    "\n",
    "            # get rg\n",
    "            if line.startswith(\"Genetic Correlation:\"):\n",
    "                # print(line)\n",
    "                rg = line.split(\" \")[2].strip()\n",
    "                rg = float(rg)\n",
    "                # print(rg)\n",
    "                se = line.split(\" \")[3].strip()\n",
    "                se = se[1:-1]\n",
    "                se = float(se)\n",
    "            if line.startswith(\"Z-score:\"):\n",
    "                z = line.split(\" \")[1].strip()\n",
    "                z = float(z)\n",
    "            if line.startswith(\"P:\"):\n",
    "                p_value = line.split(\" \")[1].strip()\n",
    "                p_value = float(p_value)\n",
    "                neglog10p = -np.log10(p_value)\n",
    "            if line.startswith(\"ERROR\"):\n",
    "                rg = \"NA\"\n",
    "                se = \"NA\"\n",
    "                z = \"NA\"\n",
    "                p_value = \"NA\"\n",
    "                neglog10p = \"NA\"\n",
    "                break\n",
    "\n",
    "        df = df.append({'pheno_pelvic': pheno1, 'pheno_pelvic_h2g': h2g_1, 'pheno_pelvic_h2g_se': h2g_1_se, 'pheno_pelvic_lambda': pheno1_lambda,\n",
    "                        'pheno_outcome': pheno2, 'pheno_outcome_h2g': h2g_2, 'pheno_outcome_h2g_se': h2g_2_se, 'pheno_outcome_lambda': pheno2_lambda,\n",
    "                        'cor': rg, 'se': se, 'z': z, 'p_value': p_value, 'neglog10p': neglog10p}, \n",
    "                        ignore_index=True)\n",
    "\n",
    "df.sort_values(by = ['pheno_pelvic', 'pheno_outcome'], inplace=True)\n",
    "df.to_csv(\"key_results/genetic_correlation/rg_20230909/rg_hip_pelvic_vs_outcome_both.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pheno_pelvic</th>\n",
       "      <th>pheno_pelvic_h2g</th>\n",
       "      <th>pheno_pelvic_h2g_se</th>\n",
       "      <th>pheno_pelvic_lambda</th>\n",
       "      <th>pheno_outcome</th>\n",
       "      <th>pheno_outcome_h2g</th>\n",
       "      <th>pheno_outcome_h2g_se</th>\n",
       "      <th>pheno_outcome_lambda</th>\n",
       "      <th>cor</th>\n",
       "      <th>se</th>\n",
       "      <th>z</th>\n",
       "      <th>p_value</th>\n",
       "      <th>neglog10p</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1089</th>\n",
       "      <td>acetabular_diameter</td>\n",
       "      <td>0.2930</td>\n",
       "      <td>0.0311</td>\n",
       "      <td>1.1459</td>\n",
       "      <td>EGG_Maternal_GWAS_META_gestational_duration</td>\n",
       "      <td>0.0357</td>\n",
       "      <td>0.0046</td>\n",
       "      <td>1.1459</td>\n",
       "      <td>0.0111</td>\n",
       "      <td>0.0703</td>\n",
       "      <td>0.158</td>\n",
       "      <td>0.8745</td>\n",
       "      <td>0.05824</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>745</th>\n",
       "      <td>acetabular_diameter</td>\n",
       "      <td>0.2921</td>\n",
       "      <td>0.0308</td>\n",
       "      <td>1.1459</td>\n",
       "      <td>EGG_Maternal_GWAS_META_postterm_delivery</td>\n",
       "      <td>0.0148</td>\n",
       "      <td>0.0033</td>\n",
       "      <td>1.1459</td>\n",
       "      <td>0.1404</td>\n",
       "      <td>0.1018</td>\n",
       "      <td>1.3785</td>\n",
       "      <td>0.168</td>\n",
       "      <td>0.774691</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>606</th>\n",
       "      <td>acetabular_diameter</td>\n",
       "      <td>0.2921</td>\n",
       "      <td>0.0312</td>\n",
       "      <td>1.1459</td>\n",
       "      <td>EGG_Maternal_GWAS_META_preterm_delivery</td>\n",
       "      <td>0.0109</td>\n",
       "      <td>0.0020</td>\n",
       "      <td>1.1459</td>\n",
       "      <td>0.0271</td>\n",
       "      <td>0.0876</td>\n",
       "      <td>0.3097</td>\n",
       "      <td>0.7568</td>\n",
       "      <td>0.121019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1376</th>\n",
       "      <td>acetabular_diameter</td>\n",
       "      <td>0.2931</td>\n",
       "      <td>0.0309</td>\n",
       "      <td>1.1459</td>\n",
       "      <td>Fetal_BW_European_meta.NG2019</td>\n",
       "      <td>0.0988</td>\n",
       "      <td>0.0058</td>\n",
       "      <td>1.1459</td>\n",
       "      <td>0.1145</td>\n",
       "      <td>0.0369</td>\n",
       "      <td>3.1049</td>\n",
       "      <td>0.0019</td>\n",
       "      <td>2.721246</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>552</th>\n",
       "      <td>acetabular_diameter</td>\n",
       "      <td>0.2931</td>\n",
       "      <td>0.0309</td>\n",
       "      <td>1.1459</td>\n",
       "      <td>Fetal_BW_transethnic_meta.NG2019</td>\n",
       "      <td>0.0941</td>\n",
       "      <td>0.0055</td>\n",
       "      <td>1.1459</td>\n",
       "      <td>0.1218</td>\n",
       "      <td>0.0358</td>\n",
       "      <td>3.402</td>\n",
       "      <td>0.0007</td>\n",
       "      <td>3.154902</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1632</th>\n",
       "      <td>trochanter_left2trochanter_right</td>\n",
       "      <td>0.3528</td>\n",
       "      <td>0.0379</td>\n",
       "      <td>1.2005</td>\n",
       "      <td>neale_Number_of_stillbirths</td>\n",
       "      <td>0.0077</td>\n",
       "      <td>0.0081</td>\n",
       "      <td>1.2005</td>\n",
       "      <td>-0.0249</td>\n",
       "      <td>0.1774</td>\n",
       "      <td>-0.1402</td>\n",
       "      <td>0.8885</td>\n",
       "      <td>0.051343</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81</th>\n",
       "      <td>trochanter_left2trochanter_right</td>\n",
       "      <td>0.3528</td>\n",
       "      <td>0.0379</td>\n",
       "      <td>1.2005</td>\n",
       "      <td>neale_Oestradiol_pmol_L</td>\n",
       "      <td>0.0235</td>\n",
       "      <td>0.0092</td>\n",
       "      <td>1.2005</td>\n",
       "      <td>0.1536</td>\n",
       "      <td>0.108</td>\n",
       "      <td>1.4233</td>\n",
       "      <td>0.1546</td>\n",
       "      <td>0.810791</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>343</th>\n",
       "      <td>trochanter_left2trochanter_right</td>\n",
       "      <td>0.3528</td>\n",
       "      <td>0.0379</td>\n",
       "      <td>1.2005</td>\n",
       "      <td>neale_Years_since_last_breast_cancer_screening...</td>\n",
       "      <td>0.0049</td>\n",
       "      <td>0.0044</td>\n",
       "      <td>1.2005</td>\n",
       "      <td>-0.1141</td>\n",
       "      <td>0.1741</td>\n",
       "      <td>-0.6556</td>\n",
       "      <td>0.5121</td>\n",
       "      <td>0.290645</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>564</th>\n",
       "      <td>trochanter_left2trochanter_right</td>\n",
       "      <td>0.3528</td>\n",
       "      <td>0.0379</td>\n",
       "      <td>1.2005</td>\n",
       "      <td>neale_Years_since_last_cervical_smear_test</td>\n",
       "      <td>0.0151</td>\n",
       "      <td>0.0028</td>\n",
       "      <td>1.2005</td>\n",
       "      <td>-0.1245</td>\n",
       "      <td>0.0881</td>\n",
       "      <td>-1.4132</td>\n",
       "      <td>0.1576</td>\n",
       "      <td>0.802444</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1101</th>\n",
       "      <td>trochanter_left2trochanter_right</td>\n",
       "      <td>0.3528</td>\n",
       "      <td>0.0379</td>\n",
       "      <td>1.2005</td>\n",
       "      <td>neale_usual_walking_pace</td>\n",
       "      <td>0.0808</td>\n",
       "      <td>0.0030</td>\n",
       "      <td>1.2005</td>\n",
       "      <td>0.0875</td>\n",
       "      <td>0.0375</td>\n",
       "      <td>2.335</td>\n",
       "      <td>0.0195</td>\n",
       "      <td>1.709965</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1743 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                          pheno_pelvic  pheno_pelvic_h2g  pheno_pelvic_h2g_se  \\\n",
       "1089               acetabular_diameter            0.2930               0.0311   \n",
       "745                acetabular_diameter            0.2921               0.0308   \n",
       "606                acetabular_diameter            0.2921               0.0312   \n",
       "1376               acetabular_diameter            0.2931               0.0309   \n",
       "552                acetabular_diameter            0.2931               0.0309   \n",
       "...                                ...               ...                  ...   \n",
       "1632  trochanter_left2trochanter_right            0.3528               0.0379   \n",
       "81    trochanter_left2trochanter_right            0.3528               0.0379   \n",
       "343   trochanter_left2trochanter_right            0.3528               0.0379   \n",
       "564   trochanter_left2trochanter_right            0.3528               0.0379   \n",
       "1101  trochanter_left2trochanter_right            0.3528               0.0379   \n",
       "\n",
       "      pheno_pelvic_lambda                                      pheno_outcome  \\\n",
       "1089               1.1459        EGG_Maternal_GWAS_META_gestational_duration   \n",
       "745                1.1459           EGG_Maternal_GWAS_META_postterm_delivery   \n",
       "606                1.1459            EGG_Maternal_GWAS_META_preterm_delivery   \n",
       "1376               1.1459                      Fetal_BW_European_meta.NG2019   \n",
       "552                1.1459                   Fetal_BW_transethnic_meta.NG2019   \n",
       "...                   ...                                                ...   \n",
       "1632               1.2005                        neale_Number_of_stillbirths   \n",
       "81                 1.2005                            neale_Oestradiol_pmol_L   \n",
       "343                1.2005  neale_Years_since_last_breast_cancer_screening...   \n",
       "564                1.2005         neale_Years_since_last_cervical_smear_test   \n",
       "1101               1.2005                           neale_usual_walking_pace   \n",
       "\n",
       "      pheno_outcome_h2g  pheno_outcome_h2g_se  pheno_outcome_lambda     cor  \\\n",
       "1089             0.0357                0.0046                1.1459  0.0111   \n",
       "745              0.0148                0.0033                1.1459  0.1404   \n",
       "606              0.0109                0.0020                1.1459  0.0271   \n",
       "1376             0.0988                0.0058                1.1459  0.1145   \n",
       "552              0.0941                0.0055                1.1459  0.1218   \n",
       "...                 ...                   ...                   ...     ...   \n",
       "1632             0.0077                0.0081                1.2005 -0.0249   \n",
       "81               0.0235                0.0092                1.2005  0.1536   \n",
       "343              0.0049                0.0044                1.2005 -0.1141   \n",
       "564              0.0151                0.0028                1.2005 -0.1245   \n",
       "1101             0.0808                0.0030                1.2005  0.0875   \n",
       "\n",
       "          se       z p_value neglog10p  \n",
       "1089  0.0703   0.158  0.8745   0.05824  \n",
       "745   0.1018  1.3785   0.168  0.774691  \n",
       "606   0.0876  0.3097  0.7568  0.121019  \n",
       "1376  0.0369  3.1049  0.0019  2.721246  \n",
       "552   0.0358   3.402  0.0007  3.154902  \n",
       "...      ...     ...     ...       ...  \n",
       "1632  0.1774 -0.1402  0.8885  0.051343  \n",
       "81     0.108  1.4233  0.1546  0.810791  \n",
       "343   0.1741 -0.6556  0.5121  0.290645  \n",
       "564   0.0881 -1.4132  0.1576  0.802444  \n",
       "1101  0.0375   2.335  0.0195  1.709965  \n",
       "\n",
       "[1743 rows x 13 columns]"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Combine male and female together to see if any difference"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [],
   "source": [
    "female_df = pd.read_csv(\"key_results/genetic_correlation/rg_20230909/rg_hip_pelvic_vs_outcome_female.csv\")\n",
    "male_df = pd.read_csv(\"key_results/genetic_correlation/rg_20230909/rg_hip_pelvic_vs_outcome_male.csv\")\n",
    "female_df = female_df[['pheno_pelvic', 'pheno_outcome', 'cor', 'se', 'z', 'p_value']]\n",
    "male_df = male_df[['pheno_pelvic', 'pheno_outcome', 'cor', 'se', 'z', 'p_value']]\n",
    "female_df.rename(columns = {'cor': 'female_cor', 'se': 'female_se', 'z': 'female_z', 'p_value': 'female_p'}, inplace = True)\n",
    "male_df.rename(columns = {'cor': 'male_cor', 'se': 'male_se', 'z': 'male_z', 'p_value': 'male_p'}, inplace = True)\n",
    "\n",
    "# Merge\n",
    "mf_df = female_df.merge(male_df, on = ['pheno_pelvic', 'pheno_outcome'], how = 'inner')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pheno_pelvic</th>\n",
       "      <th>pheno_outcome</th>\n",
       "      <th>female_cor</th>\n",
       "      <th>female_se</th>\n",
       "      <th>female_z</th>\n",
       "      <th>female_p</th>\n",
       "      <th>male_cor</th>\n",
       "      <th>male_se</th>\n",
       "      <th>male_z</th>\n",
       "      <th>male_p</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>acetabular_diameter</td>\n",
       "      <td>EGG_Maternal_GWAS_META_gestational_duration</td>\n",
       "      <td>0.1059</td>\n",
       "      <td>0.0772</td>\n",
       "      <td>1.3715</td>\n",
       "      <td>0.1702</td>\n",
       "      <td>-0.0216</td>\n",
       "      <td>0.0946</td>\n",
       "      <td>-0.2281</td>\n",
       "      <td>0.8195</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>acetabular_diameter</td>\n",
       "      <td>EGG_Maternal_GWAS_META_postterm_delivery</td>\n",
       "      <td>0.1092</td>\n",
       "      <td>0.1221</td>\n",
       "      <td>0.8947</td>\n",
       "      <td>0.3710</td>\n",
       "      <td>0.1926</td>\n",
       "      <td>0.1221</td>\n",
       "      <td>1.5778</td>\n",
       "      <td>0.1146</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>acetabular_diameter</td>\n",
       "      <td>EGG_Maternal_GWAS_META_preterm_delivery</td>\n",
       "      <td>-0.0014</td>\n",
       "      <td>0.1097</td>\n",
       "      <td>-0.0131</td>\n",
       "      <td>0.9895</td>\n",
       "      <td>0.0372</td>\n",
       "      <td>0.1062</td>\n",
       "      <td>0.3502</td>\n",
       "      <td>0.7262</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>acetabular_diameter</td>\n",
       "      <td>Fetal_BW_European_meta.NG2019</td>\n",
       "      <td>0.1096</td>\n",
       "      <td>0.0429</td>\n",
       "      <td>2.5537</td>\n",
       "      <td>0.0107</td>\n",
       "      <td>0.0912</td>\n",
       "      <td>0.0451</td>\n",
       "      <td>2.0224</td>\n",
       "      <td>0.0431</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>acetabular_diameter</td>\n",
       "      <td>Fetal_BW_transethnic_meta.NG2019</td>\n",
       "      <td>0.1177</td>\n",
       "      <td>0.0427</td>\n",
       "      <td>2.7569</td>\n",
       "      <td>0.0058</td>\n",
       "      <td>0.0987</td>\n",
       "      <td>0.0432</td>\n",
       "      <td>2.2851</td>\n",
       "      <td>0.0223</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1738</th>\n",
       "      <td>trochanter_left2trochanter_right</td>\n",
       "      <td>neale_Number_of_stillbirths</td>\n",
       "      <td>-0.1920</td>\n",
       "      <td>0.2417</td>\n",
       "      <td>-0.7943</td>\n",
       "      <td>0.4270</td>\n",
       "      <td>0.2431</td>\n",
       "      <td>0.2720</td>\n",
       "      <td>0.8937</td>\n",
       "      <td>0.3715</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1739</th>\n",
       "      <td>trochanter_left2trochanter_right</td>\n",
       "      <td>neale_Oestradiol_pmol_L</td>\n",
       "      <td>0.2034</td>\n",
       "      <td>0.1444</td>\n",
       "      <td>1.4082</td>\n",
       "      <td>0.1591</td>\n",
       "      <td>0.1240</td>\n",
       "      <td>0.1402</td>\n",
       "      <td>0.8844</td>\n",
       "      <td>0.3765</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1740</th>\n",
       "      <td>trochanter_left2trochanter_right</td>\n",
       "      <td>neale_Years_since_last_breast_cancer_screening...</td>\n",
       "      <td>0.0313</td>\n",
       "      <td>0.2276</td>\n",
       "      <td>0.1376</td>\n",
       "      <td>0.8906</td>\n",
       "      <td>-0.3138</td>\n",
       "      <td>0.2433</td>\n",
       "      <td>-1.2898</td>\n",
       "      <td>0.1971</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1741</th>\n",
       "      <td>trochanter_left2trochanter_right</td>\n",
       "      <td>neale_Years_since_last_cervical_smear_test</td>\n",
       "      <td>-0.2476</td>\n",
       "      <td>0.1090</td>\n",
       "      <td>-2.2725</td>\n",
       "      <td>0.0231</td>\n",
       "      <td>-0.0308</td>\n",
       "      <td>0.1079</td>\n",
       "      <td>-0.2851</td>\n",
       "      <td>0.7756</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1742</th>\n",
       "      <td>trochanter_left2trochanter_right</td>\n",
       "      <td>neale_usual_walking_pace</td>\n",
       "      <td>0.0487</td>\n",
       "      <td>0.0460</td>\n",
       "      <td>1.0585</td>\n",
       "      <td>0.2898</td>\n",
       "      <td>0.1091</td>\n",
       "      <td>0.0459</td>\n",
       "      <td>2.3791</td>\n",
       "      <td>0.0174</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1743 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                          pheno_pelvic  \\\n",
       "0                  acetabular_diameter   \n",
       "1                  acetabular_diameter   \n",
       "2                  acetabular_diameter   \n",
       "3                  acetabular_diameter   \n",
       "4                  acetabular_diameter   \n",
       "...                                ...   \n",
       "1738  trochanter_left2trochanter_right   \n",
       "1739  trochanter_left2trochanter_right   \n",
       "1740  trochanter_left2trochanter_right   \n",
       "1741  trochanter_left2trochanter_right   \n",
       "1742  trochanter_left2trochanter_right   \n",
       "\n",
       "                                          pheno_outcome  female_cor  \\\n",
       "0           EGG_Maternal_GWAS_META_gestational_duration      0.1059   \n",
       "1              EGG_Maternal_GWAS_META_postterm_delivery      0.1092   \n",
       "2               EGG_Maternal_GWAS_META_preterm_delivery     -0.0014   \n",
       "3                         Fetal_BW_European_meta.NG2019      0.1096   \n",
       "4                      Fetal_BW_transethnic_meta.NG2019      0.1177   \n",
       "...                                                 ...         ...   \n",
       "1738                        neale_Number_of_stillbirths     -0.1920   \n",
       "1739                            neale_Oestradiol_pmol_L      0.2034   \n",
       "1740  neale_Years_since_last_breast_cancer_screening...      0.0313   \n",
       "1741         neale_Years_since_last_cervical_smear_test     -0.2476   \n",
       "1742                           neale_usual_walking_pace      0.0487   \n",
       "\n",
       "      female_se  female_z  female_p  male_cor  male_se  male_z  male_p  \n",
       "0        0.0772    1.3715    0.1702   -0.0216   0.0946 -0.2281  0.8195  \n",
       "1        0.1221    0.8947    0.3710    0.1926   0.1221  1.5778  0.1146  \n",
       "2        0.1097   -0.0131    0.9895    0.0372   0.1062  0.3502  0.7262  \n",
       "3        0.0429    2.5537    0.0107    0.0912   0.0451  2.0224  0.0431  \n",
       "4        0.0427    2.7569    0.0058    0.0987   0.0432  2.2851  0.0223  \n",
       "...         ...       ...       ...       ...      ...     ...     ...  \n",
       "1738     0.2417   -0.7943    0.4270    0.2431   0.2720  0.8937  0.3715  \n",
       "1739     0.1444    1.4082    0.1591    0.1240   0.1402  0.8844  0.3765  \n",
       "1740     0.2276    0.1376    0.8906   -0.3138   0.2433 -1.2898  0.1971  \n",
       "1741     0.1090   -2.2725    0.0231   -0.0308   0.1079 -0.2851  0.7756  \n",
       "1742     0.0460    1.0585    0.2898    0.1091   0.0459  2.3791  0.0174  \n",
       "\n",
       "[1743 rows x 10 columns]"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mf_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [],
   "source": [
    "mf_df.to_csv(\"key_results/genetic_correlation/rg_20230909/rg_hip_pelvic_vs_outcome_male_and_female.csv\", index=False)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## rg for Handedness"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "!mkdir -p key_results/genetic_correlation/rg_20230830"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/7f/jzt97g754236ldjj13l5pgd00000gn/T/ipykernel_1794/1928971526.py:59: RuntimeWarning: divide by zero encountered in log10\n",
      "  neglog10p = -np.log10(p_value)\n"
     ]
    }
   ],
   "source": [
    "# both rg\n",
    "rg_df = pd.DataFrame(columns=['pheno1', 'pheno1_h2g', 'pheno1_h2g_se', 'pheno1_lambda',\n",
    "                                   'pheno2', 'pheno2_h2g', 'pheno2_h2g_se', 'pheno2_lambda',\n",
    "                                   'cor', 'se', 'z', 'p_value'])\n",
    "path = \"UKB_Imaging_Genetics/LDSC_OUTPUT/rg_hip_handedness_20230830\"\n",
    "for file in os.listdir(path):\n",
    "    if file.endswith(\".log\"):\n",
    "        # get phenotype name\n",
    "        pheno1, pheno2 = file.split('-')\n",
    "        pheno1 = pheno1.split('.')[1]\n",
    "        pheno2 = pheno2.split('.')\n",
    "        if len(pheno2) == 2:\n",
    "            pheno2 = pheno2[0]\n",
    "        else:\n",
    "            pheno2 = pheno2[1]\n",
    "\n",
    "        with open(os.path.join(path, file), \"r\") as f:\n",
    "            lines = f.readlines()cd\n",
    "\n",
    "        for i, line in enumerate(lines):\n",
    "            # get h2g\n",
    "            if line.startswith(\"Heritability of phenotype 1\") and i + 2 < len(lines):\n",
    "                h2g_1 = lines[i + 2].split(\" \")[4].strip()\n",
    "                h2g_1 = float(h2g_1)\n",
    "\n",
    "                h2g_1_se = lines[i + 2].split(\" \")[5].strip()\n",
    "                h2g_1_se = h2g_1_se[1:-1]\n",
    "                h2g_1_se = float(h2g_1_se)\n",
    "\n",
    "                pheno1_lambda = lines[i + 3].split(\" \")[2].strip()\n",
    "                pheno1_lambda = float(pheno1_lambda)\n",
    "            \n",
    "            if line.startswith(\"Heritability of phenotype 2/2\") and i + 2 < len(lines):\n",
    "                h2g_2 = lines[i + 2].split(\" \")[4].strip()\n",
    "                h2g_2 = float(h2g_2)\n",
    "                \n",
    "                h2g_2_se = lines[i + 2].split(\" \")[5].strip()\n",
    "                h2g_2_se = h2g_2_se[1:-1]\n",
    "                h2g_2_se = float(h2g_2_se)\n",
    "\n",
    "                pheno2_lambda = lines[i + 3].split(\" \")[2].strip()\n",
    "                pheno2_lambda = float(pheno1_lambda)\n",
    "\n",
    "            # get rg\n",
    "            if line.startswith(\"Genetic Correlation:\"):\n",
    "                # print(line)\n",
    "                rg = line.split(\" \")[2].strip()\n",
    "                rg = float(rg)\n",
    "                # print(rg)\n",
    "                se = line.split(\" \")[3].strip()\n",
    "                se = se[1:-1]\n",
    "                se = float(se)\n",
    "            if line.startswith(\"Z-score:\"):\n",
    "                z = line.split(\" \")[1].strip()\n",
    "                z = float(z)\n",
    "            if line.startswith(\"P:\"):\n",
    "                p_value = line.split(\" \")[1].strip()\n",
    "                p_value = float(p_value)\n",
    "                neglog10p = -np.log10(p_value)\n",
    "\n",
    "        rg_df = rg_df.append({'pheno1': pheno1, 'pheno1_h2g': h2g_1, 'pheno1_h2g_se': h2g_1_se, 'pheno1_lambda': pheno1_lambda,\n",
    "                                        'pheno2': pheno2, 'pheno2_h2g': h2g_2, 'pheno2_h2g_se': h2g_2_se, 'pheno2_lambda': pheno2_lambda,\n",
    "                                        'cor': rg, 'se': se, 'z': z, 'p_value': p_value, 'neglog10p': neglog10p}, \n",
    "                                        ignore_index=True)\n",
    "\n",
    "rg_df.sort_values(by = ['pheno1', 'pheno2'], inplace=True)\n",
    "rg_df.to_csv(\"key_results/genetic_correlation/rg_20230830/rg_hip_handedness.csv\", index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pheno1</th>\n",
       "      <th>pheno1_h2g</th>\n",
       "      <th>pheno1_h2g_se</th>\n",
       "      <th>pheno1_lambda</th>\n",
       "      <th>pheno2</th>\n",
       "      <th>pheno2_h2g</th>\n",
       "      <th>pheno2_h2g_se</th>\n",
       "      <th>pheno2_lambda</th>\n",
       "      <th>cor</th>\n",
       "      <th>se</th>\n",
       "      <th>z</th>\n",
       "      <th>p_value</th>\n",
       "      <th>neglog10p</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>acetabular_diameter_ratio</td>\n",
       "      <td>0.0121</td>\n",
       "      <td>0.0143</td>\n",
       "      <td>1.0046</td>\n",
       "      <td>acetabular_diameter_ratio</td>\n",
       "      <td>0.0121</td>\n",
       "      <td>0.0143</td>\n",
       "      <td>1.0046</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.000027</td>\n",
       "      <td>36445.8037</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>inf</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>acetabular_diameter_ratio</td>\n",
       "      <td>0.0121</td>\n",
       "      <td>0.0143</td>\n",
       "      <td>1.0046</td>\n",
       "      <td>acetabular_inclination_ratio</td>\n",
       "      <td>0.0128</td>\n",
       "      <td>0.0146</td>\n",
       "      <td>1.0046</td>\n",
       "      <td>-0.7196</td>\n",
       "      <td>1.066800</td>\n",
       "      <td>-0.6746</td>\n",
       "      <td>0.5000</td>\n",
       "      <td>0.301030</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>acetabular_diameter_ratio</td>\n",
       "      <td>0.0131</td>\n",
       "      <td>0.0149</td>\n",
       "      <td>1.0046</td>\n",
       "      <td>equally_handed_rg</td>\n",
       "      <td>0.0039</td>\n",
       "      <td>0.0013</td>\n",
       "      <td>1.0046</td>\n",
       "      <td>0.1713</td>\n",
       "      <td>0.455500</td>\n",
       "      <td>0.3761</td>\n",
       "      <td>0.7068</td>\n",
       "      <td>0.150703</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>acetabular_diameter_ratio</td>\n",
       "      <td>0.0121</td>\n",
       "      <td>0.0143</td>\n",
       "      <td>1.0046</td>\n",
       "      <td>iliac_isthmus_breadth_ratio</td>\n",
       "      <td>0.0277</td>\n",
       "      <td>0.0157</td>\n",
       "      <td>1.0046</td>\n",
       "      <td>0.2180</td>\n",
       "      <td>0.516200</td>\n",
       "      <td>0.4224</td>\n",
       "      <td>0.6727</td>\n",
       "      <td>0.172179</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>acetabular_diameter_ratio</td>\n",
       "      <td>0.0131</td>\n",
       "      <td>0.0149</td>\n",
       "      <td>1.0046</td>\n",
       "      <td>left_handed_rg</td>\n",
       "      <td>0.0104</td>\n",
       "      <td>0.0016</td>\n",
       "      <td>1.0046</td>\n",
       "      <td>-0.3954</td>\n",
       "      <td>0.345700</td>\n",
       "      <td>-1.1438</td>\n",
       "      <td>0.2527</td>\n",
       "      <td>0.597395</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>acetabular_diameter_ratio</td>\n",
       "      <td>0.0131</td>\n",
       "      <td>0.0149</td>\n",
       "      <td>1.0046</td>\n",
       "      <td>right_handed_rg</td>\n",
       "      <td>0.0096</td>\n",
       "      <td>0.0015</td>\n",
       "      <td>1.0046</td>\n",
       "      <td>0.3415</td>\n",
       "      <td>0.335400</td>\n",
       "      <td>1.0182</td>\n",
       "      <td>0.3086</td>\n",
       "      <td>0.510604</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>acetabular_inclination_ratio</td>\n",
       "      <td>0.0128</td>\n",
       "      <td>0.0146</td>\n",
       "      <td>1.0135</td>\n",
       "      <td>acetabular_diameter_ratio</td>\n",
       "      <td>0.0121</td>\n",
       "      <td>0.0143</td>\n",
       "      <td>1.0135</td>\n",
       "      <td>-0.7196</td>\n",
       "      <td>1.066800</td>\n",
       "      <td>-0.6746</td>\n",
       "      <td>0.5000</td>\n",
       "      <td>0.301030</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>acetabular_inclination_ratio</td>\n",
       "      <td>0.0128</td>\n",
       "      <td>0.0146</td>\n",
       "      <td>1.0135</td>\n",
       "      <td>acetabular_inclination_ratio</td>\n",
       "      <td>0.0128</td>\n",
       "      <td>0.0146</td>\n",
       "      <td>1.0135</td>\n",
       "      <td>1.0001</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>5373.2587</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>inf</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>acetabular_inclination_ratio</td>\n",
       "      <td>0.0155</td>\n",
       "      <td>0.0157</td>\n",
       "      <td>1.0105</td>\n",
       "      <td>equally_handed_rg</td>\n",
       "      <td>0.0039</td>\n",
       "      <td>0.0013</td>\n",
       "      <td>1.0105</td>\n",
       "      <td>0.4163</td>\n",
       "      <td>0.463600</td>\n",
       "      <td>0.8980</td>\n",
       "      <td>0.3692</td>\n",
       "      <td>0.432738</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>acetabular_inclination_ratio</td>\n",
       "      <td>0.0128</td>\n",
       "      <td>0.0146</td>\n",
       "      <td>1.0135</td>\n",
       "      <td>iliac_isthmus_breadth_ratio</td>\n",
       "      <td>0.0277</td>\n",
       "      <td>0.0157</td>\n",
       "      <td>1.0135</td>\n",
       "      <td>-0.5396</td>\n",
       "      <td>0.459700</td>\n",
       "      <td>-1.1738</td>\n",
       "      <td>0.2405</td>\n",
       "      <td>0.618885</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>acetabular_inclination_ratio</td>\n",
       "      <td>0.0155</td>\n",
       "      <td>0.0157</td>\n",
       "      <td>1.0105</td>\n",
       "      <td>left_handed_rg</td>\n",
       "      <td>0.0104</td>\n",
       "      <td>0.0016</td>\n",
       "      <td>1.0105</td>\n",
       "      <td>-0.3018</td>\n",
       "      <td>0.294800</td>\n",
       "      <td>-1.0235</td>\n",
       "      <td>0.3061</td>\n",
       "      <td>0.514137</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>acetabular_inclination_ratio</td>\n",
       "      <td>0.0155</td>\n",
       "      <td>0.0157</td>\n",
       "      <td>1.0105</td>\n",
       "      <td>right_handed_rg</td>\n",
       "      <td>0.0096</td>\n",
       "      <td>0.0015</td>\n",
       "      <td>1.0105</td>\n",
       "      <td>0.1862</td>\n",
       "      <td>0.276100</td>\n",
       "      <td>0.6744</td>\n",
       "      <td>0.5001</td>\n",
       "      <td>0.300943</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>iliac_isthmus_breadth_ratio</td>\n",
       "      <td>0.0277</td>\n",
       "      <td>0.0157</td>\n",
       "      <td>0.9986</td>\n",
       "      <td>acetabular_diameter_ratio</td>\n",
       "      <td>0.0121</td>\n",
       "      <td>0.0143</td>\n",
       "      <td>0.9986</td>\n",
       "      <td>0.2180</td>\n",
       "      <td>0.516200</td>\n",
       "      <td>0.4224</td>\n",
       "      <td>0.6727</td>\n",
       "      <td>0.172179</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>iliac_isthmus_breadth_ratio</td>\n",
       "      <td>0.0277</td>\n",
       "      <td>0.0157</td>\n",
       "      <td>0.9986</td>\n",
       "      <td>acetabular_inclination_ratio</td>\n",
       "      <td>0.0128</td>\n",
       "      <td>0.0146</td>\n",
       "      <td>0.9986</td>\n",
       "      <td>-0.5396</td>\n",
       "      <td>0.459700</td>\n",
       "      <td>-1.1738</td>\n",
       "      <td>0.2405</td>\n",
       "      <td>0.618885</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>iliac_isthmus_breadth_ratio</td>\n",
       "      <td>0.0275</td>\n",
       "      <td>0.0158</td>\n",
       "      <td>0.9986</td>\n",
       "      <td>equally_handed_rg</td>\n",
       "      <td>0.0039</td>\n",
       "      <td>0.0013</td>\n",
       "      <td>0.9986</td>\n",
       "      <td>-0.1458</td>\n",
       "      <td>0.295800</td>\n",
       "      <td>-0.4930</td>\n",
       "      <td>0.6220</td>\n",
       "      <td>0.206210</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>iliac_isthmus_breadth_ratio</td>\n",
       "      <td>0.0277</td>\n",
       "      <td>0.0157</td>\n",
       "      <td>0.9986</td>\n",
       "      <td>iliac_isthmus_breadth_ratio</td>\n",
       "      <td>0.0277</td>\n",
       "      <td>0.0157</td>\n",
       "      <td>0.9986</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.000009</td>\n",
       "      <td>117097.0159</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>inf</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>iliac_isthmus_breadth_ratio</td>\n",
       "      <td>0.0275</td>\n",
       "      <td>0.0158</td>\n",
       "      <td>0.9986</td>\n",
       "      <td>left_handed_rg</td>\n",
       "      <td>0.0104</td>\n",
       "      <td>0.0016</td>\n",
       "      <td>0.9986</td>\n",
       "      <td>0.1585</td>\n",
       "      <td>0.220100</td>\n",
       "      <td>0.7202</td>\n",
       "      <td>0.4714</td>\n",
       "      <td>0.326610</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>iliac_isthmus_breadth_ratio</td>\n",
       "      <td>0.0275</td>\n",
       "      <td>0.0158</td>\n",
       "      <td>0.9986</td>\n",
       "      <td>right_handed_rg</td>\n",
       "      <td>0.0096</td>\n",
       "      <td>0.0015</td>\n",
       "      <td>0.9986</td>\n",
       "      <td>-0.1170</td>\n",
       "      <td>0.211200</td>\n",
       "      <td>-0.5540</td>\n",
       "      <td>0.5796</td>\n",
       "      <td>0.236872</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                          pheno1  pheno1_h2g  pheno1_h2g_se  pheno1_lambda  \\\n",
       "10     acetabular_diameter_ratio      0.0121         0.0143         1.0046   \n",
       "17     acetabular_diameter_ratio      0.0121         0.0143         1.0046   \n",
       "7      acetabular_diameter_ratio      0.0131         0.0149         1.0046   \n",
       "12     acetabular_diameter_ratio      0.0121         0.0143         1.0046   \n",
       "11     acetabular_diameter_ratio      0.0131         0.0149         1.0046   \n",
       "6      acetabular_diameter_ratio      0.0131         0.0149         1.0046   \n",
       "15  acetabular_inclination_ratio      0.0128         0.0146         1.0135   \n",
       "8   acetabular_inclination_ratio      0.0128         0.0146         1.0135   \n",
       "14  acetabular_inclination_ratio      0.0155         0.0157         1.0105   \n",
       "1   acetabular_inclination_ratio      0.0128         0.0146         1.0135   \n",
       "4   acetabular_inclination_ratio      0.0155         0.0157         1.0105   \n",
       "3   acetabular_inclination_ratio      0.0155         0.0157         1.0105   \n",
       "9    iliac_isthmus_breadth_ratio      0.0277         0.0157         0.9986   \n",
       "16   iliac_isthmus_breadth_ratio      0.0277         0.0157         0.9986   \n",
       "0    iliac_isthmus_breadth_ratio      0.0275         0.0158         0.9986   \n",
       "2    iliac_isthmus_breadth_ratio      0.0277         0.0157         0.9986   \n",
       "13   iliac_isthmus_breadth_ratio      0.0275         0.0158         0.9986   \n",
       "5    iliac_isthmus_breadth_ratio      0.0275         0.0158         0.9986   \n",
       "\n",
       "                          pheno2  pheno2_h2g  pheno2_h2g_se  pheno2_lambda  \\\n",
       "10     acetabular_diameter_ratio      0.0121         0.0143         1.0046   \n",
       "17  acetabular_inclination_ratio      0.0128         0.0146         1.0046   \n",
       "7              equally_handed_rg      0.0039         0.0013         1.0046   \n",
       "12   iliac_isthmus_breadth_ratio      0.0277         0.0157         1.0046   \n",
       "11                left_handed_rg      0.0104         0.0016         1.0046   \n",
       "6                right_handed_rg      0.0096         0.0015         1.0046   \n",
       "15     acetabular_diameter_ratio      0.0121         0.0143         1.0135   \n",
       "8   acetabular_inclination_ratio      0.0128         0.0146         1.0135   \n",
       "14             equally_handed_rg      0.0039         0.0013         1.0105   \n",
       "1    iliac_isthmus_breadth_ratio      0.0277         0.0157         1.0135   \n",
       "4                 left_handed_rg      0.0104         0.0016         1.0105   \n",
       "3                right_handed_rg      0.0096         0.0015         1.0105   \n",
       "9      acetabular_diameter_ratio      0.0121         0.0143         0.9986   \n",
       "16  acetabular_inclination_ratio      0.0128         0.0146         0.9986   \n",
       "0              equally_handed_rg      0.0039         0.0013         0.9986   \n",
       "2    iliac_isthmus_breadth_ratio      0.0277         0.0157         0.9986   \n",
       "13                left_handed_rg      0.0104         0.0016         0.9986   \n",
       "5                right_handed_rg      0.0096         0.0015         0.9986   \n",
       "\n",
       "       cor        se            z  p_value  neglog10p  \n",
       "10  1.0000  0.000027   36445.8037   0.0000        inf  \n",
       "17 -0.7196  1.066800      -0.6746   0.5000   0.301030  \n",
       "7   0.1713  0.455500       0.3761   0.7068   0.150703  \n",
       "12  0.2180  0.516200       0.4224   0.6727   0.172179  \n",
       "11 -0.3954  0.345700      -1.1438   0.2527   0.597395  \n",
       "6   0.3415  0.335400       1.0182   0.3086   0.510604  \n",
       "15 -0.7196  1.066800      -0.6746   0.5000   0.301030  \n",
       "8   1.0001  0.000200    5373.2587   0.0000        inf  \n",
       "14  0.4163  0.463600       0.8980   0.3692   0.432738  \n",
       "1  -0.5396  0.459700      -1.1738   0.2405   0.618885  \n",
       "4  -0.3018  0.294800      -1.0235   0.3061   0.514137  \n",
       "3   0.1862  0.276100       0.6744   0.5001   0.300943  \n",
       "9   0.2180  0.516200       0.4224   0.6727   0.172179  \n",
       "16 -0.5396  0.459700      -1.1738   0.2405   0.618885  \n",
       "0  -0.1458  0.295800      -0.4930   0.6220   0.206210  \n",
       "2   1.0000  0.000009  117097.0159   0.0000        inf  \n",
       "13  0.1585  0.220100       0.7202   0.4714   0.326610  \n",
       "5  -0.1170  0.211200      -0.5540   0.5796   0.236872  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rg_df"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# SLDSC"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## for ATAC-seq from science advance paper"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "path = \"UKB_Imaging_Genetics/LDSC_OUTPUT/SLDSC_OUTPUT/part_h2g_20230920_sa_atacseq\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Both:  504\n",
      "Female:  504\n",
      "Male:  504\n"
     ]
    }
   ],
   "source": [
    "both_res_files = []\n",
    "male_res_files = [] \n",
    "female_res_files = []\n",
    "\n",
    "for f in os.listdir(path):\n",
    "    if f.endswith('.results'):\n",
    "        if \"both\" in f:\n",
    "            both_res_files.append(f)\n",
    "        elif \"female\" in f:\n",
    "            female_res_files.append(f)\n",
    "        elif \"male\" in f:\n",
    "            male_res_files.append(f)\n",
    "\n",
    "print(\"Both: \", len(both_res_files))\n",
    "print(\"Female: \", len(female_res_files))\n",
    "print(\"Male: \", len(male_res_files))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "def assign_group(i):\n",
    "    if \"Human\" in i and \"Brain-Filtered\" in i:\n",
    "        group = \"Human_Brain_Filtered\"\n",
    "    elif \"Human\" in i and \"Gain\" in i:\n",
    "        group = \"Human_Gain\"\n",
    "    elif \"Human\" in i and \"Mouse\" in i and \"Shared\" in i:\n",
    "        group = \"Human_Mouse_Shared\"\n",
    "    elif \"Human\" in i and \"Mouse\" in i and \"Intersect\" in i:\n",
    "        group = \"Human_Mouse_Intersect\"\n",
    "    elif \"Mouse\" in i and \"Brain-Filter\" in i:\n",
    "        group = \"Mouse_Brain_Filtered\"\n",
    "    elif \"Mouse\" in i and \"Gain\" in i:\n",
    "        group = \"Mouse_Gain\"\n",
    "    elif \"Mouse\" in i:\n",
    "        group = \"Mouse\"\n",
    "    elif \"Human\" in i:\n",
    "        group = \"Human\"\n",
    "    return group"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### For both male and female"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_res_df = None\n",
    "\n",
    "for f in both_res_files:\n",
    "    anno, pheno = f.split('iPSYCH-PGC_ASD_baseline_')[1].split('_both.results')[0].split('_sorted_control_')\n",
    "    df = pd.read_csv(os.path.join(path, f), sep='\\t')\n",
    "    df = df[df['Category'] == 'L2_1']\n",
    "    df['Annotation'] = anno\n",
    "    df['Phenotype'] = pheno\n",
    "\n",
    "    if all_res_df is None:\n",
    "        all_res_df = df\n",
    "    else:\n",
    "        all_res_df = pd.concat([all_res_df, df], axis=0)\n",
    "\n",
    "all_res_df['Group'] = all_res_df['Annotation'].apply(lambda x: assign_group(x))\n",
    "all_res_df['neglog10p'] = -np.log10(all_res_df['Enrichment_p'])\n",
    "all_res_df.drop(columns=['Category'], inplace=True)\n",
    "\n",
    "all_res_df.to_csv('key_results/SLDSC_results_from_sa_paper_both.csv', index=False)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### For female"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_res_df = None\n",
    "\n",
    "for f in female_res_files:\n",
    "    anno, pheno = f.split('iPSYCH-PGC_ASD_baseline_')[1].split('_female.results')[0].split('_sorted_control_')\n",
    "    df = pd.read_csv(os.path.join(path, f), sep='\\t')\n",
    "    df = df[df['Category'] == 'L2_1']\n",
    "    df['Annotation'] = anno\n",
    "    df['Phenotype'] = pheno\n",
    "\n",
    "    if all_res_df is None:\n",
    "        all_res_df = df\n",
    "    else:\n",
    "        all_res_df = pd.concat([all_res_df, df], axis=0)\n",
    "\n",
    "all_res_df['Group'] = all_res_df['Annotation'].apply(lambda x: assign_group(x))\n",
    "all_res_df['neglog10p'] = -np.log10(all_res_df['Enrichment_p'])\n",
    "all_res_df.drop(columns=['Category'], inplace=True)\n",
    "\n",
    "all_res_df.to_csv('key_results/SLDSC_results_from_sa_paper_female.csv', index=False)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### For male"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_res_df = None\n",
    "\n",
    "for f in male_res_files:\n",
    "    anno, pheno = f.split('iPSYCH-PGC_ASD_baseline_')[1].split('_male.results')[0].split('_sorted_control_')\n",
    "    df = pd.read_csv(os.path.join(path, f), sep='\\t')\n",
    "    df = df[df['Category'] == 'L2_1']\n",
    "    df['Annotation'] = anno\n",
    "    df['Phenotype'] = pheno\n",
    "\n",
    "    if all_res_df is None:\n",
    "        all_res_df = df\n",
    "    else:\n",
    "        all_res_df = pd.concat([all_res_df, df], axis=0)\n",
    "\n",
    "all_res_df['Group'] = all_res_df['Annotation'].apply(lambda x: assign_group(x))\n",
    "all_res_df['neglog10p'] = -np.log10(all_res_df['Enrichment_p'])\n",
    "all_res_df.drop(columns=['Category'], inplace=True)\n",
    "\n",
    "all_res_df.to_csv('key_results/SLDSC_results_from_sa_paper_male.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Prop._SNPs</th>\n",
       "      <th>Prop._h2</th>\n",
       "      <th>Prop._h2_std_error</th>\n",
       "      <th>Enrichment</th>\n",
       "      <th>Enrichment_std_error</th>\n",
       "      <th>Enrichment_p</th>\n",
       "      <th>Coefficient</th>\n",
       "      <th>Coefficient_std_error</th>\n",
       "      <th>Coefficient_z-score</th>\n",
       "      <th>Annotation</th>\n",
       "      <th>Phenotype</th>\n",
       "      <th>Group</th>\n",
       "      <th>neglog10p</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>0.010033</td>\n",
       "      <td>0.198410</td>\n",
       "      <td>0.050714</td>\n",
       "      <td>19.775556</td>\n",
       "      <td>5.054636</td>\n",
       "      <td>0.000040</td>\n",
       "      <td>1.087595e-06</td>\n",
       "      <td>2.862098e-07</td>\n",
       "      <td>3.799992</td>\n",
       "      <td>HumanIlium_Brain-Filtered_Peaks</td>\n",
       "      <td>subpubic_angle</td>\n",
       "      <td>Human_Brain_Filtered</td>\n",
       "      <td>4.398777</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>0.007442</td>\n",
       "      <td>0.080376</td>\n",
       "      <td>0.031066</td>\n",
       "      <td>10.800572</td>\n",
       "      <td>4.174452</td>\n",
       "      <td>0.016550</td>\n",
       "      <td>6.848862e-07</td>\n",
       "      <td>3.267313e-07</td>\n",
       "      <td>2.096175</td>\n",
       "      <td>Human_Gain_Acet_Peaks</td>\n",
       "      <td>shoulder_divide_inlet_width</td>\n",
       "      <td>Human_Gain</td>\n",
       "      <td>1.781209</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>0.002668</td>\n",
       "      <td>0.020758</td>\n",
       "      <td>0.025913</td>\n",
       "      <td>7.779889</td>\n",
       "      <td>9.712026</td>\n",
       "      <td>0.485627</td>\n",
       "      <td>2.134762e-07</td>\n",
       "      <td>6.790052e-07</td>\n",
       "      <td>0.314396</td>\n",
       "      <td>Mouse_Gain_Acet_Peaks</td>\n",
       "      <td>iliac_flare_angle</td>\n",
       "      <td>Mouse_Gain</td>\n",
       "      <td>0.313697</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>0.010033</td>\n",
       "      <td>0.240290</td>\n",
       "      <td>0.080939</td>\n",
       "      <td>23.949710</td>\n",
       "      <td>8.067230</td>\n",
       "      <td>0.001199</td>\n",
       "      <td>9.515813e-07</td>\n",
       "      <td>3.290763e-07</td>\n",
       "      <td>2.891673</td>\n",
       "      <td>HumanIlium_Brain-Filtered_Peaks</td>\n",
       "      <td>arm_devide_torso</td>\n",
       "      <td>Human_Brain_Filtered</td>\n",
       "      <td>2.921199</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>0.011133</td>\n",
       "      <td>0.189178</td>\n",
       "      <td>0.057512</td>\n",
       "      <td>16.991920</td>\n",
       "      <td>5.165734</td>\n",
       "      <td>0.000410</td>\n",
       "      <td>7.051124e-07</td>\n",
       "      <td>2.451274e-07</td>\n",
       "      <td>2.876514</td>\n",
       "      <td>HumanPubis_Brain-Filtered_Peaks</td>\n",
       "      <td>oblique_pelvic_inlet_length</td>\n",
       "      <td>Human_Brain_Filtered</td>\n",
       "      <td>3.386805</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>0.007123</td>\n",
       "      <td>-0.020490</td>\n",
       "      <td>0.037547</td>\n",
       "      <td>-2.876489</td>\n",
       "      <td>5.271019</td>\n",
       "      <td>0.460099</td>\n",
       "      <td>-2.726903e-07</td>\n",
       "      <td>3.673025e-07</td>\n",
       "      <td>-0.742413</td>\n",
       "      <td>Mouse_Acetabulum_IDR_Peaks</td>\n",
       "      <td>iliac_flare_ratio</td>\n",
       "      <td>Mouse</td>\n",
       "      <td>0.337149</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>0.000952</td>\n",
       "      <td>0.071641</td>\n",
       "      <td>0.022684</td>\n",
       "      <td>75.253736</td>\n",
       "      <td>23.827674</td>\n",
       "      <td>0.000413</td>\n",
       "      <td>4.421636e-06</td>\n",
       "      <td>1.317957e-06</td>\n",
       "      <td>3.354917</td>\n",
       "      <td>Human_Mouse_ATAC-seq_Intersect</td>\n",
       "      <td>subpubic_angle</td>\n",
       "      <td>Human_Mouse_Intersect</td>\n",
       "      <td>3.384051</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>0.017054</td>\n",
       "      <td>0.132455</td>\n",
       "      <td>0.077305</td>\n",
       "      <td>7.766840</td>\n",
       "      <td>4.532978</td>\n",
       "      <td>0.108582</td>\n",
       "      <td>3.334493e-07</td>\n",
       "      <td>3.167105e-07</td>\n",
       "      <td>1.052852</td>\n",
       "      <td>Human_Acetabulum_IDR_Peaks</td>\n",
       "      <td>pelvic_inlet_area</td>\n",
       "      <td>Human</td>\n",
       "      <td>0.964242</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>0.007123</td>\n",
       "      <td>0.015331</td>\n",
       "      <td>0.040320</td>\n",
       "      <td>2.152174</td>\n",
       "      <td>5.660377</td>\n",
       "      <td>0.838820</td>\n",
       "      <td>7.316187e-08</td>\n",
       "      <td>3.542477e-07</td>\n",
       "      <td>0.206527</td>\n",
       "      <td>Mouse_Acetabulum_IDR_Peaks</td>\n",
       "      <td>head_divide_inlet_width</td>\n",
       "      <td>Mouse</td>\n",
       "      <td>0.076331</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>0.002196</td>\n",
       "      <td>0.065317</td>\n",
       "      <td>0.027243</td>\n",
       "      <td>29.740568</td>\n",
       "      <td>12.404475</td>\n",
       "      <td>0.015992</td>\n",
       "      <td>1.616615e-06</td>\n",
       "      <td>7.369936e-07</td>\n",
       "      <td>2.193527</td>\n",
       "      <td>Mouse_Gain_Ilium_Peaks</td>\n",
       "      <td>trochanter_left2trochanter_right</td>\n",
       "      <td>Mouse_Gain</td>\n",
       "      <td>1.796110</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>504 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Prop._SNPs  Prop._h2  Prop._h2_std_error  Enrichment  \\\n",
       "97    0.010033  0.198410            0.050714   19.775556   \n",
       "97    0.007442  0.080376            0.031066   10.800572   \n",
       "97    0.002668  0.020758            0.025913    7.779889   \n",
       "97    0.010033  0.240290            0.080939   23.949710   \n",
       "97    0.011133  0.189178            0.057512   16.991920   \n",
       "..         ...       ...                 ...         ...   \n",
       "97    0.007123 -0.020490            0.037547   -2.876489   \n",
       "97    0.000952  0.071641            0.022684   75.253736   \n",
       "97    0.017054  0.132455            0.077305    7.766840   \n",
       "97    0.007123  0.015331            0.040320    2.152174   \n",
       "97    0.002196  0.065317            0.027243   29.740568   \n",
       "\n",
       "    Enrichment_std_error  Enrichment_p   Coefficient  Coefficient_std_error  \\\n",
       "97              5.054636      0.000040  1.087595e-06           2.862098e-07   \n",
       "97              4.174452      0.016550  6.848862e-07           3.267313e-07   \n",
       "97              9.712026      0.485627  2.134762e-07           6.790052e-07   \n",
       "97              8.067230      0.001199  9.515813e-07           3.290763e-07   \n",
       "97              5.165734      0.000410  7.051124e-07           2.451274e-07   \n",
       "..                   ...           ...           ...                    ...   \n",
       "97              5.271019      0.460099 -2.726903e-07           3.673025e-07   \n",
       "97             23.827674      0.000413  4.421636e-06           1.317957e-06   \n",
       "97              4.532978      0.108582  3.334493e-07           3.167105e-07   \n",
       "97              5.660377      0.838820  7.316187e-08           3.542477e-07   \n",
       "97             12.404475      0.015992  1.616615e-06           7.369936e-07   \n",
       "\n",
       "    Coefficient_z-score                       Annotation  \\\n",
       "97             3.799992  HumanIlium_Brain-Filtered_Peaks   \n",
       "97             2.096175            Human_Gain_Acet_Peaks   \n",
       "97             0.314396            Mouse_Gain_Acet_Peaks   \n",
       "97             2.891673  HumanIlium_Brain-Filtered_Peaks   \n",
       "97             2.876514  HumanPubis_Brain-Filtered_Peaks   \n",
       "..                  ...                              ...   \n",
       "97            -0.742413       Mouse_Acetabulum_IDR_Peaks   \n",
       "97             3.354917   Human_Mouse_ATAC-seq_Intersect   \n",
       "97             1.052852       Human_Acetabulum_IDR_Peaks   \n",
       "97             0.206527       Mouse_Acetabulum_IDR_Peaks   \n",
       "97             2.193527           Mouse_Gain_Ilium_Peaks   \n",
       "\n",
       "                           Phenotype                  Group  neglog10p  \n",
       "97                    subpubic_angle   Human_Brain_Filtered   4.398777  \n",
       "97       shoulder_divide_inlet_width             Human_Gain   1.781209  \n",
       "97                 iliac_flare_angle             Mouse_Gain   0.313697  \n",
       "97                  arm_devide_torso   Human_Brain_Filtered   2.921199  \n",
       "97       oblique_pelvic_inlet_length   Human_Brain_Filtered   3.386805  \n",
       "..                               ...                    ...        ...  \n",
       "97                 iliac_flare_ratio                  Mouse   0.337149  \n",
       "97                    subpubic_angle  Human_Mouse_Intersect   3.384051  \n",
       "97                 pelvic_inlet_area                  Human   0.964242  \n",
       "97           head_divide_inlet_width                  Mouse   0.076331  \n",
       "97  trochanter_left2trochanter_right             Mouse_Gain   1.796110  \n",
       "\n",
       "[504 rows x 13 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_res_df"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## for evolution annotations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [],
   "source": [
    "path = \"UKB_Imaging_Genetics/LDSC_OUTPUT/SLDSC_OUTPUT/part_h2g_20230920_evolution\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Both:  210\n",
      "Female:  210\n",
      "Male:  210\n"
     ]
    }
   ],
   "source": [
    "both_res_files = []\n",
    "male_res_files = [] \n",
    "female_res_files = []\n",
    "\n",
    "for f in os.listdir(path):\n",
    "    if f.endswith('.results'):\n",
    "        if \"both\" in f:\n",
    "            both_res_files.append(f)\n",
    "        elif \"female\" in f:\n",
    "            female_res_files.append(f)\n",
    "        elif \"male\" in f:\n",
    "            male_res_files.append(f)\n",
    "\n",
    "print(\"Both: \", len(both_res_files))\n",
    "print(\"Female: \", len(female_res_files))\n",
    "print(\"Male: \", len(male_res_files))"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### both"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_res_df = None\n",
    "\n",
    "for f in both_res_files:\n",
    "    anno, pheno = f.split('iPSYCH-PGC_ASD_baseline_')[1].split('_both.results')[0].split('_sorted_control_')\n",
    "    df = pd.read_csv(os.path.join(path, f), sep='\\t')\n",
    "    df = df[df['Category'] == 'L2_1']\n",
    "    df['Annotation'] = anno\n",
    "    df['Phenotype'] = pheno\n",
    "\n",
    "    if all_res_df is None:\n",
    "        all_res_df = df\n",
    "    else:\n",
    "        all_res_df = pd.concat([all_res_df, df], axis=0)\n",
    "\n",
    "all_res_df['neglog10p'] = -np.log10(all_res_df['Enrichment_p'])\n",
    "all_res_df.drop(columns=['Category'], inplace=True)\n",
    "\n",
    "all_res_df.to_csv('key_results/SLDSC_results_evo_anno_both.csv', index=False)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### female"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_res_df = None\n",
    "\n",
    "for f in female_res_files:\n",
    "    anno, pheno = f.split('iPSYCH-PGC_ASD_baseline_')[1].split('_female.results')[0].split('_sorted_control_')\n",
    "    df = pd.read_csv(os.path.join(path, f), sep='\\t')\n",
    "    df = df[df['Category'] == 'L2_1']\n",
    "    df['Annotation'] = anno\n",
    "    df['Phenotype'] = pheno\n",
    "\n",
    "    if all_res_df is None:\n",
    "        all_res_df = df\n",
    "    else:\n",
    "        all_res_df = pd.concat([all_res_df, df], axis=0)\n",
    "\n",
    "all_res_df['neglog10p'] = -np.log10(all_res_df['Enrichment_p'])\n",
    "all_res_df.drop(columns=['Category'], inplace=True)\n",
    "\n",
    "all_res_df.to_csv('key_results/SLDSC_results_evo_anno_female.csv', index=False)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### male"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_res_df = None\n",
    "\n",
    "for f in male_res_files:\n",
    "    anno, pheno = f.split('iPSYCH-PGC_ASD_baseline_')[1].split('_male.results')[0].split('_sorted_control_')\n",
    "    df = pd.read_csv(os.path.join(path, f), sep='\\t')\n",
    "    df = df[df['Category'] == 'L2_1']\n",
    "    df['Annotation'] = anno\n",
    "    df['Phenotype'] = pheno\n",
    "\n",
    "    if all_res_df is None:\n",
    "        all_res_df = df\n",
    "    else:\n",
    "        all_res_df = pd.concat([all_res_df, df], axis=0)\n",
    "\n",
    "all_res_df['neglog10p'] = -np.log10(all_res_df['Enrichment_p'])\n",
    "all_res_df.drop(columns=['Category'], inplace=True)\n",
    "\n",
    "all_res_df.to_csv('key_results/SLDSC_results_evo_anno_male.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "GSE63648_7pcw_ac_me2_Hu_gain                     21\n",
       "promoters_human_macaque_prefontal_cortex_hg19    21\n",
       "depleted_neanderthal_denisovan_vernot            21\n",
       "GSE63648_12Opcw_ac_me2_Hu_gain                   21\n",
       "nchaes_merged_hg19                               21\n",
       "GSE63648_8_5pcw_ac_me2_Hu_gain                   21\n",
       "selsweeps_233putativeregions_peyregne            21\n",
       "GSE63648_12Fpcw_ac_me2_Hu_gain                   21\n",
       "enhancers_human_macaque_prefontal_cortex_hg19    21\n",
       "depleted_neanderthal_vernot                      21\n",
       "Name: Annotation, dtype: int64"
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_res_df['Annotation'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Prop._SNPs</th>\n",
       "      <th>Prop._h2</th>\n",
       "      <th>Prop._h2_std_error</th>\n",
       "      <th>Enrichment</th>\n",
       "      <th>Enrichment_std_error</th>\n",
       "      <th>Enrichment_p</th>\n",
       "      <th>Coefficient</th>\n",
       "      <th>Coefficient_std_error</th>\n",
       "      <th>Coefficient_z-score</th>\n",
       "      <th>Annotation</th>\n",
       "      <th>Phenotype</th>\n",
       "      <th>neglog10p</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>0.008458</td>\n",
       "      <td>0.109611</td>\n",
       "      <td>0.031814</td>\n",
       "      <td>12.959795</td>\n",
       "      <td>3.761476</td>\n",
       "      <td>0.000358</td>\n",
       "      <td>6.722678e-07</td>\n",
       "      <td>2.152973e-07</td>\n",
       "      <td>3.122509</td>\n",
       "      <td>GSE63648_7pcw_ac_me2_Hu_gain</td>\n",
       "      <td>shoulder_width</td>\n",
       "      <td>3.446298</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>0.000780</td>\n",
       "      <td>-0.013063</td>\n",
       "      <td>0.004136</td>\n",
       "      <td>-16.753403</td>\n",
       "      <td>5.304319</td>\n",
       "      <td>0.000912</td>\n",
       "      <td>-1.497267e-06</td>\n",
       "      <td>4.409633e-07</td>\n",
       "      <td>-3.395446</td>\n",
       "      <td>promoters_human_macaque_prefontal_cortex_hg19</td>\n",
       "      <td>pelvic_inlet_width</td>\n",
       "      <td>3.040035</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>0.015396</td>\n",
       "      <td>0.006282</td>\n",
       "      <td>0.002497</td>\n",
       "      <td>0.407994</td>\n",
       "      <td>0.162164</td>\n",
       "      <td>0.000942</td>\n",
       "      <td>-4.638725e-08</td>\n",
       "      <td>1.906406e-08</td>\n",
       "      <td>-2.433231</td>\n",
       "      <td>depleted_neanderthal_denisovan_vernot</td>\n",
       "      <td>pelvic_height</td>\n",
       "      <td>3.025892</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>0.008458</td>\n",
       "      <td>0.068061</td>\n",
       "      <td>0.019651</td>\n",
       "      <td>8.047131</td>\n",
       "      <td>2.323383</td>\n",
       "      <td>0.001408</td>\n",
       "      <td>5.469099e-07</td>\n",
       "      <td>2.492603e-07</td>\n",
       "      <td>2.194132</td>\n",
       "      <td>GSE63648_7pcw_ac_me2_Hu_gain</td>\n",
       "      <td>pelvic_height</td>\n",
       "      <td>2.851522</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>0.008458</td>\n",
       "      <td>0.098975</td>\n",
       "      <td>0.031577</td>\n",
       "      <td>11.702258</td>\n",
       "      <td>3.733490</td>\n",
       "      <td>0.001501</td>\n",
       "      <td>5.600452e-07</td>\n",
       "      <td>2.070194e-07</td>\n",
       "      <td>2.705279</td>\n",
       "      <td>GSE63648_7pcw_ac_me2_Hu_gain</td>\n",
       "      <td>subpubic_angle</td>\n",
       "      <td>2.823637</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>0.004394</td>\n",
       "      <td>0.049179</td>\n",
       "      <td>0.016593</td>\n",
       "      <td>11.192451</td>\n",
       "      <td>3.776333</td>\n",
       "      <td>0.007490</td>\n",
       "      <td>8.152548e-07</td>\n",
       "      <td>4.113818e-07</td>\n",
       "      <td>1.981747</td>\n",
       "      <td>GSE63648_12Opcw_ac_me2_Hu_gain</td>\n",
       "      <td>pelvic_height</td>\n",
       "      <td>2.125509</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>0.015396</td>\n",
       "      <td>0.028858</td>\n",
       "      <td>0.005411</td>\n",
       "      <td>1.874369</td>\n",
       "      <td>0.351464</td>\n",
       "      <td>0.007921</td>\n",
       "      <td>6.717541e-08</td>\n",
       "      <td>2.269610e-08</td>\n",
       "      <td>2.959777</td>\n",
       "      <td>depleted_neanderthal_denisovan_vernot</td>\n",
       "      <td>iliac_flare_angle</td>\n",
       "      <td>2.101231</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>0.008458</td>\n",
       "      <td>0.071640</td>\n",
       "      <td>0.025119</td>\n",
       "      <td>8.470336</td>\n",
       "      <td>2.969943</td>\n",
       "      <td>0.007970</td>\n",
       "      <td>3.885626e-07</td>\n",
       "      <td>2.102545e-07</td>\n",
       "      <td>1.848058</td>\n",
       "      <td>GSE63648_7pcw_ac_me2_Hu_gain</td>\n",
       "      <td>iliac_flare_angle</td>\n",
       "      <td>2.098520</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>0.008458</td>\n",
       "      <td>0.070785</td>\n",
       "      <td>0.024150</td>\n",
       "      <td>8.369290</td>\n",
       "      <td>2.855395</td>\n",
       "      <td>0.009083</td>\n",
       "      <td>4.705241e-07</td>\n",
       "      <td>2.405948e-07</td>\n",
       "      <td>1.955670</td>\n",
       "      <td>GSE63648_7pcw_ac_me2_Hu_gain</td>\n",
       "      <td>shoulder_divide_inlet_width</td>\n",
       "      <td>2.041788</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>0.008458</td>\n",
       "      <td>0.096298</td>\n",
       "      <td>0.036353</td>\n",
       "      <td>11.385788</td>\n",
       "      <td>4.298129</td>\n",
       "      <td>0.010322</td>\n",
       "      <td>4.497949e-07</td>\n",
       "      <td>2.188284e-07</td>\n",
       "      <td>2.055468</td>\n",
       "      <td>GSE63648_7pcw_ac_me2_Hu_gain</td>\n",
       "      <td>shoulder_area_divide_pelvic_inlet_area</td>\n",
       "      <td>1.986225</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>0.000277</td>\n",
       "      <td>-0.023030</td>\n",
       "      <td>0.009279</td>\n",
       "      <td>-83.052242</td>\n",
       "      <td>33.461992</td>\n",
       "      <td>0.011405</td>\n",
       "      <td>-6.133977e-06</td>\n",
       "      <td>2.112928e-06</td>\n",
       "      <td>-2.903070</td>\n",
       "      <td>nchaes_merged_hg19</td>\n",
       "      <td>head_divide_inlet_width</td>\n",
       "      <td>1.942891</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>0.004431</td>\n",
       "      <td>0.044861</td>\n",
       "      <td>0.016880</td>\n",
       "      <td>10.123515</td>\n",
       "      <td>3.809163</td>\n",
       "      <td>0.013459</td>\n",
       "      <td>7.038741e-07</td>\n",
       "      <td>4.025970e-07</td>\n",
       "      <td>1.748334</td>\n",
       "      <td>GSE63648_8_5pcw_ac_me2_Hu_gain</td>\n",
       "      <td>pelvic_height</td>\n",
       "      <td>1.870987</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>0.004431</td>\n",
       "      <td>0.087032</td>\n",
       "      <td>0.038180</td>\n",
       "      <td>19.640099</td>\n",
       "      <td>8.615840</td>\n",
       "      <td>0.017275</td>\n",
       "      <td>7.236099e-07</td>\n",
       "      <td>3.639990e-07</td>\n",
       "      <td>1.987945</td>\n",
       "      <td>GSE63648_8_5pcw_ac_me2_Hu_gain</td>\n",
       "      <td>arm_devide_torso</td>\n",
       "      <td>1.762577</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>0.000277</td>\n",
       "      <td>-0.028462</td>\n",
       "      <td>0.014006</td>\n",
       "      <td>-102.641040</td>\n",
       "      <td>50.510549</td>\n",
       "      <td>0.021486</td>\n",
       "      <td>-5.554476e-06</td>\n",
       "      <td>2.306093e-06</td>\n",
       "      <td>-2.408609</td>\n",
       "      <td>nchaes_merged_hg19</td>\n",
       "      <td>ear_left2ear_right</td>\n",
       "      <td>1.667853</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>0.004017</td>\n",
       "      <td>-0.008726</td>\n",
       "      <td>0.006283</td>\n",
       "      <td>-2.172384</td>\n",
       "      <td>1.564323</td>\n",
       "      <td>0.022905</td>\n",
       "      <td>-1.488373e-07</td>\n",
       "      <td>6.202373e-08</td>\n",
       "      <td>-2.399683</td>\n",
       "      <td>selsweeps_233putativeregions_peyregne</td>\n",
       "      <td>head_divide_oblique_inlet_length</td>\n",
       "      <td>1.640062</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>0.004017</td>\n",
       "      <td>-0.010588</td>\n",
       "      <td>0.007252</td>\n",
       "      <td>-2.636133</td>\n",
       "      <td>1.805529</td>\n",
       "      <td>0.023518</td>\n",
       "      <td>-1.577137e-07</td>\n",
       "      <td>6.662980e-08</td>\n",
       "      <td>-2.367014</td>\n",
       "      <td>selsweeps_233putativeregions_peyregne</td>\n",
       "      <td>shoulder_divide_oblique_inlet_length</td>\n",
       "      <td>1.628595</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>0.003083</td>\n",
       "      <td>0.066958</td>\n",
       "      <td>0.032586</td>\n",
       "      <td>21.721111</td>\n",
       "      <td>10.570773</td>\n",
       "      <td>0.032542</td>\n",
       "      <td>8.350697e-07</td>\n",
       "      <td>4.554567e-07</td>\n",
       "      <td>1.833478</td>\n",
       "      <td>GSE63648_12Fpcw_ac_me2_Hu_gain</td>\n",
       "      <td>arm_devide_torso</td>\n",
       "      <td>1.487560</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>0.008458</td>\n",
       "      <td>0.089234</td>\n",
       "      <td>0.041593</td>\n",
       "      <td>10.550543</td>\n",
       "      <td>4.917787</td>\n",
       "      <td>0.040606</td>\n",
       "      <td>4.026315e-07</td>\n",
       "      <td>2.426256e-07</td>\n",
       "      <td>1.659477</td>\n",
       "      <td>GSE63648_7pcw_ac_me2_Hu_gain</td>\n",
       "      <td>acetabular_diameter</td>\n",
       "      <td>1.391407</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>0.000277</td>\n",
       "      <td>-0.015426</td>\n",
       "      <td>0.007933</td>\n",
       "      <td>-55.629368</td>\n",
       "      <td>28.609709</td>\n",
       "      <td>0.043614</td>\n",
       "      <td>-5.943663e-06</td>\n",
       "      <td>2.339894e-06</td>\n",
       "      <td>-2.540142</td>\n",
       "      <td>nchaes_merged_hg19</td>\n",
       "      <td>pelvic_inlet_width</td>\n",
       "      <td>1.360376</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>0.008458</td>\n",
       "      <td>0.077207</td>\n",
       "      <td>0.033372</td>\n",
       "      <td>9.128594</td>\n",
       "      <td>3.945671</td>\n",
       "      <td>0.044446</td>\n",
       "      <td>3.611513e-07</td>\n",
       "      <td>2.649829e-07</td>\n",
       "      <td>1.362923</td>\n",
       "      <td>GSE63648_7pcw_ac_me2_Hu_gain</td>\n",
       "      <td>trochanter_left2trochanter_right</td>\n",
       "      <td>1.352163</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Prop._SNPs  Prop._h2  Prop._h2_std_error  Enrichment  \\\n",
       "97    0.008458  0.109611            0.031814   12.959795   \n",
       "97    0.000780 -0.013063            0.004136  -16.753403   \n",
       "97    0.015396  0.006282            0.002497    0.407994   \n",
       "97    0.008458  0.068061            0.019651    8.047131   \n",
       "97    0.008458  0.098975            0.031577   11.702258   \n",
       "97    0.004394  0.049179            0.016593   11.192451   \n",
       "97    0.015396  0.028858            0.005411    1.874369   \n",
       "97    0.008458  0.071640            0.025119    8.470336   \n",
       "97    0.008458  0.070785            0.024150    8.369290   \n",
       "97    0.008458  0.096298            0.036353   11.385788   \n",
       "97    0.000277 -0.023030            0.009279  -83.052242   \n",
       "97    0.004431  0.044861            0.016880   10.123515   \n",
       "97    0.004431  0.087032            0.038180   19.640099   \n",
       "97    0.000277 -0.028462            0.014006 -102.641040   \n",
       "97    0.004017 -0.008726            0.006283   -2.172384   \n",
       "97    0.004017 -0.010588            0.007252   -2.636133   \n",
       "97    0.003083  0.066958            0.032586   21.721111   \n",
       "97    0.008458  0.089234            0.041593   10.550543   \n",
       "97    0.000277 -0.015426            0.007933  -55.629368   \n",
       "97    0.008458  0.077207            0.033372    9.128594   \n",
       "\n",
       "    Enrichment_std_error  Enrichment_p   Coefficient  Coefficient_std_error  \\\n",
       "97              3.761476      0.000358  6.722678e-07           2.152973e-07   \n",
       "97              5.304319      0.000912 -1.497267e-06           4.409633e-07   \n",
       "97              0.162164      0.000942 -4.638725e-08           1.906406e-08   \n",
       "97              2.323383      0.001408  5.469099e-07           2.492603e-07   \n",
       "97              3.733490      0.001501  5.600452e-07           2.070194e-07   \n",
       "97              3.776333      0.007490  8.152548e-07           4.113818e-07   \n",
       "97              0.351464      0.007921  6.717541e-08           2.269610e-08   \n",
       "97              2.969943      0.007970  3.885626e-07           2.102545e-07   \n",
       "97              2.855395      0.009083  4.705241e-07           2.405948e-07   \n",
       "97              4.298129      0.010322  4.497949e-07           2.188284e-07   \n",
       "97             33.461992      0.011405 -6.133977e-06           2.112928e-06   \n",
       "97              3.809163      0.013459  7.038741e-07           4.025970e-07   \n",
       "97              8.615840      0.017275  7.236099e-07           3.639990e-07   \n",
       "97             50.510549      0.021486 -5.554476e-06           2.306093e-06   \n",
       "97              1.564323      0.022905 -1.488373e-07           6.202373e-08   \n",
       "97              1.805529      0.023518 -1.577137e-07           6.662980e-08   \n",
       "97             10.570773      0.032542  8.350697e-07           4.554567e-07   \n",
       "97              4.917787      0.040606  4.026315e-07           2.426256e-07   \n",
       "97             28.609709      0.043614 -5.943663e-06           2.339894e-06   \n",
       "97              3.945671      0.044446  3.611513e-07           2.649829e-07   \n",
       "\n",
       "    Coefficient_z-score                                     Annotation  \\\n",
       "97             3.122509                   GSE63648_7pcw_ac_me2_Hu_gain   \n",
       "97            -3.395446  promoters_human_macaque_prefontal_cortex_hg19   \n",
       "97            -2.433231          depleted_neanderthal_denisovan_vernot   \n",
       "97             2.194132                   GSE63648_7pcw_ac_me2_Hu_gain   \n",
       "97             2.705279                   GSE63648_7pcw_ac_me2_Hu_gain   \n",
       "97             1.981747                 GSE63648_12Opcw_ac_me2_Hu_gain   \n",
       "97             2.959777          depleted_neanderthal_denisovan_vernot   \n",
       "97             1.848058                   GSE63648_7pcw_ac_me2_Hu_gain   \n",
       "97             1.955670                   GSE63648_7pcw_ac_me2_Hu_gain   \n",
       "97             2.055468                   GSE63648_7pcw_ac_me2_Hu_gain   \n",
       "97            -2.903070                             nchaes_merged_hg19   \n",
       "97             1.748334                 GSE63648_8_5pcw_ac_me2_Hu_gain   \n",
       "97             1.987945                 GSE63648_8_5pcw_ac_me2_Hu_gain   \n",
       "97            -2.408609                             nchaes_merged_hg19   \n",
       "97            -2.399683          selsweeps_233putativeregions_peyregne   \n",
       "97            -2.367014          selsweeps_233putativeregions_peyregne   \n",
       "97             1.833478                 GSE63648_12Fpcw_ac_me2_Hu_gain   \n",
       "97             1.659477                   GSE63648_7pcw_ac_me2_Hu_gain   \n",
       "97            -2.540142                             nchaes_merged_hg19   \n",
       "97             1.362923                   GSE63648_7pcw_ac_me2_Hu_gain   \n",
       "\n",
       "                                 Phenotype  neglog10p  \n",
       "97                          shoulder_width   3.446298  \n",
       "97                      pelvic_inlet_width   3.040035  \n",
       "97                           pelvic_height   3.025892  \n",
       "97                           pelvic_height   2.851522  \n",
       "97                          subpubic_angle   2.823637  \n",
       "97                           pelvic_height   2.125509  \n",
       "97                       iliac_flare_angle   2.101231  \n",
       "97                       iliac_flare_angle   2.098520  \n",
       "97             shoulder_divide_inlet_width   2.041788  \n",
       "97  shoulder_area_divide_pelvic_inlet_area   1.986225  \n",
       "97                 head_divide_inlet_width   1.942891  \n",
       "97                           pelvic_height   1.870987  \n",
       "97                        arm_devide_torso   1.762577  \n",
       "97                      ear_left2ear_right   1.667853  \n",
       "97        head_divide_oblique_inlet_length   1.640062  \n",
       "97    shoulder_divide_oblique_inlet_length   1.628595  \n",
       "97                        arm_devide_torso   1.487560  \n",
       "97                     acetabular_diameter   1.391407  \n",
       "97                      pelvic_inlet_width   1.360376  \n",
       "97        trochanter_left2trochanter_right   1.352163  "
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_res_df.sort_values(by=['Enrichment_p']).head(20)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "sns.scatterplot(data=all_res_df, x=\"Annotation\", y=\"neglog10p\", hue=\"Phenotype\", palette=\"deep\")\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "ml",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.1.undefined"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
